
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Molecular insights and gene prioritization at key AD loci &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/7_AD_gene_prioritized/ZYX_case_example';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="prev" title="APP" href="APP_case_example.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../1_project_overview/index.html">Multi-omics QTL atlas of the aging human brain</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../2_single_context_cis/index.html">Fine-mapped QTL landscape across brain molecular phenotypes</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_1b_num_xQTL_loci_single_context.html">Figure 1b. Number of detectable and mappable xQTL loci in single−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_EOO.html">EOO results</a></li>

<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_S2_single_context_cs_distance.html">Frequency of Events in Distance Category</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_sLDSC_GWAS_specific.html">SLDSC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_sLDSC_meta_analysis.html">Visualization of SLDSC results</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../3_single_context_multigene_trans/index.html">Coordinated gene regulation and long-range genetic effects</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../4_multi_context/index.html">Cascading genetic effects across molecular contexts</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_2b_num_xQTL_loci_multi_context.html">Figure 2b. Number of detectable and mappable xQTL loci in multi−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_2c_multicontext_effect_sign.html">Figure 2c. Marginal vs multicontext effect sign heatmap</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../5_AD_xQTL_integration/index.html">Regulatory architecture of AD susceptibility loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5a_AD_GWAS_loci_harmonized.html">AD GWAS loci harmonization and refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5b_AD_GWAS_loci_properties.html">Refined AD GWAS loci properties</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5c_AD_loci_xQTL_annotated.html">AD loci “explained” by xQTL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5d_AD_loci_table_summary.html">Summary of FunGen-xQTL AD loci table</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5e_GSEA.html">Gene Set Enrichment Analysis for AD Risk Genes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5f_PPI.html">PPI Networks for Selected AD Risk Genes in xQTL Loci</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5g_cV2F.html">Consensus Variant-to-Function Score for AD</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5h_GWAS_enrichment.html">GWAS Enrichment</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../6_AD_xQTL_genes/index.html">Molecular dissection of prioritized AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization.html">Visualization of FunGen-xQTL AD Loci Table</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="index.html">Molecular insights and gene prioritization at key AD loci</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="APP_case_example.html">APP</a></li>




<li class="toctree-l2 current active"><a class="current reference internal" href="#">Molecular insights and gene prioritization at key AD loci</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/7_AD_gene_prioritized/ZYX_case_example.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/main_text/7_AD_gene_prioritized/ZYX_case_example.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Molecular insights and gene prioritization at key AD loci</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#zyx">ZYX</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#raw-data">Raw data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-loading">Data loading</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#section-1-gwas">Section 1: GWAS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#section-2-eqtl">Section 2: eQTL</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#section-4-panel-3-msnp-pip-fine-mzyxing-and-fine-mapping">Section 4: PANEL 3 – mSNP PIP fine-mZYXing and fine-mapping</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#section-5-panel-4-snatac-effect-estimates">Section 5: PANEL 4 – SNATAC effect estimates</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#combined-plot">Combined plot</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="molecular-insights-and-gene-prioritization-at-key-ad-loci">
<h1>Molecular insights and gene prioritization at key AD loci<a class="headerlink" href="#molecular-insights-and-gene-prioritization-at-key-ad-loci" title="Link to this heading">#</a></h1>
<p>Author: Hao Sun and Jenny Empawi</p>
<section id="zyx">
<h2>ZYX<a class="headerlink" href="#zyx" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">cowplot</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>── <span class=" -Color -Color-Bold">Attaching core tidyverse packages</span> ──────────────────────── tidyverse 2.0.0 ──
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">dplyr    </span> 1.1.4     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">readr    </span> 2.1.5
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">forcats  </span> 1.0.0     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">stringr  </span> 1.5.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">lubridate</span> 1.9.4     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tibble   </span> 3.2.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">purrr    </span> 1.1.0     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tidyr    </span> 1.3.1
── <span class=" -Color -Color-Bold">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">filter()</span> masks <span class=" -Color -Color-Blue">stats</span>::filter()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">lag()</span>    masks <span class=" -Color -Color-Blue">stats</span>::lag()
<span class=" -Color -Color-Cyan">ℹ</span> Use the conflicted package (<span class=" -Color -Color-Blue">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors

Attaching package: ‘cowplot’


The following object is masked from ‘package:lubridate’:

    stamp
</pre></div>
</div>
</div>
</div>
</section>
<section id="raw-data">
<h2>Raw data<a class="headerlink" href="#raw-data" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">ad</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_ZYX/RSS_QC_RAISS_imputed.chr7_143353277_145271128.univariate_susie_rss.rds&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="data-loading">
<h2>Data loading<a class="headerlink" href="#data-loading" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#effect_table_snatac =  read_delim(&quot;/data/analysis_result/snatac_fsusie/top_loci/ROSMAP_Kelis_snatac_fsusie.exported.toploci.bed.gz&quot;) # This is what should be loaded while at the moment I loaded the copy in my local folder first. </span>
<span class="n">effect_table_snatac</span><span class="w"> </span><span class="o">=</span><span class="w">  </span><span class="nf">read_delim</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_ZYX/ROSMAP_Kelis_snatac_fsusie.exported.toploci.bed.gz&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span><span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">554</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">21</span>
<span class=" -Color -Color-Cyan">──</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">────────────────────────────────────────────────────────</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (13): #chr, a1, a2, variant_ID, event_ID, region_ID, cs_id, cs_root, epi...
<span class=" -Color -Color-Green">dbl</span>  (8): start, end, TADB_start, TADB_end, maf, PIP, cs_coverage_0.95, grid...

<span class=" -Color -Color-Cyan">ℹ</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">effect_table_snatac_zyx</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">effect_table_snatac</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">region_ID</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;chr7_139293693_145380632&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">effect_table_snatac_zyx</span><span class="o">$</span><span class="n">cs_id</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">'ROSMAP_snATAC_Kelis_Mic:chr7_139293693_145380632:29'</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">magrittr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggrepel</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">cowplot</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">source</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_ZYX/get_gene_annotations.R&quot;</span><span class="p">)</span>
<span class="nf">source</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_ZYX/interpolate_effect_estimates.R&quot;</span><span class="p">)</span>
<span class="n">gene_file</span><span class="w"> </span><span class="o">&lt;-</span>
<span class="w">  </span><span class="nf">file.path</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_ZYX/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.ERCC.gtf&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="n">genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">get_gene_annotations</span><span class="p">(</span><span class="n">gene_file</span><span class="p">)</span>


<span class="n">eQTL_1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_ZYX/ROSMAP_DeJager_eQTL.ENSG00000159840.univariate_susie_twas_weights.rds&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="n">GWAS_1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_ZYX/RSS_QC_RAISS_imputed.chr7_143353277_145271128.univariate_susie_rss.rds&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>


<span class="n">key_marker</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">143414203</span>
<span class="n">pos0</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">143353277</span>
<span class="n">pos1</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">143500000</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Attaching package: ‘data.table’


The following objects are masked from ‘package:lubridate’:

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year


The following objects are masked from ‘package:dplyr’:

    between, first, last


The following object is masked from ‘package:purrr’:

    transpose



Attaching package: ‘magrittr’


The following object is masked from ‘package:purrr’:

    set_names


The following object is masked from ‘package:tidyr’:

    extract
</pre></div>
</div>
</div>
</div>
<section id="section-1-gwas">
<h3>Section 1: GWAS<a class="headerlink" href="#section-1-gwas" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">GWAS_1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="n">GWAS_1</span><span class="p">[[</span><span class="m">1</span><span class="p">]][[</span><span class="m">2</span><span class="p">]])</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="n">variant_id</span><span class="p">,</span><span class="n">pos</span><span class="p">,</span><span class="n">beta</span><span class="p">,</span><span class="n">se</span><span class="p">,</span><span class="n">z</span><span class="p">,</span><span class="n">pval</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pvalue</span><span class="w"> </span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                </span><span class="nf">mutate</span><span class="p">(</span><span class="n">CS</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">variant_id</span><span class="w"> </span><span class="o">%in%</span><span class="w">  </span>
<span class="p">((</span><span class="n">GWAS_1</span><span class="p">[[</span><span class="m">1</span><span class="p">]][[</span><span class="m">1</span><span class="p">]][[</span><span class="s">&quot;top_loci&quot;</span><span class="p">]])</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">cs_coverage_0.95</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_id</span><span class="p">))</span><span class="w">  </span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="nf">mutate</span><span class="p">(</span><span class="n">variant_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="n">variant_id</span><span class="w"> </span><span class="p">))</span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="w"> </span><span class="n">id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">CS</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">variant_id</span><span class="p">,</span><span class="kc">NA</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="section-2-eqtl">
<h3>Section 2: eQTL<a class="headerlink" href="#section-2-eqtl" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">eQTL_pdat_1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">tibble</span><span class="p">(</span>
<span class="n">variant_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eQTL_1</span><span class="p">[[</span><span class="m">1</span><span class="p">]][[</span><span class="s">&quot;Mic_DeJager_eQTL&quot;</span><span class="p">]][[</span><span class="s">&quot;variant_names&quot;</span><span class="p">]],</span>
<span class="n">beta</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eQTL_1</span><span class="p">[[</span><span class="m">1</span><span class="p">]][[</span><span class="s">&quot;Mic_DeJager_eQTL&quot;</span><span class="p">]][[</span><span class="s">&quot;sumstats&quot;</span><span class="p">]][[</span><span class="m">1</span><span class="p">]],</span>
<span class="n">se</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">eQTL_1</span><span class="p">[[</span><span class="m">1</span><span class="p">]][[</span><span class="s">&quot;Mic_DeJager_eQTL&quot;</span><span class="p">]][[</span><span class="s">&quot;sumstats&quot;</span><span class="p">]][[</span><span class="m">2</span><span class="p">]])</span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">z</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="n">beta</span><span class="o">/</span><span class="n">se</span><span class="p">,</span><span class="w"> </span><span class="n">CS</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">variant_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span>
<span class="w">                                                                   </span><span class="p">((</span><span class="n">eQTL_1</span><span class="p">[[</span><span class="m">1</span><span class="p">]][[</span><span class="s">&quot;Mic_DeJager_eQTL&quot;</span><span class="p">]][[</span><span class="s">&quot;top_loci&quot;</span><span class="p">]])</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">cs_coverage_0.95</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_id</span><span class="p">))</span><span class="w"> </span>
<span class="w">                                                                   </span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">separate</span><span class="p">(</span><span class="n">variant_id</span><span class="p">,</span><span class="n">into</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="s">&quot;pos&quot;</span><span class="p">),</span><span class="n">remove</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">pos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">pos</span><span class="p">),</span><span class="w"> </span><span class="n">id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">CS</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">variant_id</span><span class="p">,</span><span class="kc">NA</span><span class="p">)</span><span class="w"> </span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">pval</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">pnorm</span><span class="p">(</span><span class="o">-</span><span class="nf">abs</span><span class="p">(</span><span class="n">z</span><span class="p">)))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“Expected 2 pieces. Additional pieces discarded in 15803 rows [1, 2, 3, 4, 5, 6,
7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].”
</pre></div>
</div>
</div>
</div>
</section>
<section id="section-4-panel-3-msnp-pip-fine-mzyxing-and-fine-mapping">
<h3>Section 4: PANEL 3 – mSNP PIP fine-mZYXing and fine-mapping<a class="headerlink" href="#section-4-panel-3-msnp-pip-fine-mzyxing-and-fine-mapping" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">pdat3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<span class="w">  </span><span class="n">id</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="n">effect_table_snatac_zyx</span><span class="o">$</span><span class="n">variant_ID</span><span class="p">,</span>
<span class="w">  </span><span class="n">pos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">effect_table_snatac_zyx</span><span class="o">$</span><span class="n">end</span><span class="p">,</span>
<span class="w">  </span><span class="n">pip</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">effect_table_snatac_zyx</span><span class="o">$</span><span class="n">PIP</span><span class="p">,</span>
<span class="w">  </span><span class="n">cs</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="n">effect_table_snatac_zyx</span><span class="o">$</span><span class="n">cs_coverage_0.95</span><span class="p">,</span>
<span class="w">  </span><span class="n">stringsAsFactors</span><span class="o">=</span><span class="kc">FALSE</span>
<span class="p">)</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">pdat3</span><span class="p">)</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pdat3</span><span class="o">$</span><span class="n">id</span>

<span class="n">pdat4</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">effect_table_snatac_zyx</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="n">pos</span><span class="w">	</span><span class="o">=</span><span class="w"> </span><span class="n">epi_mark_positions</span><span class="p">,</span><span class="n">effect</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">epi_mark_effects</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="nf">separate_rows</span><span class="p">(</span><span class="n">pos</span><span class="p">,</span><span class="n">effect</span><span class="p">,</span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">mutate_all</span><span class="p">(</span><span class="n">as.numeric</span><span class="p">)</span>

<span class="n">pdat5</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">genes</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">chromosome</span><span class="o">==</span><span class="s">&quot;chr7&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">end</span><span class="o">&gt;</span><span class="n">pos0</span><span class="p">,</span><span class="w"> </span><span class="n">start</span><span class="o">&lt;</span><span class="n">pos1</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="n">start</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">start</span><span class="p">,</span>
<span class="w">    </span><span class="n">end</span><span class="w">   </span><span class="o">=</span><span class="w"> </span><span class="n">end</span><span class="p">,</span>
<span class="w">    </span><span class="n">tss</span><span class="w">   </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">strand</span><span class="o">==</span><span class="s">&quot;+&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="n">end</span><span class="p">),</span>
<span class="w">    </span><span class="n">y</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="nf">seq</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="n">length.out</span><span class="o">=</span><span class="nf">n</span><span class="p">())</span>
<span class="w">  </span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">GWAS_1</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pos</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">pval</span><span class="p">))</span><span class="o">*</span><span class="nf">sign</span><span class="p">(</span><span class="n">z</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="n">CS</span><span class="p">,</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">id</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="o">=</span><span class="n">key_marker</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="o">=</span><span class="s">&quot;dotted&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;darkgray&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_text_repel</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;dimgray&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">segment.color</span><span class="o">=</span><span class="s">&quot;dimgray&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">min.segment.length</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">max.overlaps</span><span class="o">=</span><span class="kc">Inf</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="s">&quot;dodgerblue&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_x_continuous</span><span class="p">(</span><span class="n">limits</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">pos0</span><span class="p">,</span><span class="n">pos1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="s">&quot;signed -log10(pval) for AD&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">theme_cowplot</span><span class="p">(</span><span class="n">font_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">9</span><span class="p">)</span><span class="w"> </span>

<span class="n">p1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“Removed 5570 rows containing missing values or values outside the scale range
(`geom_point()`).”
Warning message:
“Removed 6143 rows containing missing values or values outside the scale range
(`geom_text_repel()`).”
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/ba8bd1cb6c49b7485632bd906264010ae77b7c93c5e6a144ddb430f22c04ad23.png"><img alt="../../_images/ba8bd1cb6c49b7485632bd906264010ae77b7c93c5e6a144ddb430f22c04ad23.png" src="../../_images/ba8bd1cb6c49b7485632bd906264010ae77b7c93c5e6a144ddb430f22c04ad23.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">eQTL_pdat_1</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pos</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">pval</span><span class="p">))</span><span class="o">*</span><span class="nf">sign</span><span class="p">(</span><span class="n">z</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="n">CS</span><span class="p">,</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">id</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="o">=</span><span class="n">key_marker</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="o">=</span><span class="s">&quot;dotted&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;darkgray&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_text_repel</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;dimgray&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">segment.color</span><span class="o">=</span><span class="s">&quot;dimgray&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">min.segment.length</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">max.overlaps</span><span class="o">=</span><span class="kc">Inf</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="s">&quot;dodgerblue&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_x_continuous</span><span class="p">(</span><span class="n">limits</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">pos0</span><span class="p">,</span><span class="n">pos1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="s">&quot;signed -log10(pval) for eQTL&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">theme_cowplot</span><span class="p">(</span><span class="n">font_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">9</span><span class="p">)</span><span class="w"> </span>

<span class="n">p2</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“Removed 15228 rows containing missing values or values outside the scale range
(`geom_point()`).”
Warning message:
“Removed 15800 rows containing missing values or values outside the scale range
(`geom_text_repel()`).”
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/1ff09a1b8c143e87f6efc9725c1c65f9b69017a82652d2571b36ba1acff85b34.png"><img alt="../../_images/1ff09a1b8c143e87f6efc9725c1c65f9b69017a82652d2571b36ba1acff85b34.png" src="../../_images/1ff09a1b8c143e87f6efc9725c1c65f9b69017a82652d2571b36ba1acff85b34.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">pdat3</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">pos</span><span class="p">,</span><span class="w"> </span><span class="n">pip</span><span class="p">,</span><span class="w">  </span><span class="n">label</span><span class="o">=</span><span class="n">id</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">0.75</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="o">=</span><span class="n">key_marker</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="o">=</span><span class="s">&quot;dotted&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;darkgray&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_text_repel</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;dimgray&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">segment.color</span><span class="o">=</span><span class="s">&quot;dimgray&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">min.segment.length</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">max.overlaps</span><span class="o">=</span><span class="kc">Inf</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_x_continuous</span><span class="p">(</span><span class="n">limits</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">pos0</span><span class="p">,</span><span class="n">pos1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="s">&quot;ca-SNP PIP&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_cowplot</span><span class="p">(</span><span class="n">font_size</span><span class="o">=</span><span class="m">9</span><span class="p">)</span>

<span class="n">p3</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/0e25042d2c3e9990f83f056367a69f7bbf8f2fff997d521ee413a72e76214a9c.png"><img alt="../../_images/0e25042d2c3e9990f83f056367a69f7bbf8f2fff997d521ee413a72e76214a9c.png" src="../../_images/0e25042d2c3e9990f83f056367a69f7bbf8f2fff997d521ee413a72e76214a9c.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="section-5-panel-4-snatac-effect-estimates">
<h3>Section 5: PANEL 4 – SNATAC effect estimates<a class="headerlink" href="#section-5-panel-4-snatac-effect-estimates" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p4</span><span class="w"> </span><span class="o">=</span><span class="w">  </span><span class="nf">ggplot</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="o">=</span><span class="s">&quot;dotted&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="nf">subset</span><span class="p">(</span><span class="n">pdat4</span><span class="p">),</span>
<span class="w">             </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">pos</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">effect</span><span class="p">),</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="o">=</span><span class="n">key_marker</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="o">=</span><span class="s">&quot;dotted&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;darkgray&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="s">&quot;effect of A allele&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_x_continuous</span><span class="p">(</span><span class="n">limits</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">pos0</span><span class="p">,</span><span class="n">pos1</span><span class="p">))</span><span class="w">  </span><span class="o">+</span><span class="nf">ylim</span><span class="p">(</span><span class="m">0.02</span><span class="p">,</span><span class="m">0.09</span><span class="p">)</span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_cowplot</span><span class="p">(</span><span class="n">font_size</span><span class="o">=</span><span class="m">9</span><span class="p">)</span>

<span class="n">p4</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_hline()`).”
Warning message:
“Removed 152 rows containing missing values or values outside the scale range
(`geom_point()`).”
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/a6f354796b15919c570dc7b652dcc5afd9f8d6de03321cd21bd839978dd282c9.png"><img alt="../../_images/a6f354796b15919c570dc7b652dcc5afd9f8d6de03321cd21bd839978dd282c9.png" src="../../_images/a6f354796b15919c570dc7b652dcc5afd9f8d6de03321cd21bd839978dd282c9.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p5</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">pdat5</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="n">xend</span><span class="o">=</span><span class="n">end</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="n">yend</span><span class="o">=</span><span class="n">y</span><span class="p">,</span><span class="w"> </span><span class="n">label</span><span class="o">=</span><span class="n">gene_name</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_segment</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="s">&quot;dodgerblue&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">0.5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">tss</span><span class="p">),</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;dodgerblue&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">shape</span><span class="o">=</span><span class="m">18</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">1.5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_text</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">fontface</span><span class="o">=</span><span class="s">&quot;italic&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">hjust</span><span class="o">=</span><span class="s">&quot;right&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">nudge_x</span><span class="o">=</span><span class="m">-0.003</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="o">=</span><span class="n">key_marker</span><span class="p">,</span><span class="w"> </span><span class="n">linetype</span><span class="o">=</span><span class="s">&quot;dotted&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;darkgray&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="c1">#scale_x_continuous(limits=c(pos0,pos1)/1e6, breaks=seq(45,48,0.05)) +</span>
<span class="w">  </span><span class="nf">scale_x_continuous</span><span class="p">(</span><span class="n">limits</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">pos0</span><span class="p">,</span><span class="n">pos1</span><span class="p">))</span><span class="o">+</span><span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s">&quot;base-pair position on chromosome 7&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_cowplot</span><span class="p">(</span><span class="n">font_size</span><span class="o">=</span><span class="m">9</span><span class="p">)</span>

<span class="n">p5</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
<span class=" -Color -Color-Cyan">ℹ</span> Please use `linewidth` instead.”
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/444e40243fc1b2d5e7793fb601040a18f6dae0d465833448a5a87533fcc0a0b8.png"><img alt="../../_images/444e40243fc1b2d5e7793fb601040a18f6dae0d465833448a5a87533fcc0a0b8.png" src="../../_images/444e40243fc1b2d5e7793fb601040a18f6dae0d465833448a5a87533fcc0a0b8.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">pdat5</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 5 × 11</caption>
<thead>
	<tr><th scope=col>chromosome</th><th scope=col>source</th><th scope=col>feature</th><th scope=col>start</th><th scope=col>end</th><th scope=col>strand</th><th scope=col>ensembl</th><th scope=col>gene_type</th><th scope=col>gene_name</th><th scope=col>tss</th><th scope=col>y</th></tr>
	<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr7</td><td>ensembl_havana</td><td>transcript</td><td>143353400</td><td>143362725</td><td>-</td><td>ENSG00000159784</td><td>protein_coding</td><td>FAM131B</td><td>143362725</td><td>0.00</td></tr>
	<tr><td>chr7</td><td>ensembl_havana</td><td>transcript</td><td>143381295</td><td>143391111</td><td>+</td><td>ENSG00000159840</td><td>protein_coding</td><td>ZYX    </td><td>143381295</td><td>0.25</td></tr>
	<tr><td>chr7</td><td>ensembl_havana</td><td>transcript</td><td>143390289</td><td>143408856</td><td>-</td><td>ENSG00000146904</td><td>protein_coding</td><td>EPHA1  </td><td>143408856</td><td>0.50</td></tr>
	<tr><td>chr7</td><td>ensembl_havana</td><td>transcript</td><td>143443453</td><td>143444409</td><td>+</td><td>ENSG00000185899</td><td>protein_coding</td><td>TAS2R60</td><td>143443453</td><td>0.75</td></tr>
	<tr><td>chr7</td><td>ensembl_havana</td><td>transcript</td><td>143477873</td><td>143478796</td><td>+</td><td>ENSG00000221855</td><td>protein_coding</td><td>TAS2R41</td><td>143477873</td><td>1.00</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="combined-plot">
<h3>Combined plot<a class="headerlink" href="#combined-plot" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">)</span>
<span class="n">fig</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">plot_grid</span><span class="p">(</span><span class="n">p1</span><span class="o">+</span><span class="nf">theme</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="m">10</span><span class="w"> </span><span class="p">)),</span><span class="w"> </span>
<span class="w">                 </span><span class="n">p2</span><span class="o">+</span><span class="nf">theme</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="m">10</span><span class="w"> </span><span class="p">)),</span>
<span class="w">                 </span><span class="n">p3</span><span class="o">+</span><span class="nf">theme</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="m">20</span><span class="w"> </span><span class="p">)),</span>
<span class="w">                 </span><span class="n">p4</span><span class="o">+</span><span class="nf">theme</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="m">20</span><span class="w"> </span><span class="p">)),</span>
<span class="w">                 </span><span class="n">p5</span><span class="o">+</span><span class="nf">theme</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="m">20</span><span class="w"> </span><span class="p">))</span>
<span class="w">                 </span><span class="c1">#p6+theme(text = element_text(size =10 ))</span>
<span class="w">                 </span><span class="c1">#,p7+theme(text = element_text(size =10 ))</span>
<span class="w">                 </span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="o">=</span><span class="m">5</span>
<span class="w">                 </span><span class="p">,</span><span class="w"> </span><span class="n">align</span><span class="o">=</span><span class="s">&quot;v&quot;</span><span class="p">)</span>
<span class="n">fig</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“Removed 5570 rows containing missing values or values outside the scale range
(`geom_point()`).”
Warning message:
“Removed 6143 rows containing missing values or values outside the scale range
(`geom_text_repel()`).”
Warning message:
“Removed 15228 rows containing missing values or values outside the scale range
(`geom_point()`).”
Warning message:
“Removed 15800 rows containing missing values or values outside the scale range
(`geom_text_repel()`).”
Warning message:
“Removed 1 row containing missing values or values outside the scale range
(`geom_hline()`).”
Warning message:
“Removed 152 rows containing missing values or values outside the scale range
(`geom_point()`).”
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/a13d2022fc40e37686925d86355383afd293df403a170ed6771b48c2c43cac6d.png"><img alt="../../_images/a13d2022fc40e37686925d86355383afd293df403a170ed6771b48c2c43cac6d.png" src="../../_images/a13d2022fc40e37686925d86355383afd293df403a170ed6771b48c2c43cac6d.png" style="width: 600px; height: 600px;" />
</a>
</div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./main_text/7_AD_gene_prioritized"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="APP_case_example.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">APP</p>
      </div>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#zyx">ZYX</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#raw-data">Raw data</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#data-loading">Data loading</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#section-1-gwas">Section 1: GWAS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#section-2-eqtl">Section 2: eQTL</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#section-4-panel-3-msnp-pip-fine-mzyxing-and-fine-mapping">Section 4: PANEL 3 – mSNP PIP fine-mZYXing and fine-mapping</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#section-5-panel-4-snatac-effect-estimates">Section 5: PANEL 4 – SNATAC effect estimates</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#combined-plot">Combined plot</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>