
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Visualization of FunGen-xQTL AD Loci Table &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Molecular insights and gene prioritization at key AD loci" href="../7_AD_gene_prioritized/index.html" />
    <link rel="prev" title="Molecular dissection of prioritized AD loci" href="index.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../1_project_overview/index.html">Multi-omics QTL atlas of the aging human brain</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../2_single_context_cis/index.html">Fine-mapped QTL landscape across brain molecular phenotypes</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_1b_num_xQTL_loci_single_context.html">Figure 1b. Number of detectable and mappable xQTL loci in single−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_EOO.html">EOO results</a></li>

<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_S2_single_context_cs_distance.html">Frequency of Events in Distance Category</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_sLDSC_GWAS_specific.html">SLDSC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_sLDSC_meta_analysis.html">Visualization of SLDSC results</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../3_single_context_multigene_trans/index.html">Coordinated gene regulation and long-range genetic effects</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../4_multi_context/index.html">Cascading genetic effects across molecular contexts</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_2b_num_xQTL_loci_multi_context.html">Number of detectable and mappable xQTL loci in ROSMAP multi−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_2c_multicontext_effect_sign.html">Marginal vs multicontext effect sign heatmap</a></li>

<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_watershed_posterior_plot.html">Watershed Posterior</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../5_AD_xQTL_integration/index.html">Regulatory architecture of AD susceptibility loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5a_AD_GWAS_loci_harmonized.html">AD GWAS loci harmonization and refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5b_AD_GWAS_loci_properties.html">Refined AD GWAS loci properties</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5c_AD_loci_xQTL_annotated.html">AD loci “explained” by xQTL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5d_AD_loci_table_summary.html">Summary of FunGen-xQTL AD loci table</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5e_GSEA.html">Gene Set Enrichment Analysis for AD Risk Genes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5f_PPI.html">PPI networks for selected AD risk genes in xQTL loci</a></li>

<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5g_cV2F.html">Consensus Variant-to-Function Score for AD</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_5h_GWAS_enrichment.html">GWAS Enrichment</a></li>
<li class="toctree-l2"><a class="reference internal" href="../5_AD_xQTL_integration/Figure_S5_prediction.html">Prediction</a></li>









</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="index.html">Molecular dissection of prioritized AD loci</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Visualization of FunGen-xQTL AD Loci Table</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../7_AD_gene_prioritized/index.html">Molecular insights and gene prioritization at key AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../7_AD_gene_prioritized/APP_case_example.html">APP</a></li>




<li class="toctree-l2"><a class="reference internal" href="../7_AD_gene_prioritized/ZYX_case_example.html">Molecular insights and gene prioritization at key AD loci</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/main_text/6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Visualization of FunGen-xQTL AD Loci Table</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#motivation">Motivation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#implementation">Implementation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-libraries">Load libraries</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-result">Load result</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#visualization">Visualization</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section id="visualization-of-fungen-xqtl-ad-loci-table">
<h1>Visualization of FunGen-xQTL AD Loci Table<a class="headerlink" href="#visualization-of-fungen-xqtl-ad-loci-table" title="Link to this heading">#</a></h1>
<p>Author: Alexandre Pelletier and Jenny Empawi</p>
<section id="motivation">
<h2>Motivation<a class="headerlink" href="#motivation" title="Link to this heading">#</a></h2>
<p>Create an information visualization summarizing key information from the comprehensive AD loci table.</p>
</section>
<section id="implementation">
<h2>Implementation<a class="headerlink" href="#implementation" title="Link to this heading">#</a></h2>
<p>Generate one figure to highlight the most critical insights from the AD loci table. Possible information to include: effect size direction for AD risk variants wrt xQTL</p>
</section>
<section id="load-libraries">
<h2>Load libraries<a class="headerlink" href="#load-libraries" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">ggtext</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">showtext</span><span class="p">)</span>
<span class="nf">font_add_google</span><span class="p">(</span><span class="s">&quot;Noto Sans&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Noto Sans&quot;</span><span class="p">)</span>
<span class="nf">showtext_auto</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Loading required package: sysfonts

Loading required package: showtextdb
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_6&#39;</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="nf">source</span><span class="p">(</span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_6/gene_prio_utils.R&#39;</span><span class="p">)</span><span class="w"> </span><span class="c1">#FIXME</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="load-result">
<h2>Load result<a class="headerlink" href="#load-result" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#update table figure</span>
<span class="n">res_adx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="nf">fp</span><span class="p">(</span><span class="n">out</span><span class="p">,</span><span class="s">&#39;res_allanalysis_ADloci_overlap.csv&#39;</span><span class="p">))</span><span class="w"> </span><span class="c1">#FIXME</span>

<span class="c1">#add the confidence level</span>
<span class="c1">#res_adxub&lt;-fread(fp(out,&#39;res_AD_variants_xQTL.csv.gz&#39;))  #FIXME</span>
<span class="n">res_adx</span><span class="p">[,</span><span class="n">top_confidence</span><span class="o">:=</span><span class="nf">str_extract</span><span class="p">(</span><span class="n">xQTL_effects</span><span class="p">,</span><span class="s">&#39;C[0-9]&#39;</span><span class="p">)]</span>


<span class="c1">#xQTL evidence summary dot plot at locus level</span>
<span class="c1">#for each QTL evidence type put in 3 cat : absent, present, high, </span>
<span class="c1">#for fp: high if cs95, coloc : high if npc &gt; 0.95, TWAS: MR</span>
<span class="nf">table</span><span class="p">(</span><span class="n">res_adx</span><span class="o">$</span><span class="n">Method</span><span class="p">)</span>
<span class="nf">table</span><span class="p">(</span><span class="n">res_adx</span><span class="o">$</span><span class="n">context_short</span><span class="p">)</span>
<span class="nf">table</span><span class="p">(</span><span class="n">res_adx</span><span class="o">$</span><span class="n">context</span><span class="p">)</span>

<span class="c1">#group contexts better: </span>
<span class="n">res_adx</span><span class="p">[,</span><span class="n">locus_gene</span><span class="o">:=</span><span class="nf">paste</span><span class="p">(</span><span class="n">locus_index</span><span class="p">,</span><span class="nf">ifelse</span><span class="p">(</span><span class="n">gene_name</span><span class="o">==</span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="s">&#39;?&#39;</span><span class="p">,</span><span class="n">gene_name</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&#39;_&#39;</span><span class="p">)]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>       AD_GWAS_finemapping     AD_meta_colocalization 
                      6552                       7825 
    AD_xQTL_colocalization           APOE interaction 
                    142480                       2661 
                     Coloc                         QR 
                      5657                     223262 
                   TWAS/MR                      cTWAS 
                      2670                        117 
        fSuSiE_finemapping           msex interaction 
                     37369                         12 
 multi_context_finemapping     multi_gene_finemapping 
                    102549                      69355 
single_context_finemapping                    sn_sQTL 
                   1479052                     227297 
         trans_finemapping 
                    173677 
</pre></div>
</div>
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>  Ast caQTL    Ast eQTL    Ast sQTL           B    Exc eQTL    Exc sQTL 
          7      120179       12744       10551      156572      162545 
   Inh eQTL    Inh sQTL           J           K   Mic caQTL    Mic eQTL 
     157924       24492         672        1118          12       62232 
   Mic sQTL    OPC eQTL    OPC sQTL    Oli eQTL    Oli sQTL           W 
          2       95833        8720      146037       18794       40897 
  bMac eQTL   bMic eQTL  bMono eQTL   bulk eQTL  bulk gpQTL  bulk haQTL 
      64755       46488       33658      863034       16770         657 
  bulk mQTL bulk p-sQTL   bulk pQTL   bulk sQTL bulk u-sQTL 
      42350      175552       25353      166250       26337 
</pre></div>
</div>
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>                        AC_DeJager_eQTL                      AD_Bellenguez_2022 
                                 130242                                    8517 
                AD_Bellenguez_EADB_2022                 AD_Bellenguez_EADI_2022 
                                   2017                                      17 
                         AD_Jansen_2021                   AD_Kunkle_Stage1_2019 
                                    672                                    1118 
      AD_Wightman_Excluding23andMe_2021 AD_Wightman_ExcludingUKBand23andME_2021 
                                  22223                                   15493 
                  AD_Wightman_Full_2021                      Ast_10_Kellis_eQTL 
                                   3181                                   20948 
                       Ast_DeJager_eQTL                         Ast_Kellis_eQTL 
                                  64726                                   19054 
                          Ast_Mega_sQTL                           Ast_mega_eQTL 
                                  12744                                   15451 
                        BM_10_MSBB_eQTL                         BM_22_MSBB_eQTL 
                                  62916                                   58807 
                        BM_36_MSBB_eQTL                         BM_44_MSBB_eQTL 
                                  62169                                   91216 
                     DLPFC_Bennett_pQTL                      DLPFC_DeJager_eQTL 
                                   5633                                  110759 
             DLPFC_Klein_gpQTL_adjusted            DLPFC_Klein_gpQTL_unadjusted 
                                   7812                                    8958 
                       Exc_DeJager_eQTL                         Exc_Kellis_eQTL 
                                  58927                                   44480 
                          Exc_Mega_sQTL                           Exc_mega_eQTL 
                                 162545                                   53165 
                       Inh_DeJager_eQTL                         Inh_Kellis_eQTL 
                                  50211                                   50502 
                          Inh_Mega_sQTL                           Inh_mega_eQTL 
                                  24492                                   57211 
                        KnightADRC_mQTL                       Knight_eQTL_brain 
                                   4211                                   69523 
                      Knight_pQTL_brain                          MSBB_BM36_pQTL 
                                   7653                                   12067 
                              MSBB_mQTL                  Metabrain_Basalganglia 
                                   9005                                   31760 
                   Metabrain_Cerebellum                        Metabrain_Cortex 
                                  29488                                   10204 
                  Metabrain_Hippocampus                    Metabrain_Spinalcord 
                                  12784                                   51744 
                          MiGA_GFM_eQTL                           MiGA_GTS_eQTL 
                                   9494                                   18514 
                          MiGA_SVZ_eQTL                           MiGA_THA_eQTL 
                                   8056                                   10424 
                     Mic_12_Kellis_eQTL                      Mic_13_Kellis_eQTL 
                                     56                                      32 
                       Mic_DeJager_eQTL                         Mic_Kellis_eQTL 
                                  22731                                   32963 
                          Mic_Mega_sQTL                           Mic_mega_eQTL 
                                      2                                    6450 
                       OPC_DeJager_eQTL                         OPC_Kellis_eQTL 
                                  39750                                   25096 
                          OPC_Mega_sQTL                           OPC_mega_eQTL 
                                   8720                                   30987 
                       Oli_DeJager_eQTL                         Oli_Kellis_eQTL 
                                  59696                                   46089 
                          Oli_Mega_sQTL                           Oli_mega_eQTL 
                                  18794                                   40252 
                       PCC_DeJager_eQTL                        ROSMAP_AC_p_sQTL 
                                 141422                                   67072 
                         ROSMAP_AC_sQTL                        ROSMAP_AC_u_sQTL 
                                  65497                                    2931 
                     ROSMAP_DLPFC_haQTL                       ROSMAP_DLPFC_mQTL 
                                    657                                   29134 
                    ROSMAP_DLPFC_p_sQTL                       ROSMAP_DLPFC_sQTL 
                                  42174                                   49093 
                    ROSMAP_DLPFC_u_sQTL                       ROSMAP_PCC_p_sQTL 
                                  11520                                   66306 
                        ROSMAP_PCC_sQTL                       ROSMAP_PCC_u_sQTL 
                                  51660                                   11886 
                ROSMAP_snATAC_Kelis_Ast                 ROSMAP_snATAC_Kelis_Mic 
                                      7                                      12 
                       STARNET_eQTL_Mac                    monocyte_ROSMAP_eQTL 
                                  64755                                   33658 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#Genome wide signif hits</span>
<span class="n">res_adx</span><span class="p">[,</span><span class="n">genomewide_sig_loc</span><span class="o">:=</span><span class="nf">any</span><span class="p">(</span><span class="n">min_pval</span><span class="o">&lt;</span><span class="m">5e-8</span><span class="p">),</span><span class="n">by</span><span class="o">=</span><span class="s">&#39;locus_gene&#39;</span><span class="p">]</span>

<span class="c1">#get at locus level</span>
<span class="n">res_adxloc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">res_adx</span><span class="p">[</span><span class="o">!</span><span class="n">context_short</span><span class="o">%in%</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;B&#39;</span><span class="p">,</span><span class="s">&#39;J&#39;</span><span class="p">,</span><span class="s">&#39;K&#39;</span><span class="p">,</span><span class="s">&#39;W&#39;</span><span class="p">)][</span><span class="nf">order</span><span class="p">(</span><span class="n">locus_gene</span><span class="p">,</span><span class="n">confidence_lvl</span><span class="p">,</span><span class="n">cV2F_rank</span><span class="p">)],</span><span class="n">by</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;locus_gene&#39;</span><span class="p">,</span><span class="s">&#39;context&#39;</span><span class="p">))</span>
<span class="n">res_adxlocf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">res_adxloc</span><span class="p">[(</span><span class="n">genomewide_sig_loc</span><span class="p">)]</span>
<span class="n">res_adxlocf</span><span class="p">[</span><span class="n">gene_name</span><span class="o">!=</span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="n">n.study.gene.locus</span><span class="o">:=</span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">context</span><span class="p">)),</span><span class="n">by</span><span class="o">=</span><span class="nf">.</span><span class="p">(</span><span class="n">gene_name</span><span class="p">,</span><span class="n">locus_index</span><span class="p">)]</span>
<span class="n">res_adxlocf</span><span class="p">[</span><span class="n">gene_name</span><span class="o">!=</span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="n">top_gene</span><span class="o">:=</span><span class="n">gene_name</span><span class="o">==</span><span class="n">gene_name</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="n">confidence_lvl</span><span class="p">,</span><span class="o">-</span><span class="n">n.study.gene.locus</span><span class="p">,</span><span class="n">cV2F_rank</span><span class="p">)][</span><span class="m">1</span><span class="p">],</span><span class="n">by</span><span class="o">=</span><span class="nf">.</span><span class="p">(</span><span class="n">locus_index</span><span class="p">)]</span>

<span class="n">res_adxlocfge</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">res_adxlocf</span><span class="p">[</span><span class="n">context_short</span><span class="o">!=</span><span class="s">&#39;&#39;</span><span class="p">][</span><span class="nf">order</span><span class="p">(</span><span class="n">locus_index</span><span class="p">,</span><span class="o">-</span><span class="n">cV2F_rank</span><span class="p">)][(</span><span class="n">top_gene</span><span class="p">)]</span>
<span class="nf">unique</span><span class="p">(</span><span class="n">res_adxlocfge</span><span class="o">$</span><span class="n">locus_index</span><span class="p">)</span><span class="o">|&gt;</span><span class="nf">length</span><span class="p">()</span><span class="w"> </span><span class="c1">#95</span>
<span class="nf">unique</span><span class="p">(</span><span class="n">res_adxlocfge</span><span class="o">$</span><span class="n">gene_name</span><span class="p">)</span><span class="o">|&gt;</span><span class="nf">length</span><span class="p">()</span><span class="c1">#83</span>

<span class="n">res_adxlocfge</span><span class="p">[,</span><span class="n">chr</span><span class="o">:=</span><span class="nf">seqid</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">[</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)][</span><span class="m">1</span><span class="p">],</span><span class="n">only_num</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">),</span><span class="n">by</span><span class="o">=</span><span class="nf">.</span><span class="p">(</span><span class="n">locus_index</span><span class="p">)]</span>
<span class="n">res_adxlocfge</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">chr</span><span class="p">),</span><span class="n">chr</span><span class="o">:=</span><span class="n">`#chr`</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">95</div><div class="output text_html">83</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># group some contexts</span>
<span class="n">res_adxlocfge</span><span class="p">[,</span><span class="n">context_group</span><span class="o">:=</span><span class="nf">ifelse</span><span class="p">(</span><span class="n">context_short</span><span class="o">%in%</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;bMono eQTL&#39;</span><span class="p">,</span><span class="s">&#39;bMac eQTL&#39;</span><span class="p">,</span><span class="s">&#39;bMic eQTL&#39;</span><span class="p">),</span><span class="s">&#39;Immune eQTL&#39;</span><span class="p">,</span>
<span class="w">                                     </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">context_short</span><span class="o">%in%</span><span class="nf">c</span><span class="p">(</span><span class="w"> </span><span class="s">&#39;bulk p-sQTL&#39;</span><span class="p">,</span><span class="s">&#39;bulk u-sQTL&#39;</span><span class="p">,</span><span class="s">&#39;bulk a-sQTL&#39;</span><span class="p">),</span><span class="s">&#39;bulk sQTL&#39;</span><span class="p">,</span>
<span class="w">                                     </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">context_short</span><span class="o">%in%</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;bulk pQTL&#39;</span><span class="p">,</span><span class="s">&#39;bulk gpQTL&#39;</span><span class="p">),</span><span class="s">&#39;bulk (g)pQTL&#39;</span><span class="p">,</span>
<span class="w">                                            </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">context_short</span><span class="o">%in%</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;bulk haQTL&#39;</span><span class="p">,</span><span class="s">&#39;bulk mQTL&#39;</span><span class="p">),</span><span class="s">&#39;bulk epiQTL&#39;</span><span class="p">,</span><span class="nf">as.character</span><span class="p">(</span><span class="n">context_short</span><span class="p">)))))]</span>


<span class="nf">unique</span><span class="p">(</span><span class="n">res_adxlocfge</span><span class="o">$</span><span class="n">context_group</span><span class="p">)</span>
<span class="n">contexts_order</span><span class="o">&lt;-</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;Exc eQTL&#39;</span><span class="p">,</span><span class="s">&#39;Exc sQTL&#39;</span><span class="p">,</span>
<span class="w">                  </span><span class="s">&#39;Inh eQTL&#39;</span><span class="p">,</span><span class="s">&#39;Inh sQTL&#39;</span><span class="p">,</span>
<span class="w">                  </span><span class="s">&#39;Oli eQTL&#39;</span><span class="p">,</span><span class="s">&#39;Oli sQTL&#39;</span><span class="p">,</span>
<span class="w">                  </span><span class="s">&#39;OPC eQTL&#39;</span><span class="p">,</span>
<span class="w">                  </span><span class="s">&#39;Ast eQTL&#39;</span><span class="p">,</span><span class="s">&#39;Ast caQTL&#39;</span><span class="p">,</span>
<span class="w">                  </span><span class="s">&#39;Mic eQTL&#39;</span><span class="p">,</span><span class="s">&#39;Mic caQTL&#39;</span><span class="p">,</span>
<span class="w">                  </span><span class="s">&#39;Immune eQTL&#39;</span><span class="p">,</span>
<span class="w">                  </span><span class="s">&#39;bulk eQTL&#39;</span><span class="p">,</span>
<span class="w">                  </span><span class="s">&#39;bulk sQTL&#39;</span><span class="p">,</span>
<span class="w">                  </span><span class="s">&#39;bulk (g)pQTL&#39;</span><span class="p">,</span>
<span class="w">                  </span><span class="s">&#39;bulk epiQTL&#39;</span><span class="p">)</span>

<span class="n">res_adxlocfge</span><span class="p">[,</span><span class="n">context_group</span><span class="o">:=</span><span class="nf">factor</span><span class="p">(</span><span class="n">context_group</span><span class="p">,</span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">contexts_order</span><span class="p">)]</span>

<span class="nf">unique</span><span class="p">(</span><span class="n">res_adxlocfge</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="n">confidence_lvl</span><span class="p">)],</span><span class="n">by</span><span class="o">=</span><span class="s">&#39;gene_name&#39;</span><span class="p">)</span><span class="o">$</span><span class="n">confidence_lvl</span><span class="o">|&gt;</span><span class="nf">table</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'bulk eQTL'</li><li>'Oli eQTL'</li><li>'bulk epiQTL'</li><li>'Immune eQTL'</li><li>'Mic eQTL'</li><li>'bulk (g)pQTL'</li><li>'Ast eQTL'</li><li>'Exc eQTL'</li><li>'Inh eQTL'</li><li>'OPC eQTL'</li><li>'bulk sQTL'</li><li>'Exc sQTL'</li><li>'Inh sQTL'</li><li>'Mic caQTL'</li><li>'Ast sQTL'</li><li>'Oli sQTL'</li><li>'Ast caQTL'</li></ol>
</div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>CL1 CL2 CL3 CL4 CL5 CL6 
 25   6  13  10  19  10 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># summarize using confidence score  and nstudy</span>
<span class="n">res_adxlocfge</span><span class="p">[,</span><span class="n">n_study_group</span><span class="o">:=</span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">context</span><span class="p">)),</span><span class="n">by</span><span class="o">=</span><span class="nf">.</span><span class="p">(</span><span class="n">gene_name</span><span class="p">,</span><span class="n">context_group</span><span class="p">)]</span>
<span class="n">res_adxlocfge</span><span class="p">[,</span><span class="n">confidence_lvl_group</span><span class="o">:=</span><span class="nf">sort</span><span class="p">(</span><span class="n">confidence_lvl</span><span class="p">)[</span><span class="m">1</span><span class="p">],</span><span class="n">by</span><span class="o">=</span><span class="nf">.</span><span class="p">(</span><span class="n">gene_name</span><span class="p">,</span><span class="n">context_group</span><span class="p">)]</span>
<span class="nf">table</span><span class="p">(</span><span class="n">res_adxlocfge</span><span class="o">$</span><span class="n">confidence_lvl_group</span><span class="p">)</span>
<span class="n">res_adxlocfge</span><span class="p">[,</span><span class="n">n_loci_gene</span><span class="o">:=</span><span class="nf">unique</span><span class="p">(</span><span class="n">locus_index</span><span class="p">)</span><span class="o">|&gt;</span><span class="nf">length</span><span class="p">(),</span><span class="n">by</span><span class="o">=</span><span class="nf">.</span><span class="p">(</span><span class="n">gene_name</span><span class="p">)]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>CL1 CL2 CL3 CL4 CL5 CL6 
195  31 138  94 174  93 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># subset gene to top 2 tiers: with C1/2/3/4 confidence</span>
<span class="n">res_adxlocfge_cont_top</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">res_adxlocfge</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;gene_name&#39;</span><span class="p">,</span><span class="s">&#39;context_group&#39;</span><span class="p">))[</span><span class="n">gene_name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">gene_name</span><span class="p">[</span><span class="n">confidence_lvl</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;CL1&#39;</span><span class="p">,</span><span class="s">&#39;CL2&#39;</span><span class="p">,</span><span class="s">&#39;CL3&#39;</span><span class="p">,</span><span class="s">&#39;CL4&#39;</span><span class="p">)]]</span>
<span class="n">res_adxlocfge_cont_top</span><span class="p">[,</span><span class="n">gene_name</span><span class="o">:=</span><span class="nf">factor</span><span class="p">(</span><span class="n">gene_name</span><span class="p">,</span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="nf">unique</span><span class="p">(</span><span class="n">gene_name</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="n">chr</span><span class="p">,</span><span class="n">pos</span><span class="p">)]))]</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="visualization">
<h2>Visualization<a class="headerlink" href="#visualization" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">conf_colors</span><span class="o">&lt;-</span><span class="nf">fread</span><span class="p">(</span><span class="nf">fp</span><span class="p">(</span><span class="n">out</span><span class="p">,</span><span class="s">&#39;pattern_coloring.tsv&#39;</span><span class="p">))[</span><span class="n">pattern</span><span class="o">%in%</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;CL&#39;</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">)]</span><span class="w">  </span><span class="c1">#FIXME</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">res_adxlocfge_cont_top</span><span class="p">)</span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">gene_name</span><span class="p">,</span><span class="n">x</span><span class="o">=</span><span class="n">context_group</span><span class="p">,</span>
<span class="w">                 </span><span class="n">size</span><span class="o">=</span><span class="n">n_study_group</span><span class="p">,</span>
<span class="w">                 </span><span class="n">col</span><span class="o">=</span><span class="n">confidence_lvl_group</span><span class="p">))</span><span class="o">+</span>
<span class="w">  </span><span class="nf">facet_grid</span><span class="p">(</span><span class="n">chr</span><span class="o">~</span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;free&#39;</span><span class="p">,</span><span class="n">space</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;free&#39;</span><span class="p">)</span><span class="o">+</span><span class="nf">scale_size</span><span class="p">(</span><span class="n">range</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1.5</span><span class="p">,</span><span class="m">5</span><span class="p">))</span><span class="o">+</span><span class="nf">theme_minimal</span><span class="p">()</span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_x_discrete</span><span class="p">(</span><span class="n">guide</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">guide_axis</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">))</span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Noto Sans&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">24</span><span class="p">),</span>
<span class="w">        </span><span class="n">strip.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">24</span><span class="p">))</span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="s">&#39;Number of datasets&#39;</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;Confidence level&#39;</span><span class="p">)</span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">conf_colors</span><span class="o">$</span><span class="n">fill_color</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">14</span><span class="p">)</span>
<span class="n">p</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/a8aafa881d605fba92e6cec93c6956830adbe4475e053ccf143c32984c46747d.png"><img alt="../../_images/a8aafa881d605fba92e6cec93c6956830adbe4475e053ccf143c32984c46747d.png" src="../../_images/a8aafa881d605fba92e6cec93c6956830adbe4475e053ccf143c32984c46747d.png" style="width: 720px; height: 840px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># sep causal vs correlated and remove all C6</span>
<span class="c1"># C6 is dropped, C1/C2/C3 are called causal, and C4/C5 are called &quot;correlative&quot; or &quot;correlated&quot;</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">res_adxlocfge_cont_top</span><span class="p">[</span><span class="n">confidence_lvl_group</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&#39;CL6&#39;</span><span class="p">]</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[,</span><span class="w"> </span><span class="n">confidence_cat_group</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">confidence_lvl_group</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;CL1&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;CL2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;CL3&#39;</span><span class="p">),</span><span class="w"> </span><span class="s">&#39;putative causal&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;correlated&#39;</span><span class="p">)]</span>

<span class="c1"># add if from APOE region, or from AD by proxy only</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">APOE_region</span><span class="p">),</span><span class="w"> </span><span class="n">APOE_region</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">]</span>

<span class="c1"># Create formatting flags</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[,</span><span class="w"> </span><span class="n">is_apoe_gene</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">any</span><span class="p">(</span><span class="n">APOE_region</span><span class="p">),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;gene_name&#39;</span><span class="p">]</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[,</span><span class="w"> </span><span class="n">has_direct_evidence</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="o">!</span><span class="nf">all</span><span class="p">(</span><span class="n">only_by_proxi</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;gene_name&#39;</span><span class="p">]</span>

<span class="c1"># Create gene_name_2 with plotmath expressions instead of HTML</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[,</span><span class="w"> </span><span class="n">gene_name_2</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">gene_name</span><span class="p">)]</span>

<span class="c1"># Convert to plotmath expressions for formatting</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[,</span><span class="w"> </span><span class="n">gene_name_plotmath</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span>
<span class="w">  </span><span class="n">has_direct_evidence</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">is_apoe_gene</span><span class="p">,</span>
<span class="w">  </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;bolditalic(&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&#39;)&quot;</span><span class="p">),</span>
<span class="w">  </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">has_direct_evidence</span><span class="p">,</span>
<span class="w">         </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;bold(&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&#39;)&quot;</span><span class="p">),</span>
<span class="w">         </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">is_apoe_gene</span><span class="p">,</span>
<span class="w">                </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;italic(&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&#39;)&quot;</span><span class="p">),</span>
<span class="w">                </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&#39;&quot;</span><span class="p">)))</span>
<span class="p">),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;gene_name&#39;</span><span class="p">]</span>

<span class="c1">#add number of GWAS</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[,</span><span class="w"> </span><span class="n">n_gwas_gene</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">strsplit</span><span class="p">(</span><span class="n">gwas_sources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;\\|&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">|&gt;</span><span class="w"> </span><span class="nf">unlist</span><span class="p">()</span><span class="w"> </span><span class="o">|&gt;</span><span class="w"> </span><span class="nf">unique</span><span class="p">()</span><span class="w"> </span><span class="o">|&gt;</span><span class="w"> </span><span class="nf">length</span><span class="p">(),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">.</span><span class="p">(</span><span class="n">gene_name</span><span class="p">)]</span>

<span class="c1">#add number of loci and create final gene_name_4 with plotmath</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[,</span><span class="w"> </span><span class="n">gene_name_4_plotmath</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span>
<span class="w">  </span><span class="n">has_direct_evidence</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">is_apoe_gene</span><span class="p">,</span>
<span class="w">  </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;bolditalic(&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&#39;)&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">n_loci_gene</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;~&#39;*&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;~&#39;(&#39;~&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">n_gwas_gene</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;~&#39;)&#39;&quot;</span><span class="p">),</span>
<span class="w">  </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">has_direct_evidence</span><span class="p">,</span>
<span class="w">         </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;bold(&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&#39;)&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">n_loci_gene</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;~&#39;*&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;~&#39;(&#39;~&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">n_gwas_gene</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;~&#39;)&#39;&quot;</span><span class="p">),</span>
<span class="w">         </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">is_apoe_gene</span><span class="p">,</span>
<span class="w">                </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;italic(&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&#39;)&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">n_loci_gene</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;~&#39;*&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;~&#39;(&#39;~&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">n_gwas_gene</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;~&#39;)&#39;&quot;</span><span class="p">),</span>
<span class="w">                </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">n_loci_gene</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;~&#39;*&#39;&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;~&#39;(&#39;~&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">n_gwas_gene</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;~&#39;)&#39;&quot;</span><span class="p">)))</span>
<span class="p">),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;gene_name&#39;</span><span class="p">]</span>

<span class="c1"># Create proper ordering</span>
<span class="n">clean_order</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">res_adxlocfge_cont_topf</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="n">gene_name</span><span class="p">),</span><span class="w"> </span><span class="nf">.</span><span class="p">(</span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="n">gene_name_4_plotmath</span><span class="p">)])</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[,</span><span class="w"> </span><span class="n">gene_name_4_plotmath</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">gene_name_4_plotmath</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">clean_order</span><span class="o">$</span><span class="n">gene_name_4_plotmath</span><span class="p">)]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Create the plot with plotmath parsing</span>
<span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">res_adxlocfge_cont_topf</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gene_name_4_plotmath</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">context_group</span><span class="p">,</span>
<span class="w">                 </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n_study_group</span><span class="p">,</span>
<span class="w">                 </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">confidence_cat_group</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">facet_grid</span><span class="p">(</span><span class="n">chr</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;free&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">space</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;free&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_size</span><span class="p">(</span><span class="n">range</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1.25</span><span class="p">,</span><span class="w"> </span><span class="m">5.5</span><span class="p">),</span><span class="w"> </span><span class="n">breaks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">7</span><span class="p">,</span><span class="w"> </span><span class="m">12</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_x_discrete</span><span class="p">(</span><span class="n">guide</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">guide_axis</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Noto Sans&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">24</span><span class="p">),</span>
<span class="w">        </span><span class="n">strip.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Number of datasets&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Confidence level&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;cyan3&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;red&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_blank</span><span class="p">())</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_y_discrete</span><span class="p">(</span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="nf">parse</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">18</span><span class="p">)</span>
<span class="n">p2</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/1fd7a4be1c052fe9a78210caa879588644f7282d6a90b778f911babe205aa5ec.png"><img alt="../../_images/1fd7a4be1c052fe9a78210caa879588644f7282d6a90b778f911babe205aa5ec.png" src="../../_images/1fd7a4be1c052fe9a78210caa879588644f7282d6a90b778f911babe205aa5ec.png" style="width: 720px; height: 1080px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># sep in 3categories</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[,</span><span class="n">confidence_cat_group2</span><span class="o">:=</span><span class="nf">ifelse</span><span class="p">(</span><span class="n">confidence_lvl_group</span><span class="o">%in%</span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;CL1&#39;</span><span class="p">,</span><span class="s">&quot;CL2&quot;</span><span class="p">),</span><span class="s">&#39;Putative causal (CL1, CL2)&#39;</span><span class="p">,</span><span class="nf">ifelse</span><span class="p">(</span><span class="n">confidence_lvl_group</span><span class="o">==</span><span class="s">&#39;CL3&#39;</span><span class="p">,</span><span class="s">&#39;Putative causal (CL3)&#39;</span><span class="p">,</span><span class="s">&#39;Associated&#39;</span><span class="p">))]</span>
<span class="n">res_adxlocfge_cont_topf</span><span class="p">[,</span><span class="n">confidence_cat_group2</span><span class="o">:=</span><span class="nf">factor</span><span class="p">(</span><span class="n">confidence_cat_group2</span><span class="p">,</span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;Putative causal (CL1, CL2)&#39;</span><span class="p">,</span><span class="s">&#39;Putative causal (CL3)&#39;</span><span class="p">,</span><span class="s">&#39;Associated&#39;</span><span class="p">))]</span>

<span class="n">res_adxlocfge_cont_topf_3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">res_adxlocfge_cont_topf</span>

<span class="n">p3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">res_adxlocfge_cont_topf_3</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">y</span><span class="o">=</span><span class="n">gene_name_4_plotmath</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="o">=</span><span class="n">context_group</span><span class="p">,</span><span class="w">  </span><span class="c1"># Use plotmath version</span>
<span class="w">                 </span><span class="n">size</span><span class="o">=</span><span class="n">n_study_group</span><span class="p">,</span>
<span class="w">                 </span><span class="n">col</span><span class="o">=</span><span class="n">confidence_cat_group2</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">facet_grid</span><span class="p">(</span><span class="n">chr</span><span class="o">~</span><span class="s">&#39;&#39;</span><span class="p">,</span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;free&#39;</span><span class="p">,</span><span class="n">space</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;free&#39;</span><span class="p">)</span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_size</span><span class="p">(</span><span class="n">range</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1.25</span><span class="p">,</span><span class="m">5.5</span><span class="p">),</span><span class="n">breaks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">7</span><span class="p">,</span><span class="m">12</span><span class="p">))</span><span class="o">+</span><span class="nf">theme_minimal</span><span class="p">()</span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_x_discrete</span><span class="p">(</span><span class="n">guide</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">guide_axis</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">))</span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Noto Sans&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">24</span><span class="p">),</span>
<span class="w">        </span><span class="n">strip.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">))</span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="s">&#39;Number of datasets&#39;</span><span class="p">,</span><span class="n">col</span><span class="o">=</span><span class="s">&#39;Confidence level&#39;</span><span class="p">)</span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;brown1&#39;</span><span class="p">,</span><span class="s">&#39;deepskyblue4&#39;</span><span class="p">,</span><span class="s">&#39;darkseagreen3&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.title.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_blank</span><span class="p">())</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_y_discrete</span><span class="p">(</span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="nf">parse</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">x</span><span class="p">))</span><span class="w">  </span><span class="c1"># Use parse instead of element_markdown</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">18</span><span class="p">)</span>
<span class="n">p3</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/f6255092a6a36865a30f4151df68cdcf4ac5a8e8949f03da6f3fe7a78c86929b.png"><img alt="../../_images/f6255092a6a36865a30f4151df68cdcf4ac5a8e8949f03da6f3fe7a78c86929b.png" src="../../_images/f6255092a6a36865a30f4151df68cdcf4ac5a8e8949f03da6f3fe7a78c86929b.png" style="width: 720px; height: 1080px;" />
</a>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "pixi-r"
        },
        kernelOptions: {
            name: "pixi-r",
            path: "./main_text/6_AD_xQTL_genes"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'pixi-r'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="index.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Molecular dissection of prioritized AD loci</p>
      </div>
    </a>
    <a class="right-next"
       href="../7_AD_gene_prioritized/index.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Molecular insights and gene prioritization at key AD loci</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#motivation">Motivation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#implementation">Implementation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-libraries">Load libraries</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-result">Load result</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#visualization">Visualization</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>