
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>PPI networks for selected AD risk genes in xQTL loci &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/5_AD_xQTL_integration/Figure_5f_PPI';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Consensus Variant-to-Function Score for AD" href="Figure_5g_cV2F.html" />
    <link rel="prev" title="Summary of FunGen-xQTL AD loci table" href="Figure_5d_AD_loci_table_summary.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../1_project_overview/index.html">Multi-omics QTL atlas of the aging human brain</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../2_single_context_cis/index.html">Fine-mapped QTL landscape across brain molecular phenotypes</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_1b_num_xQTL_loci_single_context.html">Figure 1b. Number of detectable and mappable xQTL loci in single−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_EOO.html">EOO results</a></li>

<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_S2_single_context_cs_distance.html">Frequency of Events in Distance Category</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_sLDSC_GWAS_specific.html">SLDSC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_sLDSC_meta_analysis.html">Visualization of SLDSC results</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../3_single_context_multigene_trans/index.html">Coordinated gene regulation and long-range genetic effects</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../4_multi_context/index.html">Cascading genetic effects across molecular contexts</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_2b_num_xQTL_loci_multi_context.html">Number of detectable and mappable xQTL loci in ROSMAP multi−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_2c_multicontext_effect_sign.html">Marginal vs multicontext effect sign heatmap</a></li>

<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_watershed_posterior_plot.html">Watershed Posterior</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="index.html">Regulatory architecture of AD susceptibility loci</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="Figure_5a_AD_GWAS_loci_harmonized.html">AD GWAS loci harmonization and refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5b_AD_GWAS_loci_properties.html">Refined AD GWAS loci properties</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5c_AD_loci_xQTL_annotated.html">AD loci “explained” by xQTL</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5d_AD_loci_table_summary.html">Summary of FunGen-xQTL AD loci table</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">PPI networks for selected AD risk genes in xQTL loci</a></li>

<li class="toctree-l2"><a class="reference internal" href="Figure_5g_cV2F.html">Consensus Variant-to-Function Score for AD</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5h_GWAS_enrichment.html">GWAS Enrichment</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../6_AD_xQTL_genes/index.html">Molecular dissection of prioritized AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization.html">Visualization of FunGen-xQTL AD Loci Table</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../7_AD_gene_prioritized/index.html">Molecular insights and gene prioritization at key AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../7_AD_gene_prioritized/APP_case_example.html">APP</a></li>




<li class="toctree-l2"><a class="reference internal" href="../7_AD_gene_prioritized/ZYX_case_example.html">Molecular insights and gene prioritization at key AD loci</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/5_AD_xQTL_integration/Figure_5f_PPI.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/main_text/5_AD_xQTL_integration/Figure_5f_PPI.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>PPI networks for selected AD risk genes in xQTL loci</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">PPI networks for selected AD risk genes in xQTL loci</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-libraries">Load libraries</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-gene-list-and-twas-list">Load gene list and twas list</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-genes-labelled-by-methods">xQTL only genes labelled by methods</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-genes-labelled-by-non-proxy-ad-genes">xQTL only genes labelled by non proxy AD genes</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-gvs-table2">xQTL + GVS table2</a></li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section id="ppi-networks-for-selected-ad-risk-genes-in-xqtl-loci">
<h1>PPI networks for selected AD risk genes in xQTL loci<a class="headerlink" href="#ppi-networks-for-selected-ad-risk-genes-in-xqtl-loci" title="Link to this heading">#</a></h1>
<p>Author: Ru Feng and Jenny Empawi</p>
<section id="load-libraries">
<h2>Load libraries<a class="headerlink" href="#load-libraries" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">STRINGdb</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">igraph</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidygraph</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggraph</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union



Attaching package: ‘igraph’


The following objects are masked from ‘package:dplyr’:

    as_data_frame, groups, union


The following objects are masked from ‘package:stats’:

    decompose, spectrum


The following object is masked from ‘package:base’:

    union



Attaching package: ‘tidygraph’


The following object is masked from ‘package:igraph’:

    groups


The following object is masked from ‘package:stats’:

    filter


Loading required package: ggplot2


Attaching package: ‘data.table’


The following objects are masked from ‘package:dplyr’:

    between, first, last


── <span class=" -Color -Color-Bold">Attaching core tidyverse packages</span> ──────────────────────── tidyverse 2.0.0 ──
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">forcats  </span> 1.0.0     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">stringr  </span> 1.5.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">lubridate</span> 1.9.3     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tibble   </span> 3.2.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">purrr    </span> 1.0.2     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tidyr    </span> 1.3.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">readr    </span> 2.1.5     
── <span class=" -Color -Color-Bold">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">%--%()</span>       masks <span class=" -Color -Color-Blue">igraph</span>::%--%()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">tibble</span>::<span class=" -Color -Color-Green">as_data_frame()</span> masks <span class=" -Color -Color-Blue">igraph</span>::as_data_frame(), <span class=" -Color -Color-Blue">dplyr</span>::as_data_frame()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">data.table</span>::<span class=" -Color -Color-Green">between()</span>   masks <span class=" -Color -Color-Blue">dplyr</span>::between()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">purrr</span>::<span class=" -Color -Color-Green">compose()</span>        masks <span class=" -Color -Color-Blue">igraph</span>::compose()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">tidyr</span>::<span class=" -Color -Color-Green">crossing()</span>       masks <span class=" -Color -Color-Blue">igraph</span>::crossing()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">tidygraph</span>::<span class=" -Color -Color-Green">filter()</span>     masks <span class=" -Color -Color-Blue">dplyr</span>::filter(), <span class=" -Color -Color-Blue">stats</span>::filter()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">data.table</span>::<span class=" -Color -Color-Green">first()</span>     masks <span class=" -Color -Color-Blue">dplyr</span>::first()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">hour()</span>       masks <span class=" -Color -Color-Blue">data.table</span>::hour()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">isoweek()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::isoweek()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">lag()</span>            masks <span class=" -Color -Color-Blue">stats</span>::lag()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">data.table</span>::<span class=" -Color -Color-Green">last()</span>      masks <span class=" -Color -Color-Blue">dplyr</span>::last()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">mday()</span>       masks <span class=" -Color -Color-Blue">data.table</span>::mday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">minute()</span>     masks <span class=" -Color -Color-Blue">data.table</span>::minute()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">month()</span>      masks <span class=" -Color -Color-Blue">data.table</span>::month()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">quarter()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::quarter()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">second()</span>     masks <span class=" -Color -Color-Blue">data.table</span>::second()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">purrr</span>::<span class=" -Color -Color-Green">simplify()</span>       masks <span class=" -Color -Color-Blue">igraph</span>::simplify()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">purrr</span>::<span class=" -Color -Color-Green">transpose()</span>      masks <span class=" -Color -Color-Blue">data.table</span>::transpose()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">wday()</span>       masks <span class=" -Color -Color-Blue">data.table</span>::wday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">week()</span>       masks <span class=" -Color -Color-Blue">data.table</span>::week()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">yday()</span>       masks <span class=" -Color -Color-Blue">data.table</span>::yday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">year()</span>       masks <span class=" -Color -Color-Blue">data.table</span>::year()
<span class=" -Color -Color-Cyan">ℹ</span> Use the conflicted package (<span class=" -Color -Color-Blue">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors
</pre></div>
</div>
</div>
</div>
</section>
<section id="load-gene-list-and-twas-list">
<h2>Load gene list and twas list<a class="headerlink" href="#load-gene-list-and-twas-list" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">twas_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;/Staging/variant_gene_summary/AD_xQTL_integration_summary/AD_genes_FunGen_AD_twas_GVC_xQTL_20250325.tsv&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="n">gene_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;/Staging/variant_gene_summary/AD_xQTL_integration_summary/AD_genes_FunGen_AD_GVC_xQTL_20250325.tsv&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Initialize STRINGdb (change species if needed, 9606 = Homo sapiens)</span>
<span class="n">string_db</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">STRINGdb</span><span class="o">$</span><span class="nf">new</span><span class="p">(</span><span class="n">version</span><span class="o">=</span><span class="s">&quot;11.5&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="o">=</span><span class="m">9606</span><span class="p">,</span><span class="w"> </span><span class="n">score_threshold</span><span class="o">=</span><span class="m">400</span><span class="p">,</span><span class="w"> </span><span class="n">input_directory</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">)</span>

<span class="c1"># Example gene list (use your own gene symbols or Entrez IDs)</span>
<span class="n">xQTL_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">data_resources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;FunGen_xQTL&#39;</span><span class="p">))</span>
<span class="n">genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">xQTL_list</span><span class="o">$</span><span class="n">gene_name</span>

<span class="c1"># Map gene names to STRING IDs</span>
<span class="n">mapped_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">string_db</span><span class="o">$</span><span class="nf">map</span><span class="p">(</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene</span><span class="o">=</span><span class="n">genes</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;gene&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">removeUnmappedRows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># Retrieve interaction network for the genes</span>
<span class="n">ppi_network</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">string_db</span><span class="o">$</span><span class="nf">get_interactions</span><span class="p">(</span><span class="n">mapped_genes</span><span class="o">$</span><span class="n">STRING_id</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning:  we couldn&#39;t map to STRING 2% of your identifiers
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_network_unique</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ppi_network</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">pair</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="nf">pmin</span><span class="p">(</span><span class="n">from</span><span class="p">,</span><span class="w"> </span><span class="n">to</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">pmax</span><span class="p">(</span><span class="n">from</span><span class="p">,</span><span class="w"> </span><span class="n">to</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">distinct</span><span class="p">(</span><span class="n">pair</span><span class="p">,</span><span class="w"> </span><span class="n">.keep_all</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">genes_with_ppi</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mapped_genes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">STRING_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">ppi_network</span><span class="o">$</span><span class="n">from</span><span class="p">,</span><span class="w"> </span><span class="n">ppi_network</span><span class="o">$</span><span class="n">to</span><span class="p">))</span>
<span class="n">genes_with_ppi</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span><span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>186</li><li>2</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">xQTL_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">resource_in_FunGen_xQTL</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">table</span>
<span class="n">xQTL_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
<span class="n">xQTL_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">gene_name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">genes_with_ppi</span><span class="o">$</span><span class="n">gene</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">resource_in_FunGen_xQTL</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">table</span><span class="w"> </span><span class="c1"># the xQTL genes not have any PPI</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>.
                                                  ColocBoost 
                                                         160 
                                    ColocBoost,multi_context 
                                                           2 
                         ColocBoost,multi_context,multi_gene 
                                                           1 
                                   ColocBoost,single_context 
                                                           7 
                         ColocBoost,single_context,MetaBrain 
                                                           1 
              ColocBoost,single_context,MetaBrain,multi_gene 
                                                           1 
                     ColocBoost,single_context,multi_context 
                                                          12 
           ColocBoost,single_context,multi_context,MetaBrain 
                                                           7 
ColocBoost,single_context,multi_context,MetaBrain,multi_gene 
                                                           6 
          ColocBoost,single_context,multi_context,multi_gene 
                                                           1 
                        ColocBoost,single_context,multi_gene 
                                                           2 
                                                   MetaBrain 
                                                           4 
                                        MetaBrain,ColocBoost 
                                                           1 
                                                  multi_gene 
                                                           4 
                                              single_context 
                                                          13 
                                   single_context,ColocBoost 
                                                           3 
                                    single_context,MetaBrain 
                                                           1 
                                single_context,multi_context 
                                                           1 
                     single_context,multi_context,ColocBoost 
                                                           2 
                      single_context,multi_context,MetaBrain 
                                                           3 
           single_context,multi_context,MetaBrain,multi_gene 
                                                           4 
single_context,multi_context,MetaBrain,multi_gene,ColocBoost 
                                                           1 
                     single_context,multi_context,multi_gene 
                                                           1 
                                   single_context,multi_gene 
                                                           2 
</pre></div>
</div>
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>240</li><li>7</li></ol>
</div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>.
                                                  ColocBoost 
                                                          40 
                                   ColocBoost,single_context 
                                                           2 
                     ColocBoost,single_context,multi_context 
                                                           2 
           ColocBoost,single_context,multi_context,MetaBrain 
                                                           1 
                        ColocBoost,single_context,multi_gene 
                                                           1 
                                                   MetaBrain 
                                                           2 
                                        MetaBrain,ColocBoost 
                                                           1 
                                              single_context 
                                                           1 
                                    single_context,MetaBrain 
                                                           1 
                      single_context,multi_context,MetaBrain 
                                                           2 
single_context,multi_context,MetaBrain,multi_gene,ColocBoost 
                                                           1 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">gvc_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">data_resources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;FunGen_AD_PI|ADSP_GVC&#39;</span><span class="p">))</span>
<span class="n">real_gvc_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">data_resources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;ADSP_GVC&#39;</span><span class="p">))</span><span class="w"> </span><span class="c1"># 20 genes from GVC table2</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">xQTL_only</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">Reduce</span><span class="p">(</span><span class="n">setdiff</span><span class="p">,</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">xQTL_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">gvc_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">twas_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">))</span><span class="w"> </span><span class="c1"># the genes unqiue from xQTL, NOT from gvc and twas</span>
<span class="n">xQTL_only</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">xQTL_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">xQTL_only</span><span class="p">)</span>
<span class="n">xQTL_only</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
<span class="n">xQTL_only</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">resource_in_FunGen_xQTL</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">table</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>151</li><li>7</li></ol>
</div><div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>.
                                                  ColocBoost 
                                                         116 
                         ColocBoost,multi_context,multi_gene 
                                                           1 
                                   ColocBoost,single_context 
                                                           1 
                     ColocBoost,single_context,multi_context 
                                                           4 
           ColocBoost,single_context,multi_context,MetaBrain 
                                                           2 
ColocBoost,single_context,multi_context,MetaBrain,multi_gene 
                                                           2 
                        ColocBoost,single_context,multi_gene 
                                                           2 
                                                   MetaBrain 
                                                           1 
                                                  multi_gene 
                                                           2 
                                              single_context 
                                                          10 
                                    single_context,MetaBrain 
                                                           1 
                                single_context,multi_context 
                                                           1 
                      single_context,multi_context,MetaBrain 
                                                           3 
           single_context,multi_context,MetaBrain,multi_gene 
                                                           2 
                     single_context,multi_context,multi_gene 
                                                           1 
                                   single_context,multi_gene 
                                                           2 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">cb_apr25</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/ad_xqtl_colocboost_export_filtered.bed.gz&#39;</span><span class="p">)</span><span class="w"> </span><span class="c1">#FIXME</span>
<span class="n">cb_apr25</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_ID</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">xQTL_only</span><span class="o">$</span><span class="n">gene_id</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span><span class="w"> </span><span class="c1"># 128 xQTL only genes are from colocboost</span>
<span class="c1">#cb_apr25 %&gt;% filter(gene_ID %in% xQTL_only$gene_id)  %&gt;% mutate(n_count = str_count(event_ID, &#39;;&#39;) + 1)  %&gt;% filter(n_count == 2) %&gt;% pull(gene_ID) %&gt;% unique %&gt;% length</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">128</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">xQTL_only_1_se_cb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cb_apr25</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_ID</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">xQTL_only</span><span class="o">$</span><span class="n">gene_id</span><span class="p">)</span><span class="w">  </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">              </span><span class="nf">mutate</span><span class="p">(</span><span class="n">n_count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_count</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;;&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span>
<span class="w">                     </span><span class="n">only_1_se</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">((</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;sQTL&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">n_count</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">only_1_se</span><span class="p">)</span>
<span class="n">xQTL_only_1_se_cb</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span><span class="w"> </span><span class="c1"># should we filter the genes that only colocalized between 1 splicing event and AD? although that is not much</span>
<span class="n">gene_ref</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.region_list&#39;</span><span class="p">)</span><span class="w"> </span><span class="c1">#FIXME</span>
<span class="n">xQTL_only_1_se_cb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">xQTL_only_1_se_cb</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">gene_ref</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="s">&#39;gene_id&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;gene_name&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;gene_ID&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;gene_id&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">35</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># whether the genes from ColocBoost are siginificant different from all genes and PPI genes</span>
<span class="c1"># Create contingency table</span>
<span class="c1">#            | PPI     | No PPI</span>
<span class="c1"># ---------- | ------- | -------</span>
<span class="c1"># methodA    | 120     | 40</span>
<span class="c1"># notMethodA | 66      | 14</span>

<span class="c1"># Total genes with PPI = 186</span>
<span class="c1"># Total genes without PPI = 240 - 186 = 54</span>
<span class="c1"># Total methodA = 160; of which 40 are without PPI → 120 with PPI</span>
<span class="c1"># Total not methodA = 80; of which 14 are without PPI → 66 with PPI</span>

<span class="n">mat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">120</span><span class="p">,</span><span class="w"> </span><span class="m">40</span><span class="p">,</span><span class="w"> </span><span class="m">66</span><span class="p">,</span><span class="w"> </span><span class="m">14</span><span class="p">),</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span>
<span class="w">              </span><span class="n">dimnames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;methodA&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;notMethodA&quot;</span><span class="p">),</span>
<span class="w">                              </span><span class="n">PPI</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;PPI&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;No_PPI&quot;</span><span class="p">)))</span>

<span class="c1"># Perform Fisher&#39;s exact test</span>
<span class="nf">fisher.test</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>	Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.2508
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.297715 1.302572
sample estimates:
odds ratio 
 0.6375147 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">symbol_lookup</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">mapped_genes</span><span class="p">,</span><span class="w"> </span><span class="n">STRING_id</span><span class="p">,</span><span class="w"> </span><span class="n">gene</span><span class="p">)</span>
<span class="n">symbol_lookup</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">distinct</span><span class="p">(</span><span class="n">symbol_lookup</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Build igraph object</span>
<span class="n">g</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">graph_from_data_frame</span><span class="p">(</span><span class="n">ppi_network</span><span class="p">[,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;from&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;to&quot;</span><span class="p">)],</span><span class="w"> </span><span class="n">directed</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span>
<span class="n">g</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">igraph</span><span class="o">::</span><span class="nf">simplify</span><span class="p">(</span><span class="n">g</span><span class="p">,</span><span class="w"> </span><span class="n">remove.multiple</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">remove.loops</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="c1"># Replace vertex names with gene symbols</span>
<span class="nf">V</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="o">$</span><span class="n">name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">symbol_lookup</span><span class="o">$</span><span class="n">gene</span><span class="p">[</span><span class="nf">match</span><span class="p">(</span><span class="nf">V</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="o">$</span><span class="n">name</span><span class="p">,</span><span class="w"> </span><span class="n">symbol_lookup</span><span class="o">$</span><span class="n">STRING_id</span><span class="p">)]</span>
<span class="c1"># Optional: make highlighted genes larger</span>
<span class="c1"># V(g)$size &lt;- ifelse(V(g)$name %in% highlight_genes, 8, 4)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="xqtl-only-genes-labelled-by-methods">
<h2>xQTL only genes labelled by methods<a class="headerlink" href="#xqtl-only-genes-labelled-by-methods" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">xQTL_shared_method</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">xQTL_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">genes_with_ppi</span><span class="o">$</span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="n">gene_name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">xQTL_only</span><span class="o">$</span><span class="n">gene_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">resource_in_FunGen_xQTL</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;,&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_name</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">tg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as_tbl_graph</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">highlight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">real_gvc_list</span><span class="o">$</span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;GVC&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># highlight GVC2 genes as &#39;core genes&#39;</span>
<span class="w">                            </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">xQTL_shared_method</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;xQTL_only_shared_method&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># the genes are xQTL only, but more than 1 methods</span>
<span class="w">                                   </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">xQTL_only_1_se_cb</span><span class="o">$</span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;xQTL_only_only_1_se_cb&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># the genes are only from 1 splicing event colocalized results, which are alarming</span>
<span class="w">                                       </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">xQTL_only</span><span class="o">$</span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;xQTL_only_single_method&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;shared&quot;</span><span class="p">)))))</span><span class="w"> </span><span class="c1"># the genes are xQTL only but only from one methods / shared with GVC or TWAS</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">set.seed</span><span class="p">(</span><span class="m">123</span><span class="p">)</span>
<span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">)</span>
<span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggraph</span><span class="p">(</span><span class="n">tg</span><span class="p">,</span><span class="w"> </span><span class="n">layout</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;fr&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="c1"># fr: Fruchterman-Reingold layout</span>
<span class="w">  </span><span class="nf">geom_edge_link</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey80&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_node_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">highlight</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_node_text</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">name</span><span class="p">),</span><span class="w"> </span><span class="n">repel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;GVC&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;shared&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;orange&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;xQTL_only_shared_method&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;yellow&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;xQTL_only_only_1_se_cb&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;xQTL_only_single_method&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;skyblue&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_void</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;none&quot;</span><span class="p">)</span>

<span class="n">p1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“ggrepel: 20 unlabeled data points (too many overlaps). Consider increasing max.overlaps”
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/5b5410730a08b9bd535b0eea7116e567498ea741a4aeb5e5fab6105872685455.png"><img alt="../../_images/5b5410730a08b9bd535b0eea7116e567498ea741a4aeb5e5fab6105872685455.png" src="../../_images/5b5410730a08b9bd535b0eea7116e567498ea741a4aeb5e5fab6105872685455.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
</section>
<section id="xqtl-only-genes-labelled-by-non-proxy-ad-genes">
<h2>xQTL only genes labelled by non proxy AD genes<a class="headerlink" href="#xqtl-only-genes-labelled-by-non-proxy-ad-genes" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">nonproxy_ad_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readLines</span><span class="p">(</span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/Non_proxy_AD_gene_xQTL.txt&#39;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="n">nonproxy_ad_genes_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_ref</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">nonproxy_ad_genes</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_name</span><span class="p">)</span>
<span class="n">nonproxy_ad_genes_names</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">128</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">tg_non_proxy</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as_tbl_graph</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">highlight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">nonproxy_ad_genes_names</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Non_proxy_AD_genes&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Others&quot;</span><span class="p">))</span><span class="w"> </span><span class="c1"># the genes are xQTL only but only from one methods / shared with GVC or TWAS</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">set.seed</span><span class="p">(</span><span class="m">123</span><span class="p">)</span>
<span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">)</span>
<span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggraph</span><span class="p">(</span><span class="n">tg_non_proxy</span><span class="p">,</span><span class="w"> </span><span class="n">layout</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;fr&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="c1"># fr: Fruchterman-Reingold layout</span>
<span class="w">  </span><span class="nf">geom_edge_link</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey80&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_node_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">highlight</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_node_text</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">name</span><span class="p">),</span><span class="w"> </span><span class="n">repel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Non_proxy_AD_genes&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Others&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;skyblue&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_void</span><span class="p">()</span><span class="w"> </span>
<span class="n">p2</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“ggrepel: 69 unlabeled data points (too many overlaps). Consider increasing max.overlaps”
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/ee6cf0b13ae22d3dcfe9f66bb108b4d56a6a76269d284c7a73058893fe72c371.png"><img alt="../../_images/ee6cf0b13ae22d3dcfe9f66bb108b4d56a6a76269d284c7a73058893fe72c371.png" src="../../_images/ee6cf0b13ae22d3dcfe9f66bb108b4d56a6a76269d284c7a73058893fe72c371.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">which</span><span class="p">(</span><span class="n">tg</span><span class="p">[</span><span class="s">&#39;DYSF&#39;</span><span class="p">,]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.dl-inline {width: auto; margin:0; padding: 0}
.dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}
.dl-inline>dt::after {content: ":\0020"; padding-right: .5ex}
.dl-inline>dt:not(:first-of-type) {padding-left: .5ex}
</style><dl class=dl-inline><dt>STX4</dt><dd>45</dd><dt>BIN1</dt><dd>57</dd><dt>FKRP</dt><dd>92</dd><dt>EMG1</dt><dd>182</dd></dl>
</div></div>
</div>
</section>
</section>
<section id="xqtl-gvs-table2">
<h1>xQTL + GVS table2<a class="headerlink" href="#xqtl-gvs-table2" title="Link to this heading">#</a></h1>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Example gene list (use your own gene symbols or Entrez IDs)</span>
<span class="n">genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">xQTL_list</span><span class="o">$</span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="n">real_gvc_list</span><span class="o">$</span><span class="n">gene_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span>

<span class="c1"># Map gene names to STRING IDs</span>
<span class="n">mapped_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">string_db</span><span class="o">$</span><span class="nf">map</span><span class="p">(</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene</span><span class="o">=</span><span class="n">genes</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;gene&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">removeUnmappedRows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># Retrieve interaction network for the genes</span>
<span class="n">ppi_network</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">string_db</span><span class="o">$</span><span class="nf">get_interactions</span><span class="p">(</span><span class="n">mapped_genes</span><span class="o">$</span><span class="n">STRING_id</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning:  we couldn&#39;t map to STRING 2% of your identifiers
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">ppi_network_unique</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ppi_network</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">pair</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="nf">pmin</span><span class="p">(</span><span class="n">from</span><span class="p">,</span><span class="w"> </span><span class="n">to</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">pmax</span><span class="p">(</span><span class="n">from</span><span class="p">,</span><span class="w"> </span><span class="n">to</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">distinct</span><span class="p">(</span><span class="n">pair</span><span class="p">,</span><span class="w"> </span><span class="n">.keep_all</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">symbol_lookup</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">mapped_genes</span><span class="p">,</span><span class="w"> </span><span class="n">STRING_id</span><span class="p">,</span><span class="w"> </span><span class="n">gene</span><span class="p">)</span>
<span class="n">symbol_lookup</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">distinct</span><span class="p">(</span><span class="n">symbol_lookup</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Build igraph object</span>
<span class="n">g</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">graph_from_data_frame</span><span class="p">(</span><span class="n">ppi_network</span><span class="p">[,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;from&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;to&quot;</span><span class="p">)],</span><span class="w"> </span><span class="n">directed</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span>
<span class="n">g</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">simplify</span><span class="p">(</span><span class="n">g</span><span class="p">)</span>
<span class="c1"># Replace vertex names with gene symbols</span>
<span class="nf">V</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="o">$</span><span class="n">name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">symbol_lookup</span><span class="o">$</span><span class="n">gene</span><span class="p">[</span><span class="nf">match</span><span class="p">(</span><span class="nf">V</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="o">$</span><span class="n">name</span><span class="p">,</span><span class="w"> </span><span class="n">symbol_lookup</span><span class="o">$</span><span class="n">STRING_id</span><span class="p">)]</span>
<span class="c1"># Define genes to highlight</span>
<span class="n">highlight_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">real_gvc_list</span><span class="o">$</span><span class="n">gene_name</span>

<span class="c1"># Assign color: red for highlighted, gray for others</span>
<span class="nf">V</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="o">$</span><span class="n">color</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">V</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="o">$</span><span class="n">name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">highlight_genes</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;gray&quot;</span><span class="p">)</span>

<span class="c1"># Optional: make highlighted genes larger</span>
<span class="nf">V</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="o">$</span><span class="n">size</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">V</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="o">$</span><span class="n">name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">highlight_genes</span><span class="p">,</span><span class="w"> </span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">tg_real_gvc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as_tbl_graph</span><span class="p">(</span><span class="n">g</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">highlight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">real_gvc_list</span><span class="o">$</span><span class="n">gene_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Yes&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;No&quot;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">which</span><span class="p">(</span><span class="n">tg</span><span class="p">[</span><span class="s">&#39;CORO1A&#39;</span><span class="p">,]</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.dl-inline {width: auto; margin:0; padding: 0}
.dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}
.dl-inline>dt::after {content: ":\0020"; padding-right: .5ex}
.dl-inline>dt:not(:first-of-type) {padding-left: .5ex}
</style><dl class=dl-inline><dt>TAOK2</dt><dd>53</dd><dt>DOC2A</dt><dd>71</dd><dt>PTK2B</dt><dd>84</dd><dt>TLCD3B</dt><dd>86</dd><dt>YPEL3</dt><dd>113</dd><dt>INO80E</dt><dd>114</dd><dt>STK4</dt><dd>141</dd><dt>TMEM219</dt><dd>178</dd></dl>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">)</span>
<span class="n">p3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggraph</span><span class="p">(</span><span class="n">tg_real_gvc</span><span class="p">,</span><span class="w"> </span><span class="n">layout</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;fr&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="c1"># fr: Fruchterman-Reingold s</span>
<span class="w">  </span><span class="nf">geom_edge_link</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey80&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_node_point</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">highlight</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_node_text</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">name</span><span class="p">),</span><span class="w"> </span><span class="n">repel</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Yes&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;No&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;skyblue&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_void</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;none&quot;</span><span class="p">)</span>
<span class="n">p3</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
“ggrepel: 45 unlabeled data points (too many overlaps). Consider increasing max.overlaps”
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/72dcaec0532566c7729022c48e1f6f82b4a1a42508467409a21aba70676b8700.png"><img alt="../../_images/72dcaec0532566c7729022c48e1f6f82b4a1a42508467409a21aba70676b8700.png" src="../../_images/72dcaec0532566c7729022c48e1f6f82b4a1a42508467409a21aba70676b8700.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./main_text/5_AD_xQTL_integration"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="Figure_5d_AD_loci_table_summary.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Summary of FunGen-xQTL AD loci table</p>
      </div>
    </a>
    <a class="right-next"
       href="Figure_5g_cV2F.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Consensus Variant-to-Function Score for AD</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">PPI networks for selected AD risk genes in xQTL loci</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-libraries">Load libraries</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-gene-list-and-twas-list">Load gene list and twas list</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-genes-labelled-by-methods">xQTL only genes labelled by methods</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-genes-labelled-by-non-proxy-ad-genes">xQTL only genes labelled by non proxy AD genes</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-gvs-table2">xQTL + GVS table2</a></li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>