
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Consensus Variant-to-Function Score for AD &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/5_AD_xQTL_integration/Figure_5g_cV2F';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="GWAS Enrichment" href="Figure_5h_GWAS_enrichment.html" />
    <link rel="prev" title="PPI Networks for Selected AD Risk Genes in xQTL Loci" href="Figure_5f_PPI.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../1_project_overview/index.html">Multi-omics QTL atlas of the aging human brain</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../2_single_context_cis/index.html">Fine-mapped QTL landscape across brain molecular phenotypes</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_1b_num_xQTL_loci_single_context.html">Figure 1b. Number of detectable and mappable xQTL loci in single−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_EOO.html">EOO results</a></li>

<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_S2_single_context_cs_distance.html">Frequency of Events in Distance Category</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_sLDSC_GWAS_specific.html">SLDSC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_sLDSC_meta_analysis.html">Visualization of SLDSC results</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../3_single_context_multigene_trans/index.html">Coordinated gene regulation and long-range genetic effects</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../4_multi_context/index.html">Cascading genetic effects across molecular contexts</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_2c_multicontext_effect_sign.html">Figure 2c. Marginal vs multicontext effect sign heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_watershed_posterior_plot.html">Watershed Posterior</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="index.html">Regulatory architecture of AD susceptibility loci</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="Figure_5a_AD_GWAS_loci_harmonized.html">AD GWAS loci harmonization and refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5b_AD_GWAS_loci_properties.html">Refined AD GWAS loci properties</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5c_AD_loci_xQTL_annotated.html">AD loci “explained” by xQTL</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5d_AD_loci_table_summary.html">Summary of FunGen-xQTL AD loci table</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5e_GSEA.html">Gene Set Enrichment Analysis for AD Risk Genes</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5f_PPI.html">PPI Networks for Selected AD Risk Genes in xQTL Loci</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Consensus Variant-to-Function Score for AD</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5h_GWAS_enrichment.html">GWAS Enrichment</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../6_AD_xQTL_genes/index.html">Molecular dissection of prioritized AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization.html">Visualization of FunGen-xQTL AD Loci Table</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../7_AD_gene_prioritized/index.html">Molecular insights and gene prioritization at key AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../7_AD_gene_prioritized/APP_case_example.html">APP</a></li>




<li class="toctree-l2"><a class="reference internal" href="../7_AD_gene_prioritized/ZYX_case_example.html">Molecular insights and gene prioritization at key AD loci</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/5_AD_xQTL_integration/Figure_5g_cV2F.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/main_text/5_AD_xQTL_integration/Figure_5g_cV2F.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Consensus Variant-to-Function Score for AD</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#authors-ru-feng-and-jenny-empawi">Authors: Ru Feng and Jenny Empawi</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#motivation">Motivation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#implementation">Implementation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#fungen-xqtl-cv2f-analysis">FunGen xQTL cv2f analysis</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#preparing-the-feature-sets">1. Preparing the Feature Sets</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#reformat-the-feature-sets">Reformat the feature sets</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-feature-set-data">Load feature set data</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#note-different-columns-per-chromosome">NOTE: Different columns per chromosome</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-rsid-reference">Load rsid reference</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#map-rsid-to-chr-pos">map rsid to chr_pos</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#combine-with-cv2f-feature-sets">Combine with cv2f feature sets</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#check-feature-sets">check feature sets</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#generating-positive-and-negative-sets">2. Generating Positive and Negative Sets</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#generate-variant-list-from-feature-set">generate variant list from feature set</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#positive-set">positive set</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#define-parameter-sets">Define parameter sets</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#submit-jobs-for-cv2f-analysis">3. Submit Jobs for cv2f Analysis</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#check-the-time-stamp-of-input-files">Check the time stamp of input files</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#visualization-of-results">4. Visualization of Results</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#pip0-7-cs0-vcp0-8-cos0">PIP0.7_CS0_VCP0.8_COS0</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#optional-experiment">(Optional) Experiment</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#additional-experiment-suggestion-from-gao">Additional Experiment (Suggestion from Gao)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#prepare-new-feature-sets-without-xqtl">prepare new feature sets without xQTL</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#check-new-feature-sets">check new feature sets</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#submit-jobs">submit jobs</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Visualization of Results</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#optional-experiment-2">(Optional) Experiment 2</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Additional Experiment (Suggestion from Gao)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id3">prepare new feature sets without xQTL</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id4">check new feature sets</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#prepare-new-sets">Prepare new sets</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id5">submit jobs</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id6">Visualization of Results</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#visualization-comparison">visualization comparison</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#summary">Summary</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="consensus-variant-to-function-score-for-ad">
<h1>Consensus Variant-to-Function Score for AD<a class="headerlink" href="#consensus-variant-to-function-score-for-ad" title="Link to this heading">#</a></h1>
<section id="authors-ru-feng-and-jenny-empawi">
<h2>Authors: Ru Feng and Jenny Empawi<a class="headerlink" href="#authors-ru-feng-and-jenny-empawi" title="Link to this heading">#</a></h2>
</section>
<section id="motivation">
<h2>Motivation<a class="headerlink" href="#motivation" title="Link to this heading">#</a></h2>
<p>Systematically score and prioritize variants within AD loci using the consensus Variant-to-Function (CV2F) framework. Integrate the xQTL atlas with established functional annotations to improve the CV2F framework for AD.</p>
</section>
<section id="implementation">
<h2>Implementation<a class="headerlink" href="#implementation" title="Link to this heading">#</a></h2>
<p>Leverage 339 original CV2F features and augment with xQTL-specific features, including effect sizes and posterior inclusion probabilities (PIPs) from fine-mapping results. Train a gradient boosting model using AD fine-mapped credible sets and colocalization confidence intervals as positive controls, and non-AD associated variants as negative controls. Validate the framework performance with AUPRC metrics and demonstrate the added value of xQTL features.</p>
</section>
<section id="fungen-xqtl-cv2f-analysis">
<h2>FunGen xQTL cv2f analysis<a class="headerlink" href="#fungen-xqtl-cv2f-analysis" title="Link to this heading">#</a></h2>
<p>This notebook documents the process of <strong>cv2f analysis</strong>, which includes the following key steps:</p>
<p>Here the response would be GWAS signals only, and xQTL related information is in feature set.</p>
<hr class="docutils" />
<p><strong>1. Preparing the Feature Sets</strong></p>
<p>We aggregate the ALL feature sets from the v2g project by Fabiha, where each row represents a SNP–gene pair. For cv2f, we remove the gene information to obtain SNP-level features. The steps are:</p>
<ul class="simple">
<li><p><strong>Map variants to rsIDs</strong> – Required for referencing minor allele frequencies (MAF).</p></li>
<li><p><strong>Retain SNP-only entries</strong> – Extract unique SNPs and aggregate their features by taking the <strong>maximum value across rows</strong>.</p></li>
<li><p><strong>Combine with cv2f feature sets</strong>  - Keep <strong>only</strong> the variants that appear in the xQTL feature sets.</p></li>
<li><p>No aggreagtion among different contexts in this feature set.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">exon</span></code> feature was removed because cv2f drops the first four columns.</p></li>
</ul>
<hr class="docutils" />
<p><strong>2. Generating Positive and Negative Sets for cv2f</strong></p>
<p><strong>NOTE: We do not consider any xQTL dataset in those trainning datasets to avoid information linkage, no matter for positive or negative, and no matter from fine-mapping or colocalization</strong></p>
<p>We evaluated different thresholds when defining the positive feature sets but ultimately reported results using the following criteria.<br />
The threshold combinations we considered included:</p>
<ul class="simple">
<li><p><strong>Fine-mapping thresholds</strong></p>
<ul>
<li><p>PIP &gt; 0.9 with mappable CS size ≤ 3</p></li>
<li><p>PIP &gt; 0.7 with mappable CS size ≤ 10</p></li>
<li><p>PIP &gt; 0.6 with mappable CS size ≤ 10</p></li>
<li><p>PIP &gt; 0.7 with no CS size restriction (all variants in CS)</p></li>
</ul>
</li>
<li><p><strong>Colocalization from ColocBoost thresholds</strong></p>
<ul>
<li><p>VCP &gt; 0.9 with mappable CoS size ≤ 3</p></li>
<li><p>VCP &gt; 0.8 with mappable CoS size ≤ 10</p></li>
<li><p>VCP &gt; 0.6 with mappable CoS size ≤ 10</p></li>
<li><p>VCP &gt; 0.8 with no CoS size restriction (all variants in CoS)</p></li>
</ul>
</li>
</ul>
<p><strong>(1) Positive Set</strong></p>
<ul class="simple">
<li><p>Variants with <strong>PIP &gt; 0.7</strong> in any <strong>AD GWAS fine-mapping</strong> analysis.</p></li>
<li><p>All variants within the <strong>95% credible sets (CS)</strong> from AD GWAS.</p></li>
<li><p>All variants within <strong>CoS</strong> and <strong>uCoS</strong> from <strong>GWAS-only colocalization</strong> results (no xQTL input).</p></li>
</ul>
<blockquote>
<div><p>Note: The criterion <em>“Variants with VCP &gt; 0.8 from GWAS-only colocalization results”</em> is effectively redundant,<br />
since including all variants within CoS and uCoS already encompasses all high-VCP variants from colocalization.</p>
</div></blockquote>
<p><strong>(2) Negative Set</strong></p>
<ul class="simple">
<li><p>Get <strong>NON-NEGATIVE</strong>: Identify variants from the 6 Alzheimer’s disease (AD) (no Jansen or Bellenguez_EDAI) contexts with <strong>PIP &gt; 0.025</strong> (relaxed from the original 0.01 because of upstream finemapping analysis deafult), and those with <strong>VCP &gt; 0.01</strong> from <strong>GWAS-only colocalization</strong>.</p></li>
<li><p>From the feature set, <strong>exclude</strong> variants in these NON-NEGATIVE AD-related sets and the positive set to define the negative set.</p></li>
</ul>
<hr class="docutils" />
<p><strong>3. Submit Jobs for cv2f Analysis</strong></p>
<p>The code is adapted from the <a class="reference external" href="https://github.com/Deylab999MSKCC/cv2f">cv2f GitHub repository</a>.</p>
<p>Minimal debugging edits are adapted to original script. These changes were primarily to address:</p>
<ul class="simple">
<li><p>Incorrect vector names</p></li>
<li><p>Errors caused by inconsistent column numbers across feature sets (as noted in Section 1)</p></li>
<li><p>Minor issues with file suffixes</p></li>
</ul>
<hr class="docutils" />
<p><strong>4. Visualization of Results</strong></p>
</section>
<section id="preparing-the-feature-sets">
<h2>1. Preparing the Feature Sets<a class="headerlink" href="#preparing-the-feature-sets" title="Link to this heading">#</a></h2>
<section id="reformat-the-feature-sets">
<h3>Reformat the feature sets<a class="headerlink" href="#reformat-the-feature-sets" title="Link to this heading">#</a></h3>
<p>from Fabiha’s v2g project to v2f, following the below format</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">head</span> <span class="n">baseline</span><span class="mf">.13</span><span class="o">.</span><span class="n">txt</span><span class="o">|</span><span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">1</span><span class="o">-</span><span class="mi">10</span>
<span class="n">CHR</span>	<span class="n">BP</span>	<span class="n">SNP</span>	<span class="n">CM</span>	<span class="n">base</span>	<span class="n">Coding_UCSC</span>	<span class="n">Coding_UCSC</span><span class="o">.</span><span class="n">flanking</span><span class="mf">.500</span><span class="n">Conserved_LindbladToh</span>	<span class="n">Conserved_LindbladToh</span><span class="o">.</span><span class="n">flanking</span><span class="mf">.500</span>	<span class="n">CTCF_Hoffman</span>
<span class="mi">13</span>	<span class="mi">18445955</span>	<span class="n">rs140871821</span>	<span class="mf">0.0</span>	<span class="mi">1</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">00</span>
<span class="mi">13</span>	<span class="mi">18446201</span>	<span class="n">rs184229798</span>	<span class="mf">0.0</span>	<span class="mi">1</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">00</span>
<span class="mi">13</span>	<span class="mi">18446487</span>	<span class="n">rs12018140</span>	<span class="mf">0.0</span>	<span class="mi">1</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">00</span>
<span class="mi">13</span>	<span class="mi">18446697</span>	<span class="n">rs182776109</span>	<span class="mf">0.0</span>	<span class="mi">1</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">00</span>
<span class="mi">13</span>	<span class="mi">18446716</span>	<span class="n">rs140166094</span>	<span class="mf">0.0</span>	<span class="mi">1</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">00</span>
<span class="mi">13</span>	<span class="mi">18446749</span>	<span class="n">rs149988745</span>	<span class="mf">0.0</span>	<span class="mi">1</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">00</span>
<span class="mi">13</span>	<span class="mi">18446776</span>	<span class="n">rs12017981</span>	<span class="mf">0.0</span>	<span class="mi">1</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">00</span>
<span class="mi">13</span>	<span class="mi">18446878</span>	<span class="n">rs12381094</span>	<span class="mf">0.0</span>	<span class="mi">1</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">00</span>
<span class="mi">13</span>	<span class="mi">18446879</span>	<span class="n">rs12381056</span>	<span class="mf">0.0</span>	<span class="mi">1</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">0</span>	<span class="mi">00</span>
</pre></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>── <span class=" -Color -Color-Bold">Attaching core tidyverse packages</span> ──────────────────────── tidyverse 2.0.0 ──
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">dplyr    </span> 1.1.4     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">readr    </span> 2.1.5
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">forcats  </span> 1.0.0     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">stringr  </span> 1.5.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">ggplot2  </span> 3.5.1     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tibble   </span> 3.2.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">lubridate</span> 1.9.4     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tidyr    </span> 1.3.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">purrr    </span> 1.1.0     
── <span class=" -Color -Color-Bold">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">between()</span>     masks <span class=" -Color -Color-Blue">data.table</span>::between()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">filter()</span>      masks <span class=" -Color -Color-Blue">stats</span>::filter()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">first()</span>       masks <span class=" -Color -Color-Blue">data.table</span>::first()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">hour()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::hour()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">isoweek()</span> masks <span class=" -Color -Color-Blue">data.table</span>::isoweek()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">lag()</span>         masks <span class=" -Color -Color-Blue">stats</span>::lag()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">last()</span>        masks <span class=" -Color -Color-Blue">data.table</span>::last()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">mday()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::mday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">minute()</span>  masks <span class=" -Color -Color-Blue">data.table</span>::minute()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">month()</span>   masks <span class=" -Color -Color-Blue">data.table</span>::month()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">quarter()</span> masks <span class=" -Color -Color-Blue">data.table</span>::quarter()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">second()</span>  masks <span class=" -Color -Color-Blue">data.table</span>::second()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">purrr</span>::<span class=" -Color -Color-Green">transpose()</span>   masks <span class=" -Color -Color-Blue">data.table</span>::transpose()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">wday()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::wday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">week()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::week()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">yday()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::yday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">year()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::year()
<span class=" -Color -Color-Cyan">ℹ</span> Use the conflicted package (<span class=" -Color -Color-Blue">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">snp_ref_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/00-All.variants.gz&#39;</span><span class="w"> </span><span class="c1">#dnSNP database</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="load-feature-set-data">
<h3>Load feature set data<a class="headerlink" href="#load-feature-set-data" title="Link to this heading">#</a></h3>
<p>I would use one chromosome to test the codes and prepare batch script and commands</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">chr</span><span class="o">=</span><span class="m">1</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">file_path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2g/xqtl_feat.&#39;</span><span class="p">,</span><span class="n">chr</span><span class="p">,</span><span class="s">&#39;.txt.gz&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_set</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">file_path</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_set</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 × 269</caption>
<thead>
	<tr><th scope=col>s2g</th><th scope=col>snp</th><th scope=col>gene_id</th><th scope=col>snp_bp</th><th scope=col>exon</th><th scope=col>s_in_g</th><th scope=col>same_tad</th><th scope=col>proximal_tad</th><th scope=col>g_closest_tss</th><th scope=col>single_context.Inh_mega_eQTL.PIP</th><th scope=col>⋯</th><th scope=col>trans.MSBB_eQTL_BM10.weight</th><th scope=col>EpiRaction.EPIraction_Brain_Alzheimer_Control_thresholded_predictions_0.026</th><th scope=col>ABC_dpfc</th><th scope=col>EpiMap</th><th scope=col>Baseline_Predss.dist_to_tss</th><th scope=col>Baseline_Predss.nearest_expressed_gene</th><th scope=col>EpiRaction.EPIraction_Brain_Alzheimer_Disease_thresholded_predictions_0.026</th><th scope=col>EpiRaction.EPIraction_Brain_Alzheimer_Encode_thresholded_predictions_0.026</th><th scope=col>xqtl_only_colocboost_export_filtered_MSBB.VCP</th><th scope=col>xqtl_only_colocboost_export_filtered_ROSMAP.VCP</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr1:10000044_ENSG00000049239</td><td>chr1:10000044</td><td>ENSG00000049239</td><td>10000044</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
	<tr><td>chr1:10000044_ENSG00000054523</td><td>chr1:10000044</td><td>ENSG00000054523</td><td>10000044</td><td>0</td><td>0</td><td>1</td><td>1</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
	<tr><td>chr1:10000044_ENSG00000130939</td><td>chr1:10000044</td><td>ENSG00000130939</td><td>10000044</td><td>0</td><td>0</td><td>1</td><td>1</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
	<tr><td>chr1:10000044_ENSG00000130940</td><td>chr1:10000044</td><td>ENSG00000130940</td><td>10000044</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
	<tr><td>chr1:10000044_ENSG00000142583</td><td>chr1:10000044</td><td>ENSG00000142583</td><td>10000044</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
	<tr><td>chr1:10000044_ENSG00000142655</td><td>chr1:10000044</td><td>ENSG00000142655</td><td>10000044</td><td>0</td><td>0</td><td>1</td><td>1</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="note-different-columns-per-chromosome">
<h3>NOTE: Different columns per chromosome<a class="headerlink" href="#note-different-columns-per-chromosome" title="Link to this heading">#</a></h3>
<p>In practice, I found that only chromosome 1 has 269 columns, while the other chromosomes are missing 4 columns—all of which are trans-related. These trans features may have been too sparse and were likely removed earlier.</p>
</section>
<section id="load-rsid-reference">
<h3>Load rsid reference<a class="headerlink" href="#load-rsid-reference" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Run tabix command directly in shell via R</span>
<span class="n">snp_ref</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">system</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;tabix &quot;</span><span class="p">,</span><span class="n">snp_ref_path</span><span class="p">,</span><span class="w"> </span><span class="n">chr</span><span class="p">),</span><span class="w"> </span><span class="n">intern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># Show output</span>
<span class="nf">head</span><span class="p">(</span><span class="n">snp_ref</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="n">Error</span> <span class="ow">in</span> <span class="n">system</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s2">&quot;tabix &quot;</span><span class="p">,</span> <span class="n">snp_ref_path</span><span class="p">,</span> <span class="nb">chr</span><span class="p">),</span> <span class="n">intern</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">):</span> <span class="n">error</span> <span class="ow">in</span> <span class="n">running</span> <span class="n">command</span>
<span class="ne">Traceback</span>:
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Use read.table with textConnection</span>
<span class="n">snp_ref_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">snp_ref</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span>
<span class="w">                 </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;start&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;end&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;rsid&quot;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">snp_ref_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">snp_ref_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">snp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;chr&#39;</span><span class="p">,</span><span class="n">chr</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">start</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">snp_ref_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 5</caption>
<thead>
	<tr><th></th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>rsid</th><th scope=col>snp</th></tr>
	<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>1</th><td>1</td><td>10019</td><td>10020</td><td>rs775809821 </td><td>chr1:10019</td></tr>
	<tr><th scope=row>2</th><td>1</td><td>10039</td><td>10039</td><td>rs978760828 </td><td>chr1:10039</td></tr>
	<tr><th scope=row>3</th><td>1</td><td>10043</td><td>10043</td><td>rs1008829651</td><td>chr1:10043</td></tr>
	<tr><th scope=row>4</th><td>1</td><td>10051</td><td>10051</td><td>rs1052373574</td><td>chr1:10051</td></tr>
	<tr><th scope=row>5</th><td>1</td><td>10055</td><td>10056</td><td>rs768019142 </td><td>chr1:10055</td></tr>
	<tr><th scope=row>6</th><td>1</td><td>10055</td><td>10055</td><td>rs892501864 </td><td>chr1:10055</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="map-rsid-to-chr-pos">
<h3>map rsid to chr_pos<a class="headerlink" href="#map-rsid-to-chr-pos" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_set</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>21551170</li><li>269</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_set_snp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">feat_set</span><span class="p">,</span><span class="w"> </span><span class="n">snp_ref_df</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;snp&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_set_snp</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="n">snp</span><span class="p">,</span><span class="w"> </span><span class="n">rsid</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">distinct</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">fwrite</span><span class="p">(</span><span class="n">.</span><span class="p">,</span>
<span class="w">       </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/xqtl_feature.var_list.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">chr</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt.gz&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
<span class="c1"># save variant list as intermediate file for preparing pos and neg sets </span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_set_snp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">feat_set_snp</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="n">chr</span><span class="p">,</span><span class="w"> </span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="n">rsid</span><span class="p">,</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">feat_set</span><span class="p">)[</span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">)])</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">rename</span><span class="p">(</span><span class="n">CHR</span><span class="o">=</span><span class="n">chr</span><span class="p">,</span><span class="w"> </span><span class="n">BP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="n">SNP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rsid</span><span class="p">)</span><span class="w"> </span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Save mapped feature set</span>
<span class="nf">fwrite</span><span class="p">(</span><span class="nf">distinct</span><span class="p">(</span><span class="n">feat_set_snp</span><span class="p">),</span>
<span class="w">       </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/xqtl_feature.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">chr</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt.gz&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">agg_snp_max</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">feat_set_snp</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="c1"># add head here for quick check, would remove in script </span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">SNP</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">summarise</span><span class="p">(</span><span class="nf">across</span><span class="p">(</span><span class="nf">where</span><span class="p">(</span><span class="n">is.numeric</span><span class="p">),</span><span class="w"> </span><span class="n">max</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">feat_set_snp</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">impacted_cols</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">feat_set_snp</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="nf">where</span><span class="p">(</span><span class="n">is.numeric</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">colnames</span><span class="p">()</span>
<span class="nf">ncol</span><span class="p">(</span><span class="n">feat_set_snp</span><span class="p">)</span>
<span class="nf">length</span><span class="p">(</span><span class="n">impacted_cols</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">267</div><div class="output text_html">266</div></div>
</div>
<p>That means all columns are aggregated except for the SNP ID column, which is as expected.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">agg_snp_max</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A tibble: 3 × 267</caption>
<thead>
	<tr><th scope=col>CHR</th><th scope=col>BP</th><th scope=col>SNP</th><th scope=col>exon</th><th scope=col>s_in_g</th><th scope=col>same_tad</th><th scope=col>proximal_tad</th><th scope=col>g_closest_tss</th><th scope=col>single_context.Inh_mega_eQTL.PIP</th><th scope=col>single_context.Inh_mega_eQTL.conditional_effect</th><th scope=col>⋯</th><th scope=col>trans.MSBB_eQTL_BM10.weight</th><th scope=col>EpiRaction.EPIraction_Brain_Alzheimer_Control_thresholded_predictions_0.026</th><th scope=col>ABC_dpfc</th><th scope=col>EpiMap</th><th scope=col>Baseline_Predss.dist_to_tss</th><th scope=col>Baseline_Predss.nearest_expressed_gene</th><th scope=col>EpiRaction.EPIraction_Brain_Alzheimer_Disease_thresholded_predictions_0.026</th><th scope=col>EpiRaction.EPIraction_Brain_Alzheimer_Encode_thresholded_predictions_0.026</th><th scope=col>xqtl_only_colocboost_export_filtered_MSBB.VCP</th><th scope=col>xqtl_only_colocboost_export_filtered_ROSMAP.VCP</th></tr>
	<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>21</td><td>10006045</td><td>rs540071081</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
	<tr><td>21</td><td>10009651</td><td>rs866528671</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
	<tr><td>21</td><td>10001433</td><td>rs867788656</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">fwrite</span><span class="p">(</span><span class="n">agg_snp_mean</span><span class="p">,</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/xqtl_feat.&#39;</span><span class="p">,</span><span class="n">chr</span><span class="p">,</span><span class="s">&#39;.txt.gz&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<p>aggreate above steps into R script and submit batch jobs:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/usr/bin/env Rscript</span>

<span class="c1"># Load required libraries</span>
<span class="n">library</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">table</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>

<span class="c1"># Get chromosome from command line argument</span>
<span class="n">args</span> <span class="o">&lt;-</span> <span class="n">commandArgs</span><span class="p">(</span><span class="n">trailingOnly</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>
<span class="nb">chr</span> <span class="o">&lt;-</span> <span class="k">as</span><span class="o">.</span><span class="n">integer</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>

<span class="c1"># Define paths</span>
<span class="n">snp_ref_path</span> <span class="o">&lt;-</span> <span class="s1">&#39;/lila/data/deyk/ftp_fgc_xqtl/references/00-All.variants.gz&#39;</span>
<span class="n">file_path</span> <span class="o">&lt;-</span> <span class="n">paste0</span><span class="p">(</span><span class="s1">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2g/xqtl_feat.&#39;</span><span class="p">,</span> <span class="nb">chr</span><span class="p">,</span> <span class="s1">&#39;.txt.gz&#39;</span><span class="p">)</span>

<span class="c1"># Load feature set data</span>
<span class="n">feat_set</span> <span class="o">&lt;-</span> <span class="n">fread</span><span class="p">(</span><span class="n">file_path</span><span class="p">)</span>

<span class="c1"># Load rsid reference using tabix</span>
<span class="n">snp_ref</span> <span class="o">&lt;-</span> <span class="n">system</span><span class="p">(</span><span class="n">paste</span><span class="p">(</span><span class="s2">&quot;tabix&quot;</span><span class="p">,</span> <span class="n">snp_ref_path</span><span class="p">,</span> <span class="nb">chr</span><span class="p">),</span> <span class="n">intern</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)</span>
<span class="n">snp_ref_df</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">table</span><span class="p">(</span><span class="n">text</span> <span class="o">=</span> <span class="n">snp_ref</span><span class="p">,</span> <span class="n">sep</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="n">FALSE</span><span class="p">,</span>
                         <span class="n">col</span><span class="o">.</span><span class="n">names</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;chr&quot;</span><span class="p">,</span> <span class="s2">&quot;start&quot;</span><span class="p">,</span> <span class="s2">&quot;end&quot;</span><span class="p">,</span> <span class="s2">&quot;rsid&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
  <span class="n">mutate</span><span class="p">(</span><span class="n">snp</span> <span class="o">=</span> <span class="n">paste0</span><span class="p">(</span><span class="s1">&#39;chr&#39;</span><span class="p">,</span> <span class="nb">chr</span><span class="p">,</span> <span class="s2">&quot;:&quot;</span><span class="p">,</span> <span class="n">start</span><span class="p">))</span>

<span class="c1"># Map rsid to chr_pos</span>
<span class="n">feat_set_snp</span> <span class="o">&lt;-</span> <span class="n">merge</span><span class="p">(</span><span class="n">feat_set</span><span class="p">,</span> <span class="n">snp_ref_df</span><span class="p">,</span> <span class="n">by</span> <span class="o">=</span> <span class="s1">&#39;snp&#39;</span><span class="p">)</span> 
<span class="c1"># feat_set_snp %&gt;% select(snp, rsid) %&gt;% distinct %&gt;% fwrite(.,</span>
<span class="c1">#        paste0(&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/xqtl_feature.var_list.&#39;, chr, &#39;.txt.gz&#39;), sep = &#39;\t&#39;)</span>

<span class="n">feat_set_snp</span> <span class="o">&lt;-</span> <span class="n">feat_set_snp</span> <span class="o">%&gt;%</span>
  <span class="n">select</span><span class="p">(</span><span class="nb">chr</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">rsid</span><span class="p">,</span> <span class="n">all_of</span><span class="p">(</span><span class="n">colnames</span><span class="p">(</span><span class="n">feat_set</span><span class="p">)[</span><span class="o">-</span><span class="n">c</span><span class="p">(</span><span class="mi">1</span><span class="p">:</span><span class="mi">4</span><span class="p">)]))</span> <span class="o">%&gt;%</span>
  <span class="n">rename</span><span class="p">(</span><span class="n">CHR</span> <span class="o">=</span> <span class="nb">chr</span><span class="p">,</span> <span class="n">BP</span> <span class="o">=</span> <span class="n">start</span><span class="p">,</span> <span class="n">SNP</span> <span class="o">=</span> <span class="n">rsid</span><span class="p">)</span>

<span class="c1"># Save mapped feature set</span>
<span class="n">fwrite</span><span class="p">(</span><span class="n">distinct</span><span class="p">(</span><span class="n">feat_set_snp</span><span class="p">),</span>
       <span class="n">paste0</span><span class="p">(</span><span class="s1">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/xqtl_feature.&#39;</span><span class="p">,</span> <span class="nb">chr</span><span class="p">,</span> <span class="s1">&#39;.txt.gz&#39;</span><span class="p">),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="c1"># Aggregate SNP max</span>
<span class="n">agg_snp_max</span> <span class="o">&lt;-</span> <span class="n">feat_set_snp</span> <span class="o">%&gt;%</span>
  <span class="n">group_by</span><span class="p">(</span><span class="n">SNP</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="n">summarise</span><span class="p">(</span><span class="n">across</span><span class="p">(</span><span class="n">where</span><span class="p">(</span><span class="ow">is</span><span class="o">.</span><span class="n">numeric</span><span class="p">),</span> <span class="nb">max</span><span class="p">,</span> <span class="n">na</span><span class="o">.</span><span class="n">rm</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">))</span> <span class="o">%&gt;%</span> 
  <span class="n">select</span><span class="p">(</span><span class="n">colnames</span><span class="p">(</span><span class="n">feat_set_snp</span><span class="p">))</span>


<span class="c1"># Save aggregated feature set</span>
<span class="n">fwrite</span><span class="p">(</span><span class="n">agg_snp_max</span><span class="p">,</span>
       <span class="n">paste0</span><span class="p">(</span><span class="s1">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max/xqtl_feature_max.&#39;</span><span class="p">,</span> <span class="nb">chr</span><span class="p">,</span> <span class="s1">&#39;.txt.gz&#39;</span><span class="p">),</span> <span class="n">sep</span> <span class="o">=</span> <span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#!/bin/bash</span>

<span class="kr">for</span><span class="w"> </span><span class="n">chr</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="p">{</span><span class="m">1</span><span class="kc">..22</span><span class="p">};</span><span class="w"> </span><span class="n">do</span>
<span class="w">  </span><span class="n">bsub</span><span class="w"> </span><span class="o">-</span><span class="n">q</span><span class="w"> </span><span class="n">cpuqueue</span><span class="w"> </span><span class="o">-</span><span class="n">n</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">-</span><span class="n">M</span><span class="w"> </span><span class="m">200</span><span class="n">GB</span><span class="w"> </span><span class="o">-</span><span class="n">W</span><span class="w"> </span><span class="m">10</span><span class="o">:</span><span class="m">00</span><span class="w"> </span>\
<span class="w">    </span><span class="o">-</span><span class="n">o</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">logs</span><span class="o">/</span><span class="n">aggfeat_chr</span><span class="o">$</span><span class="p">{</span><span class="n">chr</span><span class="p">}</span><span class="n">.out</span><span class="w"> </span><span class="o">-</span><span class="n">e</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">logs</span><span class="o">/</span><span class="n">aggfeat_chr</span><span class="o">$</span><span class="p">{</span><span class="n">chr</span><span class="p">}</span><span class="n">.err</span><span class="w"> </span>\
<span class="w">    </span><span class="s">&quot;Rscript ~/Work/v2f/analysis/aggregate_xqtl_features.R ${chr}&quot;</span>
<span class="n">done</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="combine-with-cv2f-feature-sets">
<h3>Combine with cv2f feature sets<a class="headerlink" href="#combine-with-cv2f-feature-sets" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&#39;~/deyk/ru/Work/v2f/analysis&#39;</span><span class="p">)</span>
<span class="n">cv2f_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&quot;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/cv2f_features/cv2f_features/&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="n">agg_xqtl_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&quot;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max/&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
<span class="c1">#### Main loop #### </span>

<span class="kr">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">22</span><span class="p">){</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
<span class="w">  </span><span class="n">cv2f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/cv2f_features/cv2f_features/baseline_annotation_chrmbpnet.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt&#39;</span><span class="p">))</span>
<span class="w">  </span><span class="n">s2g</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max//xqtl_feature_max.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt.gz&#39;</span><span class="p">))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">#### add cv2f feature #### </span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Extract cv2f features (skip first 4 columns)</span>
<span class="w">  </span><span class="n">cv2f_features</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">cv2f</span><span class="p">)[</span><span class="m">5</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">cv2f</span><span class="p">)]</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Create snp column</span>
<span class="w">  </span><span class="n">cv2f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cv2f</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">snp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">CHR</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">BP</span><span class="p">))</span>
<span class="w">  </span><span class="c1"># Merge s2g with cv2f features</span>
<span class="w">  </span><span class="n">s2g</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">s2g</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">snp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">CHR</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">BP</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">left_join</span><span class="p">(</span><span class="n">cv2f</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="nf">all_of</span><span class="p">(</span><span class="n">cv2f_features</span><span class="p">),</span><span class="w"> </span><span class="n">snp</span><span class="p">),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;snp&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="nf">across</span><span class="p">(</span><span class="nf">all_of</span><span class="p">(</span><span class="n">cv2f_features</span><span class="p">),</span><span class="w"> </span><span class="o">~</span><span class="nf">replace_na</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">snp</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="nf">fwrite</span><span class="p">(</span><span class="n">s2g</span><span class="p">,</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f/xqtl_feature_max_allcv2f.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt.gz&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
</div>
<section id="check-feature-sets">
<h4>check feature sets<a class="headerlink" href="#check-feature-sets" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f/xqtl_feature_max_allcv2f.22.txt.gz&#39;</span><span class="p">)</span><span class="w"> </span><span class="c1"># xqtl plus all feature from cv2f paper</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_tmp</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">colnames</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'CHR'</li><li>'BP'</li><li>'SNP'</li><li>'exon'</li><li>'s_in_g'</li><li>'same_tad'</li><li>'proximal_tad'</li><li>'g_closest_tss'</li><li>'single_context.Inh_mega_eQTL.PIP'</li><li>'single_context.Inh_mega_eQTL.conditional_effect'</li><li>'single_context.Inh_mega_eQTL.weight'</li><li>'single_context.Inh_Kellis_eQTL.PIP'</li><li>'single_context.Inh_Kellis_eQTL.conditional_effect'</li><li>'single_context.Inh_Kellis_eQTL.weight'</li><li>'single_context.ROSMAP_PCC_sQTL.PIP'</li><li>'single_context.ROSMAP_PCC_sQTL.conditional_effect'</li><li>'single_context.ROSMAP_PCC_sQTL.weight'</li><li>'single_context.MiGA_GTS_eQTL.PIP'</li><li>'single_context.MiGA_GTS_eQTL.conditional_effect'</li><li>'single_context.MiGA_GTS_eQTL.weight'</li><li>'single_context.monocyte_ROSMAP_eQTL.PIP'</li><li>'single_context.monocyte_ROSMAP_eQTL.conditional_effect'</li><li>'single_context.monocyte_ROSMAP_eQTL.weight'</li><li>'single_context.Ast_DeJager_eQTL.PIP'</li><li>'single_context.Ast_DeJager_eQTL.conditional_effect'</li><li>'single_context.Ast_DeJager_eQTL.weight'</li><li>'single_context.Ast_mega_eQTL.PIP'</li><li>'single_context.Ast_mega_eQTL.conditional_effect'</li><li>'single_context.Ast_mega_eQTL.weight'</li><li>'single_context.BM_10_MSBB_eQTL.PIP'</li><li>'single_context.BM_10_MSBB_eQTL.conditional_effect'</li><li>'single_context.BM_10_MSBB_eQTL.weight'</li><li>'single_context.Mic_DeJager_eQTL.PIP'</li><li>'single_context.Mic_DeJager_eQTL.conditional_effect'</li><li>'single_context.Mic_DeJager_eQTL.weight'</li><li>'single_context.Knight_pQTL_brain.PIP'</li><li>'single_context.Knight_pQTL_brain.conditional_effect'</li><li>'single_context.Knight_pQTL_brain.weight'</li><li>'single_context.STARNET_eQTL_Mac.PIP'</li><li>'single_context.STARNET_eQTL_Mac.conditional_effect'</li><li>'single_context.STARNET_eQTL_Mac.weight'</li><li>'single_context.DLPFC_Klein_gpQTL_adjusted.PIP'</li><li>'single_context.DLPFC_Klein_gpQTL_adjusted.conditional_effect'</li><li>'single_context.DLPFC_Klein_gpQTL_adjusted.weight'</li><li>'single_context.MiGA_SVZ_eQTL.PIP'</li><li>'single_context.MiGA_SVZ_eQTL.conditional_effect'</li><li>'single_context.MiGA_SVZ_eQTL.weight'</li><li>'single_context.BM_36_MSBB_eQTL.PIP'</li><li>'single_context.BM_36_MSBB_eQTL.conditional_effect'</li><li>'single_context.BM_36_MSBB_eQTL.weight'</li><li>'single_context.OPC_mega_eQTL.PIP'</li><li>'single_context.OPC_mega_eQTL.conditional_effect'</li><li>'single_context.OPC_mega_eQTL.weight'</li><li>'single_context.BM_44_MSBB_eQTL.PIP'</li><li>'single_context.BM_44_MSBB_eQTL.conditional_effect'</li><li>'single_context.BM_44_MSBB_eQTL.weight'</li><li>'single_context.Mic_mega_eQTL.PIP'</li><li>'single_context.Mic_mega_eQTL.conditional_effect'</li><li>'single_context.Mic_mega_eQTL.weight'</li><li>'single_context.Exc_mega_eQTL.PIP'</li><li>'single_context.Exc_mega_eQTL.conditional_effect'</li><li>'single_context.Exc_mega_eQTL.weight'</li><li>'single_context.AC_DeJager_eQTL.PIP'</li><li>'single_context.AC_DeJager_eQTL.conditional_effect'</li><li>'single_context.AC_DeJager_eQTL.weight'</li><li>'single_context.Exc_DeJager_eQTL.PIP'</li><li>'single_context.Exc_DeJager_eQTL.conditional_effect'</li><li>'single_context.Exc_DeJager_eQTL.weight'</li><li>'single_context.ROSMAP_AC_sQTL.PIP'</li><li>'single_context.ROSMAP_AC_sQTL.conditional_effect'</li><li>'single_context.ROSMAP_AC_sQTL.weight'</li><li>'single_context.Metabrain_Cortex.PIP'</li><li>'single_context.Metabrain_Cortex.conditional_effect'</li><li>'single_context.Metabrain_Cortex.weight'</li><li>'single_context.Mic_12_Kellis_eQTL.PIP'</li><li>'single_context.Mic_12_Kellis_eQTL.conditional_effect'</li><li>'single_context.Mic_12_Kellis_eQTL.weight'</li><li>'single_context.Knight_eQTL_brain.PIP'</li><li>'single_context.Knight_eQTL_brain.conditional_effect'</li><li>'single_context.Knight_eQTL_brain.weight'</li><li>'single_context.BM_22_MSBB_eQTL.PIP'</li><li>'single_context.BM_22_MSBB_eQTL.conditional_effect'</li><li>'single_context.BM_22_MSBB_eQTL.weight'</li><li>'single_context.DLPFC_Bennett_pQTL.PIP'</li><li>'single_context.DLPFC_Bennett_pQTL.conditional_effect'</li><li>'single_context.DLPFC_Bennett_pQTL.weight'</li><li>'single_context.Oli_Kellis_eQTL.PIP'</li><li>'single_context.Oli_Kellis_eQTL.conditional_effect'</li><li>'single_context.Oli_Kellis_eQTL.weight'</li><li>'single_context.Ast_10_Kellis_eQTL.PIP'</li><li>'single_context.Ast_10_Kellis_eQTL.conditional_effect'</li><li>'single_context.Ast_10_Kellis_eQTL.weight'</li><li>'single_context.OPC_DeJager_eQTL.PIP'</li><li>'single_context.OPC_DeJager_eQTL.conditional_effect'</li><li>'single_context.OPC_DeJager_eQTL.weight'</li><li>'single_context.Mic_13_Kellis_eQTL.PIP'</li><li>'single_context.Mic_13_Kellis_eQTL.conditional_effect'</li><li>'single_context.Mic_13_Kellis_eQTL.weight'</li><li>'single_context.Oli_DeJager_eQTL.PIP'</li><li>'single_context.Oli_DeJager_eQTL.conditional_effect'</li><li>'single_context.Oli_DeJager_eQTL.weight'</li><li>'single_context.Oli_mega_eQTL.PIP'</li><li>'single_context.Oli_mega_eQTL.conditional_effect'</li><li>'single_context.Oli_mega_eQTL.weight'</li><li>'single_context.Mic_Kellis_eQTL.PIP'</li><li>'single_context.Mic_Kellis_eQTL.conditional_effect'</li><li>'single_context.Mic_Kellis_eQTL.weight'</li><li>'single_context.PCC_DeJager_eQTL.PIP'</li><li>'single_context.PCC_DeJager_eQTL.conditional_effect'</li><li>'single_context.PCC_DeJager_eQTL.weight'</li><li>'single_context.MiGA_GFM_eQTL.PIP'</li><li>'single_context.MiGA_GFM_eQTL.conditional_effect'</li><li>'single_context.MiGA_GFM_eQTL.weight'</li><li>'single_context.DLPFC_DeJager_eQTL.PIP'</li><li>'single_context.DLPFC_DeJager_eQTL.conditional_effect'</li><li>'single_context.DLPFC_DeJager_eQTL.weight'</li><li>'single_context.DLPFC_Klein_gpQTL_unadjusted.PIP'</li><li>'single_context.DLPFC_Klein_gpQTL_unadjusted.conditional_effect'</li><li>'single_context.DLPFC_Klein_gpQTL_unadjusted.weight'</li><li>'single_context.Metabrain_Basalganglia.PIP'</li><li>'single_context.Metabrain_Basalganglia.conditional_effect'</li><li>'single_context.Metabrain_Basalganglia.weight'</li><li>'single_context.OPC_Kellis_eQTL.PIP'</li><li>'single_context.OPC_Kellis_eQTL.conditional_effect'</li><li>'single_context.OPC_Kellis_eQTL.weight'</li><li>'single_context.MiGA_THA_eQTL.PIP'</li><li>'single_context.MiGA_THA_eQTL.conditional_effect'</li><li>'single_context.MiGA_THA_eQTL.weight'</li><li>'single_context.Ast_Kellis_eQTL.PIP'</li><li>'single_context.Ast_Kellis_eQTL.conditional_effect'</li><li>'single_context.Ast_Kellis_eQTL.weight'</li><li>'single_context.ROSMAP_DLPFC_sQTL.PIP'</li><li>'single_context.ROSMAP_DLPFC_sQTL.conditional_effect'</li><li>'single_context.ROSMAP_DLPFC_sQTL.weight'</li><li>'single_context.Inh_DeJager_eQTL.PIP'</li><li>'single_context.Inh_DeJager_eQTL.conditional_effect'</li><li>'single_context.Inh_DeJager_eQTL.weight'</li><li>'single_context.Metabrain_Cerebellum.PIP'</li><li>'single_context.Metabrain_Cerebellum.conditional_effect'</li><li>'single_context.Metabrain_Cerebellum.weight'</li><li>'single_context.Exc_Kellis_eQTL.PIP'</li><li>'single_context.Exc_Kellis_eQTL.conditional_effect'</li><li>'single_context.Exc_Kellis_eQTL.weight'</li><li>'single_context.MSBB_BM36_pQTL.PIP'</li><li>'single_context.MSBB_BM36_pQTL.conditional_effect'</li><li>'single_context.MSBB_BM36_pQTL.weight'</li><li>'single_context.Metabrain_Spinalcord.PIP'</li><li>'single_context.Metabrain_Spinalcord.conditional_effect'</li><li>'single_context.Metabrain_Spinalcord.weight'</li><li>'single_context.Metabrain_Hippocampus.PIP'</li><li>'single_context.Metabrain_Hippocampus.conditional_effect'</li><li>'single_context.Metabrain_Hippocampus.weight'</li><li>'multi_context.MSBB.PIP'</li><li>'multi_context.MSBB.BM_10_MSBB_eQTL.conditional_effect'</li><li>'multi_context.MSBB.BM_10_MSBB_eQTL.weight'</li><li>'multi_context.MSBB.BM_22_MSBB_eQTL.conditional_effect'</li><li>'multi_context.MSBB.BM_22_MSBB_eQTL.weight'</li><li>'multi_context.MSBB.BM_44_MSBB_eQTL.conditional_effect'</li><li>'multi_context.MSBB.BM_44_MSBB_eQTL.weight'</li><li>'multi_context.MSBB.BM_36_MSBB_eQTL.conditional_effect'</li><li>'multi_context.MSBB.BM_36_MSBB_eQTL.weight'</li><li>'multi_context.ROSMAP.PIP'</li><li>'multi_context.ROSMAP.OPC_DeJager_eQTL.conditional_effect'</li><li>'multi_context.ROSMAP.OPC_DeJager_eQTL.weight'</li><li>'multi_context.ROSMAP.Exc_DeJager_eQTL.conditional_effect'</li><li>'multi_context.ROSMAP.Exc_DeJager_eQTL.weight'</li><li>'multi_context.ROSMAP.DLPFC_DeJager_eQTL.conditional_effect'</li><li>'multi_context.ROSMAP.DLPFC_DeJager_eQTL.weight'</li><li>'multi_context.ROSMAP.DLPFC_Bennett_pQTL.conditional_effect'</li><li>'multi_context.ROSMAP.DLPFC_Bennett_pQTL.weight'</li><li>'multi_context.ROSMAP.Mic_DeJager_eQTL.conditional_effect'</li><li>'multi_context.ROSMAP.Mic_DeJager_eQTL.weight'</li><li>'multi_context.ROSMAP.Inh_DeJager_eQTL.conditional_effect'</li><li>'multi_context.ROSMAP.Inh_DeJager_eQTL.weight'</li><li>'multi_context.ROSMAP.Oli_DeJager_eQTL.conditional_effect'</li><li>'multi_context.ROSMAP.Oli_DeJager_eQTL.weight'</li><li>'multi_context.ROSMAP.Ast_DeJager_eQTL.conditional_effect'</li><li>'multi_context.ROSMAP.Ast_DeJager_eQTL.weight'</li><li>'multi_context.ROSMAP.monocyte_ROSMAP_eQTL.conditional_effect'</li><li>'multi_context.ROSMAP.monocyte_ROSMAP_eQTL.weight'</li><li>'multi_context.ROSMAP.PCC_DeJager_eQTL.conditional_effect'</li><li>'multi_context.ROSMAP.PCC_DeJager_eQTL.weight'</li><li>'multi_context.ROSMAP.AC_DeJager_eQTL.conditional_effect'</li><li>'multi_context.ROSMAP.AC_DeJager_eQTL.weight'</li><li>'multi_gene.ROSMAP_Mic_DeJager_eQTL.PIP'</li><li>'multi_gene.ROSMAP_Mic_DeJager_eQTL.conditional_effect'</li><li>'multi_gene.ROSMAP_Mic_DeJager_eQTL.weight'</li><li>'multi_gene.ROSMAP_Exc_mega_eQTL.PIP'</li><li>'multi_gene.ROSMAP_Exc_mega_eQTL.conditional_effect'</li><li>'multi_gene.ROSMAP_Exc_mega_eQTL.weight'</li><li>'multi_gene.ROSMAP_PCC_DeJager_eQTL.PIP'</li><li>'multi_gene.ROSMAP_PCC_DeJager_eQTL.conditional_effect'</li><li>'multi_gene.ROSMAP_PCC_DeJager_eQTL.weight'</li><li>'multi_gene.ROSMAP_Ast_DeJager_eQTL.PIP'</li><li>'multi_gene.ROSMAP_Ast_DeJager_eQTL.conditional_effect'</li><li>'multi_gene.ROSMAP_Ast_DeJager_eQTL.weight'</li><li>'multi_gene.MSBB_BM_44_eQTL.PIP'</li><li>'multi_gene.MSBB_BM_44_eQTL.conditional_effect'</li><li>'multi_gene.MSBB_BM_44_eQTL.weight'</li><li>'multi_gene.ROSMAP_Oli_DeJager_eQTL.PIP'</li><li>⋯</li><li>'Enformer.Enformer_kidney_DNASE'</li><li>'Enformer.Enformer_kidney_all'</li><li>'Enformer.Enformer_liver_CHIP'</li><li>'Enformer.Enformer_liver_all'</li><li>'Enformer.Enformer_lung_CAGE'</li><li>'Enformer.Enformer_lung_CHIP'</li><li>'Enformer.Enformer_lung_DNASE'</li><li>'Enformer.Enformer_lung_H3K4me3'</li><li>'Enformer.Enformer_lung_all'</li><li>'Enformer.Enformer_skin_CHIP'</li><li>'Enformer.Enformer_skin_DNASE'</li><li>'Enformer.Enformer_skin_H3K4me3'</li><li>'Enformer.Enformer_skin_all'</li><li>'Enhancer_Andersson'</li><li>'Enhancer_Andersson.flanking.500'</li><li>'Enhancer_Hoffman'</li><li>'Enhancer_Hoffman.flanking.500'</li><li>'FetalDHS_Trynka'</li><li>'FetalDHS_Trynka.flanking.500'</li><li>'GERP.NS'</li><li>'GERP.RSsup4'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_all'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_blood'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_brain'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_fat'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_gut'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_heart'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_kidney'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_liver'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_lung'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_skin'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_all'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_blood'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_brain'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_fat'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_gut'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_heart'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_kidney'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_liver'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_lung'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_skin'</li><li>'GTEx_eQTL_MaxCPP'</li><li>'H3K27ac_Hnisz'</li><li>'H3K27ac_Hnisz.flanking.500'</li><li>'H3K27ac_PGC2'</li><li>'H3K27ac_PGC2.flanking.500'</li><li>'H3K4me1_Trynka'</li><li>'H3K4me1_Trynka.flanking.500'</li><li>'H3K4me1_peaks_Trynka'</li><li>'H3K4me3_Trynka'</li><li>'H3K4me3_Trynka.flanking.500'</li><li>'H3K4me3_peaks_Trynka'</li><li>'H3K9ac_Trynka'</li><li>'H3K9ac_Trynka.flanking.500'</li><li>'H3K9ac_peaks_Trynka'</li><li>'Hongkai_conserve.cob_H3K27ac_fdr.rsid'</li><li>'Hongkai_conserve.cob_H3K4me1_fdr.rsid'</li><li>'Hongkai_conserve.cob_H3K4me3_fdr.rsid'</li><li>'Hongkai_conserve.cob_all_fdr.rsid'</li><li>'Hongkai_conserve.cob_chromatin_accessibility_fdr.rsid'</li><li>'Hongkai_conserve.cov_H3K27ac_fdr.rsid'</li><li>'Hongkai_conserve.cov_H3K4me1_fdr.rsid'</li><li>'Hongkai_conserve.cov_H3K4me3_fdr.rsid'</li><li>'Hongkai_conserve.cov_all_fdr.rsid'</li><li>'Hongkai_conserve.cov_chromatin_accessibility_fdr.rsid'</li><li>'Hongkai_conserve.cov_cob_H3K27ac_fdr.rsid'</li><li>'Hongkai_conserve.cov_cob_H3K4me1_fdr.rsid'</li><li>'Hongkai_conserve.cov_cob_H3K4me3_fdr.rsid'</li><li>'Hongkai_conserve.cov_cob_all_fdr.rsid'</li><li>'Hongkai_conserve.cov_cob_chromatin_accessibility_fdr.rsid'</li><li>'Human_Enhancer_Villar'</li><li>'Human_Enhancer_Villar.flanking.500'</li><li>'Human_Enhancer_Villar_Species_Enhancer_Count'</li><li>'Human_Promoter_Villar'</li><li>'Human_Promoter_Villar.flanking.500'</li><li>'Human_Promoter_Villar_ExAC'</li><li>'Human_Promoter_Villar_ExAC.flanking.500'</li><li>'Intron_UCSC'</li><li>'Intron_UCSC.flanking.500'</li><li>'MPRA_2023_Ryan.GTEx.annotations.rsids_FDR10_HepG2'</li><li>'MPRA_2023_Ryan.GTEx.annotations.rsids_FDR10_K562'</li><li>'MPRA_2023_Ryan.GTEx.annotations.rsids_FDR10_SKNSH'</li><li>'MPRA_2023_Ryan.GTEx.annotations.rsids_all'</li><li>'MPRA_RyanTewhey_deeplearn_2023.All.skew.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.All.skew.bin.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.All.skew.refalt.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.HepG2.skew.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.HepG2.skew.bin.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.HepG2.skew.refalt.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.K562.skew.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.K562.skew.bin.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.K562.skew.refalt.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.SKNSH.skew.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.SKNSH.skew.bin.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.SKNSH.skew.refalt.MPRAdeep'</li><li>'Nucleotide_Diversity_10kb'</li><li>'PromoterFlanking_Hoffman'</li><li>'PromoterFlanking_Hoffman.flanking.500'</li><li>'Promoter_UCSC'</li><li>'Promoter_UCSC.flanking.500'</li><li>'Recomb_Rate_10kb'</li><li>'Repressed_Hoffman'</li><li>'Repressed_Hoffman.flanking.500'</li><li>'SuperEnhancer_Hnisz'</li><li>'SuperEnhancer_Hnisz.flanking.500'</li><li>'TFBS_ENCODE'</li><li>'TFBS_ENCODE.flanking.500'</li><li>'TSS_Hoffman'</li><li>'TSS_Hoffman.flanking.500'</li><li>'Transcr_Hoffman'</li><li>'Transcr_Hoffman.flanking.500'</li><li>'UTR_3_UCSC'</li><li>'UTR_3_UCSC.flanking.500'</li><li>'UTR_5_UCSC'</li><li>'UTR_5_UCSC.flanking.500'</li><li>'WGSTARR.rsids_sigs_AFR_STARR_75'</li><li>'WGSTARR.rsids_sigs_AFR_STARR_90'</li><li>'WeakEnhancer_Hoffman'</li><li>'WeakEnhancer_Hoffman.flanking.500'</li><li>'Zoonomia_Conserve.Zoonomia_Mammals_PhyloP_v1'</li><li>'Zoonomia_Conserve.Zoonomia_Primate_PhastCons_v1'</li><li>'base'</li><li>'cCREv4_CellType_Agnostic.ALL'</li><li>'cCREv4_CellType_Agnostic.PLS'</li><li>'cCREv4_CellType_Agnostic.dELS'</li><li>'cCREv4_CellType_Agnostic.pELS'</li><li>'cCREv4_Human_CellType.A549_PLS'</li><li>'cCREv4_Human_CellType.BLD_PLS'</li><li>'cCREv4_Human_CellType.BLD_dELS'</li><li>'cCREv4_Human_CellType.BLD_pELS'</li><li>'cCREv4_Human_CellType.BRN_PLS'</li><li>'cCREv4_Human_CellType.BRN_dELS'</li><li>'cCREv4_Human_CellType.BRN_pELS'</li><li>'cCREv4_Human_CellType.GM12878_PLS'</li><li>'cCREv4_Human_CellType.GM12878_dELS'</li><li>'cCREv4_Human_CellType.GM12878_pELS'</li><li>'cCREv4_Human_CellType.GUT_PLS'</li><li>'cCREv4_Human_CellType.GUT_dELS'</li><li>'cCREv4_Human_CellType.GUT_pELS'</li><li>'cCREv4_Human_CellType.HRT_PLS'</li><li>'cCREv4_Human_CellType.HRT_dELS'</li><li>'cCREv4_Human_CellType.HRT_pELS'</li><li>'cCREv4_Human_CellType.HepG2_PLS'</li><li>'cCREv4_Human_CellType.HepG2_dELS'</li><li>'cCREv4_Human_CellType.HepG2_pELS'</li><li>'cCREv4_Human_CellType.K562_PLS'</li><li>'cCREv4_Human_CellType.K562_dELS'</li><li>'cCREv4_Human_CellType.K562_pELS'</li><li>'cCREv4_Human_CellType.KID_PLS'</li><li>'cCREv4_Human_CellType.LIV_ALL'</li><li>'cCREv4_Human_CellType.LIV_PLS'</li><li>'cCREv4_Human_CellType.LIV_dELS'</li><li>'cCREv4_Human_CellType.LIV_pELS'</li><li>'cCREv4_Human_CellType.LNG_PLS'</li><li>'cCREv4_Human_CellType.LNG_dELS'</li><li>'cCREv4_Human_CellType.LNG_pELS'</li><li>'cCREv4_Human_CellType.SK-N-SH_PLS'</li><li>'cCREv4_Human_CellType.SK-N-SH_dELS'</li><li>'cCREv4_Human_CellType.SK-N-SH_pELS'</li><li>'cCREv4_Human_CellType.SKIN_PLS'</li><li>'cCREv4_Human_CellType.SKIN_dELS'</li><li>'cCREv4_Human_CellType.SKIN_pELS'</li><li>'chrmbpnet_GM12878_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_GM12878_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_GM12878_jsd_x_max_percentile'</li><li>'chrmbpnet_HepG2_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_HepG2_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_HepG2_jsd_x_max_percentile'</li><li>'chrmbpnet_K562_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_K562_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_K562_jsd_x_max_percentile'</li><li>'chrmbpnet_SKNSH_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_SKNSH_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_SKNSH_jsd_x_max_percentile'</li><li>'chrmbpnet_all_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_all_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_all_jsd_x_max_percentile'</li><li>'chrmbpnet_blood_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_blood_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_blood_jsd_x_max_percentile'</li><li>'chrmbpnet_brain_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_brain_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_brain_jsd_x_max_percentile'</li><li>'chrmbpnet_gut_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_gut_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_gut_jsd_x_max_percentile'</li><li>'chrmbpnet_heart_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_heart_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_heart_jsd_x_max_percentile'</li><li>'chrmbpnet_liver_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_liver_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_liver_jsd_x_max_percentile'</li><li>'chrmbpnet_lung_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_lung_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_lung_jsd_x_max_percentile'</li><li>'chrmbpnet_skin_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_skin_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_skin_jsd_x_max_percentile'</li><li>'non_synonymous'</li><li>'synonymous'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">dim</span><span class="p">(</span><span class="n">feat_tmp</span><span class="p">)</span><span class="w"> </span><span class="c1">## chrom 22</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>236695</li><li>603</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">subset_feat_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_cv2f/xqtl_feature_max_cv2f.22.txt.gz&#39;</span><span class="p">)</span><span class="w"> </span><span class="c1"># xqtl plus subset features from cv2f paper only using brain related feature</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">dim</span><span class="p">(</span><span class="n">subset_feat_tmp</span><span class="p">)</span><span class="w"> </span><span class="c1">## chrom 22</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>236695</li><li>283</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">setdiff</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">subset_feat_tmp</span><span class="p">),</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">feat_tmp</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">0</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">setdiff</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">feat_tmp</span><span class="p">),</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">subset_feat_tmp</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">320</div></div>
</div>
</section>
</section>
</section>
<section id="generating-positive-and-negative-sets">
<h2>2. Generating Positive and Negative Sets<a class="headerlink" href="#generating-positive-and-negative-sets" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>── <span class=" -Color -Color-Bold">Attaching core tidyverse packages</span> ──────────────────────── tidyverse 2.0.0 ──
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">dplyr    </span> 1.1.4     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">readr    </span> 2.1.5
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">forcats  </span> 1.0.0     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">stringr  </span> 1.5.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">ggplot2  </span> 3.5.2     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tibble   </span> 3.3.0
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">lubridate</span> 1.9.4     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tidyr    </span> 1.3.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">purrr    </span> 1.1.0     
── <span class=" -Color -Color-Bold">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">between()</span>     masks <span class=" -Color -Color-Blue">data.table</span>::between()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">filter()</span>      masks <span class=" -Color -Color-Blue">stats</span>::filter()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">first()</span>       masks <span class=" -Color -Color-Blue">data.table</span>::first()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">hour()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::hour()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">isoweek()</span> masks <span class=" -Color -Color-Blue">data.table</span>::isoweek()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">lag()</span>         masks <span class=" -Color -Color-Blue">stats</span>::lag()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">last()</span>        masks <span class=" -Color -Color-Blue">data.table</span>::last()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">mday()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::mday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">minute()</span>  masks <span class=" -Color -Color-Blue">data.table</span>::minute()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">month()</span>   masks <span class=" -Color -Color-Blue">data.table</span>::month()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">quarter()</span> masks <span class=" -Color -Color-Blue">data.table</span>::quarter()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">second()</span>  masks <span class=" -Color -Color-Blue">data.table</span>::second()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">purrr</span>::<span class=" -Color -Color-Green">transpose()</span>   masks <span class=" -Color -Color-Blue">data.table</span>::transpose()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">wday()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::wday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">week()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::week()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">yday()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::yday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">year()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::year()
<span class=" -Color -Color-Cyan">ℹ</span> Use the conflicted package (<span class=" -Color -Color-Blue">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">fmp_paths</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&#39;/home/fengr1/fungen_xQTL/&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;toploci.bed.gz$&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<section id="generate-variant-list-from-feature-set">
<h3>generate variant list from feature set<a class="headerlink" href="#generate-variant-list-from-feature-set" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># var_paths &lt;- list.files(&#39;/home/fengr1/v2f_data/xqtl_features/xqtl_features_v2f&#39;, &#39;var_list&#39;, full.names = T) # the intermediate files from last step</span>
<span class="c1"># var_paths %&gt;% length</span>

<span class="c1"># var_list &lt;- data.frame()</span>
<span class="c1"># for(var_path in var_paths){</span>
<span class="c1">#     var_list &lt;- rbind(var_list, fread(var_path))</span>
<span class="c1"># }</span>
<span class="c1"># fwrite(var_list, &#39;/home/fengr1/v2f_data/xqtl_features/xqtl_features_v2f/variant_list.txt.gz&#39;, sep = &#39;\t&#39;)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">var_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/home/fengr1/v2f_data/xqtl_features/xqtl_features_v2f/variant_list.txt.gz&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">var_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 × 2</caption>
<thead>
	<tr><th scope=col>snp</th><th scope=col>rsid</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr1:10000044 </td><td>rs112409036</td></tr>
	<tr><td>chr1:100000723</td><td>rs11166389 </td></tr>
	<tr><td>chr1:100000839</td><td>rs116740877</td></tr>
	<tr><td>chr1:100000874</td><td>rs550182761</td></tr>
	<tr><td>chr1:100000874</td><td>rs549210756</td></tr>
	<tr><td>chr1:100000874</td><td>rs796619774</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">var_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>16194875</li><li>2</li></ol>
</div></div>
</div>
</section>
<section id="positive-set">
<h3>positive set<a class="headerlink" href="#positive-set" title="Link to this heading">#</a></h3>
<section id="define-parameter-sets">
<h4>Define parameter sets<a class="headerlink" href="#define-parameter-sets" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">param_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span>
<span class="w">  </span><span class="n">strict</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">PIP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.9</span><span class="p">,</span><span class="w"> </span><span class="n">VCP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.9</span><span class="p">,</span><span class="w"> </span><span class="n">CS_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">CoS_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">),</span>
<span class="w">  </span><span class="n">intermediate</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">PIP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w"> </span><span class="n">VCP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.8</span><span class="p">,</span><span class="w"> </span><span class="n">CS_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">CoS_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">),</span>
<span class="w">  </span><span class="n">loose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">PIP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">,</span><span class="w"> </span><span class="n">VCP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">,</span><span class="w"> </span><span class="n">CS_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">CoS_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">),</span>
<span class="w">  </span><span class="n">superloose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">PIP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">,</span><span class="w"> </span><span class="n">VCP</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">,</span><span class="w"> </span><span class="n">CS_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">CoS_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>

<span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">fmp_paths</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&#39;~/ftp_fgc_xqtl/fungen_xQTL/&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;toploci.bed.gz$&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">### File paths</span>
<span class="n">AD_fmp_paths</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">file_paths</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">.</span><span class="p">[</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;GWAS&#39;</span><span class="p">)]</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">.</span><span class="p">[</span><span class="o">!</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;_Jansen|_EADI&#39;</span><span class="p">)]</span>
<span class="n">AD_cb_paths</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;/data/deyk/ftp_fgc_xqtl/fungen_xQTL/colocboost/ad_meta_colocboost_export_filtered.bed&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">### Function: process fine-mapping</span>
<span class="n">process_finemapping</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">paths</span><span class="p">,</span><span class="w"> </span><span class="n">pip_threshold</span><span class="p">,</span><span class="w"> </span><span class="n">cs_size_threshold</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">fmp_positive</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">()</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">path</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">paths</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">path</span><span class="p">)</span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">all</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;gene_ID&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;event_ID&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">dat</span><span class="p">)))</span><span class="w"> </span><span class="kr">next</span>
<span class="w">    </span><span class="n">dat_cs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dat</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">filter</span><span class="p">(</span><span class="n">cs_coverage_0.95</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">mutate</span><span class="p">(</span><span class="n">cs_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">cs_coverage_0.95</span><span class="p">))</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">dat_map</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dat_cs</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">group_by</span><span class="p">(</span><span class="n">cs_id</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">summarise</span><span class="p">(</span><span class="n">cs_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">n_distinct</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">),</span><span class="w"> </span><span class="n">.groups</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;drop&quot;</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="c1"># Only filter cs_size if threshold &gt; 0</span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">cs_size_threshold</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">dat_map</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dat_map</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">cs_size</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="n">cs_size_threshold</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">dat_map</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dat_map</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">left_join</span><span class="p">(</span><span class="n">dat_cs</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;cs_id&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">mutate</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Mappable&#39;</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">dat_pip</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dat</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">PIP</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">pip_threshold</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;PIP&#39;</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">pos</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">dat_pip</span><span class="p">,</span>
<span class="w">                 </span><span class="n">dat_map</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">cs_id</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">cs_size</span><span class="p">))</span>
<span class="w">    </span><span class="n">fmp_positive</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plyr</span><span class="o">::</span><span class="nf">rbind.fill</span><span class="p">(</span><span class="n">fmp_positive</span><span class="p">,</span><span class="w"> </span><span class="n">pos</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Positive set from fine-mapping mappable sets:&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">      </span><span class="n">fmp_positive</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;Mappable&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Positive set from fine-mapping high PIP:&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">      </span><span class="n">fmp_positive</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;PIP&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="kr">return</span><span class="p">(</span><span class="n">fmp_positive</span><span class="p">)</span>
<span class="p">}</span>


<span class="c1">### Function: process colocboost</span>
<span class="n">process_coloc</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">paths</span><span class="p">,</span><span class="w"> </span><span class="n">vcp_threshold</span><span class="p">,</span><span class="w"> </span><span class="n">cos_size_threshold</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">cb_positive</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">()</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">path</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">paths</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">path</span><span class="p">)</span>
<span class="w">    </span><span class="n">dat_map</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dat</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">group_by</span><span class="p">(</span><span class="n">cos_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">summarise</span><span class="p">(</span><span class="n">cos_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">n_distinct</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">),</span><span class="w"> </span><span class="n">.groups</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;drop&quot;</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="c1"># Only filter cos_size if threshold &gt; 0</span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">cos_size_threshold</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">dat_map</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dat_map</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">cos_size</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="n">cos_size_threshold</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">dat_map</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dat_map</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">left_join</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;cos_ID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">mutate</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Mappable&#39;</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">dat_vcp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dat</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">vcp</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">vcp_threshold</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;VCP&#39;</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">pos</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">dat_vcp</span><span class="p">,</span>
<span class="w">                 </span><span class="n">dat_map</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">cos_size</span><span class="p">))</span>
<span class="w">    </span><span class="n">cb_positive</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plyr</span><span class="o">::</span><span class="nf">rbind.fill</span><span class="p">(</span><span class="n">cb_positive</span><span class="p">,</span><span class="w"> </span><span class="n">pos</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">    </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Positive set from colocboost mappable sets:&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">        </span><span class="n">cb_positive</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;Mappable&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Positive set from colocboost high VCP:&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">        </span><span class="n">cb_positive</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;VCP&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span><span class="kr">return</span><span class="p">(</span><span class="n">cb_positive</span><span class="p">)</span>
<span class="p">}</span>


<span class="c1">### Function: extract non-negative</span>
<span class="n">extract_non_negative</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">fmp_paths</span><span class="p">,</span><span class="w"> </span><span class="n">cb_paths</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">nonneg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">()</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">path</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">fmp_paths</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">path</span><span class="p">)</span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;PIP&quot;</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">dat</span><span class="p">))</span>
<span class="w">      </span><span class="n">nonneg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plyr</span><span class="o">::</span><span class="nf">rbind.fill</span><span class="p">(</span><span class="n">nonneg</span><span class="p">,</span><span class="w"> </span><span class="n">dat</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">PIP</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">0.01</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Fine-mapping&#39;</span><span class="p">))</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;NON-Negative set from fine-maping:&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">      </span><span class="n">nonneg</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;Fine-mapping&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">path</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">cb_paths</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">path</span><span class="p">)</span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="s">&quot;vcp&quot;</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">dat</span><span class="p">))</span>
<span class="w">      </span><span class="n">nonneg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plyr</span><span class="o">::</span><span class="nf">rbind.fill</span><span class="p">(</span><span class="n">nonneg</span><span class="p">,</span><span class="w"> </span><span class="n">dat</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">vcp</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">0.01</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;colocboost&#39;</span><span class="p">))</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;NON-Negative set from colocboost:&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">      </span><span class="n">nonneg</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">category</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;colocboost&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="kr">return</span><span class="p">(</span><span class="n">nonneg</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">### Run all parameter sets</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">param_list</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">param</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">param_list</span><span class="p">[[</span><span class="n">label</span><span class="p">]]</span>
<span class="w">  </span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;Processing: Postive fine-mapping PIP as &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">PIP</span><span class="p">,</span><span class="w"> </span><span class="s">&quot; &amp; mappble cs size as &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">CS_size</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;; ColocBoost VCP as &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">VCP</span><span class="p">,</span><span class="w"> </span><span class="s">&quot; &amp; mappable cos size as &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">CoS_size</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="n">status_name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;PIP&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">PIP</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_CS&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">CS_size</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_VCP&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">VCP</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_COS&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">CoS_size</span><span class="p">)</span>
<span class="w">    </span>
<span class="w">  </span><span class="c1"># Positive set</span>
<span class="w">  </span><span class="n">fmp_positive</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">process_finemapping</span><span class="p">(</span><span class="n">AD_fmp_paths</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">PIP</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">CS_size</span><span class="p">)</span>
<span class="w">  </span><span class="n">cb_positive</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">process_coloc</span><span class="p">(</span><span class="n">AD_cb_paths</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">VCP</span><span class="p">,</span><span class="w"> </span><span class="n">param</span><span class="o">$</span><span class="n">CoS_size</span><span class="p">)</span>
<span class="w">  </span><span class="n">positive</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plyr</span><span class="o">::</span><span class="nf">rbind.fill</span><span class="p">(</span><span class="n">fmp_positive</span><span class="p">,</span><span class="w"> </span><span class="n">cb_positive</span><span class="p">)</span>
<span class="w">  </span><span class="nf">fwrite</span><span class="p">(</span><span class="n">positive</span><span class="p">,</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;~/Work/v2f/results/posneg_process/positive.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">status_name</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.df.csv.gz&#39;</span><span class="p">))</span>

<span class="w">  </span><span class="c1"># RSID for positive</span>
<span class="w">  </span><span class="n">positive_rs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">positive</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">snp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">`#chr`</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">end</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">merge</span><span class="p">(</span><span class="n">var_list</span><span class="p">)</span>
<span class="w">  </span><span class="nf">writeLines</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">positive_rs</span><span class="o">$</span><span class="n">rsid</span><span class="p">),</span>
<span class="w">             </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;~/Work/v2f/results/posneg_process/positive_set.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">status_name</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt&#39;</span><span class="p">))</span>

<span class="w">  </span><span class="c1"># Non-negative: includes positive + PIP ≥ 0.01 or VCP ≥ 0.01</span>
<span class="w">  </span><span class="n">non_negative</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">extract_non_negative</span><span class="p">(</span><span class="n">AD_fmp_paths</span><span class="p">,</span><span class="w"> </span><span class="n">AD_cb_paths</span><span class="p">)</span>
<span class="w">  </span><span class="n">non_negative</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plyr</span><span class="o">::</span><span class="nf">rbind.fill</span><span class="p">(</span><span class="n">non_negative</span><span class="p">,</span><span class="w"> </span><span class="n">positive</span><span class="p">)</span><span class="w"> </span><span class="c1"># actually not necessary that the non_negative must contain positive set by PIP or VCP</span>
<span class="w">  </span><span class="nf">fwrite</span><span class="p">(</span><span class="n">non_negative</span><span class="p">,</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;~/Work/v2f/results/posneg_process/non_negative.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">status_name</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.df.csv.gz&#39;</span><span class="p">))</span>

<span class="w">  </span><span class="c1"># RSID for non-negative</span>
<span class="w">  </span><span class="n">non_negative_rs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">non_negative</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">snp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">`#chr`</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">end</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">merge</span><span class="p">(</span><span class="n">var_list</span><span class="p">)</span>

<span class="w">  </span><span class="c1"># Negative set</span>
<span class="w">  </span><span class="n">negative_rs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">setdiff</span><span class="p">(</span><span class="n">var_list</span><span class="o">$</span><span class="n">rsid</span><span class="p">,</span><span class="w"> </span><span class="n">non_negative_rs</span><span class="o">$</span><span class="n">rsid</span><span class="p">)</span>
<span class="w">  </span><span class="nf">writeLines</span><span class="p">(</span><span class="n">negative_rs</span><span class="p">,</span>
<span class="w">             </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;~/Work/v2f/results/posneg_process/negative_set.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">status_name</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt&#39;</span><span class="p">))</span>
<span class="w">    </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;=== Summary for&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">label</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;set ===\n&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Positive set (variants):&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">positive</span><span class="o">$</span><span class="n">variant_ID</span><span class="p">)),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Positive set (intersected with feature set):**&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">positive_rs</span><span class="o">$</span><span class="n">rsid</span><span class="p">)),</span><span class="w"> </span><span class="s">&quot;**\n&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;NON-Negative set (finemapping + colocboost + positive):&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">non_negative</span><span class="o">$</span><span class="n">variant_ID</span><span class="p">)),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;NON-Negative set (intersected with feature set):&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">non_negative_rs</span><span class="o">$</span><span class="n">rsid</span><span class="p">)),</span><span class="w"> </span><span class="s">&quot;\n&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="nf">cat</span><span class="p">(</span><span class="s">&quot;Negative set (intersected with feature set):**&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">negative_rs</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;**\n\n&quot;</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Processing: Postive fine-mapping PIP as 0.9 &amp; mappble cs size as 3; ColocBoost VCP as 0.9 &amp; mappable cos size as 3
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Positive set from fine-mapping mappable sets: 57 
Positive set from fine-mapping high PIP: 28 
Positive set from colocboost mappable sets: 45 
Positive set from colocboost high VCP: 17 
NON-Negative set from fine-maping: 3490 
NON-Negative set from colocboost: 880 
=== Summary for strict set ===
Positive set (variants): 79 
Positive set (intersected with feature set):** 52 **
NON-Negative set (finemapping + colocboost + positive): 3633 
NON-Negative set (intersected with feature set): 3389 
Negative set (intersected with feature set):** 16191486 **
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Processing: Postive fine-mapping PIP as 0.7 &amp; mappble cs size as 10; ColocBoost VCP as 0.8 &amp; mappable cos size as 10
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Positive set from fine-mapping mappable sets: 200 
Positive set from fine-mapping high PIP: 48 
Positive set from colocboost mappable sets: 146 
Positive set from colocboost high VCP: 20 
NON-Negative set from fine-maping: 3490 
NON-Negative set from colocboost: 880 
=== Summary for intermediate set ===
Positive set (variants): 277 
Positive set (intersected with feature set):** 212 **
NON-Negative set (finemapping + colocboost + positive): 3643 
NON-Negative set (intersected with feature set): 3389 
Negative set (intersected with feature set):** 16191486 **
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Processing: Postive fine-mapping PIP as 0.6 &amp; mappble cs size as 10; ColocBoost VCP as 0.6 &amp; mappable cos size as 10
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Positive set from fine-mapping mappable sets: 200 
Positive set from fine-mapping high PIP: 59 
Positive set from colocboost mappable sets: 146 
Positive set from colocboost high VCP: 28 
NON-Negative set from fine-maping: 3490 
NON-Negative set from colocboost: 880 
=== Summary for loose set ===
Positive set (variants): 284 
Positive set (intersected with feature set):** 219 **
NON-Negative set (finemapping + colocboost + positive): 3643 
NON-Negative set (intersected with feature set): 3389 
Negative set (intersected with feature set):** 16191486 **
</pre></div>
</div>
</div>
</div>
<p>Additionally check the set of super loose set only, which is new and generated in R studio;</p>
<p>copy and paste the output information from Rstudio, so I don’t have to cover the exsiting information above</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Processing</span><span class="p">:</span> <span class="n">Positive</span> <span class="n">fine</span><span class="o">-</span><span class="n">mapping</span> <span class="n">PIP</span> <span class="k">as</span> <span class="mf">0.7</span> <span class="o">&amp;</span> <span class="n">mappable</span> <span class="n">cs</span> <span class="n">size</span> <span class="k">as</span> <span class="mi">0</span><span class="p">;</span> <span class="n">ColocBoost</span> <span class="n">VCP</span> <span class="k">as</span> <span class="mf">0.8</span> <span class="o">&amp;</span> <span class="n">mappable</span> <span class="n">cos</span> <span class="n">size</span> <span class="k">as</span> <span class="mi">0</span>

<span class="n">Positive</span> <span class="nb">set</span> <span class="kn">from</span><span class="w"> </span><span class="nn">fine</span><span class="o">-</span><span class="n">mapping</span> <span class="n">mappable</span> <span class="n">sets</span><span class="p">:</span> <span class="mi">1817</span> 
<span class="n">Positive</span> <span class="nb">set</span> <span class="kn">from</span><span class="w"> </span><span class="nn">fine</span><span class="o">-</span><span class="n">mapping</span> <span class="n">high</span> <span class="n">PIP</span><span class="p">:</span> <span class="mi">48</span> 
<span class="n">Positive</span> <span class="nb">set</span> <span class="kn">from</span><span class="w"> </span><span class="nn">colocboost</span> <span class="n">mappable</span> <span class="n">sets</span><span class="p">:</span> <span class="mi">5212</span> 
<span class="n">Positive</span> <span class="nb">set</span> <span class="kn">from</span><span class="w"> </span><span class="nn">colocboost</span> <span class="n">high</span> <span class="n">VCP</span><span class="p">:</span> <span class="mi">20</span> 
<span class="n">NON</span><span class="o">-</span><span class="n">Negative</span> <span class="nb">set</span> <span class="kn">from</span><span class="w"> </span><span class="nn">fine</span><span class="o">-</span><span class="n">maping</span><span class="p">:</span> <span class="mi">3490</span> 
<span class="n">NON</span><span class="o">-</span><span class="n">Negative</span> <span class="nb">set</span> <span class="kn">from</span><span class="w"> </span><span class="nn">colocboost</span><span class="p">:</span> <span class="mi">880</span> 
<span class="o">===</span> <span class="n">Summary</span> <span class="k">for</span> <span class="n">intermediate</span> <span class="nb">set</span> <span class="o">===</span>
<span class="n">Positive</span> <span class="nb">set</span> <span class="p">(</span><span class="n">variants</span><span class="p">):</span> <span class="mi">5798</span> 
<span class="n">Positive</span> <span class="nb">set</span> <span class="p">(</span><span class="n">intersected</span> <span class="k">with</span> <span class="n">feature</span> <span class="nb">set</span><span class="p">):</span><span class="o">**</span> <span class="mi">1787</span> <span class="o">**</span>
<span class="n">NON</span><span class="o">-</span><span class="n">Negative</span> <span class="nb">set</span> <span class="p">(</span><span class="n">finemapping</span> <span class="o">+</span> <span class="n">colocboost</span> <span class="o">+</span> <span class="n">positive</span><span class="p">):</span> <span class="mi">7755</span> 
<span class="n">NON</span><span class="o">-</span><span class="n">Negative</span> <span class="nb">set</span> <span class="p">(</span><span class="n">intersected</span> <span class="k">with</span> <span class="n">feature</span> <span class="nb">set</span><span class="p">):</span> <span class="mi">4150</span> 
<span class="n">Negative</span> <span class="nb">set</span> <span class="p">(</span><span class="n">intersected</span> <span class="k">with</span> <span class="n">feature</span> <span class="nb">set</span><span class="p">):</span><span class="o">**</span> <span class="mi">16190725</span> <span class="o">**</span>
</pre></div>
</div>
</section>
</section>
</section>
<section id="submit-jobs-for-cv2f-analysis">
<h2>3. Submit Jobs for cv2f Analysis<a class="headerlink" href="#submit-jobs-for-cv2f-analysis" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#!/bin/bash</span>

<span class="c1"># Parameter sets</span>
<span class="n">param_sets</span><span class="o">=</span><span class="p">(</span>
<span class="w">  </span><span class="s">&quot;PIP0.7_CS0_VCP0.8_COS0&quot;</span>
<span class="w">  </span><span class="s">&quot;PIP0.7_CS10_VCP0.8_COS10&quot;</span>
<span class="w">  </span><span class="s">&quot;PIP0.9_CS3_VCP0.9_COS3&quot;</span>
<span class="w">  </span><span class="s">&quot;PIP0.6_CS10_VCP0.6_COS10&quot;</span>
<span class="p">)</span>

<span class="c1"># Feature sets</span>
<span class="n">feature_sets</span><span class="o">=</span><span class="p">(</span>
<span class="w">  </span><span class="s">&quot;/lila/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f/xqtl_feature_max_allcv2f.&quot;</span>
<span class="p">)</span>

<span class="c1"># Shared input files</span>
<span class="n">bimpath</span><span class="o">=</span><span class="s">&quot;/home/fengr1/deyk/ftp_fgc_xqtl/v2f/1000G_BIMS_hg38/1000G.EUR.QC.&quot;</span>
<span class="n">mafpath</span><span class="o">=</span><span class="s">&quot;/data/deyk/extras/MAF_features_Aug032022.txt&quot;</span>
<span class="n">ldblockspath</span><span class="o">=</span><span class="s">&quot;/home/fengr1/deyk/ftp_fgc_xqtl/v2f/LAVA_LDblocks_published.txt&quot;</span>

<span class="c1"># Create directories</span>
<span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">score</span><span class="o">/</span><span class="n">FunGen_xQTL</span>
<span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">metric</span><span class="o">/</span><span class="n">FunGen_xQTL</span>
<span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">logs</span>

<span class="kr">for</span><span class="w"> </span><span class="n">status</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="s">&quot;${param_sets[@]}&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">do</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="n">feat</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="s">&quot;${feature_sets[@]}&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">do</span>
<span class="w">    </span><span class="n">feat_label</span><span class="o">=$</span><span class="p">(</span><span class="n">basename</span><span class="w"> </span><span class="o">$</span><span class="p">{</span><span class="n">feat</span>%<span class="n">.</span><span class="o">*</span><span class="p">})</span>
<span class="w">    </span><span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">score</span><span class="o">/</span><span class="n">FunGen_xQTL</span><span class="o">/$</span><span class="p">{</span><span class="n">status</span><span class="p">}</span><span class="o">/$</span><span class="p">{</span><span class="n">feat_label</span><span class="p">}</span>
<span class="w">    </span><span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">metric</span><span class="o">/</span><span class="n">FunGen_xQTL</span><span class="o">/$</span><span class="p">{</span><span class="n">status</span><span class="p">}</span><span class="o">/$</span><span class="p">{</span><span class="n">feat_label</span><span class="p">}</span>

<span class="w">    </span><span class="n">positive_set</span><span class="o">=</span><span class="s">&quot;/lila/data/deyk/ru/Work/v2f/results/posneg_process/positive_set.${status}.txt&quot;</span>
<span class="w">    </span><span class="n">negative_set</span><span class="o">=</span><span class="s">&quot;/lila/data/deyk/ru/Work/v2f/results/posneg_process/negative_set.${status}.txt&quot;</span>
<span class="w">    </span><span class="n">output_metrics</span><span class="o">=</span><span class="s">&quot;/home/fengr1/Work/v2f/results/metric/FunGen_xQTL/${status}/${feat_label}/FunGen_xQTL.${status}.${feat_label}.metrics&quot;</span>

<span class="w">    </span><span class="n">echo</span><span class="w"> </span><span class="s">&quot;&gt;&gt;&gt; Running metrics: $status with $feat_label&quot;</span>
<span class="w">    </span><span class="n">Rscript</span><span class="w"> </span><span class="o">/</span><span class="n">lila</span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">codes</span><span class="o">/</span><span class="n">cv2f</span><span class="o">/</span><span class="n">metric</span><span class="o">/</span><span class="n">run_cv2f_MAF5LDmatch_metrics.R</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">positive_set</span><span class="w"> </span><span class="o">$</span><span class="n">positive_set</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">negative_set</span><span class="w"> </span><span class="o">$</span><span class="n">negative_set</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">feature_file</span><span class="w"> </span><span class="o">$</span><span class="n">feat</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">bimpath</span><span class="w"> </span><span class="o">$</span><span class="n">bimpath</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">mafpath</span><span class="w"> </span><span class="o">$</span><span class="n">mafpath</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">ldblockspath</span><span class="w"> </span><span class="o">$</span><span class="n">ldblockspath</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">output_metrics</span><span class="w"> </span><span class="o">$</span><span class="n">output_metrics</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">logs</span><span class="o">/</span><span class="n">xqtl_metric.</span><span class="o">$</span><span class="p">{</span><span class="n">status</span><span class="p">}</span><span class="n">.</span><span class="o">$</span><span class="p">{</span><span class="n">feat_label</span><span class="p">}</span><span class="n">.log</span><span class="w"> </span><span class="m">2</span><span class="o">&gt;&amp;</span><span class="m">1</span>

<span class="w">    </span><span class="kr">for</span><span class="w"> </span><span class="n">chrm</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="p">{</span><span class="m">1</span><span class="kc">..22</span><span class="p">};</span><span class="w"> </span><span class="n">do</span>
<span class="w">      </span><span class="n">output_cv2f</span><span class="o">=</span><span class="s">&quot;/home/fengr1/Work/v2f/results/score/FunGen_xQTL/${status}/${feat_label}/FunGen_xQTL.${status}.${feat_label}.chr${chrm}.cv2f&quot;</span>

<span class="w">      </span><span class="n">echo</span><span class="w"> </span><span class="s">&quot;&gt;&gt;&gt; Running score: $status, $feat_label, chr${chrm}&quot;</span>
<span class="w">      </span><span class="n">Rscript</span><span class="w"> </span><span class="o">/</span><span class="n">lila</span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">codes</span><span class="o">/</span><span class="n">cv2f</span><span class="o">/</span><span class="n">score</span><span class="o">/</span><span class="n">run_cv2f_MAF5LDmatch_cv2f_scores.R</span><span class="w"> </span>\
<span class="w">        </span><span class="o">--</span><span class="n">positive_set</span><span class="w"> </span><span class="o">$</span><span class="n">positive_set</span><span class="w"> </span>\
<span class="w">        </span><span class="o">--</span><span class="n">negative_set</span><span class="w"> </span><span class="o">$</span><span class="n">negative_set</span><span class="w"> </span>\
<span class="w">        </span><span class="o">--</span><span class="n">feature_file</span><span class="w"> </span><span class="o">$</span><span class="n">feat</span><span class="w"> </span>\
<span class="w">        </span><span class="o">--</span><span class="n">bimpath</span><span class="w"> </span><span class="o">$</span><span class="n">bimpath</span><span class="w"> </span>\
<span class="w">        </span><span class="o">--</span><span class="n">mafpath</span><span class="w"> </span><span class="o">$</span><span class="n">mafpath</span><span class="w"> </span>\
<span class="w">        </span><span class="o">--</span><span class="n">ldblockspath</span><span class="w"> </span><span class="o">$</span><span class="n">ldblockspath</span><span class="w"> </span>\
<span class="w">        </span><span class="o">--</span><span class="n">output_cv2f</span><span class="w"> </span><span class="o">$</span><span class="n">output_cv2f</span><span class="w"> </span>\
<span class="w">        </span><span class="o">--</span><span class="n">chrm</span><span class="w"> </span><span class="o">$</span><span class="n">chrm</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">logs</span><span class="o">/</span><span class="n">xqtl_score.</span><span class="o">$</span><span class="p">{</span><span class="n">status</span><span class="p">}</span><span class="n">.</span><span class="o">$</span><span class="p">{</span><span class="n">feat_label</span><span class="p">}</span><span class="n">.chr</span><span class="o">$</span><span class="p">{</span><span class="n">chrm</span><span class="p">}</span><span class="n">.log</span><span class="w"> </span><span class="m">2</span><span class="o">&gt;&amp;</span><span class="m">1</span>
<span class="w">    </span><span class="n">done</span>
<span class="w">  </span><span class="n">done</span>
<span class="n">done</span>
</pre></div>
</div>
</div>
</div>
<section id="check-the-time-stamp-of-input-files">
<h3>Check the time stamp of input files<a class="headerlink" href="#check-the-time-stamp-of-input-files" title="Link to this heading">#</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="n">positive_set</span><span class="o">.*</span> <span class="o">-</span><span class="n">hl</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mf">2.3</span><span class="n">K</span> <span class="n">Jul</span> <span class="mi">25</span> <span class="mi">15</span><span class="p">:</span><span class="mi">41</span> <span class="n">positive_set</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.6</span><span class="n">_CS10_VCP0</span><span class="mf">.6</span><span class="n">_COS10</span><span class="o">.</span><span class="n">txt</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span>  <span class="mi">19</span><span class="n">K</span> <span class="n">Jul</span> <span class="mi">29</span> <span class="mi">10</span><span class="p">:</span><span class="mi">26</span> <span class="n">positive_set</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.7</span><span class="n">_CS0_VCP0</span><span class="mf">.8</span><span class="n">_COS0</span><span class="o">.</span><span class="n">txt</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mf">2.3</span><span class="n">K</span> <span class="n">Jul</span> <span class="mi">25</span> <span class="mi">15</span><span class="p">:</span><span class="mi">41</span> <span class="n">positive_set</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.7</span><span class="n">_CS10_VCP0</span><span class="mf">.8</span><span class="n">_COS10</span><span class="o">.</span><span class="n">txt</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span>  <span class="mi">560</span> <span class="n">Jul</span> <span class="mi">25</span> <span class="mi">15</span><span class="p">:</span><span class="mi">40</span> <span class="n">positive_set</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.9</span><span class="n">_CS3_VCP0</span><span class="mf">.9</span><span class="n">_COS3</span><span class="o">.</span><span class="n">txt</span>

<span class="n">ls</span> <span class="n">negative_set</span><span class="o">.*</span> <span class="o">-</span><span class="n">hl</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mi">174</span><span class="n">M</span> <span class="n">Jul</span> <span class="mi">25</span> <span class="mi">15</span><span class="p">:</span><span class="mi">41</span> <span class="n">negative_set</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.6</span><span class="n">_CS10_VCP0</span><span class="mf">.6</span><span class="n">_COS10</span><span class="o">.</span><span class="n">txt</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mi">174</span><span class="n">M</span> <span class="n">Jul</span> <span class="mi">29</span> <span class="mi">10</span><span class="p">:</span><span class="mi">27</span> <span class="n">negative_set</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.7</span><span class="n">_CS0_VCP0</span><span class="mf">.8</span><span class="n">_COS0</span><span class="o">.</span><span class="n">txt</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mi">174</span><span class="n">M</span> <span class="n">Jul</span> <span class="mi">25</span> <span class="mi">15</span><span class="p">:</span><span class="mi">41</span> <span class="n">negative_set</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.7</span><span class="n">_CS10_VCP0</span><span class="mf">.8</span><span class="n">_COS10</span><span class="o">.</span><span class="n">txt</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mi">174</span><span class="n">M</span> <span class="n">Jul</span> <span class="mi">25</span> <span class="mi">15</span><span class="p">:</span><span class="mi">41</span> <span class="n">negative_set</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.9</span><span class="n">_CS3_VCP0</span><span class="mf">.9</span><span class="n">_COS3</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">-</span><span class="n">hl</span> <span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">deyk</span><span class="o">/</span><span class="n">ftp_fgc_xqtl</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">xqtl_features</span><span class="o">/</span><span class="n">xqtl_features_v2f</span><span class="o">/</span><span class="n">agg_snp_max_allcv2f</span><span class="o">/*</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mi">51</span><span class="n">M</span> <span class="n">Jul</span> <span class="mi">28</span> <span class="mi">21</span><span class="p">:</span><span class="mi">20</span> <span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">deyk</span><span class="o">/</span><span class="n">ftp_fgc_xqtl</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">xqtl_features</span><span class="o">/</span><span class="n">xqtl_features_v2f</span><span class="o">/</span><span class="n">agg_snp_max_allcv2f</span><span class="o">/</span><span class="n">xqtl_feature_max_allcv2f</span><span class="mf">.10</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mi">50</span><span class="n">M</span> <span class="n">Jul</span> <span class="mi">28</span> <span class="mi">21</span><span class="p">:</span><span class="mi">20</span> <span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">deyk</span><span class="o">/</span><span class="n">ftp_fgc_xqtl</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">xqtl_features</span><span class="o">/</span><span class="n">xqtl_features_v2f</span><span class="o">/</span><span class="n">agg_snp_max_allcv2f</span><span class="o">/</span><span class="n">xqtl_feature_max_allcv2f</span><span class="mf">.11</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mi">50</span><span class="n">M</span> <span class="n">Jul</span> <span class="mi">28</span> <span class="mi">21</span><span class="p">:</span><span class="mi">21</span> <span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">deyk</span><span class="o">/</span><span class="n">ftp_fgc_xqtl</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">xqtl_features</span><span class="o">/</span><span class="n">xqtl_features_v2f</span><span class="o">/</span><span class="n">agg_snp_max_allcv2f</span><span class="o">/</span><span class="n">xqtl_feature_max_allcv2f</span><span class="mf">.12</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mi">33</span><span class="n">M</span> <span class="n">Jul</span> <span class="mi">28</span> <span class="mi">21</span><span class="p">:</span><span class="mi">21</span> <span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">deyk</span><span class="o">/</span><span class="n">ftp_fgc_xqtl</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">xqtl_features</span><span class="o">/</span><span class="n">xqtl_features_v2f</span><span class="o">/</span><span class="n">agg_snp_max_allcv2f</span><span class="o">/</span><span class="n">xqtl_feature_max_allcv2f</span><span class="mf">.13</span><span class="o">.</span><span class="n">txt</span><span class="o">.</span><span class="n">gz</span>

</pre></div>
</div>
</section>
</section>
<section id="visualization-of-results">
<h2>4. Visualization of Results<a class="headerlink" href="#visualization-of-results" title="Link to this heading">#</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">*/</span><span class="n">xqtl_feature_max_cv2f</span><span class="o">/*</span><span class="n">metrics</span> <span class="o">-</span><span class="n">hl</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mi">116</span> <span class="n">Jul</span> <span class="mi">29</span> <span class="mi">17</span><span class="p">:</span><span class="mi">01</span> <span class="n">PIP0</span><span class="mf">.7</span><span class="n">_CS0_VCP0</span><span class="mf">.8</span><span class="n">_COS0</span><span class="o">/</span><span class="n">xqtl_feature_max_cv2f</span><span class="o">/</span><span class="n">FunGen_xQTL</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.7</span><span class="n">_CS0_VCP0</span><span class="mf">.8</span><span class="n">_COS0</span><span class="o">.</span><span class="n">xqtl_feature_max_allcv2f</span><span class="o">.</span><span class="n">metrics</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span> <span class="mi">113</span> <span class="n">Jul</span> <span class="mi">29</span> <span class="mi">15</span><span class="p">:</span><span class="mi">10</span> <span class="n">PIP0</span><span class="mf">.7</span><span class="n">_CS10_VCP0</span><span class="mf">.8</span><span class="n">_COS10</span><span class="o">/</span><span class="n">xqtl_feature_max_cv2f</span><span class="o">/</span><span class="n">FunGen_xQTL</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.7</span><span class="n">_CS10_VCP0</span><span class="mf">.8</span><span class="n">_COS10</span><span class="o">.</span><span class="n">xqtl_feature_max_allcv2f</span><span class="o">.</span><span class="n">metrics</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">fengr1</span> <span class="n">deyk</span>  <span class="mi">52</span> <span class="n">Jul</span> <span class="mi">28</span> <span class="mi">23</span><span class="p">:</span><span class="mi">27</span> <span class="n">PIP0</span><span class="mf">.9</span><span class="n">_CS3_VCP0</span><span class="mf">.9</span><span class="n">_COS3</span><span class="o">/</span><span class="n">xqtl_feature_max_cv2f</span><span class="o">/</span><span class="n">FunGen_xQTL</span><span class="o">.</span><span class="n">PIP0</span><span class="mf">.9</span><span class="n">_CS3_VCP0</span><span class="mf">.9</span><span class="n">_COS3</span><span class="o">.</span><span class="n">xqtl_feature_max_allcv2f</span><span class="o">.</span><span class="n">metrics</span>
</pre></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>

<span class="c1"># Define statuses (adjust to match your naming scheme)</span>
<span class="n">status</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;PIP0.7_CS0_VCP0.8_COS0&quot;</span><span class="p">)</span>

<span class="c1"># Output folder for plots</span>
<span class="n">plot_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_cv2f/metric&quot;</span><span class="w"> </span><span class="c1">#FIXME</span>
<span class="nf">dir.create</span><span class="p">(</span><span class="n">plot_dir</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="n">metric_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">figure_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span>

<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">status</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">statuses</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;Processing status: &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Load metrics</span>
<span class="w">  </span><span class="n">metric_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">plot_dir</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;/xqtl_feature_max_allcv2f/FunGen_xQTL.&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.xqtl_feature_max_allcv2f.metrics&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="w">  </span><span class="n">shap_file</span><span class="w">   </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">plot_dir</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;/xqtl_feature_max_allcv2f/FunGen_xQTL.&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.xqtl_feature_max_allcv2f.shap_values.csv&quot;</span><span class="p">)</span><span class="w">   </span><span class="c1">#FIXME</span>
<span class="w">  </span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">file.exists</span><span class="p">(</span><span class="n">metric_file</span><span class="p">)</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="o">!</span><span class="nf">file.exists</span><span class="p">(</span><span class="n">shap_file</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">warning</span><span class="p">(</span><span class="s">&quot;Missing file(s) for status: &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">)</span>
<span class="w">    </span><span class="kr">next</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span>
<span class="w">  </span><span class="n">metric_list</span><span class="p">[[</span><span class="n">status</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">metric_file</span><span class="p">)</span>
<span class="w">  </span><span class="n">shap_dt</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">shap_file</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Aggregate SHAP values</span>
<span class="w">  </span><span class="n">shap_summary</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">shap_dt</span><span class="p">[,</span><span class="w"> </span><span class="nf">.</span><span class="p">(</span>
<span class="w">    </span><span class="n">mean_abs_rfvalue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">mean</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">rfvalue</span><span class="p">)),</span>
<span class="w">    </span><span class="n">max_rfvalue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">rfvalue</span><span class="p">)),</span>
<span class="w">    </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">.N</span>
<span class="w">  </span><span class="p">),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">variable</span><span class="p">]</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="n">mean_abs_rfvalue</span><span class="p">))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Plot: Top 20 Features</span>
<span class="w">  </span><span class="n">figure_list</span><span class="p">[[</span><span class="n">status</span><span class="p">]]</span><span class="o">$</span><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">shap_summary</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">],</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">reorder</span><span class="p">(</span><span class="n">variable</span><span class="p">,</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">),</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_col</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">coord_flip</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Top 20 Features by Mean |SHAP| -&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">),</span>
<span class="w">         </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Feature&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mean |SHAP| Value&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme_minimal</span><span class="p">()</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># ggsave(filename = paste0(plot_dir, &quot;Top20_SHAP_&quot;, status, &quot;.png&quot;), plot = p1, width = 10, height = 6)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Plot: Top 20 Non-dominant Non-sQTL Features</span>
<span class="w">  </span><span class="n">figure_list</span><span class="p">[[</span><span class="n">status</span><span class="p">]]</span><span class="o">$</span><span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">shap_summary</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">slice</span><span class="p">(</span><span class="o">-</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">variable</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sQTL&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">slice_max</span><span class="p">(</span><span class="n">order_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">reorder</span><span class="p">(</span><span class="n">variable</span><span class="p">,</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">),</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_col</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">coord_flip</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Top 20 Non-dominant Non-sQTL Features -&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">),</span>
<span class="w">         </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Feature&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mean |SHAP| Value&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme_minimal</span><span class="p">()</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># ggsave(filename = paste0(plot_dir, &quot;Top20_NonSQTLSHAP_&quot;, status, &quot;.png&quot;), plot = p2, width = 10, height = 6)</span>

<span class="p">}</span>
<span class="w">     </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Processing status: PIP0.7_CS0_VCP0.8_COS0
</pre></div>
</div>
</div>
</div>
<section id="pip0-7-cs0-vcp0-8-cos0">
<h3>PIP0.7_CS0_VCP0.8_COS0<a class="headerlink" href="#pip0-7-cs0-vcp0-8-cos0" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">metric_list</span><span class="o">$</span><span class="n">PIP0.7_CS0_VCP0.8_COS0</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 1 × 4</caption>
<thead>
	<tr><th scope=col>AUROC.mean</th><th scope=col>AUROC.sd</th><th scope=col>AUPRC.mean</th><th scope=col>AUPRC.sd</th></tr>
	<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>0.9181583</td><td>0.01473063</td><td>0.9244869</td><td>0.01402415</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">figure_list</span><span class="o">$</span><span class="n">PIP0.7_CS0_VCP0.8_COS0</span><span class="o">$</span><span class="n">p1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/6a5164a1389843cde3bd46b4598123546be6d97a14557c8e1b39fb3b6628c8a3.png"><img alt="../../_images/6a5164a1389843cde3bd46b4598123546be6d97a14557c8e1b39fb3b6628c8a3.png" src="../../_images/6a5164a1389843cde3bd46b4598123546be6d97a14557c8e1b39fb3b6628c8a3.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">png</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">grid</span><span class="p">)</span>

<span class="n">img</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readPNG</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_cv2f/metric/xqtl_feature_max_allcv2f/FunGen_xQTL.PIP0.7_CS0_VCP0.8_COS0.xqtl_feature_max_allcv2f.png&quot;</span><span class="p">)</span><span class="w"> </span><span class="c1">#FIXME</span>
<span class="nf">grid.raster</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/659d0d0acccd6293e430efa362cedd5e2fb62ada964dab19613a41b33d3a9d88.png"><img alt="../../_images/659d0d0acccd6293e430efa362cedd5e2fb62ada964dab19613a41b33d3a9d88.png" src="../../_images/659d0d0acccd6293e430efa362cedd5e2fb62ada964dab19613a41b33d3a9d88.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
</section>
<section id="optional-experiment">
<h2>(Optional) Experiment<a class="headerlink" href="#optional-experiment" title="Link to this heading">#</a></h2>
<section id="additional-experiment-suggestion-from-gao">
<h3>Additional Experiment (Suggestion from Gao)<a class="headerlink" href="#additional-experiment-suggestion-from-gao" title="Link to this heading">#</a></h3>
<p>As suggested by Gao, a valuable experiment to strengthen our findings is to assess model performance when excluding all xQTL features.<br />
Specifically, we would train the model on the same positive and negative sets described above<br />
(using <strong>PIP &gt; 0.7 with no CS size restriction</strong> and <strong>VCP &gt; 0.8 with no CoS size restriction</strong>).</p>
<p>In this setting, the feature set would consist only of Fabiha’s annotations, without any xQTL-derived features.<br />
This approach mirrors strategies used in prior studies, where sources of annotation were systematically removed to evaluate their individual contributions.</p>
<p>We anticipate a similar outcome here: removing xQTL features is expected to substantially reduce the AUC compared to the full model,<br />
thereby highlighting the unique and critical contribution of xQTL information.</p>
<p>experiment by taking out xQTL information from feature sets to check AUC</p>
</section>
<section id="prepare-new-feature-sets-without-xqtl">
<h3>prepare new feature sets without xQTL<a class="headerlink" href="#prepare-new-feature-sets-without-xqtl" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&#39;~/deyk/ru/Work/v2f/analysis&#39;</span><span class="p">)</span>
<span class="c1"># cv2f_files &lt;- list.files(&quot;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/cv2f_features/cv2f_features/&quot;, full.names = T)</span>
<span class="c1"># agg_xqtl_files &lt;- list.files(&quot;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max/&quot;, full.names = T)</span>


<span class="c1">#### Main loop #### </span>

<span class="kr">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">22</span><span class="p">){</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
<span class="w">  </span><span class="n">cv2f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/cv2f_features/cv2f_features/baseline_annotation_chrmbpnet.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt&#39;</span><span class="p">))</span>
<span class="w">  </span><span class="n">s2g</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max//xqtl_feature_max.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt.gz&#39;</span><span class="p">))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">#### add cv2f feature #### </span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Extract cv2f features (skip first 4 columns)</span>
<span class="w">  </span><span class="n">cv2f_features</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">cv2f</span><span class="p">)[</span><span class="m">5</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">cv2f</span><span class="p">)]</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Create snp column</span>
<span class="w">  </span><span class="n">cv2f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cv2f</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">snp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">CHR</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">BP</span><span class="p">))</span>
<span class="w">  </span><span class="c1"># Merge s2g with cv2f features</span>
<span class="w">  </span><span class="n">s2g</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">s2g</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">snp</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">CHR</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;:&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">BP</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">left_join</span><span class="p">(</span><span class="n">cv2f</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="nf">all_of</span><span class="p">(</span><span class="n">cv2f_features</span><span class="p">),</span><span class="w"> </span><span class="n">snp</span><span class="p">),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;snp&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="nf">across</span><span class="p">(</span><span class="nf">all_of</span><span class="p">(</span><span class="n">cv2f_features</span><span class="p">),</span><span class="w"> </span><span class="o">~</span><span class="nf">replace_na</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">snp</span><span class="p">)</span>
<span class="w">  </span><span class="n">s2g_xqtl_elements</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">s2g</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">.</span><span class="p">[</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;QTL|qtl|Metabrain|single_context|multi_context&#39;</span><span class="p">)]</span>
<span class="w">  </span><span class="c1"># remove all xqtl feature</span>
<span class="w">  </span><span class="n">s2g_noxqtl</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">s2g</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="nf">any_of</span><span class="p">(</span><span class="n">s2g_xqtl_elements</span><span class="p">))</span><span class="w"> </span>
<span class="w">  </span><span class="c1"># s2g_noxqtl %&gt;% mutate(snp=paste0(&#39;chr&#39;, CHR, &#39;:&#39;, BP), rsid=SNP) %&gt;% select(snp, rsid) %&gt;% </span>
<span class="w">  </span><span class="c1">#   fwrite(paste0(&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/agg_snp_max_allcv2f_noxqtl/var_list.&#39;, i, &#39;.txt&#39;), sep = &#39;\t&#39;)</span>
<span class="w">  </span><span class="nf">fwrite</span><span class="p">(</span><span class="n">s2g_noxqtl</span><span class="p">,</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f_noxqtl/xqtl_feature_max_allcv2f_noxqtl.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt.gz&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
<span class="w">   </span><span class="c1">## the difference with v2f/xqtl_features/no_xqtl_features_v2f is this does not remain the variants in cv2f only and that one does</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>── <span class=" -Color -Color-Bold">Attaching core tidyverse packages</span> ──────────────────────── tidyverse 2.0.0 ──
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">dplyr    </span> 1.1.4     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">readr    </span> 2.1.5
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">forcats  </span> 1.0.0     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">stringr  </span> 1.5.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">ggplot2  </span> 3.5.2     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tibble   </span> 3.3.0
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">lubridate</span> 1.9.4     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tidyr    </span> 1.3.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">purrr    </span> 1.1.0     
── <span class=" -Color -Color-Bold">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">filter()</span> masks <span class=" -Color -Color-Blue">stats</span>::filter()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">lag()</span>    masks <span class=" -Color -Color-Blue">stats</span>::lag()
<span class=" -Color -Color-Cyan">ℹ</span> Use the conflicted package (<span class=" -Color -Color-Blue">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors

Attaching package: ‘data.table’


The following objects are masked from ‘package:lubridate’:

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year


The following objects are masked from ‘package:dplyr’:

    between, first, last


The following object is masked from ‘package:purrr’:

    transpose
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in left_join(., cv2f %&gt;% select(all_of(cv2f_features), snp), by = &quot;snp&quot;):
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 213138 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 631086 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>23
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in left_join(., cv2f %&gt;% select(all_of(cv2f_features), snp), by = &quot;snp&quot;):
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 769626 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 36549 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>456
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in left_join(., cv2f %&gt;% select(all_of(cv2f_features), snp), by = &quot;snp&quot;):
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 305543 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 229009 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>7
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in left_join(., cv2f %&gt;% select(all_of(cv2f_features), snp), by = &quot;snp&quot;):
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 198030 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 328203 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>8
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in left_join(., cv2f %&gt;% select(all_of(cv2f_features), snp), by = &quot;snp&quot;):
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 477267 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 383828 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>9
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in left_join(., cv2f %&gt;% select(all_of(cv2f_features), snp), by = &quot;snp&quot;):
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 162469 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 113111 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>10
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in left_join(., cv2f %&gt;% select(all_of(cv2f_features), snp), by = &quot;snp&quot;):
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 183338 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 150368 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>11
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in left_join(., cv2f %&gt;% select(all_of(cv2f_features), snp), by = &quot;snp&quot;):
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 164173 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 142465 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1213141516171819
</pre></div>
</div>
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message in left_join(., cv2f %&gt;% select(all_of(cv2f_features), snp), by = &quot;snp&quot;):
“Detected an unexpected many-to-many relationship between `x` and `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 8572 of `x` matches multiple rows in `y`.
<span class=" -Color -Color-Cyan">ℹ</span> Row 87830 of `y` matches multiple rows in `x`.
<span class=" -Color -Color-Cyan">ℹ</span> If a many-to-many relationship is expected, set `relationship =
  &quot;many-to-many&quot;` to silence this warning.”
</pre></div>
</div>
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>202122
</pre></div>
</div>
</div>
</div>
<section id="check-new-feature-sets">
<h4>check new feature sets<a class="headerlink" href="#check-new-feature-sets" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f_noxqtl/xqtl_feature_max_allcv2f_noxqtl.22.txt.gz&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="n">Error</span> <span class="ow">in</span> <span class="n">fread</span><span class="p">(</span><span class="s2">&quot;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f_noxqtl/xqtl_feature_max_allcv2f_noxqtl.22.txt.gz&quot;</span><span class="p">):</span> <span class="n">could</span> <span class="ow">not</span> <span class="n">find</span> <span class="n">function</span> <span class="s2">&quot;fread&quot;</span>
<span class="ne">Traceback</span>:
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_tmp</span><span class="o">$</span><span class="n">exon</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_tmp</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">colnames</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'CHR'</li><li>'BP'</li><li>'SNP'</li><li>'exon'</li><li>'s_in_g'</li><li>'same_tad'</li><li>'proximal_tad'</li><li>'g_closest_tss'</li><li>'EpiRaction.EPIraction_Brain_Alzheimer_Control_thresholded_predictions_0.026'</li><li>'ABC_dpfc'</li><li>'EpiMap'</li><li>'Baseline_Predss.dist_to_tss'</li><li>'Baseline_Predss.nearest_expressed_gene'</li><li>'EpiRaction.EPIraction_Brain_Alzheimer_Disease_thresholded_predictions_0.026'</li><li>'EpiRaction.EPIraction_Brain_Alzheimer_Encode_thresholded_predictions_0.026'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_BLD'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_BRN'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_GM12878'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_GUT'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_HRT'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_HepG2'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_K562'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_KID'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_LIV'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_LNG'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_SKIN'</li><li>'ABC_ENCODE_E2G_2022.ABC_DNaseH3K27ac_UNION'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_A549'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_GM12878'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_HepG2'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_K562'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_SK-N-SH'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_all'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_blood'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_brain'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_fat'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_gut'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_heart'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_kidney'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_liver'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_lung'</li><li>'ADASTRA2021_BillCipher.ADASTRA_UseAvgOnNonTested_skin'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_A549'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_Blood'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_Brain'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_Colon'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_GM12878'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_Heart'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_K562'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_Kidney'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_Liver'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_Lung'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_SK-N-DZ'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_Skin'</li><li>'Altius_2024_DNase.annotations.rsids_FDR10_all'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_A549'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_Blood'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_Brain'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_Colon'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_Fat pad'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_GM12878'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_Heart'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_HepG2'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_K562'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_Kidney'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_Liver'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_Lung'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_SK-N-DZ'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_Skin'</li><li>'Altius_2024_DNase.annotations.rsids_Nominal_all'</li><li>'Ancient_Sequence_Age_Human_Enhancer'</li><li>'Ancient_Sequence_Age_Human_Enhancer.flanking.500'</li><li>'Ancient_Sequence_Age_Human_Promoter'</li><li>'Ancient_Sequence_Age_Human_Promoter.flanking.500'</li><li>'BLUEPRINT_DNA_methylation_MaxCPP'</li><li>'Backgrd_Selection_Stat'</li><li>'BivFlnk'</li><li>'BivFlnk.flanking.500'</li><li>'CTCF_Hoffman'</li><li>'CTCF_Hoffman.flanking.500'</li><li>'Coding_UCSC'</li><li>'Coding_UCSC.flanking.500'</li><li>'Conserved_LindbladToh'</li><li>'Conserved_LindbladToh.flanking.500'</li><li>'Conserved_Mammal_phastCons46way'</li><li>'Conserved_Mammal_phastCons46way.flanking.500'</li><li>'Conserved_Primate_phastCons46way'</li><li>'Conserved_Primate_phastCons46way.flanking.500'</li><li>'Conserved_Vertebrate_phastCons46way'</li><li>'Conserved_Vertebrate_phastCons46way.flanking.500'</li><li>'CpG_Content_50kb'</li><li>'DGF_ENCODE'</li><li>'DGF_ENCODE.flanking.500'</li><li>'DHS_Trynka'</li><li>'DHS_Trynka.flanking.500'</li><li>'DHS_peaks_Trynka'</li><li>'Enformer.Enformer_A549_CHIP'</li><li>'Enformer.Enformer_A549_all'</li><li>'Enformer.Enformer_GM12878_CHIP'</li><li>'Enformer.Enformer_GM12878_all'</li><li>'Enformer.Enformer_HepG2_CHIP'</li><li>'Enformer.Enformer_HepG2_all'</li><li>'Enformer.Enformer_K562_CHIP'</li><li>'Enformer.Enformer_K562_all'</li><li>'Enformer.Enformer_SKNSH_CHIP'</li><li>'Enformer.Enformer_SKNSH_all'</li><li>'Enformer.Enformer_all_CAGE'</li><li>'Enformer.Enformer_all_CHIP'</li><li>'Enformer.Enformer_all_DNASE'</li><li>'Enformer.Enformer_all_H3K27ac'</li><li>'Enformer.Enformer_all_H3K4me1'</li><li>'Enformer.Enformer_all_H3K4me3'</li><li>'Enformer.Enformer_all_all'</li><li>'Enformer.Enformer_blood_CAGE'</li><li>'Enformer.Enformer_blood_CHIP'</li><li>'Enformer.Enformer_blood_DNASE'</li><li>'Enformer.Enformer_blood_H3K27ac'</li><li>'Enformer.Enformer_blood_H3K4me1'</li><li>'Enformer.Enformer_blood_H3K4me3'</li><li>'Enformer.Enformer_blood_all'</li><li>'Enformer.Enformer_brain_CAGE'</li><li>'Enformer.Enformer_brain_CHIP'</li><li>'Enformer.Enformer_brain_DNASE'</li><li>'Enformer.Enformer_brain_H3K27ac'</li><li>'Enformer.Enformer_brain_H3K4me1'</li><li>'Enformer.Enformer_brain_H3K4me3'</li><li>'Enformer.Enformer_brain_all'</li><li>'Enformer.Enformer_fat_CAGE'</li><li>'Enformer.Enformer_fat_CHIP'</li><li>'Enformer.Enformer_fat_all'</li><li>'Enformer.Enformer_gut_CAGE'</li><li>'Enformer.Enformer_gut_CHIP'</li><li>'Enformer.Enformer_gut_DNASE'</li><li>'Enformer.Enformer_gut_H3K27ac'</li><li>'Enformer.Enformer_gut_H3K4me1'</li><li>'Enformer.Enformer_gut_H3K4me3'</li><li>'Enformer.Enformer_gut_all'</li><li>'Enformer.Enformer_heart_CHIP'</li><li>'Enformer.Enformer_heart_DNASE'</li><li>'Enformer.Enformer_heart_H3K4me3'</li><li>'Enformer.Enformer_heart_all'</li><li>'Enformer.Enformer_kidney_CAGE'</li><li>'Enformer.Enformer_kidney_CHIP'</li><li>'Enformer.Enformer_kidney_DNASE'</li><li>'Enformer.Enformer_kidney_all'</li><li>'Enformer.Enformer_liver_CHIP'</li><li>'Enformer.Enformer_liver_all'</li><li>'Enformer.Enformer_lung_CAGE'</li><li>'Enformer.Enformer_lung_CHIP'</li><li>'Enformer.Enformer_lung_DNASE'</li><li>'Enformer.Enformer_lung_H3K4me3'</li><li>'Enformer.Enformer_lung_all'</li><li>'Enformer.Enformer_skin_CHIP'</li><li>'Enformer.Enformer_skin_DNASE'</li><li>'Enformer.Enformer_skin_H3K4me3'</li><li>'Enformer.Enformer_skin_all'</li><li>'Enhancer_Andersson'</li><li>'Enhancer_Andersson.flanking.500'</li><li>'Enhancer_Hoffman'</li><li>'Enhancer_Hoffman.flanking.500'</li><li>'FetalDHS_Trynka'</li><li>'FetalDHS_Trynka.flanking.500'</li><li>'GERP.NS'</li><li>'GERP.RSsup4'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_all'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_blood'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_brain'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_fat'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_gut'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_heart'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_kidney'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_liver'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_lung'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_10_skin'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_all'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_blood'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_brain'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_fat'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_gut'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_heart'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_kidney'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_liver'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_lung'</li><li>'GTEx_Flagship.GTEx_Finemap_PIP_50_skin'</li><li>'H3K27ac_Hnisz'</li><li>'H3K27ac_Hnisz.flanking.500'</li><li>'H3K27ac_PGC2'</li><li>'H3K27ac_PGC2.flanking.500'</li><li>'H3K4me1_Trynka'</li><li>'H3K4me1_Trynka.flanking.500'</li><li>'H3K4me1_peaks_Trynka'</li><li>'H3K4me3_Trynka'</li><li>'H3K4me3_Trynka.flanking.500'</li><li>'H3K4me3_peaks_Trynka'</li><li>'H3K9ac_Trynka'</li><li>'H3K9ac_Trynka.flanking.500'</li><li>'H3K9ac_peaks_Trynka'</li><li>'Hongkai_conserve.cob_H3K27ac_fdr.rsid'</li><li>'Hongkai_conserve.cob_H3K4me1_fdr.rsid'</li><li>'Hongkai_conserve.cob_H3K4me3_fdr.rsid'</li><li>'Hongkai_conserve.cob_all_fdr.rsid'</li><li>'Hongkai_conserve.cob_chromatin_accessibility_fdr.rsid'</li><li>'Hongkai_conserve.cov_H3K27ac_fdr.rsid'</li><li>'Hongkai_conserve.cov_H3K4me1_fdr.rsid'</li><li>'Hongkai_conserve.cov_H3K4me3_fdr.rsid'</li><li>'Hongkai_conserve.cov_all_fdr.rsid'</li><li>'Hongkai_conserve.cov_chromatin_accessibility_fdr.rsid'</li><li>'Hongkai_conserve.cov_cob_H3K27ac_fdr.rsid'</li><li>'Hongkai_conserve.cov_cob_H3K4me1_fdr.rsid'</li><li>'Hongkai_conserve.cov_cob_H3K4me3_fdr.rsid'</li><li>'Hongkai_conserve.cov_cob_all_fdr.rsid'</li><li>'Hongkai_conserve.cov_cob_chromatin_accessibility_fdr.rsid'</li><li>'Human_Enhancer_Villar'</li><li>'Human_Enhancer_Villar.flanking.500'</li><li>'Human_Enhancer_Villar_Species_Enhancer_Count'</li><li>'Human_Promoter_Villar'</li><li>'Human_Promoter_Villar.flanking.500'</li><li>'Human_Promoter_Villar_ExAC'</li><li>'Human_Promoter_Villar_ExAC.flanking.500'</li><li>'Intron_UCSC'</li><li>'Intron_UCSC.flanking.500'</li><li>'MPRA_2023_Ryan.GTEx.annotations.rsids_FDR10_HepG2'</li><li>'MPRA_2023_Ryan.GTEx.annotations.rsids_FDR10_K562'</li><li>'MPRA_2023_Ryan.GTEx.annotations.rsids_FDR10_SKNSH'</li><li>'MPRA_2023_Ryan.GTEx.annotations.rsids_all'</li><li>'MPRA_RyanTewhey_deeplearn_2023.All.skew.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.All.skew.bin.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.All.skew.refalt.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.HepG2.skew.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.HepG2.skew.bin.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.HepG2.skew.refalt.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.K562.skew.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.K562.skew.bin.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.K562.skew.refalt.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.SKNSH.skew.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.SKNSH.skew.bin.MPRAdeep'</li><li>'MPRA_RyanTewhey_deeplearn_2023.SKNSH.skew.refalt.MPRAdeep'</li><li>'Nucleotide_Diversity_10kb'</li><li>'PromoterFlanking_Hoffman'</li><li>'PromoterFlanking_Hoffman.flanking.500'</li><li>'Promoter_UCSC'</li><li>'Promoter_UCSC.flanking.500'</li><li>'Recomb_Rate_10kb'</li><li>'Repressed_Hoffman'</li><li>'Repressed_Hoffman.flanking.500'</li><li>'SuperEnhancer_Hnisz'</li><li>'SuperEnhancer_Hnisz.flanking.500'</li><li>'TFBS_ENCODE'</li><li>'TFBS_ENCODE.flanking.500'</li><li>'TSS_Hoffman'</li><li>'TSS_Hoffman.flanking.500'</li><li>'Transcr_Hoffman'</li><li>'Transcr_Hoffman.flanking.500'</li><li>'UTR_3_UCSC'</li><li>'UTR_3_UCSC.flanking.500'</li><li>'UTR_5_UCSC'</li><li>'UTR_5_UCSC.flanking.500'</li><li>'WGSTARR.rsids_sigs_AFR_STARR_75'</li><li>'WGSTARR.rsids_sigs_AFR_STARR_90'</li><li>'WeakEnhancer_Hoffman'</li><li>'WeakEnhancer_Hoffman.flanking.500'</li><li>'Zoonomia_Conserve.Zoonomia_Mammals_PhyloP_v1'</li><li>'Zoonomia_Conserve.Zoonomia_Primate_PhastCons_v1'</li><li>'base'</li><li>'cCREv4_CellType_Agnostic.ALL'</li><li>'cCREv4_CellType_Agnostic.PLS'</li><li>'cCREv4_CellType_Agnostic.dELS'</li><li>'cCREv4_CellType_Agnostic.pELS'</li><li>'cCREv4_Human_CellType.A549_PLS'</li><li>'cCREv4_Human_CellType.BLD_PLS'</li><li>'cCREv4_Human_CellType.BLD_dELS'</li><li>'cCREv4_Human_CellType.BLD_pELS'</li><li>'cCREv4_Human_CellType.BRN_PLS'</li><li>'cCREv4_Human_CellType.BRN_dELS'</li><li>'cCREv4_Human_CellType.BRN_pELS'</li><li>'cCREv4_Human_CellType.GM12878_PLS'</li><li>'cCREv4_Human_CellType.GM12878_dELS'</li><li>'cCREv4_Human_CellType.GM12878_pELS'</li><li>'cCREv4_Human_CellType.GUT_PLS'</li><li>'cCREv4_Human_CellType.GUT_dELS'</li><li>'cCREv4_Human_CellType.GUT_pELS'</li><li>'cCREv4_Human_CellType.HRT_PLS'</li><li>'cCREv4_Human_CellType.HRT_dELS'</li><li>'cCREv4_Human_CellType.HRT_pELS'</li><li>'cCREv4_Human_CellType.HepG2_PLS'</li><li>'cCREv4_Human_CellType.HepG2_dELS'</li><li>'cCREv4_Human_CellType.HepG2_pELS'</li><li>'cCREv4_Human_CellType.K562_PLS'</li><li>'cCREv4_Human_CellType.K562_dELS'</li><li>'cCREv4_Human_CellType.K562_pELS'</li><li>'cCREv4_Human_CellType.KID_PLS'</li><li>'cCREv4_Human_CellType.LIV_ALL'</li><li>'cCREv4_Human_CellType.LIV_PLS'</li><li>'cCREv4_Human_CellType.LIV_dELS'</li><li>'cCREv4_Human_CellType.LIV_pELS'</li><li>'cCREv4_Human_CellType.LNG_PLS'</li><li>'cCREv4_Human_CellType.LNG_dELS'</li><li>'cCREv4_Human_CellType.LNG_pELS'</li><li>'cCREv4_Human_CellType.SK-N-SH_PLS'</li><li>'cCREv4_Human_CellType.SK-N-SH_dELS'</li><li>'cCREv4_Human_CellType.SK-N-SH_pELS'</li><li>'cCREv4_Human_CellType.SKIN_PLS'</li><li>'cCREv4_Human_CellType.SKIN_dELS'</li><li>'cCREv4_Human_CellType.SKIN_pELS'</li><li>'chrmbpnet_GM12878_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_GM12878_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_GM12878_jsd_x_max_percentile'</li><li>'chrmbpnet_HepG2_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_HepG2_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_HepG2_jsd_x_max_percentile'</li><li>'chrmbpnet_K562_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_K562_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_K562_jsd_x_max_percentile'</li><li>'chrmbpnet_SKNSH_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_SKNSH_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_SKNSH_jsd_x_max_percentile'</li><li>'chrmbpnet_all_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_all_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_all_jsd_x_max_percentile'</li><li>'chrmbpnet_blood_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_blood_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_blood_jsd_x_max_percentile'</li><li>'chrmbpnet_brain_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_brain_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_brain_jsd_x_max_percentile'</li><li>'chrmbpnet_gut_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_gut_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_gut_jsd_x_max_percentile'</li><li>'chrmbpnet_heart_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_heart_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_heart_jsd_x_max_percentile'</li><li>'chrmbpnet_liver_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_liver_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_liver_jsd_x_max_percentile'</li><li>'chrmbpnet_lung_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_lung_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_lung_jsd_x_max_percentile'</li><li>'chrmbpnet_skin_abs_logfc_x_jsd_x_max_percentile'</li><li>'chrmbpnet_skin_abs_logfc_x_max_percentile'</li><li>'chrmbpnet_skin_jsd_x_max_percentile'</li><li>'non_synonymous'</li><li>'synonymous'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_tmp</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span><span class="w"> </span><span class="c1">#chrom 22</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>236695</li><li>342</li></ol>
</div></div>
</div>
</section>
</section>
<section id="submit-jobs">
<h3>submit jobs<a class="headerlink" href="#submit-jobs" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#!/bin/bash</span>

<span class="c1"># Parameter sets</span>
<span class="n">param_sets</span><span class="o">=</span><span class="p">(</span>
<span class="w">  </span><span class="s">&quot;PIP0.7_CS0_VCP0.8_COS0&quot;</span>
<span class="p">)</span>

<span class="c1"># Feature sets</span>
<span class="n">feature_sets</span><span class="o">=</span><span class="p">(</span>
<span class="w">  </span><span class="s">&quot;/lila/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f_noxqtl/xqtl_feature_max_allcv2f_noxqtl.&quot;</span>
<span class="p">)</span>

<span class="c1"># Shared input files</span>
<span class="n">bimpath</span><span class="o">=</span><span class="s">&quot;/home/fengr1/deyk/ftp_fgc_xqtl/v2f/1000G_BIMS_hg38/1000G.EUR.QC.&quot;</span>
<span class="n">mafpath</span><span class="o">=</span><span class="s">&quot;/data/deyk/extras/MAF_features_Aug032022.txt&quot;</span>
<span class="n">ldblockspath</span><span class="o">=</span><span class="s">&quot;/home/fengr1/deyk/ftp_fgc_xqtl/v2f/LAVA_LDblocks_published.txt&quot;</span>

<span class="c1"># Create directories</span>
<span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">score</span><span class="o">/</span><span class="n">FunGen_xQTL</span>
<span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">metric</span><span class="o">/</span><span class="n">FunGen_xQTL</span>
<span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">logs</span>

<span class="kr">for</span><span class="w"> </span><span class="n">status</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="s">&quot;${param_sets[@]}&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">do</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="n">feat</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="s">&quot;${feature_sets[@]}&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">do</span>
<span class="w">    </span><span class="n">feat_label</span><span class="o">=$</span><span class="p">(</span><span class="n">basename</span><span class="w"> </span><span class="o">$</span><span class="p">{</span><span class="n">feat</span>%<span class="n">.</span><span class="o">*</span><span class="p">})</span>
<span class="w">    </span><span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">score</span><span class="o">/</span><span class="n">FunGen_xQTL</span><span class="o">/$</span><span class="p">{</span><span class="n">status</span><span class="p">}</span><span class="o">/$</span><span class="p">{</span><span class="n">feat_label</span><span class="p">}</span>
<span class="w">    </span><span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">metric</span><span class="o">/</span><span class="n">FunGen_xQTL</span><span class="o">/$</span><span class="p">{</span><span class="n">status</span><span class="p">}</span><span class="o">/$</span><span class="p">{</span><span class="n">feat_label</span><span class="p">}</span>

<span class="w">    </span><span class="n">positive_set</span><span class="o">=</span><span class="s">&quot;/lila/data/deyk/ru/Work/v2f/results/posneg_process/positive_set.${status}.txt&quot;</span>
<span class="w">    </span><span class="n">negative_set</span><span class="o">=</span><span class="s">&quot;/lila/data/deyk/ru/Work/v2f/results/posneg_process/negative_set.${status}.txt&quot;</span>
<span class="w">    </span><span class="n">output_metrics</span><span class="o">=</span><span class="s">&quot;/home/fengr1/Work/v2f/results/metric/FunGen_xQTL/${status}/${feat_label}/FunGen_xQTL.${status}.${feat_label}.metrics&quot;</span>

<span class="w">    </span><span class="n">echo</span><span class="w"> </span><span class="s">&quot;&gt;&gt;&gt; Running metrics: $status with $feat_label&quot;</span>
<span class="w">    </span><span class="n">Rscript</span><span class="w"> </span><span class="o">/</span><span class="n">lila</span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">codes</span><span class="o">/</span><span class="n">cv2f</span><span class="o">/</span><span class="n">metric</span><span class="o">/</span><span class="n">run_cv2f_MAF5LDmatch_metrics.R</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">positive_set</span><span class="w"> </span><span class="o">$</span><span class="n">positive_set</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">negative_set</span><span class="w"> </span><span class="o">$</span><span class="n">negative_set</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">feature_file</span><span class="w"> </span><span class="o">$</span><span class="n">feat</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">bimpath</span><span class="w"> </span><span class="o">$</span><span class="n">bimpath</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">mafpath</span><span class="w"> </span><span class="o">$</span><span class="n">mafpath</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">ldblockspath</span><span class="w"> </span><span class="o">$</span><span class="n">ldblockspath</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">output_metrics</span><span class="w"> </span><span class="o">$</span><span class="n">output_metrics</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">logs</span><span class="o">/</span><span class="n">xqtl_metric.</span><span class="o">$</span><span class="p">{</span><span class="n">status</span><span class="p">}</span><span class="n">.</span><span class="o">$</span><span class="p">{</span><span class="n">feat_label</span><span class="p">}</span><span class="n">.log</span><span class="w"> </span><span class="m">2</span><span class="o">&gt;&amp;</span><span class="m">1</span>

<span class="c1">#     for chrm in {1..22}; do</span>
<span class="c1">#       output_cv2f=&quot;/home/fengr1/Work/v2f/results/score/FunGen_xQTL/${status}/${feat_label}/FunGen_xQTL.GWASonly_response.${status}.${feat_label}.chr${chrm}.cv2f&quot;</span>

<span class="c1">#       echo &quot;&gt;&gt;&gt; Running score: $status, $feat_label, chr${chrm}&quot;</span>
<span class="c1">#       Rscript /lila/home/fengr1/codes/cv2f/score/run_cv2f_MAF5LDmatch_cv2f_scores.R \</span>
<span class="c1">#         --positive_set $positive_set \</span>
<span class="c1">#         --negative_set $negative_set \</span>
<span class="c1">#         --feature_file $feat \</span>
<span class="c1">#         --bimpath $bimpath \</span>
<span class="c1">#         --mafpath $mafpath \</span>
<span class="c1">#         --ldblockspath $ldblockspath \</span>
<span class="c1">#         --output_cv2f $output_cv2f \</span>
<span class="c1">#         --chrm $chrm &gt; /home/fengr1/Work/v2f/logs/xqtl_score.${status}.${feat_label}.chr${chrm}.log 2&gt;&amp;1</span>
<span class="c1">#     done</span>
<span class="w">  </span><span class="n">done</span>
<span class="n">done</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="id1">
<h3>Visualization of Results<a class="headerlink" href="#id1" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>

<span class="c1"># Define statuses (adjust to match your naming scheme)</span>
<span class="n">statuses</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;PIP0.7_CS0_VCP0.8_COS0&quot;</span><span class="p">)</span>

<span class="c1"># Output folder for plots</span>
<span class="n">plot_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_cv2f/metric&quot;</span><span class="w"> </span><span class="c1">#FIXME</span>
<span class="nf">dir.create</span><span class="p">(</span><span class="n">plot_dir</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="n">metric_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">figure_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span>

<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">status</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">statuses</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;Processing status: &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Load metrics</span>
<span class="w">  </span><span class="n">metric_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">plot_dir</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;/xqtl_feature_max_allcv2f_noxqtl/FunGen_xQTL.&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.xqtl_feature_max_allcv2f_noxqtl.metrics&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="w">  </span><span class="n">shap_file</span><span class="w">   </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">plot_dir</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;/xqtl_feature_max_allcv2f_noxqtl/FunGen_xQTL.&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.xqtl_feature_max_allcv2f_noxqtl.shap_values.csv&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="w">  </span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">file.exists</span><span class="p">(</span><span class="n">metric_file</span><span class="p">)</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="o">!</span><span class="nf">file.exists</span><span class="p">(</span><span class="n">shap_file</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">warning</span><span class="p">(</span><span class="s">&quot;Missing file(s) for status: &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">)</span>
<span class="w">    </span><span class="kr">next</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span>
<span class="w">  </span><span class="n">metric_list</span><span class="p">[[</span><span class="n">status</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">metric_file</span><span class="p">)</span>
<span class="w">  </span><span class="n">shap_dt</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">shap_file</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Aggregate SHAP values</span>
<span class="w">  </span><span class="n">shap_summary</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">shap_dt</span><span class="p">[,</span><span class="w"> </span><span class="nf">.</span><span class="p">(</span>
<span class="w">    </span><span class="n">mean_abs_rfvalue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">mean</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">rfvalue</span><span class="p">)),</span>
<span class="w">    </span><span class="n">max_rfvalue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">rfvalue</span><span class="p">)),</span>
<span class="w">    </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">.N</span>
<span class="w">  </span><span class="p">),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">variable</span><span class="p">]</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="n">mean_abs_rfvalue</span><span class="p">))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Plot: Top 20 Features</span>
<span class="w">  </span><span class="n">figure_list</span><span class="p">[[</span><span class="n">status</span><span class="p">]]</span><span class="o">$</span><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">shap_summary</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">],</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">reorder</span><span class="p">(</span><span class="n">variable</span><span class="p">,</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">),</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_col</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">coord_flip</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Top 20 Features by Mean |SHAP| -&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">),</span>
<span class="w">         </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Feature&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mean |SHAP| Value&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme_minimal</span><span class="p">()</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># ggsave(filename = paste0(plot_dir, &quot;Top20_SHAP_&quot;, status, &quot;.png&quot;), plot = p1, width = 10, height = 6)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Plot: Top 20 Non-dominant Non-sQTL Features</span>
<span class="w">  </span><span class="n">figure_list</span><span class="p">[[</span><span class="n">status</span><span class="p">]]</span><span class="o">$</span><span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">shap_summary</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">slice</span><span class="p">(</span><span class="o">-</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">variable</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sQTL&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">slice_max</span><span class="p">(</span><span class="n">order_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">reorder</span><span class="p">(</span><span class="n">variable</span><span class="p">,</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">),</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_col</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">coord_flip</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Top 20 Non-dominant Non-sQTL Features -&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">),</span>
<span class="w">         </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Feature&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mean |SHAP| Value&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme_minimal</span><span class="p">()</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># ggsave(filename = paste0(plot_dir, &quot;Top20_NonSQTLSHAP_&quot;, status, &quot;.png&quot;), plot = p2, width = 10, height = 6)</span>

<span class="p">}</span>
<span class="w">     </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Processing status: PIP0.7_CS0_VCP0.8_COS0
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">metric_list</span><span class="o">$</span><span class="n">PIP0.7_CS0_VCP0.8_COS0</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 1 × 4</caption>
<thead>
	<tr><th scope=col>AUROC.mean</th><th scope=col>AUROC.sd</th><th scope=col>AUPRC.mean</th><th scope=col>AUPRC.sd</th></tr>
	<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>0.855358</td><td>0.002755411</td><td>0.8496754</td><td>0.00733925</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">figure_list</span><span class="o">$</span><span class="n">PIP0.7_CS0_VCP0.8_COS0</span><span class="o">$</span><span class="n">p1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/d876f8fb50ba9b1e6306666d1cbc6acf5cb862730727ce641289681cc8318483.png"><img alt="../../_images/d876f8fb50ba9b1e6306666d1cbc6acf5cb862730727ce641289681cc8318483.png" src="../../_images/d876f8fb50ba9b1e6306666d1cbc6acf5cb862730727ce641289681cc8318483.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">png</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">grid</span><span class="p">)</span>

<span class="n">img</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readPNG</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_cv2f/metric/xqtl_feature_max_allcv2f_noxqtl/FunGen_xQTL.PIP0.7_CS0_VCP0.8_COS0.xqtl_feature_max_allcv2f_noxqtl.png&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="nf">grid.raster</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/88e11c46e6fad9c05b1a9bf643e8fc73c613ff0df8e6e3631fe78a53f8e0eb6e.png"><img alt="../../_images/88e11c46e6fad9c05b1a9bf643e8fc73c613ff0df8e6e3631fe78a53f8e0eb6e.png" src="../../_images/88e11c46e6fad9c05b1a9bf643e8fc73c613ff0df8e6e3631fe78a53f8e0eb6e.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
</section>
<section id="optional-experiment-2">
<h2>(Optional) Experiment 2<a class="headerlink" href="#optional-experiment-2" title="Link to this heading">#</a></h2>
<section id="id2">
<h3>Additional Experiment (Suggestion from Gao)<a class="headerlink" href="#id2" title="Link to this heading">#</a></h3>
<p>only xQTL, kept <code class="docutils literal notranslate"><span class="pre">g_closest_tss</span></code> to make sure we have all same variants, but actually that was removed by below codes</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>pos_feature_tabb_list[[numchr]] = dff[match(intersect(positive_snps, dff$SNP), dff$SNP), -(1:4)]
neg_feature_tabb_list[[numchr]] = dff[match(intersect(negative_snps, dff$SNP), dff$SNP), -(1:4)]

</pre></div>
</div>
<p>and for the same reason, all other tests do not have <code class="docutils literal notranslate"><span class="pre">exon</span></code> track.</p>
<p>experiment by taking out xQTL information from feature sets to check AUC</p>
</section>
<section id="id3">
<h3>prepare new feature sets without xQTL<a class="headerlink" href="#id3" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">setwd</span><span class="p">(</span><span class="s">&#39;~/deyk/ru/Work/v2f/analysis&#39;</span><span class="p">)</span>
<span class="c1"># cv2f_files &lt;- list.files(&quot;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/cv2f_features/cv2f_features/&quot;, full.names = T)</span>
<span class="c1"># agg_xqtl_files &lt;- list.files(&quot;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max/&quot;, full.names = T)</span>


<span class="c1">#### Main loop #### </span>

<span class="kr">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="m">22</span><span class="p">){</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="n">i</span><span class="p">)</span>
<span class="w">  </span><span class="n">s2g</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max//xqtl_feature_max.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt.gz&#39;</span><span class="p">))</span>
<span class="w"> </span>
<span class="w">  </span><span class="n">s2g_xqtl_elements</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">s2g</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">.</span><span class="p">[</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;QTL|qtl|Metabrain|single_context|multi_context&#39;</span><span class="p">)]</span>
<span class="w">  </span><span class="c1"># remove all xqtl feature</span>
<span class="w">  </span><span class="n">s2g_xqtl</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">s2g</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">select</span><span class="p">(</span><span class="s">&#39;CHR&#39;</span><span class="p">,</span><span class="s">&#39;BP&#39;</span><span class="p">,</span><span class="s">&#39;SNP&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;g_closest_tss&#39;</span><span class="p">,</span><span class="w"> </span><span class="nf">any_of</span><span class="p">(</span><span class="n">s2g_xqtl_elements</span><span class="p">))</span><span class="w">  </span>
<span class="w">  </span><span class="c1">#   %&gt;% </span>
<span class="w">  </span><span class="c1"># filter(rowSums(select(., any_of(s2g_xqtl_elements))) &gt; 0) </span>
<span class="w">  </span><span class="c1"># s2g_xqtl %&gt;% mutate(snp=paste0(&#39;chr&#39;, CHR, &#39;:&#39;, BP), rsid=SNP) %&gt;% select(snp, rsid) %&gt;% </span>
<span class="w">  </span><span class="c1">#   fwrite(paste0(&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f_xqtlonly_tss/var_list.&#39;, i, &#39;.txt&#39;), sep = &#39;\t&#39;)</span>
<span class="w">  </span><span class="nf">fwrite</span><span class="p">(</span><span class="n">s2g_xqtl</span><span class="p">,</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f_xqtlonly_tss/xqtl_feature_max_allcv2f_xqtlonly_tss.&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;.txt.gz&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
<span class="w">    </span><span class="c1">## the difference with v2f/xqtl_features/no_xqtl_features_v2f is this does not remain the variants in cv2f only and that one does</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>12345678910111213141516171819202122
</pre></div>
</div>
</div>
</div>
<section id="id4">
<h4>check new feature sets<a class="headerlink" href="#id4" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f_xqtlonly_tss/xqtl_feature_max_allcv2f_xqtlonly_tss.22.txt.gz&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">s2g_xqtl</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>90316</li><li>252</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">feat_tmp</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span><span class="w"> </span><span class="c1">#chrom 22</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>90316</li><li>252</li></ol>
</div></div>
</div>
</section>
</section>
<section id="prepare-new-sets">
<h3>Prepare new sets<a class="headerlink" href="#prepare-new-sets" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">var_paths</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f_xqtlonly_tss/&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;var_list&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span><span class="c1"># the intermediate files from last step</span>
<span class="n">var_paths</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>

<span class="n">var_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">()</span>
<span class="kr">for</span><span class="p">(</span><span class="n">var_path</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">var_paths</span><span class="p">){</span>
<span class="w">    </span><span class="n">var_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">var_list</span><span class="p">,</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">var_path</span><span class="p">))</span>
<span class="p">}</span>
<span class="nf">fwrite</span><span class="p">(</span><span class="n">var_list</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f_xqtlonly_tss/variant_list.txt.gz&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;\t&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">22</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">var_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 × 2</caption>
<thead>
	<tr><th scope=col>snp</th><th scope=col>rsid</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr1:33092249 </td><td>rs1000016877</td></tr>
	<tr><td>chr1:2106661  </td><td>rs1000019726</td></tr>
	<tr><td>chr1:226392686</td><td>rs1000033   </td></tr>
	<tr><td>chr1:203840605</td><td>rs1000050843</td></tr>
	<tr><td>chr1:52456463 </td><td>rs1000057276</td></tr>
	<tr><td>chr1:66392232 </td><td>rs1000085   </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">pos_set</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readLines</span><span class="p">(</span><span class="s">&#39;/lila/data/deyk/ru/Work/v2f/results/posneg_process/positive_set.PIP0.7_CS0_VCP0.8_COS0.txt&#39;</span><span class="p">)</span>
<span class="n">neg_set</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readLines</span><span class="p">(</span><span class="s">&#39;/lila/data/deyk/ru/Work/v2f/results/posneg_process/negative_set.PIP0.7_CS0_VCP0.8_COS0.txt&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">pos_set</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
<span class="n">neg_set</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">1787</div><div class="output text_html">16190725</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">pos_set</span><span class="p">[</span><span class="n">pos_set</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">var_list</span><span class="o">$</span><span class="n">rsid</span><span class="p">]</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
<span class="n">neg_set</span><span class="p">[</span><span class="n">neg_set</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">var_list</span><span class="o">$</span><span class="n">rsid</span><span class="p">]</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">1594</div><div class="output text_html">4505174</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">var_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">writeLines</span><span class="p">(</span><span class="n">pos_set</span><span class="p">[</span><span class="n">pos_set</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">var_list</span><span class="o">$</span><span class="n">rsid</span><span class="p">],</span><span class="s">&#39;/lila/data/deyk/ru/Work/v2f/results/posneg_process/positive_set.PIP0.7_CS0_VCP0.8_COS0.xqtl_feature_max_allcv2f_xqtlonly_tss.txt&#39;</span><span class="p">)</span>
<span class="nf">writeLines</span><span class="p">(</span><span class="n">neg_set</span><span class="p">[</span><span class="n">neg_set</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">var_list</span><span class="o">$</span><span class="n">rsid</span><span class="p">],</span><span class="s">&#39;/lila/data/deyk/ru/Work/v2f/results/posneg_process/negative_set.PIP0.7_CS0_VCP0.8_COS0.xqtl_feature_max_allcv2f_xqtlonly_tss.txt&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="id5">
<h3>submit jobs<a class="headerlink" href="#id5" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#!/bin/bash</span>

<span class="c1"># Parameter sets</span>
<span class="n">param_sets</span><span class="o">=</span><span class="p">(</span>
<span class="w">  </span><span class="s">&quot;PIP0.7_CS0_VCP0.8_COS0&quot;</span>
<span class="p">)</span>

<span class="c1"># Feature sets</span>
<span class="n">feature_sets</span><span class="o">=</span><span class="p">(</span>
<span class="w">  </span><span class="s">&quot;/lila/data/deyk/ftp_fgc_xqtl/v2f/xqtl_features/xqtl_features_v2f/agg_snp_max_allcv2f_xqtlonly_tss/xqtl_feature_max_allcv2f_xqtlonly_tss.&quot;</span>
<span class="p">)</span>

<span class="c1"># Shared input files</span>
<span class="n">bimpath</span><span class="o">=</span><span class="s">&quot;/home/fengr1/deyk/ftp_fgc_xqtl/v2f/1000G_BIMS_hg38/1000G.EUR.QC.&quot;</span>
<span class="n">mafpath</span><span class="o">=</span><span class="s">&quot;/data/deyk/extras/MAF_features_Aug032022.txt&quot;</span>
<span class="n">ldblockspath</span><span class="o">=</span><span class="s">&quot;/home/fengr1/deyk/ftp_fgc_xqtl/v2f/LAVA_LDblocks_published.txt&quot;</span>

<span class="c1"># Create directories</span>
<span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">score</span><span class="o">/</span><span class="n">FunGen_xQTL</span>
<span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">metric</span><span class="o">/</span><span class="n">FunGen_xQTL</span>
<span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">logs</span>

<span class="kr">for</span><span class="w"> </span><span class="n">status</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="s">&quot;${param_sets[@]}&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">do</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="n">feat</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="s">&quot;${feature_sets[@]}&quot;</span><span class="p">;</span><span class="w"> </span><span class="n">do</span>
<span class="w">    </span><span class="n">feat_label</span><span class="o">=$</span><span class="p">(</span><span class="n">basename</span><span class="w"> </span><span class="o">$</span><span class="p">{</span><span class="n">feat</span>%<span class="n">.</span><span class="o">*</span><span class="p">})</span>
<span class="w">    </span><span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">score</span><span class="o">/</span><span class="n">FunGen_xQTL</span><span class="o">/$</span><span class="p">{</span><span class="n">status</span><span class="p">}</span><span class="o">/$</span><span class="p">{</span><span class="n">feat_label</span><span class="p">}</span>
<span class="w">    </span><span class="n">mkdir</span><span class="w"> </span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">metric</span><span class="o">/</span><span class="n">FunGen_xQTL</span><span class="o">/$</span><span class="p">{</span><span class="n">status</span><span class="p">}</span><span class="o">/$</span><span class="p">{</span><span class="n">feat_label</span><span class="p">}</span>

<span class="w">    </span><span class="n">positive_set</span><span class="o">=</span><span class="s">&quot;/lila/data/deyk/ru/Work/v2f/results/posneg_process/positive_set.${status}.txt&quot;</span>
<span class="w">    </span><span class="n">negative_set</span><span class="o">=</span><span class="s">&quot;/lila/data/deyk/ru/Work/v2f/results/posneg_process/negative_set.${status}.txt&quot;</span>
<span class="w">    </span><span class="n">output_metrics</span><span class="o">=</span><span class="s">&quot;/home/fengr1/Work/v2f/results/metric/FunGen_xQTL/${status}/${feat_label}/FunGen_xQTL.${status}.${feat_label}.metrics&quot;</span>

<span class="w">    </span><span class="n">echo</span><span class="w"> </span><span class="s">&quot;&gt;&gt;&gt; Running metrics: $status with $feat_label to $output_metrics&quot;</span>
<span class="w">    </span><span class="n">Rscript</span><span class="w"> </span><span class="o">/</span><span class="n">lila</span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">codes</span><span class="o">/</span><span class="n">cv2f</span><span class="o">/</span><span class="n">metric</span><span class="o">/</span><span class="n">run_cv2f_MAF5LDmatch_metrics.R</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">positive_set</span><span class="w"> </span><span class="o">$</span><span class="n">positive_set</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">negative_set</span><span class="w"> </span><span class="o">$</span><span class="n">negative_set</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">feature_file</span><span class="w"> </span><span class="o">$</span><span class="n">feat</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">bimpath</span><span class="w"> </span><span class="o">$</span><span class="n">bimpath</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">mafpath</span><span class="w"> </span><span class="o">$</span><span class="n">mafpath</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">ldblockspath</span><span class="w"> </span><span class="o">$</span><span class="n">ldblockspath</span><span class="w"> </span>\
<span class="w">      </span><span class="o">--</span><span class="n">output_metrics</span><span class="w"> </span><span class="o">$</span><span class="n">output_metrics</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">fengr1</span><span class="o">/</span><span class="n">Work</span><span class="o">/</span><span class="n">v2f</span><span class="o">/</span><span class="n">logs</span><span class="o">/</span><span class="n">xqtl_metric.</span><span class="o">$</span><span class="p">{</span><span class="n">status</span><span class="p">}</span><span class="n">.</span><span class="o">$</span><span class="p">{</span><span class="n">feat_label</span><span class="p">}</span><span class="n">.log</span><span class="w"> </span><span class="m">2</span><span class="o">&gt;&amp;</span><span class="m">1</span>
<span class="w">    </span><span class="n">echo</span><span class="w"> </span><span class="s">&#39;Finished: $output_metrics&#39;</span>
<span class="c1">#     for chrm in {1..22}; do</span>
<span class="c1">#       output_cv2f=&quot;/home/fengr1/Work/v2f/results/score/FunGen_xQTL/${status}/${feat_label}/FunGen_xQTL.GWASonly_response.${status}.${feat_label}.chr${chrm}.cv2f&quot;</span>

<span class="c1">#       echo &quot;&gt;&gt;&gt; Running score: $status, $feat_label, chr${chrm}&quot;</span>
<span class="c1">#       Rscript /lila/home/fengr1/codes/cv2f/score/run_cv2f_MAF5LDmatch_cv2f_scores.R \</span>
<span class="c1">#         --positive_set $positive_set \</span>
<span class="c1">#         --negative_set $negative_set \</span>
<span class="c1">#         --feature_file $feat \</span>
<span class="c1">#         --bimpath $bimpath \</span>
<span class="c1">#         --mafpath $mafpath \</span>
<span class="c1">#         --ldblockspath $ldblockspath \</span>
<span class="c1">#         --output_cv2f $output_cv2f \</span>
<span class="c1">#         --chrm $chrm &gt; /home/fengr1/Work/v2f/logs/xqtl_score.${status}.${feat_label}.chr${chrm}.log 2&gt;&amp;1</span>
<span class="c1">#     done</span>
<span class="w">  </span><span class="n">done</span>
<span class="n">done</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="id6">
<h3>Visualization of Results<a class="headerlink" href="#id6" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>

<span class="c1"># Define statuses (adjust to match your naming scheme)</span>
<span class="n">statuses</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;PIP0.7_CS0_VCP0.8_COS0&quot;</span><span class="p">)</span>

<span class="c1"># Output folder for plots</span>
<span class="n">plot_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_cv2f/metric&quot;</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="nf">dir.create</span><span class="p">(</span><span class="n">plot_dir</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="n">metric_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">figure_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span>

<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">status</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">statuses</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">message</span><span class="p">(</span><span class="s">&quot;Processing status: &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Load metrics</span>
<span class="w">  </span><span class="n">metric_file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">plot_dir</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;/xqtl_feature_max_allcv2f_xqtlonly_tss/FunGen_xQTL.&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.xqtl_feature_max_allcv2f_xqtlonly_tss.metrics&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="w">  </span><span class="n">shap_file</span><span class="w">   </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">plot_dir</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;/xqtl_feature_max_allcv2f_xqtlonly_tss/FunGen_xQTL.&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;.xqtl_feature_max_allcv2f_xqtlonly_tss.shap_values.csv&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="w">  </span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">file.exists</span><span class="p">(</span><span class="n">metric_file</span><span class="p">)</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="o">!</span><span class="nf">file.exists</span><span class="p">(</span><span class="n">shap_file</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">warning</span><span class="p">(</span><span class="s">&quot;Missing file(s) for status: &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">)</span>
<span class="w">    </span><span class="kr">next</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span>
<span class="w">  </span><span class="n">metric_list</span><span class="p">[[</span><span class="n">status</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">metric_file</span><span class="p">)</span>
<span class="w">  </span><span class="n">shap_dt</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">shap_file</span><span class="p">)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Aggregate SHAP values</span>
<span class="w">  </span><span class="n">shap_summary</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">shap_dt</span><span class="p">[,</span><span class="w"> </span><span class="nf">.</span><span class="p">(</span>
<span class="w">    </span><span class="n">mean_abs_rfvalue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">mean</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">rfvalue</span><span class="p">)),</span>
<span class="w">    </span><span class="n">max_rfvalue</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">rfvalue</span><span class="p">)),</span>
<span class="w">    </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">.N</span>
<span class="w">  </span><span class="p">),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">variable</span><span class="p">]</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="n">mean_abs_rfvalue</span><span class="p">))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Plot: Top 20 Features</span>
<span class="w">  </span><span class="n">figure_list</span><span class="p">[[</span><span class="n">status</span><span class="p">]]</span><span class="o">$</span><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">shap_summary</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">],</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">reorder</span><span class="p">(</span><span class="n">variable</span><span class="p">,</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">),</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_col</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">coord_flip</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Top 20 Features by Mean |SHAP| -&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">),</span>
<span class="w">         </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Feature&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mean |SHAP| Value&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme_minimal</span><span class="p">()</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># ggsave(filename = paste0(plot_dir, &quot;Top20_SHAP_&quot;, status, &quot;.png&quot;), plot = p1, width = 10, height = 6)</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1">### Plot: Top 20 Non-dominant Non-sQTL Features</span>
<span class="w">  </span><span class="n">figure_list</span><span class="p">[[</span><span class="n">status</span><span class="p">]]</span><span class="o">$</span><span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">shap_summary</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">slice</span><span class="p">(</span><span class="o">-</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">variable</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sQTL&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">    </span><span class="nf">slice_max</span><span class="p">(</span><span class="n">order_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">ggplot</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">reorder</span><span class="p">(</span><span class="n">variable</span><span class="p">,</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">),</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">mean_abs_rfvalue</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_col</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">coord_flip</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Top 20 Non-dominant Non-sQTL Features -&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">status</span><span class="p">),</span>
<span class="w">         </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Feature&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Mean |SHAP| Value&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme_minimal</span><span class="p">()</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># ggsave(filename = paste0(plot_dir, &quot;Top20_NonSQTLSHAP_&quot;, status, &quot;.png&quot;), plot = p2, width = 10, height = 6)</span>

<span class="p">}</span>
<span class="w">     </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Processing status: PIP0.7_CS0_VCP0.8_COS0
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">metric_list</span><span class="o">$</span><span class="n">PIP0.7_CS0_VCP0.8_COS0</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 1 × 4</caption>
<thead>
	<tr><th scope=col>AUROC.mean</th><th scope=col>AUROC.sd</th><th scope=col>AUPRC.mean</th><th scope=col>AUPRC.sd</th></tr>
	<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>0.8841824</td><td>0.01026659</td><td>0.9093274</td><td>0.00878147</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">figure_list</span><span class="o">$</span><span class="n">PIP0.7_CS0_VCP0.8_COS0</span><span class="o">$</span><span class="n">p1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/33b4dbb12f035478001e79f68437a038ab74efcde0ebe57b69d74bcffc277476.png"><img alt="../../_images/33b4dbb12f035478001e79f68437a038ab74efcde0ebe57b69d74bcffc277476.png" src="../../_images/33b4dbb12f035478001e79f68437a038ab74efcde0ebe57b69d74bcffc277476.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">png</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">grid</span><span class="p">)</span>

<span class="n">img</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readPNG</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_cv2f/metric/xqtl_feature_max_allcv2f_xqtlonly_tss/FunGen_xQTL.PIP0.7_CS0_VCP0.8_COS0.xqtl_feature_max_allcv2f_xqtlonly_tss.png&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="nf">grid.raster</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/36088e6175c5919c666864d0b18a174fe3de7100a80dca9cd2e4c2aa4040ff66.png"><img alt="../../_images/36088e6175c5919c666864d0b18a174fe3de7100a80dca9cd2e4c2aa4040ff66.png" src="../../_images/36088e6175c5919c666864d0b18a174fe3de7100a80dca9cd2e4c2aa4040ff66.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="visualization-comparison">
<h3>visualization comparison<a class="headerlink" href="#visualization-comparison" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">metric_withxqtl</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_cv2f/metric/xqtl_feature_max_allcv2f/FunGen_xQTL.PIP0.7_CS0_VCP0.8_COS0.xqtl_feature_max_allcv2f.metrics&#39;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="n">metric_noxqtl</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_cv2f/metric/xqtl_feature_max_allcv2f_noxqtl/FunGen_xQTL.PIP0.7_CS0_VCP0.8_COS0.xqtl_feature_max_allcv2f_noxqtl.metrics&#39;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="n">metric_xqtlonly</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_cv2f/metric/xqtl_feature_max_allcv2f_xqtlonly_tss/FunGen_xQTL.PIP0.7_CS0_VCP0.8_COS0.xqtl_feature_max_allcv2f_xqtlonly_tss.metrics&#39;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">metric_withxqtl</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">melt</span><span class="p">(</span><span class="n">metric_withxqtl</span><span class="p">,</span><span class="w"> </span><span class="n">measure</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">patterns</span><span class="p">(</span><span class="n">mean</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\\.mean$&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\\.sd$&quot;</span><span class="p">))</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">metric_withxqtl</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;Metric&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Mean&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;SD&#39;</span><span class="p">)</span>
<span class="n">metric_withxqtl</span><span class="o">$</span><span class="n">Metric</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;AUROC&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;AUPRC&#39;</span><span class="p">)</span>

<span class="n">metric_noxqtl</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">melt</span><span class="p">(</span><span class="n">metric_noxqtl</span><span class="p">,</span><span class="w"> </span><span class="n">measure</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">patterns</span><span class="p">(</span><span class="n">mean</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\\.mean$&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\\.sd$&quot;</span><span class="p">))</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">metric_noxqtl</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;Metric&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Mean&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;SD&#39;</span><span class="p">)</span>
<span class="n">metric_noxqtl</span><span class="o">$</span><span class="n">Metric</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;AUROC&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;AUPRC&#39;</span><span class="p">)</span>

<span class="n">metric_xqtlonly</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">melt</span><span class="p">(</span><span class="n">metric_xqtlonly</span><span class="p">,</span><span class="w"> </span><span class="n">measure</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">patterns</span><span class="p">(</span><span class="n">mean</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\\.mean$&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\\.sd$&quot;</span><span class="p">))</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">metric_xqtlonly</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;Metric&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Mean&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;SD&#39;</span><span class="p">)</span>
<span class="n">metric_xqtlonly</span><span class="o">$</span><span class="n">Metric</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;AUROC&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;AUPRC&#39;</span><span class="p">)</span>

<span class="n">metric</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">metric_withxqtl</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Set</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;With xQTL&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">metric_noxqtl</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Set</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;Without xQTL&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">metric_xqtlonly</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Set</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;xQTL only&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">ggplot</span><span class="p">(</span><span class="n">metric</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Metric</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Mean</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Set</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;identity&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">position_dodge</span><span class="p">(</span><span class="m">0.7</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_errorbar</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">ymin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Mean</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">SD</span><span class="p">,</span><span class="w"> </span><span class="n">ymax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Mean</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">SD</span><span class="p">),</span>
<span class="w">                </span><span class="n">position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">position_dodge</span><span class="p">(</span><span class="m">0.7</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.2</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;AUC Score&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Metric&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_classic</span><span class="p">(</span><span class="n">base_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">14</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/5d0533d5ab6c94c3ff5e9cbe4b88814a3c1fbd75cd921f75e1b66b726bb27b40.png"><img alt="../../_images/5d0533d5ab6c94c3ff5e9cbe4b88814a3c1fbd75cd921f75e1b66b726bb27b40.png" src="../../_images/5d0533d5ab6c94c3ff5e9cbe4b88814a3c1fbd75cd921f75e1b66b726bb27b40.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="summary">
<h3>Summary<a class="headerlink" href="#summary" title="Link to this heading">#</a></h3>
<p>By including all variants from CS, CoS, and uCoS, along with high-PIP variants (PIP &gt; 0.7) outside the CS, we obtained a positive set of over 1,700 variants.<br />
The resulting model achieved an AUC of <strong>0.91</strong>, with xQTL features contributing substantially to this performance.</p>
<p>As a comparative experiment, we removed all xQTL features and trained the model using only Fabiha’s feature set.<br />
In this setting, the AUC dropped to <strong>0.86</strong>, underscoring the critical contribution of xQTL features to predictive accuracy.</p>
<p>Another a comparative experiment, we kept only all xQTL features and trained the model.<br />
In this setting, the AUC dropped to <strong>0.88</strong>.</p>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./main_text/5_AD_xQTL_integration"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="Figure_5f_PPI.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">PPI Networks for Selected AD Risk Genes in xQTL Loci</p>
      </div>
    </a>
    <a class="right-next"
       href="Figure_5h_GWAS_enrichment.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">GWAS Enrichment</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#authors-ru-feng-and-jenny-empawi">Authors: Ru Feng and Jenny Empawi</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#motivation">Motivation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#implementation">Implementation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#fungen-xqtl-cv2f-analysis">FunGen xQTL cv2f analysis</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#preparing-the-feature-sets">1. Preparing the Feature Sets</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#reformat-the-feature-sets">Reformat the feature sets</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-feature-set-data">Load feature set data</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#note-different-columns-per-chromosome">NOTE: Different columns per chromosome</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-rsid-reference">Load rsid reference</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#map-rsid-to-chr-pos">map rsid to chr_pos</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#combine-with-cv2f-feature-sets">Combine with cv2f feature sets</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#check-feature-sets">check feature sets</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#generating-positive-and-negative-sets">2. Generating Positive and Negative Sets</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#generate-variant-list-from-feature-set">generate variant list from feature set</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#positive-set">positive set</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#define-parameter-sets">Define parameter sets</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#submit-jobs-for-cv2f-analysis">3. Submit Jobs for cv2f Analysis</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#check-the-time-stamp-of-input-files">Check the time stamp of input files</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#visualization-of-results">4. Visualization of Results</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#pip0-7-cs0-vcp0-8-cos0">PIP0.7_CS0_VCP0.8_COS0</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#optional-experiment">(Optional) Experiment</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#additional-experiment-suggestion-from-gao">Additional Experiment (Suggestion from Gao)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#prepare-new-feature-sets-without-xqtl">prepare new feature sets without xQTL</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#check-new-feature-sets">check new feature sets</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#submit-jobs">submit jobs</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">Visualization of Results</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#optional-experiment-2">(Optional) Experiment 2</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">Additional Experiment (Suggestion from Gao)</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id3">prepare new feature sets without xQTL</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id4">check new feature sets</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#prepare-new-sets">Prepare new sets</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id5">submit jobs</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id6">Visualization of Results</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#visualization-comparison">visualization comparison</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#summary">Summary</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>