
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>AD loci “explained” by xQTL &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/5_AD_xQTL_integration/Figure_5c_AD_loci_xQTL_annotated';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Summary of FunGen-xQTL AD loci table" href="Figure_5d_AD_loci_table_summary.html" />
    <link rel="prev" title="Refined AD GWAS loci properties" href="Figure_5b_AD_GWAS_loci_properties.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../1_project_overview/index.html">Multi-omics QTL atlas of the aging human brain</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../2_single_context_cis/index.html">Fine-mapped QTL landscape across brain molecular phenotypes</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_1b_num_xQTL_loci_single_context.html">Figure 1b. Number of detectable and mappable xQTL loci in single−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_EOO.html">EOO results</a></li>

<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_S2_single_context_cs_distance.html">Frequency of Events in Distance Category</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_sLDSC_GWAS_specific.html">SLDSC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../2_single_context_cis/Figure_sLDSC_meta_analysis.html">Visualization of SLDSC results</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../3_single_context_multigene_trans/index.html">Coordinated gene regulation and long-range genetic effects</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../4_multi_context/index.html">Cascading genetic effects across molecular contexts</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_2b_num_xQTL_loci_multi_context.html">Number of detectable and mappable xQTL loci in ROSMAP multi−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_2c_multicontext_effect_sign.html">Marginal vs multicontext effect sign heatmap</a></li>

<li class="toctree-l2"><a class="reference internal" href="../4_multi_context/Figure_watershed_posterior_plot.html">Watershed Posterior</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="index.html">Regulatory architecture of AD susceptibility loci</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="Figure_5a_AD_GWAS_loci_harmonized.html">AD GWAS loci harmonization and refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5b_AD_GWAS_loci_properties.html">Refined AD GWAS loci properties</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">AD loci “explained” by xQTL</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5d_AD_loci_table_summary.html">Summary of FunGen-xQTL AD loci table</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5e_GSEA.html">Gene Set Enrichment Analysis for AD Risk Genes</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5f_PPI.html">PPI Networks for Selected AD Risk Genes in xQTL Loci</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5g_cV2F.html">Consensus Variant-to-Function Score for AD</a></li>
<li class="toctree-l2"><a class="reference internal" href="Figure_5h_GWAS_enrichment.html">GWAS Enrichment</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../6_AD_xQTL_genes/index.html">Molecular dissection of prioritized AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization.html">Visualization of FunGen-xQTL AD Loci Table</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../7_AD_gene_prioritized/index.html">Molecular insights and gene prioritization at key AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../7_AD_gene_prioritized/APP_case_example.html">APP</a></li>




<li class="toctree-l2"><a class="reference internal" href="../7_AD_gene_prioritized/ZYX_case_example.html">Molecular insights and gene prioritization at key AD loci</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/5_AD_xQTL_integration/Figure_5c_AD_loci_xQTL_annotated.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/main_text/5_AD_xQTL_integration/Figure_5c_AD_loci_xQTL_annotated.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>AD loci “explained” by xQTL</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#motivation">Motivation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#methods-overview">Methods Overview</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#implementation">Implementation</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overlap-analysis">Overlap Analysis</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-involvement-evidence">Gene Involvement Evidence</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-single-context-results">Summarize single context results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-multi-context-results">summarize multi context results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-metabrian-results">summarize metabrian results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-multi-gene-results">summarize multi gene results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-the-coloboost-results">summarize the coloboost results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-trans-results">summarize trans results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-fsusie-results">summarize fsusie results</a><ul class="nav section-nav flex-column">
<li class="toc-h5 nav-item toc-entry"><a class="reference internal nav-link" href="#combine-all">combine all</a></li>
</ul>
</li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#plot-susie-upsetr">plot susie upsetR</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#plot-the-stacked-bar-plot-only-four-categories">plot the stacked bar plot (only four categories)</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#check-trans-only">check trans only</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#add-gene-information-to-gpqtl">add gene information to gpQTL</a></li>
</ul>
</li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="ad-loci-explained-by-xqtl">
<h1>AD loci “explained” by xQTL<a class="headerlink" href="#ad-loci-explained-by-xqtl" title="Link to this heading">#</a></h1>
<p><strong>Author:</strong> Jenny Empawi</p>
<section id="motivation">
<h2>Motivation<a class="headerlink" href="#motivation" title="Link to this heading">#</a></h2>
<p>Answer the question: How many AD risk loci overlap (via fine-mapping credible sets) and/or formally colocalize (via colocalization confident sets) with various xQTL modalities and contexts? This analysis aims to quantify the extent to which Alzheimer’s disease risk loci can be ‘explained’ through molecular quantitative trait loci across multiple biological contexts and data types.</p>
</section>
<section id="methods-overview">
<h2>Methods Overview<a class="headerlink" href="#methods-overview" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p>Leverage AD GWAS fine-mapping and xQTL fine-mapping discussed in previous sections</p></li>
<li><p>Perform AD GWAS fine-mapping with multi-trait xQTL data genome-wide</p></li>
<li><p>Summarize results by GWAS studies to identify the proportion of AD loci explained</p></li>
</ul>
</section>
<section id="implementation">
<h2>Implementation<a class="headerlink" href="#implementation" title="Link to this heading">#</a></h2>
<section id="overlap-analysis">
<h3>Overlap Analysis<a class="headerlink" href="#overlap-analysis" title="Link to this heading">#</a></h3>
<p>Create a bar plot showing overlap between AD loci and multiple xQTL data types, quantifying explained loci and distinguishing between detectable and mappable categories. This analysis will highlight overlap types with illustrative cartoons and bar plots to visualize the intersection patterns between AD risk variants and molecular QTL signals across different experimental contexts.</p>
</section>
<section id="gene-involvement-evidence">
<h3>Gene Involvement Evidence<a class="headerlink" href="#gene-involvement-evidence" title="Link to this heading">#</a></h3>
<p>Generate a stacked bar plot for each AD locus showing evidence of multiple gene involvement, defined by overlap between 95% GWAS credible sets and 95% xQTL credible sets. This section will summarize xQTL-based gene targets per AD locus, including both cis and trans targets, providing comprehensive insights into the molecular mechanisms potentially underlying AD risk at each genomic region.</p>
<section id="summarize-single-context-results">
<h4>Summarize single context results<a class="headerlink" href="#summarize-single-context-results" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Attaching package: ‘dplyr’


The following objects are masked from ‘package:data.table’:

    between, first, last


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#gwas &lt;- fread(&#39;analysis_result/AD_GWAS_finemapping/export/context_specific/data/109_blocks_top_loci.csv.gz&#39;) ## From Ruixi on Mar19, 2025</span>
<span class="n">gwas</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/Staging/variant_gene_summary/AD_xQTL_integration_summary/AD_GWAS_finemapping_109_blocks_top_loci_unified_any0.8ANDmin0.5.csv.gz&#39;</span><span class="p">)</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95_min_corr&quot;</span>
<span class="n">gwas_95</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gwas</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="o">!!</span><span class="n">coverage</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gwas_95</span><span class="w"> </span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="n">region</span><span class="p">,</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]],</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;:&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">variant_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;chr&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">variant_id</span><span class="p">))</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;AD_Bellenguez_2022:chr15_47690138_49901356:1&quot;</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">pip</span><span class="o">==</span><span class="m">0.998973148941477</span><span class="p">))</span>
<span class="n">single_context_meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;/Staging/variant_gene_summary/xQTL_loci_summary/xQTL_finemapping/single_context_finemapping/consolidated/FunGen_xQTL_single_context_top_loci_integrated.rds&quot;</span><span class="p">)</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">single_context_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">single_context_meta_df</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="n">pip_sum</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID_n_detectable</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID_detectable_variant_count</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID_n_mappable</span><span class="p">))</span><span class="o">%&gt;%</span>
<span class="w">                                    </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="p">,</span><span class="w">  </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;variant_ID&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">suffixes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;QTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;GWAS&#39;</span><span class="p">))</span><span class="o">%&gt;%</span>
<span class="w">                                    </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">event_ID</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                    </span><span class="nf">rename</span><span class="p">(</span><span class="n">event_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">resource</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">single_95_info</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">single_context_meta_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">cs_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">cs_coverage_0.95</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">single_95_info</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">cs_id</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">270965</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">single_context_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">saveRDS</span><span class="p">(</span><span class="s">&quot;single_context_AD_GWAS_meta_df.rds&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">head</span><span class="p">(</span><span class="n">single_context_AD_GWAS_meta_df</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">single_context_AD_GWAS_meta_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 22</caption>
<thead>
	<tr><th></th><th scope=col>variant_ID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>a1</th><th scope=col>a2</th><th scope=col>gene_ID</th><th scope=col>cs_coverage_0.95</th><th scope=col>cs_coverage_0.7</th><th scope=col>cs_coverage_0.5</th><th scope=col>⋯</th><th scope=col>event_ID</th><th scope=col>study</th><th scope=col>variant_id</th><th scope=col>region</th><th scope=col>cs_coverage_0.95_min_corr</th><th scope=col>cs_coverage_0.7_min_corr</th><th scope=col>cs_coverage_0.5_min_corr</th><th scope=col>pip</th><th scope=col>z</th><th scope=col>gwas_cs</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>1</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>1</td><td>1</td><td>1</td><td>⋯</td><td>Mic_mega_eQTL   </td><td>AD_Bellenguez_2022               </td><td>10:11678309:A:G</td><td>chr10_10500888_12817813</td><td>1</td><td>-1</td><td>-1</td><td>        NA</td><td>      NA</td><td>AD_Bellenguez_2022:chr10_10500888_12817813:1               </td></tr>
	<tr><th scope=row>2</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>1</td><td>1</td><td>1</td><td>⋯</td><td>Mic_mega_eQTL   </td><td>AD_Jansen_2021                   </td><td>10:11678309:A:G</td><td>chr10_10500888_12817813</td><td>1</td><td>-1</td><td>-1</td><td>0.18080328</td><td>5.645878</td><td>AD_Jansen_2021:chr10_10500888_12817813:1                   </td></tr>
	<tr><th scope=row>3</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>1</td><td>1</td><td>1</td><td>⋯</td><td>Mic_mega_eQTL   </td><td>AD_Wightman_Full_2021            </td><td>10:11678309:A:G</td><td>chr10_10500888_12817813</td><td>1</td><td>-1</td><td>-1</td><td>0.05051167</td><td>7.374065</td><td>AD_Wightman_Full_2021:chr10_10500888_12817813:1            </td></tr>
	<tr><th scope=row>4</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>1</td><td>1</td><td>1</td><td>⋯</td><td>Mic_mega_eQTL   </td><td>AD_Wightman_Excluding23andMe_2021</td><td>10:11678309:A:G</td><td>chr10_10500888_12817813</td><td>1</td><td> 1</td><td> 1</td><td>0.49184952</td><td>6.603571</td><td>AD_Wightman_Excluding23andMe_2021:chr10_10500888_12817813:1</td></tr>
	<tr><th scope=row>5</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>1</td><td>1</td><td>1</td><td>⋯</td><td>Mic_DeJager_eQTL</td><td>AD_Bellenguez_2022               </td><td>10:11678309:A:G</td><td>chr10_10500888_12817813</td><td>1</td><td>-1</td><td>-1</td><td>        NA</td><td>      NA</td><td>AD_Bellenguez_2022:chr10_10500888_12817813:1               </td></tr>
	<tr><th scope=row>6</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>1</td><td>1</td><td>1</td><td>⋯</td><td>Mic_DeJager_eQTL</td><td>AD_Jansen_2021                   </td><td>10:11678309:A:G</td><td>chr10_10500888_12817813</td><td>1</td><td>-1</td><td>-1</td><td>0.18080328</td><td>5.645878</td><td>AD_Jansen_2021:chr10_10500888_12817813:1                   </td></tr>
</tbody>
</table>
</div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'variant_ID'</li><li>'#chr'</li><li>'start'</li><li>'end'</li><li>'a1'</li><li>'a2'</li><li>'gene_ID'</li><li>'cs_coverage_0.95'</li><li>'cs_coverage_0.7'</li><li>'cs_coverage_0.5'</li><li>'PIP'</li><li>'conditional_effect'</li><li>'event_ID'</li><li>'study'</li><li>'variant_id'</li><li>'region'</li><li>'cs_coverage_0.95_min_corr'</li><li>'cs_coverage_0.7_min_corr'</li><li>'cs_coverage_0.5_min_corr'</li><li>'pip'</li><li>'z'</li><li>'gwas_cs'</li></ol>
</div></div>
</div>
</section>
<section id="summarize-multi-context-results">
<h4>summarize multi context results<a class="headerlink" href="#summarize-multi-context-results" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="n">ROSMAP_multicontext_top_loci</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;/Staging/variant_gene_summary/xQTL_loci_summary/xQTL_finemapping/multi_context_finemapping/ROSMAP.exported.toploci.bed.gz&quot;</span><span class="p">)</span>
<span class="n">MSBB_multicontext_top_loci</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;/Staging/variant_gene_summary/xQTL_loci_summary/xQTL_finemapping/multi_context_finemapping/MSBB.exported.toploci.bed.gz&quot;</span><span class="p">)</span>
<span class="n">multicontext_top_loci</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">ROSMAP_multicontext_top_loci</span><span class="p">,</span><span class="w"> </span><span class="n">MSBB_multicontext_top_loci</span><span class="p">)</span>
<span class="c1">#gwas &lt;- fread(&#39;/data/interactive_analysis/rl3328/image_QTL/AD_GWAS_finemapping/export/output/extracted_info/109_blocks_top_loci.csv.gz&#39;) ## From Ruixi on Mar19, 2025</span>
<span class="n">gwas</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/Staging/variant_gene_summary/AD_xQTL_integration_summary/AD_GWAS_finemapping_109_blocks_top_loci_unified_any0.8ANDmin0.5.csv.gz&#39;</span><span class="p">)</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95_min_corr&quot;</span>
<span class="n">gwas_95</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gwas</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="o">!!</span><span class="n">coverage</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gwas_95</span><span class="w"> </span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="n">region</span><span class="p">,</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]],</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;:&quot;</span><span class="p">))</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">variant_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;chr&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">variant_id</span><span class="p">))</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;AD_Bellenguez_2022:chr15_47690138_49901356:1&quot;</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">pip</span><span class="o">==</span><span class="m">0.998973148941477</span><span class="p">))</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">multi_context_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">multicontext_top_loci</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="p">,</span><span class="w">  </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;variant_ID&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">suffixes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;QTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;GWAS&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">head</span><span class="p">(</span><span class="n">multi_context_AD_GWAS_meta_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 × 23</caption>
<thead>
	<tr><th scope=col>variant_ID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>a1</th><th scope=col>a2</th><th scope=col>gene_ID</th><th scope=col>event_ID</th><th scope=col>cs_coverage_0.95</th><th scope=col>cs_coverage_0.7</th><th scope=col>⋯</th><th scope=col>lfsr</th><th scope=col>study</th><th scope=col>variant_id</th><th scope=col>region</th><th scope=col>cs_coverage_0.95_min_corr</th><th scope=col>cs_coverage_0.7_min_corr</th><th scope=col>cs_coverage_0.5_min_corr</th><th scope=col>pip</th><th scope=col>z</th><th scope=col>gwas_cs</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>AC_DeJager_eQTL;Inh_DeJager_eQTL;Mic_DeJager_eQTL</td><td>1</td><td>1</td><td>⋯</td><td>5.39940054691537e-10;5.77822064232386e-06;1.31041603191121e-34</td><td>AD_Bellenguez_2022               </td><td>10:11678309:A:G</td><td>chr10_10500888_12817813</td><td>1</td><td>-1</td><td>-1</td><td>        NA</td><td>      NA</td><td>AD_Bellenguez_2022:chr10_10500888_12817813:1               </td></tr>
	<tr><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>AC_DeJager_eQTL;Inh_DeJager_eQTL;Mic_DeJager_eQTL</td><td>1</td><td>1</td><td>⋯</td><td>5.39940054691537e-10;5.77822064232386e-06;1.31041603191121e-34</td><td>AD_Jansen_2021                   </td><td>10:11678309:A:G</td><td>chr10_10500888_12817813</td><td>1</td><td>-1</td><td>-1</td><td>0.18080328</td><td>5.645878</td><td>AD_Jansen_2021:chr10_10500888_12817813:1                   </td></tr>
	<tr><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>AC_DeJager_eQTL;Inh_DeJager_eQTL;Mic_DeJager_eQTL</td><td>1</td><td>1</td><td>⋯</td><td>5.39940054691537e-10;5.77822064232386e-06;1.31041603191121e-34</td><td>AD_Wightman_Excluding23andMe_2021</td><td>10:11678309:A:G</td><td>chr10_10500888_12817813</td><td>1</td><td> 1</td><td> 1</td><td>0.49184952</td><td>6.603571</td><td>AD_Wightman_Excluding23andMe_2021:chr10_10500888_12817813:1</td></tr>
	<tr><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>AC_DeJager_eQTL;Inh_DeJager_eQTL;Mic_DeJager_eQTL</td><td>1</td><td>1</td><td>⋯</td><td>5.39940054691537e-10;5.77822064232386e-06;1.31041603191121e-34</td><td>AD_Wightman_Full_2021            </td><td>10:11678309:A:G</td><td>chr10_10500888_12817813</td><td>1</td><td>-1</td><td>-1</td><td>0.05051167</td><td>7.374065</td><td>AD_Wightman_Full_2021:chr10_10500888_12817813:1            </td></tr>
	<tr><td>chr10:11678621:C:T</td><td>10</td><td>11678620</td><td>11678621</td><td>T</td><td>C</td><td>ENSG00000148429</td><td>AC_DeJager_eQTL;Inh_DeJager_eQTL;Mic_DeJager_eQTL</td><td>1</td><td>0</td><td>⋯</td><td>5.39940054691537e-10;5.77822064232386e-06;1.31041603191121e-34</td><td>AD_Bellenguez_2022               </td><td>10:11678621:C:T</td><td>chr10_10500888_12817813</td><td>1</td><td>-1</td><td>-1</td><td>        NA</td><td>      NA</td><td>AD_Bellenguez_2022:chr10_10500888_12817813:1               </td></tr>
	<tr><td>chr10:11678621:C:T</td><td>10</td><td>11678620</td><td>11678621</td><td>T</td><td>C</td><td>ENSG00000148429</td><td>AC_DeJager_eQTL;Inh_DeJager_eQTL;Mic_DeJager_eQTL</td><td>1</td><td>0</td><td>⋯</td><td>5.39940054691537e-10;5.77822064232386e-06;1.31041603191121e-34</td><td>AD_Jansen_2021                   </td><td>10:11678621:C:T</td><td>chr10_10500888_12817813</td><td>1</td><td>-1</td><td>-1</td><td>0.04382463</td><td>5.378474</td><td>AD_Jansen_2021:chr10_10500888_12817813:1                   </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">colnames</span><span class="p">(</span><span class="n">multi_context_AD_GWAS_meta_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'variant_ID'</li><li>'#chr'</li><li>'start'</li><li>'end'</li><li>'a1'</li><li>'a2'</li><li>'gene_ID'</li><li>'event_ID'</li><li>'cs_coverage_0.95'</li><li>'cs_coverage_0.7'</li><li>'cs_coverage_0.5'</li><li>'PIP'</li><li>'conditional_effect'</li><li>'lfsr'</li><li>'study'</li><li>'variant_id'</li><li>'region'</li><li>'cs_coverage_0.95_min_corr'</li><li>'cs_coverage_0.7_min_corr'</li><li>'cs_coverage_0.5_min_corr'</li><li>'pip'</li><li>'z'</li><li>'gwas_cs'</li></ol>
</div></div>
</div>
</section>
<section id="summarize-metabrian-results">
<h4>summarize metabrian results<a class="headerlink" href="#summarize-metabrian-results" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="n">metabrain_sing_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/data/analysis_result/single_context/MetaBrain/export/summary/MetaBrain.exported.toploci.bed.gz&#39;</span><span class="p">)</span>
<span class="n">metabrain_sing_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">metabrain_sing_df</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_extract</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;^[^_]+_[^_]+&quot;</span><span class="p">),</span><span class="w">  </span><span class="c1"># Remove last part (gene name) to get context</span>
<span class="w">    </span><span class="n">gene_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_extract</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;[^_]+$&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1"># Extract last part (gene name)</span>
<span class="w">  </span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">gene_name</span><span class="p">)</span>
<span class="c1">#gwas &lt;- fread(&#39;/data/interactive_analysis/rl3328/image_QTL/AD_GWAS_finemapping/export/output/extracted_info/109_blocks_top_loci.csv.gz&#39;) ## From Ruixi on Mar19, 2025</span>
<span class="n">gwas</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/Staging/variant_gene_summary/AD_xQTL_integration_summary/AD_GWAS_finemapping_109_blocks_top_loci_unified_any0.8ANDmin0.5.csv.gz&#39;</span><span class="p">)</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95_min_corr&quot;</span>
<span class="n">gwas_95</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gwas</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="o">!!</span><span class="n">coverage</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gwas_95</span><span class="w"> </span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="n">region</span><span class="p">,</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]],</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;:&quot;</span><span class="p">))</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">variant_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;chr&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">variant_id</span><span class="p">))</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;AD_Bellenguez_2022:chr15_47690138_49901356:1&quot;</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">pip</span><span class="o">==</span><span class="m">0.998973148941477</span><span class="p">))</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">metabrain_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">metabrain_sing_df</span><span class="o">%&gt;%</span>
<span class="w">                                    </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="p">,</span><span class="w">  </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;variant_ID&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">suffixes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;QTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;GWAS&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">colnames</span><span class="p">(</span><span class="n">metabrain_AD_GWAS_meta_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'variant_ID'</li><li>'#chr'</li><li>'start'</li><li>'end'</li><li>'a1'</li><li>'a2'</li><li>'gene_ID'</li><li>'event_ID'</li><li>'cs_coverage_0.95'</li><li>'cs_coverage_0.7'</li><li>'cs_coverage_0.5'</li><li>'PIP'</li><li>'conditional_effect'</li><li>'resource'</li><li>'study'</li><li>'variant_id'</li><li>'region'</li><li>'cs_coverage_0.95_min_corr'</li><li>'cs_coverage_0.7_min_corr'</li><li>'cs_coverage_0.5_min_corr'</li><li>'pip'</li><li>'z'</li><li>'gwas_cs'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">metabrain_AD_GWAS_meta_df</span><span class="o">$</span><span class="n">resource</span><span class="o">%&gt;%</span><span class="nf">unique</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'Metabrain_Cortex'</li><li>'Metabrain_Cerebellum'</li><li>'Metabrain_Basalganglia'</li><li>'Metabrain_Hippocampus'</li><li>'Metabrain_Spinalcord'</li></ol>
</div></div>
</div>
</section>
<section id="summarize-multi-gene-results">
<h4>summarize multi gene results<a class="headerlink" href="#summarize-multi-gene-results" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&quot;/Staging/variant_gene_summary/xQTL_loci_summary/xQTL_finemapping/multi_gene_finemapping/eQTL/&quot;</span><span class="p">,</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.bed.gz$&quot;</span><span class="p">,</span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">()</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">files</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">multi_gene_export</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">files</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
<span class="w">    </span><span class="n">resource</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sub</span><span class="p">(</span><span class="s">&quot;.*ROSMAP_(.*?)\\.exported.*&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\\1&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">files</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
<span class="w">    </span><span class="kr">if</span><span class="p">(</span><span class="n">resource</span><span class="o">==</span><span class="s">&quot;eQTL&quot;</span><span class="p">){</span><span class="n">resource</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;monocyte_ROSMAP_eQTL&quot;</span><span class="p">}</span>
<span class="w">    </span><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multi_gene_export</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">resource</span><span class="p">)</span>
<span class="w">    </span><span class="n">tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">tmp</span><span class="p">,</span><span class="w"> </span><span class="n">df</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">tmp</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">resource</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">unique</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'AC_DeJager_eQTL'</li><li>'Ast_DeJager_eQTL'</li><li>'Ast_mega_eQTL'</li><li>'DLPFC_Bennett_pQTL'</li><li>'DLPFC_DeJager_eQTL'</li><li>'Exc_DeJager_eQTL'</li><li>'Exc_mega_eQTL'</li><li>'Inh_DeJager_eQTL'</li><li>'Inh_mega_eQTL'</li><li>'Mic_DeJager_eQTL'</li><li>'Mic_mega_eQTL'</li><li>'OPC_DeJager_eQTL'</li><li>'OPC_mega_eQTL'</li><li>'Oli_DeJager_eQTL'</li><li>'Oli_mega_eQTL'</li><li>'PCC_DeJager_eQTL'</li><li>'monocyte_ROSMAP_eQTL'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="n">files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&quot;/Staging/variant_gene_summary/xQTL_loci_summary/xQTL_finemapping/multi_gene_finemapping/eQTL/&quot;</span><span class="p">,</span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;.bed.gz$&quot;</span><span class="p">,</span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">MSBB_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">()</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">files</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="n">multi_gene_export</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="n">files</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
<span class="n">resource</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_extract</span><span class="p">(</span><span class="n">files</span><span class="p">[</span><span class="n">i</span><span class="p">],</span><span class="w"> </span><span class="s">&quot;(?&lt;=/)[^/]+_eQTL(?=\\.exported)&quot;</span><span class="p">)</span>
<span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multi_gene_export</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">resource</span><span class="p">)</span>
<span class="n">MSBB_tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">MSBB_tmp</span><span class="p">,</span><span class="w"> </span><span class="n">df</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multi_gene_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">tmp</span><span class="p">,</span><span class="w"> </span><span class="n">MSBB_tmp</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">head</span><span class="p">(</span><span class="n">multi_gene_res</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 × 14</caption>
<thead>
	<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>a1</th><th scope=col>a2</th><th scope=col>variant_ID</th><th scope=col>gene_ID</th><th scope=col>event_ID</th><th scope=col>cs_coverage_0.95</th><th scope=col>cs_coverage_0.95_purity0.5</th><th scope=col>PIP</th><th scope=col>conditional_effect</th><th scope=col>lfsr</th><th scope=col>resource</th></tr>
	<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>1</td><td>1303111</td><td>1303112</td><td>A</td><td>G</td><td>chr1:1303112:G:A</td><td>chr1_0_7520074</td><td>ENSG00000242485</td><td>10</td><td>10</td><td>0.30181490</td><td>0.213604181524862</td><td>3.79477490270115e-10</td><td>AC_DeJager_eQTL</td></tr>
	<tr><td>1</td><td>1303799</td><td>1303800</td><td>A</td><td>G</td><td>chr1:1303800:G:A</td><td>chr1_0_7520074</td><td>ENSG00000242485</td><td>10</td><td>10</td><td>0.20795258</td><td>0.21639498867546 </td><td>3.79477490270115e-10</td><td>AC_DeJager_eQTL</td></tr>
	<tr><td>1</td><td>1304554</td><td>1304555</td><td>C</td><td>A</td><td>chr1:1304555:A:C</td><td>chr1_0_7520074</td><td>ENSG00000242485</td><td>10</td><td>10</td><td>0.20795258</td><td>0.21639498867546 </td><td>3.79477490270115e-10</td><td>AC_DeJager_eQTL</td></tr>
	<tr><td>1</td><td>1325752</td><td>1325753</td><td>T</td><td>G</td><td>chr1:1325753:G:T</td><td>chr1_0_7520074</td><td>ENSG00000242485</td><td>10</td><td>10</td><td>0.03537254</td><td>0.213134189797448</td><td>3.79477490270115e-10</td><td>AC_DeJager_eQTL</td></tr>
	<tr><td>1</td><td>1344047</td><td>1344048</td><td>A</td><td>G</td><td>chr1:1344048:G:A</td><td>chr1_0_7520074</td><td>ENSG00000242485</td><td>10</td><td>10</td><td>0.05459853</td><td>0.214012593164653</td><td>3.79477490270115e-10</td><td>AC_DeJager_eQTL</td></tr>
	<tr><td>1</td><td>1364252</td><td>1364253</td><td>G</td><td>C</td><td>chr1:1364253:C:G</td><td>chr1_0_7520074</td><td>ENSG00000242485</td><td>10</td><td>10</td><td>0.03961189</td><td>0.211097094914644</td><td>3.79477490270115e-10</td><td>AC_DeJager_eQTL</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#gwas &lt;- fread(&#39;/data/interactive_analysis/rl3328/image_QTL/AD_GWAS_finemapping/export/output/extracted_info/109_blocks_top_loci.csv.gz&#39;) ## From Ruixi on Mar19, 2025</span>
<span class="n">gwas</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/Staging/variant_gene_summary/AD_xQTL_integration_summary/AD_GWAS_finemapping_109_blocks_top_loci_unified_any0.8ANDmin0.5.csv.gz&#39;</span><span class="p">)</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95_min_corr&quot;</span>
<span class="n">gwas_95</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gwas</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="o">!!</span><span class="n">coverage</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gwas_95</span><span class="w"> </span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="n">region</span><span class="p">,</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]],</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;:&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">variant_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;chr&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">variant_id</span><span class="p">))</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;AD_Bellenguez_2022:chr15_47690138_49901356:1&quot;</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">pip</span><span class="o">==</span><span class="m">0.998973148941477</span><span class="p">))</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">multigene_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">multi_gene_res</span><span class="o">%&gt;%</span>
<span class="w">                                    </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="p">,</span><span class="w">  </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;variant_ID&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">suffixes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;QTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;GWAS&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">head</span><span class="p">(</span><span class="n">multigene_AD_GWAS_meta_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 × 23</caption>
<thead>
	<tr><th scope=col>variant_ID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>a1</th><th scope=col>a2</th><th scope=col>gene_ID</th><th scope=col>event_ID</th><th scope=col>cs_coverage_0.95</th><th scope=col>cs_coverage_0.95_purity0.5</th><th scope=col>⋯</th><th scope=col>resource</th><th scope=col>study</th><th scope=col>variant_id</th><th scope=col>region</th><th scope=col>cs_coverage_0.95_min_corr</th><th scope=col>cs_coverage_0.7_min_corr</th><th scope=col>cs_coverage_0.5_min_corr</th><th scope=col>pip</th><th scope=col>z</th><th scope=col>gwas_cs</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr15:63277703:C:T</td><td>15</td><td>63277702</td><td>63277703</td><td>T</td><td>C</td><td>chr15_58574103_63343138</td><td>ENSG00000138613</td><td>2</td><td>2</td><td>⋯</td><td>AC_DeJager_eQTL</td><td>AD_Bellenguez_2022               </td><td>15:63277703:C:T</td><td>chr15_63051119_66680537</td><td>1</td><td>1</td><td>1</td><td>0.9813510</td><td>9.495798</td><td>AD_Bellenguez_2022:chr15_63051119_66680537:1               </td></tr>
	<tr><td>chr15:63277703:C:T</td><td>15</td><td>63277702</td><td>63277703</td><td>T</td><td>C</td><td>chr15_58574103_63343138</td><td>ENSG00000138613</td><td>2</td><td>2</td><td>⋯</td><td>AC_DeJager_eQTL</td><td>AD_Wightman_Excluding23andMe_2021</td><td>15:63277703:C:T</td><td>chr15_63051119_66680537</td><td>1</td><td>1</td><td>1</td><td>0.9148197</td><td>6.436907</td><td>AD_Wightman_Excluding23andMe_2021:chr15_63051119_66680537:1</td></tr>
	<tr><td>chr15:63277703:C:T</td><td>15</td><td>63277702</td><td>63277703</td><td>T</td><td>C</td><td>chr15_58574103_63343138</td><td>ENSG00000138613</td><td>2</td><td>2</td><td>⋯</td><td>AC_DeJager_eQTL</td><td>AD_Wightman_Full_2021            </td><td>15:63277703:C:T</td><td>chr15_63051119_66680537</td><td>1</td><td>1</td><td>1</td><td>0.8383659</td><td>6.857665</td><td>AD_Wightman_Full_2021:chr15_63051119_66680537:1            </td></tr>
	<tr><td>chr15:63277703:C:T</td><td>15</td><td>63277702</td><td>63277703</td><td>T</td><td>C</td><td>chr15_60834681_64158021</td><td>ENSG00000138613</td><td>2</td><td>2</td><td>⋯</td><td>AC_DeJager_eQTL</td><td>AD_Bellenguez_2022               </td><td>15:63277703:C:T</td><td>chr15_63051119_66680537</td><td>1</td><td>1</td><td>1</td><td>0.9813510</td><td>9.495798</td><td>AD_Bellenguez_2022:chr15_63051119_66680537:1               </td></tr>
	<tr><td>chr15:63277703:C:T</td><td>15</td><td>63277702</td><td>63277703</td><td>T</td><td>C</td><td>chr15_60834681_64158021</td><td>ENSG00000138613</td><td>2</td><td>2</td><td>⋯</td><td>AC_DeJager_eQTL</td><td>AD_Wightman_Excluding23andMe_2021</td><td>15:63277703:C:T</td><td>chr15_63051119_66680537</td><td>1</td><td>1</td><td>1</td><td>0.9148197</td><td>6.436907</td><td>AD_Wightman_Excluding23andMe_2021:chr15_63051119_66680537:1</td></tr>
	<tr><td>chr15:63277703:C:T</td><td>15</td><td>63277702</td><td>63277703</td><td>T</td><td>C</td><td>chr15_60834681_64158021</td><td>ENSG00000138613</td><td>2</td><td>2</td><td>⋯</td><td>AC_DeJager_eQTL</td><td>AD_Wightman_Full_2021            </td><td>15:63277703:C:T</td><td>chr15_63051119_66680537</td><td>1</td><td>1</td><td>1</td><td>0.8383659</td><td>6.857665</td><td>AD_Wightman_Full_2021:chr15_63051119_66680537:1            </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multigene_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">saveRDS</span><span class="p">(</span><span class="s">&quot;multigene_AD_GWAS_meta_df.rds&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">colnames</span><span class="p">(</span><span class="n">multigene_AD_GWAS_meta_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'variant_ID'</li><li>'#chr'</li><li>'start'</li><li>'end'</li><li>'a1'</li><li>'a2'</li><li>'gene_ID'</li><li>'event_ID'</li><li>'cs_coverage_0.95'</li><li>'cs_coverage_0.95_purity0.5'</li><li>'PIP'</li><li>'conditional_effect'</li><li>'lfsr'</li><li>'resource'</li><li>'study'</li><li>'variant_id'</li><li>'region'</li><li>'cs_coverage_0.95_min_corr'</li><li>'cs_coverage_0.7_min_corr'</li><li>'cs_coverage_0.5_min_corr'</li><li>'pip'</li><li>'z'</li><li>'gwas_cs'</li></ol>
</div></div>
</div>
</section>
<section id="summarize-the-coloboost-results">
<h4>summarize the coloboost results<a class="headerlink" href="#summarize-the-coloboost-results" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="n">cb_files</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list.files</span><span class="p">(</span><span class="s">&#39;/Staging/variant_gene_summary/AD_loci_summary/AD_xQTL_colocalization/&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;bed.gz$&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">.</span><span class="p">[</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;AD_xQTL_&#39;</span><span class="p">)]</span><span class="w"> </span><span class="c1"># data upadted on Sep 3rd, 2025</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Attaching package: ‘dplyr’


The following objects are masked from ‘package:data.table’:

    between, first, last


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">cb_files</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">cb_AD_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">()</span>
<span class="kr">for</span><span class="p">(</span><span class="n">cb_file</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">cb_files</span><span class="p">){</span>
<span class="w">    </span><span class="n">cb_AD_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">cb_AD_res</span><span class="p">,</span><span class="w"> </span>
<span class="w">                      </span><span class="nf">fread</span><span class="p">(</span><span class="n">cb_file</span><span class="p">))</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Define your tag patterns as before</span>
<span class="n">tag_patterns</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span>
<span class="w">  </span><span class="s">&#39;Knight_eQTL_brain&#39;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;KNIGHT_eQTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="c1"># left single right coloc id</span>
<span class="w">  </span><span class="s">&#39;Knight_pQTL_brain&#39;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;KNIGHT_pQTL&#39;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&#39;MiGA_THA_eQTL&#39;</span><span class="o">=</span><span class="w"> </span><span class="s">&#39;MIGA_THA&#39;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&#39;MiGA_GFM_eQTL&#39;</span><span class="o">=</span><span class="w"> </span><span class="s">&#39;MIGA_GFM&#39;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&#39;MiGA_GTS_eQTL&#39;</span><span class="o">=</span><span class="w"> </span><span class="s">&#39;MIGA_GTS&#39;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&#39;MiGA_SVZ_eQTL&#39;</span><span class="o">=</span><span class="w"> </span><span class="s">&#39;MIGA_SVZ&#39;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&#39;BM_10_MSBB_eQTL&#39;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;MSBB_eQTL_BM_10&#39;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&#39;BM_22_MSBB_eQTL&#39;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;MSBB_eQTL_BM_22&#39;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&#39;BM_36_MSBB_eQTL&#39;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;MSBB_eQTL_BM_36&#39;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&#39;BM_44_MSBB_eQTL&#39;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;MSBB_eQTL_BM_44&#39;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&#39;BM_36_MSBB_pQTL&#39;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;MSBB_pQTL_BM_36&#39;</span><span class="p">,</span>

<span class="w">   </span><span class="c1"># ROSMAP</span>
<span class="w">  </span><span class="s">&#39;DLPFC_Klein_gpQTL_adjusted&#39;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;DLPFC_Klein_gpQTL_adjusted&#39;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&#39;DLPFC_Klein_gpQTL_adjusted&#39;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;DLPFC_Klein_gpQTL_unadjusted&#39;</span><span class="p">,</span><span class="w"> </span><span class="c1">## please double check this line above </span>

<span class="w">  </span><span class="s">&quot;ROSMAP_AC_sQTL_NE&quot;</span><span class="w">    </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^AC_.*sQTL.*NE&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;ROSMAP_AC_sQTL_UP&quot;</span><span class="w">    </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^AC_.*sQTL.*UP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;ROSMAP_AC_sQTL_PR&quot;</span><span class="w">    </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^AC_.*sQTL.*PR&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;ROSMAP_PCC_sQTL_NE&quot;</span><span class="w">   </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^PCC_.*sQTL.*NE&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;ROSMAP_PCC_sQTL_UP&quot;</span><span class="w">   </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^PCC_.*sQTL.*UP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;ROSMAP_PCC_sQTL_PR&quot;</span><span class="w">   </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^PCC_.*sQTL.*PR&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;ROSMAP_DLPFC_sQTL_NE&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^DLPFC_.*sQTL.*NE&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;ROSMAP_DLPFC_sQTL_UP&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^DLPFC_.*sQTL.*UP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;ROSMAP_DLPFC_sQTL_PR&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;^DLPFC_.*sQTL.*PR&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="c1"># General patterns fallback</span>
<span class="w">  </span><span class="s">&quot;Mic_DeJager_eQTL&quot;</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;DeJager_Mic&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Mic_mega_eQTL&quot;</span><span class="w">        </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;mega_Mic&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Mic_Kellis_eQTL&quot;</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Kellis_Mic&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Inh_Kellis_eQTL&quot;</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Kellis_Inh&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Inh_DeJager_eQTL&quot;</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;DeJager_Inh&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;AC_DeJager_eQTL&quot;</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_AC&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Oli_Kellis_eQTL&quot;</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Kellis_Oli&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Exc_Kellis_eQTL&quot;</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Kellis_Exc&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Oli_mega_eQTL&quot;</span><span class="w">        </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;mega_Oli&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Exc_mega_eQTL&quot;</span><span class="w">        </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;mega_Exc&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Exc_DeJager_eQTL&quot;</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;DeJager_Exc&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Inh_mega_eQTL&quot;</span><span class="w">        </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;mega_Inh&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Ast_DeJager_eQTL&quot;</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;DeJager_Ast&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Ast_mega_eQTL&quot;</span><span class="w">        </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;mega_Ast&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;DLPFC_DeJager_eQTL&quot;</span><span class="w">   </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_DLPFC&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Oli_DeJager_eQTL&quot;</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;DeJager_Oli&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Ast_Kellis_eQTL&quot;</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Kellis_Ast&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;OPC_Kellis_eQTL&quot;</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Kellis_OPC&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;Ast_DeJager_eQTL&quot;</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;DeJager_Ast&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;OPC_mega_eQTL&quot;</span><span class="w">        </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;mega_OPC&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;OPC_DeJager_eQTL&quot;</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;DeJager_OPC&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;PCC_DeJager_eQTL&quot;</span><span class="w">     </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_PCC&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;DLPFC_Bennett_pQTL&quot;</span><span class="w">   </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;DLPFC_Bennett_pQTL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="s">&quot;monocyte_ROSMAP_eQTL&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ROSMAP_Monocyte&quot;</span>
<span class="p">)</span>

<span class="c1"># Function to extract tags per row</span>
<span class="n">extract_tags</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">event_id_str</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">substrings</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unlist</span><span class="p">(</span><span class="nf">str_split</span><span class="p">(</span><span class="n">event_id_str</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Match predefined tags</span>
<span class="w">  </span><span class="n">mapped_tags</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unlist</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">substrings</span><span class="p">,</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">sub</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">names</span><span class="p">(</span><span class="n">tag_patterns</span><span class="p">)[</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">sub</span><span class="p">,</span><span class="w"> </span><span class="n">tag_patterns</span><span class="p">)]</span>
<span class="w">  </span><span class="p">}))</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Also extract raw AD-related tags</span>
<span class="w">  </span><span class="n">ad_tags</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">substrings</span><span class="p">[</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">substrings</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AD&quot;</span><span class="p">)]</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Combine both, remove duplicates, sort optionally</span>
<span class="w">  </span><span class="n">final_tags</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">mapped_tags</span><span class="p">,</span><span class="w"> </span><span class="n">ad_tags</span><span class="p">)))</span>
<span class="w">  </span><span class="nf">paste</span><span class="p">(</span><span class="n">final_tags</span><span class="p">,</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;, &quot;</span><span class="p">)</span>
<span class="p">}</span>

<span class="c1"># Apply to your dataframe</span>
<span class="n">cb_AD_res</span><span class="o">$</span><span class="n">contexts</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sapply</span><span class="p">(</span><span class="n">cb_AD_res</span><span class="o">$</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">extract_tags</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">purrr</span><span class="p">)</span>

<span class="n">cb_AD_res_study</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cb_AD_res</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="c1"># Split contexts string into list, and trim each element</span>
<span class="w">    </span><span class="n">context_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="n">contexts</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;,\\s*&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">context_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">map</span><span class="p">(</span><span class="n">context_list</span><span class="p">,</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">str_trim</span><span class="p">(</span><span class="n">.x</span><span class="p">)),</span><span class="w">  </span><span class="c1"># Trim whitespace</span>

<span class="w">    </span><span class="c1"># Extract AD-related context</span>
<span class="w">    </span><span class="n">study</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">map_chr</span><span class="p">(</span><span class="n">context_list</span><span class="p">,</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">.x</span><span class="p">[</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">.x</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;^AD_&quot;</span><span class="p">)],</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;, &quot;</span><span class="p">)),</span>

<span class="w">    </span><span class="c1"># Remove AD-related context</span>
<span class="w">    </span><span class="n">context_cleaned</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">map_chr</span><span class="p">(</span><span class="n">context_list</span><span class="p">,</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">.x</span><span class="p">[</span><span class="o">!</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">.x</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;^AD_&quot;</span><span class="p">)],</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;, &quot;</span><span class="p">))</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="n">context_list</span><span class="p">,</span><span class="n">contexts</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">rename</span><span class="p">(</span><span class="n">contexts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">context_cleaned</span><span class="p">)</span>

<span class="c1"># Check the result</span>
<span class="nf">head</span><span class="p">(</span><span class="n">cb_AD_res_study</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="m">2</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Attaching package: ‘purrr’


The following object is masked from ‘package:data.table’:

    transpose
</pre></div>
</div>
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 2 × 15</caption>
<thead>
	<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>a1</th><th scope=col>a2</th><th scope=col>variant_ID</th><th scope=col>region_ID</th><th scope=col>event_ID</th><th scope=col>cos_ID</th><th scope=col>vcp</th><th scope=col>cos_npc</th><th scope=col>min_npc_outcome</th><th scope=col>coef</th><th scope=col>study</th><th scope=col>contexts</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr1</td><td>207573951</td><td>207573951</td><td>G</td><td>A</td><td>chr1:207573951:A:G</td><td>ENSG00000203710</td><td>KNIGHT_eQTL_ENSG00000203710; AD_Bellenguez        </td><td>ENSG00000203710:cos1:y1_y2</td><td>0.02131227</td><td>0.9741</td><td>0.9741</td><td>-0.0113326428446495;-0.000118137670560767</td><td>AD_Bellenguez        </td><td>Knight_eQTL_brain</td></tr>
	<tr><td>chr1</td><td>207573951</td><td>207573951</td><td>G</td><td>A</td><td>chr1:207573951:A:G</td><td>ENSG00000203710</td><td>KNIGHT_eQTL_ENSG00000203710; AD_Kunkle_Stage1_2019</td><td>ENSG00000203710:cos1:y1_y2</td><td>0.05200296</td><td>0.9820</td><td>0.9818</td><td>-0.0113432399994088;-0.00164700157001807 </td><td>AD_Kunkle_Stage1_2019</td><td>Knight_eQTL_brain</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#cb_AD_res &lt;- readRDS(&quot;cb_AD_res_contexts.rds&quot;)</span>
<span class="n">gwas</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/Staging/variant_gene_summary/AD_xQTL_integration_summary/AD_GWAS_finemapping_109_blocks_top_loci_unified_any0.8ANDmin0.5.csv.gz&#39;</span><span class="p">)</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95_min_corr&quot;</span>
<span class="n">gwas_95</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gwas</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="o">!!</span><span class="n">coverage</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gwas_95</span><span class="w"> </span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="n">region</span><span class="p">,</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]],</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;:&quot;</span><span class="p">))</span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">variant_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;chr&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">variant_id</span><span class="p">))</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;AD_Bellenguez_2022:chr15_47690138_49901356:1&quot;</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">pip</span><span class="o">==</span><span class="m">0.998973148941477</span><span class="p">))</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">cb_AD_combined_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">cb_AD_res_study</span><span class="p">,</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="p">,</span><span class="w">  </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;variant_ID&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">suffixes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;QTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;GWAS&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#cb_AD_combined_AD_GWAS_meta_df &lt;- readRDS(&quot;cb_AD_combined_AD_GWAS_meta_df.rds&quot;)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#head(cb_AD_combined_AD_GWAS_meta_df)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">cb_AD_combined_AD_GWAS_meta_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'variant_ID'</li><li>'#chr'</li><li>'start'</li><li>'end'</li><li>'a1'</li><li>'a2'</li><li>'region_ID'</li><li>'event_ID'</li><li>'cos_ID'</li><li>'vcp'</li><li>'cos_npc'</li><li>'min_npc_outcome'</li><li>'coef'</li><li>'studyQTL'</li><li>'contexts'</li><li>'studyGWAS'</li><li>'variant_id'</li><li>'region'</li><li>'cs_coverage_0.95_min_corr'</li><li>'cs_coverage_0.7_min_corr'</li><li>'cs_coverage_0.5_min_corr'</li><li>'pip'</li><li>'z'</li><li>'gwas_cs'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">cb_AD_combined_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cb_AD_combined_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">rename</span><span class="p">(</span><span class="n">study</span><span class="o">=</span><span class="n">studyGWAS</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">colnames</span><span class="p">(</span><span class="n">cb_AD_combined_AD_GWAS_meta_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'variant_ID'</li><li>'#chr'</li><li>'start'</li><li>'end'</li><li>'a1'</li><li>'a2'</li><li>'region_ID'</li><li>'event_ID'</li><li>'cos_ID'</li><li>'vcp'</li><li>'cos_npc'</li><li>'min_npc_outcome'</li><li>'coef'</li><li>'studyQTL'</li><li>'contexts'</li><li>'study'</li><li>'variant_id'</li><li>'region'</li><li>'cs_coverage_0.95_min_corr'</li><li>'cs_coverage_0.7_min_corr'</li><li>'cs_coverage_0.5_min_corr'</li><li>'pip'</li><li>'z'</li><li>'gwas_cs'</li></ol>
</div></div>
</div>
</section>
<section id="summarize-trans-results">
<h4>summarize trans results<a class="headerlink" href="#summarize-trans-results" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="n">trans_sing_df_raw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/data/analysis_result/trans/FunGen_xQTL.trans.nofiltered.exported.toploci.bed.gz&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">p</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="nf">pnorm</span><span class="p">(</span><span class="o">-</span><span class="nf">abs</span><span class="p">(</span><span class="n">z</span><span class="p">)))</span><span class="w"> </span><span class="c1">#not on synapse, to be updated</span>

<span class="n">trans_sing_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trans_sing_df_raw</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">MAF</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span><span class="w"> </span><span class="n">p</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">1E-05</span><span class="p">)</span><span class="w"> </span><span class="c1"># no MIGA </span>
<span class="n">trans_sing_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trans_sing_df</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="c1"># resource = str_extract(event_ID, &quot;^[^_]+_[^_]+&quot;),  # Remove last part (gene name) to get context</span>
<span class="w">    </span><span class="n">gene_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_extract</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;[^_]+$&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1"># Extract last part (gene name)</span>
<span class="w">  </span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">gene_name</span><span class="p">)</span>
<span class="c1">#gwas &lt;- fread(&#39;/data/interactive_analysis/rl3328/image_QTL/AD_GWAS_finemapping/export/output/extracted_info/109_blocks_top_loci.csv.gz&#39;) ## From Ruixi on Mar19, 2025</span>
<span class="n">gwas</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/Staging/variant_gene_summary/AD_xQTL_integration_summary/AD_GWAS_finemapping_109_blocks_top_loci_unified_any0.8ANDmin0.5.csv.gz&#39;</span><span class="p">)</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95_min_corr&quot;</span>
<span class="n">gwas_95</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gwas</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="o">!!</span><span class="n">coverage</span><span class="w"> </span><span class="o">:=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gwas_95</span><span class="w"> </span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="n">region</span><span class="p">,</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]],</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;:&quot;</span><span class="p">))</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">variant_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;chr&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">variant_id</span><span class="p">))</span>
<span class="n">AD_gwas_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">gwas_cs</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;AD_Bellenguez_2022:chr15_47690138_49901356:1&quot;</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">pip</span><span class="o">==</span><span class="m">0.998973148941477</span><span class="p">))</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">trans_sing_df</span><span class="o">%&gt;%</span>
<span class="w">                                    </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="n">AD_gwas_df</span><span class="p">,</span><span class="w">  </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;variant_ID&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">suffixes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;QTL&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;GWAS&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_sing_df_raw</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
<span class="n">trans_sing_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>5724704</li><li>19</li></ol>
</div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>3321438</li><li>19</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#cumsum pip &gt; 0.5</span>
<span class="n">trans_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trans_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="c1"># create a unique cs_id by combining gene_ID, event_ID, and cs_coverage_0.95</span>
<span class="w">    </span><span class="n">cs_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">cs_coverage_0.95</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="p">)</span>
<span class="n">trans_AD_GWAS_meta_df_cumsum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trans_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">cs_id</span><span class="p">,</span><span class="w"> </span><span class="n">gwas_cs</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">summarise</span><span class="p">(</span>
<span class="w">    </span><span class="c1"># sum PIP within each cs_id group</span>
<span class="w">    </span><span class="n">PIP_sum</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">PIP</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span>
<span class="w">    </span><span class="n">.groups</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;drop&quot;</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">PIP_sum</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">left_join</span><span class="p">(</span><span class="n">trans_AD_GWAS_meta_df</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Joining with `by = join_by(cs_id, gwas_cs)`
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
<span class="n">trans_AD_GWAS_meta_df_cumsum</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dim</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>13523</li><li>29</li></ol>
</div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>12952</li><li>30</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_95_info_raw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trans_sing_df_raw</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">cs_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">cs_coverage_0.95</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_95_info_raw</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">cs_id</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">223017</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_95_info</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trans_sing_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">cs_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">cs_coverage_0.95</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_95_info</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">cs_id</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">84787</div></div>
</div>
</section>
<section id="summarize-fsusie-results">
<h4>summarize fsusie results<a class="headerlink" href="#summarize-fsusie-results" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">fsusie_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;/Staging/variant_gene_summary/xQTL_loci_summary/xQTL_finemapping/single_context_finemapping/consolidated/AD_loci_xQTL_context_method_summary.fsusie.rds&quot;</span><span class="p">)</span>
<span class="n">snATAC</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;ROSMAP_snATAC_Kelis_Ast&#39;</span><span class="p">,</span><span class="s">&#39;ROSMAP_snATAC_Kelis_Exc&#39;</span><span class="p">,</span><span class="s">&#39;ROSMAP_snATAC_Kelis_Inh&#39;</span><span class="p">,</span><span class="s">&#39;ROSMAP_snATAC_Kelis_Mic&#39;</span><span class="p">,</span><span class="s">&#39;ROSMAP_snATAC_Kelis_OPC&#39;</span><span class="p">,</span><span class="s">&#39;ROSMAP_snATAC_Kelis_Oli&#39;</span><span class="p">)</span>
<span class="n">fsusie_AD_GWAS_meta_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fsusie_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;builk_fsusie_coloc&quot;</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="n">event_ID</span><span class="o">%in%</span><span class="w"> </span><span class="n">snATAC</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">resource</span><span class="p">)</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;single_context and single_gene&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<section id="combine-all">
<h5>combine all<a class="headerlink" href="#combine-all" title="Link to this heading">#</a></h5>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">extracted_colnames</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;variant_ID&#39;</span><span class="p">,</span><span class="s">&#39;#chr&#39;</span><span class="p">,</span><span class="s">&#39;start&#39;</span><span class="p">,</span><span class="s">&#39;end&#39;</span><span class="p">,</span><span class="s">&#39;a1&#39;</span><span class="p">,</span><span class="s">&#39;a2&#39;</span><span class="p">,</span><span class="s">&#39;gene_ID&#39;</span><span class="p">,</span><span class="s">&#39;event_ID&#39;</span><span class="p">,</span><span class="s">&#39;variant_id&#39;</span><span class="p">,</span><span class="s">&#39;z&#39;</span><span class="p">,</span><span class="s">&#39;pip&#39;</span><span class="p">,</span><span class="s">&#39;study&#39;</span><span class="p">,</span><span class="s">&#39;region&#39;</span><span class="p">,</span><span class="s">&#39;gwas_cs&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">all_95_95</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plyr</span><span class="o">::</span><span class="nf">rbind.fill</span><span class="p">(</span><span class="n">single_context_AD_GWAS_meta_df</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="nf">any_of</span><span class="p">(</span><span class="n">extracted_colnames</span><span class="p">))</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;single_context&#39;</span><span class="p">),</span>
<span class="w">                   </span><span class="n">multi_context_AD_GWAS_meta_df</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="nf">any_of</span><span class="p">(</span><span class="n">extracted_colnames</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;multi_context&#39;</span><span class="p">),</span>
<span class="w">                   </span><span class="n">metabrain_AD_GWAS_meta_df</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="nf">any_of</span><span class="p">(</span><span class="n">extracted_colnames</span><span class="p">))</span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;single_context&#39;</span><span class="p">),</span>
<span class="w">                   </span><span class="n">trans_AD_GWAS_meta_df_cumsum</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="nf">any_of</span><span class="p">(</span><span class="n">extracted_colnames</span><span class="p">))</span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;trans&#39;</span><span class="p">),</span>
<span class="w">                   </span><span class="n">fsusie_AD_GWAS_meta_df</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="nf">any_of</span><span class="p">(</span><span class="n">extracted_colnames</span><span class="p">))</span><span class="o">%&gt;%</span><span class="nf">mutate</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;epigenomic&#39;</span><span class="p">),</span>
<span class="w">                   </span><span class="n">cb_AD_combined_AD_GWAS_meta_df</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="nf">any_of</span><span class="p">(</span><span class="n">extracted_colnames</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;multi_context&#39;</span><span class="p">),</span>
<span class="w">                   </span><span class="n">multigene_AD_GWAS_meta_df</span><span class="o">%&gt;%</span><span class="nf">select</span><span class="p">(</span><span class="nf">any_of</span><span class="p">(</span><span class="n">extracted_colnames</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;multi_gene&#39;</span><span class="p">))</span>
<span class="n">all_95_95_cs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">all_95_95</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gwas_cs</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">summarise</span><span class="p">(</span><span class="n">resources</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">resource</span><span class="p">),</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">))</span>
<span class="n">all_95_95_cs_explain</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">AD_gwas_df</span><span class="p">,</span><span class="w"> </span><span class="n">all_95_95_cs</span><span class="p">,</span><span class="w"> </span><span class="n">all.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="n">gwas_cs</span><span class="p">,</span><span class="w"> </span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="n">resources</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">distinct</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">all_95_95</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">saveRDS</span><span class="p">(</span><span class="s">&quot;all_methods_95_df.rds&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">saveRDS</span><span class="p">(</span><span class="n">all_95_95_cs_explain</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;method_explain_AD_loci.rds&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">all_95_95_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">saveRDS</span><span class="p">(</span><span class="s">&quot;AD_loci_explained_by_methods.rds&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">all_95_95_cs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;AD_loci_explained_by_methods.rds&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># all_95_95 &lt;- rbind(single_context_AD_GWAS_meta_df%&gt;%select(any_of(extracted_colnames))%&gt;% mutate(resource = &#39;single_context&#39;),</span>
<span class="c1">#                    multi_context_AD_GWAS_meta_df%&gt;%select(any_of(extracted_colnames)) %&gt;% mutate(resource = &#39;multi_context&#39;),</span>
<span class="c1">#                    metabrain_AD_GWAS_meta_df%&gt;%select(any_of(extracted_colnames))%&gt;%mutate(resource = &#39;single_context&#39;),</span>
<span class="c1">#                    cb_AD_combined_AD_GWAS_meta_df%&gt;%select(any_of(extracted_colnames)) %&gt;% mutate(resource = &#39;ColocBoost&#39;),</span>
<span class="c1">#                    multigene_AD_GWAS_meta_df%&gt;%select(any_of(extracted_colnames)) %&gt;% mutate(resource = &#39;multi_gene&#39;), fill=T)</span>
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="plot-susie-upsetr">
<h4>plot susie upsetR<a class="headerlink" href="#plot-susie-upsetr" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">head</span><span class="p">(</span><span class="n">all_95_95</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 15</caption>
<thead>
	<tr><th></th><th scope=col>variant_ID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>a1</th><th scope=col>a2</th><th scope=col>gene_ID</th><th scope=col>event_ID</th><th scope=col>variant_id</th><th scope=col>z</th><th scope=col>pip</th><th scope=col>study</th><th scope=col>region</th><th scope=col>gwas_cs</th><th scope=col>resource</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>1</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>Mic_mega_eQTL   </td><td>10:11678309:A:G</td><td>      NA</td><td>        NA</td><td>AD_Bellenguez_2022               </td><td>chr10_10500888_12817813</td><td>AD_Bellenguez_2022:chr10_10500888_12817813:1               </td><td>single_context</td></tr>
	<tr><th scope=row>2</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>Mic_mega_eQTL   </td><td>10:11678309:A:G</td><td>5.645878</td><td>0.18080328</td><td>AD_Jansen_2021                   </td><td>chr10_10500888_12817813</td><td>AD_Jansen_2021:chr10_10500888_12817813:1                   </td><td>single_context</td></tr>
	<tr><th scope=row>3</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>Mic_mega_eQTL   </td><td>10:11678309:A:G</td><td>7.374065</td><td>0.05051167</td><td>AD_Wightman_Full_2021            </td><td>chr10_10500888_12817813</td><td>AD_Wightman_Full_2021:chr10_10500888_12817813:1            </td><td>single_context</td></tr>
	<tr><th scope=row>4</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>Mic_mega_eQTL   </td><td>10:11678309:A:G</td><td>6.603571</td><td>0.49184952</td><td>AD_Wightman_Excluding23andMe_2021</td><td>chr10_10500888_12817813</td><td>AD_Wightman_Excluding23andMe_2021:chr10_10500888_12817813:1</td><td>single_context</td></tr>
	<tr><th scope=row>5</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>Mic_DeJager_eQTL</td><td>10:11678309:A:G</td><td>      NA</td><td>        NA</td><td>AD_Bellenguez_2022               </td><td>chr10_10500888_12817813</td><td>AD_Bellenguez_2022:chr10_10500888_12817813:1               </td><td>single_context</td></tr>
	<tr><th scope=row>6</th><td>chr10:11678309:A:G</td><td>10</td><td>11678308</td><td>11678309</td><td>G</td><td>A</td><td>ENSG00000148429</td><td>Mic_DeJager_eQTL</td><td>10:11678309:A:G</td><td>5.645878</td><td>0.18080328</td><td>AD_Jansen_2021                   </td><td>chr10_10500888_12817813</td><td>AD_Jansen_2021:chr10_10500888_12817813:1                   </td><td>single_context</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">all_95_95_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A tibble: 6 × 2</caption>
<thead>
	<tr><th scope=col>gwas_cs</th><th scope=col>resources</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>AD_Bellenguez_2022:chr10_10500888_12817813:1  </td><td>single_context,multi_context,trans           </td></tr>
	<tr><td>AD_Bellenguez_2022:chr10_57667362_59558848:1  </td><td>single_context,multi_context,trans,epigenomic</td></tr>
	<tr><td>AD_Bellenguez_2022:chr10_59558848_62446953:1  </td><td>single_context,multi_context                 </td></tr>
	<tr><td>AD_Bellenguez_2022:chr10_80126158_82231647:1  </td><td>single_context,multi_context                 </td></tr>
	<tr><td>AD_Bellenguez_2022:chr10_95280986_98320874:1  </td><td>single_context,multi_context                 </td></tr>
	<tr><td>AD_Bellenguez_2022:chr11_119829118_122612025:1</td><td>single_context,multi_context                 </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">all_95_95_cs_explain</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;method_explain_AD_loci.rds&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="n">all_95_95_cs_explain</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">resources</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">table</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>.
                                           multi_context 
                                                      15 
                                multi_context,epigenomic 
                                                       3 
                                     multi_context,trans 
                                                       2 
                                          single_context 
                                                       7 
                            single_context,multi_context 
                                                      38 
                 single_context,multi_context,multi_gene 
                                                       2 
                      single_context,multi_context,trans 
                                                      27 
           single_context,multi_context,trans,epigenomic 
                                                       1 
single_context,multi_context,trans,epigenomic,multi_gene 
                                                       6 
           single_context,multi_context,trans,multi_gene 
                                                      11 
                                    single_context,trans 
                                                       4 
                      single_context,trans,multi_context 
                                                       6 
                                                   trans 
                                                      13 
                                     trans,multi_context 
                                                       8 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="n">all_95_95_cs_explain</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">resources</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">table</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>.
                                           multi_context 
                                                      15 
                                multi_context,epigenomic 
                                                       3 
                                     multi_context,trans 
                                                       2 
                                          single_context 
                                                       7 
                            single_context,multi_context 
                                                      38 
                 single_context,multi_context,multi_gene 
                                                       2 
                      single_context,multi_context,trans 
                                                      27 
           single_context,multi_context,trans,epigenomic 
                                                       1 
single_context,multi_context,trans,epigenomic,multi_gene 
                                                       6 
           single_context,multi_context,trans,multi_gene 
                                                      11 
                                    single_context,trans 
                                                       4 
                      single_context,trans,multi_context 
                                                       6 
                                                   trans 
                                                      13 
                                     trans,multi_context 
                                                       8 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">all_95_95_cs_explain</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">resources</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">unique</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'single_context,multi_context,trans'</li><li>'single_context,multi_context,trans,epigenomic'</li><li>'single_context,multi_context'</li><li>'trans'</li><li>'multi_context'</li><li>NA</li><li>'single_context,trans'</li><li>'multi_context,epigenomic'</li><li>'single_context,multi_context,trans,multi_gene'</li><li>'single_context,trans,multi_context'</li><li>'single_context'</li><li>'trans,multi_context'</li><li>'single_context,multi_context,trans,epigenomic,multi_gene'</li><li>'single_context,multi_context,multi_gene'</li><li>'multi_context,trans'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">all_95_95_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A tibble: 6 × 2</caption>
<thead>
	<tr><th scope=col>gwas_cs</th><th scope=col>resources</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>AD_Bellenguez_2022:chr10_10500888_12817813:1  </td><td>single_context,multi_context,trans           </td></tr>
	<tr><td>AD_Bellenguez_2022:chr10_57667362_59558848:1  </td><td>single_context,multi_context,trans,epigenomic</td></tr>
	<tr><td>AD_Bellenguez_2022:chr10_59558848_62446953:1  </td><td>single_context,multi_context                 </td></tr>
	<tr><td>AD_Bellenguez_2022:chr10_80126158_82231647:1  </td><td>single_context,multi_context                 </td></tr>
	<tr><td>AD_Bellenguez_2022:chr10_95280986_98320874:1  </td><td>single_context,multi_context                 </td></tr>
	<tr><td>AD_Bellenguez_2022:chr11_119829118_122612025:1</td><td>single_context,multi_context                 </td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="plot-the-stacked-bar-plot-only-four-categories">
<h4>plot the stacked bar plot (only four categories)<a class="headerlink" href="#plot-the-stacked-bar-plot-only-four-categories" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">scales</span><span class="p">)</span><span class="w">  </span><span class="c1"># For formatting percentages</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span>

<span class="n">resource_levels</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">all_95_95_cs_explain</span><span class="o">$</span><span class="n">resources</span><span class="p">)</span>

<span class="n">n_colors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">resource_levels</span><span class="p">)</span>

<span class="n">set2_colors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colorRampPalette</span><span class="p">(</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Set2&quot;</span><span class="p">))(</span><span class="n">n_colors</span><span class="p">)</span>

<span class="nf">names</span><span class="p">(</span><span class="n">set2_colors</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">resource_levels</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Attaching package: ‘scales’


The following object is masked from ‘package:purrr’:

    discard
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Aggregate data to count occurrences of each (study, resources) pair</span>
<span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">all_95_95_cs_explain</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="n">resources</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">summarise</span><span class="p">(</span><span class="n">count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">n</span><span class="p">(),</span><span class="w"> </span><span class="n">.groups</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;drop&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">resources</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">resources</span><span class="p">),</span><span class="w"> </span><span class="s">&#39;Not explained&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">resources</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="n">study</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;AD_Jansen_2021&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;AD_Bellenguez_EADI_2022&quot;</span><span class="p">))</span>

<span class="c1"># Compute NA percentage per study</span>
<span class="n">study_totals</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">study</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">summarise</span><span class="p">(</span><span class="n">total_count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">count</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span>
<span class="w">    </span><span class="n">plot_data</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">resources</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;Not explained&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">group_by</span><span class="p">(</span><span class="n">study</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">      </span><span class="nf">summarise</span><span class="p">(</span><span class="n">not_explained_count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">count</span><span class="p">),</span><span class="w"> </span><span class="n">.groups</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;drop&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;study&quot;</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="n">not_explained_count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">replace_na</span><span class="p">(</span><span class="n">not_explained_count</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">),</span>
<span class="w">    </span><span class="n">non_na_percentage</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">round</span><span class="p">(</span><span class="m">1</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">not_explained_count</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_count</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">),</span>
<span class="w">    </span><span class="n">unexplained_percentage</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">non_na_percentage</span><span class="w">  </span><span class="c1"># Calculate unexplained percentage</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="n">total_count</span><span class="p">,</span><span class="w"> </span><span class="n">non_na_percentage</span><span class="p">,</span><span class="w"> </span><span class="n">unexplained_percentage</span><span class="p">)</span>

<span class="c1"># Merge total counts back into plot data</span>
<span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">left_join</span><span class="p">(</span><span class="n">plot_data</span><span class="p">,</span><span class="w"> </span><span class="n">study_totals</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;study&quot;</span><span class="p">)</span>

<span class="c1"># Reorder studies by total count for better visualization</span>
<span class="n">study_order</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">study_totals</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">arrange</span><span class="p">(</span><span class="n">total_count</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">study</span><span class="p">)</span>
<span class="n">plot_data</span><span class="o">$</span><span class="n">study</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">study_order</span><span class="p">)</span>

<span class="c1"># Create the horizontal stacked bar plot</span>
<span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">32</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">12</span><span class="p">)</span><span class="w">  </span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">plot_data</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">study</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">resources</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;identity&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;stack&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.8</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="c1"># Add percentage labels (unexplained %)</span>
<span class="w">  </span><span class="nf">geom_text</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">study_totals</span><span class="p">,</span><span class="w"> </span>
<span class="w">            </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">total_count</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">1.02</span><span class="p">,</span><span class="w">  </span><span class="c1"># Position just to the right of the bar</span>
<span class="w">                </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">study</span><span class="p">,</span><span class="w"> </span>
<span class="w">                </span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">percent</span><span class="p">(</span><span class="n">unexplained_percentage</span><span class="p">,</span><span class="w"> </span><span class="n">accuracy</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)),</span><span class="w"> </span>
<span class="w">            </span><span class="n">inherit.aes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span>
<span class="w">            </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;grey20&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Number of loci&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">       </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Study&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">       </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Resources&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">       </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Percentage Explained by 95% CS AD Loci Across Methods&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span>
<span class="w">    </span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">38</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;candara&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">margin</span><span class="p">(</span><span class="n">b</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">)),</span>
<span class="w">    </span><span class="n">plot.title.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;plot&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">axis.title.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">32</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;candara&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">margin</span><span class="p">(</span><span class="n">t</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">50</span><span class="p">)),</span><span class="w">  </span><span class="c1"># X-axis title: &quot;Number of loci&quot;</span>
<span class="w">    </span><span class="n">axis.title.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">32</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">,</span><span class="w">  </span><span class="n">family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;candara&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">margin</span><span class="p">(</span><span class="n">r</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">50</span><span class="p">)),</span><span class="w">  </span><span class="c1"># Y-axis title: &quot;Study&quot;</span>
<span class="w">    </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">28</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">),</span><span class="w">  </span><span class="c1"># Larger text for study names</span>
<span class="w">    </span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">28</span><span class="p">),</span>
<span class="w">    </span><span class="n">axis.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">14</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">legend.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">24</span><span class="p">),</span>
<span class="w">    </span><span class="n">legend.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">24</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">axis.line.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_line</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">),</span>
<span class="w">    </span><span class="n">panel.grid.major.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_blank</span><span class="p">(),</span><span class="w">  </span><span class="c1"># Remove horizontal grid lines</span>
<span class="w">    </span><span class="n">panel.grid.minor.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_blank</span><span class="p">(),</span>
<span class="w">    </span><span class="n">panel.grid.major.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_line</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey80&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">panel.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">),</span>
<span class="w">    </span><span class="n">plot.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">),</span>
<span class="w">    </span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w">  </span><span class="nf">c</span><span class="p">(</span><span class="m">0.725</span><span class="p">,</span><span class="w"> </span><span class="m">0.45</span><span class="p">),</span>
<span class="w">    </span><span class="n">plot.margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">margin</span><span class="p">(</span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="m">20</span><span class="p">)</span><span class="w">  </span><span class="c1"># Add right margin for labels</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">   </span><span class="nf">scale_x_continuous</span><span class="p">(</span>
<span class="w">    </span><span class="n">breaks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="m">20</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="m">30</span><span class="p">,</span><span class="w"> </span><span class="m">40</span><span class="p">,</span><span class="w"> </span><span class="m">50</span><span class="p">,</span><span class="w"> </span><span class="m">60</span><span class="p">),</span>
<span class="w">    </span><span class="n">expand</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">expansion</span><span class="p">(</span><span class="n">mult</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0.15</span><span class="p">)),</span>
<span class="w">    </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">scales</span><span class="o">::</span><span class="n">comma</span>
<span class="w">  </span><span class="p">)</span><span class="w">  </span><span class="o">+</span><span class="w">  </span><span class="c1"># Add space for labels</span>
<span class="w">  </span><span class="nf">scale_fill_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">set2_colors</span><span class="p">)</span>

<span class="c1"># dev.off()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/d9909f31b73f8e61d699be11f8d88c5f3432b26ea91878369bff47b1ef5e86d6.png"><img alt="../../_images/d9909f31b73f8e61d699be11f8d88c5f3432b26ea91878369bff47b1ef5e86d6.png" src="../../_images/d9909f31b73f8e61d699be11f8d88c5f3432b26ea91878369bff47b1ef5e86d6.png" style="width: 1920px; height: 720px;" />
</a>
</div>
</div>
</section>
<section id="check-trans-only">
<h4>check trans only<a class="headerlink" href="#check-trans-only" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_only_gwas_cs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">all_95_95_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">resources</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;trans&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">left_join</span><span class="p">(</span><span class="n">all_95_95</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Joining with `by = join_by(gwas_cs)`
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">single_95_n_var</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="n">single_95_info</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
<span class="n">single_95_n_cs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">single_95_info</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">cs_id</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>

<span class="n">trans_95_raw_n_var</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="n">trans_95_info_raw</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
<span class="n">trans_95_raw_n_cs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">trans_95_info_raw</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">cs_id</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>

<span class="n">trans_95_n_var</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="n">trans_95_info</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
<span class="n">trans_95_n_cs</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">trans_95_info</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">cs_id</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unique</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>

<span class="n">single_95_n_gwas_cs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">all_95_95_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">resources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;single&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gwas_cs</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
<span class="n">trans_95_n_gwas_cs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">all_95_95_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">resources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;trans&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gwas_cs</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>

<span class="n">single_95_n_bellenuez_gwas_cs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">all_95_95_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">resources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;single&#39;</span><span class="p">),</span><span class="w"> </span><span class="nf">str_detect</span><span class="p">(</span><span class="n">gwas_cs</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;AD_Bellenguez_2022&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gwas_cs</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
<span class="n">trans_95_n_bellenuez_gwas_cs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">all_95_95_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">resources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;trans&#39;</span><span class="p">),</span><span class="w"> </span><span class="nf">str_detect</span><span class="p">(</span><span class="n">gwas_cs</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;AD_Bellenguez_2022&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gwas_cs</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">length</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="add-gene-information-to-gpqtl">
<h4>add gene information to gpQTL<a class="headerlink" href="#add-gene-information-to-gpqtl" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_only_gwas_cs_gp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trans_only_gwas_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;gpQTL&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">gpID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_extract</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;gp_.+&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_only_gwas_cs_gp</span><span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A tibble: 3 × 17</caption>
<thead>
	<tr><th scope=col>gwas_cs</th><th scope=col>resources</th><th scope=col>variant_ID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>a1</th><th scope=col>a2</th><th scope=col>gene_ID</th><th scope=col>event_ID</th><th scope=col>variant_id</th><th scope=col>z</th><th scope=col>pip</th><th scope=col>study</th><th scope=col>region</th><th scope=col>resource</th><th scope=col>gpID</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>AD_Kunkle_Stage1_2019:chr16_78676505_80260790:1</td><td>trans</td><td>chr16:79321960:G:A</td><td>16</td><td>79321959</td><td>79321960</td><td>A</td><td>G</td><td>chr16_78676505_80260790</td><td>ROSMAP_gpQTL_gp_3231</td><td>16:79321960:G:A</td><td>NA</td><td>0.9771629</td><td>AD_Kunkle_Stage1_2019</td><td>chr16_78676505_80260790</td><td>trans</td><td>gp_3231</td></tr>
	<tr><td>AD_Wightman_Full_2021:chr8_26225312_27515963:2 </td><td>trans</td><td>chr8:27462330:A:T </td><td>8 </td><td>27462329</td><td>27462330</td><td>T</td><td>A</td><td>chr8_26225312_27515963 </td><td>ROSMAP_gpQTL_gp_7827</td><td>8:27462330:A:T </td><td>NA</td><td>0.3067440</td><td>AD_Wightman_Full_2021</td><td>chr8_26225312_27515963 </td><td>trans</td><td>gp_7827</td></tr>
	<tr><td>AD_Wightman_Full_2021:chr8_26225312_27515963:2 </td><td>trans</td><td>chr8:27465457:G:A </td><td>8 </td><td>27465456</td><td>27465457</td><td>A</td><td>G</td><td>chr8_26225312_27515963 </td><td>ROSMAP_gpQTL_gp_7827</td><td>8:27465457:G:A </td><td>NA</td><td>0.3913480</td><td>AD_Wightman_Full_2021</td><td>chr8_26225312_27515963 </td><td>trans</td><td>gp_7827</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">unadj_gpQTL_phenoID</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/data/gpQTL/unadjusted/analysis_ready/phenotype_preprocessing_v202505/ROSMAP_glycoprotein_unadjusted_PhenotypeID.txt&#39;</span><span class="p">)</span><span class="w"> </span><span class="c1">#not on synapse but it&#39;s in github StatFunGen/xqtl-resources/data/ROSMAP</span>
<span class="n">adj_gpQTL_phenoID</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/data/gpQTL/adjusted/analysis_ready/phenotype_preprocessing_v202505/ROSMAP_glycoprotein_adjusted_PhenotypeID.txt&#39;</span><span class="p">)</span><span class="w"> </span><span class="c1">#not on synapse but it&#39;s in github StatFunGen/xqtl-resources/data/ROSMAP</span>
<span class="n">gpQTL_phenoID</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="n">unadj_gpQTL_phenoID</span><span class="p">,</span><span class="w"> </span><span class="n">adj_gpQTL_phenoID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">distinct</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">gpQTL_phenoID</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;gpID&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;gene_id&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_only_gwas_cs_gp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trans_only_gwas_cs_gp</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">gpQTL_phenoID</span><span class="p">)</span>

<span class="n">trans_only_gwas_cs_gp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trans_only_gwas_cs_gp</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">event_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="nf">str_remove</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;gp_.+&#39;</span><span class="p">),</span><span class="w">  </span><span class="n">gene_id</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_only_gwas_cs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rbind</span><span class="p">(</span><span class="w"> </span><span class="n">trans_only_gwas_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;gpQTL&#39;</span><span class="p">)),</span><span class="w"> </span><span class="n">trans_only_gwas_cs_gp</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">gpID</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">gene_id</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trans_only_gwas_cs_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trans_only_gwas_cs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">group_by</span><span class="p">(</span><span class="n">region</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">summarise</span><span class="p">(</span><span class="n">n_event</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">n_distinct</span><span class="p">(</span><span class="n">event_ID</span><span class="p">),</span><span class="w"> </span><span class="n">events</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">event_ID</span><span class="p">),</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;,&#39;</span><span class="p">),</span><span class="w"> </span><span class="n">studies</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">study</span><span class="p">),</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;,&#39;</span><span class="p">))</span>

<span class="c1"># trans_only_gwas_cs_sum %&gt;% saveRDS(&#39;trans_only.loci.info.rds&#39;)</span>
<span class="n">trans_only_gwas_cs_sum</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A tibble: 9 × 4</caption>
<thead>
	<tr><th scope=col>region</th><th scope=col>n_event</th><th scope=col>events</th><th scope=col>studies</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr11_119829118_122612025</td><td>1</td><td>MSBB_eQTL_BM10_ENSG00000198380                                                            </td><td>AD_Bellenguez_2022,AD_Wightman_Excluding23andMe_2021,AD_Wightman_ExcludingUKBand23andME_2021,AD_Wightman_Full_2021</td></tr>
	<tr><td>chr15_47690138_49901356  </td><td>1</td><td>ROSMAP_DeJager_eQTL_Monocyte_ENSG00000087263                                              </td><td>AD_Bellenguez_2022                                                                                                </td></tr>
	<tr><td>chr15_49901356_51733753  </td><td>1</td><td>ROSMAP_DeJager_eQTL_Monocyte_ENSG00000087263                                              </td><td>AD_Bellenguez_2022                                                                                                </td></tr>
	<tr><td>chr16_78676505_80260790  </td><td>2</td><td>ROSMAP_pQTL_ENSG00000120697,ROSMAP_gpQTL_ENSG00000106809                                  </td><td>AD_Wightman_ExcludingUKBand23andME_2021,AD_Kunkle_Stage1_2019                                                     </td></tr>
	<tr><td>chr1_158933969_162352348 </td><td>1</td><td>ROSMAP_DeJager_eQTL_AC_ENSG00000237637                                                    </td><td>AD_Jansen_2021                                                                                                    </td></tr>
	<tr><td>chr20_40697409_42726168  </td><td>1</td><td>ROSMAP_DeJager_eQTL_OPC_ENSG00000114861                                                   </td><td>AD_Bellenguez_EADB_2022                                                                                           </td></tr>
	<tr><td>chr2_125689597_127728648 </td><td>1</td><td>MSBB_eQTL_BM10_ENSG00000168077                                                            </td><td>AD_Bellenguez_2022                                                                                                </td></tr>
	<tr><td>chr2_158720603_160577800 </td><td>2</td><td>MSBB_eQTL_BM10_ENSG00000213579,ROSMAP_DeJager_eQTL_Monocyte_ENSG00000105220               </td><td>AD_Bellenguez_2022                                                                                                </td></tr>
	<tr><td>chr8_26225312_27515963   </td><td>3</td><td>MSBB_eQTL_BM22_ENSG00000137270,MSBB_eQTL_BM36_ENSG00000140987,ROSMAP_gpQTL_ENSG00000113621</td><td>AD_Wightman_Full_2021                                                                                             </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">)</span><span class="w">  </span>

<span class="c1"># Custom formatting function for y-axis only</span>
<span class="n">format_axis_count</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">case_when</span><span class="p">(</span>
<span class="w">    </span><span class="n">x</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1000000</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="nf">round</span><span class="p">(</span><span class="n">x</span><span class="o">/</span><span class="m">1000000</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;M&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">x</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1000</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="nf">round</span><span class="p">(</span><span class="n">x</span><span class="o">/</span><span class="m">1000</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;K&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="kc">TRUE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>
<span class="w">  </span><span class="p">)</span>
<span class="p">}</span>

<span class="nf">ggplot</span><span class="p">(</span><span class="n">summary_df</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">type</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">value</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">type</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_col</span><span class="p">(</span><span class="n">position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">position_dodge</span><span class="p">(</span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.9</span><span class="p">),</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.7</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_text</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">scales</span><span class="o">::</span><span class="nf">comma</span><span class="p">(</span><span class="n">value</span><span class="p">)),</span>
<span class="w">            </span><span class="n">position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">position_dodge</span><span class="p">(</span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.9</span><span class="p">),</span>
<span class="w">            </span><span class="n">vjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-0.5</span><span class="p">,</span>
<span class="w">            </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">,</span><span class="w"> </span><span class="n">fontface</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">show.legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="w"> </span><span class="n">category</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;free_y&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_y_continuous</span><span class="p">(</span>
<span class="w">    </span><span class="n">expand</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">expansion</span><span class="p">(</span><span class="n">mult</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0.15</span><span class="p">)),</span>
<span class="w">    </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">format_axis_count</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">(</span><span class="n">base_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">18</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="c1"># Use theme_minimal instead of theme_classic</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Type&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Count&quot;</span><span class="p">,</span>
<span class="w">       </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Single vs Trans (95% CS categories)&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme</span><span class="p">(</span>
<span class="w">    </span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">24</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">margin</span><span class="p">(</span><span class="n">b</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">)),</span>
<span class="w">    </span><span class="n">axis.title.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">margin</span><span class="p">(</span><span class="n">t</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">)),</span>
<span class="w">    </span><span class="n">axis.title.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">margin</span><span class="p">(</span><span class="n">r</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">)),</span>
<span class="w">    </span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">18</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">18</span><span class="p">),</span>
<span class="w">    </span><span class="n">strip.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="w"> </span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;none&quot;</span><span class="p">,</span>
<span class="w">    </span><span class="n">panel.spacing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">unit</span><span class="p">(</span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;lines&quot;</span><span class="p">),</span>
<span class="w">    </span><span class="c1"># Classic theme with balanced borders</span>
<span class="w">    </span><span class="n">panel.border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.5</span><span class="p">),</span><span class="w">  </span><span class="c1"># Consistent border size</span>
<span class="w">    </span><span class="n">strip.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey90&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.5</span><span class="p">),</span>
<span class="w">    </span><span class="n">panel.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_rect</span><span class="p">(</span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_fill_brewer</span><span class="p">(</span><span class="n">palette</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Set2&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/800e829d3c3df5ec3ad1423752d20b5475b59d7ee2568f909bd3cdee36a06580.png"><img alt="../../_images/800e829d3c3df5ec3ad1423752d20b5475b59d7ee2568f909bd3cdee36a06580.png" src="../../_images/800e829d3c3df5ec3ad1423752d20b5475b59d7ee2568f909bd3cdee36a06580.png" style="width: 600px; height: 600px;" />
</a>
</div>
</div>
</section>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./main_text/5_AD_xQTL_integration"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="Figure_5b_AD_GWAS_loci_properties.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Refined AD GWAS loci properties</p>
      </div>
    </a>
    <a class="right-next"
       href="Figure_5d_AD_loci_table_summary.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Summary of FunGen-xQTL AD loci table</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#motivation">Motivation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#methods-overview">Methods Overview</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#implementation">Implementation</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#overlap-analysis">Overlap Analysis</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gene-involvement-evidence">Gene Involvement Evidence</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-single-context-results">Summarize single context results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-multi-context-results">summarize multi context results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-metabrian-results">summarize metabrian results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-multi-gene-results">summarize multi gene results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-the-coloboost-results">summarize the coloboost results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-trans-results">summarize trans results</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#summarize-fsusie-results">summarize fsusie results</a><ul class="nav section-nav flex-column">
<li class="toc-h5 nav-item toc-entry"><a class="reference internal nav-link" href="#combine-all">combine all</a></li>
</ul>
</li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#plot-susie-upsetr">plot susie upsetR</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#plot-the-stacked-bar-plot-only-four-categories">plot the stacked bar plot (only four categories)</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#check-trans-only">check trans only</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#add-gene-information-to-gpqtl">add gene information to gpQTL</a></li>
</ul>
</li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>