
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Marginal vs multicontext effect sign heatmap &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/3_multi_context/Figure_2_multicontext_effect_sign';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../1_project_overview/index.html">Multi-omics QTL atlas of the aging human brain</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../2_single_context_cis/index.html">Fine-mapped QTL landscape across brain molecular phenotypes</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="simple">
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/3_multi_context/Figure_2_multicontext_effect_sign.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/main_text/3_multi_context/Figure_2_multicontext_effect_sign.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Marginal vs multicontext effect sign heatmap</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Marginal vs multicontext effect sign heatmap</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-libraries">Load libraries</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#marginal-posterior-effect-sign">Marginal Posterior Effect Sign</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#marginal-trait-matrix">Marginal trait matrix</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#marginal-posterior-effect-sign-heatmap">Marginal posterior effect sign heatmap</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#multicontext-effect-posterior-effect-sign">Multicontext Effect Posterior Effect Sign</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multicontext-trait-matrix">Multicontext trait matrix</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multicontext-posterior-effect-sign-heatmap">Multicontext posterior effect sign heatmap</a></li>
</ul>
</li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section id="marginal-vs-multicontext-effect-sign-heatmap">
<h1>Marginal vs multicontext effect sign heatmap<a class="headerlink" href="#marginal-vs-multicontext-effect-sign-heatmap" title="Link to this heading">#</a></h1>
<p>Author: Anqi Wang and Jenny Empawi</p>
<section id="load-libraries">
<h2>Load libraries<a class="headerlink" href="#load-libraries" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">reshape2</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="marginal-posterior-effect-sign">
<h2>Marginal Posterior Effect Sign<a class="headerlink" href="#marginal-posterior-effect-sign" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">posterior_dat</span><span class="w"> </span><span class="o">&lt;-</span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/mash_34_contexts_ed_bovy_1e3.EZ.posterior.rds&quot;</span><span class="p">)</span><span class="w"> </span><span class="c1">#FIXME</span>
<span class="c1">### transfer the lfsr to the four columns</span>
<span class="n">posterior_dat_lfsr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">lfsr</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">                             </span><span class="n">split_info</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">lfsr</span><span class="p">)),</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span>
<span class="w">                             </span><span class="n">variant</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span>
<span class="w">                             </span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="p">]</span>
<span class="w">                           </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">split_info</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">pivot_longer</span><span class="p">(</span>
<span class="w">                            </span><span class="n">cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="n">variant</span><span class="p">,</span><span class="w"> </span><span class="n">gene</span><span class="p">),</span><span class="w">  </span><span class="c1"># Keep &#39;variant&#39; and &#39;gene&#39; columns fixed</span>
<span class="w">                            </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;context&quot;</span><span class="p">,</span><span class="w">      </span><span class="c1"># New column for context (column names)</span>
<span class="w">                            </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;lfsr&quot;</span><span class="w">  </span><span class="c1"># New column for effect sizes (values)</span>
<span class="w">                     </span><span class="p">)</span><span class="w"> </span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">posterior_dat_lfsr</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span>
<span class="n">posterior_dat_negative_prob</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">NegativeProb</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">                             </span><span class="n">split_info</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">NegativeProb</span><span class="p">)),</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span>
<span class="w">                             </span><span class="n">variant</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span>
<span class="w">                             </span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="p">]</span>
<span class="w">                           </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">split_info</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">pivot_longer</span><span class="p">(</span>
<span class="w">                            </span><span class="n">cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="n">variant</span><span class="p">,</span><span class="w"> </span><span class="n">gene</span><span class="p">),</span><span class="w">  </span><span class="c1"># Keep &#39;variant&#39; and &#39;gene&#39; columns fixed</span>
<span class="w">                            </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;context&quot;</span><span class="p">,</span><span class="w">      </span><span class="c1"># New column for context (column names)</span>
<span class="w">                            </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;prob&quot;</span><span class="w">  </span><span class="c1"># New column for effect sizes (values)</span>
<span class="w">                     </span><span class="p">)</span>
<span class="c1">### transfer the posterior_mean to the four columns</span>
<span class="n">posterior_dat_effect</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">PosteriorMean</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">                             </span><span class="n">split_info</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">posterior_dat</span><span class="o">$</span><span class="n">PosteriorMean</span><span class="p">)),</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span>
<span class="w">                             </span><span class="n">variant</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">1</span><span class="p">],</span>
<span class="w">                             </span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">split_info</span><span class="p">[,</span><span class="w"> </span><span class="m">2</span><span class="p">]</span>
<span class="w">                           </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">split_info</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                     </span><span class="nf">pivot_longer</span><span class="p">(</span>
<span class="w">                            </span><span class="n">cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="n">variant</span><span class="p">,</span><span class="w"> </span><span class="n">gene</span><span class="p">),</span><span class="w">  </span><span class="c1"># Keep &#39;variant&#39; and &#39;gene&#39; columns fixed</span>
<span class="w">                            </span><span class="n">names_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;context&quot;</span><span class="p">,</span><span class="w">      </span><span class="c1"># New column for context (column names)</span>
<span class="w">                            </span><span class="n">values_to</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;effect&quot;</span><span class="w">  </span><span class="c1"># New column for effect sizes (values)</span>
<span class="w">                     </span><span class="p">)</span>
<span class="n">posterior_dat_effect</span><span class="o">$</span><span class="n">effect_sign</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">posterior_dat_effect</span><span class="o">$</span><span class="n">effect</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span>
<span class="w">                                           </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span>
<span class="w">                                           </span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;-1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;+1&quot;</span><span class="p">))</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">posterior_dat_effect</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span>
<span class="c1">### integrate lfsr and effec, also effect sign</span>
<span class="n">posterior_dat_lfsr_effect</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr</span><span class="o">%&gt;%</span>
<span class="w">                                </span><span class="nf">left_join</span><span class="p">(</span><span class="n">posterior_dat_effect</span><span class="p">,</span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;variant&quot;</span><span class="p">,</span><span class="s">&quot;gene&quot;</span><span class="p">,</span><span class="s">&quot;context&quot;</span><span class="p">))</span><span class="o">%&gt;%</span>
<span class="w">                                </span><span class="nf">left_join</span><span class="p">(</span><span class="n">posterior_dat_negative_prob</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;variant&quot;</span><span class="p">,</span><span class="s">&quot;gene&quot;</span><span class="p">,</span><span class="s">&quot;context&quot;</span><span class="p">))</span>
<span class="n">lfsr_cutoff</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0.01</span>
<span class="n">posterior_dat_lfsr_effect_filtered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">lfsr</span><span class="o">&lt;</span><span class="n">lfsr_cutoff</span><span class="p">)</span>
<span class="n">ROSMAP_multicontext_top_loci</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;/Staging/variant_gene_summary/xQTL_loci_summary/xQTL_finemapping/multi_context_finemapping/ROSMAP.exported.toploci.bed.gz&quot;</span><span class="p">)</span>
<span class="n">ROSMAP_multicontext_top_loci_pip_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_top_loci</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">)</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">pip_sum</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">PIP</span><span class="p">))</span>
<span class="n">mvsusie_export_lfsr_integrated_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_top_loci_pip_sum</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">separate_rows</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">conditional_effect</span><span class="p">,</span><span class="w"> </span><span class="n">lfsr</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="n">effect_sign</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">case_when</span><span class="p">(</span>
<span class="w">      </span><span class="n">conditional_effect</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;+1&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">conditional_effect</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;-1&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">conditional_effect</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;0&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="kc">TRUE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="kc">NA_character_</span><span class="w">  </span><span class="c1"># handle any unexpected NAs</span>
<span class="w">    </span><span class="p">)</span>
<span class="w">  </span><span class="p">)</span>
<span class="c1">#mvsusie_export_lfsr_integrated_df &lt;- readRDS(&quot;~/project/Figure_2b/multicontext_top_loci_integrated.rds&quot;)</span>
<span class="n">traits</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">event_ID</span><span class="p">)))</span>
<span class="n">posterior_dat_lfsr_effect_filtered_context</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">context</span><span class="o">%in%</span><span class="n">traits</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">%in%</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                                                           </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">))))</span>
<span class="n">posterior_dat_lfsr_effect_filtered_context_top_variants</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                                                                </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="n">context</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                                                                </span><span class="nf">filter</span><span class="p">(</span><span class="n">lfsr</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="nf">min</span><span class="p">(</span><span class="n">lfsr</span><span class="p">),</span><span class="n">prob</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">prob</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                                                                </span><span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                                                                </span><span class="nf">unique</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">intersected_gene_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context_top_variants</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">context</span><span class="o">%in%</span><span class="n">traits</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">%in%</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                                                           </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">))))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">unique</span><span class="p">()</span>
<span class="nf">length</span><span class="p">(</span><span class="n">intersected_gene_names</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">7069</div></div>
</div>
</section>
<section id="marginal-trait-matrix">
<h2>Marginal trait matrix<a class="headerlink" href="#marginal-trait-matrix" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#trait_matrix &lt;- readRDS(&quot;~/project/Figure_2b/marginal_efect_sign_0429_4063.rds&quot;)</span>
<span class="c1"># replace with the new version multicontext top loci table</span>
<span class="n">ROSMAP_multicontext_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;/Staging/variant_gene_summary/xQTL_loci_summary/xQTL_finemapping/multi_context_finemapping/ROSMAP.exported.toploci.bed.gz&quot;</span><span class="p">)</span>
<span class="n">ROSMAP_multicontext_top_loci_pip_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_res</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">)</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">pip_sum</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">PIP</span><span class="p">))</span>
<span class="n">mvsusie_export_lfsr_integrated_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_top_loci_pip_sum</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">separate_rows</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">conditional_effect</span><span class="p">,</span><span class="w"> </span><span class="n">lfsr</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="n">effect_sign</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">case_when</span><span class="p">(</span>
<span class="w">      </span><span class="n">conditional_effect</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;+1&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">conditional_effect</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;-1&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">conditional_effect</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;0&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="kc">TRUE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="kc">NA_character_</span><span class="w">  </span><span class="c1"># handle any unexpected NAs</span>
<span class="w">    </span><span class="p">)</span>
<span class="w">  </span><span class="p">)</span>
<span class="c1"># we split the event_ID in the multicontext top loci table to get `mvsusie_export_lfsr_integrated_df`</span>
<span class="c1">#mvsusie_export_lfsr_integrated_df &lt;- readRDS(&quot;~/project/Figure_2b/multicontext_top_loci_integrated.rds&quot;)</span>
<span class="n">traits</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">event_ID</span><span class="p">)))</span>
<span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">intersected_gene_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context_top_variants</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">context</span><span class="o">%in%</span><span class="n">traits</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">%in%</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                                                           </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">))))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">unique</span><span class="p">()</span>
<span class="n">ROSMAP_multicontext_extracted</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_res</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_ID</span><span class="o">%in%</span><span class="n">intersected_gene_names</span><span class="p">)</span>
<span class="n">posterior_dat_lfsr_effect_filtered_context</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">context</span><span class="o">%in%</span><span class="n">traits</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">%in%</span><span class="nf">unique</span><span class="p">(</span><span class="n">ROSMAP_multicontext_extracted</span><span class="o">$</span><span class="n">gene_ID</span><span class="p">))</span>
<span class="n">posterior_dat_lfsr_effect_filtered_context_grouped</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gene</span><span class="p">,</span><span class="w"> </span><span class="n">variant</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">summarise</span><span class="p">(</span>
<span class="w">    </span><span class="n">contexts</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">context</span><span class="p">),</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">),</span><span class="w"> </span>
<span class="w">    </span><span class="n">.groups</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;drop&quot;</span>
<span class="w">  </span><span class="p">)</span>
<span class="n">num_traits</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">traits</span><span class="p">)</span>
<span class="n">trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">num_traits</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">num_traits</span><span class="p">)</span><span class="w">  </span><span class="c1"># Initialize matrix</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">traits</span><span class="w">   </span><span class="c1"># Assign trait names</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">gene_name</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">posterior_dat_lfsr_effect_filtered_context_grouped</span><span class="o">$</span><span class="n">gene</span><span class="p">)){</span>
<span class="c1">#gene_name &lt;- as.character(unique(mvsusie_export_lfsr_integrated_df$gene))[1]</span>
<span class="n">strong_variants_lfsr_thres</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context_grouped</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">gene_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">unique</span><span class="p">()</span>
<span class="c1">#Iterate through variants in df2</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">variant_ID</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">strong_variants_lfsr_thres</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1"># Extract rows from df1 where the marker matches the variant in df2</span>
<span class="w">  </span><span class="n">variant_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">==</span><span class="n">gene_name</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">variant</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">variant_ID</span><span class="p">)</span>
<span class="w">  </span><span class="c1"># If variant is found in multiple traits, compare effect signs</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span>
<span class="w">    </span><span class="c1"># Get unique trait pairs</span>
<span class="w">    </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="o">:</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span>
<span class="w">        </span><span class="n">trait_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">context</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
<span class="w">        </span><span class="n">trait_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">context</span><span class="p">[</span><span class="n">j</span><span class="p">]</span>
<span class="w">        </span>
<span class="w">        </span><span class="n">effect_sign_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">effect_sign</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
<span class="w">        </span><span class="n">effect_sign_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">effect_sign</span><span class="p">[</span><span class="n">j</span><span class="p">]</span>

<span class="w">        </span><span class="c1"># Find indices in the trait matrix</span>
<span class="w">        </span><span class="n">idx_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">trait_i</span><span class="p">)</span>
<span class="w">        </span><span class="n">idx_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">trait_j</span><span class="p">)</span>
<span class="w">        </span>
<span class="w">        </span><span class="c1"># Ensure idx_i is always smaller than idx_j for upper triangle updates</span>
<span class="w">        </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">idx_i</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">idx_j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">              </span><span class="n">temp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">idx_i</span>
<span class="w">              </span><span class="n">idx_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">idx_j</span>
<span class="w">              </span><span class="n">idx_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">temp</span>
<span class="w">        </span><span class="p">}</span>

<span class="w">        </span><span class="c1"># Update matrix correctly with only +1 increments</span>
<span class="w">        </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">effect_sign_i</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">effect_sign_j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">            </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_i</span><span class="p">,</span><span class="w"> </span><span class="n">idx_j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_i</span><span class="p">,</span><span class="w"> </span><span class="n">idx_j</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Always upper triangle</span>
<span class="w">        </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">            </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_j</span><span class="p">,</span><span class="w"> </span><span class="n">idx_i</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_j</span><span class="p">,</span><span class="w"> </span><span class="n">idx_i</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Always lower triangle</span>
<span class="w">        </span><span class="p">}</span>
<span class="w">      </span><span class="p">}</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">trait_idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">context</span><span class="p">)</span>
<span class="w">    </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">trait_idx</span><span class="p">,</span><span class="w"> </span><span class="n">trait_idx</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">trait_idx</span><span class="p">,</span><span class="w"> </span><span class="n">trait_idx</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="c1"># Diagonal increment</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>

<span class="n">trait_matrix</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A matrix: 23  23 of type dbl</caption>
<thead>
	<tr><th></th><th scope=col>AC_DeJager_eQTL</th><th scope=col>DLPFC_DeJager_eQTL</th><th scope=col>PCC_DeJager_eQTL</th><th scope=col>Exc_mega_eQTL</th><th scope=col>Ast_Kellis_eQTL</th><th scope=col>Ast_mega_eQTL</th><th scope=col>Exc_DeJager_eQTL</th><th scope=col>Exc_Kellis_eQTL</th><th scope=col>Inh_DeJager_eQTL</th><th scope=col>Inh_Kellis_eQTL</th><th scope=col></th><th scope=col>Ast_DeJager_eQTL</th><th scope=col>OPC_Kellis_eQTL</th><th scope=col>Oli_Kellis_eQTL</th><th scope=col>DLPFC_Bennett_pQTL</th><th scope=col>OPC_DeJager_eQTL</th><th scope=col>OPC_mega_eQTL</th><th scope=col>monocyte_ROSMAP_eQTL</th><th scope=col>Mic_Kellis_eQTL</th><th scope=col>Mic_mega_eQTL</th><th scope=col>Mic_DeJager_eQTL</th></tr>
</thead>
<tbody>
	<tr><th scope=row>AC_DeJager_eQTL</th><td> 310</td><td>9786</td><td>10111</td><td>5336</td><td>3626</td><td>3673</td><td>5836</td><td>5093</td><td>4655</td><td>4392</td><td></td><td>4411</td><td>2788</td><td>3896</td><td>2762</td><td>2112</td><td>2174</td><td>2325</td><td>1305</td><td> 941</td><td>1365</td></tr>
	<tr><th scope=row>DLPFC_DeJager_eQTL</th><td> 324</td><td> 126</td><td>10568</td><td>6482</td><td>4010</td><td>4039</td><td>6984</td><td>6228</td><td>5352</td><td>5090</td><td></td><td>4839</td><td>3144</td><td>4297</td><td>2889</td><td>2386</td><td>2454</td><td>2356</td><td>1385</td><td> 991</td><td>1445</td></tr>
	<tr><th scope=row>PCC_DeJager_eQTL</th><td> 116</td><td>  88</td><td>   17</td><td>6029</td><td>3615</td><td>3653</td><td>6561</td><td>5785</td><td>4972</td><td>4682</td><td></td><td>4421</td><td>2793</td><td>3807</td><td>2949</td><td>2119</td><td>2185</td><td>2218</td><td>1268</td><td> 914</td><td>1337</td></tr>
	<tr><th scope=row>Exc_mega_eQTL</th><td>1291</td><td> 783</td><td>  837</td><td>   2</td><td>3794</td><td>3852</td><td>8974</td><td>8566</td><td>6617</td><td>6476</td><td></td><td>4100</td><td>3499</td><td>4216</td><td>1740</td><td>2694</td><td>2860</td><td>1183</td><td>1361</td><td>1016</td><td>1369</td></tr>
	<tr><th scope=row>Ast_Kellis_eQTL</th><td> 721</td><td> 518</td><td>  602</td><td> 458</td><td>   7</td><td>5458</td><td>3658</td><td>3808</td><td>3614</td><td>3771</td><td></td><td>5346</td><td>3339</td><td>3540</td><td>1151</td><td>2535</td><td>2610</td><td>1017</td><td>1314</td><td> 875</td><td>1265</td></tr>
	<tr><th scope=row>Ast_mega_eQTL</th><td> 730</td><td> 544</td><td>  627</td><td> 503</td><td>   0</td><td>   0</td><td>3717</td><td>3760</td><td>3656</td><td>3684</td><td></td><td>5475</td><td>3281</td><td>3475</td><td>1166</td><td>2622</td><td>2712</td><td>1033</td><td>1359</td><td> 949</td><td>1335</td></tr>
	<tr><th scope=row>Exc_DeJager_eQTL</th><td>1237</td><td> 742</td><td>  785</td><td>   1</td><td> 456</td><td> 494</td><td>   7</td><td>8335</td><td>6852</td><td>6294</td><td></td><td>4429</td><td>3373</td><td>4098</td><td>1790</td><td>2715</td><td>2751</td><td>1471</td><td>1319</td><td> 990</td><td>1396</td></tr>
	<tr><th scope=row>Exc_Kellis_eQTL</th><td>1205</td><td> 700</td><td>  784</td><td>   6</td><td> 368</td><td> 422</td><td>  16</td><td>   6</td><td>6285</td><td>6584</td><td></td><td>3919</td><td>3529</td><td>4259</td><td>1640</td><td>2580</td><td>2741</td><td>1184</td><td>1389</td><td> 992</td><td>1345</td></tr>
	<tr><th scope=row>Inh_DeJager_eQTL</th><td>1128</td><td> 779</td><td>  837</td><td> 175</td><td> 324</td><td> 351</td><td> 153</td><td> 148</td><td>  11</td><td>7025</td><td></td><td>4123</td><td>3369</td><td>3930</td><td>1274</td><td>2679</td><td>2716</td><td>1170</td><td>1343</td><td> 985</td><td>1413</td></tr>
	<tr><th scope=row>Inh_Kellis_eQTL</th><td>1120</td><td> 758</td><td>  838</td><td> 149</td><td> 270</td><td> 316</td><td> 144</td><td> 101</td><td>   9</td><td>  15</td><td></td><td>3841</td><td>3547</td><td>4148</td><td>1220</td><td>2545</td><td>2716</td><td>1135</td><td>1389</td><td> 984</td><td>1331</td></tr>
	<tr><th scope=row>Inh_mega_eQTL</th><td>1150</td><td> 811</td><td>  873</td><td> 188</td><td> 327</td><td> 361</td><td> 175</td><td> 158</td><td>   1</td><td>   2</td><td></td><td>3966</td><td>3506</td><td>4053</td><td>1259</td><td>2645</td><td>2862</td><td>1145</td><td>1421</td><td>1048</td><td>1409</td></tr>
	<tr><th scope=row>Oli_DeJager_eQTL</th><td> 843</td><td> 677</td><td>  774</td><td> 599</td><td> 286</td><td> 299</td><td> 594</td><td> 540</td><td> 419</td><td> 405</td><td></td><td>3535</td><td>3074</td><td>5581</td><td>1049</td><td>2573</td><td>2531</td><td>1407</td><td>1249</td><td> 892</td><td>1316</td></tr>
	<tr><th scope=row>Oli_mega_eQTL</th><td> 868</td><td> 700</td><td>  790</td><td> 592</td><td> 287</td><td> 310</td><td> 585</td><td> 527</td><td> 414</td><td> 390</td><td></td><td>3499</td><td>3302</td><td>5856</td><td>1059</td><td>2564</td><td>2694</td><td>1372</td><td>1320</td><td> 929</td><td>1299</td></tr>
	<tr><th scope=row>Ast_DeJager_eQTL</th><td> 733</td><td> 538</td><td>  613</td><td> 487</td><td>   2</td><td>   3</td><td> 478</td><td> 416</td><td> 332</td><td> 308</td><td></td><td>  17</td><td>3241</td><td>3566</td><td>1267</td><td>2740</td><td>2703</td><td>1366</td><td>1292</td><td> 894</td><td>1370</td></tr>
	<tr><th scope=row>OPC_Kellis_eQTL</th><td> 897</td><td> 704</td><td>  770</td><td> 322</td><td> 130</td><td> 143</td><td> 510</td><td> 272</td><td> 229</td><td> 194</td><td></td><td> 355</td><td>  15</td><td>3450</td><td> 870</td><td>3745</td><td>3833</td><td> 975</td><td>1340</td><td> 865</td><td>1279</td></tr>
	<tr><th scope=row>Oli_Kellis_eQTL</th><td> 895</td><td> 681</td><td>  797</td><td> 565</td><td> 253</td><td> 281</td><td> 564</td><td> 483</td><td> 405</td><td> 359</td><td></td><td> 278</td><td> 180</td><td>  28</td><td>1122</td><td>2543</td><td>2646</td><td>1403</td><td>1336</td><td> 911</td><td>1292</td></tr>
	<tr><th scope=row>DLPFC_Bennett_pQTL</th><td> 293</td><td> 270</td><td>  210</td><td> 588</td><td> 397</td><td> 429</td><td> 538</td><td> 553</td><td> 591</td><td> 594</td><td></td><td> 405</td><td> 400</td><td> 518</td><td> 194</td><td> 653</td><td> 657</td><td> 690</td><td> 451</td><td> 355</td><td> 484</td></tr>
	<tr><th scope=row>OPC_DeJager_eQTL</th><td> 798</td><td> 650</td><td>  683</td><td> 300</td><td> 120</td><td> 122</td><td> 504</td><td> 271</td><td> 216</td><td> 198</td><td></td><td> 340</td><td>   1</td><td> 167</td><td> 321</td><td>   5</td><td>3840</td><td> 728</td><td>1140</td><td> 816</td><td>1208</td></tr>
	<tr><th scope=row>OPC_mega_eQTL</th><td> 826</td><td> 683</td><td>  715</td><td> 308</td><td> 134</td><td> 141</td><td> 525</td><td> 275</td><td> 221</td><td> 200</td><td></td><td> 369</td><td>   1</td><td> 176</td><td> 340</td><td>   0</td><td>   0</td><td> 731</td><td>1189</td><td> 881</td><td>1188</td></tr>
	<tr><th scope=row>monocyte_ROSMAP_eQTL</th><td> 286</td><td> 290</td><td>  284</td><td> 375</td><td> 219</td><td> 223</td><td> 380</td><td> 352</td><td> 296</td><td> 281</td><td></td><td> 233</td><td> 368</td><td> 226</td><td> 186</td><td> 361</td><td> 383</td><td>  66</td><td> 999</td><td> 742</td><td>1041</td></tr>
	<tr><th scope=row>Mic_Kellis_eQTL</th><td> 340</td><td> 290</td><td>  302</td><td> 224</td><td> 127</td><td> 128</td><td> 222</td><td> 200</td><td> 186</td><td> 179</td><td></td><td> 154</td><td> 102</td><td> 208</td><td> 245</td><td>  90</td><td>  87</td><td>  44</td><td>  11</td><td>1570</td><td>2057</td></tr>
	<tr><th scope=row>Mic_mega_eQTL</th><td> 262</td><td> 237</td><td>  236</td><td> 168</td><td> 126</td><td> 125</td><td> 168</td><td> 155</td><td> 135</td><td> 132</td><td></td><td> 122</td><td> 101</td><td> 156</td><td> 193</td><td>  90</td><td>  89</td><td>  36</td><td>   0</td><td>   0</td><td>1595</td></tr>
	<tr><th scope=row>Mic_DeJager_eQTL</th><td> 337</td><td> 300</td><td>  302</td><td> 233</td><td> 133</td><td> 132</td><td> 238</td><td> 207</td><td> 204</td><td> 193</td><td></td><td> 163</td><td> 110</td><td> 217</td><td> 229</td><td>  98</td><td>  94</td><td>  41</td><td>   0</td><td>   0</td><td>  14</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="marginal-posterior-effect-sign-heatmap">
<h2>Marginal posterior effect sign heatmap<a class="headerlink" href="#marginal-posterior-effect-sign-heatmap" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Ensure &quot;AC_DeJager_eQTL&quot; stays on top</span>
<span class="c1">#trait_matrix &lt;- readRDS(&quot;~/project/Figure_2b/marginal_efect_sign_0429_4063.rds&quot;)</span>
<span class="n">trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">),</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))),</span><span class="w"> </span>
<span class="w">                           </span><span class="nf">order</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">),</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)))]</span>
<span class="n">rev_trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[,</span><span class="nf">ncol</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="c1">#rev_trait_matrix &lt;- t(trait_matrix)</span>
<span class="n">rev_trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_trait_matrix</span><span class="p">)</span>

<span class="c1"># Calculate the fraction matrix</span>
<span class="c1"># Assuming rev_trait_matrix is your input square matrix</span>
<span class="n">context_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">rev_trait_matrix</span><span class="p">)</span>
<span class="n">n</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">context_names</span><span class="p">)</span>

<span class="c1"># Initialize output matrix</span>
<span class="n">proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">context_names</span><span class="w"> </span>

<span class="n">pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">context_names</span><span class="w"> </span>

<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">seq_len</span><span class="p">(</span><span class="n">n</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">seq_len</span><span class="p">(</span><span class="n">n</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="n">j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="c1"># Extract 2x2 submatrix</span>
<span class="w">      </span><span class="n">trait_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)[</span><span class="n">i</span><span class="p">]</span>
<span class="w">      </span><span class="n">trait_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)[</span><span class="n">j</span><span class="p">]</span>
<span class="w">      </span><span class="n">pattern</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_c</span><span class="p">(</span><span class="n">trait_i</span><span class="p">,</span><span class="w"> </span><span class="n">trait_j</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;|&quot;</span><span class="p">)</span>
<span class="w">      </span><span class="n">submat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">)]</span>
<span class="w">      </span><span class="n">marginal_multicontext_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w">  </span><span class="n">posterior_dat_lfsr_effect_filtered_context_grouped</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                                        </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">contexts</span><span class="p">,</span><span class="n">pattern</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                                        </span><span class="nf">unique</span><span class="p">()</span><span class="w"> </span>
<span class="w">      </span><span class="n">pairwise_total_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">submat</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="nf">diag</span><span class="p">(</span><span class="n">submat</span><span class="p">))</span>
<span class="w">      </span><span class="n">total_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise_total_sum</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="nf">nrow</span><span class="p">(</span><span class="n">marginal_multicontext_res</span><span class="p">)</span>
<span class="w">      </span><span class="n">value</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span>

<span class="w">      </span><span class="c1"># Compute and assign proportion</span>
<span class="w">      </span><span class="n">proportion_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">round</span><span class="p">((</span><span class="n">value</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_sum</span><span class="p">)</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
<span class="w">      </span><span class="n">pairwise_proportion_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">round</span><span class="p">((</span><span class="n">value</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">pairwise_total_sum</span><span class="p">)</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>
<span class="c1"># View or use proportion_matrix</span>
<span class="c1">#rev_proportion_matrix</span>

<span class="n">rev_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">proportion_matrix</span><span class="p">[,</span><span class="w"> </span><span class="nf">ncol</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="n">rev_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)</span>


<span class="c1"># Convert matrix into long format for ggplot</span>
<span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">reshape2</span><span class="o">::</span><span class="nf">melt</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">plot_data</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Trait1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Trait2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Count&quot;</span><span class="p">)</span>

<span class="c1"># 1. Identify which rows need filling</span>
<span class="n">na_idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">is.na</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">Count</span><span class="p">)</span>

<span class="c1"># 2. Pull from the matrix by name for those rows</span>
<span class="c1"># Find numeric row/col indices</span>
<span class="n">r</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">match</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">Trait1</span><span class="p">[</span><span class="n">na_idx</span><span class="p">],</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))</span>
<span class="n">c</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">match</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">Trait2</span><span class="p">[</span><span class="n">na_idx</span><span class="p">],</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))</span>

<span class="c1"># Do element-wise lookup</span>
<span class="n">plot_data</span><span class="o">$</span><span class="n">Count</span><span class="p">[</span><span class="n">na_idx</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">cbind</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="w"> </span><span class="n">c</span><span class="p">)]</span>

<span class="c1">#proportion_matrix</span>
<span class="n">rev_pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise_proportion_matrix</span><span class="p">[,</span><span class="w"> </span><span class="nf">ncol</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="n">rev_pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_pairwise_proportion_matrix</span><span class="p">)</span>

<span class="w"> </span>
<span class="n">plot_data</span><span class="o">$</span><span class="n">Fraction</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.vector</span><span class="p">(</span><span class="n">reshape2</span><span class="o">::</span><span class="nf">melt</span><span class="p">(</span><span class="n">rev_pairwise_proportion_matrix</span><span class="p">)</span><span class="o">$</span><span class="n">value</span><span class="p">)</span>

<span class="c1">#credible_sum &lt;- posterior_dat_lfsr_effect_filtered_context_grouped %&gt;% unique() %&gt;%nrow()</span>


<span class="n">plot_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">rowwise</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="n">Label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Fraction</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nf">paste0</span><span class="p">(</span><span class="n">Fraction</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n(&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">Count</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;)&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">total_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context_grouped</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">contexts</span><span class="p">,</span><span class="nf">as.character</span><span class="p">(</span><span class="n">Trait1</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">nrow</span><span class="p">()</span>
<span class="w">      </span>
<span class="w">      </span><span class="nf">round</span><span class="p">((</span><span class="n">Count</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_genes</span><span class="p">)</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">as.character</span><span class="p">()</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">ungroup</span><span class="p">()</span>
<span class="c1"># df$Label &lt;- ifelse(!is.na(df$Fraction), </span>
<span class="c1">#                    paste0(df$Fraction, &quot;\n(&quot;, df$Count, &quot;)&quot;), </span>
<span class="c1">#                    as.character(round((df$Count/credible_sum)*100,2)))</span>

<span class="n">plot_data</span><span class="o">$</span><span class="n">Trait1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">Trait1</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rev</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)))</span>
<span class="n">plot_data</span><span class="o">$</span><span class="n">Trait2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">plot_data</span><span class="o">$</span><span class="n">Trait2</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">))</span>
<span class="c1"># df$Trait1 &lt;- factor(df$Trait1, levels = rev(rownames(rev_trait_matrix)))</span>
<span class="c1"># df$Trait2 &lt;- factor(df$Trait2, levels = rownames(rev_trait_matrix))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#pdf(&quot;ROSMAP_marginal_effect_sign_heatmap_version_0429_4063.pdf&quot;, width =10, height = 8)</span>
<span class="c1"># Plot heatmap using ggplot2</span>
<span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">plot_data</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Trait1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Trait2</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Fraction</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_tile</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_fill_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">na.value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_text</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Label</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="c1"># Add text with count and fraction</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Proportion&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Marginal Posterior Effect Sign Heatmap&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">45</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">),</span>
<span class="w">        </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">),</span>
<span class="w">        </span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">25</span><span class="p">,</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">        </span><span class="n">legend.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">       </span><span class="n">legend.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">)</span>
<span class="n">p1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/d2a579adb44af498bed12ea0a746ab4664969683abd517de2d273649adca812a.png"><img alt="../../_images/d2a579adb44af498bed12ea0a746ab4664969683abd517de2d273649adca812a.png" src="../../_images/d2a579adb44af498bed12ea0a746ab4664969683abd517de2d273649adca812a.png" style="width: 1800px; height: 1800px;" />
</a>
</div>
</div>
</section>
</section>
<section id="multicontext-effect-posterior-effect-sign">
<h1>Multicontext Effect Posterior Effect Sign<a class="headerlink" href="#multicontext-effect-posterior-effect-sign" title="Link to this heading">#</a></h1>
<section id="multicontext-trait-matrix">
<h2>Multicontext trait matrix<a class="headerlink" href="#multicontext-trait-matrix" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">coverage</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;cs_coverage_0.95&quot;</span>
<span class="n">mvsusie_export_lfsr_integrated_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Staging/variant_gene_summary/xQTL_loci_summary/xQTL_finemapping/multi_context_finemapping/consolidated/FunGen_xQTL_multi_context_top_loci_integrated.rds&quot;</span><span class="p">)</span>
<span class="n">ROSMAP_multicontext_top_loci_pip_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_res</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">group_by</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">)</span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">pip_sum</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">PIP</span><span class="p">))</span>
<span class="n">mvsusie_export_lfsr_integrated_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_top_loci_pip_sum</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">separate_rows</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">conditional_effect</span><span class="p">,</span><span class="w"> </span><span class="n">lfsr</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="n">effect_sign</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">case_when</span><span class="p">(</span>
<span class="w">      </span><span class="n">conditional_effect</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;+1&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">conditional_effect</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;-1&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="n">conditional_effect</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;0&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="kc">TRUE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="kc">NA_character_</span><span class="w">  </span><span class="c1"># handle any unexpected NAs</span>
<span class="w">    </span><span class="p">)</span>
<span class="w">  </span><span class="p">)</span>
<span class="n">traits</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">event_ID</span><span class="p">)))</span>
<span class="n">intersected_gene_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">posterior_dat_lfsr_effect_filtered_context_top_variants</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">context</span><span class="o">%in%</span><span class="n">traits</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene</span><span class="o">%in%</span><span class="nf">as.character</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                                                                           </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">))))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">unique</span><span class="p">()</span>
<span class="n">ROSMAP_multicontext_extracted</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_res</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_ID</span><span class="o">%in%</span><span class="n">intersected_gene_names</span><span class="p">)</span>
<span class="n">num_traits</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">traits</span><span class="p">)</span>
<span class="n">trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">num_traits</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">num_traits</span><span class="p">)</span><span class="w">  </span><span class="c1"># Initialize matrix</span>
<span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">traits</span><span class="w">   </span><span class="c1"># Assign trait names</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">gene_name</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">intersected_gene_names</span><span class="p">){</span>
<span class="c1">#gene_name &lt;- as.character(unique(mvsusie_export_lfsr_integrated_df$gene))[1]</span>
<span class="c1">#gene_name</span>
<span class="c1"># top_variants_lfsr_thres &lt;-  mvsusie_export_lfsr_integrated_df%&gt;%filter(gene_ID==gene_name)%&gt;%</span>
<span class="c1">#                             filter(.data[[coverage]]!=0)%&gt;%group_by(.data[[coverage]])%&gt;% </span>
<span class="c1">#                             filter(PIP == max(PIP, na.rm = TRUE)) %&gt;%</span>
<span class="c1">#                             distinct(variant_ID, .keep_all = TRUE) %&gt;%  # Ensure unique SNPs</span>
<span class="c1">#                             summarise(variant_ID = sample(variant_ID, 1), .groups = &quot;drop&quot;)%&gt;%</span>
<span class="c1">#                             pull(variant_ID) #top_variants_lfsr_thres</span>
<span class="n">top_variants_lfsr_thres</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_extracted</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_ID</span><span class="o">==</span><span class="n">gene_name</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                            </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">group_by</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]],</span><span class="n">event_ID</span><span class="p">)</span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                            </span><span class="nf">filter</span><span class="p">(</span><span class="n">PIP</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">PIP</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">slice_sample</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                            </span><span class="nf">pull</span><span class="p">(</span><span class="n">variant_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">unique</span><span class="p">()</span>
<span class="c1"># Step 2: Iterate through variants in df2</span>
<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">variant</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">top_variants_lfsr_thres</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1">#variant &lt;- &#39;chr20:50961558:T:C&#39;</span>
<span class="w">  </span><span class="c1"># Extract rows from df1 where the marker matches the variant in df2</span>
<span class="w">  </span><span class="n">variant_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mvsusie_export_lfsr_integrated_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_ID</span><span class="o">==</span><span class="n">gene_name</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">filter</span><span class="p">(</span><span class="n">variant_ID</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">variant</span><span class="p">)</span>
<span class="w">  </span><span class="c1">#variant_data</span>
<span class="w">  </span><span class="c1"># If variant is found in multiple traits, compare effect signs</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span>
<span class="w">    </span><span class="c1"># Get unique trait pairs</span>
<span class="w">    </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="o">:</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span>
<span class="w">        </span><span class="n">trait_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">event_ID</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
<span class="w">        </span><span class="n">trait_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">event_ID</span><span class="p">[</span><span class="n">j</span><span class="p">]</span>
<span class="w">        </span>
<span class="w">        </span><span class="n">effect_sign_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">effect_sign</span><span class="p">[</span><span class="n">i</span><span class="p">]</span>
<span class="w">        </span><span class="n">effect_sign_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">effect_sign</span><span class="p">[</span><span class="n">j</span><span class="p">]</span>

<span class="w">        </span><span class="c1"># Find indices in the trait matrix</span>
<span class="n">idx_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">trait_i</span><span class="p">)</span>
<span class="n">idx_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">trait_j</span><span class="p">)</span>
<span class="w">        </span>
<span class="w">       </span><span class="c1"># Ensure idx_i is always smaller than idx_j for upper triangle updates</span>
<span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">idx_i</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">idx_j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">temp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">idx_i</span>
<span class="w">  </span><span class="n">idx_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">idx_j</span>
<span class="w">  </span><span class="n">idx_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">temp</span>
<span class="p">}</span>

<span class="c1"># Update matrix correctly with only +1 increments</span>
<span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">effect_sign_i</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">effect_sign_j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">          </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_i</span><span class="p">,</span><span class="w"> </span><span class="n">idx_j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_i</span><span class="p">,</span><span class="w"> </span><span class="n">idx_j</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Always upper triangle</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">          </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_j</span><span class="p">,</span><span class="w"> </span><span class="n">idx_i</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">idx_j</span><span class="p">,</span><span class="w"> </span><span class="n">idx_i</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Always lower triangle</span>
<span class="p">}</span>
<span class="w">      </span><span class="p">}</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span>
<span class="w">  </span><span class="c1"># Step 3: Check if the variant is **context-specific**</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">nrow</span><span class="p">(</span><span class="n">variant_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">trait_idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">which</span><span class="p">(</span><span class="n">traits</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">variant_data</span><span class="o">$</span><span class="n">event_ID</span><span class="p">)</span>
<span class="w">    </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">trait_idx</span><span class="p">,</span><span class="w"> </span><span class="n">trait_idx</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">trait_idx</span><span class="p">,</span><span class="w"> </span><span class="n">trait_idx</span><span class="p">]</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">  </span><span class="c1"># Diagonal increment</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>
<span class="p">}</span>


<span class="n">trait_matrix</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A matrix: 23  23 of type dbl</caption>
<thead>
	<tr><th></th><th scope=col>AC_DeJager_eQTL</th><th scope=col>DLPFC_DeJager_eQTL</th><th scope=col>PCC_DeJager_eQTL</th><th scope=col>Exc_mega_eQTL</th><th scope=col>Ast_Kellis_eQTL</th><th scope=col>Ast_mega_eQTL</th><th scope=col>Exc_DeJager_eQTL</th><th scope=col>Exc_Kellis_eQTL</th><th scope=col>Inh_DeJager_eQTL</th><th scope=col>Inh_Kellis_eQTL</th><th scope=col></th><th scope=col>Ast_DeJager_eQTL</th><th scope=col>OPC_Kellis_eQTL</th><th scope=col>Oli_Kellis_eQTL</th><th scope=col>DLPFC_Bennett_pQTL</th><th scope=col>OPC_DeJager_eQTL</th><th scope=col>OPC_mega_eQTL</th><th scope=col>monocyte_ROSMAP_eQTL</th><th scope=col>Mic_Kellis_eQTL</th><th scope=col>Mic_mega_eQTL</th><th scope=col>Mic_DeJager_eQTL</th></tr>
</thead>
<tbody>
	<tr><th scope=row>AC_DeJager_eQTL</th><td>368</td><td>9297</td><td>9384</td><td>4624</td><td>2394</td><td>2900</td><td>4807</td><td>4253</td><td>3661</td><td>3370</td><td></td><td>3531</td><td>2017</td><td>3321</td><td>1949</td><td>1074</td><td>1323</td><td>1666</td><td>686</td><td>520</td><td>798</td></tr>
	<tr><th scope=row>DLPFC_DeJager_eQTL</th><td>262</td><td> 148</td><td>9594</td><td>5587</td><td>2543</td><td>3081</td><td>5776</td><td>5205</td><td>4154</td><td>3826</td><td></td><td>3716</td><td>2165</td><td>3555</td><td>1995</td><td>1161</td><td>1426</td><td>1645</td><td>680</td><td>513</td><td>792</td></tr>
	<tr><th scope=row>PCC_DeJager_eQTL</th><td> 91</td><td> 105</td><td>   7</td><td>5117</td><td>2299</td><td>2772</td><td>5320</td><td>4762</td><td>3812</td><td>3507</td><td></td><td>3386</td><td>1978</td><td>3069</td><td>1991</td><td>1035</td><td>1265</td><td>1576</td><td>643</td><td>474</td><td>743</td></tr>
	<tr><th scope=row>Exc_mega_eQTL</th><td>976</td><td> 611</td><td> 600</td><td>   9</td><td>2005</td><td>2497</td><td>7352</td><td>6910</td><td>4837</td><td>4651</td><td></td><td>2765</td><td>2044</td><td>3109</td><td>1110</td><td>1160</td><td>1479</td><td> 827</td><td>590</td><td>481</td><td>638</td></tr>
	<tr><th scope=row>Ast_Kellis_eQTL</th><td>295</td><td> 187</td><td> 218</td><td> 272</td><td>   0</td><td>3002</td><td>1920</td><td>1973</td><td>1906</td><td>1937</td><td></td><td>2994</td><td>1722</td><td>1918</td><td> 464</td><td>1053</td><td>1264</td><td> 598</td><td>478</td><td>331</td><td>526</td></tr>
	<tr><th scope=row>Ast_mega_eQTL</th><td>370</td><td> 246</td><td> 290</td><td> 345</td><td>   0</td><td>   4</td><td>2382</td><td>2369</td><td>2331</td><td>2270</td><td></td><td>3577</td><td>1956</td><td>2330</td><td> 580</td><td>1206</td><td>1537</td><td> 690</td><td>578</td><td>452</td><td>649</td></tr>
	<tr><th scope=row>Exc_DeJager_eQTL</th><td>913</td><td> 579</td><td> 581</td><td>   4</td><td> 248</td><td> 312</td><td>   8</td><td>6673</td><td>4998</td><td>4510</td><td></td><td>2886</td><td>1968</td><td>3000</td><td>1130</td><td>1163</td><td>1404</td><td> 913</td><td>541</td><td>457</td><td>635</td></tr>
	<tr><th scope=row>Exc_Kellis_eQTL</th><td>818</td><td> 518</td><td> 532</td><td>   5</td><td> 189</td><td> 255</td><td>   7</td><td>  30</td><td>4515</td><td>4602</td><td></td><td>2611</td><td>2002</td><td>3044</td><td> 966</td><td>1079</td><td>1381</td><td> 771</td><td>562</td><td>446</td><td>600</td></tr>
	<tr><th scope=row>Inh_DeJager_eQTL</th><td>693</td><td> 478</td><td> 481</td><td> 106</td><td> 143</td><td> 180</td><td>  91</td><td>  89</td><td>   6</td><td>4674</td><td></td><td>2757</td><td>1954</td><td>2796</td><td> 736</td><td>1143</td><td>1363</td><td> 802</td><td>522</td><td>410</td><td>618</td></tr>
	<tr><th scope=row>Inh_Kellis_eQTL</th><td>651</td><td> 450</td><td> 465</td><td> 101</td><td> 131</td><td> 171</td><td>  91</td><td>  73</td><td>   4</td><td>  18</td><td></td><td>2481</td><td>1975</td><td>2860</td><td> 656</td><td>1040</td><td>1332</td><td> 752</td><td>545</td><td>415</td><td>572</td></tr>
	<tr><th scope=row>Inh_mega_eQTL</th><td>736</td><td> 504</td><td> 523</td><td> 115</td><td> 160</td><td> 210</td><td> 101</td><td>  94</td><td>  10</td><td>   5</td><td></td><td>2698</td><td>2070</td><td>2948</td><td> 751</td><td>1156</td><td>1507</td><td> 788</td><td>620</td><td>498</td><td>686</td></tr>
	<tr><th scope=row>Oli_DeJager_eQTL</th><td>439</td><td> 307</td><td> 358</td><td> 350</td><td>  86</td><td> 102</td><td> 309</td><td> 253</td><td> 191</td><td> 164</td><td></td><td>2235</td><td>1709</td><td>3724</td><td> 475</td><td>1106</td><td>1332</td><td> 794</td><td>501</td><td>416</td><td>590</td></tr>
	<tr><th scope=row>Oli_mega_eQTL</th><td>489</td><td> 344</td><td> 384</td><td> 375</td><td> 101</td><td> 128</td><td> 331</td><td> 272</td><td> 198</td><td> 171</td><td></td><td>2362</td><td>1895</td><td>4136</td><td> 530</td><td>1150</td><td>1495</td><td> 829</td><td>571</td><td>470</td><td>627</td></tr>
	<tr><th scope=row>Ast_DeJager_eQTL</th><td>394</td><td> 267</td><td> 310</td><td> 337</td><td>   1</td><td>   2</td><td> 315</td><td> 245</td><td> 197</td><td> 166</td><td></td><td>   9</td><td>2035</td><td>2459</td><td> 711</td><td>1240</td><td>1428</td><td> 851</td><td>540</td><td>383</td><td>644</td></tr>
	<tr><th scope=row>OPC_Kellis_eQTL</th><td>335</td><td> 226</td><td> 238</td><td> 253</td><td>  29</td><td>  53</td><td> 226</td><td> 201</td><td> 155</td><td> 129</td><td></td><td>  45</td><td>   6</td><td>2060</td><td> 373</td><td>1707</td><td>1930</td><td> 554</td><td>480</td><td>349</td><td>531</td></tr>
	<tr><th scope=row>Oli_Kellis_eQTL</th><td>467</td><td> 340</td><td> 372</td><td> 365</td><td>  63</td><td>  85</td><td> 326</td><td> 264</td><td> 191</td><td> 162</td><td></td><td>  91</td><td>  53</td><td>  16</td><td> 604</td><td>1128</td><td>1423</td><td> 845</td><td>542</td><td>427</td><td>576</td></tr>
	<tr><th scope=row>DLPFC_Bennett_pQTL</th><td>176</td><td> 181</td><td> 130</td><td> 324</td><td> 139</td><td> 194</td><td> 295</td><td> 283</td><td> 259</td><td> 240</td><td></td><td> 187</td><td> 157</td><td> 218</td><td> 519</td><td> 200</td><td> 240</td><td> 368</td><td>121</td><td>111</td><td>164</td></tr>
	<tr><th scope=row>OPC_DeJager_eQTL</th><td>273</td><td> 176</td><td> 189</td><td> 219</td><td>  22</td><td>  33</td><td> 197</td><td> 168</td><td> 156</td><td> 125</td><td></td><td>  32</td><td>   0</td><td>  46</td><td> 115</td><td>   2</td><td>1761</td><td> 304</td><td>360</td><td>273</td><td>426</td></tr>
	<tr><th scope=row>OPC_mega_eQTL</th><td>300</td><td> 197</td><td> 212</td><td> 244</td><td>  27</td><td>  45</td><td> 218</td><td> 183</td><td> 147</td><td> 130</td><td></td><td>  39</td><td>   0</td><td>  52</td><td> 133</td><td>   0</td><td>   2</td><td> 365</td><td>455</td><td>386</td><td>503</td></tr>
	<tr><th scope=row>monocyte_ROSMAP_eQTL</th><td>155</td><td> 150</td><td> 145</td><td> 207</td><td>  57</td><td>  72</td><td> 205</td><td> 177</td><td> 151</td><td> 129</td><td></td><td>  91</td><td>  56</td><td>  81</td><td>  77</td><td>  46</td><td>  58</td><td>  65</td><td>387</td><td>269</td><td>443</td></tr>
	<tr><th scope=row>Mic_Kellis_eQTL</th><td>137</td><td> 112</td><td> 106</td><td>  90</td><td>  33</td><td>  40</td><td>  82</td><td>  65</td><td>  64</td><td>  51</td><td></td><td>  36</td><td>  22</td><td>  48</td><td>  78</td><td>  25</td><td>  28</td><td>  23</td><td>  2</td><td>708</td><td>874</td></tr>
	<tr><th scope=row>Mic_mega_eQTL</th><td>128</td><td> 113</td><td> 100</td><td>  77</td><td>  37</td><td>  47</td><td>  71</td><td>  57</td><td>  54</td><td>  39</td><td></td><td>  41</td><td>  27</td><td>  48</td><td>  65</td><td>  27</td><td>  35</td><td>  22</td><td>  0</td><td>  2</td><td>754</td></tr>
	<tr><th scope=row>Mic_DeJager_eQTL</th><td>157</td><td> 143</td><td> 131</td><td> 120</td><td>  37</td><td>  50</td><td> 120</td><td>  85</td><td>  85</td><td>  74</td><td></td><td>  49</td><td>  31</td><td>  59</td><td>  70</td><td>  30</td><td>  35</td><td>  25</td><td>  0</td><td>  2</td><td> 15</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="multicontext-posterior-effect-sign-heatmap">
<h2>Multicontext posterior effect sign heatmap<a class="headerlink" href="#multicontext-posterior-effect-sign-heatmap" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">),</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))),</span><span class="w"> </span>
<span class="w">                           </span><span class="nf">order</span><span class="p">(</span><span class="nf">factor</span><span class="p">(</span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">),</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)))]</span>
<span class="n">rev_trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[,</span><span class="nf">ncol</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="c1">#rev_trait_matrix &lt;- t(trait_matrix)</span>
<span class="n">rev_trait_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_trait_matrix</span><span class="p">)</span>

<span class="c1"># Calculate the fraction matrix</span>
<span class="c1"># Assuming rev_trait_matrix is your input square matrix</span>
<span class="n">context_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">rev_trait_matrix</span><span class="p">)</span>
<span class="n">n</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">context_names</span><span class="p">)</span>

<span class="c1"># Initialize output matrix</span>
<span class="n">proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">context_names</span><span class="w"> </span>

<span class="n">pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">matrix</span><span class="p">(</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">context_names</span><span class="w"> </span>

<span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">seq_len</span><span class="p">(</span><span class="n">n</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">j</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="nf">seq_len</span><span class="p">(</span><span class="n">n</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="n">j</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="c1"># Extract 2x2 submatrix</span>
<span class="w">      </span><span class="n">trait_i</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)[</span><span class="n">i</span><span class="p">]</span>
<span class="w">      </span><span class="n">trait_j</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">)[</span><span class="n">j</span><span class="p">]</span>
<span class="w">      </span><span class="n">pattern</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_c</span><span class="p">(</span><span class="n">trait_i</span><span class="p">,</span><span class="w"> </span><span class="n">trait_j</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;|&quot;</span><span class="p">)</span>
<span class="w">      </span><span class="n">submat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">)]</span>
<span class="w">      </span><span class="n">ROSMAP_multicontext_res_filtered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_extracted</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                        </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="n">pattern</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">                                        </span><span class="nf">select</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]])</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">unique</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span>
<span class="w">      </span><span class="n">pairwise_total_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">submat</span><span class="p">)</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="nf">diag</span><span class="p">(</span><span class="n">submat</span><span class="p">))</span>
<span class="w">      </span><span class="n">total_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise_total_sum</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="nf">dim</span><span class="p">(</span><span class="n">ROSMAP_multicontext_res_filtered</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span>
<span class="w">      </span><span class="n">value</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span>

<span class="w">      </span><span class="c1"># Compute and assign proportion</span>
<span class="w">      </span><span class="n">proportion_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">round</span><span class="p">((</span><span class="n">value</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_sum</span><span class="p">)</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
<span class="w">      </span><span class="n">pairwise_proportion_matrix</span><span class="p">[</span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">j</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">round</span><span class="p">((</span><span class="n">value</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">pairwise_total_sum</span><span class="p">)</span><span class="o">*</span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>

<span class="c1">#proportion_matrix</span>
<span class="n">rev_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">proportion_matrix</span><span class="p">[,</span><span class="w"> </span><span class="nf">ncol</span><span class="p">(</span><span class="n">proportion_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="n">rev_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)</span>

<span class="c1"># Convert matrix into long format for ggplot</span>
<span class="n">plot_data2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">reshape2</span><span class="o">::</span><span class="nf">melt</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">plot_data2</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Trait1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Trait2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Count&quot;</span><span class="p">)</span>

<span class="c1"># 1. Identify which rows need filling</span>
<span class="n">na_idx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">is.na</span><span class="p">(</span><span class="n">plot_data2</span><span class="o">$</span><span class="n">Count</span><span class="p">)</span>

<span class="c1"># 2. Pull from the matrix by name for those rows</span>
<span class="c1"># Find numeric row/col indices</span>
<span class="n">r</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">match</span><span class="p">(</span><span class="n">plot_data2</span><span class="o">$</span><span class="n">Trait1</span><span class="p">[</span><span class="n">na_idx</span><span class="p">],</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))</span>
<span class="n">c</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">match</span><span class="p">(</span><span class="n">plot_data2</span><span class="o">$</span><span class="n">Trait2</span><span class="p">[</span><span class="n">na_idx</span><span class="p">],</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">trait_matrix</span><span class="p">))</span>

<span class="c1"># Do element-wise lookup</span>
<span class="n">plot_data2</span><span class="o">$</span><span class="n">Count</span><span class="p">[</span><span class="n">na_idx</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">trait_matrix</span><span class="p">[</span><span class="nf">cbind</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="w"> </span><span class="n">c</span><span class="p">)]</span>

<span class="c1">#proportion_matrix</span>

<span class="n">rev_pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise_proportion_matrix</span><span class="p">[,</span><span class="w"> </span><span class="nf">ncol</span><span class="p">(</span><span class="n">pairwise_proportion_matrix</span><span class="p">)</span><span class="o">:</span><span class="m">1</span><span class="p">]</span>
<span class="n">rev_pairwise_proportion_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">t</span><span class="p">(</span><span class="n">rev_pairwise_proportion_matrix</span><span class="p">)</span>
<span class="w"> </span>
<span class="n">plot_data2</span><span class="o">$</span><span class="n">Fraction</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.vector</span><span class="p">(</span><span class="n">reshape2</span><span class="o">::</span><span class="nf">melt</span><span class="p">(</span><span class="n">rev_pairwise_proportion_matrix</span><span class="p">)</span><span class="o">$</span><span class="n">value</span><span class="p">)</span>

<span class="n">credible_sum</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_extracted</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="o">!=</span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="o">%&gt;%</span>
<span class="w">                                        </span><span class="nf">select</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]])</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">unique</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                                        </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="o">!=</span><span class="m">0</span><span class="p">)</span><span class="o">%&gt;%</span><span class="nf">nrow</span><span class="p">()</span>

<span class="n">plot_data2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_data2</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">rowwise</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="n">Label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Fraction</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nf">paste0</span><span class="p">(</span><span class="n">Fraction</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;\n(&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">Count</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;)&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">total_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ROSMAP_multicontext_extracted</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">filter</span><span class="p">(</span><span class="n">event_ID</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">event_ID</span><span class="p">,</span><span class="nf">as.character</span><span class="p">(</span><span class="n">Trait1</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">select</span><span class="p">(</span><span class="n">gene_ID</span><span class="p">,</span><span class="w"> </span><span class="n">event_ID</span><span class="p">,</span><span class="w"> </span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]])</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">unique</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">filter</span><span class="p">(</span><span class="n">.data</span><span class="p">[[</span><span class="n">coverage</span><span class="p">]]</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">        </span><span class="nf">nrow</span><span class="p">()</span>
<span class="w">      </span>
<span class="w">      </span><span class="nf">round</span><span class="p">((</span><span class="n">Count</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">total_genes</span><span class="p">)</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">as.character</span><span class="p">()</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">ungroup</span><span class="p">()</span>

<span class="n">plot_data2</span><span class="o">$</span><span class="n">Trait1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">plot_data2</span><span class="o">$</span><span class="n">Trait1</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rev</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">)))</span>
<span class="n">plot_data2</span><span class="o">$</span><span class="n">Trait2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">plot_data2</span><span class="o">$</span><span class="n">Trait2</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">rev_proportion_matrix</span><span class="p">))</span>

<span class="n">plot_data2</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A tibble: 529  5</caption>
<thead>
	<tr><th scope=col>Trait1</th><th scope=col>Trait2</th><th scope=col>Count</th><th scope=col>Fraction</th><th scope=col>Label</th></tr>
	<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>Mic_DeJager_eQTL    </td><td>AC_DeJager_eQTL   </td><td>  6.11</td><td>83.56</td><td>83.56
(6.11) </td></tr>
	<tr><td>Mic_mega_eQTL       </td><td>AC_DeJager_eQTL   </td><td>  4.11</td><td>80.25</td><td>80.25
(4.11) </td></tr>
	<tr><td>Mic_Kellis_eQTL     </td><td>AC_DeJager_eQTL   </td><td>  5.33</td><td>83.35</td><td>83.35
(5.33) </td></tr>
	<tr><td>monocyte_ROSMAP_eQTL</td><td>AC_DeJager_eQTL   </td><td> 12.04</td><td>91.49</td><td>91.49
(12.04)</td></tr>
	<tr><td>OPC_mega_eQTL       </td><td>AC_DeJager_eQTL   </td><td>  9.31</td><td>81.52</td><td>81.52
(9.31) </td></tr>
	<tr><td>OPC_DeJager_eQTL    </td><td>AC_DeJager_eQTL   </td><td>  7.78</td><td>79.73</td><td>79.73
(7.78) </td></tr>
	<tr><td>DLPFC_Bennett_pQTL  </td><td>AC_DeJager_eQTL   </td><td> 13.14</td><td>91.72</td><td>91.72
(13.14)</td></tr>
	<tr><td>Oli_Kellis_eQTL     </td><td>AC_DeJager_eQTL   </td><td> 19.61</td><td>87.67</td><td>87.67
(19.61)</td></tr>
	<tr><td>OPC_Kellis_eQTL     </td><td>AC_DeJager_eQTL   </td><td> 13.37</td><td>85.76</td><td>85.76
(13.37)</td></tr>
	<tr><td>Ast_DeJager_eQTL    </td><td>AC_DeJager_eQTL   </td><td> 21.12</td><td>89.96</td><td>89.96
(21.12)</td></tr>
	<tr><td>Oli_mega_eQTL       </td><td>AC_DeJager_eQTL   </td><td> 18.56</td><td>86.33</td><td>86.33
(18.56)</td></tr>
	<tr><td>Oli_DeJager_eQTL    </td><td>AC_DeJager_eQTL   </td><td> 18.05</td><td>86.98</td><td>86.98
(18.05)</td></tr>
	<tr><td>Inh_mega_eQTL       </td><td>AC_DeJager_eQTL   </td><td> 20.36</td><td>83.23</td><td>83.23
(20.36)</td></tr>
	<tr><td>Inh_Kellis_eQTL     </td><td>AC_DeJager_eQTL   </td><td> 19.43</td><td>83.81</td><td>83.81
(19.43)</td></tr>
	<tr><td>Inh_DeJager_eQTL    </td><td>AC_DeJager_eQTL   </td><td> 20.62</td><td>84.08</td><td>84.08
(20.62)</td></tr>
	<tr><td>Exc_Kellis_eQTL     </td><td>AC_DeJager_eQTL   </td><td> 22.29</td><td>83.87</td><td>83.87
(22.29)</td></tr>
	<tr><td>Exc_DeJager_eQTL    </td><td>AC_DeJager_eQTL   </td><td> 24.20</td><td>84.04</td><td>84.04
(24.2) </td></tr>
	<tr><td>Ast_mega_eQTL       </td><td>AC_DeJager_eQTL   </td><td> 18.08</td><td>88.69</td><td>88.69
(18.08)</td></tr>
	<tr><td>Ast_Kellis_eQTL     </td><td>AC_DeJager_eQTL   </td><td> 15.79</td><td>89.03</td><td>89.03
(15.79)</td></tr>
	<tr><td>Exc_mega_eQTL       </td><td>AC_DeJager_eQTL   </td><td> 23.34</td><td>82.57</td><td>82.57
(23.34)</td></tr>
	<tr><td>PCC_DeJager_eQTL    </td><td>AC_DeJager_eQTL   </td><td> 41.73</td><td>99.04</td><td>99.04
(41.73)</td></tr>
	<tr><td>DLPFC_DeJager_eQTL  </td><td>AC_DeJager_eQTL   </td><td> 39.82</td><td>97.26</td><td>97.26
(39.82)</td></tr>
	<tr><td>AC_DeJager_eQTL     </td><td>AC_DeJager_eQTL   </td><td>368.00</td><td>   NA</td><td>3.15          </td></tr>
	<tr><td>Mic_DeJager_eQTL    </td><td>DLPFC_DeJager_eQTL</td><td>  6.06</td><td>84.71</td><td>84.71
(6.06) </td></tr>
	<tr><td>Mic_mega_eQTL       </td><td>DLPFC_DeJager_eQTL</td><td>  4.05</td><td>81.95</td><td>81.95
(4.05) </td></tr>
	<tr><td>Mic_Kellis_eQTL     </td><td>DLPFC_DeJager_eQTL</td><td>  5.28</td><td>85.86</td><td>85.86
(5.28) </td></tr>
	<tr><td>monocyte_ROSMAP_eQTL</td><td>DLPFC_DeJager_eQTL</td><td> 11.87</td><td>91.64</td><td>91.64
(11.87)</td></tr>
	<tr><td>OPC_mega_eQTL       </td><td>DLPFC_DeJager_eQTL</td><td> 10.02</td><td>87.86</td><td>87.86
(10.02)</td></tr>
	<tr><td>OPC_DeJager_eQTL    </td><td>DLPFC_DeJager_eQTL</td><td>  8.40</td><td>86.84</td><td>86.84
(8.4)  </td></tr>
	<tr><td>DLPFC_Bennett_pQTL  </td><td>DLPFC_DeJager_eQTL</td><td> 13.44</td><td>91.68</td><td>91.68
(13.44)</td></tr>
	<tr><td></td><td></td><td></td><td></td><td></td></tr>
	<tr><td>Exc_DeJager_eQTL    </td><td>Mic_mega_eQTL   </td><td> 0.77</td><td>13.45</td><td>13.45
(0.77)</td></tr>
	<tr><td>Ast_mega_eQTL       </td><td>Mic_mega_eQTL   </td><td> 0.88</td><td> 9.42</td><td>9.42
(0.88) </td></tr>
	<tr><td>Ast_Kellis_eQTL     </td><td>Mic_mega_eQTL   </td><td> 0.82</td><td>10.05</td><td>10.05
(0.82)</td></tr>
	<tr><td>Exc_mega_eQTL       </td><td>Mic_mega_eQTL   </td><td> 0.84</td><td>13.80</td><td>13.8
(0.84) </td></tr>
	<tr><td>PCC_DeJager_eQTL    </td><td>Mic_mega_eQTL   </td><td> 0.85</td><td>17.42</td><td>17.42
(0.85)</td></tr>
	<tr><td>DLPFC_DeJager_eQTL  </td><td>Mic_mega_eQTL   </td><td> 0.89</td><td>18.05</td><td>18.05
(0.89)</td></tr>
	<tr><td>AC_DeJager_eQTL     </td><td>Mic_mega_eQTL   </td><td> 1.01</td><td>19.75</td><td>19.75
(1.01)</td></tr>
	<tr><td>Mic_DeJager_eQTL    </td><td>Mic_DeJager_eQTL</td><td>15.00</td><td>   NA</td><td>1.07         </td></tr>
	<tr><td>Mic_mega_eQTL       </td><td>Mic_DeJager_eQTL</td><td> 0.08</td><td> 0.26</td><td>0.26
(0.08) </td></tr>
	<tr><td>Mic_Kellis_eQTL     </td><td>Mic_DeJager_eQTL</td><td> 0.00</td><td> 0.00</td><td>0
(0)       </td></tr>
	<tr><td>monocyte_ROSMAP_eQTL</td><td>Mic_DeJager_eQTL</td><td> 0.70</td><td> 5.34</td><td>5.34
(0.7)  </td></tr>
	<tr><td>OPC_mega_eQTL       </td><td>Mic_DeJager_eQTL</td><td> 0.89</td><td> 6.51</td><td>6.51
(0.89) </td></tr>
	<tr><td>OPC_DeJager_eQTL    </td><td>Mic_DeJager_eQTL</td><td> 0.85</td><td> 6.58</td><td>6.58
(0.85) </td></tr>
	<tr><td>DLPFC_Bennett_pQTL  </td><td>Mic_DeJager_eQTL</td><td> 1.54</td><td>29.91</td><td>29.91
(1.54)</td></tr>
	<tr><td>Oli_Kellis_eQTL     </td><td>Mic_DeJager_eQTL</td><td> 0.88</td><td> 9.29</td><td>9.29
(0.88) </td></tr>
	<tr><td>OPC_Kellis_eQTL     </td><td>Mic_DeJager_eQTL</td><td> 0.64</td><td> 5.52</td><td>5.52
(0.64) </td></tr>
	<tr><td>Ast_DeJager_eQTL    </td><td>Mic_DeJager_eQTL</td><td> 0.76</td><td> 7.07</td><td>7.07
(0.76) </td></tr>
	<tr><td>Oli_mega_eQTL       </td><td>Mic_DeJager_eQTL</td><td> 1.24</td><td>11.19</td><td>11.19
(1.24)</td></tr>
	<tr><td>Oli_DeJager_eQTL    </td><td>Mic_DeJager_eQTL</td><td> 1.15</td><td>10.47</td><td>10.47
(1.15)</td></tr>
	<tr><td>Inh_mega_eQTL       </td><td>Mic_DeJager_eQTL</td><td> 1.15</td><td>11.37</td><td>11.37
(1.15)</td></tr>
	<tr><td>Inh_Kellis_eQTL     </td><td>Mic_DeJager_eQTL</td><td> 1.05</td><td>11.46</td><td>11.46
(1.05)</td></tr>
	<tr><td>Inh_DeJager_eQTL    </td><td>Mic_DeJager_eQTL</td><td> 1.14</td><td>12.09</td><td>12.09
(1.14)</td></tr>
	<tr><td>Exc_Kellis_eQTL     </td><td>Mic_DeJager_eQTL</td><td> 0.96</td><td>12.41</td><td>12.41
(0.96)</td></tr>
	<tr><td>Exc_DeJager_eQTL    </td><td>Mic_DeJager_eQTL</td><td> 1.25</td><td>15.89</td><td>15.89
(1.25)</td></tr>
	<tr><td>Ast_mega_eQTL       </td><td>Mic_DeJager_eQTL</td><td> 0.87</td><td> 7.15</td><td>7.15
(0.87) </td></tr>
	<tr><td>Ast_Kellis_eQTL     </td><td>Mic_DeJager_eQTL</td><td> 0.76</td><td> 6.57</td><td>6.57
(0.76) </td></tr>
	<tr><td>Exc_mega_eQTL       </td><td>Mic_DeJager_eQTL</td><td> 1.26</td><td>15.83</td><td>15.83
(1.26)</td></tr>
	<tr><td>PCC_DeJager_eQTL    </td><td>Mic_DeJager_eQTL</td><td> 1.07</td><td>14.99</td><td>14.99
(1.07)</td></tr>
	<tr><td>DLPFC_DeJager_eQTL  </td><td>Mic_DeJager_eQTL</td><td> 1.09</td><td>15.29</td><td>15.29
(1.09)</td></tr>
	<tr><td>AC_DeJager_eQTL     </td><td>Mic_DeJager_eQTL</td><td> 1.20</td><td>16.44</td><td>16.44
(1.2) </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#pdf(&quot;ROSMAP_multicontext_effect_sign_heatmap_4063_genes.pdf&quot;, width =10, height = 8)</span>
<span class="c1"># Plot heatmap using ggplot2</span>
<span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">plot_data2</span><span class="p">,</span><span class="w"> </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Trait1</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Trait2</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Fraction</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_tile</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_fill_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">na.value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_text</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Label</span><span class="p">),</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">  </span><span class="c1"># Add text with count and fraction</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Proportion&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Multicontext Posterior Effect Sign Heatmap&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">45</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">),</span>
<span class="w">        </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">),</span>
<span class="w">        </span><span class="n">plot.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">25</span><span class="p">,</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">        </span><span class="n">legend.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span><span class="n">face</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;bold&quot;</span><span class="p">),</span>
<span class="w">       </span><span class="n">legend.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">))</span>
<span class="c1">#dev.off()</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">options</span><span class="p">(</span><span class="n">repr.plot.width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">,</span><span class="w"> </span><span class="n">repr.plot.height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">)</span>
<span class="n">p2</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/bf4f4b1cba14262086b1e9b5effd8b9153f082da0c0c92dde2b0a0eb0a5acb7e.png"><img alt="../../_images/bf4f4b1cba14262086b1e9b5effd8b9153f082da0c0c92dde2b0a0eb0a5acb7e.png" src="../../_images/bf4f4b1cba14262086b1e9b5effd8b9153f082da0c0c92dde2b0a0eb0a5acb7e.png" style="width: 1800px; height: 1800px;" />
</a>
</div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./main_text/3_multi_context"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">Marginal vs multicontext effect sign heatmap</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#load-libraries">Load libraries</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#marginal-posterior-effect-sign">Marginal Posterior Effect Sign</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#marginal-trait-matrix">Marginal trait matrix</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#marginal-posterior-effect-sign-heatmap">Marginal posterior effect sign heatmap</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#multicontext-effect-posterior-effect-sign">Multicontext Effect Posterior Effect Sign</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multicontext-trait-matrix">Multicontext trait matrix</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multicontext-posterior-effect-sign-heatmap">Multicontext posterior effect sign heatmap</a></li>
</ul>
</li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
       Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>