
<!DOCTYPE html>


<html lang="en" data-content_root="../../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Generate Gene List for FunGen xQTL Analysis &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/5_AD_xQTL_integration/staging/FunGen_xQTL_gene_enrichment';</script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../1_project_overview/index.html">Multi-omics QTL atlas of the aging human brain</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../2_single_context_cis/index.html">Fine-mapped QTL landscape across brain molecular phenotypes</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../2_single_context_cis/Figure_1b_num_xQTL_loci_single_context.html">Figure 1b. Number of detectable and mappable xQTL loci in single−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../2_single_context_cis/Figure_EOO.html">EOO results</a></li>

<li class="toctree-l2"><a class="reference internal" href="../../2_single_context_cis/Figure_S2_single_context_cs_distance.html">Frequency of Events in Distance Category</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../2_single_context_cis/Figure_sLDSC_GWAS_specific.html">SLDSC</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../2_single_context_cis/Figure_sLDSC_meta_analysis.html">Visualization of SLDSC results</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../../3_single_context_multigene_trans/index.html">Coordinated gene regulation and long-range genetic effects</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../4_multi_context/index.html">Cascading genetic effects across molecular contexts</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../4_multi_context/Figure_2b_num_xQTL_loci_multi_context.html">Number of detectable and mappable xQTL loci in ROSMAP multi−context fine−mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../4_multi_context/Figure_2c_multicontext_effect_sign.html">Marginal vs multicontext effect sign heatmap</a></li>

<li class="toctree-l2"><a class="reference internal" href="../../4_multi_context/Figure_watershed_posterior_plot.html">Watershed Posterior</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../index.html">Regulatory architecture of AD susceptibility loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../Figure_5a_AD_GWAS_loci_harmonized.html">AD GWAS loci harmonization and refinement</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Figure_5b_AD_GWAS_loci_properties.html">Refined AD GWAS loci properties</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Figure_5c_AD_loci_xQTL_annotated.html">AD loci “explained” by xQTL</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Figure_5d_AD_loci_table_summary.html">Summary of FunGen-xQTL AD loci table</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Figure_5e_GSEA.html">Gene Set Enrichment Analysis for AD Risk Genes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Figure_5f_PPI.html">PPI networks for selected AD risk genes in xQTL loci</a></li>

<li class="toctree-l2"><a class="reference internal" href="../Figure_5g_cV2F.html">Consensus Variant-to-Function Score for AD</a></li>
<li class="toctree-l2"><a class="reference internal" href="../Figure_5h_GWAS_enrichment.html">GWAS Enrichment</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../6_AD_xQTL_genes/index.html">Molecular dissection of prioritized AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../6_AD_xQTL_genes/Figure_6_AD_loci_table_visualization.html">Visualization of FunGen-xQTL AD Loci Table</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../7_AD_gene_prioritized/index.html">Molecular insights and gene prioritization at key AD loci</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../7_AD_gene_prioritized/APP_case_example.html">APP</a></li>




<li class="toctree-l2"><a class="reference internal" href="../../7_AD_gene_prioritized/ZYX_case_example.html">Molecular insights and gene prioritization at key AD loci</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/5_AD_xQTL_integration/staging/FunGen_xQTL_gene_enrichment.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../../_sources/main_text/5_AD_xQTL_integration/staging/FunGen_xQTL_gene_enrichment.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Generate Gene List for FunGen xQTL Analysis</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#overview">Overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-gene-lists">load gene lists</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-colocboost-results-for-double-check">load colocboost results for double check</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#identify-overlapping-genes-between-metabolite-colocalized-genes-and-ad-genes-to-uncover-potential-links-between-ad-and-metabolites">Identify overlapping genes between metabolite-colocalized genes and AD genes to uncover potential links between AD and metabolites</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#split-twas-gene-lists">split twas gene lists</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#go-enrichment">GO enrichment</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#real-gvc-list-20-genes-from-gvc-table2">real gvc list - 20 genes from GVC table2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gvc-list1-76-genes">gvc list1 -76 genes</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#strong-list-overlapped-between-xqtl-gvc-and-twas">strong list - overlapped between xQTL, GVC and TWAS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-list">xQTL list</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#go">GO</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#kegg">KEGG</a><ul class="nav section-nav flex-column">
<li class="toc-h5 nav-item toc-entry"><a class="reference internal nav-link" href="#plot-ad-path">plot AD path</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-list">xQTL only list</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">GO</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">KEGG</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#amyloid-pathway-genes">Amyloid pathway genes</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#context-gene-diff-list">context: gene diff list</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#twas-only-list">twas only list</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-case-ypel3">multi gene case - YPEL3</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-case-nbeal1">multi gene case - NBEAL1</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#ppi">PPI</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl">xQTL</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-genes-labelled-by-methods">xQTL only genes labelled by methods</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-genes-labelled-by-non-proxy-ad-genes">xQTL only genes labelled by non proxy AD genes</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-gvs-table2">xQTL + GVS table2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#check-gene-expression">check gene expression</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#prediction">Prediction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#model-with-glmnet">model with glmnet</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#model-with-glm">model with glm</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#cross-sample">cross sample</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#grn">GRN</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#fisher-test-between-xqtl-and-xqtl-only-genes">fisher test between xQTL and xQTL only genes</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multigene-grn-ypel3">Multigene GRN -YPEL3</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multigene-grn-stx4">Multigene GRN - STX4</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-trans-function">Multi Gene trans function</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#program1-go">program1 GO</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#split">split</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#another-top-split">another top split</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#program1-kegg">program1 KEGG</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id3">split</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#program2-go">program2 GO</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id4">split</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id5">another top split</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id6">check gene expression</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="generate-gene-list-for-fungen-xqtl-analysis">
<h1>Generate Gene List for FunGen xQTL Analysis<a class="headerlink" href="#generate-gene-list-for-fungen-xqtl-analysis" title="Link to this heading">#</a></h1>
<section id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Link to this heading">#</a></h2>
<p>This analysis is trying to understand the functional pathway signals for the gene list we found from xQTL data. We are trying to answer the question – we have limit genes overlapped with GVC and TWAS, does that true or we did incorrect analysis. For this, we need to answer it through:</p>
<ol class="arabic simple">
<li><p>functional pathways enrichment</p>
<ol class="arabic simple">
<li><p>for GVC list – to check the pathways for gold standard</p></li>
<li><p>for genes in xQTL data we have – to check the overview pathways we have;</p></li>
<li><p>for specific genes in xQTL data only – to check the additional pathways we have;</p></li>
</ol>
</li>
<li><p>PPI analysis – for the protein level evidence may show the indirect effect</p></li>
<li><p>Are xQTL-only genes missed by TWAS/GVC because they are tissue-specific/cell type specific? – need to check the resource of those xQTL-only gene – or aggregate</p></li>
<li><p>Identify if xQTL-only genes are regulators or targets in known gene regulatory circuits.</p></li>
<li><p>The prediction to disease? – AUC</p></li>
</ol>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(tidyverse)
library(data.table)
</pre></div>
</div>
</div>
</div>
<section id="load-gene-lists">
<h3>load gene lists<a class="headerlink" href="#load-gene-lists" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gene_list &lt;- fread(&quot;/data/resource/AD_genes_FunGen_AD_GVC_xQTL_20250325.tsv&quot;)
twas_list &lt;- fread(&quot;/data/resource/AD_genes_FunGen_AD_twas_GVC_xQTL_20250325.tsv&quot;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gene_list %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 x 7</caption>
<thead>
	<tr><th scope=col>gene_id</th><th scope=col>resource_in_FunGen_xQTL</th><th scope=col>data_resources</th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>ENSG00000002919</td><td>ColocBoost</td><td>FunGen_xQTL </td><td>17</td><td> 48103356</td><td> 48103357</td><td>SNX11 </td></tr>
	<tr><td>ENSG00000005100</td><td>ColocBoost</td><td>FunGen_xQTL </td><td>17</td><td>  5468981</td><td>  5468982</td><td>DHX33 </td></tr>
	<tr><td>ENSG00000006530</td><td>ColocBoost</td><td>FunGen_xQTL </td><td> 7</td><td>141551277</td><td>141551278</td><td>AGK   </td></tr>
	<tr><td>ENSG00000008294</td><td>ColocBoost</td><td>FunGen_xQTL </td><td>17</td><td> 51120867</td><td> 51120868</td><td>SPAG9 </td></tr>
	<tr><td>ENSG00000013503</td><td>ColocBoost</td><td>FunGen_xQTL </td><td>12</td><td>106357747</td><td>106357748</td><td>POLR3B</td></tr>
	<tr><td>ENSG00000015592</td><td>          </td><td>FunGen_AD_PI</td><td> 8</td><td> 27258419</td><td> 27258420</td><td>STMN4 </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>real_gvc_list &lt;- gene_list %&gt;% filter(str_detect(data_resources, &#39;ADSP_GVC&#39;)) # 20 genes from GVC table2
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gvc_list &lt;- gene_list %&gt;% filter(str_detect(data_resources, &#39;FunGen_AD_PI|ADSP_GVC&#39;)) # 20 genes + PIs&#39; genes
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_list &lt;- gene_list %&gt;% filter(str_detect(data_resources, &#39;FunGen_xQTL&#39;)) # genes from xQTL evidence
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>intersect(gvc_list$gene_id, gene_list$gene_id) %&gt;% length # the overlap between xQTL and gvc
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">62</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Intersect multiple gene_id vectors
strong_list &lt;- Reduce(intersect, list(gvc_list$gene_id, gene_list$gene_id, twas_list$gene_id)) # the overlap between xQTL and gvc and twas
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>twas_xqtl_list &lt;- intersect(twas_list$gene_id, gene_list$gene_id)
twas_xqtl_list %&gt;% length
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">96</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gene_ref &lt;- fread(&#39;/data/resource/references/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.region_list&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gvc_table1 &lt;- fread(&#39;../Mar/GVC - table1.tsv&#39;)
# Set column names from the second row and remove header rows
colnames(gvc_table1) &lt;- gvc_table1[2,] %&gt;% as.character()
gvc_table1 &lt;- gvc_table1[-c(1:2),]

# Map gene names to gene reference data and add resource identifier
gvc_table1 &lt;- gene_ref %&gt;% 
  filter(gene_name %in% gvc_table1$`Reported Gene/ Closest gene`) %&gt;% 
  mutate(data_resource = &#39;FunGen_ADSP&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_list %&gt;% dim
gene_list %&gt;% dim
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>240</li><li>7</li></ol>
</div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>287</li><li>7</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_only &lt;- Reduce(setdiff, list(xQTL_list$gene_id, gvc_list$gene_id, twas_list$gene_id)) # the genes unqiue from xQTL, NOT from gvc and twas
xQTL_only &lt;- xQTL_list %&gt;% filter(gene_id %in% xQTL_only)
xQTL_only %&gt;% dim
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>151</li><li>7</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_only %&gt;% pull(resource_in_FunGen_xQTL) %&gt;% table # check their resources from FunGen xQTL
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>.
                                                  ColocBoost 
                                                         116 
                         ColocBoost,multi_context,multi_gene 
                                                           1 
                                   ColocBoost,single_context 
                                                           1 
                     ColocBoost,single_context,multi_context 
                                                           4 
           ColocBoost,single_context,multi_context,MetaBrain 
                                                           2 
ColocBoost,single_context,multi_context,MetaBrain,multi_gene 
                                                           2 
                        ColocBoost,single_context,multi_gene 
                                                           2 
                                                   MetaBrain 
                                                           1 
                                                  multi_gene 
                                                           2 
                                              single_context 
                                                          10 
                                    single_context,MetaBrain 
                                                           1 
                                single_context,multi_context 
                                                           1 
                      single_context,multi_context,MetaBrain 
                                                           3 
           single_context,multi_context,MetaBrain,multi_gene 
                                                           2 
                     single_context,multi_context,multi_gene 
                                                           1 
                                   single_context,multi_gene 
                                                           2 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>twas_diff_list &lt;- Reduce(setdiff, list(twas_list$gene_id, gene_list$gene_id, gvc_list$gene_id)) # the genes unqiue from twas, NOT from gvc and xQTL
twas_diff_list %&gt;% length
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">271</div></div>
</div>
</section>
<section id="load-colocboost-results-for-double-check">
<h3>load colocboost results for double check<a class="headerlink" href="#load-colocboost-results-for-double-check" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>cb_apr25 &lt;- fread(&#39;/data/analysis_result//ColocBoost//export//summary//ad_xqtl_colocboost_export_filtered.bed.gz&#39;) # load colocboost results for double check
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>cb_apr25 %&gt;% filter(gene_ID %in% xQTL_only$gene_id) %&gt;% pull(gene_ID) %&gt;% unique %&gt;% length # 128 xQTL only genes are from colocboost
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_only_1_se_cb &lt;- cb_apr25 %&gt;% filter(gene_ID %in% xQTL_only$gene_id)  %&gt;% 
              mutate(n_count = str_count(event_ID, &#39;;&#39;) + 1, 
                     only_1_se = ifelse((str_detect(event_ID, &#39;sQTL&#39;) &amp; n_count == 2), TRUE, FALSE)) %&gt;% filter(only_1_se)
xQTL_only_1_se_cb %&gt;% pull(gene_ID) %&gt;% unique %&gt;% length # should we filter the genes that only colocalized between 1 splicing event and AD? although that is not much
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gene_ref &lt;- fread(&#39;/data/resource/references/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.region_list&#39;)
xQTL_only_1_se_cb &lt;- xQTL_only_1_se_cb %&gt;% merge(., gene_ref %&gt;% select(&#39;gene_id&#39;, &#39;gene_name&#39;), by.x = &#39;gene_ID&#39;, by.y = &#39;gene_id&#39;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="n">Error</span> <span class="ow">in</span> <span class="n">h</span><span class="p">(</span><span class="n">simpleError</span><span class="p">(</span><span class="n">msg</span><span class="p">,</span> <span class="n">call</span><span class="p">)):</span> <span class="n">error</span> <span class="ow">in</span> <span class="n">evaluating</span> <span class="n">the</span> <span class="n">argument</span> <span class="s1">&#39;x&#39;</span> <span class="ow">in</span> <span class="n">selecting</span> <span class="n">a</span> <span class="n">method</span> <span class="k">for</span> <span class="n">function</span> <span class="s1">&#39;merge&#39;</span><span class="p">:</span> <span class="nb">object</span> <span class="s1">&#39;xQTL_only_1_se_cb&#39;</span> <span class="ow">not</span> <span class="n">found</span>
<span class="ne">Traceback</span>:

<span class="mi">1</span><span class="o">.</span> <span class="n">merge</span><span class="p">(</span><span class="o">.</span><span class="p">,</span> <span class="n">gene_ref</span> <span class="o">%&gt;%</span> <span class="n">select</span><span class="p">(</span><span class="s2">&quot;gene_id&quot;</span><span class="p">,</span> <span class="s2">&quot;gene_name&quot;</span><span class="p">),</span> <span class="n">by</span><span class="o">.</span><span class="n">x</span> <span class="o">=</span> <span class="s2">&quot;gene_ID&quot;</span><span class="p">,</span> 
 <span class="o">.</span>     <span class="n">by</span><span class="o">.</span><span class="n">y</span> <span class="o">=</span> <span class="s2">&quot;gene_id&quot;</span><span class="p">)</span>
<span class="mi">2</span><span class="o">.</span> <span class="o">.</span><span class="n">handleSimpleError</span><span class="p">(</span><span class="n">function</span> <span class="p">(</span><span class="n">cond</span><span class="p">)</span> 
 <span class="o">.</span> <span class="o">.</span><span class="n">Internal</span><span class="p">(</span><span class="n">C_tryCatchHelper</span><span class="p">(</span><span class="n">addr</span><span class="p">,</span> <span class="mi">1</span><span class="n">L</span><span class="p">,</span> <span class="n">cond</span><span class="p">)),</span> <span class="s2">&quot;object &#39;xQTL_only_1_se_cb&#39; not found&quot;</span><span class="p">,</span> 
 <span class="o">.</span>     <span class="n">base</span><span class="p">::</span><span class="n">quote</span><span class="p">(</span><span class="nb">eval</span><span class="p">(</span><span class="n">expr</span><span class="p">,</span> <span class="n">envir</span><span class="p">)))</span>
<span class="mi">3</span><span class="o">.</span> <span class="n">h</span><span class="p">(</span><span class="n">simpleError</span><span class="p">(</span><span class="n">msg</span><span class="p">,</span> <span class="n">call</span><span class="p">))</span>
<span class="mi">4</span><span class="o">.</span> <span class="o">.</span><span class="n">handleSimpleError</span><span class="p">(</span><span class="n">function</span> <span class="p">(</span><span class="n">cnd</span><span class="p">)</span> 
 <span class="o">.</span> <span class="p">{</span>
 <span class="o">.</span>     <span class="n">watcher</span><span class="err">$</span><span class="n">capture_plot_and_output</span><span class="p">()</span>
 <span class="o">.</span>     <span class="n">cnd</span> <span class="o">&lt;-</span> <span class="n">sanitize_call</span><span class="p">(</span><span class="n">cnd</span><span class="p">)</span>
 <span class="o">.</span>     <span class="n">watcher</span><span class="err">$</span><span class="n">push</span><span class="p">(</span><span class="n">cnd</span><span class="p">)</span>
 <span class="o">.</span>     <span class="n">switch</span><span class="p">(</span><span class="n">on_error</span><span class="p">,</span> <span class="k">continue</span> <span class="o">=</span> <span class="n">invokeRestart</span><span class="p">(</span><span class="s2">&quot;eval_continue&quot;</span><span class="p">),</span> 
 <span class="o">.</span>         <span class="n">stop</span> <span class="o">=</span> <span class="n">invokeRestart</span><span class="p">(</span><span class="s2">&quot;eval_stop&quot;</span><span class="p">),</span> <span class="n">error</span> <span class="o">=</span> <span class="n">NULL</span><span class="p">)</span>
 <span class="o">.</span> <span class="p">},</span> <span class="s2">&quot;error in evaluating the argument &#39;x&#39; in selecting a method for function &#39;merge&#39;: object &#39;xQTL_only_1_se_cb&#39; not found&quot;</span><span class="p">,</span> 
 <span class="o">.</span>     <span class="n">base</span><span class="p">::</span><span class="n">quote</span><span class="p">(</span><span class="n">h</span><span class="p">(</span><span class="n">simpleError</span><span class="p">(</span><span class="n">msg</span><span class="p">,</span> <span class="n">call</span><span class="p">))))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>cb_apr25 %&gt;% filter(gene_ID %in% xQTL_only$gene_id)  %&gt;% mutate(n_count = str_count(event_ID, &#39;;&#39;) + 1)  %&gt;% filter(n_count == 2) %&gt;% pull(gene_ID) %&gt;% unique %&gt;% length
</pre></div>
</div>
</div>
</div>
</section>
<section id="identify-overlapping-genes-between-metabolite-colocalized-genes-and-ad-genes-to-uncover-potential-links-between-ad-and-metabolites">
<h3>Identify overlapping genes between metabolite-colocalized genes and AD genes to uncover potential links between AD and metabolites<a class="headerlink" href="#identify-overlapping-genes-between-metabolite-colocalized-genes-and-ad-genes-to-uncover-potential-links-between-ad-and-metabolites" title="Link to this heading">#</a></h3>
<p>and pathway enrichment results there</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>metabolites_list &lt;- readLines(&#39;../May/FunGen_xQTL_metabolites_coloc_genes.txt&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>intersect(metabolites_list, gene_list$gene_id) # only 2 genes can overlap to the gene list from AD loci
</pre></div>
</div>
</div>
</div>
<p>Only two genes are present, which is insufficient for meaningful enrichment analysis.</p>
</section>
<section id="split-twas-gene-lists">
<h3>split twas gene lists<a class="headerlink" href="#split-twas-gene-lists" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>twas &lt;- fread(&#39;/data/analysis_result/twas/export/summary/FunGen_twas.exported.bed.gz&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>twas_xQTL &lt;- twas %&gt;%
  filter(molecular_id %in% xQTL_list$gene_id) %&gt;%
  group_by(molecular_id) %&gt;%
  slice_max(twas_z, n = 1, with_ties = FALSE) %&gt;%
  select(molecular_id, context) %&gt;% mutate(
      context = ifelse(str_detect(context, &#39;sQTL|pQTL&#39;), str_extract(context, &quot;^[^_]+_[^_]+_[^_]+&quot;), context)
  )
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Split the data frame by &#39;context&#39; column into a list of data frames
twas_xQTL_list &lt;- split(twas_xQTL, twas_xQTL$context)
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="go-enrichment">
<h2>GO enrichment<a class="headerlink" href="#go-enrichment" title="Link to this heading">#</a></h2>
<section id="real-gvc-list-20-genes-from-gvc-table2">
<h3>real gvc list - 20 genes from GVC table2<a class="headerlink" href="#real-gvc-list-20-genes-from-gvc-table2" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(clusterProfiler)
library(org.Hs.eg.db)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>clusterProfiler v4.10.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use clusterProfiler in published research, please cite:
T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141


Attaching package: &#39;clusterProfiler&#39;


The following object is masked from &#39;package:stats&#39;:

    filter


Loading required package: AnnotationDbi

Loading required package: stats4

Loading required package: BiocGenerics


Attaching package: &#39;BiocGenerics&#39;


The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs


The following objects are masked from &#39;package:base&#39;:

    Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,
    as.data.frame, basename, cbind, colnames, dirname, do.call,
    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,
    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,
    tapply, union, unique, unsplit, which.max, which.min


Loading required package: Biobase

Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.


Loading required package: IRanges

Loading required package: S4Vectors


Attaching package: &#39;S4Vectors&#39;


The following object is masked from &#39;package:clusterProfiler&#39;:

    rename


The following object is masked from &#39;package:utils&#39;:

    findMatches


The following objects are masked from &#39;package:base&#39;:

    I, expand.grid, unname



Attaching package: &#39;IRanges&#39;


The following object is masked from &#39;package:clusterProfiler&#39;:

    slice



Attaching package: &#39;AnnotationDbi&#39;


The following object is masked from &#39;package:clusterProfiler&#39;:

    select
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Run GO enrichment analysis for Biological Process (BP)
ego &lt;- enrichGO(
  gene          = real_gvc_list$gene_id,
  OrgDb         = org.Hs.eg.db,
  keyType       = &quot;ENSEMBL&quot;,
  ont           = &quot;BP&quot;,
  pAdjustMethod = &quot;BH&quot;,
  pvalueCutoff  = 0.05,
  qvalueCutoff  = 0.2
)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># View results
head(ego)

# Plot results
barplot(ego, showCategory = 10, title = &quot;GO Enrichment - BP&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process                       </td><td>8/20</td><td>83/21261 </td><td>4.626889e-15</td><td>7.139289e-12</td><td>3.248563e-12</td><td>ENSG00000064687/ENSG00000073921/ENSG00000080815/ENSG00000130203/ENSG00000136717/ENSG00000137642/ENSG00000137845/ENSG00000143801</td><td>8</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process                       </td><td>8/20</td><td>112/21261</td><td>5.496876e-14</td><td>4.240840e-11</td><td>1.929693e-11</td><td>ENSG00000064687/ENSG00000073921/ENSG00000080815/ENSG00000130203/ENSG00000136717/ENSG00000137642/ENSG00000137845/ENSG00000143801</td><td>8</td></tr>
	<tr><th scope=row>GO:0034205</th><td>GO:0034205</td><td>amyloid-beta formation                                            </td><td>7/20</td><td>65/21261 </td><td>1.344059e-13</td><td>6.912941e-11</td><td>3.145568e-11</td><td>ENSG00000064687/ENSG00000073921/ENSG00000080815/ENSG00000130203/ENSG00000136717/ENSG00000137642/ENSG00000143801                </td><td>7</td></tr>
	<tr><th scope=row>GO:0050435</th><td>GO:0050435</td><td>amyloid-beta metabolic process                                    </td><td>7/20</td><td>73/21261 </td><td>3.132134e-13</td><td>1.208221e-10</td><td>5.497719e-11</td><td>ENSG00000064687/ENSG00000073921/ENSG00000080815/ENSG00000130203/ENSG00000136717/ENSG00000137642/ENSG00000143801                </td><td>7</td></tr>
	<tr><th scope=row>GO:1902992</th><td>GO:1902992</td><td>negative regulation of amyloid precursor protein catabolic process</td><td>5/20</td><td>34/21261 </td><td>1.172045e-10</td><td>3.616931e-08</td><td>1.645798e-08</td><td>ENSG00000064687/ENSG00000073921/ENSG00000130203/ENSG00000136717/ENSG00000137642                                                </td><td>5</td></tr>
	<tr><th scope=row>GO:1902003</th><td>GO:1902003</td><td>regulation of amyloid-beta formation                              </td><td>5/20</td><td>57/21261 </td><td>1.739932e-09</td><td>4.474526e-07</td><td>2.036026e-07</td><td>ENSG00000064687/ENSG00000073921/ENSG00000130203/ENSG00000136717/ENSG00000137642                                                </td><td>5</td></tr>
</tbody>
</table>
</div><a class="reference internal image-reference" href="../../../_images/86da62c810ebef476ac801bdf79a8d966b2286f7eca24fbf09cb91640a057f15.png"><img alt="../../../_images/86da62c810ebef476ac801bdf79a8d966b2286f7eca24fbf09cb91640a057f15.png" src="../../../_images/86da62c810ebef476ac801bdf79a8d966b2286f7eca24fbf09cb91640a057f15.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
</section>
<section id="gvc-list1-76-genes">
<h3>gvc list1 -76 genes<a class="headerlink" href="#gvc-list1-76-genes" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Run GO enrichment analysis for Biological Process (BP)
gvc_table1_ego &lt;- enrichGO(
  gene          = gvc_table1$gene_id,
  OrgDb         = org.Hs.eg.db,
  keyType       = &quot;ENSEMBL&quot;,
  ont           = &quot;BP&quot;,
  pAdjustMethod = &quot;BH&quot;,
  pvalueCutoff  = 0.05,
  qvalueCutoff  = 0.2
)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># View results
head(gvc_table1_ego)

# Plot results
barplot(gvc_table1_ego, showCategory = 10, title = &quot;GO Enrichment - BP&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0050435</th><td>GO:0050435</td><td>amyloid-beta metabolic process             </td><td>7/73</td><td>73/21261 </td><td>5.699259e-09</td><td>1.238449e-05</td><td>8.296922e-06</td><td>ENSG00000136717/ENSG00000196549/ENSG00000120885/ENSG00000137642/ENSG00000138613/ENSG00000159640/ENSG00000064687                                </td><td>7</td></tr>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process</td><td>6/73</td><td>83/21261 </td><td>4.070155e-07</td><td>4.422223e-04</td><td>2.962644e-04</td><td>ENSG00000151694/ENSG00000136717/ENSG00000120885/ENSG00000137642/ENSG00000138613/ENSG00000064687                                                </td><td>6</td></tr>
	<tr><th scope=row>GO:0002274</th><td>GO:0002274</td><td>myeloid leukocyte activation               </td><td>9/73</td><td>292/21261</td><td>6.914695e-07</td><td>5.008544e-04</td><td>3.355447e-04</td><td>ENSG00000109684/ENSG00000168421/ENSG00000095970/ENSG00000120885/ENSG00000066336/ENSG00000197943/ENSG00000103241/ENSG00000030582/ENSG00000142192</td><td>9</td></tr>
	<tr><th scope=row>GO:0018108</th><td>GO:0018108</td><td>peptidyl-tyrosine phosphorylation          </td><td>9/73</td><td>316/21261</td><td>1.331243e-06</td><td>5.279251e-04</td><td>3.536805e-04</td><td>ENSG00000151694/ENSG00000095970/ENSG00000146904/ENSG00000120899/ENSG00000157368/ENSG00000197943/ENSG00000159640/ENSG00000087589/ENSG00000142192</td><td>9</td></tr>
	<tr><th scope=row>GO:0018212</th><td>GO:0018212</td><td>peptidyl-tyrosine modification             </td><td>9/73</td><td>318/21261</td><td>1.402387e-06</td><td>5.279251e-04</td><td>3.536805e-04</td><td>ENSG00000151694/ENSG00000095970/ENSG00000146904/ENSG00000120899/ENSG00000157368/ENSG00000197943/ENSG00000159640/ENSG00000087589/ENSG00000142192</td><td>9</td></tr>
	<tr><th scope=row>GO:0030183</th><td>GO:0030183</td><td>B cell differentiation                     </td><td>7/73</td><td>163/21261</td><td>1.457685e-06</td><td>5.279251e-04</td><td>3.536805e-04</td><td>ENSG00000203710/ENSG00000151694/ENSG00000168918/ENSG00000120899/ENSG00000066336/ENSG00000095585/ENSG00000197943                                </td><td>7</td></tr>
</tbody>
</table>
</div><a class="reference internal image-reference" href="../../../_images/eab194bb00c00ad56b086f9767f62db0ad7ee850a3f56477014a01d5bc5ae891.png"><img alt="../../../_images/eab194bb00c00ad56b086f9767f62db0ad7ee850a3f56477014a01d5bc5ae891.png" src="../../../_images/eab194bb00c00ad56b086f9767f62db0ad7ee850a3f56477014a01d5bc5ae891.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
</section>
<section id="strong-list-overlapped-between-xqtl-gvc-and-twas">
<h3>strong list - overlapped between xQTL, GVC and TWAS<a class="headerlink" href="#strong-list-overlapped-between-xqtl-gvc-and-twas" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>
ego &lt;- enrichGO(
  gene          = strong_list,
  OrgDb         = org.Hs.eg.db,
  keyType       = &quot;ENSEMBL&quot;,
  ont           = &quot;BP&quot;,
  pAdjustMethod = &quot;BH&quot;,
  pvalueCutoff  = 0.05,
  qvalueCutoff  = 0.2
)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:1 mapping between keys and columns
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>head(ego)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process                       </td><td>5/21</td><td>68/18870</td><td>1.018064e-08</td><td>1.487391e-05</td><td>7.330060e-06</td><td>8301/348/274/6653/102</td><td>5</td></tr>
	<tr><th scope=row>GO:1902992</th><td>GO:1902992</td><td>negative regulation of amyloid precursor protein catabolic process</td><td>4/21</td><td>32/18870</td><td>3.993694e-08</td><td>2.850115e-05</td><td>1.404574e-05</td><td>8301/348/274/6653    </td><td>4</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process                       </td><td>5/21</td><td>96/18870</td><td>5.852392e-08</td><td>2.850115e-05</td><td>1.404574e-05</td><td>8301/348/274/6653/102</td><td>5</td></tr>
	<tr><th scope=row>GO:1905245</th><td>GO:1905245</td><td>regulation of aspartic-type peptidase activity                    </td><td>3/21</td><td>10/18870</td><td>1.418278e-07</td><td>5.180262e-05</td><td>2.552901e-05</td><td>8301/274/6653        </td><td>3</td></tr>
	<tr><th scope=row>GO:1902003</th><td>GO:1902003</td><td>regulation of amyloid-beta formation                              </td><td>4/21</td><td>48/18870</td><td>2.136190e-07</td><td>6.241947e-05</td><td>3.076113e-05</td><td>8301/348/274/6653    </td><td>4</td></tr>
	<tr><th scope=row>GO:1902991</th><td>GO:1902991</td><td>regulation of amyloid precursor protein catabolic process         </td><td>4/21</td><td>55/18870</td><td>3.725395e-07</td><td>8.367501e-05</td><td>4.123615e-05</td><td>8301/348/274/6653    </td><td>4</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>barplot(ego, showCategory = 10, title = &quot;GO Enrichment - BP&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/a18de2e8c648b60bfa6b2a5713b7078cfd1ab46892fbcba41ff22989d17e2d7c.png"><img alt="../../../_images/a18de2e8c648b60bfa6b2a5713b7078cfd1ab46892fbcba41ff22989d17e2d7c.png" src="../../../_images/a18de2e8c648b60bfa6b2a5713b7078cfd1ab46892fbcba41ff22989d17e2d7c.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="xqtl-list">
<h3>xQTL list<a class="headerlink" href="#xqtl-list" title="Link to this heading">#</a></h3>
<section id="go">
<h4>GO<a class="headerlink" href="#go" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_ego &lt;- enrichGO(
  gene          = xQTL_list$gene_name,
  OrgDb         = org.Hs.eg.db,
  keyType       = &quot;SYMBOL&quot;,
  ont           = &quot;BP&quot;,
  pAdjustMethod = &quot;BH&quot;,
  pvalueCutoff  = 1,
  qvalueCutoff  = 1
)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>head(xQTL_ego, n=5)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 5 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0002274</th><td>GO:0002274</td><td>myeloid leukocyte activation                     </td><td>14/215</td><td>241/18870</td><td>7.287466e-07</td><td>0.001336395</td><td>0.001100642</td><td>GRN/TREM2/STX4/CD33/HAVCR1/CLU/NECTIN2/HAVCR2/DYSF/APP/FCER1G/RHOH/C5AR1/PLCG2                                     </td><td>14</td></tr>
	<tr><th scope=row>GO:0030100</th><td>GO:0030100</td><td>regulation of endocytosis                        </td><td>15/215</td><td>291/18870</td><td>1.315310e-06</td><td>0.001336395</td><td>0.001100642</td><td>ATG5/PICALM/TREM2/RIN3/SIRPB1/LILRB1/CLU/APOC1/DYSF/BIN1/DGKQ/FCER1G/PLCG2/SIRPA/APOC2                             </td><td>15</td></tr>
	<tr><th scope=row>GO:0002281</th><td>GO:0002281</td><td>macrophage activation involved in immune response</td><td>5/215 </td><td>19/18870 </td><td>1.871302e-06</td><td>0.001336395</td><td>0.001100642</td><td>GRN/TREM2/HAVCR2/DYSF/PLCG2                                                                                        </td><td> 5</td></tr>
	<tr><th scope=row>GO:0045806</th><td>GO:0045806</td><td>negative regulation of endocytosis               </td><td>8/215 </td><td>75/18870 </td><td>2.184779e-06</td><td>0.001336395</td><td>0.001100642</td><td>ATG5/PICALM/RIN3/LILRB1/APOC1/DYSF/SIRPA/APOC2                                                                     </td><td> 8</td></tr>
	<tr><th scope=row>GO:0045785</th><td>GO:0045785</td><td>positive regulation of cell adhesion             </td><td>19/215</td><td>482/18870</td><td>2.856012e-06</td><td>0.001336395</td><td>0.001100642</td><td>CBFB/NCK2/ACTL6B/CASS4/STK4/SIRPB1/CORO1A/STX4/LILRB1/ITGB1BP1/PTK2B/HAVCR2/EPHA1/STX3/SERPINF2/RHOH/CD55/SIRPA/CR1</td><td>19</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_ego %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;amyloid&#39;)) %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0050435</th><td>GO:0050435</td><td>amyloid-beta metabolic process                                    </td><td>6/215</td><td>65/18870</td><td>9.625198e-05</td><td>0.008784523</td><td>0.007234844</td><td>PICALM/CLU/BIN1/SORL1/APH1B/ACE   </td><td>6</td></tr>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process                       </td><td>6/215</td><td>68/18870</td><td>1.239814e-04</td><td>0.009283489</td><td>0.007645788</td><td>PICALM/CLU/BIN1/SORL1/APH1B/ADAM17</td><td>6</td></tr>
	<tr><th scope=row>GO:0034205</th><td>GO:0034205</td><td>amyloid-beta formation                                            </td><td>5/215</td><td>56/18870</td><td>4.365566e-04</td><td>0.014128981</td><td>0.011636486</td><td>PICALM/CLU/BIN1/SORL1/APH1B       </td><td>5</td></tr>
	<tr><th scope=row>GO:1902992</th><td>GO:1902992</td><td>negative regulation of amyloid precursor protein catabolic process</td><td>4/215</td><td>32/18870</td><td>4.588274e-04</td><td>0.014228284</td><td>0.011718271</td><td>PICALM/CLU/BIN1/SORL1             </td><td>4</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process                       </td><td>6/215</td><td>96/18870</td><td>8.066540e-04</td><td>0.019811423</td><td>0.016316488</td><td>PICALM/CLU/BIN1/SORL1/APH1B/ADAM17</td><td>6</td></tr>
	<tr><th scope=row>GO:1905906</th><td>GO:1905906</td><td>regulation of amyloid fibril formation                            </td><td>3/215</td><td>17/18870</td><td>8.814851e-04</td><td>0.020977978</td><td>0.017277251</td><td>TREM2/CLU/APP                     </td><td>3</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_ego %&gt;%
  as.data.frame() %&gt;%
  filter(str_detect(Description, regex(&#39;lysosome|lysosomal membrane|lysosomal lumen|autophagy|macroautophagy|vacuole|endosome|phagosome|lipid droplet&#39;, ignore_case = TRUE))) %&gt;%
  head() # GO terms related to lysosome
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0007040</th><td>GO:0007040</td><td>lysosome organization              </td><td>6/215</td><td>109/18870</td><td>0.001559714</td><td>0.02816660</td><td>0.02319773</td><td>GRN/CORO1A/TMEM106B/TMEM175/ACP2/ABCA1              </td><td>6</td></tr>
	<tr><th scope=row>GO:0080171</th><td>GO:0080171</td><td>lytic vacuole organization         </td><td>6/215</td><td>109/18870</td><td>0.001559714</td><td>0.02816660</td><td>0.02319773</td><td>GRN/CORO1A/TMEM106B/TMEM175/ACP2/ABCA1              </td><td>6</td></tr>
	<tr><th scope=row>GO:0035751</th><td>GO:0035751</td><td>regulation of lysosomal lumen pH   </td><td>3/215</td><td>30/18870 </td><td>0.004720251</td><td>0.05682812</td><td>0.04680307</td><td>GRN/TMEM106B/TMEM175                                </td><td>3</td></tr>
	<tr><th scope=row>GO:0007033</th><td>GO:0007033</td><td>vacuole organization               </td><td>8/215</td><td>232/18870</td><td>0.005230031</td><td>0.06151798</td><td>0.05066559</td><td>GRN/ATG5/CORO1A/TMEM106B/TMEM175/ACP2/TP53INP1/ABCA1</td><td>8</td></tr>
	<tr><th scope=row>GO:1905671</th><td>GO:1905671</td><td>regulation of lysosome organization</td><td>2/215</td><td>12/18870 </td><td>0.007910467</td><td>0.07489316</td><td>0.06168125</td><td>GRN/TMEM106B                                        </td><td>2</td></tr>
	<tr><th scope=row>GO:0061684</th><td>GO:0061684</td><td>chaperone-mediated autophagy       </td><td>2/215</td><td>15/18870 </td><td>0.012305649</td><td>0.09661980</td><td>0.07957509</td><td>ATG5/CLU                                            </td><td>2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_ego %&gt;%
  as.data.frame() %&gt;%
  filter(str_detect(Description, regex(&#39;synaptic transmission|synapse organization|cognition|synaptic plasticity|neuron projection morphogenesis|axon guidance|nervous system development|modulation of chemical synaptic transmission|generation of neurons|learning or memory|neuron projection|postsynaptic membrane|dendrite|axon|cell body|postsynapse|presynapse|synaptic membrane|synapse&#39;, ignore_case = TRUE))) %&gt;% head  
# GO terms related to neuron function
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0098814</th><td>GO:0098814</td><td>spontaneous synaptic transmission                           </td><td>3/215</td><td>14/18870 </td><td>0.0004839086</td><td>0.01428461</td><td>0.01176466</td><td>APP/DOC2A/DOC2B                                   </td><td>3</td></tr>
	<tr><th scope=row>GO:0099003</th><td>GO:0099003</td><td>vesicle-mediated transport in synapse                       </td><td>9/215</td><td>220/18870</td><td>0.0009731495</td><td>0.02231430</td><td>0.01837783</td><td>SLC4A8/PICALM/BIN1/DGKQ/DOC2A/STX3/CPLX1/GAK/DOC2B</td><td>9</td></tr>
	<tr><th scope=row>GO:0099643</th><td>GO:0099643</td><td>signal release from synapse                                 </td><td>6/215</td><td>144/18870</td><td>0.0061904976</td><td>0.06811766</td><td>0.05610102</td><td>SLC4A8/DYSF/DOC2A/STX3/CPLX1/DOC2B                </td><td>6</td></tr>
	<tr><th scope=row>GO:0002753</th><td>GO:0002753</td><td>cytosolic pattern recognition receptor signaling pathway    </td><td>6/215</td><td>146/18870</td><td>0.0066103893</td><td>0.06955848</td><td>0.05728766</td><td>DHX33/TREM2/HAVCR2/OTULIN/SCIMP/PLCG2             </td><td>6</td></tr>
	<tr><th scope=row>GO:0001771</th><td>GO:0001771</td><td>immunological synapse formation                             </td><td>2/215</td><td>15/18870 </td><td>0.0123056490</td><td>0.09661980</td><td>0.07957509</td><td>NCK2/HAVCR2                                       </td><td>2</td></tr>
	<tr><th scope=row>GO:0098969</th><td>GO:0098969</td><td>neurotransmitter receptor transport to postsynaptic membrane</td><td>2/215</td><td>16/18870 </td><td>0.0139590195</td><td>0.10276784</td><td>0.08463856</td><td>STX3/CPLX1                                        </td><td>2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_ego %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;lipid|Lipid|fatty|Fatty&#39;)) %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0033700</th><td>GO:0033700</td><td>phospholipid efflux                      </td><td>3/215</td><td>14/18870 </td><td>0.0004839086</td><td>0.01428461</td><td>0.01176466</td><td>APOC1/ABCA1/APOC2                         </td><td>3</td></tr>
	<tr><th scope=row>GO:0071825</th><td>GO:0071825</td><td>protein-lipid complex organization       </td><td>4/215</td><td>53/18870 </td><td>0.0031013959</td><td>0.04442263</td><td>0.03658603</td><td>APOC1/BIN1/ABCA1/APOC2                    </td><td>4</td></tr>
	<tr><th scope=row>GO:0055088</th><td>GO:0055088</td><td>lipid homeostasis                        </td><td>7/215</td><td>175/18870</td><td>0.0039847419</td><td>0.05183541</td><td>0.04269112</td><td>POLD1/TREM2/MTCH2/THADA/TLCD3B/ABCA1/APOC2</td><td>7</td></tr>
	<tr><th scope=row>GO:0065005</th><td>GO:0065005</td><td>protein-lipid complex assembly           </td><td>3/215</td><td>30/18870 </td><td>0.0047202510</td><td>0.05682812</td><td>0.04680307</td><td>APOC1/BIN1/ABCA1                          </td><td>3</td></tr>
	<tr><th scope=row>GO:0046834</th><td>GO:0046834</td><td>lipid phosphorylation                    </td><td>2/215</td><td>11/18870 </td><td>0.0066416292</td><td>0.06955848</td><td>0.05728766</td><td>AGK/DGKQ                                  </td><td>2</td></tr>
	<tr><th scope=row>GO:1905953</th><td>GO:1905953</td><td>negative regulation of lipid localization</td><td>4/215</td><td>66/18870 </td><td>0.0068071776</td><td>0.06966012</td><td>0.05737137</td><td>TREM2/APOC1/ABCA1/APOC2                   </td><td>4</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>barplot(xQTL_ego,
        showCategory = 10,
        title = &quot;GO Enrichment - BP&quot;) +
  theme(plot.margin = margin(t = 5, r = 5, b = 5, l = 10, unit = &quot;mm&quot;))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/b6a79e8ef0c2dff702ce63ba4e72019cb50a3086105e1971287f48335bf875e7.png"><img alt="../../../_images/b6a79e8ef0c2dff702ce63ba4e72019cb50a3086105e1971287f48335bf875e7.png" src="../../../_images/b6a79e8ef0c2dff702ce63ba4e72019cb50a3086105e1971287f48335bf875e7.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>ggsave(&#39;./figs/xQTL_list_GO.pdf&#39;, width = 6, height = 6)
</pre></div>
</div>
</div>
</div>
</section>
<section id="kegg">
<h4>KEGG<a class="headerlink" href="#kegg" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>entrez_ids &lt;- bitr(xQTL_list$gene_id, fromType = &quot;ENSEMBL&quot;, 
                   toType = &quot;ENTREZID&quot;, OrgDb = org.Hs.eg.db)

kegg_enrich &lt;- enrichKEGG(gene = entrez_ids$ENTREZID,
                          organism = &#39;hsa&#39;,
                          pvalueCutoff = 1,
                          qvalueCutoff = 1) 
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:many mapping between keys and columns

Warning message in bitr(xQTL_list$gene_id, fromType = &quot;ENSEMBL&quot;, toType = &quot;ENTREZID&quot;, :
&quot;2.08% of input gene IDs are fail to map...&quot;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>kegg_enrich %&gt;% head(n=4)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 4 x 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa04610</th><td>Organismal Systems</td><td>Immune system               </td><td>hsa04610</td><td>Complement and coagulation cascades</td><td>7/106</td><td>88/8866 </td><td>8.205886e-05</td><td>0.01127777</td><td>0.01069490</td><td>1380/1191/5345/716/1604/728/1378               </td><td>7</td></tr>
	<tr><th scope=row>hsa04662</th><td>Organismal Systems</td><td>Immune system               </td><td>hsa04662</td><td>B cell receptor signaling pathway  </td><td>7/106</td><td>91/8866 </td><td>1.016015e-04</td><td>0.01127777</td><td>0.01069490</td><td>29760/5595/10859/1380/3635/353514/5336         </td><td>7</td></tr>
	<tr><th scope=row>hsa04380</th><td>Organismal Systems</td><td>Development and regeneration</td><td>hsa04380</td><td>Osteoclast differentiation         </td><td>8/106</td><td>143/8866</td><td>2.999088e-04</td><td>0.02219325</td><td>0.02104623</td><td>29760/54209/10326/5595/10859/353514/5336/140885</td><td>8</td></tr>
	<tr><th scope=row>hsa04148</th><td>NA                </td><td>NA                          </td><td>hsa04148</td><td>Efferocytosis                      </td><td>8/106</td><td>157/8866</td><td>5.618405e-04</td><td>0.03118215</td><td>0.02957055</td><td>10326/5595/26762/84868/91937/6868/19/140885    </td><td>8</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>kegg_enrich %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;lipid|Lipid|fatty|Fatty|Alzheimer|Cholesterol&#39;))
kegg_enrich %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;Alzheimer&#39;))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 9 x 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa04979</th><td>Organismal Systems                  </td><td>Digestive system                  </td><td>hsa04979</td><td>Cholesterol metabolism                                    </td><td>3/106</td><td>51/8866 </td><td>0.02284626</td><td>0.3088180</td><td>0.2928573</td><td>341/19/344                         </td><td>3</td></tr>
	<tr><th scope=row>hsa00604</th><td>Metabolism                          </td><td>Glycan biosynthesis and metabolism</td><td>hsa00604</td><td>Glycosphingolipid biosynthesis - ganglio series           </td><td>1/106</td><td>15/8866 </td><td>0.16519819</td><td>0.7616641</td><td>0.7222988</td><td>6483                               </td><td>1</td></tr>
	<tr><th scope=row>hsa00603</th><td>Metabolism                          </td><td>Glycan biosynthesis and metabolism</td><td>hsa00603</td><td>Glycosphingolipid biosynthesis - globo and isoglobo series</td><td>1/106</td><td>16/8866 </td><td>0.17519581</td><td>0.7616641</td><td>0.7222988</td><td>6483                               </td><td>1</td></tr>
	<tr><th scope=row>hsa00561</th><td>Metabolism                          </td><td>Lipid metabolism                  </td><td>hsa00561</td><td>Glycerolipid metabolism                                   </td><td>2/106</td><td>65/8866 </td><td>0.18221177</td><td>0.7616641</td><td>0.7222988</td><td>55750/1609                         </td><td>2</td></tr>
	<tr><th scope=row>hsa05010</th><td>Human Diseases                      </td><td>Neurodegenerative disease         </td><td>hsa05010</td><td>Alzheimer disease                                         </td><td>7/106</td><td>391/8866</td><td>0.18673497</td><td>0.7616641</td><td>0.7222988</td><td>5595/1350/83464/351/6868/91252/4722</td><td>7</td></tr>
	<tr><th scope=row>hsa04071</th><td>Environmental Information Processing</td><td>Signal transduction               </td><td>hsa04071</td><td>Sphingolipid signaling pathway                            </td><td>2/106</td><td>122/8866</td><td>0.43021701</td><td>0.7977285</td><td>0.7564993</td><td>5595/2207                          </td><td>2</td></tr>
	<tr><th scope=row>hsa04932</th><td>Human Diseases                      </td><td>Endocrine and metabolic disease   </td><td>hsa04932</td><td>Non-alcoholic fatty liver disease                         </td><td>2/106</td><td>157/8866</td><td>0.56344595</td><td>0.8140899</td><td>0.7720151</td><td>1350/4722                          </td><td>2</td></tr>
	<tr><th scope=row>hsa00564</th><td>Metabolism                          </td><td>Lipid metabolism                  </td><td>hsa00564</td><td>Glycerophospholipid metabolism                            </td><td>1/106</td><td>103/8866</td><td>0.71237338</td><td>0.8364174</td><td>0.7931886</td><td>1609                               </td><td>1</td></tr>
	<tr><th scope=row>hsa05417</th><td>Human Diseases                      </td><td>Cardiovascular disease            </td><td>hsa05417</td><td>Lipid and atherosclerosis                                 </td><td>2/106</td><td>216/8866</td><td>0.73485076</td><td>0.8364174</td><td>0.7931886</td><td>5595/19                            </td><td>2</td></tr>
</tbody>
</table>
</div><div class="output text_html"><table class="dataframe">
<caption>A data.frame: 1 x 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa05010</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05010</td><td>Alzheimer disease</td><td>7/106</td><td>391/8866</td><td>0.186735</td><td>0.7616641</td><td>0.7222988</td><td>5595/1350/83464/351/6868/91252/4722</td><td>7</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>entrez_ids %&gt;% filter(ENTREZID == &#39;1609&#39;) %&gt;% merge(.,gene_ref, by.x = &#39;ENSEMBL&#39;, by.y = &#39;gene_id&#39;)
entrez_ids %&gt;% filter(ENTREZID == &#39;6483&#39;) %&gt;% merge(.,gene_ref, by.x = &#39;ENSEMBL&#39;, by.y = &#39;gene_id&#39;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 1 x 6</caption>
<thead>
	<tr><th scope=col>ENSEMBL</th><th scope=col>ENTREZID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>ENSG00000145214</td><td>1609</td><td>chr4</td><td>986894</td><td>986895</td><td>DGKQ</td></tr>
</tbody>
</table>
</div><div class="output text_html"><table class="dataframe">
<caption>A data.frame: 1 x 6</caption>
<thead>
	<tr><th scope=col>ENSEMBL</th><th scope=col>ENTREZID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>ENSG00000157350</td><td>6483</td><td>chr16</td><td>70439236</td><td>70439237</td><td>ST3GAL2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>entrez_ids %&gt;% filter(ENTREZID%in% (kegg_enrich %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;Alzheimer&#39;)) %&gt;% pull(geneID) %&gt;% str_split(., &#39;/&#39;) %&gt;% unlist)) %&gt;% merge(.,gene_ref, by.x = &#39;ENSEMBL&#39;, by.y = &#39;gene_id&#39;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 7 x 6</caption>
<thead>
	<tr><th scope=col>ENSEMBL</th><th scope=col>ENTREZID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>ENSG00000102882</td><td>5595 </td><td>chr16</td><td>30123505</td><td>30123506</td><td>MAPK3   </td></tr>
	<tr><td>ENSG00000127184</td><td>1350 </td><td>chr5 </td><td>86617927</td><td>86617928</td><td>COX7C   </td></tr>
	<tr><td>ENSG00000138613</td><td>83464</td><td>chr15</td><td>63276017</td><td>63276018</td><td>APH1B   </td></tr>
	<tr><td>ENSG00000142192</td><td>351  </td><td>chr21</td><td>26171127</td><td>26171128</td><td>APP     </td></tr>
	<tr><td>ENSG00000151694</td><td>6868 </td><td>chr2 </td><td> 9556731</td><td> 9556732</td><td>ADAM17  </td></tr>
	<tr><td>ENSG00000165915</td><td>91252</td><td>chr11</td><td>47407131</td><td>47407132</td><td>SLC39A13</td></tr>
	<tr><td>ENSG00000213619</td><td>4722 </td><td>chr11</td><td>47565335</td><td>47565336</td><td>NDUFS3  </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>barplot(kegg_enrich, showCategory = 10, title = &quot;KEGG Pathway Enrichment&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/2b373784a70591194a29f1ec89c400502b2374e70645082e6ea5a5dab9b5b67f.png"><img alt="../../../_images/2b373784a70591194a29f1ec89c400502b2374e70645082e6ea5a5dab9b5b67f.png" src="../../../_images/2b373784a70591194a29f1ec89c400502b2374e70645082e6ea5a5dab9b5b67f.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>ggsave(&#39;./figs/xQTL_list_KEGG.pdf&#39;, width = 6, height = 6)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>kegg_enrich %&gt;%
  as.data.frame() %&gt;%
  filter(str_detect(Description, regex(&#39;lysosome|autophagy|endocytosis|phagosome|vacuole|lipid&#39;, ignore_case = TRUE))) %&gt;%
  head()
# KEGG pathways related to lysosome functions
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa04142</th><td>Cellular Processes</td><td>Transport and catabolism          </td><td>hsa04142</td><td>Lysosome                                                  </td><td>5/106</td><td>133/8866</td><td>0.02139461</td><td>0.3088180</td><td>0.2928573</td><td>23062/1512/3425/53/1508</td><td>5</td></tr>
	<tr><th scope=row>hsa00604</th><td>Metabolism        </td><td>Glycan biosynthesis and metabolism</td><td>hsa00604</td><td>Glycosphingolipid biosynthesis - ganglio series           </td><td>1/106</td><td>15/8866 </td><td>0.16519819</td><td>0.7616641</td><td>0.7222988</td><td>6483                   </td><td>1</td></tr>
	<tr><th scope=row>hsa00603</th><td>Metabolism        </td><td>Glycan biosynthesis and metabolism</td><td>hsa00603</td><td>Glycosphingolipid biosynthesis - globo and isoglobo series</td><td>1/106</td><td>16/8866 </td><td>0.17519581</td><td>0.7616641</td><td>0.7222988</td><td>6483                   </td><td>1</td></tr>
	<tr><th scope=row>hsa00561</th><td>Metabolism        </td><td>Lipid metabolism                  </td><td>hsa00561</td><td>Glycerolipid metabolism                                   </td><td>2/106</td><td>65/8866 </td><td>0.18221177</td><td>0.7616641</td><td>0.7222988</td><td>55750/1609             </td><td>2</td></tr>
	<tr><th scope=row>hsa04136</th><td>Cellular Processes</td><td>Transport and catabolism          </td><td>hsa04136</td><td>Autophagy - other                                         </td><td>1/106</td><td>32/8866 </td><td>0.31993655</td><td>0.7977285</td><td>0.7564993</td><td>9474                   </td><td>1</td></tr>
	<tr><th scope=row>hsa04140</th><td>Cellular Processes</td><td>Transport and catabolism          </td><td>hsa04140</td><td>Autophagy - animal                                        </td><td>3/106</td><td>169/8866</td><td>0.32892441</td><td>0.7977285</td><td>0.7564993</td><td>9474/5595/1508         </td><td>3</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>kegg_enrich %&gt;%
  as.data.frame() %&gt;%
  filter(str_detect(Description, regex(&#39;synapse|neuro|axon|dendrite|memory|learning|brain|glutamatergic|GABAergic|cholinergic|dopaminergic|serotonergic&#39;, ignore_case = TRUE))) %&gt;%
  head()

# KEGG pathways related to neuron function
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa04360</th><td>Organismal Systems</td><td>Development and regeneration</td><td>hsa04360</td><td>Axon guidance                                    </td><td>7/106</td><td>184/8866</td><td>0.006415158</td><td>0.2034522</td><td>0.1929371</td><td>8440/5595/5921/2041/3985/23365/5336</td><td>7</td></tr>
	<tr><th scope=row>hsa04722</th><td>Organismal Systems</td><td>Nervous system              </td><td>hsa04722</td><td>Neurotrophin signaling pathway                   </td><td>3/106</td><td>120/8866</td><td>0.173133337</td><td>0.7616641</td><td>0.7222988</td><td>4214/5595/5336                     </td><td>3</td></tr>
	<tr><th scope=row>hsa04726</th><td>Organismal Systems</td><td>Nervous system              </td><td>hsa04726</td><td>Serotonergic synapse                             </td><td>2/106</td><td>115/8866</td><td>0.401087493</td><td>0.7977285</td><td>0.7564993</td><td>5595/351                           </td><td>2</td></tr>
	<tr><th scope=row>hsa04727</th><td>Organismal Systems</td><td>Nervous system              </td><td>hsa04727</td><td>GABAergic synapse                                </td><td>1/106</td><td>89/8866 </td><td>0.658994838</td><td>0.8359820</td><td>0.7927757</td><td>4905                               </td><td>1</td></tr>
	<tr><th scope=row>hsa05022</th><td>Human Diseases    </td><td>Neurodegenerative disease   </td><td>hsa05022</td><td>Pathways of neurodegeneration - multiple diseases</td><td>5/106</td><td>483/8866</td><td>0.692328902</td><td>0.8364174</td><td>0.7931886</td><td>5595/83544/1350/351/4722           </td><td>5</td></tr>
	<tr><th scope=row>hsa04724</th><td>Organismal Systems</td><td>Nervous system              </td><td>hsa04724</td><td>Glutamatergic synapse                            </td><td>1/106</td><td>116/8866</td><td>0.754488647</td><td>0.8374824</td><td>0.7941986</td><td>5595                               </td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
<section id="plot-ad-path">
<h5>plot AD path<a class="headerlink" href="#plot-ad-path" title="Link to this heading">#</a></h5>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gene.data &lt;- kegg_enrich %&gt;%
  as.data.frame() %&gt;%
  filter(str_detect(Description, &#39;Alzheimer&#39;)) %&gt;%
  pull(geneID) %&gt;%
  str_split(&#39;/&#39;) %&gt;%
  unlist()

gene.data &lt;- setNames(as.numeric(rep(1, length(gene.data))), gene.data)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(pathview)


# Generate the pathway plot with highlighted genes
pathview(
  gene.data = gene.data,
  pathway.id = &quot;hsa05010&quot;,
  species = &quot;hsa&quot;,
  out.suffix = &quot;xQTL_list&quot;,
  kegg.native = TRUE
)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:1 mapping between keys and columns

Info: Working in directory /home/user/project/FunGen_xQTL/2025/May

Info: Writing image file hsa05010.xQTL_list.png
</pre></div>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="xqtl-only-list">
<h3>xQTL only list<a class="headerlink" href="#xqtl-only-list" title="Link to this heading">#</a></h3>
<section id="id1">
<h4>GO<a class="headerlink" href="#id1" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xqtl_only_ego &lt;- enrichGO(
  gene          = xQTL_only$gene_id,
  OrgDb         = org.Hs.eg.db,
  keyType       = &quot;ENSEMBL&quot;,
  ont           = &quot;BP&quot;,
  pAdjustMethod = &quot;BH&quot;,
  pvalueCutoff  = 1,
  qvalueCutoff  = 1
)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xqtl_only_ego %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0016482</th><td>GO:0016482</td><td>cytosolic transport          </td><td>7/134</td><td>205/21261</td><td>0.0003266830</td><td>0.3308869</td><td>0.3181081</td><td>ENSG00000008294/ENSG00000099992/ENSG00000102879/ENSG00000141258/ENSG00000167716/ENSG00000178950/ENSG00000188529</td><td>7</td></tr>
	<tr><th scope=row>GO:0031338</th><td>GO:0031338</td><td>regulation of vesicle fusion </td><td>3/134</td><td>26/21261 </td><td>0.0005723755</td><td>0.3308869</td><td>0.3181081</td><td>ENSG00000102879/ENSG00000168993/ENSG00000272636                                                                </td><td>3</td></tr>
	<tr><th scope=row>GO:0048284</th><td>GO:0048284</td><td>organelle fusion             </td><td>6/134</td><td>170/21261</td><td>0.0007385666</td><td>0.3308869</td><td>0.3181081</td><td>ENSG00000102879/ENSG00000103496/ENSG00000127419/ENSG00000140379/ENSG00000168993/ENSG00000272636                </td><td>6</td></tr>
	<tr><th scope=row>GO:1902017</th><td>GO:1902017</td><td>regulation of cilium assembly</td><td>4/134</td><td>75/21261 </td><td>0.0012972001</td><td>0.3308869</td><td>0.3181081</td><td>ENSG00000057663/ENSG00000103540/ENSG00000138036/ENSG00000182541                                                </td><td>4</td></tr>
	<tr><th scope=row>GO:0006906</th><td>GO:0006906</td><td>vesicle fusion               </td><td>5/134</td><td>129/21261</td><td>0.0013601269</td><td>0.3308869</td><td>0.3181081</td><td>ENSG00000102879/ENSG00000103496/ENSG00000127419/ENSG00000168993/ENSG00000272636                                </td><td>5</td></tr>
	<tr><th scope=row>GO:0090174</th><td>GO:0090174</td><td>organelle membrane fusion    </td><td>5/134</td><td>131/21261</td><td>0.0014560405</td><td>0.3308869</td><td>0.3181081</td><td>ENSG00000102879/ENSG00000103496/ENSG00000127419/ENSG00000168993/ENSG00000272636                                </td><td>5</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xqtl_only_ego %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;amyloid&#39;))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 3 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0097242</th><td>GO:0097242</td><td>amyloid-beta clearance                     </td><td>1/134</td><td>46/21261 </td><td>0.2525944</td><td>0.5085526</td><td>0.4889124</td><td>ENSG00000197405</td><td>1</td></tr>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process</td><td>1/134</td><td>83/21261 </td><td>0.4089122</td><td>0.5944998</td><td>0.5715403</td><td>ENSG00000151694</td><td>1</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process</td><td>1/134</td><td>112/21261</td><td>0.5083500</td><td>0.6579564</td><td>0.6325463</td><td>ENSG00000151694</td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>amyloid_genes &lt;- (xqtl_only_ego %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;amyloid&#39;)) %&gt;% pull(geneID) %&gt;% str_split(., &#39;/&#39;) %&gt;% unlist )
gene_ref %&gt;% filter(gene_id %in% amyloid_genes)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 2 x 5</caption>
<thead>
	<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_id</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr2 </td><td> 9556731</td><td> 9556732</td><td>ENSG00000151694</td><td>ADAM17</td></tr>
	<tr><td>chr19</td><td>47290022</td><td>47290023</td><td>ENSG00000197405</td><td>C5AR1 </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>barplot(xqtl_only_ego, showCategory = 10, title = &quot;GO Enrichment - BP&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/7aff41127e7b7f2a8023aee642288701c726744f8671cbfe4dbc165891504ccf.png"><img alt="../../../_images/7aff41127e7b7f2a8023aee642288701c726744f8671cbfe4dbc165891504ccf.png" src="../../../_images/7aff41127e7b7f2a8023aee642288701c726744f8671cbfe4dbc165891504ccf.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
</section>
<section id="id2">
<h4>KEGG<a class="headerlink" href="#id2" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>entrez_ids &lt;- bitr(xQTL_only$gene_id, fromType = &quot;ENSEMBL&quot;, 
                   toType = &quot;ENTREZID&quot;, OrgDb = org.Hs.eg.db)

kegg_enrich_only &lt;- enrichKEGG(gene         = entrez_ids$ENTREZID,
                          organism     = &#39;hsa&#39;,
                          pvalueCutoff = 1,
                         qvalueCutoff = 1)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:many mapping between keys and columns

Warning message in bitr(xQTL_only$gene_id, fromType = &quot;ENSEMBL&quot;, toType = &quot;ENTREZID&quot;, :
&quot;1.32% of input gene IDs are fail to map...&quot;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>kegg_enrich_only %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;lipid|Lipid|fatty|Fatty|Alzheimer|Cholesterol&#39;))
kegg_enrich_only %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;Alzheimer&#39;))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa00561</th><td>Metabolism    </td><td>Lipid metabolism                  </td><td>hsa00561</td><td>Glycerolipid metabolism                                   </td><td>2/59</td><td>65/8863 </td><td>0.06939863</td><td>0.7798039</td><td>0.7798039</td><td>55750/1609</td><td>2</td></tr>
	<tr><th scope=row>hsa00604</th><td>Metabolism    </td><td>Glycan biosynthesis and metabolism</td><td>hsa00604</td><td>Glycosphingolipid biosynthesis - ganglio series           </td><td>1/59</td><td>15/8863 </td><td>0.09540382</td><td>0.7798039</td><td>0.7798039</td><td>6483      </td><td>1</td></tr>
	<tr><th scope=row>hsa00603</th><td>Metabolism    </td><td>Glycan biosynthesis and metabolism</td><td>hsa00603</td><td>Glycosphingolipid biosynthesis - globo and isoglobo series</td><td>1/59</td><td>16/8863 </td><td>0.10143582</td><td>0.7798039</td><td>0.7798039</td><td>6483      </td><td>1</td></tr>
	<tr><th scope=row>hsa00564</th><td>Metabolism    </td><td>Lipid metabolism                  </td><td>hsa00564</td><td>Glycerophospholipid metabolism                            </td><td>1/59</td><td>103/8863</td><td>0.49940342</td><td>0.7838754</td><td>0.7838754</td><td>1609      </td><td>1</td></tr>
	<tr><th scope=row>hsa04932</th><td>Human Diseases</td><td>Endocrine and metabolic disease   </td><td>hsa04932</td><td>Non-alcoholic fatty liver disease                         </td><td>1/59</td><td>157/8863</td><td>0.65284917</td><td>0.8107516</td><td>0.8107516</td><td>1350      </td><td>1</td></tr>
	<tr><th scope=row>hsa05010</th><td>Human Diseases</td><td>Neurodegenerative disease         </td><td>hsa05010</td><td>Alzheimer disease                                         </td><td>2/59</td><td>391/8863</td><td>0.74112307</td><td>0.8497582</td><td>0.8497582</td><td>1350/6868 </td><td>2</td></tr>
</tbody>
</table>
</div><div class="output text_html"><table class="dataframe">
<caption>A data.frame: 1 x 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa05010</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05010</td><td>Alzheimer disease</td><td>2/59</td><td>391/8863</td><td>0.7411231</td><td>0.8497582</td><td>0.8497582</td><td>1350/6868</td><td>2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>kegg_enrich_only %&gt;% as.data.frame %&gt;% filter(str_detect(subcategory, &#39;Neurodegenerative&#39;))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa05016</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05016</td><td>Huntington disease                               </td><td>3/59</td><td>311/8863</td><td>0.3426881</td><td>0.7798039</td><td>0.7798039</td><td>7019/83544/1350</td><td>3</td></tr>
	<tr><th scope=row>hsa05014</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05014</td><td>Amyotrophic lateral sclerosis                    </td><td>2/59</td><td>371/8863</td><td>0.7139881</td><td>0.8474438</td><td>0.8474438</td><td>83544/1350     </td><td>2</td></tr>
	<tr><th scope=row>hsa05010</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05010</td><td>Alzheimer disease                                </td><td>2/59</td><td>391/8863</td><td>0.7411231</td><td>0.8497582</td><td>0.8497582</td><td>1350/6868      </td><td>2</td></tr>
	<tr><th scope=row>hsa05022</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05022</td><td>Pathways of neurodegeneration - multiple diseases</td><td>2/59</td><td>483/8863</td><td>0.8396301</td><td>0.8612552</td><td>0.8612552</td><td>83544/1350     </td><td>2</td></tr>
	<tr><th scope=row>hsa05012</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05012</td><td>Parkinson disease                                </td><td>1/59</td><td>271/8863</td><td>0.8409106</td><td>0.8612552</td><td>0.8612552</td><td>1350           </td><td>1</td></tr>
	<tr><th scope=row>hsa05020</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05020</td><td>Prion disease                                    </td><td>1/59</td><td>278/8863</td><td>0.8484045</td><td>0.8619790</td><td>0.8619790</td><td>1350           </td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>entrez_ids %&gt;% filter(ENTREZID%in% (kegg_enrich %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;Alzheimer&#39;)) %&gt;% pull(geneID) %&gt;% str_split(., &#39;/&#39;) %&gt;% unlist)) %&gt;% merge(.,gene_ref, by.x = &#39;ENSEMBL&#39;, by.y = &#39;gene_id&#39;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 2 x 6</caption>
<thead>
	<tr><th scope=col>ENSEMBL</th><th scope=col>ENTREZID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>ENSG00000127184</td><td>1350</td><td>chr5</td><td>86617927</td><td>86617928</td><td>COX7C </td></tr>
	<tr><td>ENSG00000151694</td><td>6868</td><td>chr2</td><td> 9556731</td><td> 9556732</td><td>ADAM17</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gene.data # xQTL list
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.dl-inline {width: auto; margin:0; padding: 0}
.dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}
.dl-inline>dt::after {content: ":\0020"; padding-right: .5ex}
.dl-inline>dt:not(:first-of-type) {padding-left: .5ex}
</style><dl class=dl-inline><dt>5595</dt><dd>1</dd><dt>1350</dt><dd>1</dd><dt>83464</dt><dd>1</dd><dt>351</dt><dd>1</dd><dt>6868</dt><dd>1</dd><dt>91252</dt><dd>1</dd><dt>4722</dt><dd>1</dd></dl>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gene.data[names(gene.data) %in% c(&#39;1350&#39;, &#39;6868&#39;)] &lt;- 2
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gene.data # xQTL list highlight xQTL only
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.dl-inline {width: auto; margin:0; padding: 0}
.dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}
.dl-inline>dt::after {content: ":\0020"; padding-right: .5ex}
.dl-inline>dt:not(:first-of-type) {padding-left: .5ex}
</style><dl class=dl-inline><dt>5595</dt><dd>1</dd><dt>1350</dt><dd>2</dd><dt>83464</dt><dd>1</dd><dt>351</dt><dd>1</dd><dt>6868</dt><dd>2</dd><dt>91252</dt><dd>1</dd><dt>4722</dt><dd>1</dd></dl>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Make sure gene.data is numeric named vector with correct format and proper range for color mapping
pathview(
  gene.data = gene.data,
  pathway.id = &quot;hsa05010&quot;,
  species = &quot;hsa&quot;,
  out.suffix = &quot;xQTL_only&quot;,
  kegg.native = TRUE,
  limit = list(gene = max(abs(gene.data))),  # set gene color scale limit properly
  low = &quot;green&quot;,
  mid = &quot;yellow&quot;,
  high = &quot;red&quot;,
  na.col = &quot;white&quot;   # color for genes without data
)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:1 mapping between keys and columns

Info: Working in directory /home/user/project/FunGen_xQTL/2025/May

Info: Writing image file hsa05010.xQTL_only.png
</pre></div>
</div>
</div>
</div>
</section>
<section id="amyloid-pathway-genes">
<h4>Amyloid pathway genes<a class="headerlink" href="#amyloid-pathway-genes" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>go_genes &lt;- AnnotationDbi::select(org.Hs.eg.db, keytype=&quot;GOALL&quot;, keys=&quot;GO:0097242&quot;, columns=c(&quot;ENSEMBL&quot;, &quot;SYMBOL&quot;))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:many mapping between keys and columns
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>go_genes %&gt;% pull(ENSEMBL) %&gt;% unique %&gt;% length
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">46</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>go_genes[,-2] %&gt;% distinct %&gt;% pull(SYMBOL)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'APOE'</li><li>'C3'</li><li>'C5AR1'</li><li>'CD36'</li><li>'CLU'</li><li>'CLTC'</li><li>'CYP51A1'</li><li>'HDAC1'</li><li>'HMGCR'</li><li>'IDE'</li><li>'IFNG'</li><li>'IFNGR1'</li><li>'IGF1R'</li><li>'IL4'</li><li>'INSR'</li><li>'ITGAM'</li><li>'ITGB2'</li><li>'LDLR'</li><li>'LRP1'</li><li>'LRP2'</li><li>'LRP4'</li><li>'LRPAP1'</li><li>'MME'</li><li>'MSR1'</li><li>'RAB5A'</li><li>'ROCK1'</li><li>'SRF'</li><li>'TNF'</li><li>'TNF'</li><li>'TNF'</li><li>'TNF'</li><li>'TNF'</li><li>'TNF'</li><li>'TNF'</li><li>'TNF'</li><li>'TTPA'</li><li>'PICALM'</li><li>'MARCO'</li><li>'RAB11A'</li><li>'RAB11B'</li><li>'ABCA7'</li><li>'PLA2G3'</li><li>'TREM2'</li><li>'MYOCD'</li><li>'MIR34A'</li><li>'MIR1908'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>go_genes[,-2] %&gt;% distinct %&gt;% filter(ENSEMBL %in% xQTL_list$gene_id)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 4 x 4</caption>
<thead>
	<tr><th scope=col>GOALL</th><th scope=col>ONTOLOGYALL</th><th scope=col>ENSEMBL</th><th scope=col>SYMBOL</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000197405</td><td>C5AR1 </td></tr>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000120885</td><td>CLU   </td></tr>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000073921</td><td>PICALM</td></tr>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000095970</td><td>TREM2 </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>go_genes[,-2] %&gt;% distinct %&gt;% filter(ENSEMBL %in% c(gene_list$gene_id, twas_list$gene_id))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 7 x 4</caption>
<thead>
	<tr><th scope=col>GOALL</th><th scope=col>ONTOLOGYALL</th><th scope=col>ENSEMBL</th><th scope=col>SYMBOL</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000130203</td><td>APOE  </td></tr>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000197405</td><td>C5AR1 </td></tr>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000120885</td><td>CLU   </td></tr>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000119912</td><td>IDE   </td></tr>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000073921</td><td>PICALM</td></tr>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000064687</td><td>ABCA7 </td></tr>
	<tr><td>GO:0097242</td><td>BP</td><td>ENSG00000095970</td><td>TREM2 </td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="context-gene-diff-list">
<h4>context: gene diff list<a class="headerlink" href="#context-gene-diff-list" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>context_res &lt;- list()
for(i in 1:length(twas_xQTL_list)){
    # entrez_ids &lt;- bitr(diff_list[diff_list %in% twas_xQTL_list[[i]]$molecular_id], fromType = &quot;ENSEMBL&quot;, 
    #                toType = &quot;ENTREZID&quot;, OrgDb = org.Hs.eg.db)

    context_res[[i]] &lt;- enrichGO(
      gene          = diff_list[diff_list %in% twas_xQTL_list[[i]]$molecular_id],
      OrgDb         = org.Hs.eg.db,
      keyType       = &quot;ENSEMBL&quot;,
      ont           = &quot;BP&quot;,
      pAdjustMethod = &quot;BH&quot;,
      pvalueCutoff  = 0.05,
      qvalueCutoff  = 0.2
    )
}
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>names(context_res) &lt;- names(twas_xQTL_list)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>res &lt;- data.frame()
for(i in seq_along(context_res)){
    if(nrow(head(context_res[[i]])) &gt; 0) {
        print(names(context_res)[[i]])
        res &lt;- rbind(res, as.data.frame(context_res[[i]]) %&gt;% mutate(context = names(context_res)[[i]]))
    }
}
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;Ast_DeJager_eQTL&quot;
[1] &quot;Ast_mega_eQTL&quot;
[1] &quot;BM_10_MSBB_eQTL&quot;
[1] &quot;BM_22_MSBB_eQTL&quot;
[1] &quot;BM_36_MSBB_eQTL&quot;
[1] &quot;BM_44_MSBB_eQTL&quot;
[1] &quot;DLPFC_Bennett_pQTL&quot;
[1] &quot;DLPFC_Klein_gpQTL&quot;
[1] &quot;Exc_DeJager_eQTL&quot;
[1] &quot;Exc_mega_eQTL&quot;
[1] &quot;Inh_DeJager_eQTL&quot;
[1] &quot;Inh_mega_eQTL&quot;
[1] &quot;Mic_DeJager_eQTL&quot;
[1] &quot;Mic_mega_eQTL&quot;
[1] &quot;OPC_mega_eQTL&quot;
[1] &quot;Oli_DeJager_eQTL&quot;
[1] &quot;Oli_mega_eQTL&quot;
[1] &quot;PCC_DeJager_eQTL&quot;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>res
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 584 x 10</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th><th scope=col>context</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0045746</th><td>GO:0045746</td><td>negative regulation of Notch signaling pathway     </td><td>1/2</td><td>41/21261 </td><td>0.0038531988</td><td>0.01989337</td><td>NA</td><td>ENSG00000139405</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0005978</th><td>GO:0005978</td><td>glycogen biosynthetic process                      </td><td>1/2</td><td>47/21261 </td><td>0.0044164577</td><td>0.01989337</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0009250</th><td>GO:0009250</td><td>glucan biosynthetic process                        </td><td>1/2</td><td>47/21261 </td><td>0.0044164577</td><td>0.01989337</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0000271</th><td>GO:0000271</td><td>polysaccharide biosynthetic process                </td><td>1/2</td><td>67/21261 </td><td>0.0062928368</td><td>0.01989337</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0005977</th><td>GO:0005977</td><td>glycogen metabolic process                         </td><td>1/2</td><td>79/21261 </td><td>0.0074178148</td><td>0.01989337</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0044042</th><td>GO:0044042</td><td>glucan metabolic process                           </td><td>1/2</td><td>82/21261 </td><td>0.0076989597</td><td>0.01989337</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0006112</th><td>GO:0006112</td><td>energy reserve metabolic process                   </td><td>1/2</td><td>92/21261 </td><td>0.0086358219</td><td>0.01989337</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0008593</th><td>GO:0008593</td><td>regulation of Notch signaling pathway              </td><td>1/2</td><td>103/21261</td><td>0.0096658592</td><td>0.01989337</td><td>NA</td><td>ENSG00000139405</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0005976</th><td>GO:0005976</td><td>polysaccharide metabolic process                   </td><td>1/2</td><td>106/21261</td><td>0.0099466856</td><td>0.01989337</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0043524</th><td>GO:0043524</td><td>negative regulation of neuron apoptotic process    </td><td>1/2</td><td>176/21261</td><td>0.0164879954</td><td>0.02766924</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0051168</th><td>GO:0051168</td><td>nuclear export                                     </td><td>1/2</td><td>186/21261</td><td>0.0174206983</td><td>0.02766924</td><td>NA</td><td>ENSG00000139405</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0007219</th><td>GO:0007219</td><td>Notch signaling pathway                            </td><td>1/2</td><td>197/21261</td><td>0.0184461605</td><td>0.02766924</td><td>NA</td><td>ENSG00000139405</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0016051</th><td>GO:0016051</td><td>carbohydrate biosynthetic process                  </td><td>1/2</td><td>229/21261</td><td>0.0214262791</td><td>0.02966716</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0043523</th><td>GO:0043523</td><td>regulation of neuron apoptotic process             </td><td>1/2</td><td>255/21261</td><td>0.0238442892</td><td>0.03065694</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0051402</th><td>GO:0051402</td><td>neuron apoptotic process                           </td><td>1/2</td><td>307/21261</td><td>0.0286713362</td><td>0.03440560</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0006913</th><td>GO:0006913</td><td>nucleocytoplasmic transport                        </td><td>1/2</td><td>362/21261</td><td>0.0337638470</td><td>0.03441102</td><td>NA</td><td>ENSG00000139405</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0051169</th><td>GO:0051169</td><td>nuclear transport                                  </td><td>1/2</td><td>362/21261</td><td>0.0337638470</td><td>0.03441102</td><td>NA</td><td>ENSG00000139405</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0015980</th><td>GO:0015980</td><td>energy derivation by oxidation of organic compounds</td><td>1/2</td><td>369/21261</td><td>0.0344110246</td><td>0.03441102</td><td>NA</td><td>ENSG00000114480</td><td>1</td><td>Ast_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0075509</th><td>GO:0075509</td><td>endocytosis involved in viral entry into host cell </td><td>1/1</td><td>13/21261 </td><td>0.0006114482</td><td>0.01528620</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0046794</th><td>GO:0046794</td><td>transport of virus                                 </td><td>1/1</td><td>30/21261 </td><td>0.0014110343</td><td>0.01763793</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0051209</th><td>GO:0051209</td><td>release of sequestered calcium ion into cytosol    </td><td>1/1</td><td>144/21261</td><td>0.0067729646</td><td>0.02085195</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0051283</th><td>GO:0051283</td><td>negative regulation of sequestering of calcium ion </td><td>1/1</td><td>145/21261</td><td>0.0068199991</td><td>0.02085195</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0051282</th><td>GO:0051282</td><td>regulation of sequestering of calcium ion          </td><td>1/1</td><td>147/21261</td><td>0.0069140680</td><td>0.02085195</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0051208</th><td>GO:0051208</td><td>sequestering of calcium ion                        </td><td>1/1</td><td>151/21261</td><td>0.0071022059</td><td>0.02085195</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0010508</th><td>GO:0010508</td><td>positive regulation of autophagy                   </td><td>1/1</td><td>161/21261</td><td>0.0075725507</td><td>0.02085195</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0035725</th><td>GO:0035725</td><td>sodium ion transmembrane transport                 </td><td>1/1</td><td>187/21261</td><td>0.0087954471</td><td>0.02085195</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0046718</th><td>GO:0046718</td><td>viral entry into host cell                         </td><td>1/1</td><td>202/21261</td><td>0.0095009642</td><td>0.02085195</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0044409</th><td>GO:0044409</td><td>entry into host                                    </td><td>1/1</td><td>210/21261</td><td>0.0098772400</td><td>0.02085195</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0097553</th><td>GO:0097553</td><td>calcium ion transmembrane import into cytosol      </td><td>1/1</td><td>216/21261</td><td>0.0101594469</td><td>0.02085195</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0044000</th><td>GO:0044000</td><td>movement in host                                   </td><td>1/1</td><td>236/21261</td><td>0.0111001364</td><td>0.02085195</td><td>NA</td><td>ENSG00000186815</td><td>1</td><td>Ast_mega_eQTL   </td></tr>
	<tr><th scope=row>...</th><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr>
	<tr><th scope=row>GO:0043484</th><td>GO:0043484</td><td>regulation of RNA splicing                                          </td><td>1/1 </td><td>194/21261</td><td>0.0091246884</td><td>0.009124688</td><td>        NA</td><td>ENSG00000075292                </td><td>1</td><td>OPC_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0009650</th><td>GO:0009650</td><td>UV protection                                                       </td><td>1/1 </td><td>12/21261 </td><td>0.0005644137</td><td>0.002822069</td><td>        NA</td><td>ENSG00000168077                </td><td>1</td><td>Oli_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0009411</th><td>GO:0009411</td><td>response to UV                                                      </td><td>1/1 </td><td>161/21261</td><td>0.0075725507</td><td>0.018931377</td><td>        NA</td><td>ENSG00000168077                </td><td>1</td><td>Oli_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0009416</th><td>GO:0009416</td><td>response to light stimulus                                          </td><td>1/1 </td><td>344/21261</td><td>0.0161798598</td><td>0.021871031</td><td>        NA</td><td>ENSG00000168077                </td><td>1</td><td>Oli_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0006979</th><td>GO:0006979</td><td>response to oxidative stress                                        </td><td>1/1 </td><td>450/21261</td><td>0.0211655143</td><td>0.021871031</td><td>        NA</td><td>ENSG00000168077                </td><td>1</td><td>Oli_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0009314</th><td>GO:0009314</td><td>response to radiation                                               </td><td>1/1 </td><td>465/21261</td><td>0.0218710315</td><td>0.021871031</td><td>        NA</td><td>ENSG00000168077                </td><td>1</td><td>Oli_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0097067</th><td>GO:0097067</td><td>cellular response to thyroid hormone stimulus                       </td><td>1/1 </td><td>18/21261 </td><td>0.0008466206</td><td>0.006381011</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0006590</th><td>GO:0006590</td><td>thyroid hormone generation                                          </td><td>1/1 </td><td>24/21261 </td><td>0.0011288274</td><td>0.006381011</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0046697</th><td>GO:0046697</td><td>decidualization                                                     </td><td>1/1 </td><td>27/21261 </td><td>0.0012699309</td><td>0.006381011</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0042403</th><td>GO:0042403</td><td>thyroid hormone metabolic process                                   </td><td>1/1 </td><td>30/21261 </td><td>0.0014110343</td><td>0.006381011</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0001893</th><td>GO:0001893</td><td>maternal placenta development                                       </td><td>1/1 </td><td>37/21261 </td><td>0.0017402756</td><td>0.006381011</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0097066</th><td>GO:0097066</td><td>response to thyroid hormone                                         </td><td>1/1 </td><td>37/21261 </td><td>0.0017402756</td><td>0.006381011</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0030574</th><td>GO:0030574</td><td>collagen catabolic process                                          </td><td>1/1 </td><td>50/21261 </td><td>0.0023517238</td><td>0.007391132</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0060135</th><td>GO:0060135</td><td>maternal process involved in female pregnancy                       </td><td>1/1 </td><td>61/21261 </td><td>0.0028691031</td><td>0.007890033</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0032963</th><td>GO:0032963</td><td>collagen metabolic process                                          </td><td>1/1 </td><td>119/21261</td><td>0.0055971027</td><td>0.012624053</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0018958</th><td>GO:0018958</td><td>phenol-containing compound metabolic process                        </td><td>1/1 </td><td>122/21261</td><td>0.0057382061</td><td>0.012624053</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:00065751</th><td>GO:0006575</td><td>cellular modified amino acid metabolic process                      </td><td>1/1 </td><td>139/21261</td><td>0.0065377922</td><td>0.013075584</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0001890</th><td>GO:0001890</td><td>placenta development                                                </td><td>1/1 </td><td>157/21261</td><td>0.0073844128</td><td>0.013538090</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0007565</th><td>GO:0007565</td><td>female pregnancy                                                    </td><td>1/1 </td><td>198/21261</td><td>0.0093128263</td><td>0.013566833</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:00467182</th><td>GO:0046718</td><td>viral entry into host cell                                          </td><td>1/1 </td><td>202/21261</td><td>0.0095009642</td><td>0.013566833</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:00444092</th><td>GO:0044409</td><td>entry into host                                                     </td><td>1/1 </td><td>210/21261</td><td>0.0098772400</td><td>0.013566833</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0044703</th><td>GO:0044703</td><td>multi-organism reproductive process                                 </td><td>1/1 </td><td>224/21261</td><td>0.0105357227</td><td>0.013566833</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0044706</th><td>GO:0044706</td><td>multi-multicellular organism process                                </td><td>1/1 </td><td>229/21261</td><td>0.0107708951</td><td>0.013566833</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:00440002</th><td>GO:0044000</td><td>movement in host                                                    </td><td>1/1 </td><td>236/21261</td><td>0.0111001364</td><td>0.013566833</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:00517012</th><td>GO:0051701</td><td>biological process involved in interaction with host                </td><td>1/1 </td><td>266/21261</td><td>0.0125111707</td><td>0.014434881</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:0042445</th><td>GO:0042445</td><td>hormone metabolic process                                           </td><td>1/1 </td><td>279/21261</td><td>0.0131226189</td><td>0.014434881</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:00190581</th><td>GO:0019058</td><td>viral life cycle                                                    </td><td>1/1 </td><td>396/21261</td><td>0.0186256526</td><td>0.018719722</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:00444031</th><td>GO:0044403</td><td>biological process involved in symbiotic interaction                </td><td>1/1 </td><td>398/21261</td><td>0.0187197216</td><td>0.018719722</td><td>        NA</td><td>ENSG00000164733                </td><td>1</td><td>Oli_mega_eQTL   </td></tr>
	<tr><th scope=row>GO:1904357</th><td>GO:1904357</td><td>negative regulation of telomere maintenance via telomere lengthening</td><td>2/12</td><td>32/21261 </td><td>0.0001434903</td><td>0.044116591</td><td>0.02902407</td><td>ENSG00000140451/ENSG00000173273</td><td>2</td><td>PCC_DeJager_eQTL</td></tr>
	<tr><th scope=row>GO:0032205</th><td>GO:0032205</td><td>negative regulation of telomere maintenance                         </td><td>2/12</td><td>40/21261 </td><td>0.0002250847</td><td>0.044116591</td><td>0.02902407</td><td>ENSG00000140451/ENSG00000173273</td><td>2</td><td>PCC_DeJager_eQTL</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
</section>
<section id="twas-only-list">
<h3>twas only list<a class="headerlink" href="#twas-only-list" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>twas_diff_ego &lt;- enrichGO(
  gene          = twas_diff_list,
  OrgDb         = org.Hs.eg.db,
  keyType       = &quot;ENSEMBL&quot;,
  ont           = &quot;BP&quot;,
  pAdjustMethod = &quot;BH&quot;,
  pvalueCutoff  = 1,
  qvalueCutoff  = 1
)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>twas_diff_ego %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0030198</th><td>GO:0030198</td><td>extracellular matrix organization            </td><td>14/242</td><td>376/21261</td><td>0.0001118404</td><td>0.1186644</td><td>0.1127558</td><td>ENSG00000060718/ENSG00000069399/ENSG00000082293/ENSG00000085998/ENSG00000104884/ENSG00000111799/ENSG00000120708/ENSG00000133392/ENSG00000135862/ENSG00000154175/ENSG00000164692/ENSG00000178031/ENSG00000187498/ENSG00000196569</td><td>14</td></tr>
	<tr><th scope=row>GO:0043062</th><td>GO:0043062</td><td>extracellular structure organization         </td><td>14/242</td><td>377/21261</td><td>0.0001149963</td><td>0.1186644</td><td>0.1127558</td><td>ENSG00000060718/ENSG00000069399/ENSG00000082293/ENSG00000085998/ENSG00000104884/ENSG00000111799/ENSG00000120708/ENSG00000133392/ENSG00000135862/ENSG00000154175/ENSG00000164692/ENSG00000178031/ENSG00000187498/ENSG00000196569</td><td>14</td></tr>
	<tr><th scope=row>GO:0045229</th><td>GO:0045229</td><td>external encapsulating structure organization</td><td>14/242</td><td>379/21261</td><td>0.0001215408</td><td>0.1186644</td><td>0.1127558</td><td>ENSG00000060718/ENSG00000069399/ENSG00000082293/ENSG00000085998/ENSG00000104884/ENSG00000111799/ENSG00000120708/ENSG00000133392/ENSG00000135862/ENSG00000154175/ENSG00000164692/ENSG00000178031/ENSG00000187498/ENSG00000196569</td><td>14</td></tr>
	<tr><th scope=row>GO:0009743</th><td>GO:0009743</td><td>response to carbohydrate                     </td><td>11/242</td><td>255/21261</td><td>0.0001733033</td><td>0.1269013</td><td>0.1205826</td><td>ENSG00000010310/ENSG00000012061/ENSG00000012232/ENSG00000075391/ENSG00000121691/ENSG00000127616/ENSG00000136504/ENSG00000145242/ENSG00000156711/ENSG00000169375/ENSG00000225697                                                </td><td>11</td></tr>
	<tr><th scope=row>GO:0051458</th><td>GO:0051458</td><td>corticotropin secretion                      </td><td>3/242 </td><td>11/21261 </td><td>0.0002246324</td><td>0.1315897</td><td>0.1250375</td><td>ENSG00000120088/ENSG00000166128/ENSG00000263715                                                                                                                                                                                </td><td> 3</td></tr>
	<tr><th scope=row>GO:0009650</th><td>GO:0009650</td><td>UV protection                                </td><td>3/242 </td><td>12/21261 </td><td>0.0002969951</td><td>0.1428420</td><td>0.1357296</td><td>ENSG00000012061/ENSG00000104884/ENSG00000121691                                                                                                                                                                                </td><td> 3</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>barplot(twas_diff_ego, showCategory = 10, title = &quot;GO Enrichment - BP&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/5e312cff8f3dc7cfdcf380399887d85085e2a6b4750024a9e8518710e1245126.png"><img alt="../../../_images/5e312cff8f3dc7cfdcf380399887d85085e2a6b4750024a9e8518710e1245126.png" src="../../../_images/5e312cff8f3dc7cfdcf380399887d85085e2a6b4750024a9e8518710e1245126.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>twas_diff_ego %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;amyloid&#39;))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 10 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0050435</th><td>GO:0050435</td><td>amyloid-beta metabolic process                                    </td><td>2/242</td><td>73/21261 </td><td>0.2018901</td><td>0.6088825</td><td>0.5785647</td><td>ENSG00000119912/ENSG00000125744</td><td>2</td></tr>
	<tr><th scope=row>GO:1902430</th><td>GO:1902430</td><td>negative regulation of amyloid-beta formation                     </td><td>1/242</td><td>31/21261 </td><td>0.2989164</td><td>0.6298750</td><td>0.5985120</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:1902992</th><td>GO:1902992</td><td>negative regulation of amyloid precursor protein catabolic process</td><td>1/242</td><td>34/21261 </td><td>0.3226202</td><td>0.6467313</td><td>0.6145289</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:0097242</th><td>GO:0097242</td><td>amyloid-beta clearance                                            </td><td>1/242</td><td>46/21261 </td><td>0.4097161</td><td>0.6749486</td><td>0.6413412</td><td>ENSG00000119912                </td><td>1</td></tr>
	<tr><th scope=row>GO:1902003</th><td>GO:1902003</td><td>regulation of amyloid-beta formation                              </td><td>1/242</td><td>57/21261 </td><td>0.4797158</td><td>0.6962773</td><td>0.6616079</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:1990000</th><td>GO:1990000</td><td>amyloid fibril formation                                          </td><td>1/242</td><td>61/21261 </td><td>0.5030658</td><td>0.7075611</td><td>0.6723299</td><td>ENSG00000186868                </td><td>1</td></tr>
	<tr><th scope=row>GO:0034205</th><td>GO:0034205</td><td>amyloid-beta formation                                            </td><td>1/242</td><td>65/21261 </td><td>0.5253720</td><td>0.7150625</td><td>0.6794578</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:1902991</th><td>GO:1902991</td><td>regulation of amyloid precursor protein catabolic process         </td><td>1/242</td><td>70/21261 </td><td>0.5518575</td><td>0.7277760</td><td>0.6915382</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process                       </td><td>1/242</td><td>83/21261 </td><td>0.6140317</td><td>0.7582204</td><td>0.7204667</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process                       </td><td>1/242</td><td>112/21261</td><td>0.7234908</td><td>0.8074249</td><td>0.7672212</td><td>ENSG00000125744                </td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="multi-gene-case-ypel3">
<h3>multi gene case - YPEL3<a class="headerlink" href="#multi-gene-case-ypel3" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># multi_gene_list &lt;- c(&#39;YPEL3&#39;, &#39;INO80E&#39;, &#39;DOC2A&#39;, &#39;MAPK3&#39;, &#39;CORO1A&#39;, &#39;BCKDK&#39;, &#39;TLCD3B&#39;, &#39;ZNF267&#39;)
multi_gene_list &lt;- c(&#39;ENSG00000090238&#39;,&#39;ENSG00000169592&#39;,&#39;ENSG00000149927&#39;,&#39;ENSG00000102879&#39;,&#39;ENSG00000149930&#39;,&#39;ENSG00000149932&#39;,&#39;ENSG00000102882&#39;,&#39;ENSG00000103507&#39;,&#39;ENSG00000149926&#39;,&#39;ENSG00000185947&#39;, &#39;ENSG00000285043&#39;, &#39;ENSG00000052344&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gene_ref %&gt;% filter(gene_id %in% multi_gene_list)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 12 x 5</caption>
<thead>
	<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_id</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr16</td><td>29940884</td><td>29940885</td><td>ENSG00000149932</td><td>TMEM219   </td></tr>
	<tr><td>chr16</td><td>29973867</td><td>29973868</td><td>ENSG00000149930</td><td>TAOK2     </td></tr>
	<tr><td>chr16</td><td>29995714</td><td>29995715</td><td>ENSG00000169592</td><td>INO80E    </td></tr>
	<tr><td>chr16</td><td>30023269</td><td>30023270</td><td>ENSG00000149927</td><td>DOC2A     </td></tr>
	<tr><td>chr16</td><td>30052977</td><td>30052978</td><td>ENSG00000149926</td><td>TLCD3B    </td></tr>
	<tr><td>chr16</td><td>30053089</td><td>30053090</td><td>ENSG00000285043</td><td>AC093512.2</td></tr>
	<tr><td>chr16</td><td>30096914</td><td>30096915</td><td>ENSG00000090238</td><td>YPEL3     </td></tr>
	<tr><td>chr16</td><td>30123505</td><td>30123506</td><td>ENSG00000102882</td><td>MAPK3     </td></tr>
	<tr><td>chr16</td><td>30182826</td><td>30182827</td><td>ENSG00000102879</td><td>CORO1A    </td></tr>
	<tr><td>chr16</td><td>31106106</td><td>31106107</td><td>ENSG00000103507</td><td>BCKDK     </td></tr>
	<tr><td>chr16</td><td>31135726</td><td>31135727</td><td>ENSG00000052344</td><td>PRSS8     </td></tr>
	<tr><td>chr16</td><td>31873806</td><td>31873807</td><td>ENSG00000185947</td><td>ZNF267    </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multigene_ego &lt;- enrichGO(
  gene          = multi_gene_list,
  OrgDb         = org.Hs.eg.db,
  keyType       = &quot;ENSEMBL&quot;,
  ont           = &quot;BP&quot;,
  pAdjustMethod = &quot;BH&quot;,
  pvalueCutoff  = 1,
  qvalueCutoff  = 1
)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multigene_ego %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0031338</th><td>GO:0031338</td><td>regulation of vesicle fusion                             </td><td>2/11</td><td>26/21261 </td><td>7.855737e-05</td><td>0.03055882</td><td>0.01852300</td><td>ENSG00000149927/ENSG00000102879</td><td>2</td></tr>
	<tr><th scope=row>GO:0098927</th><td>GO:0098927</td><td>vesicle-mediated transport between endosomal compartments</td><td>2/11</td><td>53/21261 </td><td>3.305544e-04</td><td>0.06429282</td><td>0.03897062</td><td>ENSG00000102879/ENSG00000102882</td><td>2</td></tr>
	<tr><th scope=row>GO:0032206</th><td>GO:0032206</td><td>positive regulation of telomere maintenance              </td><td>2/11</td><td>92/21261 </td><td>9.931434e-04</td><td>0.07133644</td><td>0.04324006</td><td>ENSG00000169592/ENSG00000102882</td><td>2</td></tr>
	<tr><th scope=row>GO:0006906</th><td>GO:0006906</td><td>vesicle fusion                                           </td><td>2/11</td><td>129/21261</td><td>1.938415e-03</td><td>0.07133644</td><td>0.04324006</td><td>ENSG00000149927/ENSG00000102879</td><td>2</td></tr>
	<tr><th scope=row>GO:0032204</th><td>GO:0032204</td><td>regulation of telomere maintenance                       </td><td>2/11</td><td>130/21261</td><td>1.968147e-03</td><td>0.07133644</td><td>0.04324006</td><td>ENSG00000169592/ENSG00000102882</td><td>2</td></tr>
	<tr><th scope=row>GO:0090174</th><td>GO:0090174</td><td>organelle membrane fusion                                </td><td>2/11</td><td>131/21261</td><td>1.998097e-03</td><td>0.07133644</td><td>0.04324006</td><td>ENSG00000149927/ENSG00000102879</td><td>2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>barplot(multigene_ego, showCategory = 10, title = &quot;GO Enrichment - BP&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/a25cfeda03fa1579e7c090b23a77d2ed27c27ba9ae6ed1314bac568cea707f2e.png"><img alt="../../../_images/a25cfeda03fa1579e7c090b23a77d2ed27c27ba9ae6ed1314bac568cea707f2e.png" src="../../../_images/a25cfeda03fa1579e7c090b23a77d2ed27c27ba9ae6ed1314bac568cea707f2e.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multigene_ego %&gt;% as.data.frame %&gt;% filter(str_detect(Description, &#39;amyloid&#39;))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 0 x 9</caption>
<thead>
	<tr><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="multi-gene-case-nbeal1">
<h3>multi gene case - NBEAL1<a class="headerlink" href="#multi-gene-case-nbeal1" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multi_gene_list2 &lt;- c(&#39;WDR81&#39;, &#39;TSC22D4&#39;, &#39;ICA1L&#39;, &#39;FAM117B&#39;, &#39;NBEAL1&#39;, &#39;CARF&#39;, &#39;WDR12&#39;, &quot;DHX33&quot;, &quot;EMG1&quot;, &quot;DYSF&quot;)
multi_gene_list2 &lt;- gene_ref %&gt;% filter(gene_name %in% multi_gene_list2) %&gt;% pull(gene_id)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>
multigene_ego &lt;- enrichGO(
  gene          = multi_gene_list2,
  OrgDb         = org.Hs.eg.db,
  keyType       = &quot;ENSEMBL&quot;,
  ont           = &quot;BP&quot;,
  pAdjustMethod = &quot;BH&quot;,
  pvalueCutoff  = 1,
  qvalueCutoff  = 1
)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multigene_ego %&gt;% as.data.frame %&gt;% arrange(desc(Count))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 92 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0006364</th><td>GO:0006364</td><td>rRNA processing                                                                          </td><td>2/9</td><td>250/21261</td><td>0.004694911</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000138442/ENSG00000126749</td><td>2</td></tr>
	<tr><th scope=row>GO:0016072</th><td>GO:0016072</td><td>rRNA metabolic process                                                                   </td><td>2/9</td><td>292/21261</td><td>0.006349674</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000138442/ENSG00000126749</td><td>2</td></tr>
	<tr><th scope=row>GO:0042254</th><td>GO:0042254</td><td>ribosome biogenesis                                                                      </td><td>2/9</td><td>365/21261</td><td>0.009769967</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000138442/ENSG00000126749</td><td>2</td></tr>
	<tr><th scope=row>GO:0034470</th><td>GO:0034470</td><td>ncRNA processing                                                                         </td><td>2/9</td><td>490/21261</td><td>0.017141532</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000138442/ENSG00000126749</td><td>2</td></tr>
	<tr><th scope=row>GO:0043551</th><td>GO:0043551</td><td>regulation of phosphatidylinositol 3-kinase activity                                     </td><td>1/9</td><td>10/21261 </td><td>0.004225941</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000167716                </td><td>1</td></tr>
	<tr><th scope=row>GO:0035973</th><td>GO:0035973</td><td>aggrephagy                                                                               </td><td>1/9</td><td>11/21261 </td><td>0.004647661</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000167716                </td><td>1</td></tr>
	<tr><th scope=row>GO:0070475</th><td>GO:0070475</td><td>rRNA base methylation                                                                    </td><td>1/9</td><td>11/21261 </td><td>0.004647661</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000126749                </td><td>1</td></tr>
	<tr><th scope=row>GO:0000463</th><td>GO:0000463</td><td>maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA) </td><td>1/9</td><td>17/21261 </td><td>0.007174647</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000138442                </td><td>1</td></tr>
	<tr><th scope=row>GO:0042117</th><td>GO:0042117</td><td>monocyte activation                                                                      </td><td>1/9</td><td>18/21261 </td><td>0.007595257</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000135636                </td><td>1</td></tr>
	<tr><th scope=row>GO:0007000</th><td>GO:0007000</td><td>nucleolus organization                                                                   </td><td>1/9</td><td>19/21261 </td><td>0.008015708</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000126749                </td><td>1</td></tr>
	<tr><th scope=row>GO:0050765</th><td>GO:0050765</td><td>negative regulation of phagocytosis                                                      </td><td>1/9</td><td>23/21261 </td><td>0.009695929</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000135636                </td><td>1</td></tr>
	<tr><th scope=row>GO:1900227</th><td>GO:1900227</td><td>positive regulation of NLRP3 inflammasome complex assembly                               </td><td>1/9</td><td>26/21261 </td><td>0.010954435</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0002281</th><td>GO:0002281</td><td>macrophage activation involved in immune response                                        </td><td>1/9</td><td>27/21261 </td><td>0.011373620</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000135636                </td><td>1</td></tr>
	<tr><th scope=row>GO:0141087</th><td>GO:0141087</td><td>positive regulation of inflammasome-mediated signaling pathway                           </td><td>1/9</td><td>27/21261 </td><td>0.011373620</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0000466</th><td>GO:0000466</td><td>maturation of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)</td><td>1/9</td><td>28/21261 </td><td>0.011792648</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000138442                </td><td>1</td></tr>
	<tr><th scope=row>GO:0031167</th><td>GO:0031167</td><td>rRNA methylation                                                                         </td><td>1/9</td><td>29/21261 </td><td>0.012211518</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000126749                </td><td>1</td></tr>
	<tr><th scope=row>GO:0000470</th><td>GO:0000470</td><td>maturation of LSU-rRNA                                                                   </td><td>1/9</td><td>30/21261 </td><td>0.012630230</td><td>0.07262383</td><td>0.04154681</td><td>ENSG00000138442                </td><td>1</td></tr>
	<tr><th scope=row>GO:0045943</th><td>GO:0045943</td><td>positive regulation of transcription by RNA polymerase I                                 </td><td>1/9</td><td>36/21261 </td><td>0.015139192</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0000154</th><td>GO:0000154</td><td>rRNA modification                                                                        </td><td>1/9</td><td>39/21261 </td><td>0.016391546</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000126749                </td><td>1</td></tr>
	<tr><th scope=row>GO:0000460</th><td>GO:0000460</td><td>maturation of 5.8S rRNA                                                                  </td><td>1/9</td><td>39/21261 </td><td>0.016391546</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000138442                </td><td>1</td></tr>
	<tr><th scope=row>GO:0043550</th><td>GO:0043550</td><td>regulation of lipid kinase activity                                                      </td><td>1/9</td><td>42/21261 </td><td>0.017642485</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000167716                </td><td>1</td></tr>
	<tr><th scope=row>GO:0045022</th><td>GO:0045022</td><td>early endosome to late endosome transport                                                </td><td>1/9</td><td>47/21261 </td><td>0.019724240</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000167716                </td><td>1</td></tr>
	<tr><th scope=row>GO:1900225</th><td>GO:1900225</td><td>regulation of NLRP3 inflammasome complex assembly                                        </td><td>1/9</td><td>48/21261 </td><td>0.020140120</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0006356</th><td>GO:0006356</td><td>regulation of transcription by RNA polymerase I                                          </td><td>1/9</td><td>49/21261 </td><td>0.020555843</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0044546</th><td>GO:0044546</td><td>NLRP3 inflammasome complex assembly                                                      </td><td>1/9</td><td>51/21261 </td><td>0.021386820</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0141085</th><td>GO:0141085</td><td>regulation of inflammasome-mediated signaling pathway                                    </td><td>1/9</td><td>51/21261 </td><td>0.021386820</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0098927</th><td>GO:0098927</td><td>vesicle-mediated transport between endosomal compartments                                </td><td>1/9</td><td>53/21261 </td><td>0.022217169</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000167716                </td><td>1</td></tr>
	<tr><th scope=row>GO:0140632</th><td>GO:0140632</td><td>canonical inflammasome complex assembly                                                  </td><td>1/9</td><td>53/21261 </td><td>0.022217169</td><td>0.07299927</td><td>0.04176160</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0141084</th><td>GO:0141084</td><td>inflammasome-mediated signaling pathway                                                  </td><td>1/9</td><td>56/21261 </td><td>0.023461520</td><td>0.07442965</td><td>0.04257989</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0070050</th><td>GO:0070050</td><td>neuron cellular homeostasis                                                              </td><td>1/9</td><td>70/21261 </td><td>0.029249895</td><td>0.08969968</td><td>0.05131561</td><td>ENSG00000166925                </td><td>1</td></tr>
	<tr><th scope=row>...</th><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr>
	<tr><th scope=row>GO:0071248</th><td>GO:0071248</td><td>cellular response to metal ion                                  </td><td>1/9</td><td>213/21261</td><td>0.08665073</td><td>0.1265376</td><td>0.07238992</td><td>ENSG00000138380</td><td>1</td></tr>
	<tr><th scope=row>GO:0006836</th><td>GO:0006836</td><td>neurotransmitter transport                                      </td><td>1/9</td><td>223/21261</td><td>0.09054949</td><td>0.1270057</td><td>0.07265774</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0043414</th><td>GO:0043414</td><td>macromolecule methylation                                       </td><td>1/9</td><td>226/21261</td><td>0.09171622</td><td>0.1270057</td><td>0.07265774</td><td>ENSG00000126749</td><td>1</td></tr>
	<tr><th scope=row>GO:1902075</th><td>GO:1902075</td><td>cellular response to salt                                       </td><td>1/9</td><td>226/21261</td><td>0.09171622</td><td>0.1270057</td><td>0.07265774</td><td>ENSG00000138380</td><td>1</td></tr>
	<tr><th scope=row>GO:0031334</th><td>GO:0031334</td><td>positive regulation of protein-containing complex assembly      </td><td>1/9</td><td>228/21261</td><td>0.09249331</td><td>0.1270057</td><td>0.07265774</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0071241</th><td>GO:0071241</td><td>cellular response to inorganic substance                        </td><td>1/9</td><td>244/21261</td><td>0.09868875</td><td>0.1335201</td><td>0.07638448</td><td>ENSG00000138380</td><td>1</td></tr>
	<tr><th scope=row>GO:0050821</th><td>GO:0050821</td><td>protein stabilization                                           </td><td>1/9</td><td>252/21261</td><td>0.10177235</td><td>0.1356965</td><td>0.07762956</td><td>ENSG00000167716</td><td>1</td></tr>
	<tr><th scope=row>GO:0048588</th><td>GO:0048588</td><td>developmental cell growth                                       </td><td>1/9</td><td>260/21261</td><td>0.10484657</td><td>0.1377983</td><td>0.07883201</td><td>ENSG00000166925</td><td>1</td></tr>
	<tr><th scope=row>GO:0060560</th><td>GO:0060560</td><td>developmental growth involved in morphogenesis                  </td><td>1/9</td><td>264/21261</td><td>0.10638017</td><td>0.1378447</td><td>0.07885854</td><td>ENSG00000166925</td><td>1</td></tr>
	<tr><th scope=row>GO:0006909</th><td>GO:0006909</td><td>phagocytosis                                                    </td><td>1/9</td><td>273/21261</td><td>0.10982223</td><td>0.1398540</td><td>0.08000799</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0042593</th><td>GO:0042593</td><td>glucose homeostasis                                             </td><td>1/9</td><td>279/21261</td><td>0.11211039</td><td>0.1398540</td><td>0.08000799</td><td>ENSG00000166925</td><td>1</td></tr>
	<tr><th scope=row>GO:0033500</th><td>GO:0033500</td><td>carbohydrate homeostasis                                        </td><td>1/9</td><td>280/21261</td><td>0.11249124</td><td>0.1398540</td><td>0.08000799</td><td>ENSG00000166925</td><td>1</td></tr>
	<tr><th scope=row>GO:0002274</th><td>GO:0002274</td><td>myeloid leukocyte activation                                    </td><td>1/9</td><td>292/21261</td><td>0.11705013</td><td>0.1435815</td><td>0.08214044</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0032259</th><td>GO:0032259</td><td>methylation                                                     </td><td>1/9</td><td>302/21261</td><td>0.12083329</td><td>0.1447460</td><td>0.08280666</td><td>ENSG00000126749</td><td>1</td></tr>
	<tr><th scope=row>GO:0002221</th><td>GO:0002221</td><td>pattern recognition receptor signaling pathway                  </td><td>1/9</td><td>303/21261</td><td>0.12121082</td><td>0.1447460</td><td>0.08280666</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0051091</th><td>GO:0051091</td><td>positive regulation of DNA-binding transcription factor activity</td><td>1/9</td><td>307/21261</td><td>0.12271947</td><td>0.1447460</td><td>0.08280666</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0002758</th><td>GO:0002758</td><td>innate immune response-activating signaling pathway             </td><td>1/9</td><td>333/21261</td><td>0.13246971</td><td>0.1542685</td><td>0.08825430</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0030100</th><td>GO:0030100</td><td>regulation of endocytosis                                       </td><td>1/9</td><td>340/21261</td><td>0.13507826</td><td>0.1553400</td><td>0.08886728</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0002366</th><td>GO:0002366</td><td>leukocyte activation involved in immune response                </td><td>1/9</td><td>353/21261</td><td>0.13990422</td><td>0.1571249</td><td>0.08988838</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0002263</th><td>GO:0002263</td><td>cell activation involved in immune response                     </td><td>1/9</td><td>357/21261</td><td>0.14138431</td><td>0.1571249</td><td>0.08988838</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0002218</th><td>GO:0002218</td><td>activation of innate immune response                            </td><td>1/9</td><td>358/21261</td><td>0.14175398</td><td>0.1571249</td><td>0.08988838</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0016236</th><td>GO:0016236</td><td>macroautophagy                                                  </td><td>1/9</td><td>367/21261</td><td>0.14507463</td><td>0.1588913</td><td>0.09089889</td><td>ENSG00000167716</td><td>1</td></tr>
	<tr><th scope=row>GO:0031647</th><td>GO:0031647</td><td>regulation of protein stability                                 </td><td>1/9</td><td>376/21261</td><td>0.14838385</td><td>0.1606037</td><td>0.09187855</td><td>ENSG00000167716</td><td>1</td></tr>
	<tr><th scope=row>GO:0010038</th><td>GO:0010038</td><td>response to metal ion                                           </td><td>1/9</td><td>384/21261</td><td>0.15131583</td><td>0.1618727</td><td>0.09260454</td><td>ENSG00000138380</td><td>1</td></tr>
	<tr><th scope=row>GO:1902074</th><td>GO:1902074</td><td>response to salt                                                </td><td>1/9</td><td>410/21261</td><td>0.16078286</td><td>0.1700233</td><td>0.09726731</td><td>ENSG00000138380</td><td>1</td></tr>
	<tr><th scope=row>GO:0030522</th><td>GO:0030522</td><td>intracellular receptor signaling pathway                        </td><td>1/9</td><td>426/21261</td><td>0.16656196</td><td>0.1740357</td><td>0.09956278</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0001701</th><td>GO:0001701</td><td>in utero embryonic development                                  </td><td>1/9</td><td>431/21261</td><td>0.16836067</td><td>0.1740357</td><td>0.09956278</td><td>ENSG00000126749</td><td>1</td></tr>
	<tr><th scope=row>GO:0043254</th><td>GO:0043254</td><td>regulation of protein-containing complex assembly               </td><td>1/9</td><td>470/21261</td><td>0.18227252</td><td>0.1863230</td><td>0.10659212</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0051090</th><td>GO:0051090</td><td>regulation of DNA-binding transcription factor activity         </td><td>1/9</td><td>488/21261</td><td>0.18862332</td><td>0.1900286</td><td>0.10871203</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0045089</th><td>GO:0045089</td><td>positive regulation of innate immune response                   </td><td>1/9</td><td>492/21261</td><td>0.19002864</td><td>0.1900286</td><td>0.10871203</td><td>ENSG00000005100</td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>entrez_ids &lt;- bitr(multi_gene_list2, fromType = &quot;ENSEMBL&quot;, 
                   toType = &quot;ENTREZID&quot;, OrgDb = org.Hs.eg.db)

kegg_enrich_only &lt;- enrichKEGG(gene         = entrez_ids$ENTREZID,
                          organism     = &#39;hsa&#39;,
                          pvalueCutoff = 1,
                         qvalueCutoff = 1)
kegg_enrich_only %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:1 mapping between keys and columns
</pre></div>
</div>
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 2 x 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa04621</th><td>Organismal Systems            </td><td>Immune system</td><td>hsa04621</td><td>NOD-like receptor signaling pathway</td><td>1/2</td><td>189/8863</td><td>0.04219683</td><td>0.08439366</td><td>0.04441772</td><td>56919</td><td>1</td></tr>
	<tr><th scope=row>hsa03008</th><td>Genetic Information Processing</td><td>Translation  </td><td>hsa03008</td><td>Ribosome biogenesis in eukaryotes  </td><td>1/2</td><td>467/8863</td><td>0.10261122</td><td>0.10261122</td><td>0.05400591</td><td>10436</td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
</section>
<section id="ppi">
<h2>PPI<a class="headerlink" href="#ppi" title="Link to this heading">#</a></h2>
<section id="xqtl">
<h3>xQTL<a class="headerlink" href="#xqtl" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Load necessary libraries
library(STRINGdb)
library(dplyr)
library(igraph)

# Initialize STRINGdb (change species if needed, 9606 = Homo sapiens)
string_db &lt;- STRINGdb$new(version=&quot;11.5&quot;, species=9606, score_threshold=400, input_directory=&quot;&quot;)

# Example gene list (use your own gene symbols or Entrez IDs)
genes &lt;- xQTL_list$gene_name

# Map gene names to STRING IDs
mapped_genes &lt;- string_db$map(data.frame(gene=genes), &quot;gene&quot;, removeUnmappedRows = TRUE)

# Retrieve interaction network for the genes
ppi_network &lt;- string_db$get_interactions(mapped_genes$STRING_id)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning:  we couldn&#39;t map to STRING 2% of your identifiers
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>ppi_network_unique &lt;- ppi_network %&gt;%
  mutate(pair = paste0(pmin(from, to), &quot;_&quot;, pmax(from, to))) %&gt;%
  distinct(pair, .keep_all = TRUE)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>genes_with_ppi &lt;- mapped_genes %&gt;% filter(STRING_id %in% c(ppi_network$from, ppi_network$to))
genes_with_ppi %&gt;% dim 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_list %&gt;% pull(resource_in_FunGen_xQTL) %&gt;% table
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_list %&gt;% dim
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>240</li><li>7</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_list %&gt;% filter(!(gene_name %in% genes_with_ppi$gene)) %&gt;% pull(resource_in_FunGen_xQTL) %&gt;% table # the xQTL genes not have any PPI
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>.
                                                  ColocBoost 
                                                          40 
                                   ColocBoost,single_context 
                                                           2 
                     ColocBoost,single_context,multi_context 
                                                           2 
           ColocBoost,single_context,multi_context,MetaBrain 
                                                           1 
                        ColocBoost,single_context,multi_gene 
                                                           1 
                                                   MetaBrain 
                                                           2 
                                        MetaBrain,ColocBoost 
                                                           1 
                                              single_context 
                                                           1 
                                    single_context,MetaBrain 
                                                           1 
                      single_context,multi_context,MetaBrain 
                                                           2 
single_context,multi_context,MetaBrain,multi_gene,ColocBoost 
                                                           1 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># whether the genes from ColocBoost are siginificant different from all genes and PPI genes
# Create contingency table
#            | PPI     | No PPI
# ---------- | ------- | -------
# methodA    | 120     | 40
# notMethodA | 66      | 14

# Total genes with PPI = 186
# Total genes without PPI = 240 - 186 = 54
# Total methodA = 160; of which 40 are without PPI → 120 with PPI
# Total not methodA = 80; of which 14 are without PPI → 66 with PPI

mat &lt;- matrix(c(120, 40, 66, 14), nrow = 2,
              dimnames = list(Method = c(&quot;methodA&quot;, &quot;notMethodA&quot;),
                              PPI = c(&quot;PPI&quot;, &quot;No_PPI&quot;)))

# Perform Fisher&#39;s exact test
fisher.test(mat)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>	Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.2508
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.297715 1.302572
sample estimates:
odds ratio 
 0.6375147 
</pre></div>
</div>
</div>
</div>
<p>Even it is not significant, I am still wondering if there are more genes in PPI are detected by &gt;1 methods</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>symbol_lookup &lt;- dplyr::select(mapped_genes, STRING_id, gene)
symbol_lookup &lt;- dplyr::distinct(symbol_lookup)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>options(repr.plot.width = 12, repr.plot.height = 12)
library(igraph)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Build igraph object
g &lt;- graph_from_data_frame(ppi_network[, c(&quot;from&quot;, &quot;to&quot;)], directed=FALSE)
g &lt;- igraph::simplify(g, remove.multiple = TRUE, remove.loops = TRUE)
# Replace vertex names with gene symbols
V(g)$name &lt;- symbol_lookup$gene[match(V(g)$name, symbol_lookup$STRING_id)]
# Optional: make highlighted genes larger
# V(g)$size &lt;- ifelse(V(g)$name %in% highlight_genes, 8, 4)
</pre></div>
</div>
</div>
</div>
<section id="xqtl-only-genes-labelled-by-methods">
<h4>xQTL only genes labelled by methods<a class="headerlink" href="#xqtl-only-genes-labelled-by-methods" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_shared_method &lt;- xQTL_list %&gt;% filter(gene_name %in% genes_with_ppi$gene, gene_name %in% xQTL_only$gene_name) %&gt;% filter(str_detect(resource_in_FunGen_xQTL, &#39;,&#39;)) %&gt;% pull(gene_name)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(tidygraph)
library(ggraph)
tg &lt;- as_tbl_graph(g) %&gt;% 
  mutate(highlight = ifelse(name %in% real_gvc_list$gene_name, &quot;GVC&quot;, # highlight GVC2 genes as &#39;core genes&#39;
                            ifelse(name %in% xQTL_shared_method, &quot;xQTL_only_shared_method&quot;, # the genes are xQTL only, but more than 1 methods
                                   ifelse(name %in% xQTL_only_1_se_cb$gene_name, &quot;xQTL_only_only_1_se_cb&quot;, # the genes are only from 1 splicing event colocalized results, which are alarming
                                       ifelse(name %in% xQTL_only$gene_name, &quot;xQTL_only_single_method&quot;, &quot;shared&quot;))))) # the genes are xQTL only but only from one methods / shared with GVC or TWAS

set.seed(123)
ggraph(tg, layout = &quot;fr&quot;) +  # fr: Fruchterman-Reingold layout
  geom_edge_link(color = &quot;grey80&quot;, alpha = 0.6) +
  geom_node_point(aes(color = highlight), size = 5) +
  geom_node_text(aes(label = name), repel = TRUE, size = 3) +
  scale_color_manual(values = c(&quot;GVC&quot; = &quot;red&quot;, &quot;shared&quot; = &quot;orange&quot;, &quot;xQTL_only_shared_method&quot; = &quot;yellow&quot;, &quot;xQTL_only_only_1_se_cb&quot; = &quot;black&quot;, &quot;xQTL_only_single_method&quot; = &quot;skyblue&quot;)) +
  theme_void() +
  theme(legend.position = &quot;none&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Attaching package: &#39;tidygraph&#39;


The following object is masked from &#39;package:igraph&#39;:

    groups


The following object is masked from &#39;package:stats&#39;:

    filter


Warning message:
&quot;ggrepel: 15 unlabeled data points (too many overlaps). Consider increasing max.overlaps&quot;
</pre></div>
</div>
<a class="reference internal image-reference" href="../../../_images/56b79c5667ae57f27493728d145263bd1f31e3a7211654c03a7ff37457535b4c.png"><img alt="../../../_images/56b79c5667ae57f27493728d145263bd1f31e3a7211654c03a7ff37457535b4c.png" src="../../../_images/56b79c5667ae57f27493728d145263bd1f31e3a7211654c03a7ff37457535b4c.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
</section>
<section id="xqtl-only-genes-labelled-by-non-proxy-ad-genes">
<h4>xQTL only genes labelled by non proxy AD genes<a class="headerlink" href="#xqtl-only-genes-labelled-by-non-proxy-ad-genes" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>nonproxy_ad_genes &lt;- readLines(&#39;Non_proxy_AD_gene_xQTL.txt&#39;)
nonproxy_ad_genes_names &lt;- gene_ref %&gt;% filter(gene_id %in% nonproxy_ad_genes) %&gt;% pull(gene_name)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>nonproxy_ad_genes_names %&gt;% length
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">128</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(tidygraph)
library(ggraph)
tg &lt;- as_tbl_graph(g) %&gt;% 
  mutate(highlight = ifelse(name %in% nonproxy_ad_genes_names, &quot;Non_proxy_AD_genes&quot;, &quot;Others&quot;)) # the genes are xQTL only but only from one methods / shared with GVC or TWAS

set.seed(123)
ggraph(tg, layout = &quot;fr&quot;) +  # fr: Fruchterman-Reingold layout
  geom_edge_link(color = &quot;grey80&quot;, alpha = 0.6) +
  geom_node_point(aes(color = highlight), size = 5) +
  geom_node_text(aes(label = name), repel = TRUE, size = 3) +
  scale_color_manual(values = c(&quot;Non_proxy_AD_genes&quot; = &quot;red&quot;, &quot;Others&quot; = &quot;skyblue&quot;)) +
  theme_void() 
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
&quot;ggrepel: 22 unlabeled data points (too many overlaps). Consider increasing max.overlaps&quot;
</pre></div>
</div>
<a class="reference internal image-reference" href="../../../_images/dbe43e06fa616b9beb909d1955eba9ab8f83a4045a9d5fc053ef404f3158b134.png"><img alt="../../../_images/dbe43e06fa616b9beb909d1955eba9ab8f83a4045a9d5fc053ef404f3158b134.png" src="../../../_images/dbe43e06fa616b9beb909d1955eba9ab8f83a4045a9d5fc053ef404f3158b134.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>ggsave(&#39;./figs/xQTL_list_PPI.pdf&#39;, width=10, height=10)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>which(tg[&#39;DYSF&#39;,]&gt;0)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.dl-inline {width: auto; margin:0; padding: 0}
.dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}
.dl-inline>dt::after {content: ":\0020"; padding-right: .5ex}
.dl-inline>dt:not(:first-of-type) {padding-left: .5ex}
</style><dl class=dl-inline><dt>STX4</dt><dd>45</dd><dt>BIN1</dt><dd>57</dd><dt>FKRP</dt><dd>92</dd><dt>EMG1</dt><dd>182</dd></dl>
</div></div>
</div>
</section>
</section>
<section id="xqtl-gvs-table2">
<h3>xQTL + GVS table2<a class="headerlink" href="#xqtl-gvs-table2" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Load necessary libraries
library(STRINGdb)
library(dplyr)
library(igraph)

# Initialize STRINGdb (change species if needed, 9606 = Homo sapiens)
string_db &lt;- STRINGdb$new(version=&quot;11.5&quot;, species=9606, score_threshold=400, input_directory=&quot;&quot;)

# Example gene list (use your own gene symbols or Entrez IDs)
genes &lt;- c(xQTL_list$gene_name, real_gvc_list$gene_name) %&gt;% unique 

# Map gene names to STRING IDs
mapped_genes &lt;- string_db$map(data.frame(gene=genes), &quot;gene&quot;, removeUnmappedRows = TRUE)

# Retrieve interaction network for the genes
ppi_network &lt;- string_db$get_interactions(mapped_genes$STRING_id)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning:  we couldn&#39;t map to STRING 2% of your identifiers
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>ppi_network_unique &lt;- ppi_network %&gt;%
  mutate(pair = paste0(pmin(from, to), &quot;_&quot;, pmax(from, to))) %&gt;%
  distinct(pair, .keep_all = TRUE)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>symbol_lookup &lt;- dplyr::select(mapped_genes, STRING_id, gene)
symbol_lookup &lt;- dplyr::distinct(symbol_lookup)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>options(repr.plot.width = 12, repr.plot.height = 12)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Build igraph object
g &lt;- graph_from_data_frame(ppi_network[, c(&quot;from&quot;, &quot;to&quot;)], directed=FALSE)
g &lt;- simplify(g, remove.multiple = TRUE, remove.loops = TRUE)
# Replace vertex names with gene symbols
V(g)$name &lt;- symbol_lookup$gene[match(V(g)$name, symbol_lookup$STRING_id)]
# Define genes to highlight
highlight_genes &lt;- real_gvc_list$gene_name

# Assign color: red for highlighted, gray for others
V(g)$color &lt;- ifelse(V(g)$name %in% highlight_genes, &quot;red&quot;, &quot;gray&quot;)

# Optional: make highlighted genes larger
V(g)$size &lt;- ifelse(V(g)$name %in% highlight_genes, 8, 4)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(tidygraph)
library(ggraph)
tg &lt;- as_tbl_graph(g) %&gt;% 
  mutate(highlight = ifelse(name %in% real_gvc_list$gene_name, &quot;Yes&quot;, &quot;No&quot;))
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>which(tg[&#39;CORO1A&#39;,] &gt; 0)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.dl-inline {width: auto; margin:0; padding: 0}
.dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}
.dl-inline>dt::after {content: ":\0020"; padding-right: .5ex}
.dl-inline>dt:not(:first-of-type) {padding-left: .5ex}
</style><dl class=dl-inline><dt>SPI1</dt><dd>15</dd><dt>TAOK2</dt><dd>62</dd><dt>DOC2A</dt><dd>83</dd><dt>PTK2B</dt><dd>96</dd><dt>TLCD3B</dt><dd>98</dd><dt>YPEL3</dt><dd>128</dd><dt>INO80E</dt><dd>129</dd><dt>STK4</dt><dd>155</dd><dt>TMEM219</dt><dd>191</dd></dl>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>ggraph(tg, layout = &quot;fr&quot;) +  # fr: Fruchterman-Reingold s
  geom_edge_link(color = &quot;grey80&quot;, alpha = 0.6) +
  geom_node_point(aes(color = highlight), size = 5) +
  geom_node_text(aes(label = name), repel = TRUE, size = 3) +
  scale_color_manual(values = c(&quot;Yes&quot; = &quot;red&quot;, &quot;No&quot; = &quot;skyblue&quot;)) +
  theme_void() +
  theme(legend.position = &quot;none&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
&quot;ggrepel: 26 unlabeled data points (too many overlaps). Consider increasing max.overlaps&quot;
</pre></div>
</div>
<a class="reference internal image-reference" href="../../../_images/95dd91b5e1c1ba43c11ecb6d3ba13e8591742df2e06d593626504703c937d35b.png"><img alt="../../../_images/95dd91b5e1c1ba43c11ecb6d3ba13e8591742df2e06d593626504703c937d35b.png" src="../../../_images/95dd91b5e1c1ba43c11ecb6d3ba13e8591742df2e06d593626504703c937d35b.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
</section>
<section id="check-gene-expression">
<h3>check gene expression<a class="headerlink" href="#check-gene-expression" title="Link to this heading">#</a></h3>
<p>we are wondering why CORO1A was picked by cTWAS, wondering if that gene has a higher expression correlation with others</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>exp &lt;- fread(&#39;/data/eQTL/ROSMAP/DLPFC/analysis_ready/phenotype_preprocessing/dlpfc_batch_all.rnaseqc.low_expression_filtered.outlier_removed.tmm.expression.bed.chr16.mol_phe.bed.gz&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>exp %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 x 1145</caption>
<thead>
	<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_id</th><th scope=col>SM-CTEFX</th><th scope=col>SM-CJIWX</th><th scope=col>03_120405</th><th scope=col>04_120405</th><th scope=col>SM-CJIXW</th><th scope=col>07_120410</th><th scope=col>...</th><th scope=col>RISK_97</th><th scope=col>RISK_99</th><th scope=col>SM-CTEE8</th><th scope=col>SM-2T6Z5</th><th scope=col>SM-2T717</th><th scope=col>SM-CJEHY</th><th scope=col>SM-AYO4G</th><th scope=col>SM-CJFL9</th><th scope=col>SM-AYOSS</th><th scope=col>SM-CJJ2G</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>...</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr16</td><td>18067</td><td>18068</td><td>ENSG00000234769</td><td> 0.03732274</td><td> 0.09452324</td><td> 0.7263884</td><td>-0.2870415</td><td>-1.24833355</td><td>-0.755269049</td><td>...</td><td>-1.1492866</td><td>-1.0189406</td><td>-0.33542703</td><td> 0.07029612</td><td> 0.3146008</td><td>-0.6539642</td><td>-0.02853817</td><td>-2.3403738</td><td>-2.41438445</td><td> 2.277320807</td></tr>
	<tr><td>chr16</td><td>22909</td><td>22910</td><td>ENSG00000231439</td><td> 0.24383900</td><td> 0.39171142</td><td> 0.1564774</td><td> 1.1450483</td><td> 0.01536523</td><td> 0.587425922</td><td>...</td><td>-1.0115907</td><td>-0.9072527</td><td> 0.19437225</td><td>-1.02263626</td><td>-0.6925110</td><td>-0.7523529</td><td>-0.93068987</td><td>-1.9607137</td><td> 0.01756047</td><td> 0.160922913</td></tr>
	<tr><td>chr16</td><td>35194</td><td>35195</td><td>ENSG00000260803</td><td>-0.91723547</td><td>-0.21454434</td><td>-0.3587397</td><td> 0.1631469</td><td>-1.16212698</td><td>-1.794838030</td><td>...</td><td>-1.2201268</td><td> 1.6897891</td><td>-0.76994756</td><td>-1.79483803</td><td>-0.1077605</td><td>-1.7948380</td><td>-0.62701716</td><td>-1.7948380</td><td>-1.79483803</td><td> 1.124158694</td></tr>
	<tr><td>chr16</td><td>38320</td><td>38321</td><td>ENSG00000226942</td><td> 0.31229530</td><td> 0.89081249</td><td> 1.1037484</td><td> 0.6925110</td><td>-1.19285876</td><td> 0.947748620</td><td>...</td><td>-1.1928588</td><td> 0.2574290</td><td> 0.07910046</td><td>-1.19285876</td><td> 0.6621409</td><td> 0.5666850</td><td>-1.19285876</td><td>-1.1928588</td><td> 0.27790431</td><td>-0.006584885</td></tr>
	<tr><td>chr16</td><td>53607</td><td>53608</td><td>ENSG00000161980</td><td>-0.16537162</td><td> 1.36698910</td><td> 0.8940812</td><td> 1.2063807</td><td> 0.05709961</td><td>-0.006584885</td><td>...</td><td>-0.9650877</td><td>-1.8347016</td><td>-1.28756214</td><td>-1.01525878</td><td>-1.0681251</td><td>-2.4569939</td><td>-0.14981486</td><td>-1.6126061</td><td>-0.47859867</td><td> 0.940892269</td></tr>
	<tr><td>chr16</td><td>53827</td><td>53828</td><td>ENSG00000161981</td><td>-0.04611020</td><td> 1.44356404</td><td> 1.2875621</td><td> 0.7263884</td><td> 0.33078670</td><td> 0.528434354</td><td>...</td><td>-0.8681925</td><td>-1.0300698</td><td>-1.06039107</td><td>-2.45699387</td><td>-1.6126061</td><td>-2.3403738</td><td>-0.53602111</td><td>-0.9546498</td><td>-0.44926579</td><td> 0.203327376</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multi_gene_list &lt;- c(&#39;YPEL3&#39;, &#39;INO80E&#39;, &#39;DOC2A&#39;, &#39;MAPK3&#39;, &#39;CORO1A&#39;, &#39;BCKDK&#39;, &#39;TLCD3B&#39;, &#39;ZNF267&#39;, &#39;AC093512.2&#39;, &#39;PRSS8&#39;, &#39;TAOK2&#39;, &#39;TMEM219&#39;)
library(clusterProfiler)
library(org.Hs.eg.db)
ensembl_ids &lt;- bitr(multi_gene_list, fromType = &quot;SYMBOL&quot;, 
                   toType = &quot;ENSEMBL&quot;, OrgDb = org.Hs.eg.db)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:1 mapping between keys and columns

Warning message in bitr(multi_gene_list, fromType = &quot;SYMBOL&quot;, toType = &quot;ENSEMBL&quot;, :
&quot;8.33% of input gene IDs are fail to map...&quot;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multigene_exp &lt;- exp %&gt;% filter(gene_id %in% ensembl_ids$ENSEMBL) %&gt;% dplyr::select(-c(1:3)) %&gt;% t
multigene_exp %&gt;% dim
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>1142</li><li>8</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>colnames(multigene_exp) &lt;- multigene_exp[1,]
multigene_exp &lt;- multigene_exp[-1,]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>colnames(multigene_exp) &lt;- ensembl_ids$SYMBOL[match(colnames(multigene_exp), ensembl_ids$ENSEMBL)]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multigene_exp %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A matrix: 6 x 8 of type chr</caption>
<thead>
	<tr><th></th><th scope=col>INO80E</th><th scope=col>DOC2A</th><th scope=col>TLCD3B</th><th scope=col>YPEL3</th><th scope=col>MAPK3</th><th scope=col>CORO1A</th><th scope=col>BCKDK</th><th scope=col>ZNF267</th></tr>
</thead>
<tbody>
	<tr><th scope=row>SM-CTEFX</th><td> 1.1535456 </td><td> 0.1832001 </td><td> 0.2235373 </td><td> 0.3917114 </td><td> 0.2122988 </td><td> 1.8466451 </td><td>-0.5822173 </td><td>-1.3782501 </td></tr>
	<tr><th scope=row>SM-CJIWX</th><td> 1.43123012</td><td> 1.84664506</td><td> 1.44981422</td><td> 1.20184892</td><td> 0.49341939</td><td> 1.68070816</td><td> 0.93748063</td><td>-0.07249667</td></tr>
	<tr><th scope=row>03_120405</th><td> 1.9386875 </td><td> 1.2531321 </td><td> 2.2489074 </td><td> 2.9197941 </td><td> 0.7093479 </td><td> 0.7758656 </td><td>-0.6350527 </td><td>-0.5770244 </td></tr>
	<tr><th scope=row>04_120405</th><td>0.3377499  </td><td>1.5367961  </td><td>0.9443149  </td><td>0.3775306  </td><td>0.2779043  </td><td>1.4312301  </td><td>0.7523529  </td><td>0.1876662  </td></tr>
	<tr><th scope=row>SM-CJIXW</th><td> 2.62141469</td><td> 0.74075149</td><td> 0.54363885</td><td> 1.17955152</td><td> 0.35406193</td><td> 0.03512636</td><td>-0.30768933</td><td>-2.41438445</td></tr>
	<tr><th scope=row>07_120410</th><td> 2.41438445</td><td> 0.75526905</td><td> 1.11185698</td><td> 2.79103738</td><td> 1.52968559</td><td> 0.07469757</td><td>-0.45169514</td><td>-1.19734174</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>cor(multigene_exp[,&#39;YPEL3&#39;] %&gt;% as.numeric, multigene_exp[,&#39;INO80E&#39;] %&gt;% as.numeric) ## YPEL3 &amp; INO80E
cor(multigene_exp[,&#39;YPEL3&#39;] %&gt;% as.numeric, multigene_exp[,&#39;DOC2A&#39;] %&gt;% as.numeric)## YPEL3 &amp; DOC2A
cor(multigene_exp[,&#39;YPEL3&#39;] %&gt;% as.numeric, multigene_exp[,&#39;CORO1A&#39;] %&gt;% as.numeric)## YPEL3 &amp; CORO1A
cor(multigene_exp[,&#39;INO80E&#39;] %&gt;% as.numeric, multigene_exp[,&#39;CORO1A&#39;] %&gt;% as.numeric)## SPI1 &amp; CORO1A
cor(multigene_exp[,&#39;DOC2A&#39;] %&gt;% as.numeric, multigene_exp[,&#39;CORO1A&#39;] %&gt;% as.numeric)## 
cor(multigene_exp[,&#39;DOC2A&#39;] %&gt;% as.numeric, multigene_exp[,&#39;INO80E&#39;] %&gt;% as.numeric)## 
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">0.81142088770498</div><div class="output text_html">0.813293608488586</div><div class="output text_html">0.712913901774005</div><div class="output text_html">0.700308459119743</div><div class="output text_html">0.851877392881107</div><div class="output text_html">0.702018207860162</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multigene_exp[,c(&#39;YPEL3&#39;, &#39;DOC2A&#39;, &#39;INO80E&#39;)] %&gt;% apply(., 2, as.numeric) %&gt;% cor
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A matrix: 3 x 3 of type dbl</caption>
<thead>
	<tr><th></th><th scope=col>YPEL3</th><th scope=col>DOC2A</th><th scope=col>INO80E</th></tr>
</thead>
<tbody>
	<tr><th scope=row>YPEL3</th><td>1.0000000</td><td>0.8132936</td><td>0.8114209</td></tr>
	<tr><th scope=row>DOC2A</th><td>0.8132936</td><td>1.0000000</td><td>0.7020182</td></tr>
	<tr><th scope=row>INO80E</th><td>0.8114209</td><td>0.7020182</td><td>1.0000000</td></tr>
</tbody>
</table>
</div></div>
</div>
<p>the conclusion is CORO1A not really showing a higher correlation than others</p>
</section>
</section>
<section id="prediction">
<h2>Prediction<a class="headerlink" href="#prediction" title="Link to this heading">#</a></h2>
<p>Use gene set expression, show their prediction to disease</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(pROC)
library(ggplot2)
library(caret)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>exp_raw &lt;- fread(&#39;/data/eQTL/ROSMAP/DLPFC/analysis_ready/phenotype_preprocessing/DLPFC_samples_list.rnaseqc.gene_tpm.low_expression_filtered.outlier_removed.tmm.expression.remove_duplicates.bed.gz&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>exp_raw %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 x 1097</caption>
<thead>
	<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_id</th><th scope=col>SM-CTDS5</th><th scope=col>SM-CJIXK</th><th scope=col>SM-CJGH9</th><th scope=col>SM-CTDS7</th><th scope=col>SM-CJGLT</th><th scope=col>SM-CTDT7</th><th scope=col>...</th><th scope=col>RISK_442</th><th scope=col>RISK_45_rerun</th><th scope=col>SM-CTEGK</th><th scope=col>RISK_51_rerun</th><th scope=col>RISK_67_rerun</th><th scope=col>RISK_73_redo</th><th scope=col>RISK_81</th><th scope=col>SM-CTEE8</th><th scope=col>RISK_93</th><th scope=col>RISK_97</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>...</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr1</td><td> 29569</td><td> 29570</td><td>ENSG00000227232</td><td> 0.86713223</td><td>0.4357151</td><td>-0.3668001</td><td> 0.006811543</td><td> 0.2688521</td><td> 1.79619337</td><td>...</td><td>-0.2383699</td><td>-1.377110</td><td>-0.5275293</td><td>-0.4862194</td><td>-1.43189486</td><td>-1.5490204</td><td>-2.685304</td><td>-1.5959174</td><td>-2.6098584</td><td>-0.9006891</td></tr>
	<tr><td>chr1</td><td> 52472</td><td> 52473</td><td>ENSG00000268020</td><td>-0.07955121</td><td>1.6378678</td><td> 0.2688521</td><td>-0.182650237</td><td> 0.2081076</td><td>-0.02270692</td><td>...</td><td>-1.1703806</td><td>-1.170381</td><td>-1.1703806</td><td>-1.1703806</td><td>-1.17038063</td><td> 0.7120250</td><td>-1.170381</td><td>-0.4307273</td><td>-1.1703806</td><td>-0.4034813</td></tr>
	<tr><td>chr1</td><td> 57597</td><td> 57598</td><td>ENSG00000240361</td><td>-0.08410743</td><td>1.5720367</td><td> 0.9143990</td><td>-0.027249331</td><td> 0.8905169</td><td>-0.11376979</td><td>...</td><td>-1.2334946</td><td>-1.233495</td><td>-0.5458851</td><td>-1.2334946</td><td>-0.62127258</td><td>-1.2334946</td><td>-1.233495</td><td>-1.2334946</td><td>-1.2334946</td><td>-0.5912576</td></tr>
	<tr><td>chr1</td><td> 65418</td><td> 65419</td><td>ENSG00000186092</td><td>-0.39610308</td><td>1.5798971</td><td> 0.9638026</td><td>-0.271206929</td><td> 0.9352923</td><td>-0.76268350</td><td>...</td><td>-1.1243382</td><td>-1.875759</td><td>-0.6240288</td><td>-0.4257502</td><td>-0.93529230</td><td> 1.3654825</td><td>-1.172635</td><td>-1.2888111</td><td>-1.8757588</td><td>-1.8757588</td></tr>
	<tr><td>chr1</td><td>133722</td><td>133723</td><td>ENSG00000238009</td><td> 0.10691669</td><td>0.8804360</td><td> 0.6916925</td><td> 0.217397537</td><td>-0.2150733</td><td> 0.77795727</td><td>...</td><td>-0.3789700</td><td>-1.757666</td><td> 0.4836656</td><td>-1.7576660</td><td>-0.03633594</td><td> 0.4862194</td><td>-1.757666</td><td> 0.9783555</td><td>-0.5804752</td><td> 0.3474399</td></tr>
	<tr><td>chr1</td><td>135894</td><td>135895</td><td>ENSG00000268903</td><td> 2.78006615</td><td>1.8077066</td><td> 0.4658759</td><td> 0.897288076</td><td> 1.2190477</td><td> 0.36922970</td><td>...</td><td>-1.3626043</td><td>-1.362604</td><td>-1.3626043</td><td>-1.3626043</td><td>-1.36260432</td><td>-1.3626043</td><td>-1.362604</td><td>-1.3626043</td><td>-1.3626043</td><td>-1.3626043</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>exp &lt;- exp_raw %&gt;% select(-c(1:3)) %&gt;% t()
colnames(exp) &lt;- exp[1,]
exp &lt;- exp[-1,]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_list %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 x 7</caption>
<thead>
	<tr><th scope=col>gene_id</th><th scope=col>resource_in_FunGen_xQTL</th><th scope=col>data_resources</th><th scope=col>chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>ENSG00000002919</td><td>ColocBoost</td><td>FunGen_xQTL</td><td>17</td><td> 48103356</td><td> 48103357</td><td>SNX11 </td></tr>
	<tr><td>ENSG00000005100</td><td>ColocBoost</td><td>FunGen_xQTL</td><td>17</td><td>  5468981</td><td>  5468982</td><td>DHX33 </td></tr>
	<tr><td>ENSG00000006530</td><td>ColocBoost</td><td>FunGen_xQTL</td><td> 7</td><td>141551277</td><td>141551278</td><td>AGK   </td></tr>
	<tr><td>ENSG00000008294</td><td>ColocBoost</td><td>FunGen_xQTL</td><td>17</td><td> 51120867</td><td> 51120868</td><td>SPAG9 </td></tr>
	<tr><td>ENSG00000013503</td><td>ColocBoost</td><td>FunGen_xQTL</td><td>12</td><td>106357747</td><td>106357748</td><td>POLR3B</td></tr>
	<tr><td>ENSG00000029725</td><td>ColocBoost</td><td>FunGen_xQTL</td><td>17</td><td>  5282264</td><td>  5282265</td><td>RABEP1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>meta &lt;- fread(&#39;ROSMAP_clinical_metadata.tsv&#39;)
exp_meta &lt;- meta %&gt;% filter(specimenID %in% rownames(exp)) %&gt;% select(specimenID, group)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>exp_meta &lt;- exp_meta %&gt;% arrange(match(specimenID, colnames(exp_raw))) 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gvc_exp &lt;- exp_raw %&gt;% filter(gene_id %in% real_gvc_list$gene_id) 
rownames(gvc_exp) &lt;- gvc_exp$gene_id
gvc_exp &lt;- gvc_exp %&gt;% select(any_of(exp_meta$specimenID)) %&gt;% t
</pre></div>
</div>
</div>
</div>
<section id="model-with-glmnet">
<h3>model with glmnet<a class="headerlink" href="#model-with-glmnet" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>
library(dplyr)
library(stringr)
library(caret)
library(pROC)
library(ggplot2)
library(glmnet)

# Function to process xQTL expression data
process_xQTL_exp &lt;- function(exp_raw, gene_ids, exp_meta) {
  exp_data &lt;- exp_raw %&gt;% 
    filter(gene_id %in% gene_ids) %&gt;% 
    column_to_rownames(&quot;gene_id&quot;) %&gt;% 
    select(any_of(exp_meta$specimenID)) %&gt;% 
    t()
  return(exp_data)
}

# Calculate n_methods and create shared gene sets
xQTL_list &lt;- xQTL_list %&gt;% 
  mutate(n_methods = str_count(resource_in_FunGen_xQTL, &#39;,&#39;) + 1)

# Store gene sets for different n_methods thresholds
gene_sets &lt;- list(
  &quot;Gene Set 1&quot; = list(genes = gvc_exp, is_matrix = TRUE),
  &quot;Gene Set 2&quot; = list(genes = xQTL_list$gene_id, is_matrix = FALSE),
  &quot;Gene Set 3&quot; = list(genes = xQTL_list %&gt;% filter(n_methods &gt; 1) %&gt;% pull(gene_id), is_matrix = FALSE),
  &quot;Gene Set 4&quot; = list(genes = xQTL_list %&gt;% filter(n_methods &gt; 2) %&gt;% pull(gene_id), is_matrix = FALSE),
  &quot;Gene Set 5&quot; = list(genes = xQTL_list %&gt;% filter(n_methods &gt; 3) %&gt;% pull(gene_id), is_matrix = FALSE),
  &quot;Gene Set 6&quot; = list(genes = xQTL_list %&gt;% filter(n_methods &gt; 4) %&gt;% pull(gene_id), is_matrix = FALSE)
)

# Prepare labels
labels &lt;- exp_meta %&gt;% 
  mutate(labels = ifelse(group == &#39;ad&#39;, 1, 0)) %&gt;% 
  pull(labels)

# Process datasets and remove near-zero variance predictors
data_list &lt;- lapply(names(gene_sets), function(set_name) {
  if (gene_sets[[set_name]]$is_matrix) {
    data &lt;- data.frame(gvc_exp, Disease = as.factor(labels))
  } else {
    exp_data &lt;- process_xQTL_exp(exp_raw, gene_sets[[set_name]]$genes, exp_meta)
    data &lt;- data.frame(exp_data, Disease = as.factor(labels))
  }
  # Remove near-zero variance predictors
  nzv &lt;- nearZeroVar(data[, -ncol(data)], saveMetrics = TRUE)
  data &lt;- data[, !nzv$zeroVar &amp; !nzv$nzv]
  return(data)
})
names(data_list) &lt;- names(gene_sets)

# Split data into train/test
set.seed(123)
trainIndex &lt;- createDataPartition(labels, p = 0.7, list = FALSE)
train_data &lt;- lapply(data_list, function(data) data[trainIndex, ])
test_data &lt;- lapply(data_list, function(data) data[-trainIndex, ])

# Train models with glmnet (L1 regularization) to handle high-dimensional data
models &lt;- lapply(train_data, function(data) {
  x &lt;- as.matrix(data[, -ncol(data)])
  y &lt;- as.factor(data$Disease)
  # Use cross-validation to select lambda
  cv_model &lt;- cv.glmnet(x, y, family = &quot;binomial&quot;, alpha = 1, maxit = 1e6)
  return(cv_model)
})

# Make predictions
predictions &lt;- mapply(function(model, test) {
  x_test &lt;- as.matrix(test[, -ncol(test)])
  predict(model, newx = x_test, s = &quot;lambda.min&quot;, type = &quot;response&quot;)[, 1]
}, models, test_data, SIMPLIFY = FALSE)

# Calculate ROC and AUC
roc_results &lt;- mapply(function(pred, test) {
  roc(test$Disease, pred, levels = c(0, 1), direction = &quot;&lt;&quot;)
}, predictions, test_data, SIMPLIFY = FALSE)

auc_values &lt;- lapply(roc_results, auc)

# Prepare ROC data for plotting
roc_data &lt;- do.call(rbind, mapply(function(roc, set_name) {
  data.frame(
    Specificity = 1 - roc$specificities,
    Sensitivity = roc$sensitivities,
    Set = set_name
  )
}, roc_results, names(gene_sets), SIMPLIFY = FALSE))

# Plot ROC curves
ggplot(roc_data, aes(x = Specificity, y = Sensitivity, color = Set)) +
  geom_line() +
  geom_abline(slope = 1, intercept = 0, linetype = &quot;dashed&quot;, color = &quot;gray&quot;) +
  labs(title = paste(&quot;ROC Curves (AUC: &quot;, 
                     paste(names(auc_values), &quot;=&quot;, round(unlist(auc_values), 3), collapse = &quot;, &quot;), &quot;)&quot;, sep = &quot;&quot;),
       x = &quot;1 - Specificity&quot;, y = &quot;Sensitivity&quot;) +
  scale_color_manual(values = c(&quot;Gene Set 1&quot; = &quot;blue&quot;, &quot;Gene Set 2&quot; = &quot;red&quot;, 
                                &quot;Gene Set 3&quot; = &quot;pink&quot;, &quot;Gene Set 4&quot; = &quot;orange&quot;, 
                                &quot;Gene Set 5&quot; = &quot;yellow&quot;, &quot;Gene Set 6&quot; = &quot;black&quot;)) +
  theme_minimal()
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Loading required package: Matrix


Attaching package: &#39;Matrix&#39;


The following object is masked from &#39;package:S4Vectors&#39;:

    expand


The following objects are masked from &#39;package:tidyr&#39;:

    expand, pack, unpack


Loaded glmnet 4.1-8
</pre></div>
</div>
<a class="reference internal image-reference" href="../../../_images/589dc6e946137dde5124bfb8edab78543178113ffb1a9469222b8530a7a41e4d.png"><img alt="../../../_images/589dc6e946137dde5124bfb8edab78543178113ffb1a9469222b8530a7a41e4d.png" src="../../../_images/589dc6e946137dde5124bfb8edab78543178113ffb1a9469222b8530a7a41e4d.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
</section>
<section id="model-with-glm">
<h3>model with glm<a class="headerlink" href="#model-with-glm" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>real_gvc_list_small &lt;- intersect(xQTL_list$gene_id, real_gvc_list$gene_id)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xQTL_list %&gt;% filter(n_methods &gt; 1) %&gt;% pull(gene_id) %&gt;% unique %&gt;% length
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">59</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(dplyr)
library(pROC)
library(ggplot2)
library(caret)
library(stringr)
set.seed(123456)
# Calculate number of methods for each gene
xQTL_list &lt;- xQTL_list %&gt;%
  mutate(n_methods = str_count(resource_in_FunGen_xQTL, &quot;,&quot;) + 1)

# Create filtered lists based on number of methods
xQTL_list_shared &lt;- lapply(2:5, function(n) {
  xQTL_list %&gt;% filter(n_methods &gt;= n)
})
names(xQTL_list_shared) &lt;- paste0(&quot;shared_&quot;, 2:5)

# Function to get transposed expression matrix filtered by gene list
get_exp_matrix &lt;- function(exp, gene_ids) {
  tmp &lt;- exp_raw %&gt;% filter(gene_id %in% gene_ids)
  rownames(tmp) &lt;- tmp$gene_id
  t(tmp %&gt;% select(any_of(exp_meta$specimenID)))
}

# Prepare expression matrices
gvc_small &lt;- get_exp_matrix(gvc_exp, real_gvc_list_small)
xQTL_exp_all &lt;- get_exp_matrix(exp_raw, xQTL_list$gene_id)
xQTL_exp_shared2 &lt;- get_exp_matrix(exp_raw, xQTL_list_shared$shared_2$gene_id)
xQTL_exp_shared3 &lt;- get_exp_matrix(exp_raw, xQTL_list_shared$shared_3$gene_id)
xQTL_exp_shared4 &lt;- get_exp_matrix(exp_raw, xQTL_list_shared$shared_4$gene_id)
xQTL_exp_shared5 &lt;- get_exp_matrix(exp_raw, xQTL_list_shared$shared_5$gene_id)
twas_xQTL_exp_all &lt;- get_exp_matrix(exp_raw,twas_xqtl_list)
# Prepare labels
labels &lt;- exp_meta %&gt;% mutate(labels = ifelse(group == &quot;ad&quot;, 1, 0)) %&gt;% pull(labels)

# Create data frames with expression and labels
datasets &lt;- list(
  data1 = data.frame(gvc_exp, Disease = factor(labels)),
  data2 = data.frame(xQTL_exp_all, Disease = factor(labels)),
  data3 = data.frame(xQTL_exp_shared2, Disease = factor(labels)),
  data4 = data.frame(xQTL_exp_shared3, Disease = factor(labels)),
  data5 = data.frame(xQTL_exp_shared4, Disease = factor(labels)),
  data6 = data.frame(xQTL_exp_shared5, Disease = factor(labels)),
  data7 = data.frame(gvc_small, Disease = factor(labels)),
  data8 = data.frame(twas_xQTL_exp_all, Disease = factor(labels))
)

# Split train/test indices
trainIndex &lt;- createDataPartition(labels, p = 0.7, list = FALSE)

# Split datasets into train/test
train_test &lt;- lapply(datasets, function(df) {
  list(train = df[trainIndex, ], test = df[-trainIndex, ])
})

# Fit logistic regression models
models &lt;- lapply(train_test, function(data) {
  glm(Disease ~ ., data = data$train, family = binomial)
})

# Predict probabilities on test sets
predictions &lt;- mapply(function(model, data) {
  predict(model, data$test, type = &quot;response&quot;)
}, models, train_test, SIMPLIFY = FALSE)

# Calculate ROC and AUC
rocs &lt;- mapply(function(pred, data) {
  roc(data$test$Disease, pred)
}, predictions, train_test, SIMPLIFY = FALSE)

aucs &lt;- sapply(rocs, auc)

# Prepare ROC data for plotting
roc_data &lt;- do.call(rbind, lapply(seq_along(rocs), function(i) {
  data.frame(
    Specificity = 1 - rocs[[i]]$specificities,
    Sensitivity = rocs[[i]]$sensitivities,
    Set = paste0(&quot;Gene Set &quot;, i)
  )
}))

roc_data &lt;- roc_data %&gt;%
  mutate(Set = case_when(
    Set == &quot;Gene Set 1&quot; ~ &quot;GVC&quot;,
    Set == &quot;Gene Set 2&quot; ~ &quot;all xQTL genes&quot;,
    Set == &quot;Gene Set 3&quot; ~ &quot;xQTL genes with 2 methods&quot;,
    Set == &quot;Gene Set 4&quot; ~ &quot;xQTL genes with 3 methods&quot;,
    Set == &quot;Gene Set 5&quot; ~ &quot;xQTL genes with 4 methods&quot;,
    Set == &quot;Gene Set 6&quot; ~ &quot;xQTL genes with 5 methods&quot;,
    Set == &quot;Gene Set 7&quot; ~ &quot;GVC small&quot;,
    Set == &quot;Gene Set 8&quot; ~ &quot;xQTL and TWAS&quot;,
    TRUE ~ Set
  ))

ggplot(roc_data, aes(x = Specificity, y = Sensitivity, color = Set)) +
  geom_line(size = 1) +
  geom_abline(slope = 1, intercept = 0, linetype = &quot;dashed&quot;, color = &quot;gray&quot;) +
  labs(
    title = paste0(
      &quot;ROC Curves (AUC: \n&quot;,
      paste0(c(&quot;GVC&quot;, &quot;all xQTL genes&quot;, &quot;xQTL genes with 2 methods&quot;,
               &quot;xQTL genes with 3 methods&quot;, &quot;xQTL genes with 4 methods&quot;,
               &quot;xQTL genes with 5 methods&quot;, &quot;GVC small&quot;, &quot;xQTL and TWAS&quot;),
             &quot; = &quot;, round(aucs, 3), collapse = &quot;, \n&quot;),
      &quot;)&quot;
    ),
    x = &quot;1 - Specificity&quot;,
    y = &quot;Sensitivity&quot;
  ) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
&quot;glm.fit: fitted probabilities numerically 0 or 1 occurred&quot;
Setting levels: control = 0, case = 1

Setting direction: controls &lt; cases

Setting levels: control = 0, case = 1

Setting direction: controls &lt; cases

Setting levels: control = 0, case = 1

Setting direction: controls &lt; cases

Setting levels: control = 0, case = 1

Setting direction: controls &lt; cases

Setting levels: control = 0, case = 1

Setting direction: controls &lt; cases

Setting levels: control = 0, case = 1

Setting direction: controls &lt; cases

Setting levels: control = 0, case = 1

Setting direction: controls &lt; cases

Setting levels: control = 0, case = 1

Setting direction: controls &lt; cases
</pre></div>
</div>
<a class="reference internal image-reference" href="../../../_images/0073452d321e57431895c0fd9bc128195530af968e530d3f3cc3a116a1f62dfc.png"><img alt="../../../_images/0073452d321e57431895c0fd9bc128195530af968e530d3f3cc3a116a1f62dfc.png" src="../../../_images/0073452d321e57431895c0fd9bc128195530af968e530d3f3cc3a116a1f62dfc.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>ggsave(&#39;./figs/xQTL_list_AUC_glm.pdf&#39;, width=8, height=8)
</pre></div>
</div>
</div>
</div>
<section id="cross-sample">
<h4>cross sample<a class="headerlink" href="#cross-sample" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># double check to make the outcome as the factor, so to preserve the class distribution of disease 
str(datasets[[1]]$Disease)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span> Factor w/ 2 levels &quot;0&quot;,&quot;1&quot;: 2 1 2 2 1 1 2 1 2 1 ...
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>                
# Replace train/test split and model training section with cross-validation
ctrl &lt;- trainControl(method = &quot;cv&quot;, number = 5, classProbs = TRUE, summaryFunction = twoClassSummary)

# Fit models with cross-validation
models_cv &lt;- lapply(datasets, function(df) {
  df$Disease &lt;- factor(df$Disease, levels = c(0, 1), labels = c(&quot;Control&quot;, &quot;AD&quot;))  # caret needs factor with levels
  train(Disease ~ ., data = df, method = &quot;glm&quot;, family = &quot;binomial&quot;,
        trControl = ctrl, metric = &quot;ROC&quot;)
})

# Extract AUC from cross-validation
aucs &lt;- sapply(models_cv, function(model) max(model$results$ROC))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning message:
&quot;glm.fit: fitted probabilities numerically 0 or 1 occurred&quot;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Prepare ROC data for plotting
roc_data &lt;- do.call(rbind, lapply(seq_along(rocs), function(i) {
  data.frame(
    Specificity = 1 - rocs[[i]]$specificities,
    Sensitivity = rocs[[i]]$sensitivities,
    Set = paste0(&quot;Gene Set &quot;, i)
  )
}))

roc_data &lt;- roc_data %&gt;%
  mutate(Set = case_when(
    Set == &quot;Gene Set 1&quot; ~ &quot;GVC&quot;,
    Set == &quot;Gene Set 2&quot; ~ &quot;all xQTL genes&quot;,
    Set == &quot;Gene Set 3&quot; ~ &quot;xQTL genes with 2 methods&quot;,
    Set == &quot;Gene Set 4&quot; ~ &quot;xQTL genes with 3 methods&quot;,
    Set == &quot;Gene Set 5&quot; ~ &quot;xQTL genes with 4 methods&quot;,
    Set == &quot;Gene Set 6&quot; ~ &quot;xQTL genes with 5 methods&quot;,
    Set == &quot;Gene Set 7&quot; ~ &quot;GVC small&quot;,
    Set == &quot;Gene Set 8&quot; ~ &quot;xQTL and TWAS&quot;,
    TRUE ~ Set
  ))

ggplot(roc_data, aes(x = Specificity, y = Sensitivity, color = Set)) +
  geom_line(size = 1) +
  geom_abline(slope = 1, intercept = 0, linetype = &quot;dashed&quot;, color = &quot;gray&quot;) +
  labs(
    title = paste0(
      &quot;ROC Curves (AUC: \n&quot;,
      paste0(c(&quot;GVC&quot;, &quot;all xQTL genes&quot;, &quot;xQTL genes with 2 methods&quot;,
               &quot;xQTL genes with 3 methods&quot;, &quot;xQTL genes with 4 methods&quot;,
               &quot;xQTL genes with 5 methods&quot;, &quot;GVC small&quot;, &quot;xQTL and TWAS&quot;),
             &quot; = &quot;, round(aucs, 3), collapse = &quot;, \n&quot;),
      &quot;)&quot;
    ),
    x = &quot;1 - Specificity&quot;,
    y = &quot;Sensitivity&quot;
  ) +
  theme_minimal() +
  theme(plot.title = element_text(hjust = 0.5))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/58768ae366bd391cd0e1511c2a250d54eac3eb5070d9f58911d16358c099c5ea.png"><img alt="../../../_images/58768ae366bd391cd0e1511c2a250d54eac3eb5070d9f58911d16358c099c5ea.png" src="../../../_images/58768ae366bd391cd0e1511c2a250d54eac3eb5070d9f58911d16358c099c5ea.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
</section>
</section>
</section>
<section id="grn">
<h2>GRN<a class="headerlink" href="#grn" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Load required packages
library(igraph)
library(dplyr)
library(readr)
reg_net_files &lt;- list.files(&#39;/data/resource/Functional_annotation/GRN/&#39;, pattern = &#39;txt$&#39;, full.names = T)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Attaching package: &#39;igraph&#39;


The following objects are masked from &#39;package:lubridate&#39;:

    %--%, union


The following objects are masked from &#39;package:dplyr&#39;:

    as_data_frame, groups, union


The following objects are masked from &#39;package:purrr&#39;:

    compose, simplify


The following object is masked from &#39;package:tidyr&#39;:

    crossing


The following object is masked from &#39;package:tibble&#39;:

    as_data_frame


The following object is masked from &#39;package:IRanges&#39;:

    union


The following object is masked from &#39;package:S4Vectors&#39;:

    union


The following objects are masked from &#39;package:BiocGenerics&#39;:

    normalize, path, union


The following object is masked from &#39;package:clusterProfiler&#39;:

    simplify


The following objects are masked from &#39;package:stats&#39;:

    decompose, spectrum


The following object is masked from &#39;package:base&#39;:

    union
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xqtl_roles_df &lt;- data.frame()
    xqtl_genes &lt;- xQTL_list$gene_name

for(i in 1:length(reg_net_files)){
    reg_net &lt;- read_tsv(reg_net_files[i])
    g &lt;- graph_from_data_frame(reg_net, directed = TRUE)
       
    xqtl_roles &lt;- data.frame(
      gene = xqtl_genes,
      is_in_graph = xqtl_genes %in% V(g)$name,
      out_degree = sapply(xqtl_genes, function(gene) if (gene %in% V(g)$name) degree(g, v = gene, mode = &quot;out&quot;) else NA),
      in_degree = sapply(xqtl_genes, function(gene) if (gene %in% V(g)$name) degree(g, v = gene, mode = &quot;in&quot;) else NA)
    )
    
    xqtl_roles &lt;- xqtl_roles %&gt;%
      mutate(
        role = case_when(
          is.na(in_degree) ~ &quot;not_in_network&quot;,
          out_degree &gt; 0 &amp; in_degree == 0 ~ &quot;regulator&quot;,
          in_degree &gt; 0 &amp; out_degree == 0 ~ &quot;target&quot;,
          out_degree &gt; 0 &amp; in_degree &gt; 0 ~ &quot;intermediary&quot;,
          TRUE ~ &quot;unclassified&quot;),
        file = basename(reg_net_files[i] %&gt;% gsub(&#39;.txt&#39;,&#39;&#39;,.))
        
      )
    xqtl_roles_df &lt;- rbind(xqtl_roles_df, xqtl_roles)
}
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span><span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">119370</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">111592</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">147186</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">365162</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">1121325</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">1118973</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">1120860</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">1123715</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">1119682</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">1123627</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">1125598</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">1123519</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">1124340</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">115917</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">114227</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">363591</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">75432</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">386902</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">115493</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">114768</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">114458</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">115461</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">146107</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">114087</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">9</span>
<span class=" -Color -Color-Cyan">--</span> <span class=" -Color -Color-Bold">Column specification</span> <span class=" -Color -Color-Cyan">--------------------------------------------------------</span>
<span class=" -Color -Color-Bold">Delimiter:</span> &quot;\t&quot;
<span class=" -Color -Color-Red">chr</span> (7): TF, enhancer, promoter, TG, method, celltype, Regulation
<span class=" -Color -Color-Green">dbl</span> (2): edgeWeight, Correlation

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
Warning message in graph_from_data_frame(reg_net, directed = TRUE):
&quot;In `d&#39; `NA&#39; elements were replaced with string &quot;NA&quot;&quot;
<span class=" -Color -Color-Bold">Rows: </span><span class=" -Color -Color-Blue">0</span> <span class=" -Color -Color-Bold">Columns: </span><span class=" -Color -Color-Blue">0</span>

<span class=" -Color -Color-Cyan">i</span> Use `spec()` to retrieve the full column specification for this data.
<span class=" -Color -Color-Cyan">i</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.
</pre></div>
</div>
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="n">Error</span> <span class="ow">in</span> <span class="n">graph_from_data_frame</span><span class="p">(</span><span class="n">reg_net</span><span class="p">,</span> <span class="n">directed</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">):</span> <span class="n">the</span> <span class="n">data</span> <span class="n">frame</span> <span class="n">should</span> <span class="n">contain</span> <span class="n">at</span> <span class="n">least</span> <span class="n">two</span> <span class="n">columns</span>
<span class="ne">Traceback</span>:

<span class="mi">1</span><span class="o">.</span> <span class="n">stop</span><span class="p">(</span><span class="s2">&quot;the data frame should contain at least two columns&quot;</span><span class="p">)</span>
<span class="mi">2</span><span class="o">.</span> <span class="o">.</span><span class="n">handleSimpleError</span><span class="p">(</span><span class="n">function</span> <span class="p">(</span><span class="n">cnd</span><span class="p">)</span> 
 <span class="o">.</span> <span class="p">{</span>
 <span class="o">.</span>     <span class="n">watcher</span><span class="err">$</span><span class="n">capture_plot_and_output</span><span class="p">()</span>
 <span class="o">.</span>     <span class="n">cnd</span> <span class="o">&lt;-</span> <span class="n">sanitize_call</span><span class="p">(</span><span class="n">cnd</span><span class="p">)</span>
 <span class="o">.</span>     <span class="n">watcher</span><span class="err">$</span><span class="n">push</span><span class="p">(</span><span class="n">cnd</span><span class="p">)</span>
 <span class="o">.</span>     <span class="n">switch</span><span class="p">(</span><span class="n">on_error</span><span class="p">,</span> <span class="k">continue</span> <span class="o">=</span> <span class="n">invokeRestart</span><span class="p">(</span><span class="s2">&quot;eval_continue&quot;</span><span class="p">),</span> 
 <span class="o">.</span>         <span class="n">stop</span> <span class="o">=</span> <span class="n">invokeRestart</span><span class="p">(</span><span class="s2">&quot;eval_stop&quot;</span><span class="p">),</span> <span class="n">error</span> <span class="o">=</span> <span class="n">NULL</span><span class="p">)</span>
 <span class="o">.</span> <span class="p">},</span> <span class="s2">&quot;the data frame should contain at least two columns&quot;</span><span class="p">,</span> <span class="n">base</span><span class="p">::</span><span class="n">quote</span><span class="p">(</span><span class="n">graph_from_data_frame</span><span class="p">(</span><span class="n">reg_net</span><span class="p">,</span> 
 <span class="o">.</span>     <span class="n">directed</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)))</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xqtl_roles_df %&gt;% head
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>fwrite(xqtl_roles_df, &quot;xQTL_roles_df.GRN.tsv&quot;, sep = &#39;\t&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>xqtl_roles_df &lt;- fread(&quot;xQTL_roles_df.GRN.tsv&quot;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>ggplot(xqtl_roles_df, aes(x = role)) +
  geom_bar(fill = &quot;#59A14F&quot;) +
  facet_wrap(~ file, scales = &quot;free_y&quot;) +
  labs(title = &quot;xQTL Gene Roles Across GRN Files&quot;, x = &quot;Role&quot;, y = &quot;Count&quot;) +
  theme_minimal(base_size = 13) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/1a85e0d091b492355042dbce8251280d0c608000fa2c6be046a13b9b4c52c61e.png"><img alt="../../../_images/1a85e0d091b492355042dbce8251280d0c608000fa2c6be046a13b9b4c52c61e.png" src="../../../_images/1a85e0d091b492355042dbce8251280d0c608000fa2c6be046a13b9b4c52c61e.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<section id="fisher-test-between-xqtl-and-xqtl-only-genes">
<h3>fisher test between xQTL and xQTL only genes<a class="headerlink" href="#fisher-test-between-xqtl-and-xqtl-only-genes" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Function to process one GRN file
process_grn &lt;- function(grn_file) {
  grn &lt;- read_tsv(grn_file, show_col_types = FALSE)
  if (!all(c(&quot;TF&quot;, &quot;TG&quot;) %in% names(grn))) return(NULL)
  
  reg_net &lt;- grn %&gt;% select(source = TF, target = TG) %&gt;% distinct()
  g &lt;- graph_from_data_frame(reg_net, directed = TRUE)
  all_genes &lt;- V(g)$name
  
  get_role &lt;- function(gene) {
    if (!(gene %in% V(g)$name)) return(&quot;not_in_network&quot;)
    in_deg &lt;- degree(g, gene, mode = &quot;in&quot;)
    out_deg &lt;- degree(g, gene, mode = &quot;out&quot;)
    if (out_deg &gt; 0 &amp; in_deg == 0) return(&quot;regulator&quot;)
    else if (in_deg &gt; 0 &amp; out_deg == 0) return(&quot;target&quot;)
    else if (in_deg &gt; 0 &amp; out_deg &gt; 0) return(&quot;intermediary&quot;)
    else return(&quot;unclassified&quot;)
  }
  
  role_df &lt;- data.frame(
    gene = all_genes,
    role = sapply(all_genes, get_role),
    group = ifelse(all_genes %in% xQTL_list$gene_name, &quot;xQTL_only&quot;, &quot;background&quot;),
    file = basename(grn_file),
    stringsAsFactors = FALSE
  )
  
  return(role_df)
}
# Function to run Fisher test per file (regulator vs others)
run_fisher_per_file &lt;- function(df, set_role) {
  reg_table &lt;- df %&gt;%
    filter(role %in% c(&quot;regulator&quot;, &quot;target&quot;, &quot;intermediary&quot;)) %&gt;%
    mutate(role_bin = ifelse(role == set_role, set_role, &quot;other&quot;)) %&gt;%
    count(group, role_bin) %&gt;%
    pivot_wider(names_from = role_bin, values_from = n, values_fill = 0)

  # Check that we have exactly two groups and both role types
  required_cols &lt;- c(set_role, &quot;other&quot;)
  if (nrow(reg_table) != 2 || !all(required_cols %in% names(reg_table))) {
    return(NA_real_)
  }

  # Safely extract the contingency table
  mat &lt;- matrix(c(
    reg_table[[set_role]][1], reg_table$other[1],
    reg_table[[set_role]][2], reg_table$other[2]
  ), nrow = 2, byrow = TRUE)

  # Ensure matrix is valid for Fisher test
  if (any(dim(mat) != 2) || any(rowSums(mat) == 0) || any(colSums(mat) == 0)) {
    return(NA_real_)
  }

  fisher.test(mat)$p.value
}
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Process all GRNs and combine
all_roles &lt;- map_dfr(reg_net_files, process_grn)
all_roles_filtered &lt;- all_roles %&gt;% filter(role != &quot;not_in_network&quot;)

count_table &lt;- all_roles_filtered %&gt;%
  count(file, group, role) %&gt;%
  pivot_wider(names_from = role, values_from = n, values_fill = 0)

print(count_table)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span># A tibble: 48 x 5
   file               group      intermediary regulator target
   &lt;chr&gt;              &lt;chr&gt;             &lt;int&gt;     &lt;int&gt;  &lt;int&gt;
 1 Ast_GRN.txt        background          257       127   7899
 2 Ast_GRN.txt        xQTL_only             4         1    127
 3 Chandelier_GRN.txt background          230       178   7441
 4 Chandelier_GRN.txt xQTL_only             3         1    128
 5 End_GRN.txt        background          239       114   7367
 6 End_GRN.txt        xQTL_only             4         1    117
 7 Immune_GRN.txt     background          242        98   7979
 8 Immune_GRN.txt     xQTL_only             6         0    152
 9 L2.3.IT_GRN.txt    background          279       107   9862
10 L2.3.IT_GRN.txt    xQTL_only             4         0    146
# i 38 more rows
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Calculate Fisher p-values per file
fisher_results &lt;- all_roles_filtered %&gt;%
  group_by(file) %&gt;%
  group_split() %&gt;%
  map_dbl(~run_fisher_per_file(.x, set_role = &#39;regulator&#39;))

fisher_summary &lt;- tibble(
  file = unique(all_roles_filtered$file),
  fisher_p_value = fisher_results
)

print(fisher_summary %&gt;% arrange(fisher_p_value))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span># A tibble: 24 x 2
   file               fisher_p_value
   &lt;chr&gt;                       &lt;dbl&gt;
 1 Immune_GRN.txt              0.267
 2 Mic_GRN.txt                 0.267
 3 Oli_GRN.txt                 0.267
 4 Sncg_GRN.txt                0.373
 5 Vip_GRN.txt                 0.373
 6 Chandelier_GRN.txt          0.374
 7 Lamp5_GRN.txt               0.375
 8 L6.IT.Car3_GRN.txt          0.408
 9 L5.IT_GRN.txt               0.409
10 L6b_GRN.txt                 0.409
# i 14 more rows
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Calculate Fisher p-values per file
fisher_results &lt;- all_roles_filtered %&gt;%
  group_by(file) %&gt;%
  group_split() %&gt;%
  map_dbl(~run_fisher_per_file(.x, set_role = &#39;target&#39;))

fisher_summary &lt;- tibble(
  file = unique(all_roles_filtered$file),
  fisher_p_value = fisher_results
)

print(fisher_summary %&gt;% arrange(fisher_p_value))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span># A tibble: 24 x 2
   file               fisher_p_value
   &lt;chr&gt;                       &lt;dbl&gt;
 1 Chandelier_GRN.txt          0.325
 2 Pvalb_GRN.txt               0.325
 3 Sst_GRN.txt                 0.421
 4 OPC_GRN.txt                 0.530
 5 Lamp5_GRN.txt               0.558
 6 L5.IT_GRN.txt               0.663
 7 L4.IT_GRN.txt               0.663
 8 L6.CT_GRN.txt               0.664
 9 L5.6.NP_GRN.txt             0.664
10 L6.IT.Car3_GRN.txt          0.664
# i 14 more rows
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Calculate Fisher p-values per file
fisher_results &lt;- all_roles_filtered %&gt;%
  group_by(file) %&gt;%
  group_split() %&gt;%
  map_dbl(~run_fisher_per_file(.x, set_role = &#39;intermediary&#39;))

fisher_summary &lt;- tibble(
  file = unique(all_roles_filtered$file),
  fisher_p_value = fisher_results
)

print(fisher_summary %&gt;% arrange(fisher_p_value))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span># A tibble: 24 x 2
   file            fisher_p_value
   &lt;chr&gt;                    &lt;dbl&gt;
 1 Mic_GRN.txt              0.465
 2 Immune_GRN.txt           0.470
 3 Pvalb_GRN.txt            0.596
 4 Sncg_GRN.txt             0.788
 5 End_GRN.txt              0.792
 6 Vip_GRN.txt              0.792
 7 Pax6_GRN.txt             0.793
 8 L5.6.NP_GRN.txt          0.799
 9 Lamp5_GRN.txt            0.799
10 Sst_GRN.txt              0.800
# i 14 more rows
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Visualization: Role distribution per GRN file by group
ggplot(all_roles_filtered, aes(x = role, fill = group)) +
  geom_bar(position = &quot;fill&quot;) +
  facet_wrap(~ file, scales = &quot;free_y&quot;) +
  labs(title = &quot;xQTL Gene Role Proportions Across GRNs&quot;,
       x = &quot;Role&quot;, y = &quot;Proportion&quot;) +
  scale_fill_manual(values = c(&quot;xQTL_only&quot; = &quot;#E15759&quot;, &quot;background&quot; = &quot;#4E79A7&quot;)) +
  theme_minimal(base_size = 13) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/5f9864f09a365361519db9f84af31cf3f22214a1ed6a9d234804b6c102635c0f.png"><img alt="../../../_images/5f9864f09a365361519db9f84af31cf3f22214a1ed6a9d234804b6c102635c0f.png" src="../../../_images/5f9864f09a365361519db9f84af31cf3f22214a1ed6a9d234804b6c102635c0f.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>all_roles_filtered %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 4</caption>
<thead>
	<tr><th></th><th scope=col>gene</th><th scope=col>role</th><th scope=col>group</th><th scope=col>file</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>RXRG...1</th><td>RXRG   </td><td>intermediary</td><td>background</td><td>Ast_GRN.txt</td></tr>
	<tr><th scope=row>NR2F1...2</th><td>NR2F1  </td><td>intermediary</td><td>background</td><td>Ast_GRN.txt</td></tr>
	<tr><th scope=row>IRF1...3</th><td>IRF1   </td><td>intermediary</td><td>background</td><td>Ast_GRN.txt</td></tr>
	<tr><th scope=row>ESR1...4</th><td>ESR1   </td><td>intermediary</td><td>background</td><td>Ast_GRN.txt</td></tr>
	<tr><th scope=row>ZNF143...5</th><td>ZNF143 </td><td>intermediary</td><td>background</td><td>Ast_GRN.txt</td></tr>
	<tr><th scope=row>CREB3L1...6</th><td>CREB3L1</td><td>intermediary</td><td>background</td><td>Ast_GRN.txt</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>df &lt;- data.frame()
for(i in reg_net_files){
    df &lt;- rbind(df, fread(i) %&gt;% mutate(file = basename(i)))
}
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>df %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 x 10</caption>
<thead>
	<tr><th scope=col>TF</th><th scope=col>enhancer</th><th scope=col>promoter</th><th scope=col>TG</th><th scope=col>edgeWeight</th><th scope=col>method</th><th scope=col>celltype</th><th scope=col>Correlation</th><th scope=col>Regulation</th><th scope=col>file</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>RXRG   </td><td>chr1:1019276-1019776</td><td>chr1:958584-960584</td><td>KLHL17</td><td>0.12987922</td><td>scGRNom</td><td>Ast</td><td> 0.6773105</td><td>Activating</td><td>Ast_GRN.txt</td></tr>
	<tr><td>NR2F1  </td><td>chr1:1019276-1019776</td><td>chr1:958584-960584</td><td>KLHL17</td><td>0.06547182</td><td>scGRNom</td><td>Ast</td><td>-0.1551904</td><td>Repressing</td><td>Ast_GRN.txt</td></tr>
	<tr><td>IRF1   </td><td>chr1:1019276-1019776</td><td>chr1:958584-960584</td><td>KLHL17</td><td>0.06022639</td><td>scGRNom</td><td>Ast</td><td> 0.6686182</td><td>Activating</td><td>Ast_GRN.txt</td></tr>
	<tr><td>ESR1   </td><td>chr1:1019276-1019776</td><td>chr1:958584-960584</td><td>KLHL17</td><td>0.06465946</td><td>scGRNom</td><td>Ast</td><td> 0.5641387</td><td>Activating</td><td>Ast_GRN.txt</td></tr>
	<tr><td>ZNF143 </td><td>chr1:1019276-1019776</td><td>chr1:958584-960584</td><td>KLHL17</td><td>0.04856990</td><td>scGRNom</td><td>Ast</td><td> 0.3650754</td><td>Activating</td><td>Ast_GRN.txt</td></tr>
	<tr><td>CREB3L1</td><td>chr1:1019276-1019776</td><td>chr1:958584-960584</td><td>KLHL17</td><td>0.09256433</td><td>scGRNom</td><td>Ast</td><td> 0.7666757</td><td>Activating</td><td>Ast_GRN.txt</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>saveRDS(df, &#39;all_GRN.rds&#39;)
</pre></div>
</div>
</div>
</div>
</section>
<section id="multigene-grn-ypel3">
<h3>Multigene GRN -YPEL3<a class="headerlink" href="#multigene-grn-ypel3" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>
# grn &lt;- read_tsv(reg_net_files[5], show_col_types = FALSE)
df &lt;- readRDS(&#39;all_GRN.rds&#39;)
reg_net &lt;- df %&gt;% select(source = TF, target = TG) %&gt;% distinct()
g &lt;- graph_from_data_frame(reg_net, directed = TRUE)

focus_gene &lt;- &quot;CORO1A&quot; 
neighbors_g &lt;- ego(g, order = 1, nodes = focus_gene, mode = &quot;in&quot;)[[1]]
subg &lt;- induced_subgraph(g, neighbors_g)

plot(subg,
     vertex.label.cex = 0.8,
     vertex.label.color = &quot;black&quot;,
     vertex.color = &quot;lightgreen&quot;,
     edge.arrow.size = 0.3,
     layout = layout_with_fr,
     main = paste(&quot;Target Network of&quot;, focus_gene))
coro1a_genes &lt;- names(neighbors_g)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/f53148d715839c75768594a2d01b3013ffec2223cde6ccfef791b185e20b0598.png"><img alt="../../../_images/f53148d715839c75768594a2d01b3013ffec2223cde6ccfef791b185e20b0598.png" src="../../../_images/f53148d715839c75768594a2d01b3013ffec2223cde6ccfef791b185e20b0598.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>
# grn &lt;- read_tsv(reg_net_files[5], show_col_types = FALSE)

reg_net &lt;- df %&gt;% select(source = TF, target = TG) %&gt;% distinct()
g &lt;- graph_from_data_frame(reg_net, directed = TRUE)

focus_gene &lt;- &quot;YPEL3&quot; 
neighbors_g &lt;- ego(g, order = 1, nodes = focus_gene, mode = &quot;in&quot;)[[1]]
subg &lt;- induced_subgraph(g, neighbors_g)

plot(subg,
     vertex.label.cex = 0.8,
     vertex.label.color = &quot;black&quot;,
     vertex.color = &quot;lightgreen&quot;,
     edge.arrow.size = 0.3,
     layout = layout_with_fr,
     main = paste(&quot;Target Network of&quot;, focus_gene))
ypel3_genes &lt;- names(neighbors_g)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/a04508a9c1c4e1c64ce1946c5ffb01afda3ec3573eae9cefa95cb727a47b2b13.png"><img alt="../../../_images/a04508a9c1c4e1c64ce1946c5ffb01afda3ec3573eae9cefa95cb727a47b2b13.png" src="../../../_images/a04508a9c1c4e1c64ce1946c5ffb01afda3ec3573eae9cefa95cb727a47b2b13.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>
# grn &lt;- read_tsv(reg_net_files[5], show_col_types = FALSE)

reg_net &lt;- df %&gt;% select(source = TF, target = TG) %&gt;% distinct()
g &lt;- graph_from_data_frame(reg_net, directed = TRUE)

focus_gene &lt;- &quot;DOC2A&quot; 
neighbors_g &lt;- ego(g, order = 1, nodes = focus_gene, mode = &quot;in&quot;)[[1]]
subg &lt;- induced_subgraph(g, neighbors_g)

plot(subg,
     vertex.label.cex = 0.8,
     vertex.label.color = &quot;black&quot;,
     vertex.color = &quot;lightgreen&quot;,
     edge.arrow.size = 0.3,
     layout = layout_with_fr,
     main = paste(&quot;Target Network of&quot;, focus_gene))

doc2a_genes &lt;- names(neighbors_g)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/c9b4cd392296f60412119307ab161f39eaa1d5c6075a83501b201cceda94177f.png"><img alt="../../../_images/c9b4cd392296f60412119307ab161f39eaa1d5c6075a83501b201cceda94177f.png" src="../../../_images/c9b4cd392296f60412119307ab161f39eaa1d5c6075a83501b201cceda94177f.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>
# grn &lt;- read_tsv(reg_net_files[5], show_col_types = FALSE)

reg_net &lt;- df %&gt;% select(source = TF, target = TG) %&gt;% distinct()
g &lt;- graph_from_data_frame(reg_net, directed = TRUE)

focus_gene &lt;- &quot;INO80E&quot; 
neighbors_g &lt;- ego(g, order = 1, nodes = focus_gene, mode = &quot;in&quot;)[[1]]
subg &lt;- induced_subgraph(g, neighbors_g)

plot(subg,
     vertex.label.cex = 0.8,
     vertex.label.color = &quot;black&quot;,
     vertex.color = &quot;lightgreen&quot;,
     edge.arrow.size = 0.3,
     layout = layout_with_fr,
     main = paste(&quot;Target Network of&quot;, focus_gene))

ino80e_genes &lt;- names(neighbors_g)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/351475c4acd443e6b5938252e2caee4c829dc69411a09f636edec35695992e00.png"><img alt="../../../_images/351475c4acd443e6b5938252e2caee4c829dc69411a09f636edec35695992e00.png" src="../../../_images/351475c4acd443e6b5938252e2caee4c829dc69411a09f636edec35695992e00.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multi_gene_list %&gt;% length
valid_genes %&gt;% length
V(g)$name %&gt;% length
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">12</div><div class="output text_html">2</div><div class="output text_html">14484</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multi_gene_list
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'YPEL3'</li><li>'INO80E'</li><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li><li>'BCKDK'</li><li>'TLCD3B'</li><li>'ZNF267'</li><li>'AC093512.2'</li><li>'PRSS8'</li><li>'TAOK2'</li><li>'TMEM219'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># multi_gene_list_name &lt;- gene_ref %&gt;% filter(gene_id %in% multi_gene_list) %&gt;% pull(gene_name)
multi_gene_list_name &lt;- multi_gene_list
valid_genes &lt;- multi_gene_list_name[multi_gene_list_name %in% V(g)$name]

tf_genes &lt;- list()
for (gene in valid_genes) {
  neighbors_g &lt;- ego(g, order = 1, nodes = gene, mode = &quot;in&quot;)[[1]]
  tf_genes[[gene]] &lt;- setdiff(names(neighbors_g), gene)
}
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>tf_genes
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><dl>
	<dt>$YPEL3</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'NR2F1'</li><li>'ESR1'</li><li>'CREB3L1'</li><li>'JUN'</li><li>'ATF2'</li><li>'JUND'</li><li>'BACH1'</li><li>'RXRA'</li><li>'THRB'</li><li>'ZNF148'</li><li>'MAFK'</li><li>'TP53'</li><li>'PPARD'</li><li>'FOXK1'</li><li>'ZNF263'</li><li>'NFYB'</li><li>'SP2'</li><li>'FOS'</li><li>'RARA'</li><li>'BACH2'</li><li>'RREB1'</li><li>'IRF2'</li><li>'SPI1'</li><li>'NR2C2'</li><li>'KLF13'</li><li>'NR1H2'</li><li>'RXRB'</li><li>'ZNF135'</li><li>'PPARA'</li><li>'EGR1'</li><li>'NFIB'</li><li>'PKNOX2'</li><li>'NR6A1'</li><li>'PKNOX1'</li><li>'ASCL1'</li><li>'PPARG'</li><li>'SMAD3'</li><li>'ZNF684'</li><li>'NFYA'</li><li>'SMAD2'</li><li>'SMAD4'</li><li>'SP1'</li><li>'MAZ'</li><li>'IRF9'</li><li>'POU2F2'</li><li>'ZNF274'</li><li>'ATF6'</li><li>'ZNF449'</li><li>'INSM1'</li><li>'PURA'</li><li>'NPDC1'</li></ol>
</dd>
	<dt>$INO80E</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'RXRG'</li><li>'ESR1'</li><li>'ELF1'</li><li>'RXRA'</li><li>'TEAD1'</li><li>'PLAG1'</li><li>'ZBTB26'</li><li>'ZNF263'</li><li>'TCF7L1'</li><li>'ZBTB6'</li><li>'TCF7L2'</li><li>'SP2'</li><li>'NR1D1'</li><li>'RARA'</li><li>'HNF4G'</li><li>'NR1D2'</li><li>'IRF3'</li><li>'IRF2'</li><li>'SPI1'</li><li>'NR2C2'</li><li>'NR3C1'</li><li>'REST'</li><li>'YY1'</li><li>'NR6A1'</li><li>'PKNOX1'</li><li>'ASCL1'</li><li>'ZKSCAN5'</li><li>'HSF1'</li><li>'IRF9'</li><li>'SCRT1'</li><li>'ETV1'</li><li>'SRF'</li><li>'GLIS2'</li><li>'RELA'</li></ol>
</dd>
	<dt>$DOC2A</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'NR2F1'</li><li>'IRF1'</li><li>'ZBTB7A'</li><li>'ZNF528'</li><li>'JUN'</li><li>'SP3'</li><li>'MAFG'</li><li>'JUND'</li><li>'BACH1'</li><li>'STAT1'</li><li>'FOXP1'</li><li>'RXRA'</li><li>'MEF2A'</li><li>'RFX5'</li><li>'POU3F3'</li><li>'ZNF148'</li><li>'RFX3'</li><li>'STAT2'</li><li>'NFIC'</li><li>'MAFK'</li><li>'SP4'</li><li>'ZNF460'</li><li>'SOX4'</li><li>'ZNF384'</li><li>'E2F6'</li><li>'ZNF263'</li><li>'SP2'</li><li>'ZNF682'</li><li>'FOS'</li><li>'PRDM1'</li><li>'SREBF1'</li><li>'NR1D1'</li><li>'RFX2'</li><li>'NR1D2'</li><li>'BACH2'</li><li>'RREB1'</li><li>'SPI1'</li><li>'NR2C2'</li><li>'KLF13'</li><li>'MEF2D'</li><li>'ZNF135'</li><li>'EGR1'</li><li>'MAF'</li><li>'GABPA'</li><li>'SP1'</li><li>'MAZ'</li><li>'JUNB'</li><li>'RFX4'</li><li>'KLF16'</li><li>'RFX1'</li><li>'NEUROD1'</li><li>'SP9'</li><li>'ZNF75D'</li><li>'INSM1'</li><li>'FOXO1'</li><li>'MGA'</li><li>'ZNF75A'</li><li>'WT1'</li><li>'PURA'</li><li>'NR0B1'</li><li>'KLF4'</li><li>'MEIS3'</li><li>'MYOD1'</li><li>'FERD3L'</li><li>'TAL1'</li><li>'TAL2'</li><li>'NFE2'</li><li>'ZNF219'</li><li>'KLF1'</li><li>'GTF2F1'</li><li>'ETS2'</li><li>'EPAS1'</li><li>'EOMES'</li><li>'PGR'</li><li>'NFAT5'</li><li>'TBX10'</li><li>'ZNF75C'</li></ol>
</dd>
	<dt>$MAPK3</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'RXRG'</li><li>'NR2F1'</li><li>'IRF1'</li><li>'KLF15'</li><li>'JUN'</li><li>'MAFG'</li><li>'CEBPG'</li><li>'ATF4'</li><li>'STAT1'</li><li>'FOXP1'</li><li>'RXRA'</li><li>'MEF2A'</li><li>'USF2'</li><li>'THRB'</li><li>'PBX3'</li><li>'NFIC'</li><li>'SP4'</li><li>'ZNF460'</li><li>'ZNF384'</li><li>'E2F6'</li><li>'ZNF263'</li><li>'SP2'</li><li>'MEF2C'</li><li>'PRDM1'</li><li>'RARG'</li><li>'RARA'</li><li>'IRF3'</li><li>'SPI1'</li><li>'KLF13'</li><li>'MEF2D'</li><li>'RXRB'</li><li>'EGR1'</li><li>'NR3C1'</li><li>'NFIB'</li><li>'PKNOX2'</li><li>'SMAD3'</li><li>'SMAD2'</li><li>'CEBPA'</li><li>'ZNF382'</li><li>'SP1'</li><li>'MAZ'</li><li>'RARB'</li><li>'MITF'</li><li>'BHLHE41'</li><li>'USF1'</li><li>'CEBPD'</li><li>'ELF2'</li><li>'RELA'</li><li>'ETV6'</li><li>'ZNF76'</li><li>'RUNX1'</li><li>'ETS2'</li><li>'ELF5'</li><li>'SPIB'</li><li>'ZNF683'</li><li>'ATF5'</li><li>'SPIC'</li></ol>
</dd>
	<dt>$CORO1A</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'RXRG'</li><li>'NR2F1'</li><li>'IRF1'</li><li>'ELF1'</li><li>'ONECUT2'</li><li>'JUN'</li><li>'ATF2'</li><li>'JUND'</li><li>'CEBPG'</li><li>'BACH1'</li><li>'STAT1'</li><li>'FOXP1'</li><li>'RXRA'</li><li>'MEF2A'</li><li>'USF2'</li><li>'STAT2'</li><li>'NFIC'</li><li>'EBF1'</li><li>'MAFK'</li><li>'PLAG1'</li><li>'MEF2B'</li><li>'FOXK1'</li><li>'ZBTB26'</li><li>'ZNF384'</li><li>'ZNF263'</li><li>'SP2'</li><li>'ZNF682'</li><li>'XBP1'</li><li>'FOS'</li><li>'MEF2C'</li><li>'PRDM1'</li><li>'SREBF1'</li><li>'NR1D1'</li><li>'RARA'</li><li>'BACH2'</li><li>'IRF3'</li><li>'RREB1'</li><li>'SPI1'</li><li>'MEF2D'</li><li>'ZNF135'</li><li>'PPARA'</li><li>'NFIB'</li><li>'REST'</li><li>'AR'</li><li>'NR6A1'</li><li>'MAF'</li><li>'PKNOX1'</li><li>'SMAD3'</li><li>'ZNF684'</li><li>'RUNX2'</li><li>'SMAD2'</li><li>'MAZ'</li><li>'IRF9'</li><li>'SCRT1'</li><li>'NR4A1'</li><li>'ELF2'</li><li>'STAT3'</li><li>'ETV6'</li><li>'CEBPB'</li><li>'PML'</li><li>'BCL11A'</li><li>'NR2C1'</li><li>'ETS2'</li><li>'KLF2'</li><li>'ELF5'</li><li>'ARID3A'</li><li>'SPIB'</li><li>'BCL11B'</li><li>'ENSG00000250096'</li><li>'IRF4'</li><li>'ATF5'</li><li>'SPIC'</li></ol>
</dd>
	<dt>$BCKDK</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'ZNF148'</li><li>'MAZ'</li><li>'PURA'</li></ol>
</dd>
	<dt>$ZNF267</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'IRF1'</li><li>'ESR1'</li><li>'STAT1'</li><li>'MZF1'</li><li>'SP4'</li><li>'SP2'</li><li>'PRDM1'</li><li>'RREB1'</li><li>'IRF2'</li><li>'NFIB'</li><li>'YY1'</li><li>'SMAD2'</li><li>'IKZF1'</li><li>'ZNF75D'</li><li>'PML'</li><li>'ETV3'</li><li>'REL'</li></ol>
</dd>
	<dt>$TAOK2</dt>
		<dd>'ATF4'</dd>
	<dt>$TMEM219</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'NR2F1'</li><li>'ELF1'</li><li>'ONECUT2'</li><li>'JUN'</li><li>'ATF2'</li><li>'JUND'</li><li>'ATF4'</li><li>'FOXP1'</li><li>'RXRA'</li><li>'MEF2A'</li><li>'USF2'</li><li>'PBX3'</li><li>'RFX3'</li><li>'NFIC'</li><li>'MAFK'</li><li>'PLAG1'</li><li>'ZNF460'</li><li>'SOX4'</li><li>'ZNF384'</li><li>'GLIS1'</li><li>'KLF9'</li><li>'ZNF263'</li><li>'RORA'</li><li>'MEF2C'</li><li>'PRDM1'</li><li>'ZNF282'</li><li>'SREBF1'</li><li>'NR1D1'</li><li>'RFX2'</li><li>'NFIX'</li><li>'RARA'</li><li>'NR2F6'</li><li>'NR1D2'</li><li>'IRF3'</li><li>'GLI3'</li><li>'MEF2D'</li><li>'ZNF135'</li><li>'NFIB'</li><li>'PKNOX1'</li><li>'SMAD3'</li><li>'ZNF136'</li><li>'GABPA'</li><li>'SMAD2'</li><li>'SMAD4'</li><li>'NFKB1'</li><li>'GLI2'</li><li>'CREB3'</li><li>'MITF'</li><li>'RFX4'</li><li>'ETV1'</li><li>'RFX1'</li><li>'ATF6'</li><li>'NR4A2'</li><li>'SREBF2'</li><li>'ETS2'</li></ol>
</dd>
</dl>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>tf_genes$YPEL3 %&gt;% .[. %in% multi_gene_list]
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>tf_genes$INO80E %&gt;% .[ . %in% multi_gene_list]
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>tf_genes$DOC2A %&gt;% .[. %in% multi_gene_list]
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>tf_genes$MAPK3 %&gt;% .[. %in% multi_gene_list]
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>tf_genes$CORO1A %&gt;% .[. %in% multi_gene_list]
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(UpSetR)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Convert list to binary presence/absence data frame
gene_matrix &lt;- fromList(tf_genes)

# Create the UpSet plot
upset(gene_matrix, sets = names(tf_genes), keep.order = TRUE, order.by = &quot;freq&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/6398fbec9d6837b41958e3c6a01d1fd8505a650e17a60ba2b288646a25467203.png"><img alt="../../../_images/6398fbec9d6837b41958e3c6a01d1fd8505a650e17a60ba2b288646a25467203.png" src="../../../_images/6398fbec9d6837b41958e3c6a01d1fd8505a650e17a60ba2b288646a25467203.png" style="width: 960px; height: 360px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>tf_genes %&gt;% str
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>List of 9
 $ YPEL3  : chr [1:51] &quot;NR2F1&quot; &quot;ESR1&quot; &quot;CREB3L1&quot; &quot;JUN&quot; ...
 $ INO80E : chr [1:34] &quot;RXRG&quot; &quot;ESR1&quot; &quot;ELF1&quot; &quot;RXRA&quot; ...
 $ DOC2A  : chr [1:77] &quot;NR2F1&quot; &quot;IRF1&quot; &quot;ZBTB7A&quot; &quot;ZNF528&quot; ...
 $ MAPK3  : chr [1:57] &quot;RXRG&quot; &quot;NR2F1&quot; &quot;IRF1&quot; &quot;KLF15&quot; ...
 $ CORO1A : chr [1:72] &quot;RXRG&quot; &quot;NR2F1&quot; &quot;IRF1&quot; &quot;ELF1&quot; ...
 $ BCKDK  : chr [1:3] &quot;ZNF148&quot; &quot;MAZ&quot; &quot;PURA&quot;
 $ ZNF267 : chr [1:17] &quot;IRF1&quot; &quot;ESR1&quot; &quot;STAT1&quot; &quot;MZF1&quot; ...
 $ TAOK2  : chr &quot;ATF4&quot;
 $ TMEM219: chr [1:55] &quot;NR2F1&quot; &quot;ELF1&quot; &quot;ONECUT2&quot; &quot;JUN&quot; ...
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(UpSetR)

# Build the binary matrix from tf_genes
all_tfs &lt;- unique(unlist(tf_genes))
gene_matrix &lt;- as.data.frame(
  sapply(tf_genes, function(gene_tfs) as.integer(all_tfs %in% gene_tfs))
)
rownames(gene_matrix) &lt;- all_tfs

# Check column names
print(colnames(gene_matrix))  # Should include &quot;YPEL3&quot;, &quot;INO80E&quot;, &quot;DOC2A&quot;
options(repr.plot.width = 10, repr.plot.height = 6)

# Generate the UpSet plot with highlighted intersections
upset(
  gene_matrix,
  sets = c(&quot;YPEL3&quot;, &quot;INO80E&quot;, &quot;DOC2A&quot;, &quot;CORO1A&quot;, &quot;MAPK3&quot;),
  keep.order = TRUE,
  order.by = &quot;freq&quot;,
  queries = list(
    list(
      query = intersects,
      params = list(&quot;YPEL3&quot;, &quot;INO80E&quot;, &quot;DOC2A&quot;),
      color = &quot;red&quot;,
      active = TRUE,
      query.name = &quot;Shared: YPEL3+INO80E+DOC2A&quot;
    ),
    list(
      query = intersects,
      params = list(&quot;YPEL3&quot;, &quot;INO80E&quot;, &quot;DOC2A&quot;, &quot;CORO1A&quot;, &quot;MAPK3&quot;),
      color = &quot;blue&quot;,
      active = TRUE,
      query.name = &quot;Shared: All Five Genes&quot;
    )
  ),
  text.scale = 2
)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[1] &quot;YPEL3&quot;   &quot;INO80E&quot;  &quot;DOC2A&quot;   &quot;MAPK3&quot;   &quot;CORO1A&quot;  &quot;BCKDK&quot;   &quot;ZNF267&quot; 
[8] &quot;TAOK2&quot;   &quot;TMEM219&quot;
</pre></div>
</div>
<a class="reference internal image-reference" href="../../../_images/8f480d2a0b5b18d0dd53d751331f516844b9b81a1c53e36589351cfd110d7851.png"><img alt="../../../_images/8f480d2a0b5b18d0dd53d751331f516844b9b81a1c53e36589351cfd110d7851.png" src="../../../_images/8f480d2a0b5b18d0dd53d751331f516844b9b81a1c53e36589351cfd110d7851.png" style="width: 600px; height: 360px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Filter rows where all three genes are present (shared)
rownames(gene_matrix) &lt;- unique(unlist(tf_genes))
shared_rows &lt;- gene_matrix %&gt;%
  filter(YPEL3 == 1, INO80E == 1, DOC2A == 1)
shared_rows
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 5 x 9</caption>
<thead>
	<tr><th></th><th scope=col>YPEL3</th><th scope=col>INO80E</th><th scope=col>DOC2A</th><th scope=col>MAPK3</th><th scope=col>CORO1A</th><th scope=col>BCKDK</th><th scope=col>ZNF267</th><th scope=col>TAOK2</th><th scope=col>TMEM219</th></tr>
	<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>RXRA</th><td>1</td><td>1</td><td>1</td><td>1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>
	<tr><th scope=row>ZNF263</th><td>1</td><td>1</td><td>1</td><td>1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>
	<tr><th scope=row>SP2</th><td>1</td><td>1</td><td>1</td><td>1</td><td>1</td><td>0</td><td>1</td><td>0</td><td>0</td></tr>
	<tr><th scope=row>SPI1</th><td>1</td><td>1</td><td>1</td><td>1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
	<tr><th scope=row>NR2C2</th><td>1</td><td>1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>names(tf_genes)
Reduce(intersect, tf_genes[-c(9)])
names(tf_genes[-c(9)])
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'YPEL3'</li><li>'INO80E'</li><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li><li>'BCKDK'</li><li>'ZNF267'</li><li>'TAOK2'</li><li>'TMEM219'</li></ol>
</div><div class="output text_html"></div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'YPEL3'</li><li>'INO80E'</li><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li><li>'BCKDK'</li><li>'ZNF267'</li><li>'TAOK2'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>names(tf_genes)
Reduce(intersect, tf_genes[-c(2,8:9)])
names(tf_genes[-c(2,8:9)])
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'YPEL3'</li><li>'INO80E'</li><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li><li>'BCKDK'</li><li>'ZNF267'</li><li>'TAOK2'</li><li>'TMEM219'</li></ol>
</div><div class="output text_html"></div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'YPEL3'</li><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li><li>'BCKDK'</li><li>'ZNF267'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>names(tf_genes)
Reduce(intersect, tf_genes[-c(2, 7:9)])
names(tf_genes[-c(2, 7:9)])
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'YPEL3'</li><li>'INO80E'</li><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li><li>'BCKDK'</li><li>'ZNF267'</li><li>'TAOK2'</li><li>'TMEM219'</li></ol>
</div><div class="output text_html">'MAZ'</div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'YPEL3'</li><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li><li>'BCKDK'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>names(tf_genes)
Reduce(intersect, tf_genes[-c(2, 6:9)])
names(tf_genes[-c(2, 6:9)])
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'YPEL3'</li><li>'INO80E'</li><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li><li>'BCKDK'</li><li>'ZNF267'</li><li>'TAOK2'</li><li>'TMEM219'</li></ol>
</div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'NR2F1'</li><li>'JUN'</li><li>'RXRA'</li><li>'ZNF263'</li><li>'SP2'</li><li>'SPI1'</li><li>'MAZ'</li></ol>
</div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'YPEL3'</li><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>names(tf_genes)
Reduce(intersect, tf_genes[-c(1:2, 6:9)])
names(tf_genes[-c(1:2, 6:9)])
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'YPEL3'</li><li>'INO80E'</li><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li><li>'BCKDK'</li><li>'ZNF267'</li><li>'TAOK2'</li><li>'TMEM219'</li></ol>
</div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'NR2F1'</li><li>'IRF1'</li><li>'JUN'</li><li>'STAT1'</li><li>'FOXP1'</li><li>'RXRA'</li><li>'MEF2A'</li><li>'NFIC'</li><li>'ZNF384'</li><li>'ZNF263'</li><li>'SP2'</li><li>'PRDM1'</li><li>'SPI1'</li><li>'MEF2D'</li><li>'MAZ'</li><li>'ETS2'</li></ol>
</div><div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'DOC2A'</li><li>'MAPK3'</li><li>'CORO1A'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>retest_genes &lt;- Reduce(intersect, tf_genes[3:4])
retest_in_tf_genes &lt;- list()
for (gene in retest_genes) {
  neighbors_g &lt;- ego(g, order = 1, nodes = gene, mode = &quot;in&quot;)[[1]]
  retest_in_tf_genes[[gene]] &lt;- setdiff(names(neighbors_g), gene)
}
retest_in_tf_genes
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><dl>
	<dt>$RXRA</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'RXRG'</li><li>'NR2F1'</li><li>'ESR1'</li><li>'CREB3L1'</li><li>'ZBTB7A'</li><li>'KLF15'</li><li>'SP3'</li><li>'NFE2L1'</li><li>'MAFG'</li><li>'BACH1'</li><li>'USF2'</li><li>'SCRT2'</li><li>'THRB'</li><li>'BCL6'</li><li>'POU3F3'</li><li>'ZNF148'</li><li>'NFIC'</li><li>'MZF1'</li><li>'EBF1'</li><li>'PLAG1'</li><li>'ZNF460'</li><li>'PPARD'</li><li>'ZBTB26'</li><li>'E2F6'</li><li>'KLF9'</li><li>'ZNF263'</li><li>'TCF7L1'</li><li>'SP2'</li><li>'XBP1'</li><li>'HSF4'</li><li>'RFX2'</li><li>'NFIX'</li><li>'RARA'</li><li>'RREB1'</li><li>'NFE2L2'</li><li>'ESR2'</li><li>'RXRB'</li><li>'PPARA'</li><li>'EGR1'</li><li>'NR3C1'</li><li>'NR6A1'</li><li>'MAF'</li><li>'ASCL1'</li><li>'ZKSCAN5'</li><li>'PPARG'</li><li>'GABPA'</li><li>'SMAD2'</li><li>'SMAD4'</li><li>'NFKB1'</li><li>'SP1'</li><li>'MAZ'</li><li>'HSF1'</li><li>'CREB3'</li><li>'GLIS3'</li><li>'HSF2'</li><li>'SCRT1'</li><li>'ZNF740'</li><li>'KLF16'</li><li>'ATF6'</li><li>'ZNF16'</li><li>'NEUROD1'</li><li>'SOX2'</li><li>'ZNF449'</li><li>'KLF5'</li><li>'INSM1'</li><li>'TCF12'</li><li>'SMAD5'</li><li>'KLF6'</li><li>'ZNF281'</li><li>'MXD4'</li><li>'TFCP2L1'</li><li>'PLAGL1'</li><li>'ZFY'</li><li>'MSX2'</li><li>'BSX'</li><li>'LHX9'</li><li>'PRRX2'</li><li>'SHOX2'</li><li>'RAX2'</li><li>'ISX'</li><li>'SHOX'</li><li>'ZFX'</li><li>'DLX3'</li><li>'DLX2'</li><li>'TFAP2C'</li><li>'EN1'</li><li>'ENSG00000229544'</li><li>'GBX1'</li><li>'RFX7'</li><li>'PATZ1'</li><li>'HOXA4'</li><li>'PRRX1'</li><li>'WT1'</li><li>'PURA'</li><li>'ZNF350'</li><li>'MECP2'</li><li>'TFAP2B'</li><li>'SMAD1'</li><li>'KLF4'</li><li>'DLX4'</li><li>'MYOG'</li><li>'ASCL2'</li><li>'MYF5'</li><li>'LMO2'</li><li>'MAFA'</li><li>'NHLH2'</li><li>'NHLH1'</li><li>'LYL1'</li><li>'CTCFL'</li><li>'KLF1'</li><li>'UBTF'</li><li>'SP6'</li><li>'SP5'</li><li>'ZIC4'</li><li>'TFAP2D'</li><li>'PTF1A'</li></ol>
</dd>
	<dt>$ZNF263</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'ELK4'</li><li>'TBP'</li><li>'GMEB2'</li><li>'ARID3A'</li></ol>
</dd>
	<dt>$SP2</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'RXRG'</li><li>'NR2F1'</li><li>'ESR1'</li><li>'ZNF24'</li><li>'ATF2'</li><li>'STAT1'</li><li>'FOXP1'</li><li>'MEF2A'</li><li>'THRB'</li><li>'PBX3'</li><li>'ZNF148'</li><li>'ZIC1'</li><li>'EBF1'</li><li>'PLAG1'</li><li>'SP4'</li><li>'CTCF'</li><li>'ZNF460'</li><li>'SOX4'</li><li>'PPARD'</li><li>'ZBTB6'</li><li>'NFYB'</li><li>'MEF2C'</li><li>'ZNF282'</li><li>'NR1D1'</li><li>'RARA'</li><li>'NR2C2'</li><li>'KLF13'</li><li>'NR1H2'</li><li>'RXRB'</li><li>'EGR1'</li><li>'PPARG'</li><li>'NFYA'</li><li>'SP1'</li><li>'MAZ'</li><li>'SCRT1'</li><li>'GLIS2'</li><li>'PAX6'</li><li>'MTF1'</li><li>'IKZF1'</li><li>'ZNF449'</li><li>'ZBTB17'</li></ol>
</dd>
	<dt>$NR1D1</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'CEBPG'</li><li>'CEBPB'</li><li>'HCFC1'</li><li>'GTF2F1'</li></ol>
</dd>
	<dt>$NR1D2</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'ELF1'</li><li>'SP3'</li><li>'MEF2A'</li><li>'ZNF148'</li><li>'RREB1'</li><li>'POU2F2'</li><li>'TFEC'</li><li>'SMARCC2'</li></ol>
</dd>
	<dt>$SPI1</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'IRF1'</li><li>'ELF1'</li><li>'ZBTB7A'</li><li>'SP3'</li><li>'CREM'</li><li>'STAT1'</li><li>'FOXP1'</li><li>'PBX3'</li><li>'ZNF148'</li><li>'MZF1'</li><li>'MAFK'</li><li>'ZNF263'</li><li>'SP2'</li><li>'TCF4'</li><li>'PRDM1'</li><li>'RUNX2'</li><li>'SP1'</li><li>'MAZ'</li><li>'MITF'</li><li>'KLF16'</li><li>'KLF5'</li><li>'INSM1'</li><li>'KLF6'</li><li>'ZNF281'</li><li>'ZFY'</li><li>'ZFX'</li><li>'PATZ1'</li><li>'PURA'</li><li>'KLF4'</li><li>'BCL11A'</li><li>'RUNX1'</li><li>'KLF1'</li><li>'ETS2'</li><li>'KLF7'</li><li>'SP6'</li><li>'KLF2'</li><li>'TFAP2D'</li><li>'REL'</li><li>'SPIB'</li><li>'BCL11B'</li><li>'SPIC'</li><li>'NR2E1'</li><li>'MYCL1'</li></ol>
</dd>
	<dt>$NR2C2</dt>
		<dd><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'ZNF148'</li><li>'ZNF263'</li><li>'SP2'</li><li>'RREB1'</li><li>'SP1'</li><li>'MAZ'</li><li>'KLF16'</li><li>'GLIS2'</li><li>'SOX2'</li><li>'SP9'</li><li>'KLF5'</li><li>'ZNF281'</li><li>'FOXO1'</li><li>'ZFX'</li><li>'PATZ1'</li><li>'PURA'</li><li>'SMAD1'</li><li>'RAD21'</li><li>'LMO2'</li><li>'SP8'</li><li>'SP7'</li><li>'GTF2F1'</li><li>'SP6'</li><li>'SP5'</li><li>'CREB3L2'</li><li>'SOX17'</li><li>'SOX18'</li></ol>
</dd>
</dl>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>retest_in_tf_genes$RXRA %&gt;% .[. %in% retest_genes]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>retest_in_tf_genes$ZNF263 %&gt;% .[ . %in% retest_genes]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>retest_in_tf_genes$SP2 %&gt;% .[. %in% retest_genes]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>retest_in_tf_genes$SPI1 %&gt;% .[. %in% retest_genes]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>retest_in_tf_genes$NR2C2 %&gt;% .[. %in% retest_genes]
</pre></div>
</div>
</div>
</div>
<p>(ZNF263 + SP2) -&gt; SPI1 + RXRA –&gt; c(‘YPEL3’, ‘INO80E’, ‘DOC2A’, ‘MAPK3’, ‘CORO1A’) ## seems ‘BCKDK’’ZNF267’ are not in this bundle</p>
<p>NR2C2  &lt;-&gt;  (ZNF263 + SP2) -&gt; SPI1 + RXRA –&gt; c(‘YPEL3’, ‘INO80E’, ‘DOC2A’) ## seems ‘BCKDK’’ZNF267’ are not in this bundle, and those three are closer to each other</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(igraph)
library(dplyr)

# Define all relevant nodes
focus_gene &lt;- c(&quot;YPEL3&quot;, &quot;INO80E&quot;, &quot;DOC2A&quot;)
core_nodes &lt;- c(&quot;NR2C2&quot;, &quot;ZNF263&quot;, &quot;SP2&quot;, &quot;SPI1&quot;, &quot;RXRA&quot;, focus_gene)

reg_net &lt;- df %&gt;% select(source = TF, target = TG) %&gt;% distinct()
g &lt;- graph_from_data_frame(reg_net, directed = TRUE)
# Extract subgraph with only those nodes
subg &lt;- induced_subgraph(g, vids = V(g)[name %in% core_nodes])

# Set edge colors to highlight direction (optional)
E(subg)$color &lt;- &quot;gray&quot;
V(subg)$color &lt;- ifelse(V(subg)$name %in% focus_gene, &quot;red&quot;, &quot;lightblue&quot;)

# Plot minimal network
pdf(&#39;./figs/GRN_YPEL3_INO80E_DOC2A.pdf&#39;, width = 5, height = 5, family = &quot;sans&quot;)

plot(subg,
     vertex.label.cex = 1,
     vertex.label.family = &quot;sans&quot;,  # Set label fontplot(subg,
     vertex.label.cex = 1,
     vertex.label.color = &quot;black&quot;,
     vertex.color = V(subg)$color,
     vertex.size = 25,
     edge.arrow.size = 0.5,
     layout = layout_with_fr(subg),
     main = &quot;Key Regulatory Pathway to YPEL3 / INO80E / DOC2A&quot;)
dev.off()
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><strong>pdf:</strong> 2</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>multi_gene_list
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'ENSG00000090238'</li><li>'ENSG00000169592'</li><li>'ENSG00000149927'</li><li>'ENSG00000102879'</li><li>'ENSG00000149930'</li><li>'ENSG00000149932'</li><li>'ENSG00000102882'</li><li>'ENSG00000103507'</li><li>'ENSG00000149926'</li><li>'ENSG00000185947'</li><li>'ENSG00000285043'</li><li>'ENSG00000052344'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(igraph)
library(dplyr)

# Define all relevant nodes
focus_gene &lt;- multi_gene_list_name
core_nodes &lt;- c(&quot;NR2C2&quot;, &quot;ZNF263&quot;, &quot;SP2&quot;, &quot;SPI1&quot;, &quot;RXRA&quot;, focus_gene)

reg_net &lt;- df %&gt;% dplyr::select(source = TF, target = TG) %&gt;% distinct()
g &lt;- graph_from_data_frame(reg_net, directed = TRUE)
# Extract subgraph with only those nodes
subg &lt;- induced_subgraph(g, vids = V(g)[name %in% core_nodes])

# Set edge colors to highlight direction (optional)
E(subg)$color &lt;- &quot;gray&quot;
V(subg)$color &lt;- ifelse(V(subg)$name %in% focus_gene, &quot;red&quot;, &quot;lightblue&quot;)
# Use sans font (e.g., Arial) for cleaner output
# pdf(&#39;./figs/GRN_multigene.pdf&#39;, width = 10, height = 10, family = &quot;sans&quot;)

plot(subg,
     vertex.label.cex = 1,
     vertex.label.family = &quot;sans&quot;,  # Set label font
     vertex.label.color = &quot;black&quot;,
     vertex.color = V(subg)$color,
     vertex.size = 20,
     edge.arrow.size = 0.5,
     layout = layout_with_fr(subg),
     main = &quot;Key Regulatory Pathway to Multi Gene&quot;)

dev.off()
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><strong>pdf:</strong> 3</div><a class="reference internal image-reference" href="../../../_images/e267cd117434f261e2febb78401a1f7985b3ff4fe94105c7570a1ffb6a981324.png"><img alt="../../../_images/e267cd117434f261e2febb78401a1f7985b3ff4fe94105c7570a1ffb6a981324.png" src="../../../_images/e267cd117434f261e2febb78401a1f7985b3ff4fe94105c7570a1ffb6a981324.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(igraph)
library(dplyr)

# Define nodes of interest
focus_gene &lt;- multi_gene_list_name
core_nodes &lt;- c(&quot;NR2C2&quot;, &quot;ZNF263&quot;, &quot;SP2&quot;, &quot;SPI1&quot;, &quot;RXRA&quot;, focus_gene)

# Create full network with edge attributes
reg_net &lt;- df %&gt;% dplyr::select(source = TF, target = TG, Regulation) %&gt;% distinct()
g &lt;- graph_from_data_frame(reg_net, directed = TRUE)

# Filter subgraph with selected nodes
subg &lt;- induced_subgraph(g, vids = V(g)[name %in% core_nodes])

# Set vertex color (focus genes in red)
V(subg)$color &lt;- ifelse(V(subg)$name %in% focus_gene, &#39;#F15E4E&#39;, &quot;lightblue&quot;)

# Set edge color by regulation type
E(subg)$color &lt;- ifelse(E(subg)$Regulation == &quot;Repressing&quot;, &quot;blue&quot;,
                        ifelse(E(subg)$Regulation == &quot;Activating&quot;, &quot;orange&quot;, &quot;gray&quot;))

# Optional: adjust edge width by weight or use dashed lines for repressors
# E(subg)$lty &lt;- ifelse(E(subg)$Regulation == &quot;Repressing&quot;, 2, 1)

# Plot the graph
# pdf(&#39;./figs/GRN_multigene.pdf&#39;, width = 9.5, height = 16, family = &quot;sans&quot;)

plot(subg,
     vertex.label.cex = 1,
     vertex.label.family = &quot;sans&quot;,
     vertex.label.color = &quot;black&quot;,
     vertex.color = V(subg)$color,
     vertex.size = 20,
     edge.arrow.size = 0.5,
     edge.color = E(subg)$color,
     layout = layout_with_fr(subg),
     main = &quot;Key Regulatory Pathway to Multi Gene&quot;)
dev.off()
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><strong>pdf:</strong> 3</div><a class="reference internal image-reference" href="../../../_images/0bc096f71e5fcfc8d8aead60414100315caafb7495340772cf8458de5bcfa1c2.png"><img alt="../../../_images/0bc096f71e5fcfc8d8aead60414100315caafb7495340772cf8458de5bcfa1c2.png" src="../../../_images/0bc096f71e5fcfc8d8aead60414100315caafb7495340772cf8458de5bcfa1c2.png" style="width: 720px; height: 720px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>gene_df &lt;- fread(&#39;multi_gene_share_contexts.txt&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Prepare graph
reg_net &lt;- df %&gt;% dplyr::select(source = TF, target = TG, Regulation) %&gt;% distinct()
g &lt;- graph_from_data_frame(reg_net, directed = TRUE)

# Define genes of interest
focus_gene &lt;- gene_df$gene_name
core_nodes &lt;- c(&quot;NR2C2&quot;, &quot;ZNF263&quot;, &quot;SP2&quot;, &quot;SPI1&quot;, &quot;RXRA&quot;, focus_gene)

# Subgraph
subg &lt;- induced_subgraph(g, vids = V(g)[name %in% core_nodes])
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Set vertex color
V(subg)$color &lt;- ifelse(V(subg)$name %in% focus_gene, &#39;#F15E4E&#39;, &quot;lightblue&quot;)

# Match dot size to n_all or other column
gene_size_map &lt;- gene_df %&gt;% dplyr::select(gene_name, n_count_sQTL1) %&gt;% deframe()
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>V(subg)$size &lt;- ifelse(V(subg)$name %in% names(gene_size_map),
                       gene_size_map[V(subg)$name] * 1.5, 10)  # adjust scale as needed

# Edge color by regulation
E(subg)$color &lt;- ifelse(E(subg)$Regulation == &quot;Repressing&quot;, &quot;blue&quot;,
                        ifelse(E(subg)$Regulation == &quot;Activating&quot;, &quot;orange&quot;, &quot;gray&quot;))

# Plot
# pdf(&#39;./figs/GRN_multigene.pdf&#39;, width = 10, height = 16, family = &quot;sans&quot;)

plot(subg,
     vertex.label.cex = 1,
     vertex.label.family = &quot;sans&quot;,
     vertex.label.color = &quot;black&quot;,
     vertex.color = V(subg)$color,
     vertex.size = V(subg)$size,
     edge.arrow.size = 0.5,
     edge.color = E(subg)$color,
     layout = layout_with_fr(subg),
     main = &quot;Key Regulatory Pathway to Multi Gene&quot;)
dev.off()
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><strong>null device:</strong> 1</div><a class="reference internal image-reference" href="../../../_images/3b0c0e3c4fcad2ecb65314b06d5b6645972b19a19d9fea4b1f8a31afb4977458.png"><img alt="../../../_images/3b0c0e3c4fcad2ecb65314b06d5b6645972b19a19d9fea4b1f8a31afb4977458.png" src="../../../_images/3b0c0e3c4fcad2ecb65314b06d5b6645972b19a19d9fea4b1f8a31afb4977458.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>cb &lt;- fread(&#39;/data/analysis_result/ColocBoost/export/summary/ad_xqtl_colocboost_export_filtered.bed.gz&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>cb %&gt;% filter(gene_ID %in% gene_df$gene_ID)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 862 x 13</caption>
<thead>
	<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>a1</th><th scope=col>a2</th><th scope=col>variant_ID</th><th scope=col>gene_ID</th><th scope=col>event_ID</th><th scope=col>cos_ID</th><th scope=col>vcp</th><th scope=col>cos_npc</th><th scope=col>min_npc_outcome</th><th scope=col>coef</th></tr>
	<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>16</td><td>29648903</td><td>29648905</td><td>C</td><td>T</td><td>chr16:29648905:T:C</td><td>ENSG00000149930</td><td>Kellis_Ast_ENSG00000149930; AD_Bellenguez                                                                                                                                                                                                                                                                                                                                                                                                                  </td><td>ENSG00000149930:cos1:y34_y55                             </td><td>0.15731266</td><td>0.5589</td><td>0.4728</td><td>0.00580717409251111;0.00178370917940764                                                                                                                                                                                </td></tr>
	<tr><td>16</td><td>29648903</td><td>29648905</td><td>C</td><td>T</td><td>chr16:29648905:T:C</td><td>ENSG00000149932</td><td>Kellis_Mic_ENSG00000149932; AD_Bellenguez                                                                                                                                                                                                                                                                                                                                                                                                                  </td><td>ENSG00000149932:cos2:y23_y36                             </td><td>0.24246126</td><td>0.8223</td><td>0.7876</td><td>-0.0231917908063331;0.00178356117441108                                                                                                                                                                                </td></tr>
	<tr><td>16</td><td>29649894</td><td>29649896</td><td>T</td><td>C</td><td>chr16:29649896:C:T</td><td>ENSG00000149930</td><td>Kellis_Ast_ENSG00000149930; AD_Bellenguez                                                                                                                                                                                                                                                                                                                                                                                                                  </td><td>ENSG00000149930:cos1:y34_y55                             </td><td>0.83172742</td><td>0.5589</td><td>0.4728</td><td>0.0353373888483476;0.00218527039761535                                                                                                                                                                                 </td></tr>
	<tr><td>16</td><td>29649894</td><td>29649896</td><td>T</td><td>C</td><td>chr16:29649896:C:T</td><td>ENSG00000149932</td><td>Kellis_Mic_ENSG00000149932; AD_Bellenguez                                                                                                                                                                                                                                                                                                                                                                                                                  </td><td>ENSG00000149932:cos2:y23_y36                             </td><td>0.75168536</td><td>0.8223</td><td>0.7876</td><td>-0.0732775806809154;0.00218507518494286                                                                                                                                                                                </td></tr>
	<tr><td>16</td><td>29913099</td><td>29913101</td><td>T</td><td>A</td><td>chr16:29913101:A:T</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01255222</td><td>1.0000</td><td>0.4763</td><td>-0.000181029819259899;-0.00187553089732488;-5.83452183916351e-05;-0.000177289312998618;-0.00211298422809833;-9.38159324959302e-06;0.000738502411154741;-0.00404284995029416;-0.00109699925703438;1.98810503403778e-06  </td></tr>
	<tr><td>16</td><td>29913099</td><td>29913101</td><td>T</td><td>A</td><td>chr16:29913101:A:T</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01345728</td><td>1.0000</td><td>0.4902</td><td>-0.00016961027859452;-0.00179565367026293;-6.09122239689996e-05;-0.000173326287207139;-0.00210355385889389;-8.83992116558586e-06;0.000725860885255428;-0.00377344828907428;-0.000876214192570317;-3.54135631502978e-05 </td></tr>
	<tr><td>16</td><td>29913582</td><td>29913584</td><td>G</td><td>A</td><td>chr16:29913584:A:G</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.03988393</td><td>1.0000</td><td>0.4763</td><td>-0.000197619750517417;-0.00424052143324231;-4.28374601670315e-05;-0.0106749980835483;-0.00887881248104837;-1.51544423515312e-06;0.00131710673284922;-0.00428943482718778;-0.0127293834396922;1.98680807742577e-06      </td></tr>
	<tr><td>16</td><td>29913582</td><td>29913584</td><td>G</td><td>A</td><td>chr16:29913584:A:G</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.04219279</td><td>1.0000</td><td>0.4902</td><td>-0.000185098918477567;-0.00399433567174669;-4.48486912799211e-05;-0.0105732620352061;-0.00892236631942056;-1.45023836152242e-06;0.00130284132371191;-0.00425868586367259;-0.011371840623673;-1.71238066710256e-05      </td></tr>
	<tr><td>16</td><td>29914122</td><td>29914124</td><td>T</td><td>C</td><td>chr16:29914124:C:T</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01049807</td><td>1.0000</td><td>0.4763</td><td>-9.61241829037031e-05;-0.00310526227646874;-4.89967453470016e-05;-0.000138442234215216;-0.00205641905660427;-5.56055063910532e-06;0.00310480039407823;-0.000974228085639173;-0.0012339638300155;2.01158308195266e-06   </td></tr>
	<tr><td>16</td><td>29914122</td><td>29914124</td><td>T</td><td>C</td><td>chr16:29914124:C:T</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01105238</td><td>1.0000</td><td>0.4902</td><td>-9.14207118185384e-05;-0.00295182171639148;-5.12497441327638e-05;-0.000135599201081376;-0.00204821881730936;-5.25853187360262e-06;0.00306243993674793;-0.00096096587189358;-0.00099102652318798;-2.63811640021507e-05  </td></tr>
	<tr><td>16</td><td>29916379</td><td>29916381</td><td>A</td><td>G</td><td>chr16:29916381:G:A</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01352890</td><td>1.0000</td><td>0.4763</td><td>-0.000234293693015442;-0.0040323081246738;-6.60106170830554e-05;-0.000121302850823532;-0.00132397171414951;-6.99490256206466e-06;0.00119417084667759;-0.00128489886715108;-0.0021088691645559;1.91355173243222e-06     </td></tr>
	<tr><td>16</td><td>29916379</td><td>29916381</td><td>A</td><td>G</td><td>chr16:29916381:G:A</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01409537</td><td>1.0000</td><td>0.4902</td><td>-0.000218371299480077;-0.00382023877503929;-6.88905921138878e-05;-0.000118392580749782;-0.00132431517126197;-6.59498683970073e-06;0.00117795143523661;-0.0012488987826348;-0.00172400728876902;-4.53607998344738e-05   </td></tr>
	<tr><td>16</td><td>29920741</td><td>29920743</td><td>T</td><td>C</td><td>chr16:29920743:C:T</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01016240</td><td>1.0000</td><td>0.4763</td><td>-0.000118275908889372;-0.0027010260852915;-6.61282712743761e-05;-0.000104464529345184;-0.00722192893541661;-3.35652946785189e-06;0.000712223249138011;-0.0015576184185088;-0.00123169521117886;1.96747755400499e-06    </td></tr>
	<tr><td>16</td><td>29920741</td><td>29920743</td><td>T</td><td>C</td><td>chr16:29920743:C:T</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01044092</td><td>1.0000</td><td>0.4902</td><td>-0.000111813102676793;-0.00257405058107427;-6.89536153206984e-05;-0.000101793672352537;-0.00714886856662226;-3.17568339940586e-06;0.000700836381987249;-0.001502842514421;-0.00100067005743326;-2.35263219012567e-05   </td></tr>
	<tr><td>16</td><td>29921368</td><td>29921370</td><td>C</td><td>T</td><td>chr16:29921370:T:C</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01039184</td><td>1.0000</td><td>0.4763</td><td>-0.000118943468977775;-0.00281704493855373;-4.37134322463817e-05;-0.000147438788013779;-0.00384717801903351;-4.58556695901399e-06;0.000762607099288168;-0.00157486463305489;-0.00114780089970187;1.90239704272896e-06  </td></tr>
	<tr><td>16</td><td>29921368</td><td>29921370</td><td>C</td><td>T</td><td>chr16:29921370:T:C</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01046597</td><td>1.0000</td><td>0.4902</td><td>-0.00011242503845206;-0.00268282416164886;-4.56967149569239e-05;-0.000144196396170754;-0.00382222691247161;-4.33500558051165e-06;0.000750393980831481;-0.00151893819481654;-0.000929700410107428;-3.59280391160038e-05 </td></tr>
	<tr><td>16</td><td>29925331</td><td>29925333</td><td>G</td><td>A</td><td>chr16:29925333:A:G</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.02705088</td><td>1.0000</td><td>0.4763</td><td>-0.000161306833608242;-0.00304393808205575;-2.38780167769108e-05;-0.00607237307581688;-0.00436093383691707;-2.45121172315028e-06;0.00102231385703854;-0.00515052966746284;-0.00722916381163407;1.92008263973709e-06    </td></tr>
	<tr><td>16</td><td>29925331</td><td>29925333</td><td>G</td><td>A</td><td>chr16:29925333:A:G</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.02757250</td><td>1.0000</td><td>0.4902</td><td>-0.000151712111355889;-0.00288274263306073;-2.51177715068623e-05;-0.00600983078124398;-0.004396460360479;-2.3351341251854e-06;0.00100668245848129;-0.00511024587384637;-0.00640134577593499;-1.97320828880063e-05      </td></tr>
	<tr><td>16</td><td>29926329</td><td>29926331</td><td>A</td><td>G</td><td>chr16:29926331:G:A</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.03868810</td><td>1.0000</td><td>0.4763</td><td>-0.00019874722687781;-0.00442420433829663;-2.12734265475045e-05;-0.00699126213214067;-0.010624190009625;-2.32034356903982e-06;0.00140499394760132;-0.00433258150986836;-0.0117506456934333;1.94453280264462e-06        </td></tr>
	<tr><td>16</td><td>29926329</td><td>29926331</td><td>A</td><td>G</td><td>chr16:29926331:G:A</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.03860387</td><td>1.0000</td><td>0.4902</td><td>-0.000186121150554019;-0.00416467334082071;-2.24200859934291e-05;-0.0069200834042436;-0.0106759917033564;-2.21214189998056e-06;0.00138889169822554;-0.00430046814820824;-0.0105345404355301;-1.84067797128418e-05      </td></tr>
	<tr><td>16</td><td>29927497</td><td>29927499</td><td>G</td><td>A</td><td>chr16:29927499:A:G</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.08734090</td><td>1.0000</td><td>0.4763</td><td>-0.000230835599133975;-0.0046236351433983;-3.24632283801787e-05;-0.0139369305993957;-0.0213902535891134;-4.15548998391261e-06;0.00153788099313446;-0.0155289538064806;-0.0161869063658492;1.92971519411031e-06         </td></tr>
	<tr><td>16</td><td>29927497</td><td>29927499</td><td>G</td><td>A</td><td>chr16:29927499:A:G</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.08988758</td><td>1.0000</td><td>0.4902</td><td>-0.000215338374622751;-0.00435564659404039;-3.41569411515161e-05;-0.0138369007085574;-0.0215387632739918;-3.95415208282211e-06;0.00151681680022772;-0.0148330788378569;-0.014688294223594;-3.3112010658941e-05         </td></tr>
	<tr><td>16</td><td>29927970</td><td>29927972</td><td>C</td><td>T</td><td>chr16:29927972:T:C</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.08945256</td><td>1.0000</td><td>0.4763</td><td>-0.000230835599133975;-0.0046236351433983;-3.24632283801787e-05;-0.0139369305993957;-0.0213902535891134;-4.15548998391261e-06;0.00153788099313446;-0.0155289538064806;-0.0161869063658492;1.89015721067296e-06         </td></tr>
	<tr><td>16</td><td>29927970</td><td>29927972</td><td>C</td><td>T</td><td>chr16:29927972:T:C</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.08526866</td><td>1.0000</td><td>0.4902</td><td>-0.000215338374622751;-0.00435564659404039;-3.41569411515161e-05;-0.0138369007085574;-0.0215387632739918;-3.95415208282211e-06;0.00151681680022772;-0.0148330788378569;-0.014688294223594;-2.34268381283047e-05        </td></tr>
	<tr><td>16</td><td>29928554</td><td>29928556</td><td>G</td><td>A</td><td>chr16:29928556:A:G</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.07008745</td><td>1.0000</td><td>0.4763</td><td>-0.000322688287962524;-0.00195098274862272;-3.61054127218513e-05;-0.0139369305993957;-0.0131091589299103;-5.21075459956067e-06;0.00115960743180972;-0.018860203634941;-0.00884751316534779;1.96144913696335e-06        </td></tr>
	<tr><td>16</td><td>29928554</td><td>29928556</td><td>G</td><td>A</td><td>chr16:29928556:A:G</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.06985281</td><td>1.0000</td><td>0.4902</td><td>-0.000298664577434005;-0.00186704852153259;-3.78394373740358e-05;-0.0138369007085574;-0.0131116805804861;-4.95121209367969e-06;0.00113503764656425;-0.0174975487033417;-0.00782303271037374;-2.47952836660125e-05      </td></tr>
	<tr><td>16</td><td>29932044</td><td>29932046</td><td>C</td><td>T</td><td>chr16:29932046:T:C</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01595077</td><td>1.0000</td><td>0.4763</td><td>-0.000223267349426954;-0.000349101187814905;-8.97318575445761e-05;-0.000221573638876832;-0.00911605883732243;-1.58279614776427e-05;0.000370902557649078;-0.0050341178135953;-0.000823341009574222;1.89925676879644e-06 </td></tr>
	<tr><td>16</td><td>29932044</td><td>29932046</td><td>C</td><td>T</td><td>chr16:29932046:T:C</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.02141824</td><td>1.0000</td><td>0.4902</td><td>-0.000207941371265429;-0.00034804737893259;-9.33652993666418e-05;-0.000217869592038919;-0.00894243959481919;-1.48094696372418e-05;0.000363729815458922;-0.0045925985464067;-0.000651816817929211;0.000759777976282097  </td></tr>
	<tr><td>16</td><td>29932136</td><td>29932138</td><td>A</td><td>C</td><td>chr16:29932138:C:A</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Bellenguez        </td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01526052</td><td>1.0000</td><td>0.4763</td><td>-0.000248608097422805;-0.000451999598955464;-0.000121134085206925;-0.000162822409096747;-0.00789088259648098;-1.3784224432586e-05;0.000322171744252936;-0.00403574296121077;-0.000864430086362348;1.95921484055518e-06 </td></tr>
	<tr><td>16</td><td>29932136</td><td>29932138</td><td>A</td><td>C</td><td>chr16:29932138:C:A</td><td>ENSG00000149932</td><td>AC_lf2_sQTL_chr16:29962232:29963106:clu_110638_+:NE:ENSG00000149932; AC_lf2_sQTL_chr16:29962232:29963109:clu_110638_+:NE:ENSG00000149932; DLPFC_lf2_sQTL_chr16:29962232:29963109:clu_131719_+:NE:ENSG00000149932; DeJager_Ast_ENSG00000149932; Kellis_Ast_ENSG00000149932; Kellis_OPC_ENSG00000149932; PCC_lf2_sQTL_chr16:29962132:29962390:clu_101665_+:NE:ENSG00000149932; ROSMAP_AC_ENSG00000149932; ROSMAP_DLPFC_ENSG00000149932; AD_Wightman_Full_2021</td><td>ENSG00000149932:cos3:y3_y4_y6_y14_y20_y24_y29_y32_y33_y36</td><td>0.01331298</td><td>1.0000</td><td>0.4902</td><td>-0.00023122145308121;-0.000447845620723006;-0.000125864690805603;-0.000159911375806614;-0.00775749535094516;-1.29268518132235e-05;0.000315568923608905;-0.00370028808193019;-0.000686129760381247;-1.59416883523183e-05</td></tr>
	<tr><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr>
	<tr><td>16</td><td>30102801</td><td>30102803</td><td>T</td><td>C</td><td>chr16:30102803:C:T</td><td>ENSG00000090238</td><td>AC_lf2_sQTL_chr16:30094897:30095251:clu_113741_-:PR:ENSG00000090238; AC_lf2_sQTL_chr16:30095335:30096096:clu_113742_-:NE:ENSG00000090238; AC_lf2_sQTL_chr16:30095434:30096096:clu_113742_-:PR:ENSG00000090238; DLPFC_lf2_sQTL_chr16:30094897:30095251:clu_135246_-:PR:ENSG00000090238; DLPFC_lf2_sQTL_chr16:30095335:30096096:clu_135247_-:NE:ENSG00000090238; DLPFC_lf2_sQTL_chr16:30095434:30096096:clu_135247_-:PR:ENSG00000090238; DeJager_Exc_ENSG00000090238; DeJager_Inh_ENSG00000090238; DeJager_Oli_ENSG00000090238; Kellis_Ast_ENSG00000090238; Kellis_Exc_ENSG00000090238; Kellis_Inh_ENSG00000090238; Kellis_Mic_ENSG00000090238; Kellis_Oli_ENSG00000090238; PCC_lf2_sQTL_chr16:30094897:30095251:clu_104519_-:PR:ENSG00000090238; PCC_lf2_sQTL_chr16:30095335:30096096:clu_104520_-:NE:ENSG00000090238; ROSMAP_DLPFC_ENSG00000090238; ROSMAP_PCC_ENSG00000090238; AD_Wightman_ExcludingUKBand23andME_2021</td><td>ENSG00000090238:cos3:y2_y4_y6_y15_y17_y19_y22_y23_y26_y27_y28_y29_y30_y32_y36_y38_y43_y45_y46</td><td>0.041244601</td><td>1.0000</td><td>0.5971</td><td>-0.0183200458018149;0.0117210546305083;-0.0113642624046988;-0.00948913106073303;0.0030419186064217;-0.00109356158009929;0.00153944281354599;0.0111313614036887;0.000410406459955326;0.000468023133659571;0.00297073913759643;0.00744177886817723;0.00117792813779476;0.055126352276103;-0.00647516802498812;0.00182938103950685;0.021220739718841;0.00081476596377729;-0.000237534000716698   </td></tr>
	<tr><td>16</td><td>30102801</td><td>30102803</td><td>T</td><td>C</td><td>chr16:30102803:C:T</td><td>ENSG00000090238</td><td>AC_lf2_sQTL_chr16:30094897:30095251:clu_113741_-:PR:ENSG00000090238; AC_lf2_sQTL_chr16:30095335:30096096:clu_113742_-:NE:ENSG00000090238; AC_lf2_sQTL_chr16:30095434:30096096:clu_113742_-:PR:ENSG00000090238; DLPFC_lf2_sQTL_chr16:30094897:30095251:clu_135246_-:PR:ENSG00000090238; DLPFC_lf2_sQTL_chr16:30095335:30096096:clu_135247_-:NE:ENSG00000090238; DLPFC_lf2_sQTL_chr16:30095434:30096096:clu_135247_-:PR:ENSG00000090238; DeJager_Exc_ENSG00000090238; DeJager_Inh_ENSG00000090238; DeJager_Oli_ENSG00000090238; Kellis_Ast_ENSG00000090238; Kellis_Exc_ENSG00000090238; Kellis_Inh_ENSG00000090238; Kellis_Mic_ENSG00000090238; Kellis_Oli_ENSG00000090238; PCC_lf2_sQTL_chr16:30094897:30095251:clu_104519_-:PR:ENSG00000090238; PCC_lf2_sQTL_chr16:30095335:30096096:clu_104520_-:NE:ENSG00000090238; ROSMAP_DLPFC_ENSG00000090238; ROSMAP_PCC_ENSG00000090238; AD_Wightman_Full_2021                  </td><td>ENSG00000090238:cos4:y2_y4_y6_y15_y17_y19_y22_y23_y26_y27_y28_y29_y30_y32_y36_y38_y43_y45_y46</td><td>0.037311021</td><td>1.0000</td><td>0.5989</td><td>-0.0184008038080819;0.0118202001314863;-0.0118244073889315;-0.00957053745781502;0.00303404247255839;-0.00104435960140226;0.00150350336767063;0.0111282879057801;0.000415715935646294;0.000468490538138031;0.00294659132738121;0.00746286056668423;0.00114599047199578;0.0551190736820339;-0.00648983606424825;0.00182875326019856;0.0213005478186032;0.000805307013927021;-0.00069841150439068</td></tr>
	<tr><td>16</td><td>30102801</td><td>30102803</td><td>T</td><td>C</td><td>chr16:30102803:C:T</td><td>ENSG00000102882</td><td>ROSMAP_Monocyte_ENSG00000102882; AD_Wightman_ExcludingUKBand23andME_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               </td><td>ENSG00000102882:cos7:y54_y56                                                                 </td><td>0.007747075</td><td>0.7888</td><td>0.7318</td><td>-0.00238287827894969;-0.000185499855696214                                                                                                                                                                                                                                                                                                                                                    </td></tr>
	<tr><td>16</td><td>30102801</td><td>30102803</td><td>T</td><td>C</td><td>chr16:30102803:C:T</td><td>ENSG00000103507</td><td>ROSMAP_DLPFC_ENSG00000103507; AD_Wightman_ExcludingUKBand23andME_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  </td><td>ENSG00000103507:cos3:y24_y27                                                                 </td><td>0.032195862</td><td>0.7710</td><td>0.7064</td><td>-0.00147524634933275;-0.000209522886769491                                                                                                                                                                                                                                                                                                                                                    </td></tr>
	<tr><td>16</td><td>30102801</td><td>30102803</td><td>T</td><td>C</td><td>chr16:30102803:C:T</td><td>ENSG00000169592</td><td>Kellis_Mic_ENSG00000169592; AD_Wightman_ExcludingUKBand23andME_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    </td><td>ENSG00000169592:cos4:y31_y44                                                                 </td><td>0.012456216</td><td>0.8213</td><td>0.7951</td><td>-0.000747315040504783;-0.000204910385055593                                                                                                                                                                                                                                                                                                                                                   </td></tr>
	<tr><td>16</td><td>30105930</td><td>30105932</td><td>A</td><td>G</td><td>chr16:30105932:G:A</td><td>ENSG00000149927</td><td>DLPFC_Bennett_pQTL_ENSG00000149927|Q14183; DLPFC_lf2_sQTL_chr16:30006331:30006595:clu_135221_-:UP:ENSG00000149927; DeJager_Exc_ENSG00000149927; AD_Wightman_ExcludingUKBand23andME_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                </td><td>ENSG00000149927:cos8:y15_y16_y44_y80                                                         </td><td>0.046978859</td><td>0.9625</td><td>0.3499</td><td>0.0114541974475276;-0.000788334588628517;-0.00813198850975073;-0.000101951246318738                                                                                                                                                                                                                                                                                                           </td></tr>
	<tr><td>16</td><td>30105930</td><td>30105932</td><td>A</td><td>G</td><td>chr16:30105932:G:A</td><td>ENSG00000169592</td><td>Kellis_Mic_ENSG00000169592; AD_Wightman_ExcludingUKBand23andME_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    </td><td>ENSG00000169592:cos4:y31_y44                                                                 </td><td>0.005417553</td><td>0.8213</td><td>0.7951</td><td>-0.00191982868372136;-5.72422740501575e-05                                                                                                                                                                                                                                                                                                                                                    </td></tr>
	<tr><td>16</td><td>30105930</td><td>30105932</td><td>A</td><td>G</td><td>chr16:30105932:G:A</td><td>ENSG00000185947</td><td>PCC_lf2_sQTL_chr16:31873969:31884497:clu_101907_+:PR:ENSG00000185947; PCC_lf2_sQTL_chr16:31880821:31884497:clu_101907_+:UP:ENSG00000185947; AD_Bellenguez                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              </td><td>ENSG00000185947:cos3:y22_y23_y28                                                             </td><td>0.134491707</td><td>0.6603</td><td>0.3889</td><td>-0.00908191986949523;0.00726242876625511;-4.32220395615334e-05                                                                                                                                                                                                                                                                                                                                </td></tr>
	<tr><td>16</td><td>30107022</td><td>30107024</td><td>C</td><td>T</td><td>chr16:30107024:T:C</td><td>ENSG00000102882</td><td>ROSMAP_Monocyte_ENSG00000102882; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 </td><td>ENSG00000102882:cos5:y54_y56                                                                 </td><td>0.062950985</td><td>0.7514</td><td>0.7514</td><td>-0.00182216921061399;0.0162480737742663                                                                                                                                                                                                                                                                                                                                                       </td></tr>
	<tr><td>16</td><td>30107022</td><td>30107024</td><td>C</td><td>T</td><td>chr16:30107024:T:C</td><td>ENSG00000103507</td><td>ROSMAP_DLPFC_ENSG00000103507; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    </td><td>ENSG00000103507:cos4:y24_y27                                                                 </td><td>0.054290034</td><td>0.6915</td><td>0.6915</td><td>-0.00084047906441162;0.0162311123385538                                                                                                                                                                                                                                                                                                                                                       </td></tr>
	<tr><td>16</td><td>30107022</td><td>30107024</td><td>C</td><td>T</td><td>chr16:30107024:T:C</td><td>ENSG00000149926</td><td>AC_lf2_sQTL_chr16:30025821:30026608:clu_113727_-:PR:ENSG00000149926; DLPFC_lf2_sQTL_chr16:30025821:30026608:clu_135233_-:PR:ENSG00000149926; DLPFC_lf2_sQTL_chr16:30025769:30026608:clu_135233_-:PR:ENSG00000149926; PCC_lf2_sQTL_chr16:30025769:30026608:clu_104506_-:UP:ENSG00000149926; PCC_lf2_sQTL_chr16:30025821:30026608:clu_104506_-:PR:ENSG00000149926; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 </td><td>ENSG00000149926:cos1:y1_y6_y7_y12_y13_y19                                                    </td><td>0.024373988</td><td>1.0000</td><td>0.8713</td><td>0.0125089533814097;0.00362641720213083;-0.00325266260738003;-0.00148885737362075;0.00108176989959191;0.0166594281575467                                                                                                                                                                                                                                                                       </td></tr>
	<tr><td>16</td><td>31106375</td><td>31106377</td><td>T</td><td>C</td><td>chr16:31106377:C:T</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.018932648</td><td>0.5739</td><td>0.4889</td><td>0.00429981027381748;-0.000126445863933832                                                                                                                                                                                                                                                                                                                                                     </td></tr>
	<tr><td>16</td><td>31111248</td><td>31111250</td><td>T</td><td>C</td><td>chr16:31111250:C:T</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Excluding23andMe_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.363762563</td><td>0.7782</td><td>0.7292</td><td>0.00457618597920208;-0.00711872657179389                                                                                                                                                                                                                                                                                                                                                      </td></tr>
	<tr><td>16</td><td>31111248</td><td>31111250</td><td>T</td><td>C</td><td>chr16:31111250:C:T</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.328433430</td><td>0.5739</td><td>0.4889</td><td>0.00457300135483516;-0.00516124380587855                                                                                                                                                                                                                                                                                                                                                      </td></tr>
	<tr><td>16</td><td>31114998</td><td>31115000</td><td>A</td><td>C</td><td>chr16:31115000:C:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Excluding23andMe_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.024730794</td><td>0.7782</td><td>0.7292</td><td>0.00372020763353801;-0.000205668993208773                                                                                                                                                                                                                                                                                                                                                     </td></tr>
	<tr><td>16</td><td>31114998</td><td>31115000</td><td>A</td><td>C</td><td>chr16:31115000:C:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.026767091</td><td>0.5739</td><td>0.4889</td><td>0.00371732364817883;-0.000234711193758497                                                                                                                                                                                                                                                                                                                                                     </td></tr>
	<tr><td>16</td><td>31120927</td><td>31120929</td><td>G</td><td>A</td><td>chr16:31120929:A:G</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Excluding23andMe_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.092123856</td><td>0.7782</td><td>0.7292</td><td>0.00740894767553152;-0.000422376433200108                                                                                                                                                                                                                                                                                                                                                     </td></tr>
	<tr><td>16</td><td>31120927</td><td>31120929</td><td>G</td><td>A</td><td>chr16:31120929:A:G</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.083831482</td><td>0.5739</td><td>0.4889</td><td>0.00740465162504632;-0.00047598744073794                                                                                                                                                                                                                                                                                                                                                      </td></tr>
	<tr><td>16</td><td>31121339</td><td>31121341</td><td>A</td><td>G</td><td>chr16:31121341:G:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Excluding23andMe_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.090658977</td><td>0.7782</td><td>0.7292</td><td>0.00740894767553152;-0.00034117160368764                                                                                                                                                                                                                                                                                                                                                      </td></tr>
	<tr><td>16</td><td>31121339</td><td>31121341</td><td>A</td><td>G</td><td>chr16:31121341:G:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.103017048</td><td>0.5739</td><td>0.4889</td><td>0.00740465162504632;-0.000626514946054708                                                                                                                                                                                                                                                                                                                                                     </td></tr>
	<tr><td>16</td><td>31121777</td><td>31121779</td><td>A</td><td>G</td><td>chr16:31121779:G:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Excluding23andMe_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.100649613</td><td>0.7782</td><td>0.7292</td><td>0.00689465242043429;-0.000517378686896869                                                                                                                                                                                                                                                                                                                                                     </td></tr>
	<tr><td>16</td><td>31121777</td><td>31121779</td><td>A</td><td>G</td><td>chr16:31121779:G:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.111691722</td><td>0.5739</td><td>0.4889</td><td>0.00689064659239464;-0.00076713593027436                                                                                                                                                                                                                                                                                                                                                      </td></tr>
	<tr><td>16</td><td>31122126</td><td>31122128</td><td>T</td><td>C</td><td>chr16:31122128:C:T</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Excluding23andMe_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.106459402</td><td>0.7782</td><td>0.7292</td><td>0.00740894767553152;-0.000506683422763838                                                                                                                                                                                                                                                                                                                                                     </td></tr>
	<tr><td>16</td><td>31122126</td><td>31122128</td><td>T</td><td>C</td><td>chr16:31122128:C:T</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.119344133</td><td>0.5739</td><td>0.4889</td><td>0.00740465162504632;-0.000762290896786018                                                                                                                                                                                                                                                                                                                                                     </td></tr>
	<tr><td>16</td><td>31122736</td><td>31122738</td><td>A</td><td>G</td><td>chr16:31122738:G:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Excluding23andMe_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.036896622</td><td>0.7782</td><td>0.7292</td><td>0.00433513916339601;-0.000394121743229941                                                                                                                                                                                                                                                                                                                                                     </td></tr>
	<tr><td>16</td><td>31122736</td><td>31122738</td><td>A</td><td>G</td><td>chr16:31122738:G:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.040239378</td><td>0.5739</td><td>0.4889</td><td>0.00433196653723863;-0.000338190880342647                                                                                                                                                                                                                                                                                                                                                     </td></tr>
	<tr><td>16</td><td>31142823</td><td>31142825</td><td>A</td><td>G</td><td>chr16:31142825:G:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Excluding23andMe_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.100594526</td><td>0.7782</td><td>0.7292</td><td>-0.0161406275918419;0.000290439623132993                                                                                                                                                                                                                                                                                                                                                      </td></tr>
	<tr><td>16</td><td>31142823</td><td>31142825</td><td>A</td><td>G</td><td>chr16:31142825:G:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.091852574</td><td>0.5739</td><td>0.4889</td><td>-0.0161330011114674;0.000230227218855795                                                                                                                                                                                                                                                                                                                                                      </td></tr>
	<tr><td>16</td><td>31143035</td><td>31143037</td><td>A</td><td>G</td><td>chr16:31143037:G:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Excluding23andMe_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.038883252</td><td>0.7782</td><td>0.7292</td><td>0.0116154175303511;7.53005953243271e-06                                                                                                                                                                                                                                                                                                                                                       </td></tr>
	<tr><td>16</td><td>31143035</td><td>31143037</td><td>A</td><td>G</td><td>chr16:31143037:G:A</td><td>ENSG00000052344</td><td>ROSMAP_AC_ENSG00000052344; AD_Wightman_Full_2021                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       </td><td>ENSG00000052344:cos1:y1_y5                                                                   </td><td>0.037991644</td><td>0.5739</td><td>0.4889</td><td>0.0116094047733889;-0.000105917553085748                                                                                                                                                                                                                                                                                                                                                      </td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="multigene-grn-stx4">
<h3>Multigene GRN - STX4<a class="headerlink" href="#multigene-grn-stx4" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>df &lt;- readRDS(&#39;all_GRN.rds&#39;)
reg_net &lt;- df %&gt;% select(source = TF, target = TG) %&gt;% distinct()
g &lt;- graph_from_data_frame(reg_net, directed = TRUE)


multi_gene_list_name &lt;- c(&#39;STX4&#39;, &#39;KAT8&#39;, &#39;PRSS36&#39;)
valid_genes &lt;- multi_gene_list_name[multi_gene_list_name %in% V(g)$name]

tf_genes &lt;- list()
for (gene in valid_genes) {
  neighbors_g &lt;- ego(g, order = 1, nodes = gene, mode = &quot;in&quot;)[[1]]
  tf_genes[[gene]] &lt;- setdiff(names(neighbors_g), gene)
}
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>Reduce(intersect, tf_genes)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'IRF1'</li><li>'ESR1'</li><li>'FOXP1'</li><li>'MEF2A'</li><li>'PBX3'</li><li>'MAFK'</li><li>'ZNF460'</li><li>'MEF2C'</li><li>'SREBF1'</li><li>'RARG'</li><li>'ZNF140'</li><li>'ZNF135'</li><li>'PKNOX1'</li><li>'ZNF740'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Convert list to binary presence/absence data frame
gene_matrix &lt;- fromList(tf_genes)

# Create the UpSet plot
upset(gene_matrix, sets = names(tf_genes), order.by = &quot;freq&quot;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/70a7991238f4c2c9da6e2d2253d0652803d0840ed8099fed86a21ce84e5e00cf.png"><img alt="../../../_images/70a7991238f4c2c9da6e2d2253d0652803d0840ed8099fed86a21ce84e5e00cf.png" src="../../../_images/70a7991238f4c2c9da6e2d2253d0652803d0840ed8099fed86a21ce84e5e00cf.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(igraph)
library(dplyr)

# Define nodes of interest
focus_gene &lt;- multi_gene_list_name
core_nodes &lt;- c(Reduce(intersect, tf_genes), focus_gene)

# Create full network with edge attributes
reg_net &lt;- df %&gt;% dplyr::select(source = TF, target = TG, Regulation) %&gt;% distinct()
g &lt;- graph_from_data_frame(reg_net, directed = TRUE)

# Filter subgraph with selected nodes
subg &lt;- induced_subgraph(g, vids = V(g)[name %in% core_nodes])

# Set vertex color (focus genes in red)
V(subg)$color &lt;- ifelse(V(subg)$name %in% focus_gene, &#39;#F15E4E&#39;, &quot;lightblue&quot;)

# Set edge color by regulation type
E(subg)$color &lt;- ifelse(E(subg)$Regulation == &quot;Repressing&quot;, &quot;blue&quot;,
                        ifelse(E(subg)$Regulation == &quot;Activating&quot;, &quot;orange&quot;, &quot;gray&quot;))

# Optional: adjust edge width by weight or use dashed lines for repressors
# E(subg)$lty &lt;- ifelse(E(subg)$Regulation == &quot;Repressing&quot;, 2, 1)

# Plot the graph
# pdf(&#39;./figs/GRN_multigene_STX4.pdf&#39;, width = 9.5, height = 16, family = &quot;sans&quot;)

plot(subg,
     vertex.label.cex = 1,
     vertex.label.family = &quot;sans&quot;,
     vertex.label.color = &quot;black&quot;,
     vertex.color = V(subg)$color,
     vertex.size = 20,
     edge.arrow.size = 0.5,
     edge.color = E(subg)$color,
     layout = layout_with_fr(subg),
     main = &quot;Key Regulatory Pathway to Multi Gene&quot;)
dev.off()
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><strong>null device:</strong> 1</div><a class="reference internal image-reference" href="../../../_images/f50230bbaf4b25c17683d655edae808774a2e7a4e09d00c0394e8ad71df24498.png"><img alt="../../../_images/f50230bbaf4b25c17683d655edae808774a2e7a4e09d00c0394e8ad71df24498.png" src="../../../_images/f50230bbaf4b25c17683d655edae808774a2e7a4e09d00c0394e8ad71df24498.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
</section>
<section id="multi-gene-trans-function">
<h2>Multi Gene trans function<a class="headerlink" href="#multi-gene-trans-function" title="Link to this heading">#</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm1_all &lt;- readLines(&#39;Top50_genes_Program1.txt&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm1_all_bp_ego &lt;- enrichGO(
  gene          = progm1_all,
  OrgDb         = org.Hs.eg.db,
  keyType       = &quot;SYMBOL&quot;,
  ont           = &quot;BP&quot;,
  pAdjustMethod = &quot;BH&quot;,
  pvalueCutoff  = 1,
  qvalueCutoff  = 1
)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm1_all_bp_ego %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 x 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:2001259</th><td>GO:2001259</td><td>positive regulation of cation channel activity</td><td>6/385 </td><td>49/18870 </td><td>0.0004624001</td><td>0.3333622</td><td>0.3175203</td><td>CTSS/NIPSNAP2/CRACR2A/ANTKMT/STIMATE/SCN1B                                                                      </td><td> 6</td></tr>
	<tr><th scope=row>GO:0006406</th><td>GO:0006406</td><td>mRNA export from nucleus                      </td><td>7/385 </td><td>70/18870 </td><td>0.0005534691</td><td>0.3333622</td><td>0.3175203</td><td>HNRNPA2B1/THOC5/CHTOP/NSUN2/SUPT6H/UPF2/SMG7                                                                    </td><td> 7</td></tr>
	<tr><th scope=row>GO:0006405</th><td>GO:0006405</td><td>RNA export from nucleus                       </td><td>8/385 </td><td>92/18870 </td><td>0.0005844065</td><td>0.3333622</td><td>0.3175203</td><td>HNRNPA2B1/THOC5/CHTOP/NSUN2/NOL6/SUPT6H/UPF2/SMG7                                                               </td><td> 8</td></tr>
	<tr><th scope=row>GO:0051168</th><td>GO:0051168</td><td>nuclear export                                </td><td>11/385</td><td>168/18870</td><td>0.0006858634</td><td>0.3333622</td><td>0.3175203</td><td>STRADB/HNRNPA2B1/THOC5/CHTOP/NSUN2/NOL6/RITA1/SUPT6H/RANBP2/UPF2/SMG7                                           </td><td>11</td></tr>
	<tr><th scope=row>GO:0006644</th><td>GO:0006644</td><td>phospholipid metabolic process                </td><td>18/385</td><td>375/18870</td><td>0.0007437289</td><td>0.3333622</td><td>0.3175203</td><td>GPX4/PLCH2/PIGS/PLCB1/ETNK2/PROCA1/INPP5D/CEPT1/PGP/PIP4K2C/HMGCS1/FDPS/FDFT1/INPP5J/P3R3URF/PIK3R3/PTPMT1/ITPKA</td><td>18</td></tr>
	<tr><th scope=row>GO:0009060</th><td>GO:0009060</td><td>aerobic respiration                           </td><td>12/385</td><td>197/18870</td><td>0.0007552532</td><td>0.3333622</td><td>0.3175203</td><td>NIPSNAP2/ANTKMT/NDUFA9/NDUFA7/SDHAF2/UQCC3/ARL2/NDUFB9/NDUFS1/IDH3B/UQCRC1/ACO2                                 </td><td>12</td></tr>
</tbody>
</table>
</div></div>
</div>
<section id="program1-go">
<h3>program1 GO<a class="headerlink" href="#program1-go" title="Link to this heading">#</a></h3>
<section id="split">
<h4>split<a class="headerlink" href="#split" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm1 &lt;- fread(&#39;Top50_genes_Program1_split.txt&#39;)
progm1 %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 x 9</caption>
<thead>
	<tr><th scope=col>Mic_V1</th><th scope=col>Exc_V1</th><th scope=col>Inh_V1</th><th scope=col>Ast_V1</th><th scope=col>Oli_V1</th><th scope=col>OPC_V1</th><th scope=col>DLPFC_V1</th><th scope=col>AC_V1</th><th scope=col>PCC_V1</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>MAP2K2 </td><td>ZBTB37 </td><td>B3GNT5    </td><td>ADCK5  </td><td>UQCC3 </td><td>RANBP2</td><td>BCKDK   </td><td>ARHGEF1</td><td>FHAD1 </td></tr>
	<tr><td>HLA-DMA</td><td>SMIM12 </td><td>AL162231.1</td><td>EIF5A  </td><td>MAGEF1</td><td>BIN1  </td><td>FBXO10  </td><td>DTX3   </td><td>TPM3  </td></tr>
	<tr><td>CPLX1  </td><td>TMED7  </td><td>MTHFS     </td><td>GNG3   </td><td>FUCA1 </td><td>TET3  </td><td>COLQ    </td><td>PENK   </td><td>HERC2 </td></tr>
	<tr><td>TTYH1  </td><td>PGBD5  </td><td>MCM7      </td><td>BTN3A1 </td><td>PRR7  </td><td>KIF1A </td><td>ZFC3H1  </td><td>COG1   </td><td>FGF22 </td></tr>
	<tr><td>KLHDC4 </td><td>FAM229A</td><td>RILPL2    </td><td>SDHAF2 </td><td>CMTM1 </td><td>SAP18 </td><td>TMEM151A</td><td>QRICH1 </td><td>CADPS2</td></tr>
	<tr><td>TPGS2  </td><td>FSCN1  </td><td>FBXO21    </td><td>ZFYVE19</td><td>GSTT2B</td><td>GCFC2 </td><td>PKD1L3  </td><td>SLC26A2</td><td>PEX6  </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Initialize empty data frame
progm1_df &lt;- data.frame()

# Loop through each context (column) in progm1
for(context in colnames(progm1)){
    message(context)
  # Perform GO enrichment and store results with context label
  res &lt;- enrichGO(
    gene          = progm1[[context]],  # use [[ ]] to access column by name
    OrgDb         = org.Hs.eg.db,
    keyType       = &quot;SYMBOL&quot;,
    ont           = &quot;ALL&quot;,
    pAdjustMethod = &quot;BH&quot;,
    pvalueCutoff  = 1,
    qvalueCutoff  = 1
  ) %&gt;% 
    as.data.frame() %&gt;% 
    mutate(context = context)  # add context info
  
  # Append to cumulative results
  progm1_df &lt;- rbind(progm1_df, res)
}
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Mic_V1

Exc_V1

Inh_V1

Ast_V1

Oli_V1

OPC_V1

DLPFC_V1

AC_V1

PCC_V1
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm1_df %&gt;% filter(qvalue &lt; 0.05,ONTOLOGY == &#39;BP&#39;) %&gt;% arrange(qvalue)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 58 x 11</caption>
<thead>
	<tr><th></th><th scope=col>ONTOLOGY</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th><th scope=col>context</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:00028241</th><td>BP</td><td>GO:0002824</td><td>positive regulation of adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains</td><td>5/43</td><td>124/18870</td><td>8.905590e-06</td><td>0.007593824</td><td>0.005911059</td><td>PVR/IL12A/ADA/HFE/EXOSC6      </td><td>5</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00028211</th><td>BP</td><td>GO:0002821</td><td>positive regulation of adaptive immune response                                                                                                 </td><td>5/43</td><td>130/18870</td><td>1.120860e-05</td><td>0.007593824</td><td>0.005911059</td><td>PVR/IL12A/ADA/HFE/EXOSC6      </td><td>5</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0046037</th><td>BP</td><td>GO:0046037</td><td>GMP metabolic process                                                                                                                           </td><td>3/43</td><td>27/18870 </td><td>3.103176e-05</td><td>0.014016011</td><td>0.010910113</td><td>CARD11/ADA/GMPR2              </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00067211</th><td>BP</td><td>GO:0006721</td><td>terpenoid metabolic process                                                                                                                     </td><td>4/43</td><td>98/18870 </td><td>7.226141e-05</td><td>0.017904959</td><td>0.013937284</td><td>LRP1/HMGCS1/FDPS/FDFT1        </td><td>4</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00028222</th><td>BP</td><td>GO:0002822</td><td>regulation of adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains         </td><td>5/43</td><td>193/18870</td><td>7.461184e-05</td><td>0.017904959</td><td>0.013937284</td><td>PVR/IL12A/ADA/HFE/EXOSC6      </td><td>5</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0001916</th><td>BP</td><td>GO:0001916</td><td>positive regulation of T cell mediated cytotoxicity                                                                                             </td><td>3/43</td><td>40/18870 </td><td>1.026764e-04</td><td>0.017904959</td><td>0.013937284</td><td>PVR/IL12A/HFE                 </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00091671</th><td>BP</td><td>GO:0009167</td><td>purine ribonucleoside monophosphate metabolic process                                                                                           </td><td>3/43</td><td>40/18870 </td><td>1.026764e-04</td><td>0.017904959</td><td>0.013937284</td><td>CARD11/ADA/GMPR2              </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00028192</th><td>BP</td><td>GO:0002819</td><td>regulation of adaptive immune response                                                                                                          </td><td>5/43</td><td>208/18870</td><td>1.061822e-04</td><td>0.017904959</td><td>0.013937284</td><td>PVR/IL12A/ADA/HFE/EXOSC6      </td><td>5</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00091261</th><td>BP</td><td>GO:0009126</td><td>purine nucleoside monophosphate metabolic process                                                                                               </td><td>3/43</td><td>42/18870 </td><td>1.189259e-04</td><td>0.017904959</td><td>0.013937284</td><td>CARD11/ADA/GMPR2              </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00466342</th><td>BP</td><td>GO:0046634</td><td>regulation of alpha-beta T cell activation                                                                                                      </td><td>4/43</td><td>116/18870</td><td>1.390432e-04</td><td>0.018840350</td><td>0.014665395</td><td>IL12A/CARD11/ADA/HFE          </td><td>4</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0001914</th><td>BP</td><td>GO:0001914</td><td>regulation of T cell mediated cytotoxicity                                                                                                      </td><td>3/43</td><td>50/18870 </td><td>2.004828e-04</td><td>0.021248027</td><td>0.016539541</td><td>PVR/IL12A/HFE                 </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00027081</th><td>BP</td><td>GO:0002708</td><td>positive regulation of lymphocyte mediated immunity                                                                                             </td><td>4/43</td><td>129/18870</td><td>2.092520e-04</td><td>0.021248027</td><td>0.016539541</td><td>PVR/IL12A/HFE/EXOSC6          </td><td>4</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0002420</th><td>BP</td><td>GO:0002420</td><td>natural killer cell mediated cytotoxicity directed against tumor cell target                                                                    </td><td>2/43</td><td>10/18870 </td><td>2.256187e-04</td><td>0.021248027</td><td>0.016539541</td><td>PVR/IL12A                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0002855</th><td>BP</td><td>GO:0002855</td><td>regulation of natural killer cell mediated immune response to tumor cell                                                                        </td><td>2/43</td><td>10/18870 </td><td>2.256187e-04</td><td>0.021248027</td><td>0.016539541</td><td>PVR/IL12A                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00067201</th><td>BP</td><td>GO:0006720</td><td>isoprenoid metabolic process                                                                                                                    </td><td>4/43</td><td>133/18870</td><td>2.352180e-04</td><td>0.021248027</td><td>0.016539541</td><td>LRP1/HMGCS1/FDPS/FDFT1        </td><td>4</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0002423</th><td>BP</td><td>GO:0002423</td><td>natural killer cell mediated immune response to tumor cell                                                                                      </td><td>2/43</td><td>11/18870 </td><td>2.753572e-04</td><td>0.021947589</td><td>0.017084082</td><td>PVR/IL12A                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:1905165</th><td>BP</td><td>GO:1905165</td><td>regulation of lysosomal protein catabolic process                                                                                               </td><td>2/43</td><td>11/18870 </td><td>2.753572e-04</td><td>0.021947589</td><td>0.017084082</td><td>LRP1/ATP13A2                  </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00091612</th><td>BP</td><td>GO:0009161</td><td>ribonucleoside monophosphate metabolic process                                                                                                  </td><td>3/43</td><td>59/18870 </td><td>3.278124e-04</td><td>0.021983605</td><td>0.017112117</td><td>CARD11/ADA/GMPR2              </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0001913</th><td>BP</td><td>GO:0001913</td><td>T cell mediated cytotoxicity                                                                                                                    </td><td>3/43</td><td>60/18870 </td><td>3.445190e-04</td><td>0.021983605</td><td>0.017112117</td><td>PVR/IL12A/HFE                 </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00066951</th><td>BP</td><td>GO:0006695</td><td>cholesterol biosynthetic process                                                                                                                </td><td>3/43</td><td>60/18870 </td><td>3.445190e-04</td><td>0.021983605</td><td>0.017112117</td><td>HMGCS1/FDPS/FDFT1             </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:19026531</th><td>BP</td><td>GO:1902653</td><td>secondary alcohol biosynthetic process                                                                                                          </td><td>3/43</td><td>60/18870 </td><td>3.445190e-04</td><td>0.021983605</td><td>0.017112117</td><td>HMGCS1/FDPS/FDFT1             </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:19026521</th><td>BP</td><td>GO:1902652</td><td>secondary alcohol metabolic process                                                                                                             </td><td>4/43</td><td>150/18870</td><td>3.719743e-04</td><td>0.021983605</td><td>0.017112117</td><td>IDH3B/HMGCS1/FDPS/FDFT1       </td><td>4</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00027052</th><td>BP</td><td>GO:0002705</td><td>positive regulation of leukocyte mediated immunity                                                                                              </td><td>4/43</td><td>151/18870</td><td>3.814667e-04</td><td>0.021983605</td><td>0.017112117</td><td>PVR/IL12A/HFE/EXOSC6          </td><td>4</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:1904350</th><td>BP</td><td>GO:1904350</td><td>regulation of protein catabolic process in the vacuole                                                                                          </td><td>2/43</td><td>13/18870 </td><td>3.893775e-04</td><td>0.021983605</td><td>0.017112117</td><td>LRP1/ATP13A2                  </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00161261</th><td>BP</td><td>GO:0016126</td><td>sterol biosynthetic process                                                                                                                     </td><td>3/43</td><td>67/18870 </td><td>4.769746e-04</td><td>0.025852022</td><td>0.020123306</td><td>HMGCS1/FDPS/FDFT1             </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0001912</th><td>BP</td><td>GO:0001912</td><td>positive regulation of leukocyte mediated cytotoxicity                                                                                          </td><td>3/43</td><td>69/18870 </td><td>5.200194e-04</td><td>0.026070510</td><td>0.020293378</td><td>PVR/IL12A/HFE                 </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00027111</th><td>BP</td><td>GO:0002711</td><td>positive regulation of T cell mediated immunity                                                                                                 </td><td>3/43</td><td>69/18870 </td><td>5.200194e-04</td><td>0.026070510</td><td>0.020293378</td><td>PVR/IL12A/HFE                 </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00466352</th><td>BP</td><td>GO:0046635</td><td>positive regulation of alpha-beta T cell activation                                                                                             </td><td>3/43</td><td>72/18870 </td><td>5.891304e-04</td><td>0.026070510</td><td>0.020293378</td><td>IL12A/CARD11/ADA              </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0002836</th><td>BP</td><td>GO:0002836</td><td>positive regulation of response to tumor cell                                                                                                   </td><td>2/43</td><td>16/18870 </td><td>5.964471e-04</td><td>0.026070510</td><td>0.020293378</td><td>PVR/IL12A                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0002839</th><td>BP</td><td>GO:0002839</td><td>positive regulation of immune response to tumor cell                                                                                            </td><td>2/43</td><td>16/18870 </td><td>5.964471e-04</td><td>0.026070510</td><td>0.020293378</td><td>PVR/IL12A                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0016114</th><td>BP</td><td>GO:0016114</td><td>terpenoid biosynthetic process                                                                                                                  </td><td>2/43</td><td>16/18870 </td><td>5.964471e-04</td><td>0.026070510</td><td>0.020293378</td><td>HMGCS1/FDPS                   </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00508621</th><td>BP</td><td>GO:0050862</td><td>positive regulation of T cell receptor signaling pathway                                                                                        </td><td>2/43</td><td>17/18870 </td><td>6.749961e-04</td><td>0.027500329</td><td>0.021406355</td><td>CARD11/ADA                    </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0071468</th><td>BP</td><td>GO:0071468</td><td>cellular response to acidic pH                                                                                                                  </td><td>2/43</td><td>17/18870 </td><td>6.749961e-04</td><td>0.027500329</td><td>0.021406355</td><td>PKD1L3/SLC9A1                 </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0031343</th><td>BP</td><td>GO:0031343</td><td>positive regulation of cell killing                                                                                                             </td><td>3/43</td><td>76/18870 </td><td>6.900452e-04</td><td>0.027500329</td><td>0.021406355</td><td>PVR/IL12A/HFE                 </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00091232</th><td>BP</td><td>GO:0009123</td><td>nucleoside monophosphate metabolic process                                                                                                      </td><td>3/43</td><td>77/18870 </td><td>7.168837e-04</td><td>0.027753642</td><td>0.021603534</td><td>CARD11/ADA/GMPR2              </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00466314</th><td>BP</td><td>GO:0046631</td><td>alpha-beta T cell activation                                                                                                                    </td><td>4/43</td><td>181/18870</td><td>7.545658e-04</td><td>0.028401018</td><td>0.022107454</td><td>IL12A/CARD11/ADA/HFE          </td><td>4</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00027063</th><td>BP</td><td>GO:0002706</td><td>regulation of lymphocyte mediated immunity                                                                                                      </td><td>4/43</td><td>188/18870</td><td>8.692625e-04</td><td>0.031806123</td><td>0.024758000</td><td>PVR/IL12A/HFE/EXOSC6          </td><td>4</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00508712</th><td>BP</td><td>GO:0050871</td><td>positive regulation of B cell activation                                                                                                        </td><td>3/43</td><td>84/18870 </td><td>9.235091e-04</td><td>0.031806123</td><td>0.024758000</td><td>CARD11/ADA/EXOSC6             </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0006144</th><td>BP</td><td>GO:0006144</td><td>purine nucleobase metabolic process                                                                                                             </td><td>2/43</td><td>20/18870 </td><td>9.389261e-04</td><td>0.031806123</td><td>0.024758000</td><td>ADA/GMPR2                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:1905146</th><td>BP</td><td>GO:1905146</td><td>lysosomal protein catabolic process                                                                                                             </td><td>2/43</td><td>20/18870 </td><td>9.389261e-04</td><td>0.031806123</td><td>0.024758000</td><td>LRP1/ATP13A2                  </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0002837</th><td>BP</td><td>GO:0002837</td><td>regulation of immune response to tumor cell                                                                                                     </td><td>2/43</td><td>22/18870 </td><td>1.138240e-03</td><td>0.037617451</td><td>0.029281558</td><td>PVR/IL12A                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00028341</th><td>BP</td><td>GO:0002834</td><td>regulation of response to tumor cell                                                                                                            </td><td>2/43</td><td>23/18870 </td><td>1.244843e-03</td><td>0.040161013</td><td>0.031261476</td><td>PVR/IL12A                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0007039</th><td>BP</td><td>GO:0007039</td><td>protein catabolic process in the vacuole                                                                                                        </td><td>2/43</td><td>24/18870 </td><td>1.356049e-03</td><td>0.041844027</td><td>0.032571540</td><td>LRP1/ATP13A2                  </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00019101</th><td>BP</td><td>GO:0001910</td><td>regulation of leukocyte mediated cytotoxicity                                                                                                   </td><td>3/43</td><td>96/18870 </td><td>1.358773e-03</td><td>0.041844027</td><td>0.032571540</td><td>PVR/IL12A/HFE                 </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00027091</th><td>BP</td><td>GO:0002709</td><td>regulation of T cell mediated immunity                                                                                                          </td><td>3/43</td><td>99/18870 </td><td>1.484557e-03</td><td>0.044701652</td><td>0.034795926</td><td>PVR/IL12A/HFE                 </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00060663</th><td>BP</td><td>GO:0006066</td><td>alcohol metabolic process                                                                                                                       </td><td>5/43</td><td>373/18870</td><td>1.525626e-03</td><td>0.044939629</td><td>0.034981168</td><td>IDH3B/HMGCS1/FDPS/FDFT1/INPP5J</td><td>5</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00024604</th><td>BP</td><td>GO:0002460</td><td>adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains                       </td><td>5/43</td><td>380/18870</td><td>1.655558e-03</td><td>0.047482562</td><td>0.036960596</td><td>PVR/IL12A/ADA/HFE/EXOSC6      </td><td>5</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00508572</th><td>BP</td><td>GO:0050857</td><td>positive regulation of antigen receptor-mediated signaling pathway                                                                              </td><td>2/43</td><td>27/18870 </td><td>1.717082e-03</td><td>0.047482562</td><td>0.036960596</td><td>CARD11/ADA                    </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0071467</th><td>BP</td><td>GO:0071467</td><td>cellular response to pH                                                                                                                         </td><td>2/43</td><td>27/18870 </td><td>1.717082e-03</td><td>0.047482562</td><td>0.036960596</td><td>PKD1L3/SLC9A1                 </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00026976</th><td>BP</td><td>GO:0002697</td><td>regulation of immune effector process                                                                                                           </td><td>5/43</td><td>389/18870</td><td>1.834395e-03</td><td>0.049693635</td><td>0.038681703</td><td>TWIST1/PVR/IL12A/HFE/EXOSC6   </td><td>5</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0002418</th><td>BP</td><td>GO:0002418</td><td>immune response to tumor cell                                                                                                                   </td><td>2/43</td><td>29/18870 </td><td>1.980411e-03</td><td>0.049693635</td><td>0.038681703</td><td>PVR/IL12A                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0008299</th><td>BP</td><td>GO:0008299</td><td>isoprenoid biosynthetic process                                                                                                                 </td><td>2/43</td><td>29/18870 </td><td>1.980411e-03</td><td>0.049693635</td><td>0.038681703</td><td>HMGCS1/FDPS                   </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00104471</th><td>BP</td><td>GO:0010447</td><td>response to acidic pH                                                                                                                           </td><td>2/43</td><td>29/18870 </td><td>1.980411e-03</td><td>0.049693635</td><td>0.038681703</td><td>PKD1L3/SLC9A1                 </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0045954</th><td>BP</td><td>GO:0045954</td><td>positive regulation of natural killer cell mediated cytotoxicity                                                                                </td><td>2/43</td><td>29/18870 </td><td>1.980411e-03</td><td>0.049693635</td><td>0.038681703</td><td>PVR/IL12A                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00313411</th><td>BP</td><td>GO:0031341</td><td>regulation of cell killing                                                                                                                      </td><td>3/43</td><td>111/18870</td><td>2.060007e-03</td><td>0.050751089</td><td>0.039504829</td><td>PVR/IL12A/HFE                 </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00027034</th><td>BP</td><td>GO:0002703</td><td>regulation of leukocyte mediated immunity                                                                                                       </td><td>4/43</td><td>249/18870</td><td>2.438784e-03</td><td>0.059009859</td><td>0.045933485</td><td>PVR/IL12A/HFE/EXOSC6          </td><td>4</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0002717</th><td>BP</td><td>GO:0002717</td><td>positive regulation of natural killer cell mediated immunity                                                                                    </td><td>2/43</td><td>34/18870 </td><td>2.716796e-03</td><td>0.063469985</td><td>0.049405263</td><td>PVR/IL12A                     </td><td>2</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:00603172</th><td>BP</td><td>GO:0060317</td><td>cardiac epithelial to mesenchymal transition                                                                                                    </td><td>2/43</td><td>34/18870 </td><td>2.716796e-03</td><td>0.063469985</td><td>0.049405263</td><td>TWIST1/RTN4                   </td><td>2</td><td>DLPFC_V1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm1_df %&gt;% filter(qvalue &lt; 0.05,ONTOLOGY == &#39;MF&#39;) %&gt;% arrange(qvalue)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 2 x 11</caption>
<thead>
	<tr><th></th><th scope=col>ONTOLOGY</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th><th scope=col>context</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:19016813</th><td>MF</td><td>GO:1901681</td><td>sulfur compound binding   </td><td>5/44</td><td>275/18496</td><td>0.0004735447</td><td>0.06397159</td><td>0.04682736</td><td>COLQ/PRMT8/LRP1/ATP13A2/RTN4R</td><td>5</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0004659</th><td>MF</td><td>GO:0004659</td><td>prenyltransferase activity</td><td>2/44</td><td>17/18496 </td><td>0.0007353056</td><td>0.06397159</td><td>0.04682736</td><td>FDPS/FDFT1                   </td><td>2</td><td>DLPFC_V1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(ggplot2)
library(dplyr)

# Filter, group by context, and take top 10 terms per context
df &lt;- progm1_df %&gt;%
  filter(qvalue &lt; 0.05, ONTOLOGY == &quot;BP&quot;) %&gt;%
  group_by(context) %&gt;%
  slice_min(qvalue, n = 10) %&gt;%
  ungroup() %&gt;%
  arrange(context, qvalue) %&gt;%
  mutate(Description = factor(Description, levels = rev(unique(Description))))  # keep order

options(repr.plot.width = 16, repr.plot.height = 6)


ggplot(df, aes(x = -log10(qvalue), y = Description, size = Count, color = p.adjust)) +
  geom_point() +
  scale_color_gradient(low = &quot;red&quot;, high = &quot;blue&quot;) +
  facet_wrap(~ context, scales = &quot;free_y&quot;) +
  labs(
    title = &quot;Top 10 significant BP GO terms per context (q &lt; 0.1)&quot;,
    x = &quot;-log10(q-value)&quot;,
    y = &quot;GO Term&quot;
  ) +
  theme_bw() +
  theme(
    strip.text = element_text(face = &quot;bold&quot;, size = 14),
    axis.text.y = element_text(size = 14),
    axis.text.x = element_text(size = 14),
    axis.title = element_text(size = 16),
    plot.title = element_text(size = 18, face = &quot;bold&quot;)
  )
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/d0a8fb71aeb38943956eff751d44b15ed0cfb885ce73e61a6bad31c8803f29fd.png"><img alt="../../../_images/d0a8fb71aeb38943956eff751d44b15ed0cfb885ce73e61a6bad31c8803f29fd.png" src="../../../_images/d0a8fb71aeb38943956eff751d44b15ed0cfb885ce73e61a6bad31c8803f29fd.png" style="width: 960px; height: 360px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>fwrite(progm1_df, &#39;GO.Top50_genes_Program1_split.txt&#39;, sep = &#39;\t&#39;)
</pre></div>
</div>
</div>
</div>
</section>
<section id="another-top-split">
<h4>another top split<a class="headerlink" href="#another-top-split" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm1_cor &lt;- fread(&#39;Top50_Cor_genes_Program1_split.txt&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Initialize empty data frame
progm1_cor_df &lt;- data.frame()

# Loop through each context (column) in progm1
for(context in colnames(progm1_cor)){
    message(context)
  # Perform GO enrichment and store results with context label
  res_cor &lt;- enrichGO(
    gene          = progm1_cor[[context]],  # use [[ ]] to access column by name
    OrgDb         = org.Hs.eg.db,
    keyType       = &quot;SYMBOL&quot;,
    ont           = &quot;ALL&quot;,
    pAdjustMethod = &quot;BH&quot;,
    pvalueCutoff  = 1,
    qvalueCutoff  = 1
  ) %&gt;% 
    as.data.frame() %&gt;% 
    mutate(context = context)  # add context info
  
  # Append to cumulative results
  progm1_cor_df &lt;- rbind(progm1_cor_df, res_cor)
}
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Mic_V1

Exc_V1

Inh_V1

Ast_V1

Oli_V1

OPC_V1

DLPFC_V1

AC_V1

PCC_V1
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm1_cor_df %&gt;% filter(qvalue &lt; 0.05) %&gt;% arrange(qvalue)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 5 x 10</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th><th scope=col>context</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:1904350</th><td>GO:1904350</td><td>regulation of protein catabolic process in the vacuole</td><td>3/48</td><td>13/18870 </td><td>4.339535e-06</td><td>0.006409494</td><td>0.005431271</td><td>LRP1/ATP13A2/CD81     </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0007039</th><td>GO:0007039</td><td>protein catabolic process in the vacuole              </td><td>3/48</td><td>24/18870 </td><td>3.011233e-05</td><td>0.022237953</td><td>0.018843977</td><td>LRP1/ATP13A2/CD81     </td><td>3</td><td>DLPFC_V1</td></tr>
	<tr><th scope=row>GO:0071425</th><td>GO:0071425</td><td>hematopoietic stem cell proliferation                 </td><td>3/41</td><td>36/18870 </td><td>6.467181e-05</td><td>0.047052170</td><td>0.038836071</td><td>ARIH2/ERCC2/RUNX1     </td><td>3</td><td>Oli_V1  </td></tr>
	<tr><th scope=row>GO:0072595</th><td>GO:0072595</td><td>maintenance of protein localization in organelle      </td><td>3/41</td><td>41/18870 </td><td>9.582927e-05</td><td>0.047052170</td><td>0.038836071</td><td>INSIG1/ARL2/TAF3      </td><td>3</td><td>Oli_V1  </td></tr>
	<tr><th scope=row>GO:00720893</th><td>GO:0072089</td><td>stem cell proliferation                               </td><td>4/41</td><td>125/18870</td><td>1.537188e-04</td><td>0.050317303</td><td>0.041531057</td><td>ARIH2/GJA1/ERCC2/RUNX1</td><td>4</td><td>Oli_V1  </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>fwrite(progm1_cor_df, &#39;GO.Top50_Cor_genes_Program1_split.txt&#39;, sep = &#39;\t&#39;)
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="program1-kegg">
<h3>program1 KEGG<a class="headerlink" href="#program1-kegg" title="Link to this heading">#</a></h3>
<section id="id3">
<h4>split<a class="headerlink" href="#id3" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm1 &lt;- fread(&#39;multi_gene/Top50_genes_Program1_split.txt&#39;)
progm1 %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 x 9</caption>
<thead>
	<tr><th scope=col>Mic_V1</th><th scope=col>Exc_V1</th><th scope=col>Inh_V1</th><th scope=col>Ast_V1</th><th scope=col>Oli_V1</th><th scope=col>OPC_V1</th><th scope=col>DLPFC_V1</th><th scope=col>AC_V1</th><th scope=col>PCC_V1</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>MAP2K2 </td><td>ZBTB37 </td><td>B3GNT5    </td><td>ADCK5  </td><td>UQCC3 </td><td>RANBP2</td><td>BCKDK   </td><td>ARHGEF1</td><td>FHAD1 </td></tr>
	<tr><td>HLA-DMA</td><td>SMIM12 </td><td>AL162231.1</td><td>EIF5A  </td><td>MAGEF1</td><td>BIN1  </td><td>FBXO10  </td><td>DTX3   </td><td>TPM3  </td></tr>
	<tr><td>CPLX1  </td><td>TMED7  </td><td>MTHFS     </td><td>GNG3   </td><td>FUCA1 </td><td>TET3  </td><td>COLQ    </td><td>PENK   </td><td>HERC2 </td></tr>
	<tr><td>TTYH1  </td><td>PGBD5  </td><td>MCM7      </td><td>BTN3A1 </td><td>PRR7  </td><td>KIF1A </td><td>ZFC3H1  </td><td>COG1   </td><td>FGF22 </td></tr>
	<tr><td>KLHDC4 </td><td>FAM229A</td><td>RILPL2    </td><td>SDHAF2 </td><td>CMTM1 </td><td>SAP18 </td><td>TMEM151A</td><td>QRICH1 </td><td>CADPS2</td></tr>
	<tr><td>TPGS2  </td><td>FSCN1  </td><td>FBXO21    </td><td>ZFYVE19</td><td>GSTT2B</td><td>GCFC2 </td><td>PKD1L3  </td><td>SLC26A2</td><td>PEX6  </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(clusterProfiler)
library(org.Hs.eg.db)

# Initialize empty data frame
progm1_kegg_df &lt;- data.frame()

# Loop through each context (column) in progm1
for (context in colnames(progm1)) {
  message(&quot;Processing: &quot;, context)
  
  # Convert gene symbols to Entrez IDs
  genes_entrez &lt;- bitr(
    progm1[[context]],
    fromType = &quot;SYMBOL&quot;,
    toType = &quot;ENTREZID&quot;,
    OrgDb = org.Hs.eg.db
  ) %&gt;% pull(ENTREZID)
  
  # Skip if no mapped genes
  if (length(genes_entrez) == 0) next
  
  # Perform KEGG enrichment
  res_kegg &lt;- enrichKEGG(
    gene          = genes_entrez,
    organism      = &#39;hsa&#39;,
    keyType       = &quot;kegg&quot;,
    pAdjustMethod = &quot;BH&quot;,
    pvalueCutoff  = 1,
    qvalueCutoff  = 1
  ) %&gt;%
    as.data.frame() %&gt;%
    mutate(context = context)  # label with context

  # Append results
  progm1_kegg_df &lt;- rbind(progm1_kegg_df, res_kegg)
}
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Processing: Mic_V1

&#39;select()&#39; returned 1:1 mapping between keys and columns

Reading KEGG annotation online: &quot;https://rest.kegg.jp/link/hsa/pathway&quot;...

Reading KEGG annotation online: &quot;https://rest.kegg.jp/list/pathway/hsa&quot;...

Processing: Exc_V1

&#39;select()&#39; returned 1:1 mapping between keys and columns

Processing: Inh_V1

&#39;select()&#39; returned 1:1 mapping between keys and columns

Warning message in bitr(progm1[[context]], fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, :
&quot;2% of input gene IDs are fail to map...&quot;
Processing: Ast_V1

&#39;select()&#39; returned 1:1 mapping between keys and columns

Warning message in bitr(progm1[[context]], fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, :
&quot;6.12% of input gene IDs are fail to map...&quot;
Processing: Oli_V1

&#39;select()&#39; returned 1:1 mapping between keys and columns

Warning message in bitr(progm1[[context]], fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, :
&quot;2% of input gene IDs are fail to map...&quot;
Processing: OPC_V1

&#39;select()&#39; returned 1:1 mapping between keys and columns

Processing: DLPFC_V1

&#39;select()&#39; returned 1:1 mapping between keys and columns

Warning message in bitr(progm1[[context]], fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, :
&quot;4% of input gene IDs are fail to map...&quot;
Processing: AC_V1

&#39;select()&#39; returned 1:1 mapping between keys and columns

Warning message in bitr(progm1[[context]], fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, :
&quot;6% of input gene IDs are fail to map...&quot;
Processing: PCC_V1

&#39;select()&#39; returned 1:1 mapping between keys and columns
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm1_kegg_df
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 866 x 12</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th><th scope=col>context</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa05219</th><td>Human Diseases                      </td><td>Cancer: specific types              </td><td>hsa05219</td><td>Bladder cancer                                   </td><td>2/24</td><td>41/8866 </td><td>0.005399633</td><td>0.2526374</td><td>0.2434320</td><td>5605/1612           </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04216</th><td>Cellular Processes                  </td><td>Cell growth and death               </td><td>hsa04216</td><td>Ferroptosis                                      </td><td>2/24</td><td>42/8866 </td><td>0.005660279</td><td>0.2526374</td><td>0.2434320</td><td>2512/2879           </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa03050</th><td>Genetic Information Processing      </td><td>Folding, sorting and degradation    </td><td>hsa03050</td><td>Proteasome                                       </td><td>2/24</td><td>46/8866 </td><td>0.006759480</td><td>0.2526374</td><td>0.2434320</td><td>5691/5708           </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04140</th><td>Cellular Processes                  </td><td>Transport and catabolism            </td><td>hsa04140</td><td>Autophagy - animal                               </td><td>3/24</td><td>169/8866</td><td>0.010261506</td><td>0.2526374</td><td>0.2434320</td><td>5605/53349/1612     </td><td>3</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05164</th><td>Human Diseases                      </td><td>Infectious disease: viral           </td><td>hsa05164</td><td>Influenza A                                      </td><td>3/24</td><td>173/8866</td><td>0.010934476</td><td>0.2526374</td><td>0.2434320</td><td>5605/3108/114548    </td><td>3</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04978</th><td>Organismal Systems                  </td><td>Digestive system                    </td><td>hsa04978</td><td>Mineral absorption                               </td><td>2/24</td><td>61/8866 </td><td>0.011660188</td><td>0.2526374</td><td>0.2434320</td><td>2512/475            </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05223</th><td>Human Diseases                      </td><td>Cancer: specific types              </td><td>hsa05223</td><td>Non-small cell lung cancer                       </td><td>2/24</td><td>73/8866 </td><td>0.016418000</td><td>0.3049057</td><td>0.2937958</td><td>5605/83593          </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa03040</th><td>Genetic Information Processing      </td><td>Transcription                       </td><td>hsa03040</td><td>Spliceosome                                      </td><td>3/24</td><td>235/8866</td><td>0.024679306</td><td>0.4010387</td><td>0.3864260</td><td>11017/23020/9410    </td><td>3</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05022</th><td>Human Diseases                      </td><td>Neurodegenerative disease           </td><td>hsa05022</td><td>Pathways of neurodegeneration - multiple diseases</td><td>4/24</td><td>483/8866</td><td>0.038840904</td><td>0.4231942</td><td>0.4077742</td><td>5605/5691/53349/5708</td><td>4</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa00604</th><td>Metabolism                          </td><td>Glycan biosynthesis and metabolism  </td><td>hsa00604</td><td>Glycosphingolipid biosynthesis - ganglio series  </td><td>1/24</td><td>15/8866 </td><td>0.039875000</td><td>0.4231942</td><td>0.4077742</td><td>9197                </td><td>1</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04722</th><td>Organismal Systems                  </td><td>Nervous system                      </td><td>hsa04722</td><td>Neurotrophin signaling pathway                   </td><td>2/24</td><td>120/8866</td><td>0.041304924</td><td>0.4231942</td><td>0.4077742</td><td>5605/396            </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05135</th><td>Human Diseases                      </td><td>Infectious disease: bacterial       </td><td>hsa05135</td><td>Yersinia infection                               </td><td>2/24</td><td>138/8866</td><td>0.053105929</td><td>0.4231942</td><td>0.4077742</td><td>5605/114548         </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05017</th><td>Human Diseases                      </td><td>Neurodegenerative disease           </td><td>hsa05017</td><td>Spinocerebellar ataxia                           </td><td>2/24</td><td>144/8866</td><td>0.057280363</td><td>0.4231942</td><td>0.4077742</td><td>5691/5708           </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04218</th><td>Cellular Processes                  </td><td>Cell growth and death               </td><td>hsa04218</td><td>Cellular senescence                              </td><td>2/24</td><td>157/8866</td><td>0.066706151</td><td>0.4231942</td><td>0.4077742</td><td>5605/83593          </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04150</th><td>Environmental Information Processing</td><td>Signal transduction                 </td><td>hsa04150</td><td>mTOR signaling pathway                           </td><td>2/24</td><td>158/8866</td><td>0.067451878</td><td>0.4231942</td><td>0.4077742</td><td>5605/55437          </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04217</th><td>Cellular Processes                  </td><td>Cell growth and death               </td><td>hsa04217</td><td>Necroptosis                                      </td><td>2/24</td><td>159/8866</td><td>0.068200459</td><td>0.4231942</td><td>0.4077742</td><td>2512/114548         </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04514</th><td>Environmental Information Processing</td><td>Signaling molecules and interaction </td><td>hsa04514</td><td>Cell adhesion molecules                          </td><td>2/24</td><td>160/8866</td><td>0.068951874</td><td>0.4231942</td><td>0.4077742</td><td>3108/1272           </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05310</th><td>Human Diseases                      </td><td>Immune disease                      </td><td>hsa05310</td><td>Asthma                                           </td><td>1/24</td><td>32/8866 </td><td>0.083224558</td><td>0.4231942</td><td>0.4077742</td><td>3108                </td><td>1</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05010</th><td>Human Diseases                      </td><td>Neurodegenerative disease           </td><td>hsa05010</td><td>Alzheimer disease                                </td><td>3/24</td><td>391/8866</td><td>0.086890045</td><td>0.4231942</td><td>0.4077742</td><td>5605/5691/5708      </td><td>3</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05216</th><td>Human Diseases                      </td><td>Cancer: specific types              </td><td>hsa05216</td><td>Thyroid cancer                                   </td><td>1/24</td><td>37/8866 </td><td>0.095613234</td><td>0.4231942</td><td>0.4077742</td><td>5605                </td><td>1</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05330</th><td>Human Diseases                      </td><td>Immune disease                      </td><td>hsa05330</td><td>Allograft rejection                              </td><td>1/24</td><td>39/8866 </td><td>0.100523644</td><td>0.4231942</td><td>0.4077742</td><td>3108                </td><td>1</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05169</th><td>Human Diseases                      </td><td>Infectious disease: viral           </td><td>hsa05169</td><td>Epstein-Barr virus infection                     </td><td>2/24</td><td>204/8866</td><td>0.104539011</td><td>0.4231942</td><td>0.4077742</td><td>3108/5708           </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04015</th><td>Environmental Information Processing</td><td>Signal transduction                 </td><td>hsa04015</td><td>Rap1 signaling pathway                           </td><td>2/24</td><td>212/8866</td><td>0.111475344</td><td>0.4231942</td><td>0.4077742</td><td>5605/83593          </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04940</th><td>Human Diseases                      </td><td>Endocrine and metabolic disease     </td><td>hsa04940</td><td>Type I diabetes mellitus                         </td><td>1/24</td><td>44/8866 </td><td>0.112688141</td><td>0.4231942</td><td>0.4077742</td><td>3108                </td><td>1</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04962</th><td>Organismal Systems                  </td><td>Excretory system                    </td><td>hsa04962</td><td>Vasopressin-regulated water reabsorption         </td><td>1/24</td><td>44/8866 </td><td>0.112688141</td><td>0.4231942</td><td>0.4077742</td><td>396                 </td><td>1</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05332</th><td>Human Diseases                      </td><td>Immune disease                      </td><td>hsa05332</td><td>Graft-versus-host disease                        </td><td>1/24</td><td>45/8866 </td><td>0.115102047</td><td>0.4231942</td><td>0.4077742</td><td>3108                </td><td>1</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa00860</th><td>Metabolism                          </td><td>Metabolism of cofactors and vitamins</td><td>hsa00860</td><td>Porphyrin metabolism                             </td><td>1/24</td><td>46/8866 </td><td>0.117509660</td><td>0.4231942</td><td>0.4077742</td><td>2512                </td><td>1</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa00514</th><td>Metabolism                          </td><td>Glycan biosynthesis and metabolism  </td><td>hsa00514</td><td>Other types of O-glycan biosynthesis             </td><td>1/24</td><td>47/8866 </td><td>0.119910994</td><td>0.4231942</td><td>0.4077742</td><td>56983               </td><td>1</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa05166</th><td>Human Diseases                      </td><td>Infectious disease: viral           </td><td>hsa05166</td><td>Human T-cell leukemia virus 1 infection          </td><td>2/24</td><td>224/8866</td><td>0.122106557</td><td>0.4231942</td><td>0.4077742</td><td>5605/3108           </td><td>2</td><td>Mic_V1</td></tr>
	<tr><th scope=row>hsa04672</th><td>Organismal Systems                  </td><td>Immune system                       </td><td>hsa04672</td><td>Intestinal immune network for IgA production     </td><td>1/24</td><td>50/8866 </td><td>0.127077483</td><td>0.4231942</td><td>0.4077742</td><td>3108                </td><td>1</td><td>Mic_V1</td></tr>
	<tr><th scope=row>...</th><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr>
	<tr><th scope=row>hsa050173</th><td>Human Diseases                      </td><td>Neurodegenerative disease       </td><td>hsa05017</td><td>Spinocerebellar ataxia                         </td><td>1/22</td><td>144/8866</td><td>0.3028011</td><td>0.4694877</td><td>0.4616846</td><td>5718  </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa052244</th><td>Human Diseases                      </td><td>Cancer: specific types          </td><td>hsa05224</td><td>Breast cancer                                  </td><td>1/22</td><td>148/8866</td><td>0.3098101</td><td>0.4694877</td><td>0.4616846</td><td>27006 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa052264</th><td>Human Diseases                      </td><td>Cancer: specific types          </td><td>hsa05226</td><td>Gastric cancer                                 </td><td>1/22</td><td>150/8866</td><td>0.3132893</td><td>0.4694877</td><td>0.4616846</td><td>27006 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa012401</th><td>Metabolism                          </td><td>Global and overview maps        </td><td>hsa01240</td><td>Biosynthesis of cofactors                      </td><td>1/22</td><td>154/8866</td><td>0.3201975</td><td>0.4694877</td><td>0.4616846</td><td>1503  </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa041506</th><td>Environmental Information Processing</td><td>Signal transduction             </td><td>hsa04150</td><td>mTOR signaling pathway                         </td><td>1/22</td><td>158/8866</td><td>0.3270394</td><td>0.4694877</td><td>0.4616846</td><td>528   </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa041453</th><td>Cellular Processes                  </td><td>Transport and catabolism        </td><td>hsa04145</td><td>Phagosome                                      </td><td>1/22</td><td>159/8866</td><td>0.3287396</td><td>0.4694877</td><td>0.4616846</td><td>528   </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa051605</th><td>Human Diseases                      </td><td>Infectious disease: viral       </td><td>hsa05160</td><td>Hepatitis C                                    </td><td>1/22</td><td>159/8866</td><td>0.3287396</td><td>0.4694877</td><td>0.4616846</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa051614</th><td>Human Diseases                      </td><td>Infectious disease: viral       </td><td>hsa05161</td><td>Hepatitis B                                    </td><td>1/22</td><td>163/8866</td><td>0.3354994</td><td>0.4694877</td><td>0.4616846</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa041404</th><td>Cellular Processes                  </td><td>Transport and catabolism        </td><td>hsa04140</td><td>Autophagy - animal                             </td><td>1/22</td><td>169/8866</td><td>0.3455174</td><td>0.4694877</td><td>0.4616846</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa041411</th><td>Genetic Information Processing      </td><td>Folding, sorting and degradation</td><td>hsa04141</td><td>Protein processing in endoplasmic reticulum    </td><td>1/22</td><td>171/8866</td><td>0.3488246</td><td>0.4694877</td><td>0.4616846</td><td>8720  </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa051646</th><td>Human Diseases                      </td><td>Infectious disease: viral       </td><td>hsa05164</td><td>Influenza A                                    </td><td>1/22</td><td>173/8866</td><td>0.3521158</td><td>0.4694877</td><td>0.4616846</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa051685</th><td>Human Diseases                      </td><td>Infectious disease: viral       </td><td>hsa05168</td><td>Herpes simplex virus 1 infection               </td><td>1/22</td><td>182/8866</td><td>0.3667308</td><td>0.4805438</td><td>0.4725569</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa046212</th><td>Organismal Systems                  </td><td>Immune system                   </td><td>hsa04621</td><td>NOD-like receptor signaling pathway            </td><td>1/22</td><td>189/8866</td><td>0.3778798</td><td>0.4863903</td><td>0.4783062</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa051674</th><td>Human Diseases                      </td><td>Infectious disease: viral       </td><td>hsa05167</td><td>Kaposi sarcoma-associated herpesvirus infection</td><td>1/22</td><td>196/8866</td><td>0.3888414</td><td>0.4863903</td><td>0.4783062</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa048142</th><td>Cellular Processes                  </td><td>Cell motility                   </td><td>hsa04814</td><td>Motor proteins                                 </td><td>1/22</td><td>197/8866</td><td>0.3903922</td><td>0.4863903</td><td>0.4783062</td><td>7170  </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa052032</th><td>Human Diseases                      </td><td>Cancer: overview                </td><td>hsa05203</td><td>Viral carcinogenesis                           </td><td>1/22</td><td>205/8866</td><td>0.4026641</td><td>0.4900872</td><td>0.4819417</td><td>9093  </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa040154</th><td>Environmental Information Processing</td><td>Signal transduction             </td><td>hsa04015</td><td>Rap1 signaling pathway                         </td><td>1/22</td><td>212/8866</td><td>0.4132083</td><td>0.4900872</td><td>0.4819417</td><td>27006 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa051704</th><td>Human Diseases                      </td><td>Infectious disease: viral       </td><td>hsa05170</td><td>Human immunodeficiency virus 1 infection       </td><td>1/22</td><td>213/8866</td><td>0.4147000</td><td>0.4900872</td><td>0.4819417</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa054175</th><td>Human Diseases                      </td><td>Cardiovascular disease          </td><td>hsa05417</td><td>Lipid and atherosclerosis                      </td><td>1/22</td><td>216/8866</td><td>0.4191535</td><td>0.4900872</td><td>0.4819417</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa051635</th><td>Human Diseases                      </td><td>Infectious disease: viral       </td><td>hsa05163</td><td>Human cytomegalovirus infection                </td><td>1/22</td><td>226/8866</td><td>0.4337661</td><td>0.4944074</td><td>0.4861901</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa048106</th><td>Cellular Processes                  </td><td>Cell motility                   </td><td>hsa04810</td><td>Regulation of actin cytoskeleton               </td><td>1/22</td><td>232/8866</td><td>0.4423645</td><td>0.4944074</td><td>0.4861901</td><td>27006 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa048203</th><td>NA                                  </td><td>NA                              </td><td>hsa04820</td><td>Cytoskeleton in muscle cells                   </td><td>1/22</td><td>232/8866</td><td>0.4423645</td><td>0.4944074</td><td>0.4861901</td><td>7170  </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa051714</th><td>Human Diseases                      </td><td>Infectious disease: viral       </td><td>hsa05171</td><td>Coronavirus disease - COVID-19                 </td><td>1/22</td><td>238/8866</td><td>0.4508381</td><td>0.4965753</td><td>0.4883220</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa051323</th><td>Human Diseases                      </td><td>Infectious disease: bacterial   </td><td>hsa05132</td><td>Salmonella infection                           </td><td>1/22</td><td>251/8866</td><td>0.4687781</td><td>0.4976817</td><td>0.4894100</td><td>3839  </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa041446</th><td>Cellular Processes                  </td><td>Transport and catabolism        </td><td>hsa04144</td><td>Endocytosis                                    </td><td>1/22</td><td>252/8866</td><td>0.4701347</td><td>0.4976817</td><td>0.4894100</td><td>116983</td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa051315</th><td>Human Diseases                      </td><td>Infectious disease: bacterial   </td><td>hsa05131</td><td>Shigellosis                                    </td><td>1/22</td><td>253/8866</td><td>0.4714879</td><td>0.4976817</td><td>0.4894100</td><td>29110 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa050125</th><td>Human Diseases                      </td><td>Neurodegenerative disease       </td><td>hsa05012</td><td>Parkinson disease                              </td><td>1/22</td><td>271/8866</td><td>0.4952899</td><td>0.5156442</td><td>0.5070740</td><td>5718  </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa050205</th><td>Human Diseases                      </td><td>Neurodegenerative disease       </td><td>hsa05020</td><td>Prion disease                                  </td><td>1/22</td><td>278/8866</td><td>0.5042669</td><td>0.5178958</td><td>0.5092881</td><td>5718  </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa040104</th><td>Environmental Information Processing</td><td>Signal transduction             </td><td>hsa04010</td><td>MAPK signaling pathway                         </td><td>1/22</td><td>300/8866</td><td>0.5314989</td><td>0.5385856</td><td>0.5296340</td><td>27006 </td><td>1</td><td>PCC_V1</td></tr>
	<tr><th scope=row>hsa050164</th><td>Human Diseases                      </td><td>Neurodegenerative disease       </td><td>hsa05016</td><td>Huntington disease                             </td><td>1/22</td><td>311/8866</td><td>0.5445735</td><td>0.5445735</td><td>0.5355224</td><td>5718  </td><td>1</td><td>PCC_V1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>library(ggplot2)
library(dplyr)

# Filter, group by context, and take top 10 terms per context
df &lt;- progm1_kegg_df %&gt;%
  filter(qvalue &lt; 0.1) %&gt;%
  group_by(context) %&gt;%
  slice_min(qvalue, n = 10) %&gt;%
  ungroup() %&gt;%
  arrange(context, qvalue) %&gt;%
  mutate(Description = factor(Description, levels = rev(unique(Description))))  # keep order

options(repr.plot.width = 16, repr.plot.height = 6)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>

ggplot(df, aes(x = -log10(qvalue), y = Description, size = Count, color = p.adjust)) +
  geom_point() +
  scale_color_gradient(low = &quot;red&quot;, high = &quot;blue&quot;) +
  facet_wrap(~ context, scales = &quot;free_y&quot;) +
  labs(
    title = &quot;Top 10 significant KEGG per context (q &lt; 0.1)&quot;,
    x = &quot;-log10(q-value)&quot;,
    y = &quot;KEGG pathway&quot;
  ) +
  theme_bw() +
  theme(
    strip.text = element_text(face = &quot;bold&quot;, size = 14),
    axis.text.y = element_text(size = 14),
    axis.text.x = element_text(size = 14),
    axis.title = element_text(size = 16),
    plot.title = element_text(size = 18, face = &quot;bold&quot;)
  )
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../../_images/85a778db0c3b5a7f5fe637a73fc04eb2cab78ebaa95a32bb0bcc45af17cadfed.png"><img alt="../../../_images/85a778db0c3b5a7f5fe637a73fc04eb2cab78ebaa95a32bb0bcc45af17cadfed.png" src="../../../_images/85a778db0c3b5a7f5fe637a73fc04eb2cab78ebaa95a32bb0bcc45af17cadfed.png" style="width: 960px; height: 360px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>fwrite(progm1_df, &#39;GO.Top50_genes_Program1_split.txt&#39;, sep = &#39;\t&#39;)
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="program2-go">
<h3>program2 GO<a class="headerlink" href="#program2-go" title="Link to this heading">#</a></h3>
<section id="id4">
<h4>split<a class="headerlink" href="#id4" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm2 &lt;- fread(&#39;Top50_genes_Program2_split.txt&#39;)
progm2 %&gt;% head
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 6 x 6</caption>
<thead>
	<tr><th scope=col>Mic_V2</th><th scope=col>Exc_V2</th><th scope=col>Inh_V2</th><th scope=col>Ast_V2</th><th scope=col>Oli_V2</th><th scope=col>OPC_V2</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>AP1G2 </td><td>SSU72 </td><td>PTHLH </td><td>CALY   </td><td>GABRD   </td><td>EEF1A2</td></tr>
	<tr><td>ZBTB43</td><td>PPIA  </td><td>PI4K2B</td><td>RTL6   </td><td>C1QTNF4 </td><td>RTKN  </td></tr>
	<tr><td>IKBKE </td><td>SH2D5 </td><td>MCAT  </td><td>ZDHHC23</td><td>LY6E    </td><td>CEBPD </td></tr>
	<tr><td>LRRC4 </td><td>MIA3  </td><td>ESYT3 </td><td>TAGLN3 </td><td>EBNA1BP2</td><td>PDIA6 </td></tr>
	<tr><td>ENTPD6</td><td>ZNF106</td><td>ROR2  </td><td>TARBP2 </td><td>ZNF804B </td><td>NOSIP </td></tr>
	<tr><td>WDR73 </td><td>UQCRB </td><td>REEP5 </td><td>LTBR   </td><td>EFNA1   </td><td>FABP5 </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Initialize empty data frame
progm2_df &lt;- data.frame()

# Loop through each context (column) in progm1
for(context in colnames(progm2)){
    message(context)
  # Perform GO enrichment and store results with context label
  res &lt;- enrichGO(
    gene          = progm2[[context]],  # use [[ ]] to access column by name
    OrgDb         = org.Hs.eg.db,
    keyType       = &quot;SYMBOL&quot;,
    ont           = &quot;ALL&quot;,
    pAdjustMethod = &quot;BH&quot;,
    pvalueCutoff  = 1,
    qvalueCutoff  = 1
  ) %&gt;% 
    as.data.frame() %&gt;% 
    mutate(context = context)  # add context info
  
  # Append to cumulative results
  progm2_df &lt;- rbind(progm2_df, res)
}
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Mic_V2

Exc_V2

Inh_V2

Ast_V2

Oli_V2

OPC_V2
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm2_df %&gt;% filter(qvalue &lt; 0.05,ONTOLOGY == &#39;BP&#39;) %&gt;% arrange(qvalue)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 22 x 11</caption>
<thead>
	<tr><th></th><th scope=col>ONTOLOGY</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th><th scope=col>context</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:01060271</th><td>BP</td><td>GO:0106027</td><td>neuron projection organization                                                               </td><td>4/46</td><td>91/18870 </td><td>7.073369e-05</td><td>0.03729628</td><td>0.03141501</td><td>EFNA1/DVL1/BAIAP2/MAP1A      </td><td>4</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:00609962</th><td>BP</td><td>GO:0060996</td><td>dendritic spine development                                                                  </td><td>4/46</td><td>95/18870 </td><td>8.365735e-05</td><td>0.03729628</td><td>0.03141501</td><td>EFNA1/CAMK1/DVL1/BAIAP2      </td><td>4</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:00991751</th><td>BP</td><td>GO:0099175</td><td>regulation of postsynapse organization                                                       </td><td>4/46</td><td>103/18870</td><td>1.145424e-04</td><td>0.03729628</td><td>0.03141501</td><td>EFNA1/LRFN4/DVL1/BAIAP2      </td><td>4</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:00163583</th><td>BP</td><td>GO:0016358</td><td>dendrite development                                                                         </td><td>5/46</td><td>233/18870</td><td>2.494742e-04</td><td>0.03729628</td><td>0.03141501</td><td>EFNA1/CAMK1/DVL1/BAIAP2/MAP1A</td><td>5</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:00508073</th><td>BP</td><td>GO:0050807</td><td>regulation of synapse organization                                                           </td><td>5/46</td><td>244/18870</td><td>3.086070e-04</td><td>0.03729628</td><td>0.03141501</td><td>EFNA1/LRFN4/CAMK1/DVL1/BAIAP2</td><td>5</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:1905383</th><td>BP</td><td>GO:1905383</td><td>protein localization to presynapse                                                           </td><td>2/46</td><td>11/18870 </td><td>3.153078e-04</td><td>0.03729628</td><td>0.03141501</td><td>DVL1/MAP1A                   </td><td>2</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:00609971</th><td>BP</td><td>GO:0060997</td><td>dendritic spine morphogenesis                                                                </td><td>3/46</td><td>56/18870 </td><td>3.432937e-04</td><td>0.03729628</td><td>0.03141501</td><td>EFNA1/DVL1/BAIAP2            </td><td>3</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:00508033</th><td>BP</td><td>GO:0050803</td><td>regulation of synapse structure or activity                                                  </td><td>5/46</td><td>250/18870</td><td>3.450457e-04</td><td>0.03729628</td><td>0.03141501</td><td>EFNA1/LRFN4/CAMK1/DVL1/BAIAP2</td><td>5</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:00606122</th><td>BP</td><td>GO:0060612</td><td>adipose tissue development                                                                   </td><td>3/46</td><td>57/18870 </td><td>3.617486e-04</td><td>0.03729628</td><td>0.03141501</td><td>DYRK1B/SPTLC2/CSF1           </td><td>3</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:00609981</th><td>BP</td><td>GO:0060998</td><td>regulation of dendritic spine development                                                    </td><td>3/46</td><td>57/18870 </td><td>3.617486e-04</td><td>0.03729628</td><td>0.03141501</td><td>EFNA1/CAMK1/BAIAP2           </td><td>3</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:0040018</th><td>BP</td><td>GO:0040018</td><td>positive regulation of multicellular organism growth                                         </td><td>3/48</td><td>31/18870 </td><td>6.604359e-05</td><td>0.04972069</td><td>0.04345005</td><td>PPIB/BBS2/BBS4               </td><td>3</td><td>Mic_V2</td></tr>
	<tr><th scope=row>GO:0032096</th><td>BP</td><td>GO:0032096</td><td>negative regulation of response to food                                                      </td><td>2/48</td><td>10/18870 </td><td>2.814379e-04</td><td>0.04972069</td><td>0.04345005</td><td>BBS2/BBS4                    </td><td>2</td><td>Mic_V2</td></tr>
	<tr><th scope=row>GO:0032099</th><td>BP</td><td>GO:0032099</td><td>negative regulation of appetite                                                              </td><td>2/48</td><td>10/18870 </td><td>2.814379e-04</td><td>0.04972069</td><td>0.04345005</td><td>BBS2/BBS4                    </td><td>2</td><td>Mic_V2</td></tr>
	<tr><th scope=row>GO:0032105</th><td>BP</td><td>GO:0032105</td><td>negative regulation of response to extracellular stimulus                                    </td><td>2/48</td><td>10/18870 </td><td>2.814379e-04</td><td>0.04972069</td><td>0.04345005</td><td>BBS2/BBS4                    </td><td>2</td><td>Mic_V2</td></tr>
	<tr><th scope=row>GO:0032108</th><td>BP</td><td>GO:0032108</td><td>negative regulation of response to nutrient levels                                           </td><td>2/48</td><td>10/18870 </td><td>2.814379e-04</td><td>0.04972069</td><td>0.04345005</td><td>BBS2/BBS4                    </td><td>2</td><td>Mic_V2</td></tr>
	<tr><th scope=row>GO:0033210</th><td>BP</td><td>GO:0033210</td><td>leptin-mediated signaling pathway                                                            </td><td>2/48</td><td>10/18870 </td><td>2.814379e-04</td><td>0.04972069</td><td>0.04345005</td><td>BBS2/BBS4                    </td><td>2</td><td>Mic_V2</td></tr>
	<tr><th scope=row>GO:00109763</th><td>BP</td><td>GO:0010976</td><td>positive regulation of neuron projection development                                         </td><td>4/46</td><td>158/18870</td><td>5.872434e-04</td><td>0.05504072</td><td>0.04636132</td><td>CAMK1/KATNB1/DVL1/BAIAP2     </td><td>4</td><td>Oli_V2</td></tr>
	<tr><th scope=row>GO:0042559</th><td>BP</td><td>GO:0042559</td><td>pteridine-containing compound biosynthetic process                                           </td><td>2/48</td><td>13/18870 </td><td>4.854544e-04</td><td>0.05448829</td><td>0.04761638</td><td>PTS/FPGS                     </td><td>2</td><td>Mic_V2</td></tr>
	<tr><th scope=row>GO:0035264</th><td>BP</td><td>GO:0035264</td><td>multicellular organism growth                                                                </td><td>4/48</td><td>147/18870</td><td>5.270644e-04</td><td>0.05448829</td><td>0.04761638</td><td>KAT2A/PPIB/BBS2/BBS4         </td><td>4</td><td>Mic_V2</td></tr>
	<tr><th scope=row>GO:0040014</th><td>BP</td><td>GO:0040014</td><td>regulation of multicellular organism growth                                                  </td><td>3/48</td><td>63/18870 </td><td>5.510798e-04</td><td>0.05448829</td><td>0.04761638</td><td>PPIB/BBS2/BBS4               </td><td>3</td><td>Mic_V2</td></tr>
	<tr><th scope=row>GO:0032095</th><td>BP</td><td>GO:0032095</td><td>regulation of response to food                                                               </td><td>2/48</td><td>14/18870 </td><td>5.654446e-04</td><td>0.05448829</td><td>0.04761638</td><td>BBS2/BBS4                    </td><td>2</td><td>Mic_V2</td></tr>
	<tr><th scope=row>GO:1904925</th><td>BP</td><td>GO:1904925</td><td>positive regulation of autophagy of mitochondrion in response to mitochondrial depolarization</td><td>2/48</td><td>14/18870 </td><td>5.654446e-04</td><td>0.05448829</td><td>0.04761638</td><td>CDC37/ATP5IF1                </td><td>2</td><td>Mic_V2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>fwrite(progm2_df, &#39;GO.Top50_genes_Program2_split.txt&#39;, sep = &#39;\t&#39;)
</pre></div>
</div>
</div>
</div>
</section>
<section id="id5">
<h4>another top split<a class="headerlink" href="#id5" title="Link to this heading">#</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm2_cor &lt;- fread(&#39;Top50_Cor_genes_Program2_split.txt&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span># Initialize empty data frame
progm2_cor_df &lt;- data.frame()

# Loop through each context (column) in progm1
for(context in colnames(progm2_cor)){
    message(context)
  # Perform GO enrichment and store results with context label
  res_cor &lt;- enrichGO(
    gene          = progm2_cor[[context]],  # use [[ ]] to access column by name
    OrgDb         = org.Hs.eg.db,
    keyType       = &quot;SYMBOL&quot;,
    ont           = &quot;ALL&quot;,
    pAdjustMethod = &quot;BH&quot;,
    pvalueCutoff  = 1,
    qvalueCutoff  = 1
  ) %&gt;% 
    as.data.frame() %&gt;% 
    mutate(context = context)  # add context info
  
  # Append to cumulative results
  progm2_cor_df &lt;- rbind(progm2_cor_df, res_cor)
}
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Mic_V2

Exc_V2

Inh_V2

Ast_V2

Oli_V2

OPC_V2
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>progm2_cor_df %&gt;% filter(qvalue &lt; 0.05) %&gt;% arrange(qvalue)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 53 x 11</caption>
<thead>
	<tr><th></th><th scope=col>ONTOLOGY</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th><th scope=col>context</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:00226261</th><td>CC</td><td>GO:0022626</td><td>cytosolic ribosome                                             </td><td>6/49</td><td>118/19886</td><td>4.361033e-07</td><td>8.678456e-05</td><td>6.426786e-05</td><td>RPL37/RPL31/RPL23/RPS25/RPS21/RPS24                </td><td>6</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00443911</th><td>CC</td><td>GO:0044391</td><td>ribosomal subunit                                              </td><td>6/49</td><td>188/19886</td><td>6.574894e-06</td><td>6.542020e-04</td><td>4.844659e-04</td><td>RPL37/RPL31/RPL23/RPS25/RPS21/RPS24                </td><td>6</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00037351</th><td>MF</td><td>GO:0003735</td><td>structural constituent of ribosome                             </td><td>6/48</td><td>176/18496</td><td>6.004821e-06</td><td>1.068858e-03</td><td>9.102044e-04</td><td>RPL37/RPL31/RPL23/RPS25/RPS21/RPS24                </td><td>6</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00226262</th><td>CC</td><td>GO:0022626</td><td>cytosolic ribosome                                             </td><td>5/49</td><td>118/19886</td><td>1.045671e-05</td><td>1.189522e-03</td><td>9.277818e-04</td><td>RPL37/RPS25/RPL31/RPL27A/RPL38                     </td><td>5</td><td>Inh_V2</td></tr>
	<tr><th scope=row>GO:00226252</th><td>CC</td><td>GO:0022625</td><td>cytosolic large ribosomal subunit                              </td><td>4/49</td><td>60/19886 </td><td>1.433159e-05</td><td>1.189522e-03</td><td>9.277818e-04</td><td>RPL37/RPL31/RPL27A/RPL38                           </td><td>4</td><td>Inh_V2</td></tr>
	<tr><th scope=row>GO:00058401</th><td>CC</td><td>GO:0005840</td><td>ribosome                                                       </td><td>6/49</td><td>246/19886</td><td>3.020576e-05</td><td>2.003649e-03</td><td>1.483792e-03</td><td>RPL37/RPL31/RPL23/RPS25/RPS21/RPS24                </td><td>6</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0035035</th><td>MF</td><td>GO:0035035</td><td>histone acetyltransferase binding                              </td><td>3/48</td><td>24/18496 </td><td>3.195199e-05</td><td>2.843727e-03</td><td>2.421625e-03</td><td>PCNA/SIRT2/CEBPB                                   </td><td>3</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00021811</th><td>BP</td><td>GO:0002181</td><td>cytoplasmic translation                                        </td><td>6/47</td><td>156/18870</td><td>2.352749e-06</td><td>2.980933e-03</td><td>2.503820e-03</td><td>RPL37/RPL31/RPL23/RPS25/RPS21/RPS24                </td><td>6</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00050852</th><td>MF</td><td>GO:0005085</td><td>guanyl-nucleotide exchange factor activity                     </td><td>6/49</td><td>225/18496</td><td>2.738743e-05</td><td>4.710638e-03</td><td>4.007214e-03</td><td>ARHGEF25/RAPGEF2/DENND6A/DENND5A/PSD/ITSN2         </td><td>6</td><td>OPC_V2</td></tr>
	<tr><th scope=row>GO:00443912</th><td>CC</td><td>GO:0044391</td><td>ribosomal subunit                                              </td><td>5/49</td><td>188/19886</td><td>9.744209e-05</td><td>5.391796e-03</td><td>4.205396e-03</td><td>RPL37/RPS25/RPL31/RPL27A/RPL38                     </td><td>5</td><td>Inh_V2</td></tr>
	<tr><th scope=row>GO:0022627</th><td>CC</td><td>GO:0022627</td><td>cytosolic small ribosomal subunit                              </td><td>3/49</td><td>41/19886 </td><td>1.403123e-04</td><td>6.980537e-03</td><td>5.169401e-03</td><td>RPS25/RPS21/RPS24                                  </td><td>3</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00159342</th><td>CC</td><td>GO:0015934</td><td>large ribosomal subunit                                        </td><td>4/49</td><td>117/19886</td><td>1.965645e-04</td><td>8.157427e-03</td><td>6.362482e-03</td><td>RPL37/RPL31/RPL27A/RPL38                           </td><td>4</td><td>Inh_V2</td></tr>
	<tr><th scope=row>GO:00058402</th><td>CC</td><td>GO:0005840</td><td>ribosome                                                       </td><td>5/49</td><td>246/19886</td><td>3.402385e-04</td><td>1.129592e-02</td><td>8.810387e-03</td><td>RPL37/RPS25/RPL31/RPL27A/RPL38                     </td><td>5</td><td>Inh_V2</td></tr>
	<tr><th scope=row>GO:0031209</th><td>CC</td><td>GO:0031209</td><td>SCAR complex                                                   </td><td>2/49</td><td>12/19886 </td><td>3.864240e-04</td><td>1.364694e-02</td><td>1.010617e-02</td><td>ABI2/NCKAP1                                        </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00226251</th><td>CC</td><td>GO:0022625</td><td>cytosolic large ribosomal subunit                              </td><td>3/49</td><td>60/19886 </td><td>4.358534e-04</td><td>1.364694e-02</td><td>1.010617e-02</td><td>RPL37/RPL31/RPL23                                  </td><td>3</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0035580</th><td>CC</td><td>GO:0035580</td><td>specific granule lumen                                         </td><td>3/49</td><td>62/19886 </td><td>4.800431e-04</td><td>1.364694e-02</td><td>1.010617e-02</td><td>SPTAN1/KPNB1/CTSD                                  </td><td>3</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00312522</th><td>CC</td><td>GO:0031252</td><td>cell leading edge                                              </td><td>6/49</td><td>423/19886</td><td>5.767581e-04</td><td>1.595697e-02</td><td>1.244583e-02</td><td>GABRG3/FERMT2/ATP6V1B2/SHISA8/P4HB/PTPRK           </td><td>6</td><td>Inh_V2</td></tr>
	<tr><th scope=row>GO:00037352</th><td>MF</td><td>GO:0003735</td><td>structural constituent of ribosome                             </td><td>5/49</td><td>176/18496</td><td>1.001360e-04</td><td>1.522067e-02</td><td>1.264875e-02</td><td>RPL37/RPS25/RPL31/RPL27A/RPL38                     </td><td>5</td><td>Inh_V2</td></tr>
	<tr><th scope=row>GO:00306954</th><td>MF</td><td>GO:0030695</td><td>GTPase regulator activity                                      </td><td>7/49</td><td>486/18496</td><td>2.736939e-04</td><td>1.569179e-02</td><td>1.334858e-02</td><td>ARHGEF25/RAPGEF2/DENND6A/RAP1GAP2/DENND5A/PSD/ITSN2</td><td>7</td><td>OPC_V2</td></tr>
	<tr><th scope=row>GO:00605894</th><td>MF</td><td>GO:0060589</td><td>nucleoside-triphosphatase regulator activity                   </td><td>7/49</td><td>486/18496</td><td>2.736939e-04</td><td>1.569179e-02</td><td>1.334858e-02</td><td>ARHGEF25/RAPGEF2/DENND6A/RAP1GAP2/DENND5A/PSD/ITSN2</td><td>7</td><td>OPC_V2</td></tr>
	<tr><th scope=row>GO:00159351</th><td>CC</td><td>GO:0015935</td><td>small ribosomal subunit                                        </td><td>3/49</td><td>75/19886 </td><td>8.380463e-04</td><td>2.084640e-02</td><td>1.543769e-02</td><td>RPS25/RPS21/RPS24                                  </td><td>3</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00331781</th><td>CC</td><td>GO:0033178</td><td>proton-transporting two-sector ATPase complex, catalytic domain</td><td>2/49</td><td>19/19886 </td><td>9.902206e-04</td><td>2.189488e-02</td><td>1.621414e-02</td><td>ATP6V1F/ATP5F1D                                    </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0055105</th><td>MF</td><td>GO:0055105</td><td>ubiquitin-protein transferase inhibitor activity               </td><td>2/48</td><td>11/18496 </td><td>3.573447e-04</td><td>2.120245e-02</td><td>1.805531e-02</td><td>RPL37/RPL23                                        </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00312561</th><td>CC</td><td>GO:0031256</td><td>leading edge membrane                                          </td><td>4/49</td><td>179/19886</td><td>9.805649e-04</td><td>2.325340e-02</td><td>1.813676e-02</td><td>GABRG3/FERMT2/SHISA8/PTPRK                         </td><td>4</td><td>Inh_V2</td></tr>
	<tr><th scope=row>GO:0005940</th><td>CC</td><td>GO:0005940</td><td>septin ring                                                    </td><td>2/48</td><td>14/19886 </td><td>5.096482e-04</td><td>2.538551e-02</td><td>1.935012e-02</td><td>SEPTIN3/SEPTIN7                                    </td><td>2</td><td>OPC_V2</td></tr>
	<tr><th scope=row>GO:0031105</th><td>CC</td><td>GO:0031105</td><td>septin complex                                                 </td><td>2/48</td><td>14/19886 </td><td>5.096482e-04</td><td>2.538551e-02</td><td>1.935012e-02</td><td>SEPTIN3/SEPTIN7                                    </td><td>2</td><td>OPC_V2</td></tr>
	<tr><th scope=row>GO:0032156</th><td>CC</td><td>GO:0032156</td><td>septin cytoskeleton                                            </td><td>2/48</td><td>16/19886 </td><td>6.699962e-04</td><td>2.538551e-02</td><td>1.935012e-02</td><td>SEPTIN3/SEPTIN7                                    </td><td>2</td><td>OPC_V2</td></tr>
	<tr><th scope=row>GO:00321531</th><td>CC</td><td>GO:0032153</td><td>cell division site                                             </td><td>3/48</td><td>72/19886 </td><td>7.002900e-04</td><td>2.538551e-02</td><td>1.935012e-02</td><td>SEPTIN3/SEPTIN7/PSD                                </td><td>3</td><td>OPC_V2</td></tr>
	<tr><th scope=row>GO:0042809</th><td>MF</td><td>GO:0042809</td><td>nuclear vitamin D receptor binding                             </td><td>2/49</td><td>15/18496 </td><td>7.062051e-04</td><td>3.036682e-02</td><td>2.583224e-02</td><td>MED24/MED17                                        </td><td>2</td><td>OPC_V2</td></tr>
	<tr><th scope=row>GO:0034315</th><td>BP</td><td>GO:0034315</td><td>regulation of Arp2/3 complex-mediated actin nucleation         </td><td>3/47</td><td>31/18870 </td><td>6.198475e-05</td><td>3.926734e-02</td><td>3.298241e-02</td><td>ARF1/ABI2/NCKAP1                                   </td><td>3</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0070064</th><td>MF</td><td>GO:0070064</td><td>proline-rich region binding                                    </td><td>2/48</td><td>17/18496 </td><td>8.748778e-04</td><td>3.893206e-02</td><td>3.315326e-02</td><td>ABI2/RABAC1                                        </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00021812</th><td>BP</td><td>GO:0002181</td><td>cytoplasmic translation                                        </td><td>5/49</td><td>156/18870</td><td>5.149146e-05</td><td>3.815517e-02</td><td>3.441798e-02</td><td>RPL37/RPS25/RPL31/RPL27A/RPL38                     </td><td>5</td><td>Inh_V2</td></tr>
	<tr><th scope=row>GO:0042788</th><td>CC</td><td>GO:0042788</td><td>polysomal ribosome                                             </td><td>2/49</td><td>31/19886 </td><td>2.642407e-03</td><td>4.867641e-02</td><td>3.604707e-02</td><td>RPL31/RPS21                                        </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00986871</th><td>CC</td><td>GO:0098687</td><td>chromosomal region                                             </td><td>5/49</td><td>399/19886</td><td>2.928934e-03</td><td>4.867641e-02</td><td>3.604707e-02</td><td>PCNA/SIRT2/CEBPB/BOD1/PPP2R1A                      </td><td>5</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00159341</th><td>CC</td><td>GO:0015934</td><td>large ribosomal subunit                                        </td><td>3/49</td><td>117/19886</td><td>3.002945e-03</td><td>4.867641e-02</td><td>3.604707e-02</td><td>RPL37/RPL31/RPL23                                  </td><td>3</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0034358</th><td>CC</td><td>GO:0034358</td><td>plasma lipoprotein particle                                    </td><td>2/49</td><td>36/19886 </td><td>3.552057e-03</td><td>4.867641e-02</td><td>3.604707e-02</td><td>SELENOS/CLU                                        </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:1990777</th><td>CC</td><td>GO:1990777</td><td>lipoprotein particle                                           </td><td>2/49</td><td>36/19886 </td><td>3.552057e-03</td><td>4.867641e-02</td><td>3.604707e-02</td><td>SELENOS/CLU                                        </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00059251</th><td>CC</td><td>GO:0005925</td><td>focal adhesion                                                 </td><td>5/49</td><td>421/19886</td><td>3.683860e-03</td><td>4.867641e-02</td><td>3.604707e-02</td><td>RPL31/ARF1/RPL23/CD81/NCKAP1                       </td><td>5</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00300551</th><td>CC</td><td>GO:0030055</td><td>cell-substrate junction                                        </td><td>5/49</td><td>431/19886</td><td>4.069648e-03</td><td>4.867641e-02</td><td>3.604707e-02</td><td>RPL31/ARF1/RPL23/CD81/NCKAP1                       </td><td>5</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0032994</th><td>CC</td><td>GO:0032994</td><td>protein-lipid complex                                          </td><td>2/49</td><td>39/19886 </td><td>4.158287e-03</td><td>4.867641e-02</td><td>3.604707e-02</td><td>SELENOS/CLU                                        </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0070847</th><td>CC</td><td>GO:0070847</td><td>core mediator complex                                          </td><td>2/48</td><td>26/19886 </td><td>1.786860e-03</td><td>5.181895e-02</td><td>3.949902e-02</td><td>MED24/MED17                                        </td><td>2</td><td>OPC_V2</td></tr>
	<tr><th scope=row>GO:00427881</th><td>CC</td><td>GO:0042788</td><td>polysomal ribosome                                             </td><td>2/49</td><td>31/19886 </td><td>2.642407e-03</td><td>5.482994e-02</td><td>4.276527e-02</td><td>RPL31/RPL38                                        </td><td>2</td><td>Inh_V2</td></tr>
	<tr><th scope=row>GO:0001772</th><td>CC</td><td>GO:0001772</td><td>immunological synapse                                          </td><td>2/49</td><td>45/19886 </td><td>5.503617e-03</td><td>5.970137e-02</td><td>4.421154e-02</td><td>ARHGDIA/CD81                                       </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00058741</th><td>CC</td><td>GO:0005874</td><td>microtubule                                                    </td><td>5/49</td><td>467/19886</td><td>5.700131e-03</td><td>5.970137e-02</td><td>4.421154e-02</td><td>TUBB2A/SIRT2/TUBA1A/SELENOS/BOD1                   </td><td>5</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0044389</th><td>MF</td><td>GO:0044389</td><td>ubiquitin-like protein ligase binding                          </td><td>5/48</td><td>327/18496</td><td>1.538542e-03</td><td>5.477208e-02</td><td>4.664210e-02</td><td>CKB/RPL23/CEBPB/ABI2/CLU                           </td><td>5</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0051125</th><td>BP</td><td>GO:0051125</td><td>regulation of actin nucleation                                 </td><td>3/47</td><td>40/18870 </td><td>1.341177e-04</td><td>5.664238e-02</td><td>4.757649e-02</td><td>ARF1/ABI2/NCKAP1                                   </td><td>3</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00164691</th><td>CC</td><td>GO:0016469</td><td>proton-transporting two-sector ATPase complex                  </td><td>2/49</td><td>50/19886 </td><td>6.756931e-03</td><td>6.439794e-02</td><td>4.768957e-02</td><td>ATP6V1F/ATP5F1D                                    </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0042581</th><td>CC</td><td>GO:0042581</td><td>specific granule                                               </td><td>3/49</td><td>160/19886</td><td>7.182892e-03</td><td>6.439794e-02</td><td>4.768957e-02</td><td>SPTAN1/KPNB1/CTSD                                  </td><td>3</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:0005788</th><td>CC</td><td>GO:0005788</td><td>endoplasmic reticulum lumen                                    </td><td>4/49</td><td>313/19886</td><td>7.312499e-03</td><td>6.439794e-02</td><td>4.768957e-02</td><td>RCN2/VGF/SCG2/CLU                                  </td><td>4</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00347741</th><td>CC</td><td>GO:0034774</td><td>secretory granule lumen                                        </td><td>4/49</td><td>322/19886</td><td>8.064040e-03</td><td>6.439794e-02</td><td>4.768957e-02</td><td>SPTAN1/KPNB1/CTSD/CLU                              </td><td>4</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:1904724</th><td>CC</td><td>GO:1904724</td><td>tertiary granule lumen                                         </td><td>2/49</td><td>55/19886 </td><td>8.127240e-03</td><td>6.439794e-02</td><td>4.768957e-02</td><td>SPTAN1/CTSD                                        </td><td>2</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00602051</th><td>CC</td><td>GO:0060205</td><td>cytoplasmic vesicle lumen                                      </td><td>4/49</td><td>325/19886</td><td>8.325442e-03</td><td>6.439794e-02</td><td>4.768957e-02</td><td>SPTAN1/KPNB1/CTSD/CLU                              </td><td>4</td><td>Exc_V2</td></tr>
	<tr><th scope=row>GO:00319831</th><td>CC</td><td>GO:0031983</td><td>vesicle lumen                                                  </td><td>4/49</td><td>326/19886</td><td>8.413802e-03</td><td>6.439794e-02</td><td>4.768957e-02</td><td>SPTAN1/KPNB1/CTSD/CLU                              </td><td>4</td><td>Exc_V2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>fwrite(progm2_cor_df, &#39;GO.Top50_Cor_genes_Program2_split.txt&#39;, sep = &#39;\t&#39;)
</pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="id6">
<h3>check gene expression<a class="headerlink" href="#id6" title="Link to this heading">#</a></h3>
<p>we are wondering why CORO1A was picked by cTWAS, wondering if that gene has a higher expression correlation with others</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>mic_exp &lt;- fread(&#39;/data/eQTL/ROSMAP/DLPFC_Mic/analysis_ready/phenotype_preprocessing/phenotype_by_chrom/Mic.log2cpm.bed.chr11.bed.gz&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>mic_exp %&gt;% filter(ID == &#39;ENSG00000066336&#39;)
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 1 x 428</caption>
<thead>
	<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>ID</th><th scope=col>MAP15387421</th><th scope=col>MAP22868024</th><th scope=col>MAP46246604</th><th scope=col>MAP50104134</th><th scope=col>MAP50104846</th><th scope=col>MAP50106442</th><th scope=col>...</th><th scope=col>SM-CTEMS</th><th scope=col>SM-CTEMU</th><th scope=col>SM-CTEMW</th><th scope=col>SM-CTEMY</th><th scope=col>SM-CTEN2</th><th scope=col>SM-CTEN5</th><th scope=col>SM-CTEN7</th><th scope=col>SM-CTEND</th><th scope=col>SM-CTENF</th><th scope=col>SM-CTENG</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>...</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr11</td><td>47378546</td><td>47378547</td><td>ENSG00000066336</td><td>0.3710662</td><td>-1.778669</td><td>0.7291951</td><td>-0.7291951</td><td>-0.9848335</td><td>1.054258</td><td>...</td><td>-0.168891</td><td>1.650584</td><td>-0.3964614</td><td>-0.1808688</td><td>0.661588</td><td>-1.525803</td><td>2.454256</td><td>-0.4480493</td><td>0.1509692</td><td>-0.3271831</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>spi1_fmp &lt;- readRDS(&#39;/data/analysis_result/single_context/ROSMAP_eQTL/export/ROSMAP_eQTL.ENSG00000066336.cis_results_db.export.rds&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>spi1_fmp[[1]] %&gt;% names
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.list-inline {list-style: none; margin:0; padding: 0}
.list-inline>li {display: inline-block}
.list-inline>li:not(:last-child)::after {content: "\00b7"; padding: 0 .5ex}
</style>
<ol class=list-inline><li>'Mic_DeJager_eQTL_ENSG00000066336'</li><li>'Ast_DeJager_eQTL_ENSG00000066336'</li><li>'Inh_DeJager_eQTL_ENSG00000066336'</li><li>'DLPFC_DeJager_eQTL_ENSG00000066336'</li><li>'PCC_DeJager_eQTL_ENSG00000066336'</li><li>'AC_DeJager_eQTL_ENSG00000066336'</li><li>'Mic_Kellis_eQTL_ENSG00000066336'</li><li>'Mic_mega_eQTL_ENSG00000066336'</li></ol>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>spi1_fmp[[1]][[&#39;Mic_DeJager_eQTL_ENSG00000066336&#39;]][[&#39;top_loci&#39;]]
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 0 x 8</caption>
<thead>
	<tr><th scope=col>variant_id</th><th scope=col>betahat</th><th scope=col>sebetahat</th><th scope=col>maf</th><th scope=col>pip</th><th scope=col>cs_coverage_0.95</th><th scope=col>cs_coverage_0.7</th><th scope=col>cs_coverage_0.5</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>
</thead>
<tbody>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>spi1_fmp[[1]][[&#39;Mic_Kellis_eQTL_ENSG00000066336&#39;]][[&#39;top_loci&#39;]]
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>NULL
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-sos notranslate"><div class="highlight"><pre><span></span>spi1_fmp[[1]][[&#39;Mic_mega_eQTL_ENSG00000066336&#39;]][[&#39;top_loci&#39;]]
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>NULL
</pre></div>
</div>
</div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "sos"
        },
        kernelOptions: {
            name: "sos",
            path: "./main_text/5_AD_xQTL_integration/staging"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'sos'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#overview">Overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-gene-lists">load gene lists</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-colocboost-results-for-double-check">load colocboost results for double check</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#identify-overlapping-genes-between-metabolite-colocalized-genes-and-ad-genes-to-uncover-potential-links-between-ad-and-metabolites">Identify overlapping genes between metabolite-colocalized genes and AD genes to uncover potential links between AD and metabolites</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#split-twas-gene-lists">split twas gene lists</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#go-enrichment">GO enrichment</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#real-gvc-list-20-genes-from-gvc-table2">real gvc list - 20 genes from GVC table2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gvc-list1-76-genes">gvc list1 -76 genes</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#strong-list-overlapped-between-xqtl-gvc-and-twas">strong list - overlapped between xQTL, GVC and TWAS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-list">xQTL list</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#go">GO</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#kegg">KEGG</a><ul class="nav section-nav flex-column">
<li class="toc-h5 nav-item toc-entry"><a class="reference internal nav-link" href="#plot-ad-path">plot AD path</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-list">xQTL only list</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id1">GO</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id2">KEGG</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#amyloid-pathway-genes">Amyloid pathway genes</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#context-gene-diff-list">context: gene diff list</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#twas-only-list">twas only list</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-case-ypel3">multi gene case - YPEL3</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-case-nbeal1">multi gene case - NBEAL1</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#ppi">PPI</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl">xQTL</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-genes-labelled-by-methods">xQTL only genes labelled by methods</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-genes-labelled-by-non-proxy-ad-genes">xQTL only genes labelled by non proxy AD genes</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-gvs-table2">xQTL + GVS table2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#check-gene-expression">check gene expression</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#prediction">Prediction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#model-with-glmnet">model with glmnet</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#model-with-glm">model with glm</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#cross-sample">cross sample</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#grn">GRN</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#fisher-test-between-xqtl-and-xqtl-only-genes">fisher test between xQTL and xQTL only genes</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multigene-grn-ypel3">Multigene GRN -YPEL3</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multigene-grn-stx4">Multigene GRN - STX4</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-trans-function">Multi Gene trans function</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#program1-go">program1 GO</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#split">split</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#another-top-split">another top split</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#program1-kegg">program1 KEGG</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id3">split</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#program2-go">program2 GO</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id4">split</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#id5">another top split</a></li>
</ul>
</li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#id6">check gene expression</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>