
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Gene Set Enrichment Analysis for AD Risk Genes &#8212; FunGen-xQTL Manuscript Repository</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'main_text/5_AD_xQTL_genes_cis_trans/Figure_5_GSEA';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
        
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../README.html">
  
  
  
  
  
  
    <p class="title logo__title">FunGen-xQTL Manuscript Repository</p>
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../README.html">
                    About
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../1_project_overview/index.html">Multi-omics QTL atlas of the aging human brain</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../2_single_context_cis/index.html">Fine-mapped QTL landscape across brain molecular phenotypes</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="simple">
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/StatFunGen/xqtl-paper/issues/new?title=Issue%20on%20page%20%2Fmain_text/5_AD_xQTL_genes_cis_trans/Figure_5_GSEA.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/main_text/5_AD_xQTL_genes_cis_trans/Figure_5_GSEA.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Gene Set Enrichment Analysis for AD Risk Genes</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#motivation">Motivation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#implementation">Implementation</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-libraries">Load libraries</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-gene-and-twas-list">Load gene and TWAS list</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#real-gvc-list-20-genes-from-gvc-table-2">Real GVC list - 20 genes from GVC table 2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gvc-list-1-76-genes">GVC list 1 - 76 genes</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#strong-list-overlapped-between-xqtl-gvc-and-twas">Strong list - overlapped between xQTL, GVC and TWAS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-list-go-enrichment">xQTL list - GO enrichment</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-list-kegg-enrichment">xQTL list - KEGG enrichment</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plot-ad-functional-pathway-xqtl-list">Plot AD functional pathway - xQTL list</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-list-go-enrichment">xQTL only list - GO enrichment</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-list-kegg-enrichment">xQTL only list - KEGG enrichment</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plot-ad-functional-pathway-xqtl-only-list">Plot AD functional pathway - xQTL only list</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#twas-only-list">TWAS only list</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-case-ypel3">multi gene case - YPEL3</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-case-nbeal1">multi gene case - NBEAL1</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section id="gene-set-enrichment-analysis-for-ad-risk-genes">
<h1>Gene Set Enrichment Analysis for AD Risk Genes<a class="headerlink" href="#gene-set-enrichment-analysis-for-ad-risk-genes" title="Link to this heading">#</a></h1>
<p>Author: Ru Feng and Jenny Empawi</p>
<section id="motivation">
<h2>Motivation<a class="headerlink" href="#motivation" title="Link to this heading">#</a></h2>
<p>Perform gene set enrichment analysis for AD risk genes identified via xQTL. Highlight the pathway enrichment for novel genes to uncover new insights.</p>
</section>
<section id="implementation">
<h2>Implementation<a class="headerlink" href="#implementation" title="Link to this heading">#</a></h2>
<p>This notebook shows pathway enrichment analysis for the identified AD risk genes which includes:</p>
<ul class="simple">
<li><p>Real GVC list - 20 genes from GVC Table 2 (GO enrichment only)</p></li>
<li><p>GVC list 1 - 76 genes (GO enrichment only)</p></li>
<li><p>Strong list - overlapped between xQTL, GVC and TWAS (GO enrichment only)</p></li>
<li><p>xQTL list (GO and KEGG enrichment)</p></li>
<li><p>xQTL only list (GO and KEGG enrichment)</p></li>
<li><p>TWAS only list (GO enrichment only)</p></li>
<li><p>multi gene case - YPEL3, NBEAL1 (GO and KEGG enrichment for YPEL3; no KEGG plot because only one pathway enriched)</p></li>
</ul>
<section id="load-libraries">
<h3>Load libraries<a class="headerlink" href="#load-libraries" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">clusterProfiler</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">org.Hs.eg.db</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">STRINGdb</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">pathview</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">png</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">grid</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>clusterProfiler v4.12.6 Learn more at https://yulab-smu.top/contribution-knowledge-mining/

Please cite:

S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,
W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize
multiomics data. Nature Protocols. 2024, 19(11):3292-3320


Attaching package: ‘clusterProfiler’


The following object is masked from ‘package:stats’:

    filter


Loading required package: AnnotationDbi

Loading required package: stats4

Loading required package: BiocGenerics


Attaching package: ‘BiocGenerics’


The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs


The following objects are masked from ‘package:base’:

    Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,
    as.data.frame, basename, cbind, colnames, dirname, do.call,
    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,
    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, rank, rbind, rownames, sapply, setdiff, table, tapply,
    union, unique, unsplit, which.max, which.min


Loading required package: Biobase

Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.


Loading required package: IRanges

Loading required package: S4Vectors


Attaching package: ‘S4Vectors’


The following object is masked from ‘package:clusterProfiler’:

    rename


The following object is masked from ‘package:utils’:

    findMatches


The following objects are masked from ‘package:base’:

    I, expand.grid, unname



Attaching package: ‘IRanges’


The following object is masked from ‘package:clusterProfiler’:

    slice



Attaching package: ‘AnnotationDbi’


The following object is masked from ‘package:clusterProfiler’:

    select





Attaching package: ‘data.table’


The following object is masked from ‘package:IRanges’:

    shift


The following objects are masked from ‘package:S4Vectors’:

    first, second


── <span class=" -Color -Color-Bold">Attaching core tidyverse packages</span> ──────────────────────── tidyverse 2.0.0 ──
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">dplyr    </span> 1.1.4     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">readr    </span> 2.1.5
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">forcats  </span> 1.0.0     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">stringr  </span> 1.5.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">ggplot2  </span> 3.5.1     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tibble   </span> 3.2.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">lubridate</span> 1.9.3     <span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">tidyr    </span> 1.3.1
<span class=" -Color -Color-Green">✔</span> <span class=" -Color -Color-Blue">purrr    </span> 1.0.2     
── <span class=" -Color -Color-Bold">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">%within%()</span> masks <span class=" -Color -Color-Blue">IRanges</span>::%within%()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">ggplot2</span>::<span class=" -Color -Color-Green">Position()</span>   masks <span class=" -Color -Color-Blue">BiocGenerics</span>::Position(), <span class=" -Color -Color-Blue">base</span>::Position()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">between()</span>      masks <span class=" -Color -Color-Blue">data.table</span>::between()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">collapse()</span>     masks <span class=" -Color -Color-Blue">IRanges</span>::collapse()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">combine()</span>      masks <span class=" -Color -Color-Blue">Biobase</span>::combine(), <span class=" -Color -Color-Blue">BiocGenerics</span>::combine()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">desc()</span>         masks <span class=" -Color -Color-Blue">IRanges</span>::desc()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">tidyr</span>::<span class=" -Color -Color-Green">expand()</span>       masks <span class=" -Color -Color-Blue">S4Vectors</span>::expand()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">filter()</span>       masks <span class=" -Color -Color-Blue">clusterProfiler</span>::filter(), <span class=" -Color -Color-Blue">stats</span>::filter()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">first()</span>        masks <span class=" -Color -Color-Blue">data.table</span>::first(), <span class=" -Color -Color-Blue">S4Vectors</span>::first()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">hour()</span>     masks <span class=" -Color -Color-Blue">data.table</span>::hour()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">isoweek()</span>  masks <span class=" -Color -Color-Blue">data.table</span>::isoweek()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">lag()</span>          masks <span class=" -Color -Color-Blue">stats</span>::lag()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">last()</span>         masks <span class=" -Color -Color-Blue">data.table</span>::last()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">mday()</span>     masks <span class=" -Color -Color-Blue">data.table</span>::mday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">minute()</span>   masks <span class=" -Color -Color-Blue">data.table</span>::minute()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">month()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::month()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">quarter()</span>  masks <span class=" -Color -Color-Blue">data.table</span>::quarter()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">purrr</span>::<span class=" -Color -Color-Green">reduce()</span>       masks <span class=" -Color -Color-Blue">IRanges</span>::reduce()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">rename()</span>       masks <span class=" -Color -Color-Blue">S4Vectors</span>::rename(), <span class=" -Color -Color-Blue">clusterProfiler</span>::rename()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">second()</span>   masks <span class=" -Color -Color-Blue">data.table</span>::second(), <span class=" -Color -Color-Blue">S4Vectors</span>::second()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">second&lt;-()</span> masks <span class=" -Color -Color-Blue">S4Vectors</span>::second&lt;-()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">select()</span>       masks <span class=" -Color -Color-Blue">AnnotationDbi</span>::select(), <span class=" -Color -Color-Blue">clusterProfiler</span>::select()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">purrr</span>::<span class=" -Color -Color-Green">simplify()</span>     masks <span class=" -Color -Color-Blue">clusterProfiler</span>::simplify()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">dplyr</span>::<span class=" -Color -Color-Green">slice()</span>        masks <span class=" -Color -Color-Blue">IRanges</span>::slice(), <span class=" -Color -Color-Blue">clusterProfiler</span>::slice()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">purrr</span>::<span class=" -Color -Color-Green">transpose()</span>    masks <span class=" -Color -Color-Blue">data.table</span>::transpose()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">wday()</span>     masks <span class=" -Color -Color-Blue">data.table</span>::wday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">week()</span>     masks <span class=" -Color -Color-Blue">data.table</span>::week()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">yday()</span>     masks <span class=" -Color -Color-Blue">data.table</span>::yday()
<span class=" -Color -Color-Red">✖</span> <span class=" -Color -Color-Blue">lubridate</span>::<span class=" -Color -Color-Green">year()</span>     masks <span class=" -Color -Color-Blue">data.table</span>::year()
<span class=" -Color -Color-Cyan">ℹ</span> Use the conflicted package (<span class=" -Color -Color-Blue">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors
##############################################################################
Pathview is an open source software package distributed under GNU General
Public License version 3 (GPLv3). Details of GPLv3 is available at
http://www.gnu.org/licenses/gpl-3.0.html. Particullary, users are required to
formally cite the original Pathview paper (not just mention it) in publications
or products. For details, do citation(&quot;pathview&quot;) within R.

The pathview downloads and uses KEGG data. Non-academic uses may require a KEGG
license agreement (details at http://www.kegg.jp/kegg/legal.html).
##############################################################################
</pre></div>
</div>
</div>
</div>
</section>
<section id="load-gene-and-twas-list">
<h3>Load gene and TWAS list<a class="headerlink" href="#load-gene-and-twas-list" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">gene_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Staging/variant_gene_summary/AD_xQTL_integration_summary/AD_genes_FunGen_AD_GVC_xQTL_20250325.tsv&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
<span class="n">twas_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&quot;/restricted/projectnb/xqtl/jaempawi/xqtl/Staging/variant_gene_summary/AD_xQTL_integration_summary/AD_genes_FunGen_AD_twas_GVC_xQTL_20250325.tsv&quot;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="real-gvc-list-20-genes-from-gvc-table-2">
<h3>Real GVC list - 20 genes from GVC table 2<a class="headerlink" href="#real-gvc-list-20-genes-from-gvc-table-2" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load real gvc list</span>
<span class="n">real_gvc_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">data_resources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;ADSP_GVC&#39;</span><span class="p">))</span><span class="w"> </span><span class="c1"># 20 genes from GVC table2</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Run GO enrichment analysis for Biological Process (BP)</span>
<span class="n">ego</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span>
<span class="w">  </span><span class="n">gene</span><span class="w">          </span><span class="o">=</span><span class="w"> </span><span class="n">real_gvc_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span>
<span class="w">  </span><span class="n">OrgDb</span><span class="w">         </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">  </span><span class="n">keyType</span><span class="w">       </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">ont</span><span class="w">           </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">  </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">0.2</span>
<span class="p">)</span>

<span class="c1"># View results</span>
<span class="nf">head</span><span class="p">(</span><span class="n">ego</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process                       </td><td>9/21</td><td>85/21288 </td><td>4.713703e-17</td><td>7.188397e-14</td><td>3.349210e-14</td><td>ENSG00000064687/ENSG00000073921/ENSG00000080815/ENSG00000130203/ENSG00000136717/ENSG00000137642/ENSG00000137845/ENSG00000143801/ENSG00000288674</td><td>9</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process                       </td><td>9/21</td><td>114/21288</td><td>7.311827e-16</td><td>5.510498e-13</td><td>2.567445e-13</td><td>ENSG00000064687/ENSG00000073921/ENSG00000080815/ENSG00000130203/ENSG00000136717/ENSG00000137642/ENSG00000137845/ENSG00000143801/ENSG00000288674</td><td>9</td></tr>
	<tr><th scope=row>GO:0034205</th><td>GO:0034205</td><td>amyloid-beta formation                                            </td><td>8/21</td><td>66/21288 </td><td>1.084032e-15</td><td>5.510498e-13</td><td>2.567445e-13</td><td>ENSG00000064687/ENSG00000073921/ENSG00000080815/ENSG00000130203/ENSG00000136717/ENSG00000137642/ENSG00000143801/ENSG00000288674                </td><td>8</td></tr>
	<tr><th scope=row>GO:0050435</th><td>GO:0050435</td><td>amyloid-beta metabolic process                                    </td><td>8/21</td><td>74/21288 </td><td>2.832204e-15</td><td>1.079778e-12</td><td>5.030889e-13</td><td>ENSG00000064687/ENSG00000073921/ENSG00000080815/ENSG00000130203/ENSG00000136717/ENSG00000137642/ENSG00000143801/ENSG00000288674                </td><td>8</td></tr>
	<tr><th scope=row>GO:1902992</th><td>GO:1902992</td><td>negative regulation of amyloid precursor protein catabolic process</td><td>5/21</td><td>34/21288 </td><td>1.526875e-10</td><td>4.656968e-08</td><td>2.169769e-08</td><td>ENSG00000064687/ENSG00000073921/ENSG00000130203/ENSG00000136717/ENSG00000137642                                                                </td><td>5</td></tr>
	<tr><th scope=row>GO:1902003</th><td>GO:1902003</td><td>regulation of amyloid-beta formation                              </td><td>5/21</td><td>57/21288 </td><td>2.264684e-09</td><td>5.395839e-07</td><td>2.514023e-07</td><td>ENSG00000064687/ENSG00000073921/ENSG00000130203/ENSG00000136717/ENSG00000137642                                                                </td><td>5</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Plot results</span>
<span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">ego</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GO Enrichment - BP&quot;</span><span class="p">)</span>
<span class="n">p1</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/55fd2632e7e4fddcbf50af0e6f203eccc4aefe9f1de283d520ab5d4c157afd54.png"><img alt="../../_images/55fd2632e7e4fddcbf50af0e6f203eccc4aefe9f1de283d520ab5d4c157afd54.png" src="../../_images/55fd2632e7e4fddcbf50af0e6f203eccc4aefe9f1de283d520ab5d4c157afd54.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="gvc-list-1-76-genes">
<h3>GVC list 1 - 76 genes<a class="headerlink" href="#gvc-list-1-76-genes" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load gene reference</span>
<span class="n">gene_ref</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/Figure_5f/Homo_sapiens.GRCh38.103.chr.reformatted.collapse_only.gene.region_list&#39;</span><span class="p">)</span><span class="w"> </span><span class="c1">#FIXME</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Load gvc list1</span>
<span class="n">gvc_table1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">fread</span><span class="p">(</span><span class="s">&#39;/restricted/projectnb/xqtl/jaempawi/xqtl/GVC - table1.tsv&#39;</span><span class="p">)</span><span class="w">  </span><span class="c1">#FIXME</span>

<span class="c1"># Set column names from the second row and remove header rows</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">gvc_table1</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gvc_table1</span><span class="p">[</span><span class="m">2</span><span class="p">,]</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">as.character</span><span class="p">()</span>
<span class="n">gvc_table1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gvc_table1</span><span class="p">[</span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">),]</span>

<span class="c1"># Map gene names to gene reference data and add resource identifier</span>
<span class="n">gvc_table1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_ref</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">gvc_table1</span><span class="o">$</span><span class="n">`Reported Gene/ Closest gene`</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">data_resource</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;FunGen_ADSP&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Run GO enrichment analysis for Biological Process (BP)</span>
<span class="n">gvc_table1_ego</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span>
<span class="w">  </span><span class="n">gene</span><span class="w">          </span><span class="o">=</span><span class="w"> </span><span class="n">gvc_table1</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span>
<span class="w">  </span><span class="n">OrgDb</span><span class="w">         </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">  </span><span class="n">keyType</span><span class="w">       </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">ont</span><span class="w">           </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">  </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">0.2</span>
<span class="p">)</span>

<span class="c1"># View results</span>
<span class="nf">head</span><span class="p">(</span><span class="n">gvc_table1_ego</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0050435</th><td>GO:0050435</td><td>amyloid-beta metabolic process                     </td><td>7/73</td><td>74/21288 </td><td>6.223514e-09</td><td>1.338056e-05</td><td>9.047024e-06</td><td>ENSG00000136717/ENSG00000196549/ENSG00000120885/ENSG00000137642/ENSG00000138613/ENSG00000159640/ENSG00000064687                                </td><td>7</td></tr>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process        </td><td>6/73</td><td>85/21288 </td><td>4.656406e-07</td><td>4.904041e-04</td><td>3.315780e-04</td><td>ENSG00000151694/ENSG00000136717/ENSG00000120885/ENSG00000137642/ENSG00000138613/ENSG00000064687                                                </td><td>6</td></tr>
	<tr><th scope=row>GO:0002274</th><td>GO:0002274</td><td>myeloid leukocyte activation                       </td><td>9/73</td><td>292/21288</td><td>6.842848e-07</td><td>4.904041e-04</td><td>3.315780e-04</td><td>ENSG00000109684/ENSG00000168421/ENSG00000095970/ENSG00000120885/ENSG00000066336/ENSG00000197943/ENSG00000103241/ENSG00000030582/ENSG00000142192</td><td>9</td></tr>
	<tr><th scope=row>GO:0030183</th><td>GO:0030183</td><td>B cell differentiation                             </td><td>7/73</td><td>165/21288</td><td>1.568380e-06</td><td>8.430040e-04</td><td>5.699822e-04</td><td>ENSG00000203710/ENSG00000151694/ENSG00000168918/ENSG00000120899/ENSG00000066336/ENSG00000095585/ENSG00000197943                                </td><td>7</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process        </td><td>6/73</td><td>114/21288</td><td>2.625738e-06</td><td>9.624278e-04</td><td>6.507284e-04</td><td>ENSG00000151694/ENSG00000136717/ENSG00000120885/ENSG00000137642/ENSG00000138613/ENSG00000064687                                                </td><td>6</td></tr>
	<tr><th scope=row>GO:0002275</th><td>GO:0002275</td><td>myeloid cell activation involved in immune response</td><td>6/73</td><td>115/21288</td><td>2.762813e-06</td><td>9.624278e-04</td><td>6.507284e-04</td><td>ENSG00000109684/ENSG00000095970/ENSG00000066336/ENSG00000197943/ENSG00000103241/ENSG00000030582                                                </td><td>6</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Plot results</span>
<span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">gvc_table1_ego</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GO Enrichment - BP&quot;</span><span class="p">)</span>
<span class="n">p2</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/7d5382b939618bf7b174991a2b05625d255d7074b8773c4802bf8ed5eeced149.png"><img alt="../../_images/7d5382b939618bf7b174991a2b05625d255d7074b8773c4802bf8ed5eeced149.png" src="../../_images/7d5382b939618bf7b174991a2b05625d255d7074b8773c4802bf8ed5eeced149.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="strong-list-overlapped-between-xqtl-gvc-and-twas">
<h3>Strong list - overlapped between xQTL, GVC and TWAS<a class="headerlink" href="#strong-list-overlapped-between-xqtl-gvc-and-twas" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Intersect multiple gene_id vectors</span>
<span class="n">gvc_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">data_resources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;FunGen_AD_PI|ADSP_GVC&#39;</span><span class="p">))</span><span class="w"> </span><span class="c1"># 20 genes + PIs&#39; genes</span>
<span class="n">strong_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">Reduce</span><span class="p">(</span><span class="n">intersect</span><span class="p">,</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">gvc_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">gene_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">twas_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">))</span><span class="w"> </span><span class="c1"># the overlap between xQTL and gvc and twas</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">strong_list_ego</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span>
<span class="w">  </span><span class="n">gene</span><span class="w">          </span><span class="o">=</span><span class="w"> </span><span class="n">strong_list</span><span class="p">,</span>
<span class="w">  </span><span class="n">OrgDb</span><span class="w">         </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">  </span><span class="n">keyType</span><span class="w">       </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">ont</span><span class="w">           </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">  </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">0.2</span>
<span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">strong_list_ego</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process                       </td><td>5/21</td><td>85/21288 </td><td>1.743235e-08</td><td>2.256673e-05</td><td>1.137466e-05</td><td>ENSG00000073921/ENSG00000130203/ENSG00000136717/ENSG00000137642/ENSG00000137845</td><td>5</td></tr>
	<tr><th scope=row>GO:1902992</th><td>GO:1902992</td><td>negative regulation of amyloid precursor protein catabolic process</td><td>4/21</td><td>34/21288 </td><td>3.182895e-08</td><td>2.256673e-05</td><td>1.137466e-05</td><td>ENSG00000073921/ENSG00000130203/ENSG00000136717/ENSG00000137642                </td><td>4</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process                       </td><td>5/21</td><td>114/21288</td><td>7.661054e-08</td><td>3.621125e-05</td><td>1.825213e-05</td><td>ENSG00000073921/ENSG00000130203/ENSG00000136717/ENSG00000137642/ENSG00000137845</td><td>5</td></tr>
	<tr><th scope=row>GO:1902003</th><td>GO:1902003</td><td>regulation of amyloid-beta formation                              </td><td>4/21</td><td>57/21288 </td><td>2.671464e-07</td><td>9.470339e-05</td><td>4.773484e-05</td><td>ENSG00000073921/ENSG00000130203/ENSG00000136717/ENSG00000137642                </td><td>4</td></tr>
	<tr><th scope=row>GO:0034205</th><td>GO:0034205</td><td>amyloid-beta formation                                            </td><td>4/21</td><td>66/21288 </td><td>4.846271e-07</td><td>1.374402e-04</td><td>6.927616e-05</td><td>ENSG00000073921/ENSG00000130203/ENSG00000136717/ENSG00000137642                </td><td>4</td></tr>
	<tr><th scope=row>GO:1902991</th><td>GO:1902991</td><td>regulation of amyloid precursor protein catabolic process         </td><td>4/21</td><td>70/21288 </td><td>6.149636e-07</td><td>1.453364e-04</td><td>7.325620e-05</td><td>ENSG00000073921/ENSG00000130203/ENSG00000136717/ENSG00000137642                </td><td>4</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">strong_list_ego</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GO Enrichment - BP&quot;</span><span class="p">)</span>
<span class="n">p3</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/731ace587f233f20c512feb6eca532ce8c05a4d145c01a896c2ab2dc65e75526.png"><img alt="../../_images/731ace587f233f20c512feb6eca532ce8c05a4d145c01a896c2ab2dc65e75526.png" src="../../_images/731ace587f233f20c512feb6eca532ce8c05a4d145c01a896c2ab2dc65e75526.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="xqtl-list-go-enrichment">
<h3>xQTL list - GO enrichment<a class="headerlink" href="#xqtl-list-go-enrichment" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Initialize STRINGdb (change species if needed, 9606 = Homo sapiens)</span>
<span class="n">string_db</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">STRINGdb</span><span class="o">$</span><span class="nf">new</span><span class="p">(</span><span class="n">version</span><span class="o">=</span><span class="s">&quot;11.5&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="o">=</span><span class="m">9606</span><span class="p">,</span><span class="w"> </span><span class="n">score_threshold</span><span class="o">=</span><span class="m">400</span><span class="p">,</span><span class="w"> </span><span class="n">input_directory</span><span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">)</span>

<span class="c1"># Load xQTL list</span>
<span class="n">xQTL_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">data_resources</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;FunGen_xQTL&#39;</span><span class="p">))</span>

<span class="c1"># Load xQTL list</span>
<span class="n">genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">xQTL_list</span><span class="o">$</span><span class="n">gene_name</span>

<span class="c1"># Map gene names to STRING IDs</span>
<span class="n">mapped_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">string_db</span><span class="o">$</span><span class="nf">map</span><span class="p">(</span><span class="nf">data.frame</span><span class="p">(</span><span class="n">gene</span><span class="o">=</span><span class="n">genes</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;gene&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">removeUnmappedRows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># Retrieve interaction network for the genes</span>
<span class="n">ppi_network</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">string_db</span><span class="o">$</span><span class="nf">get_interactions</span><span class="p">(</span><span class="n">mapped_genes</span><span class="o">$</span><span class="n">STRING_id</span><span class="p">)</span>
<span class="n">genes_with_ppi</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mapped_genes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">STRING_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">ppi_network</span><span class="o">$</span><span class="n">from</span><span class="p">,</span><span class="w"> </span><span class="n">ppi_network</span><span class="o">$</span><span class="n">to</span><span class="p">))</span>

<span class="n">xQTL_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">resource_in_FunGen_xQTL</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">table</span>
<span class="n">xQTL_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="n">gene_name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">genes_with_ppi</span><span class="o">$</span><span class="n">gene</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">resource_in_FunGen_xQTL</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">table</span><span class="w"> </span><span class="c1"># the xQTL genes not have any PPI</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Warning:  we couldn&#39;t map to STRING 2% of your identifiers
</pre></div>
</div>
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>.
                                                  ColocBoost 
                                                         160 
                                    ColocBoost,multi_context 
                                                           2 
                         ColocBoost,multi_context,multi_gene 
                                                           1 
                                   ColocBoost,single_context 
                                                           7 
                         ColocBoost,single_context,MetaBrain 
                                                           1 
              ColocBoost,single_context,MetaBrain,multi_gene 
                                                           1 
                     ColocBoost,single_context,multi_context 
                                                          12 
           ColocBoost,single_context,multi_context,MetaBrain 
                                                           7 
ColocBoost,single_context,multi_context,MetaBrain,multi_gene 
                                                           6 
          ColocBoost,single_context,multi_context,multi_gene 
                                                           1 
                        ColocBoost,single_context,multi_gene 
                                                           2 
                                                   MetaBrain 
                                                           4 
                                        MetaBrain,ColocBoost 
                                                           1 
                                                  multi_gene 
                                                           4 
                                              single_context 
                                                          13 
                                   single_context,ColocBoost 
                                                           3 
                                    single_context,MetaBrain 
                                                           1 
                                single_context,multi_context 
                                                           1 
                     single_context,multi_context,ColocBoost 
                                                           2 
                      single_context,multi_context,MetaBrain 
                                                           3 
           single_context,multi_context,MetaBrain,multi_gene 
                                                           4 
single_context,multi_context,MetaBrain,multi_gene,ColocBoost 
                                                           1 
                     single_context,multi_context,multi_gene 
                                                           1 
                                   single_context,multi_gene 
                                                           2 
</pre></div>
</div>
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>.
                                                  ColocBoost 
                                                          40 
                                   ColocBoost,single_context 
                                                           2 
                     ColocBoost,single_context,multi_context 
                                                           2 
           ColocBoost,single_context,multi_context,MetaBrain 
                                                           1 
                        ColocBoost,single_context,multi_gene 
                                                           1 
                                                   MetaBrain 
                                                           2 
                                        MetaBrain,ColocBoost 
                                                           1 
                                              single_context 
                                                           1 
                                    single_context,MetaBrain 
                                                           1 
                      single_context,multi_context,MetaBrain 
                                                           2 
single_context,multi_context,MetaBrain,multi_gene,ColocBoost 
                                                           1 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">xQTL_ego</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span>
<span class="w">  </span><span class="n">gene</span><span class="w">          </span><span class="o">=</span><span class="w"> </span><span class="n">xQTL_list</span><span class="o">$</span><span class="n">gene_name</span><span class="p">,</span>
<span class="w">  </span><span class="n">OrgDb</span><span class="w">         </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">  </span><span class="n">keyType</span><span class="w">       </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">ont</span><span class="w">           </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<span class="w">  </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">1</span>
<span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">xQTL_ego</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="m">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 5 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0002274</th><td>GO:0002274</td><td>myeloid leukocyte activation                     </td><td>14/215</td><td>241/18888</td><td>7.206455e-07</td><td>0.001682472</td><td>0.001379365</td><td>GRN/TREM2/STX4/CD33/HAVCR1/CLU/NECTIN2/HAVCR2/DYSF/APP/FCER1G/RHOH/C5AR1/PLCG2                                     </td><td>14</td></tr>
	<tr><th scope=row>GO:0030100</th><td>GO:0030100</td><td>regulation of endocytosis                        </td><td>15/215</td><td>291/18888</td><td>1.300061e-06</td><td>0.001682472</td><td>0.001379365</td><td>ATG5/PICALM/TREM2/RIN3/SIRPB1/LILRB1/CLU/APOC1/DYSF/BIN1/DGKQ/FCER1G/PLCG2/SIRPA/APOC2                             </td><td>15</td></tr>
	<tr><th scope=row>GO:0002281</th><td>GO:0002281</td><td>macrophage activation involved in immune response</td><td>5/215 </td><td>19/18888 </td><td>1.862633e-06</td><td>0.001682472</td><td>0.001379365</td><td>GRN/TREM2/HAVCR2/DYSF/PLCG2                                                                                        </td><td> 5</td></tr>
	<tr><th scope=row>GO:0045806</th><td>GO:0045806</td><td>negative regulation of endocytosis               </td><td>8/215 </td><td>75/18888 </td><td>2.169532e-06</td><td>0.001682472</td><td>0.001379365</td><td>ATG5/PICALM/RIN3/LILRB1/APOC1/DYSF/SIRPA/APOC2                                                                     </td><td> 8</td></tr>
	<tr><th scope=row>GO:0045785</th><td>GO:0045785</td><td>positive regulation of cell adhesion             </td><td>19/215</td><td>485/18888</td><td>3.082937e-06</td><td>0.001912654</td><td>0.001568079</td><td>CBFB/NCK2/ACTL6B/CASS4/STK4/SIRPB1/CORO1A/STX4/LILRB1/ITGB1BP1/PTK2B/HAVCR2/EPHA1/STX3/SERPINF2/RHOH/CD55/SIRPA/CR1</td><td>19</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># GO terms related to amyloid</span>
<span class="n">xQTL_ego</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;amyloid&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0050435</th><td>GO:0050435</td><td>amyloid-beta metabolic process                                    </td><td>6/215</td><td>65/18888</td><td>9.575398e-05</td><td>0.008486538</td><td>0.006957643</td><td>PICALM/CLU/BIN1/SORL1/APH1B/ACE   </td><td>6</td></tr>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process                       </td><td>6/215</td><td>68/18888</td><td>1.233433e-04</td><td>0.008897927</td><td>0.007294917</td><td>PICALM/CLU/BIN1/SORL1/APH1B/ADAM17</td><td>6</td></tr>
	<tr><th scope=row>GO:0034205</th><td>GO:0034205</td><td>amyloid-beta formation                                            </td><td>5/215</td><td>56/18888</td><td>4.346757e-04</td><td>0.015562055</td><td>0.012758467</td><td>PICALM/CLU/BIN1/SORL1/APH1B       </td><td>5</td></tr>
	<tr><th scope=row>GO:1902992</th><td>GO:1902992</td><td>negative regulation of amyloid precursor protein catabolic process</td><td>4/215</td><td>32/18888</td><td>4.571898e-04</td><td>0.015584647</td><td>0.012776989</td><td>PICALM/CLU/BIN1/SORL1             </td><td>4</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process                       </td><td>6/215</td><td>96/18888</td><td>8.027018e-04</td><td>0.021294972</td><td>0.017458567</td><td>PICALM/CLU/BIN1/SORL1/APH1B/ADAM17</td><td>6</td></tr>
	<tr><th scope=row>GO:1905906</th><td>GO:1905906</td><td>regulation of amyloid fibril formation                            </td><td>3/215</td><td>17/18888</td><td>8.790662e-04</td><td>0.021814907</td><td>0.017884833</td><td>TREM2/CLU/APP                     </td><td>3</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># GO terms related to lysosome</span>
<span class="n">xQTL_ego</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">as.data.frame</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="nf">regex</span><span class="p">(</span><span class="s">&#39;lysosome|lysosomal membrane|lysosomal lumen|autophagy|macroautophagy|vacuole|endosome|phagosome|lipid droplet&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">ignore_case</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">head</span><span class="p">()</span><span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0007040</th><td>GO:0007040</td><td>lysosome organization              </td><td>6/215</td><td>111/18888</td><td>0.001703434</td><td>0.03108266</td><td>0.02548295</td><td>GRN/CORO1A/TMEM106B/TMEM175/ACP2/ABCA1              </td><td>6</td></tr>
	<tr><th scope=row>GO:0080171</th><td>GO:0080171</td><td>lytic vacuole organization         </td><td>6/215</td><td>111/18888</td><td>0.001703434</td><td>0.03108266</td><td>0.02548295</td><td>GRN/CORO1A/TMEM106B/TMEM175/ACP2/ABCA1              </td><td>6</td></tr>
	<tr><th scope=row>GO:0007033</th><td>GO:0007033</td><td>vacuole organization               </td><td>8/215</td><td>234/18888</td><td>0.005470877</td><td>0.06262236</td><td>0.05134061</td><td>GRN/ATG5/CORO1A/TMEM106B/TMEM175/ACP2/TP53INP1/ABCA1</td><td>8</td></tr>
	<tr><th scope=row>GO:0035751</th><td>GO:0035751</td><td>regulation of lysosomal lumen pH   </td><td>3/215</td><td>32/18888 </td><td>0.005656367</td><td>0.06403668</td><td>0.05250012</td><td>GRN/TMEM106B/TMEM175                                </td><td>3</td></tr>
	<tr><th scope=row>GO:1905671</th><td>GO:1905671</td><td>regulation of lysosome organization</td><td>2/215</td><td>12/18888 </td><td>0.007895963</td><td>0.07542693</td><td>0.06183836</td><td>GRN/TMEM106B                                        </td><td>2</td></tr>
	<tr><th scope=row>GO:0061684</th><td>GO:0061684</td><td>chaperone-mediated autophagy       </td><td>2/215</td><td>15/18888 </td><td>0.012283347</td><td>0.09486872</td><td>0.07777761</td><td>ATG5/CLU                                            </td><td>2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># GO terms related to neuron function</span>
<span class="n">xQTL_ego</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">as.data.frame</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="nf">regex</span><span class="p">(</span><span class="s">&#39;synaptic transmission|synapse organization|cognition|synaptic plasticity|neuron projection morphogenesis|axon guidance|nervous system development|modulation of chemical synaptic transmission|generation of neurons|learning or memory|neuron projection|postsynaptic membrane|dendrite|axon|cell body|postsynapse|presynapse|synaptic membrane|synapse&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">ignore_case</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span><span class="w">  </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0098814</th><td>GO:0098814</td><td>spontaneous synaptic transmission                           </td><td>3/215</td><td>14/18888 </td><td>0.0004825692</td><td>0.01567266</td><td>0.01284914</td><td>APP/DOC2A/DOC2B                                   </td><td>3</td></tr>
	<tr><th scope=row>GO:0001771</th><td>GO:0001771</td><td>immunological synapse formation                             </td><td>3/215</td><td>16/18888 </td><td>0.0007300409</td><td>0.02021953</td><td>0.01657687</td><td>NCK2/HAVCR2/CD2AP                                 </td><td>3</td></tr>
	<tr><th scope=row>GO:0099003</th><td>GO:0099003</td><td>vesicle-mediated transport in synapse                       </td><td>9/215</td><td>221/18888</td><td>0.0009980271</td><td>0.02363267</td><td>0.01937512</td><td>SLC4A8/PICALM/BIN1/DGKQ/DOC2A/STX3/CPLX1/GAK/DOC2B</td><td>9</td></tr>
	<tr><th scope=row>GO:0099643</th><td>GO:0099643</td><td>signal release from synapse                                 </td><td>6/215</td><td>144/18888</td><td>0.0061627519</td><td>0.06803152</td><td>0.05577527</td><td>SLC4A8/DYSF/DOC2A/STX3/CPLX1/DOC2B                </td><td>6</td></tr>
	<tr><th scope=row>GO:0002753</th><td>GO:0002753</td><td>cytoplasmic pattern recognition receptor signaling pathway  </td><td>6/215</td><td>163/18888</td><td>0.0110057492</td><td>0.08984167</td><td>0.07365621</td><td>DHX33/TREM2/HAVCR2/OTULIN/SCIMP/PLCG2             </td><td>6</td></tr>
	<tr><th scope=row>GO:0098969</th><td>GO:0098969</td><td>neurotransmitter receptor transport to postsynaptic membrane</td><td>2/215</td><td>16/18888 </td><td>0.0139338188</td><td>0.10291120</td><td>0.08437119</td><td>STX3/CPLX1                                        </td><td>2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># GO terms related to lipid</span>
<span class="n">xQTL_ego</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;lipid|Lipid|fatty|Fatty&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0033700</th><td>GO:0033700</td><td>phospholipid efflux                      </td><td>3/215</td><td>14/18888 </td><td>0.0004825692</td><td>0.01567266</td><td>0.01284914</td><td>APOC1/ABCA1/APOC2                         </td><td>3</td></tr>
	<tr><th scope=row>GO:0071825</th><td>GO:0071825</td><td>protein-lipid complex organization       </td><td>4/215</td><td>53/18888 </td><td>0.0030908703</td><td>0.04561033</td><td>0.03739339</td><td>APOC1/BIN1/ABCA1/APOC2                    </td><td>4</td></tr>
	<tr><th scope=row>GO:0055088</th><td>GO:0055088</td><td>lipid homeostasis                        </td><td>7/215</td><td>176/18888</td><td>0.0040895779</td><td>0.05398243</td><td>0.04425720</td><td>POLD1/TREM2/MTCH2/THADA/TLCD3B/ABCA1/APOC2</td><td>7</td></tr>
	<tr><th scope=row>GO:0065005</th><td>GO:0065005</td><td>protein-lipid complex assembly           </td><td>3/215</td><td>30/18888 </td><td>0.0047077831</td><td>0.05841417</td><td>0.04789054</td><td>APOC1/BIN1/ABCA1                          </td><td>3</td></tr>
	<tr><th scope=row>GO:0046834</th><td>GO:0046834</td><td>lipid phosphorylation                    </td><td>2/215</td><td>11/18888 </td><td>0.0066294041</td><td>0.06947436</td><td>0.05695818</td><td>AGK/DGKQ                                  </td><td>2</td></tr>
	<tr><th scope=row>GO:1905953</th><td>GO:1905953</td><td>negative regulation of lipid localization</td><td>4/215</td><td>67/18888 </td><td>0.0071525138</td><td>0.07322474</td><td>0.06003291</td><td>TREM2/APOC1/ABCA1/APOC2                   </td><td>4</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p4</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">xQTL_ego</span><span class="p">,</span>
<span class="w">        </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span>
<span class="w">        </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GO Enrichment - BP&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">        </span><span class="nf">theme</span><span class="p">(</span><span class="n">plot.margin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">margin</span><span class="p">(</span><span class="n">t</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">r</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">b</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">l</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">unit</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;mm&quot;</span><span class="p">))</span>
<span class="n">p4</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/15360adb778127083ff7c29259ecf69616bef13e2a3555d806fc40ed3e793177.png"><img alt="../../_images/15360adb778127083ff7c29259ecf69616bef13e2a3555d806fc40ed3e793177.png" src="../../_images/15360adb778127083ff7c29259ecf69616bef13e2a3555d806fc40ed3e793177.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="xqtl-list-kegg-enrichment">
<h3>xQTL list - KEGG enrichment<a class="headerlink" href="#xqtl-list-kegg-enrichment" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">entrez_ids</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">bitr</span><span class="p">(</span><span class="n">xQTL_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">fromType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                   </span><span class="n">toType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENTREZID&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">)</span>

<span class="n">kegg_enrich</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichKEGG</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">entrez_ids</span><span class="o">$</span><span class="n">ENTREZID</span><span class="p">,</span>
<span class="w">                          </span><span class="n">organism</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;hsa&#39;</span><span class="p">,</span>
<span class="w">                          </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<span class="w">                          </span><span class="n">qvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:many mapping between keys and columns

Warning message in bitr(xQTL_list$gene_id, fromType = &quot;ENSEMBL&quot;, toType = &quot;ENTREZID&quot;, :
“2.08% of input gene IDs are fail to map...”
Reading KEGG annotation online: &quot;https://rest.kegg.jp/link/hsa/pathway&quot;...

Reading KEGG annotation online: &quot;https://rest.kegg.jp/list/pathway/hsa&quot;...
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">kegg_enrich</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">head</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="m">4</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 4 × 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa04610</th><td>Organismal Systems</td><td>Immune system               </td><td>hsa04610</td><td>Complement and coagulation cascades</td><td>7/108</td><td>88/9447 </td><td>6.227564e-05</td><td>0.008569564</td><td>0.008045394</td><td>1380/1191/5345/716/1604/728/1378               </td><td>7</td></tr>
	<tr><th scope=row>hsa04662</th><td>Organismal Systems</td><td>Immune system               </td><td>hsa04662</td><td>B cell receptor signaling pathway  </td><td>7/108</td><td>91/9447 </td><td>7.720328e-05</td><td>0.008569564</td><td>0.008045394</td><td>29760/5595/10859/1380/3635/353514/5336         </td><td>7</td></tr>
	<tr><th scope=row>hsa04380</th><td>Organismal Systems</td><td>Development and regeneration</td><td>hsa04380</td><td>Osteoclast differentiation         </td><td>8/108</td><td>143/9447</td><td>2.229312e-04</td><td>0.016496910</td><td>0.015487853</td><td>29760/54209/10326/5595/10859/353514/5336/140885</td><td>8</td></tr>
	<tr><th scope=row>hsa04148</th><td>Cellular Processes</td><td>Transport and catabolism    </td><td>hsa04148</td><td>Efferocytosis                      </td><td>8/108</td><td>157/9447</td><td>4.200806e-04</td><td>0.023314471</td><td>0.021888408</td><td>10326/5595/26762/84868/91937/6868/19/140885    </td><td>8</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># KEGG pathways related to lipid</span>
<span class="n">kegg_enrich</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;lipid|Lipid|fatty|Fatty|Alzheimer|Cholesterol&#39;</span><span class="p">))</span>

<span class="c1"># KEGG pathways related to AD</span>
<span class="n">kegg_enrich</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Alzheimer&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 9 × 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa04979</th><td>Organismal Systems                  </td><td>Digestive system                  </td><td>hsa04979</td><td>Cholesterol metabolism                                    </td><td>3/108</td><td>51/9447 </td><td>0.02034414</td><td>0.2656705</td><td>0.2494204</td><td>341/19/344                         </td><td>3</td></tr>
	<tr><th scope=row>hsa00604</th><td>Metabolism                          </td><td>Glycan biosynthesis and metabolism</td><td>hsa00604</td><td>Glycosphingolipid biosynthesis - ganglio series           </td><td>1/108</td><td>15/9447 </td><td>0.15852538</td><td>0.7172266</td><td>0.6733564</td><td>6483                               </td><td>1</td></tr>
	<tr><th scope=row>hsa05010</th><td>Human Diseases                      </td><td>Neurodegenerative disease         </td><td>hsa05010</td><td>Alzheimer disease                                         </td><td>7/108</td><td>391/9447</td><td>0.15982022</td><td>0.7172266</td><td>0.6733564</td><td>5595/1350/83464/351/6868/91252/4722</td><td>7</td></tr>
	<tr><th scope=row>hsa00603</th><td>Metabolism                          </td><td>Glycan biosynthesis and metabolism</td><td>hsa00603</td><td>Glycosphingolipid biosynthesis - globo and isoglobo series</td><td>1/108</td><td>16/9447 </td><td>0.16816059</td><td>0.7172266</td><td>0.6733564</td><td>6483                               </td><td>1</td></tr>
	<tr><th scope=row>hsa00561</th><td>Metabolism                          </td><td>Lipid metabolism                  </td><td>hsa00561</td><td>Glycerolipid metabolism                                   </td><td>2/108</td><td>65/9447 </td><td>0.17005472</td><td>0.7172266</td><td>0.6733564</td><td>55750/1609                         </td><td>2</td></tr>
	<tr><th scope=row>hsa04071</th><td>Environmental Information Processing</td><td>Signal transduction               </td><td>hsa04071</td><td>Sphingolipid signaling pathway                            </td><td>2/108</td><td>122/9447</td><td>0.40799561</td><td>0.7549129</td><td>0.7087376</td><td>5595/2207                          </td><td>2</td></tr>
	<tr><th scope=row>hsa04932</th><td>Human Diseases                      </td><td>Endocrine and metabolic disease   </td><td>hsa04932</td><td>Non-alcoholic fatty liver disease                         </td><td>2/108</td><td>157/9447</td><td>0.53896386</td><td>0.7786545</td><td>0.7310270</td><td>1350/4722                          </td><td>2</td></tr>
	<tr><th scope=row>hsa00564</th><td>Metabolism                          </td><td>Lipid metabolism                  </td><td>hsa00564</td><td>Glycerophospholipid metabolism                            </td><td>1/108</td><td>103/9447</td><td>0.69601495</td><td>0.8122626</td><td>0.7625794</td><td>1609                               </td><td>1</td></tr>
	<tr><th scope=row>hsa05417</th><td>Human Diseases                      </td><td>Cardiovascular disease            </td><td>hsa05417</td><td>Lipid and atherosclerosis                                 </td><td>2/108</td><td>216/9447</td><td>0.71165338</td><td>0.8122626</td><td>0.7625794</td><td>5595/19                            </td><td>2</td></tr>
</tbody>
</table>
</div><div class="output text_html"><table class="dataframe">
<caption>A data.frame: 1 × 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa05010</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05010</td><td>Alzheimer disease</td><td>7/108</td><td>391/9447</td><td>0.1598202</td><td>0.7172266</td><td>0.6733564</td><td>5595/1350/83464/351/6868/91252/4722</td><td>7</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">entrez_ids</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">ENTREZID</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;1609&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="n">gene_ref</span><span class="p">,</span><span class="w"> </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;ENSEMBL&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;gene_id&#39;</span><span class="p">)</span>
<span class="n">entrez_ids</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">ENTREZID</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;6483&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="n">gene_ref</span><span class="p">,</span><span class="w"> </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;ENSEMBL&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;gene_id&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 1 × 6</caption>
<thead>
	<tr><th scope=col>ENSEMBL</th><th scope=col>ENTREZID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>ENSG00000145214</td><td>1609</td><td>chr4</td><td>986894</td><td>986895</td><td>DGKQ</td></tr>
</tbody>
</table>
</div><div class="output text_html"><table class="dataframe">
<caption>A data.frame: 1 × 6</caption>
<thead>
	<tr><th scope=col>ENSEMBL</th><th scope=col>ENTREZID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>ENSG00000157350</td><td>6483</td><td>chr16</td><td>70439236</td><td>70439237</td><td>ST3GAL2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">entrez_ids</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">ENTREZID</span><span class="o">%in%</span><span class="w"> </span><span class="p">(</span><span class="n">kegg_enrich</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Alzheimer&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">geneID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;/&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unlist</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="n">gene_ref</span><span class="p">,</span><span class="w"> </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;ENSEMBL&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;gene_id&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 7 × 6</caption>
<thead>
	<tr><th scope=col>ENSEMBL</th><th scope=col>ENTREZID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>ENSG00000102882</td><td>5595 </td><td>chr16</td><td>30123505</td><td>30123506</td><td>MAPK3   </td></tr>
	<tr><td>ENSG00000127184</td><td>1350 </td><td>chr5 </td><td>86617927</td><td>86617928</td><td>COX7C   </td></tr>
	<tr><td>ENSG00000138613</td><td>83464</td><td>chr15</td><td>63276017</td><td>63276018</td><td>APH1B   </td></tr>
	<tr><td>ENSG00000142192</td><td>351  </td><td>chr21</td><td>26171127</td><td>26171128</td><td>APP     </td></tr>
	<tr><td>ENSG00000151694</td><td>6868 </td><td>chr2 </td><td> 9556731</td><td> 9556732</td><td>ADAM17  </td></tr>
	<tr><td>ENSG00000165915</td><td>91252</td><td>chr11</td><td>47407131</td><td>47407132</td><td>SLC39A13</td></tr>
	<tr><td>ENSG00000213619</td><td>4722 </td><td>chr11</td><td>47565335</td><td>47565336</td><td>NDUFS3  </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p5</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">kegg_enrich</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;KEGG Pathway Enrichment&quot;</span><span class="p">)</span>
<span class="n">p5</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/194383f3f6811c753f950989ef7da4dde32c97975eb26bd1d1ca598eea0e0f28.png"><img alt="../../_images/194383f3f6811c753f950989ef7da4dde32c97975eb26bd1d1ca598eea0e0f28.png" src="../../_images/194383f3f6811c753f950989ef7da4dde32c97975eb26bd1d1ca598eea0e0f28.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># KEGG pathways related to lysosome functions</span>
<span class="n">kegg_enrich</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">as.data.frame</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="nf">regex</span><span class="p">(</span><span class="s">&#39;lysosome|autophagy|endocytosis|phagosome|vacuole|lipid&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">ignore_case</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa04142</th><td>Cellular Processes</td><td>Transport and catabolism          </td><td>hsa04142</td><td>Lysosome                                                  </td><td>5/108</td><td>133/9447</td><td>0.01803538</td><td>0.2655746</td><td>0.2493303</td><td>23062/1512/3425/53/1508</td><td>5</td></tr>
	<tr><th scope=row>hsa00604</th><td>Metabolism        </td><td>Glycan biosynthesis and metabolism</td><td>hsa00604</td><td>Glycosphingolipid biosynthesis - ganglio series           </td><td>1/108</td><td>15/9447 </td><td>0.15852538</td><td>0.7172266</td><td>0.6733564</td><td>6483                   </td><td>1</td></tr>
	<tr><th scope=row>hsa00603</th><td>Metabolism        </td><td>Glycan biosynthesis and metabolism</td><td>hsa00603</td><td>Glycosphingolipid biosynthesis - globo and isoglobo series</td><td>1/108</td><td>16/9447 </td><td>0.16816059</td><td>0.7172266</td><td>0.6733564</td><td>6483                   </td><td>1</td></tr>
	<tr><th scope=row>hsa00561</th><td>Metabolism        </td><td>Lipid metabolism                  </td><td>hsa00561</td><td>Glycerolipid metabolism                                   </td><td>2/108</td><td>65/9447 </td><td>0.17005472</td><td>0.7172266</td><td>0.6733564</td><td>55750/1609             </td><td>2</td></tr>
	<tr><th scope=row>hsa04140</th><td>Cellular Processes</td><td>Transport and catabolism          </td><td>hsa04140</td><td>Autophagy - animal                                        </td><td>3/108</td><td>169/9447</td><td>0.30462437</td><td>0.7549129</td><td>0.7087376</td><td>9474/5595/1508         </td><td>3</td></tr>
	<tr><th scope=row>hsa04136</th><td>Cellular Processes</td><td>Transport and catabolism          </td><td>hsa04136</td><td>Autophagy - other                                         </td><td>1/108</td><td>32/9447 </td><td>0.30826072</td><td>0.7549129</td><td>0.7087376</td><td>9474                   </td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># KEGG pathways related to neuron function</span>
<span class="n">kegg_enrich</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">as.data.frame</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="nf">regex</span><span class="p">(</span><span class="s">&#39;synapse|neuro|axon|dendrite|memory|learning|brain|glutamatergic|GABAergic|cholinergic|dopaminergic|serotonergic&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">ignore_case</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa04360</th><td>Organismal Systems</td><td>Development and regeneration</td><td>hsa04360</td><td>Axon guidance                                    </td><td>7/108</td><td>184/9447</td><td>0.005065273</td><td>0.1606415</td><td>0.1508156</td><td>8440/5595/5921/2041/3985/23365/5336</td><td>7</td></tr>
	<tr><th scope=row>hsa04722</th><td>Organismal Systems</td><td>Nervous system              </td><td>hsa04722</td><td>Neurotrophin signaling pathway                   </td><td>3/108</td><td>120/9447</td><td>0.157892330</td><td>0.7172266</td><td>0.6733564</td><td>4214/5595/5336                     </td><td>3</td></tr>
	<tr><th scope=row>hsa04726</th><td>Organismal Systems</td><td>Nervous system              </td><td>hsa04726</td><td>Serotonergic synapse                             </td><td>2/108</td><td>115/9447</td><td>0.379673022</td><td>0.7549129</td><td>0.7087376</td><td>5595/351                           </td><td>2</td></tr>
	<tr><th scope=row>hsa04727</th><td>Organismal Systems</td><td>Nervous system              </td><td>hsa04727</td><td>GABAergic synapse                                </td><td>1/108</td><td>89/9447 </td><td>0.642332162</td><td>0.8122626</td><td>0.7625794</td><td>4905                               </td><td>1</td></tr>
	<tr><th scope=row>hsa05022</th><td>Human Diseases    </td><td>Neurodegenerative disease   </td><td>hsa05022</td><td>Pathways of neurodegeneration - multiple diseases</td><td>5/108</td><td>483/9447</td><td>0.653400224</td><td>0.8122626</td><td>0.7625794</td><td>5595/83544/1350/351/4722           </td><td>5</td></tr>
	<tr><th scope=row>hsa04724</th><td>Organismal Systems</td><td>Nervous system              </td><td>hsa04724</td><td>Glutamatergic synapse                            </td><td>1/108</td><td>116/9447</td><td>0.738679177</td><td>0.8122626</td><td>0.7625794</td><td>5595                               </td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="plot-ad-functional-pathway-xqtl-list">
<h3>Plot AD functional pathway - xQTL list<a class="headerlink" href="#plot-ad-functional-pathway-xqtl-list" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">gene.data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">kegg_enrich</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">as.data.frame</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Alzheimer&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">pull</span><span class="p">(</span><span class="n">geneID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">str_split</span><span class="p">(</span><span class="s">&#39;/&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">unlist</span><span class="p">()</span>

<span class="n">gene.data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">setNames</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">gene.data</span><span class="p">))),</span><span class="w"> </span><span class="n">gene.data</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Generate the pathway plot with highlighted genes</span>
<span class="nf">pathview</span><span class="p">(</span>
<span class="w">  </span><span class="n">gene.data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gene.data</span><span class="p">,</span>
<span class="w">  </span><span class="n">pathway.id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;hsa05010&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;hsa&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">out.suffix</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NULL</span><span class="p">,</span>
<span class="w">  </span><span class="n">kegg.native</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span>
<span class="p">)</span>

<span class="n">p6</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readPNG</span><span class="p">(</span><span class="s">&quot;hsa05010.xQTL_list.png&quot;</span><span class="p">)</span>
<span class="nf">grid.raster</span><span class="p">(</span><span class="n">p6</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:1 mapping between keys and columns

Info: Working in directory /restricted/projectnb/xqtl/jaempawi/xqtl

Info: Writing image file hsa05010..png
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/ab1f57991050abf1953a8ea9063268b9e98a23bbcc3e690e347ec2a3fd046d9c.png"><img alt="../../_images/ab1f57991050abf1953a8ea9063268b9e98a23bbcc3e690e347ec2a3fd046d9c.png" src="../../_images/ab1f57991050abf1953a8ea9063268b9e98a23bbcc3e690e347ec2a3fd046d9c.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="xqtl-only-list-go-enrichment">
<h3>xQTL only list - GO enrichment<a class="headerlink" href="#xqtl-only-list-go-enrichment" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">xQTL_only</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">Reduce</span><span class="p">(</span><span class="n">setdiff</span><span class="p">,</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">xQTL_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">gvc_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">twas_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">))</span><span class="w"> </span><span class="c1"># the genes unqiue from xQTL, NOT from gvc and twas</span>
<span class="n">xQTL_only</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">xQTL_list</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">xQTL_only</span><span class="p">)</span>
<span class="n">xQTL_only</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">resource_in_FunGen_xQTL</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">table</span><span class="w"> </span><span class="c1"># check their resources from FunGen xQTL</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>.
                                                  ColocBoost 
                                                         116 
                         ColocBoost,multi_context,multi_gene 
                                                           1 
                                   ColocBoost,single_context 
                                                           1 
                     ColocBoost,single_context,multi_context 
                                                           4 
           ColocBoost,single_context,multi_context,MetaBrain 
                                                           2 
ColocBoost,single_context,multi_context,MetaBrain,multi_gene 
                                                           2 
                        ColocBoost,single_context,multi_gene 
                                                           2 
                                                   MetaBrain 
                                                           1 
                                                  multi_gene 
                                                           2 
                                              single_context 
                                                          10 
                                    single_context,MetaBrain 
                                                           1 
                                single_context,multi_context 
                                                           1 
                      single_context,multi_context,MetaBrain 
                                                           3 
           single_context,multi_context,MetaBrain,multi_gene 
                                                           2 
                     single_context,multi_context,multi_gene 
                                                           1 
                                   single_context,multi_gene 
                                                           2 
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">xqtl_only_ego</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span>
<span class="w">  </span><span class="n">gene</span><span class="w">          </span><span class="o">=</span><span class="w"> </span><span class="n">xQTL_only</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span>
<span class="w">  </span><span class="n">OrgDb</span><span class="w">         </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">  </span><span class="n">keyType</span><span class="w">       </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">ont</span><span class="w">           </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<span class="w">  </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">1</span>
<span class="p">)</span>
<span class="n">xqtl_only_ego</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:1902017</th><td>GO:1902017</td><td>regulation of cilium assembly                          </td><td>5/134</td><td>78/21288 </td><td>0.0001339299</td><td>0.2795117</td><td>0.2692696</td><td>ENSG00000057663/ENSG00000103540/ENSG00000136536/ENSG00000138036/ENSG00000182541                                                                </td><td>5</td></tr>
	<tr><th scope=row>GO:1902018</th><td>GO:1902018</td><td>negative regulation of cilium assembly                 </td><td>3/134</td><td>21/21288 </td><td>0.0002985075</td><td>0.3114926</td><td>0.3000786</td><td>ENSG00000103540/ENSG00000136536/ENSG00000182541                                                                                                </td><td>3</td></tr>
	<tr><th scope=row>GO:0031338</th><td>GO:0031338</td><td>regulation of vesicle fusion                           </td><td>3/134</td><td>25/21288 </td><td>0.0005068010</td><td>0.3220401</td><td>0.3102397</td><td>ENSG00000102879/ENSG00000168993/ENSG00000272636                                                                                                </td><td>3</td></tr>
	<tr><th scope=row>GO:0048284</th><td>GO:0048284</td><td>organelle fusion                                       </td><td>6/134</td><td>172/21288</td><td>0.0007799122</td><td>0.3220401</td><td>0.3102397</td><td>ENSG00000102879/ENSG00000103496/ENSG00000127419/ENSG00000140379/ENSG00000168993/ENSG00000272636                                                </td><td>6</td></tr>
	<tr><th scope=row>GO:0060271</th><td>GO:0060271</td><td>cilium assembly                                        </td><td>9/134</td><td>401/21288</td><td>0.0010053721</td><td>0.3220401</td><td>0.3102397</td><td>ENSG00000057663/ENSG00000103540/ENSG00000112981/ENSG00000119661/ENSG00000121289/ENSG00000136536/ENSG00000138036/ENSG00000157423/ENSG00000182541</td><td>9</td></tr>
	<tr><th scope=row>GO:0061098</th><td>GO:0061098</td><td>positive regulation of protein tyrosine kinase activity</td><td>3/134</td><td>33/21288 </td><td>0.0011588387</td><td>0.3220401</td><td>0.3102397</td><td>ENSG00000151694/ENSG00000167191/ENSG00000187116                                                                                                </td><td>3</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># GO term related to amyloid</span>
<span class="n">xqtl_only_ego</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;amyloid&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 3 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0097242</th><td>GO:0097242</td><td>amyloid-beta clearance                     </td><td>1/134</td><td>46/21288 </td><td>0.2523171</td><td>0.5002484</td><td>0.4819179</td><td>ENSG00000197405</td><td>1</td></tr>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process</td><td>1/134</td><td>85/21288 </td><td>0.4159679</td><td>0.5978823</td><td>0.5759742</td><td>ENSG00000151694</td><td>1</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process</td><td>1/134</td><td>114/21288</td><td>0.5141128</td><td>0.6562406</td><td>0.6321941</td><td>ENSG00000151694</td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">amyloid_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="p">(</span><span class="n">xqtl_only_ego</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;amyloid&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">geneID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;/&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unlist</span><span class="w"> </span><span class="p">)</span>
<span class="n">gene_ref</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">amyloid_genes</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 2 × 5</caption>
<thead>
	<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_id</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr2 </td><td> 9556731</td><td> 9556732</td><td>ENSG00000151694</td><td>ADAM17</td></tr>
	<tr><td>chr19</td><td>47290022</td><td>47290023</td><td>ENSG00000197405</td><td>C5AR1 </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p7</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">xqtl_only_ego</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GO Enrichment - BP&quot;</span><span class="p">)</span>
<span class="n">p7</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/05a4de6653b8374364b21ad4a8e3d3daeb9edf5af5c6666532c84626959b1e52.png"><img alt="../../_images/05a4de6653b8374364b21ad4a8e3d3daeb9edf5af5c6666532c84626959b1e52.png" src="../../_images/05a4de6653b8374364b21ad4a8e3d3daeb9edf5af5c6666532c84626959b1e52.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="xqtl-only-list-kegg-enrichment">
<h3>xQTL only list - KEGG enrichment<a class="headerlink" href="#xqtl-only-list-kegg-enrichment" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">entrez_ids</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">bitr</span><span class="p">(</span><span class="n">xQTL_only</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">fromType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                   </span><span class="n">toType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENTREZID&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">)</span>

<span class="n">kegg_enrich_only</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichKEGG</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">entrez_ids</span><span class="o">$</span><span class="n">ENTREZID</span><span class="p">,</span>
<span class="w">                          </span><span class="n">organism</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;hsa&#39;</span><span class="p">,</span>
<span class="w">                          </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<span class="w">                         </span><span class="n">qvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:many mapping between keys and columns

Warning message in bitr(xQTL_only$gene_id, fromType = &quot;ENSEMBL&quot;, toType = &quot;ENTREZID&quot;, :
“1.32% of input gene IDs are fail to map...”
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># KEGG pathway related to lipid</span>
<span class="n">kegg_enrich_only</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;lipid|Lipid|fatty|Fatty|Alzheimer|Cholesterol&#39;</span><span class="p">))</span>

<span class="c1"># KEGG pathway related to AD</span>
<span class="n">kegg_enrich_only</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Alzheimer&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa00561</th><td>Metabolism    </td><td>Lipid metabolism                  </td><td>hsa00561</td><td>Glycerolipid metabolism                                   </td><td>2/61</td><td>65/9447 </td><td>0.06583244</td><td>0.7570456</td><td>0.7570456</td><td>55750/1609</td><td>2</td></tr>
	<tr><th scope=row>hsa00604</th><td>Metabolism    </td><td>Glycan biosynthesis and metabolism</td><td>hsa00604</td><td>Glycosphingolipid biosynthesis - ganglio series           </td><td>1/61</td><td>15/9447 </td><td>0.09266406</td><td>0.7570456</td><td>0.7570456</td><td>6483      </td><td>1</td></tr>
	<tr><th scope=row>hsa00603</th><td>Metabolism    </td><td>Glycan biosynthesis and metabolism</td><td>hsa00603</td><td>Glycosphingolipid biosynthesis - globo and isoglobo series</td><td>1/61</td><td>16/9447 </td><td>0.09853211</td><td>0.7570456</td><td>0.7570456</td><td>6483      </td><td>1</td></tr>
	<tr><th scope=row>hsa00564</th><td>Metabolism    </td><td>Lipid metabolism                  </td><td>hsa00564</td><td>Glycerophospholipid metabolism                            </td><td>1/61</td><td>103/9447</td><td>0.48873928</td><td>0.7580112</td><td>0.7580112</td><td>1609      </td><td>1</td></tr>
	<tr><th scope=row>hsa04932</th><td>Human Diseases</td><td>Endocrine and metabolic disease   </td><td>hsa04932</td><td>Non-alcoholic fatty liver disease                         </td><td>1/61</td><td>157/9447</td><td>0.64140700</td><td>0.7889887</td><td>0.7889887</td><td>1350      </td><td>1</td></tr>
	<tr><th scope=row>hsa05010</th><td>Human Diseases</td><td>Neurodegenerative disease         </td><td>hsa05010</td><td>Alzheimer disease                                         </td><td>2/61</td><td>391/9447</td><td>0.72521663</td><td>0.8303681</td><td>0.8303681</td><td>1350/6868 </td><td>2</td></tr>
</tbody>
</table>
</div><div class="output text_html"><table class="dataframe">
<caption>A data.frame: 1 × 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa05010</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05010</td><td>Alzheimer disease</td><td>2/61</td><td>391/9447</td><td>0.7252166</td><td>0.8303681</td><td>0.8303681</td><td>1350/6868</td><td>2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># KEGG pathway related to neurogenerative</span>
<span class="n">kegg_enrich_only</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">subcategory</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Neurodegenerative&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa05016</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05016</td><td>Huntington disease                               </td><td>3/61</td><td>311/9447</td><td>0.3255188</td><td>0.7570456</td><td>0.7570456</td><td>7019/83544/1350</td><td>3</td></tr>
	<tr><th scope=row>hsa05014</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05014</td><td>Amyotrophic lateral sclerosis                    </td><td>2/61</td><td>371/9447</td><td>0.6976572</td><td>0.8203932</td><td>0.8203932</td><td>83544/1350     </td><td>2</td></tr>
	<tr><th scope=row>hsa05010</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05010</td><td>Alzheimer disease                                </td><td>2/61</td><td>391/9447</td><td>0.7252166</td><td>0.8303681</td><td>0.8303681</td><td>1350/6868      </td><td>2</td></tr>
	<tr><th scope=row>hsa05022</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05022</td><td>Pathways of neurodegeneration - multiple diseases</td><td>2/61</td><td>483/9447</td><td>0.8264184</td><td>0.8448696</td><td>0.8448696</td><td>83544/1350     </td><td>2</td></tr>
	<tr><th scope=row>hsa05012</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05012</td><td>Parkinson disease                                </td><td>1/61</td><td>271/9447</td><td>0.8315646</td><td>0.8448696</td><td>0.8448696</td><td>1350           </td><td>1</td></tr>
	<tr><th scope=row>hsa05020</th><td>Human Diseases</td><td>Neurodegenerative disease</td><td>hsa05020</td><td>Prion disease                                    </td><td>1/61</td><td>278/9447</td><td>0.8392505</td><td>0.8459112</td><td>0.8459112</td><td>1350           </td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">entrez_ids</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">ENTREZID</span><span class="o">%in%</span><span class="w"> </span><span class="p">(</span><span class="n">kegg_enrich</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;Alzheimer&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">geneID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">str_split</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;/&#39;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">unlist</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">merge</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="n">gene_ref</span><span class="p">,</span><span class="w"> </span><span class="n">by.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;ENSEMBL&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">by.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;gene_id&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 2 × 6</caption>
<thead>
	<tr><th scope=col>ENSEMBL</th><th scope=col>ENTREZID</th><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>ENSG00000127184</td><td>1350</td><td>chr5</td><td>86617927</td><td>86617928</td><td>COX7C </td></tr>
	<tr><td>ENSG00000151694</td><td>6868</td><td>chr2</td><td> 9556731</td><td> 9556732</td><td>ADAM17</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p8</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">kegg_enrich_only</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;KEGG Pathway Enrichment&quot;</span><span class="p">)</span>
<span class="n">p8</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/1b4c7f00987639b31b2435054f3183f0123030ee5d2468942f27de84e3cd086d.png"><img alt="../../_images/1b4c7f00987639b31b2435054f3183f0123030ee5d2468942f27de84e3cd086d.png" src="../../_images/1b4c7f00987639b31b2435054f3183f0123030ee5d2468942f27de84e3cd086d.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># xQTL list</span>
<span class="n">gene.data</span><span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.dl-inline {width: auto; margin:0; padding: 0}
.dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}
.dl-inline>dt::after {content: ":\0020"; padding-right: .5ex}
.dl-inline>dt:not(:first-of-type) {padding-left: .5ex}
</style><dl class=dl-inline><dt>5595</dt><dd>1</dd><dt>1350</dt><dd>1</dd><dt>83464</dt><dd>1</dd><dt>351</dt><dd>1</dd><dt>6868</dt><dd>1</dd><dt>91252</dt><dd>1</dd><dt>4722</dt><dd>1</dd></dl>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">gene.data</span><span class="p">[</span><span class="nf">names</span><span class="p">(</span><span class="n">gene.data</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;1350&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;6868&#39;</span><span class="p">)]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">2</span>

<span class="c1"># xQTL list highlight xQTL only</span>
<span class="n">gene.data_xqtl_only</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene.data</span><span class="w"> </span>
<span class="n">gene.data_xqtl_only</span><span class="w"> </span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><style>
.dl-inline {width: auto; margin:0; padding: 0}
.dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}
.dl-inline>dt::after {content: ":\0020"; padding-right: .5ex}
.dl-inline>dt:not(:first-of-type) {padding-left: .5ex}
</style><dl class=dl-inline><dt>5595</dt><dd>1</dd><dt>1350</dt><dd>2</dd><dt>83464</dt><dd>1</dd><dt>351</dt><dd>1</dd><dt>6868</dt><dd>2</dd><dt>91252</dt><dd>1</dd><dt>4722</dt><dd>1</dd></dl>
</div></div>
</div>
</section>
<section id="plot-ad-functional-pathway-xqtl-only-list">
<h3>Plot AD functional pathway - xQTL only list<a class="headerlink" href="#plot-ad-functional-pathway-xqtl-only-list" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Make sure gene.data is numeric named vector with correct format and proper range for color mapping</span>
<span class="nf">pathview</span><span class="p">(</span>
<span class="w">  </span><span class="n">gene.data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gene.data_xqtl_only</span><span class="p">,</span>
<span class="w">  </span><span class="n">pathway.id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;hsa05010&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;hsa&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">out.suffix</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;xQTL_only&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">kegg.native</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<span class="w">  </span><span class="n">limit</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">gene.data</span><span class="p">))),</span><span class="w">  </span><span class="c1"># set gene color scale limit properly</span>
<span class="w">  </span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;green&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">mid</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;yellow&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">na.col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="w">   </span><span class="c1"># color for genes without data</span>
<span class="p">)</span>

<span class="n">p9</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readPNG</span><span class="p">(</span><span class="s">&quot;hsa05010.xQTL_only.png&quot;</span><span class="p">)</span>
<span class="nf">grid.raster</span><span class="p">(</span><span class="n">p9</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:1 mapping between keys and columns

Info: Working in directory /restricted/projectnb/xqtl/jaempawi/xqtl

Info: Writing image file hsa05010.xQTL_only.png
</pre></div>
</div>
<a class="reference internal image-reference" href="../../_images/e38453bf9e65f8374d1c69bb39e431e2bd47384453373bc8e15c224ae792cac6.png"><img alt="../../_images/e38453bf9e65f8374d1c69bb39e431e2bd47384453373bc8e15c224ae792cac6.png" src="../../_images/e38453bf9e65f8374d1c69bb39e431e2bd47384453373bc8e15c224ae792cac6.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="twas-only-list">
<h3>TWAS only list<a class="headerlink" href="#twas-only-list" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">twas_diff_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">Reduce</span><span class="p">(</span><span class="n">setdiff</span><span class="p">,</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">twas_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">gene_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="w"> </span><span class="n">gvc_list</span><span class="o">$</span><span class="n">gene_id</span><span class="p">))</span><span class="w"> </span><span class="c1"># the genes unqiue from twas, NOT from gvc and xQTL</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">twas_diff_ego</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span>
<span class="w">  </span><span class="n">gene</span><span class="w">          </span><span class="o">=</span><span class="w"> </span><span class="n">twas_diff_list</span><span class="p">,</span>
<span class="w">  </span><span class="n">OrgDb</span><span class="w">         </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">  </span><span class="n">keyType</span><span class="w">       </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">ont</span><span class="w">           </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<span class="w">  </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">1</span>
<span class="p">)</span>
<span class="n">twas_diff_ego</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0030198</th><td>GO:0030198</td><td>extracellular matrix organization            </td><td>14/242</td><td>387/21288</td><td>0.0001491059</td><td>0.1338011</td><td>0.1271048</td><td>ENSG00000060718/ENSG00000069399/ENSG00000082293/ENSG00000085998/ENSG00000104884/ENSG00000111799/ENSG00000120708/ENSG00000133392/ENSG00000135862/ENSG00000154175/ENSG00000164692/ENSG00000178031/ENSG00000187498/ENSG00000196569</td><td>14</td></tr>
	<tr><th scope=row>GO:0043062</th><td>GO:0043062</td><td>extracellular structure organization         </td><td>14/242</td><td>388/21288</td><td>0.0001531484</td><td>0.1338011</td><td>0.1271048</td><td>ENSG00000060718/ENSG00000069399/ENSG00000082293/ENSG00000085998/ENSG00000104884/ENSG00000111799/ENSG00000120708/ENSG00000133392/ENSG00000135862/ENSG00000154175/ENSG00000164692/ENSG00000178031/ENSG00000187498/ENSG00000196569</td><td>14</td></tr>
	<tr><th scope=row>GO:0045229</th><td>GO:0045229</td><td>external encapsulating structure organization</td><td>14/242</td><td>390/21288</td><td>0.0001615187</td><td>0.1338011</td><td>0.1271048</td><td>ENSG00000060718/ENSG00000069399/ENSG00000082293/ENSG00000085998/ENSG00000104884/ENSG00000111799/ENSG00000120708/ENSG00000133392/ENSG00000135862/ENSG00000154175/ENSG00000164692/ENSG00000178031/ENSG00000187498/ENSG00000196569</td><td>14</td></tr>
	<tr><th scope=row>GO:0051458</th><td>GO:0051458</td><td>corticotropin secretion                      </td><td>3/242 </td><td>11/21288 </td><td>0.0002237980</td><td>0.1338011</td><td>0.1271048</td><td>ENSG00000120088/ENSG00000166128/ENSG00000263715                                                                                                                                                                                </td><td> 3</td></tr>
	<tr><th scope=row>GO:0009743</th><td>GO:0009743</td><td>response to carbohydrate                     </td><td>11/242</td><td>264/21288</td><td>0.0002314097</td><td>0.1338011</td><td>0.1271048</td><td>ENSG00000010310/ENSG00000012061/ENSG00000012232/ENSG00000075391/ENSG00000121691/ENSG00000127616/ENSG00000136504/ENSG00000145242/ENSG00000156711/ENSG00000169375/ENSG00000225697                                                </td><td>11</td></tr>
	<tr><th scope=row>GO:0009650</th><td>GO:0009650</td><td>UV protection                                </td><td>3/242 </td><td>12/21288 </td><td>0.0002958951</td><td>0.1425721</td><td>0.1354369</td><td>ENSG00000012061/ENSG00000104884/ENSG00000121691                                                                                                                                                                                </td><td> 3</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p10</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">twas_diff_ego</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GO Enrichment - BP&quot;</span><span class="p">)</span>
<span class="n">p10</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/47b7ea75cc19b1a1ce5219052c3183115aa4d86bd1a716559d9ef779edec25d0.png"><img alt="../../_images/47b7ea75cc19b1a1ce5219052c3183115aa4d86bd1a716559d9ef779edec25d0.png" src="../../_images/47b7ea75cc19b1a1ce5219052c3183115aa4d86bd1a716559d9ef779edec25d0.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># GO terms related to amyloid</span>
<span class="n">twas_diff_ego</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;amyloid&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 10 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0050435</th><td>GO:0050435</td><td>amyloid-beta metabolic process                                    </td><td>2/242</td><td>74/21288 </td><td>0.2056536</td><td>0.6088354</td><td>0.5783654</td><td>ENSG00000119912/ENSG00000125744</td><td>2</td></tr>
	<tr><th scope=row>GO:1902430</th><td>GO:1902430</td><td>negative regulation of amyloid-beta formation                     </td><td>1/242</td><td>31/21288 </td><td>0.2985986</td><td>0.6275205</td><td>0.5961154</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:1902992</th><td>GO:1902992</td><td>negative regulation of amyloid precursor protein catabolic process</td><td>1/242</td><td>34/21288 </td><td>0.3222833</td><td>0.6456832</td><td>0.6133691</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:0097242</th><td>GO:0097242</td><td>amyloid-beta clearance                                            </td><td>1/242</td><td>46/21288 </td><td>0.4093186</td><td>0.6719534</td><td>0.6383245</td><td>ENSG00000119912                </td><td>1</td></tr>
	<tr><th scope=row>GO:1902003</th><td>GO:1902003</td><td>regulation of amyloid-beta formation                              </td><td>1/242</td><td>57/21288 </td><td>0.4792814</td><td>0.6924550</td><td>0.6578001</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:1990000</th><td>GO:1990000</td><td>amyloid fibril formation                                          </td><td>1/242</td><td>61/21288 </td><td>0.5026217</td><td>0.7019706</td><td>0.6668395</td><td>ENSG00000186868                </td><td>1</td></tr>
	<tr><th scope=row>GO:0034205</th><td>GO:0034205</td><td>amyloid-beta formation                                            </td><td>1/242</td><td>66/21288 </td><td>0.5303371</td><td>0.7122686</td><td>0.6766221</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:1902991</th><td>GO:1902991</td><td>regulation of amyloid precursor protein catabolic process         </td><td>1/242</td><td>70/21288 </td><td>0.5513976</td><td>0.7232716</td><td>0.6870744</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:0042987</th><td>GO:0042987</td><td>amyloid precursor protein catabolic process                       </td><td>1/242</td><td>85/21288 </td><td>0.6223320</td><td>0.7584999</td><td>0.7205397</td><td>ENSG00000125744                </td><td>1</td></tr>
	<tr><th scope=row>GO:0042982</th><td>GO:0042982</td><td>amyloid precursor protein metabolic process                       </td><td>1/242</td><td>114/21288</td><td>0.7293301</td><td>0.8094024</td><td>0.7688947</td><td>ENSG00000125744                </td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
<section id="multi-gene-case-ypel3">
<h3>multi gene case - YPEL3<a class="headerlink" href="#multi-gene-case-ypel3" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># multi_gene_list &lt;- c(&#39;YPEL3&#39;, &#39;INO80E&#39;, &#39;DOC2A&#39;, &#39;MAPK3&#39;, &#39;CORO1A&#39;, &#39;BCKDK&#39;, &#39;TLCD3B&#39;, &#39;ZNF267&#39;)</span>
<span class="n">multi_gene_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;ENSG00000090238&#39;</span><span class="p">,</span><span class="s">&#39;ENSG00000169592&#39;</span><span class="p">,</span><span class="s">&#39;ENSG00000149927&#39;</span><span class="p">,</span><span class="s">&#39;ENSG00000102879&#39;</span><span class="p">,</span><span class="s">&#39;ENSG00000149930&#39;</span><span class="p">,</span><span class="s">&#39;ENSG00000149932&#39;</span><span class="p">,</span><span class="s">&#39;ENSG00000102882&#39;</span><span class="p">,</span><span class="s">&#39;ENSG00000103507&#39;</span><span class="p">,</span><span class="s">&#39;ENSG00000149926&#39;</span><span class="p">,</span><span class="s">&#39;ENSG00000185947&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;ENSG00000285043&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;ENSG00000052344&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">gene_ref</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">multi_gene_list</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.table: 12 × 5</caption>
<thead>
	<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_id</th><th scope=col>gene_name</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>
</thead>
<tbody>
	<tr><td>chr16</td><td>29940884</td><td>29940885</td><td>ENSG00000149932</td><td>TMEM219   </td></tr>
	<tr><td>chr16</td><td>29973867</td><td>29973868</td><td>ENSG00000149930</td><td>TAOK2     </td></tr>
	<tr><td>chr16</td><td>29995714</td><td>29995715</td><td>ENSG00000169592</td><td>INO80E    </td></tr>
	<tr><td>chr16</td><td>30023269</td><td>30023270</td><td>ENSG00000149927</td><td>DOC2A     </td></tr>
	<tr><td>chr16</td><td>30052977</td><td>30052978</td><td>ENSG00000149926</td><td>TLCD3B    </td></tr>
	<tr><td>chr16</td><td>30053089</td><td>30053090</td><td>ENSG00000285043</td><td>AC093512.2</td></tr>
	<tr><td>chr16</td><td>30096914</td><td>30096915</td><td>ENSG00000090238</td><td>YPEL3     </td></tr>
	<tr><td>chr16</td><td>30123505</td><td>30123506</td><td>ENSG00000102882</td><td>MAPK3     </td></tr>
	<tr><td>chr16</td><td>30182826</td><td>30182827</td><td>ENSG00000102879</td><td>CORO1A    </td></tr>
	<tr><td>chr16</td><td>31106106</td><td>31106107</td><td>ENSG00000103507</td><td>BCKDK     </td></tr>
	<tr><td>chr16</td><td>31135726</td><td>31135727</td><td>ENSG00000052344</td><td>PRSS8     </td></tr>
	<tr><td>chr16</td><td>31873806</td><td>31873807</td><td>ENSG00000185947</td><td>ZNF267    </td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multigene_ego</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span>
<span class="w">  </span><span class="n">gene</span><span class="w">          </span><span class="o">=</span><span class="w"> </span><span class="n">multi_gene_list</span><span class="p">,</span>
<span class="w">  </span><span class="n">OrgDb</span><span class="w">         </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">  </span><span class="n">keyType</span><span class="w">       </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">ont</span><span class="w">           </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<span class="w">  </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">1</span>
<span class="p">)</span>
<span class="n">multigene_ego</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0031338</th><td>GO:0031338</td><td>regulation of vesicle fusion                                   </td><td>2/11</td><td>25/21288</td><td>7.235168e-05</td><td>0.02229795</td><td>0.01339537</td><td>ENSG00000149927/ENSG00000102879</td><td>2</td></tr>
	<tr><th scope=row>GO:0032872</th><td>GO:0032872</td><td>regulation of stress-activated MAPK cascade                    </td><td>2/11</td><td>35/21288</td><td>1.430935e-04</td><td>0.02229795</td><td>0.01339537</td><td>ENSG00000149930/ENSG00000102882</td><td>2</td></tr>
	<tr><th scope=row>GO:0070302</th><td>GO:0070302</td><td>regulation of stress-activated protein kinase signaling cascade</td><td>2/11</td><td>38/21288</td><td>1.689239e-04</td><td>0.02229795</td><td>0.01339537</td><td>ENSG00000149930/ENSG00000102882</td><td>2</td></tr>
	<tr><th scope=row>GO:0051403</th><td>GO:0051403</td><td>stress-activated MAPK cascade                                  </td><td>2/11</td><td>50/21288</td><td>2.933612e-04</td><td>0.02611401</td><td>0.01568784</td><td>ENSG00000149930/ENSG00000102882</td><td>2</td></tr>
	<tr><th scope=row>GO:0098927</th><td>GO:0098927</td><td>vesicle-mediated transport between endosomal compartments      </td><td>2/11</td><td>53/21288</td><td>3.297224e-04</td><td>0.02611401</td><td>0.01568784</td><td>ENSG00000102879/ENSG00000102882</td><td>2</td></tr>
	<tr><th scope=row>GO:0031098</th><td>GO:0031098</td><td>stress-activated protein kinase signaling cascade              </td><td>2/11</td><td>59/21288</td><td>4.087094e-04</td><td>0.02697482</td><td>0.01620497</td><td>ENSG00000149930/ENSG00000102882</td><td>2</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p11</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">multigene_ego</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GO Enrichment - BP&quot;</span><span class="p">)</span>
<span class="n">p11</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/0f380c035b984cb03d61326a45b4d2e1ec34d438c9004fbad9f43f36b20e1ae9.png"><img alt="../../_images/0f380c035b984cb03d61326a45b4d2e1ec34d438c9004fbad9f43f36b20e1ae9.png" src="../../_images/0f380c035b984cb03d61326a45b4d2e1ec34d438c9004fbad9f43f36b20e1ae9.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multigene_ego</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;amyloid&#39;</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 0 × 9</caption>
<thead>
	<tr><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">entrez_ids</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">bitr</span><span class="p">(</span><span class="n">multi_gene_list</span><span class="p">,</span><span class="w"> </span><span class="n">fromType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                   </span><span class="n">toType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENTREZID&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">)</span>

<span class="n">kegg_enrich_YPEL3_only</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichKEGG</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">entrez_ids</span><span class="o">$</span><span class="n">ENTREZID</span><span class="p">,</span>
<span class="w">                          </span><span class="n">organism</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;hsa&#39;</span><span class="p">,</span>
<span class="w">                          </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<span class="w">                          </span><span class="n">qvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
<span class="n">kegg_enrich_YPEL3_only</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:1 mapping between keys and columns
</pre></div>
</div>
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 6 × 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa05152</th><td>Human Diseases                      </td><td>Infectious disease: bacterial  </td><td>hsa05152</td><td>Tuberculosis                             </td><td>2/5</td><td>182/9447</td><td>0.003552830</td><td>0.1148963</td><td>0.08667617</td><td>11151/5595</td><td>2</td></tr>
	<tr><th scope=row>hsa04010</th><td>Environmental Information Processing</td><td>Signal transduction            </td><td>hsa04010</td><td>MAPK signaling pathway                   </td><td>2/5</td><td>300/9447</td><td>0.009432492</td><td>0.1148963</td><td>0.08667617</td><td>9344/5595 </td><td>2</td></tr>
	<tr><th scope=row>hsa05216</th><td>Human Diseases                      </td><td>Cancer: specific types         </td><td>hsa05216</td><td>Thyroid cancer                           </td><td>1/5</td><td>37/9447 </td><td>0.019434222</td><td>0.1148963</td><td>0.08667617</td><td>5595      </td><td>1</td></tr>
	<tr><th scope=row>hsa04960</th><td>Organismal Systems                  </td><td>Excretory system               </td><td>hsa04960</td><td>Aldosterone-regulated sodium reabsorption</td><td>1/5</td><td>38/9447 </td><td>0.019955245</td><td>0.1148963</td><td>0.08667617</td><td>5595      </td><td>1</td></tr>
	<tr><th scope=row>hsa05219</th><td>Human Diseases                      </td><td>Cancer: specific types         </td><td>hsa05219</td><td>Bladder cancer                           </td><td>1/5</td><td>41/9447 </td><td>0.021516986</td><td>0.1148963</td><td>0.08667617</td><td>5595      </td><td>1</td></tr>
	<tr><th scope=row>hsa04930</th><td>Human Diseases                      </td><td>Endocrine and metabolic disease</td><td>hsa04930</td><td>Type II diabetes mellitus                </td><td>1/5</td><td>47/9447 </td><td>0.024634496</td><td>0.1148963</td><td>0.08667617</td><td>5595      </td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p12</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">kegg_enrich_YPEL3_only</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;KEGG Pathway Enrichment&quot;</span><span class="p">)</span>
<span class="n">p12</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/6513a70220365ee40bbac0bc3750483f7b6819cc41cdd075c1084f9970aca37d.png"><img alt="../../_images/6513a70220365ee40bbac0bc3750483f7b6819cc41cdd075c1084f9970aca37d.png" src="../../_images/6513a70220365ee40bbac0bc3750483f7b6819cc41cdd075c1084f9970aca37d.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
</section>
<section id="multi-gene-case-nbeal1">
<h3>multi gene case - NBEAL1<a class="headerlink" href="#multi-gene-case-nbeal1" title="Link to this heading">#</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multi_gene_list2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;WDR81&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;TSC22D4&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;ICA1L&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;FAM117B&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;NBEAL1&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;CARF&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;WDR12&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;DHX33&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;EMG1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;DYSF&quot;</span><span class="p">)</span>
<span class="n">multi_gene_list2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_ref</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gene_name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">multi_gene_list2</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">gene_id</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">multigene_ego2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span>
<span class="w">  </span><span class="n">gene</span><span class="w">          </span><span class="o">=</span><span class="w"> </span><span class="n">multi_gene_list2</span><span class="p">,</span>
<span class="w">  </span><span class="n">OrgDb</span><span class="w">         </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">  </span><span class="n">keyType</span><span class="w">       </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">ont</span><span class="w">           </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span>
<span class="w">  </span><span class="n">pvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<span class="w">  </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">1</span>
<span class="p">)</span>
<span class="n">multigene_ego2</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">as.data.frame</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="n">Count</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 85 × 9</caption>
<thead>
	<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>GO:0006364</th><td>GO:0006364</td><td>rRNA processing                                                                          </td><td>2/9</td><td>249/21288</td><td>0.004646887</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000138442/ENSG00000126749</td><td>2</td></tr>
	<tr><th scope=row>GO:0016072</th><td>GO:0016072</td><td>rRNA metabolic process                                                                   </td><td>2/9</td><td>294/21288</td><td>0.006418484</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000138442/ENSG00000126749</td><td>2</td></tr>
	<tr><th scope=row>GO:0042254</th><td>GO:0042254</td><td>ribosome biogenesis                                                                      </td><td>2/9</td><td>359/21288</td><td>0.009440409</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000138442/ENSG00000126749</td><td>2</td></tr>
	<tr><th scope=row>GO:0034470</th><td>GO:0034470</td><td>ncRNA processing                                                                         </td><td>2/9</td><td>482/21288</td><td>0.016575172</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000138442/ENSG00000126749</td><td>2</td></tr>
	<tr><th scope=row>GO:0070475</th><td>GO:0070475</td><td>rRNA base methylation                                                                    </td><td>1/9</td><td>11/21288 </td><td>0.004641777</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000126749                </td><td>1</td></tr>
	<tr><th scope=row>GO:0035973</th><td>GO:0035973</td><td>aggrephagy                                                                               </td><td>1/9</td><td>16/21288 </td><td>0.006745337</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000167716                </td><td>1</td></tr>
	<tr><th scope=row>GO:0000463</th><td>GO:0000463</td><td>maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA) </td><td>1/9</td><td>17/21288 </td><td>0.007165575</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000138442                </td><td>1</td></tr>
	<tr><th scope=row>GO:0042117</th><td>GO:0042117</td><td>monocyte activation                                                                      </td><td>1/9</td><td>18/21288 </td><td>0.007585654</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000135636                </td><td>1</td></tr>
	<tr><th scope=row>GO:0007000</th><td>GO:0007000</td><td>nucleolus organization                                                                   </td><td>1/9</td><td>19/21288 </td><td>0.008005576</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000126749                </td><td>1</td></tr>
	<tr><th scope=row>GO:0050765</th><td>GO:0050765</td><td>negative regulation of phagocytosis                                                      </td><td>1/9</td><td>23/21288 </td><td>0.009683683</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000135636                </td><td>1</td></tr>
	<tr><th scope=row>GO:1900227</th><td>GO:1900227</td><td>positive regulation of NLRP3 inflammasome complex assembly                               </td><td>1/9</td><td>26/21288 </td><td>0.010940606</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0002281</th><td>GO:0002281</td><td>macrophage activation involved in immune response                                        </td><td>1/9</td><td>27/21288 </td><td>0.011359266</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000135636                </td><td>1</td></tr>
	<tr><th scope=row>GO:0141087</th><td>GO:0141087</td><td>positive regulation of inflammasome-mediated signaling pathway                           </td><td>1/9</td><td>27/21288 </td><td>0.011359266</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0000466</th><td>GO:0000466</td><td>maturation of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)</td><td>1/9</td><td>28/21288 </td><td>0.011777767</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000138442                </td><td>1</td></tr>
	<tr><th scope=row>GO:0031167</th><td>GO:0031167</td><td>rRNA methylation                                                                         </td><td>1/9</td><td>29/21288 </td><td>0.012196112</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000126749                </td><td>1</td></tr>
	<tr><th scope=row>GO:0000470</th><td>GO:0000470</td><td>maturation of LSU-rRNA                                                                   </td><td>1/9</td><td>30/21288 </td><td>0.012614299</td><td>0.07148102</td><td>0.03983463</td><td>ENSG00000138442                </td><td>1</td></tr>
	<tr><th scope=row>GO:0045943</th><td>GO:0045943</td><td>positive regulation of transcription by RNA polymerase I                                 </td><td>1/9</td><td>36/21288 </td><td>0.015120117</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0000154</th><td>GO:0000154</td><td>rRNA modification                                                                        </td><td>1/9</td><td>39/21288 </td><td>0.016370905</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000126749                </td><td>1</td></tr>
	<tr><th scope=row>GO:0000460</th><td>GO:0000460</td><td>maturation of 5.8S rRNA                                                                  </td><td>1/9</td><td>39/21288 </td><td>0.016370905</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000138442                </td><td>1</td></tr>
	<tr><th scope=row>GO:0045022</th><td>GO:0045022</td><td>early endosome to late endosome transport                                                </td><td>1/9</td><td>47/21288 </td><td>0.019699439</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000167716                </td><td>1</td></tr>
	<tr><th scope=row>GO:1900225</th><td>GO:1900225</td><td>regulation of NLRP3 inflammasome complex assembly                                        </td><td>1/9</td><td>48/21288 </td><td>0.020114801</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0006356</th><td>GO:0006356</td><td>regulation of transcription by RNA polymerase I                                          </td><td>1/9</td><td>49/21288 </td><td>0.020530007</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0044546</th><td>GO:0044546</td><td>NLRP3 inflammasome complex assembly                                                      </td><td>1/9</td><td>51/21288 </td><td>0.021359949</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0141085</th><td>GO:0141085</td><td>regulation of inflammasome-mediated signaling pathway                                    </td><td>1/9</td><td>51/21288 </td><td>0.021359949</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0098927</th><td>GO:0098927</td><td>vesicle-mediated transport between endosomal compartments                                </td><td>1/9</td><td>53/21288 </td><td>0.022189266</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000167716                </td><td>1</td></tr>
	<tr><th scope=row>GO:0140632</th><td>GO:0140632</td><td>canonical inflammasome complex assembly                                                  </td><td>1/9</td><td>53/21288 </td><td>0.022189266</td><td>0.07254183</td><td>0.04042579</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0141084</th><td>GO:0141084</td><td>inflammasome-mediated signaling pathway                                                  </td><td>1/9</td><td>56/21288 </td><td>0.023432070</td><td>0.07376763</td><td>0.04110889</td><td>ENSG00000005100                </td><td>1</td></tr>
	<tr><th scope=row>GO:0070050</th><td>GO:0070050</td><td>neuron cellular homeostasis                                                              </td><td>1/9</td><td>72/21288 </td><td>0.030036675</td><td>0.09118276</td><td>0.05081392</td><td>ENSG00000166925                </td><td>1</td></tr>
	<tr><th scope=row>GO:0042273</th><td>GO:0042273</td><td>ribosomal large subunit biogenesis                                                       </td><td>1/9</td><td>78/21288 </td><td>0.032503147</td><td>0.09159128</td><td>0.05104158</td><td>ENSG00000138442                </td><td>1</td></tr>
	<tr><th scope=row>GO:0046928</th><td>GO:0046928</td><td>regulation of neurotransmitter secretion                                                 </td><td>1/9</td><td>81/21288 </td><td>0.033734291</td><td>0.09159128</td><td>0.05104158</td><td>ENSG00000135636                </td><td>1</td></tr>
	<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>
	<tr><th scope=row>GO:0007219</th><td>GO:0007219</td><td>Notch signaling pathway                                         </td><td>1/9</td><td>200/21288</td><td>0.08146055</td><td>0.1236455</td><td>0.06890460</td><td>ENSG00000138442</td><td>1</td></tr>
	<tr><th scope=row>GO:0071248</th><td>GO:0071248</td><td>cellular response to metal ion                                  </td><td>1/9</td><td>213/21288</td><td>0.08654518</td><td>0.1275717</td><td>0.07109261</td><td>ENSG00000138380</td><td>1</td></tr>
	<tr><th scope=row>GO:0002753</th><td>GO:0002753</td><td>cytoplasmic pattern recognition receptor signaling pathway      </td><td>1/9</td><td>217/21288</td><td>0.08810464</td><td>0.1275717</td><td>0.07109261</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0006836</th><td>GO:0006836</td><td>neurotransmitter transport                                      </td><td>1/9</td><td>222/21288</td><td>0.09005064</td><td>0.1275717</td><td>0.07109261</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0062207</th><td>GO:0062207</td><td>regulation of pattern recognition receptor signaling pathway    </td><td>1/9</td><td>222/21288</td><td>0.09005064</td><td>0.1275717</td><td>0.07109261</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0031334</th><td>GO:0031334</td><td>positive regulation of protein-containing complex assembly      </td><td>1/9</td><td>227/21288</td><td>0.09199295</td><td>0.1281869</td><td>0.07143542</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0071241</th><td>GO:0071241</td><td>cellular response to inorganic substance                        </td><td>1/9</td><td>244/21288</td><td>0.09856926</td><td>0.1351353</td><td>0.07530758</td><td>ENSG00000138380</td><td>1</td></tr>
	<tr><th scope=row>GO:0050821</th><td>GO:0050821</td><td>protein stabilization                                           </td><td>1/9</td><td>254/21288</td><td>0.10241786</td><td>0.1378248</td><td>0.07680639</td><td>ENSG00000167716</td><td>1</td></tr>
	<tr><th scope=row>GO:0048588</th><td>GO:0048588</td><td>developmental cell growth                                       </td><td>1/9</td><td>259/21288</td><td>0.10433668</td><td>0.1378248</td><td>0.07680639</td><td>ENSG00000166925</td><td>1</td></tr>
	<tr><th scope=row>GO:0060560</th><td>GO:0060560</td><td>developmental growth involved in morphogenesis                  </td><td>1/9</td><td>263/21288</td><td>0.10586911</td><td>0.1378248</td><td>0.07680639</td><td>ENSG00000166925</td><td>1</td></tr>
	<tr><th scope=row>GO:0006909</th><td>GO:0006909</td><td>phagocytosis                                                    </td><td>1/9</td><td>266/21288</td><td>0.10701691</td><td>0.1378248</td><td>0.07680639</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0032259</th><td>GO:0032259</td><td>methylation                                                     </td><td>1/9</td><td>279/21288</td><td>0.11197556</td><td>0.1396609</td><td>0.07782960</td><td>ENSG00000126749</td><td>1</td></tr>
	<tr><th scope=row>GO:0042593</th><td>GO:0042593</td><td>glucose homeostasis                                             </td><td>1/9</td><td>286/21288</td><td>0.11463545</td><td>0.1396609</td><td>0.07782960</td><td>ENSG00000166925</td><td>1</td></tr>
	<tr><th scope=row>GO:0051091</th><td>GO:0051091</td><td>positive regulation of DNA-binding transcription factor activity</td><td>1/9</td><td>286/21288</td><td>0.11463545</td><td>0.1396609</td><td>0.07782960</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0033500</th><td>GO:0033500</td><td>carbohydrate homeostasis                                        </td><td>1/9</td><td>287/21288</td><td>0.11501485</td><td>0.1396609</td><td>0.07782960</td><td>ENSG00000166925</td><td>1</td></tr>
	<tr><th scope=row>GO:0002274</th><td>GO:0002274</td><td>myeloid leukocyte activation                                    </td><td>1/9</td><td>292/21288</td><td>0.11690972</td><td>0.1399623</td><td>0.07799759</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0016197</th><td>GO:0016197</td><td>endosomal transport                                             </td><td>1/9</td><td>318/21288</td><td>0.12670499</td><td>0.1495823</td><td>0.08335854</td><td>ENSG00000167716</td><td>1</td></tr>
	<tr><th scope=row>GO:0002221</th><td>GO:0002221</td><td>pattern recognition receptor signaling pathway                  </td><td>1/9</td><td>328/21288</td><td>0.13044661</td><td>0.1515519</td><td>0.08445614</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0030100</th><td>GO:0030100</td><td>regulation of endocytosis                                       </td><td>1/9</td><td>332/21288</td><td>0.13193926</td><td>0.1515519</td><td>0.08445614</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0002758</th><td>GO:0002758</td><td>innate immune response-activating signaling pathway             </td><td>1/9</td><td>357/21288</td><td>0.14121685</td><td>0.1583322</td><td>0.08823464</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0002366</th><td>GO:0002366</td><td>leukocyte activation involved in immune response                </td><td>1/9</td><td>359/21288</td><td>0.14195523</td><td>0.1583322</td><td>0.08823464</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0002263</th><td>GO:0002263</td><td>cell activation involved in immune response                     </td><td>1/9</td><td>363/21288</td><td>0.14343031</td><td>0.1583322</td><td>0.08823464</td><td>ENSG00000135636</td><td>1</td></tr>
	<tr><th scope=row>GO:0031647</th><td>GO:0031647</td><td>regulation of protein stability                                 </td><td>1/9</td><td>379/21288</td><td>0.14930811</td><td>0.1594178</td><td>0.08883965</td><td>ENSG00000167716</td><td>1</td></tr>
	<tr><th scope=row>GO:0002218</th><td>GO:0002218</td><td>activation of innate immune response                            </td><td>1/9</td><td>381/21288</td><td>0.15004031</td><td>0.1594178</td><td>0.08883965</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0016236</th><td>GO:0016236</td><td>macroautophagy                                                  </td><td>1/9</td><td>381/21288</td><td>0.15004031</td><td>0.1594178</td><td>0.08883965</td><td>ENSG00000167716</td><td>1</td></tr>
	<tr><th scope=row>GO:0010038</th><td>GO:0010038</td><td>response to metal ion                                           </td><td>1/9</td><td>387/21288</td><td>0.15223354</td><td>0.1597512</td><td>0.08902546</td><td>ENSG00000138380</td><td>1</td></tr>
	<tr><th scope=row>GO:0001701</th><td>GO:0001701</td><td>in utero embryonic development                                  </td><td>1/9</td><td>441/21288</td><td>0.17174741</td><td>0.1769840</td><td>0.09862888</td><td>ENSG00000126749</td><td>1</td></tr>
	<tr><th scope=row>GO:0030522</th><td>GO:0030522</td><td>intracellular receptor signaling pathway                        </td><td>1/9</td><td>444/21288</td><td>0.17281971</td><td>0.1769840</td><td>0.09862888</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0051090</th><td>GO:0051090</td><td>regulation of DNA-binding transcription factor activity         </td><td>1/9</td><td>455/21288</td><td>0.17674093</td><td>0.1788450</td><td>0.09966594</td><td>ENSG00000005100</td><td>1</td></tr>
	<tr><th scope=row>GO:0043254</th><td>GO:0043254</td><td>regulation of protein-containing complex assembly               </td><td>1/9</td><td>467/21288</td><td>0.18099977</td><td>0.1809998</td><td>0.10086674</td><td>ENSG00000005100</td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">p13</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">barplot</span><span class="p">(</span><span class="n">multigene_ego2</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;GO Enrichment - BP&quot;</span><span class="p">)</span>
<span class="n">p13</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<a class="reference internal image-reference" href="../../_images/446d75352f8005655bf32702600451751447f6598de52de0226cf485b0e17a9c.png"><img alt="../../_images/446d75352f8005655bf32702600451751447f6598de52de0226cf485b0e17a9c.png" src="../../_images/446d75352f8005655bf32702600451751447f6598de52de0226cf485b0e17a9c.png" style="width: 420px; height: 420px;" />
</a>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">entrez_ids</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">bitr</span><span class="p">(</span><span class="n">multi_gene_list2</span><span class="p">,</span><span class="w"> </span><span class="n">fromType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                   </span><span class="n">toType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENTREZID&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">)</span>

<span class="n">kegg_enrich_NBEAL1_only</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichKEGG</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">entrez_ids</span><span class="o">$</span><span class="n">ENTREZID</span><span class="p">,</span>
<span class="w">                          </span><span class="n">organism</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;hsa&#39;</span><span class="p">,</span>
<span class="w">                          </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span>
<span class="w">                         </span><span class="n">qvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
<span class="n">kegg_enrich_NBEAL1_only</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">head</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&#39;select()&#39; returned 1:1 mapping between keys and columns
</pre></div>
</div>
<div class="output text_html"><table class="dataframe">
<caption>A data.frame: 1 × 11</caption>
<thead>
	<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>
	<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>
</thead>
<tbody>
	<tr><th scope=row>hsa04621</th><td>Organismal Systems</td><td>Immune system</td><td>hsa04621</td><td>NOD-like receptor signaling pathway</td><td>1/2</td><td>189/9447</td><td>0.03961452</td><td>0.03961452</td><td>NA</td><td>56919</td><td>1</td></tr>
</tbody>
</table>
</div></div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "r"
        },
        kernelOptions: {
            name: "ir",
            path: "./main_text/5_AD_xQTL_genes_cis_trans"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'ir'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#motivation">Motivation</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#implementation">Implementation</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-libraries">Load libraries</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#load-gene-and-twas-list">Load gene and TWAS list</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#real-gvc-list-20-genes-from-gvc-table-2">Real GVC list - 20 genes from GVC table 2</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#gvc-list-1-76-genes">GVC list 1 - 76 genes</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#strong-list-overlapped-between-xqtl-gvc-and-twas">Strong list - overlapped between xQTL, GVC and TWAS</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-list-go-enrichment">xQTL list - GO enrichment</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-list-kegg-enrichment">xQTL list - KEGG enrichment</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plot-ad-functional-pathway-xqtl-list">Plot AD functional pathway - xQTL list</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-list-go-enrichment">xQTL only list - GO enrichment</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#xqtl-only-list-kegg-enrichment">xQTL only list - KEGG enrichment</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#plot-ad-functional-pathway-xqtl-only-list">Plot AD functional pathway - xQTL only list</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#twas-only-list">TWAS only list</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-case-ypel3">multi gene case - YPEL3</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-gene-case-nbeal1">multi gene case - NBEAL1</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By FunGen-AD Consortium
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>